diff --git a/apps/BUILD.gn b/apps/BUILD.gn
index f93f4fbcc1ff16dc6b2831a9e97004ebeb32d821..484ddccef1ed1206eface84850461a7df86212f7 100644
--- a/apps/BUILD.gn
+++ b/apps/BUILD.gn
@@ -7,7 +7,7 @@ import("//build/config/features.gni")
 import("//build/config/ui.gni")
 import("//extensions/buildflags/buildflags.gni")
 
-assert(!is_android && !is_ios)
+#assert(!is_android && !is_ios)
 assert(enable_extensions,
        "Cannot depend on extensions because enable_extensions=false.")
 
