diff --git a/apps/BUILD.gn b/apps/BUILD.gn
index d14c89436b97b6e4eddce91a9d257496f3fe5a59..1ae93bc3998f1f7c2bd99167d8744be0b43cbd13 100644
--- a/apps/BUILD.gn
+++ b/apps/BUILD.gn
@@ -7,7 +7,7 @@ import("//build/config/features.gni")
 import("//build/config/ui.gni")
 import("//extensions/buildflags/buildflags.gni")
 
-assert(!is_android && !is_ios)
+#assert(!is_android && !is_ios)
 assert(enable_extensions,
        "Cannot depend on extensions because enable_extensions=false.")
 
