diff --git a/base/android/java/src/org/chromium/base/shared_preferences/SharedPreferencesManager.java b/base/android/java/src/org/chromium/base/shared_preferences/SharedPreferencesManager.java
index ea4756405e16064a8150039dbe4ca47e83f91466..dc674f4cac7ebffc37364ba5a2aba90f9fb60128 100644
--- a/base/android/java/src/org/chromium/base/shared_preferences/SharedPreferencesManager.java
+++ b/base/android/java/src/org/chromium/base/shared_preferences/SharedPreferencesManager.java
@@ -258,7 +258,7 @@ public class SharedPreferencesManager {
         writeIntUnchecked(key, value);
     }
 
-    private void writeIntUnchecked(String key, int value) {
+    public void writeIntUnchecked(String key, int value) {
         SharedPreferences.Editor ed = ContextUtils.getAppSharedPreferences().edit();
         ed.putInt(key, value);
         ed.apply();
@@ -632,4 +632,33 @@ public class SharedPreferencesManager {
         }
         return allPrefsWithPrefix;
     }
+
+    public static final String MISES_USER_INFO = "mises_user_info";
+    public static final String MISES_FIRST_RUN = "mises_first_run";
+    public String getMisesUserInfo() {
+        // Convention to keep all the key in preference lower case.
+        return readString(MISES_USER_INFO, "");
+    }
+
+    public void setMisesUserInfo(String userinfo) {
+        writeString(MISES_USER_INFO, userinfo);
+    }
+
+    public boolean isMisesFirstRun() {
+        return readBoolean(MISES_FIRST_RUN, true);
+    }
+
+    public void setMisesFirstRun(boolean isFirst) {
+        writeBoolean(MISES_FIRST_RUN, isFirst);
+    }
+    public void writeStringUnchecked(String key, String value) {
+        SharedPreferences.Editor ed = ContextUtils.getAppSharedPreferences().edit();
+        ed.putString(key, value);
+        ed.apply();
+    }
+    public void writeBooleanUnchecked(String key, boolean value) {
+        SharedPreferences.Editor ed = ContextUtils.getAppSharedPreferences().edit();
+        ed.putBoolean(key, value);
+        ed.apply();
+    }
 }
