diff --git a/base/third_party/double_conversion/double-conversion/string-to-double.h b/base/third_party/double_conversion/double-conversion/string-to-double.h
index 770ccbc3067548199978e88f5d5a748973fc54fd..dae5c55b9410ac554517edc83e3694b03810a4ee 100644
--- a/base/third_party/double_conversion/double-conversion/string-to-double.h
+++ b/base/third_party/double_conversion/double-conversion/string-to-double.h
@@ -230,6 +230,12 @@ class StringToDoubleConverter {
                       bool read_as_double,
                       int* processed_characters_count) const;
 
+  double StringToIeee(
+    const char* buffer,
+    int length,
+    int* processed_characters_count,
+    bool read_as_double)const ;
+
   DOUBLE_CONVERSION_DISALLOW_IMPLICIT_CONSTRUCTORS(StringToDoubleConverter);
 };
 
