diff --git a/base/third_party/double_conversion/double-conversion/string-to-double.h b/base/third_party/double_conversion/double-conversion/string-to-double.h
index 770ccbc3067548199978e88f5d5a748973fc54fd..21a4d6eb99a5c4dce00549943de43583f3a556c1 100644
--- a/base/third_party/double_conversion/double-conversion/string-to-double.h
+++ b/base/third_party/double_conversion/double-conversion/string-to-double.h
@@ -215,6 +215,12 @@ class StringToDoubleConverter {
   T StringTo(const uc16* buffer,
              int length,
              int* processed_characters_count) const;
+   
+  double StringToIeee(
+    const char* buffer,
+    int length,
+    int* processed_characters_count,
+    bool read_as_double)const ;
 
  private:
   const int flags_;
