diff --git a/build/config/compiler/BUILD.gn b/build/config/compiler/BUILD.gn
index ef3285ae74e1bf95389e2c30da001b0d6ef7a8e0..cf5b83fbbae40538b8b7bf1e260850d423714122 100644
--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -769,7 +769,7 @@ config("compiler") {
   # Pass flag to LLD so Android builds can allow debuggerd to properly symbolize
   # stack crashes (http://crbug.com/919499).
   if (use_lld && is_android) {
-    ldflags += [ "-Wl,--no-rosegment" ]
+    ldflags += [ "-Wl,--no-rosegment", "-Wl,--error-limit=0" ]
   }
 
   # TODO(crbug.com/1374347): Cleanup undefined symbol errors caught by
