diff --git a/build/config/compiler/compiler.gni b/build/config/compiler/compiler.gni
index 7bb39b929553b0d93a1146f0f7d86b0d511aecc7..68a261a00a230755af1c400bbf03bc2cbdc61057 100644
--- a/build/config/compiler/compiler.gni
+++ b/build/config/compiler/compiler.gni
@@ -228,7 +228,7 @@ can_unwind_with_cfi_table = is_android && !is_component_build &&
 # Whether or not cfi table should be enabled on arm.
 # TODO(crbug.com/40133751): Replace can_unwind_with_cfi_table with this once
 # sampling profiler is enabled on android.
-enable_arm_cfi_table = is_android && !is_component_build && current_cpu == "arm"
+enable_arm_cfi_table = !is_official_build && is_android && !is_component_build && current_cpu == "arm"
 
 # Use relative paths for debug info. This is important to make the build
 # results independent of the checkout and build directory names, which
