diff --git a/build/android/gyp/proguard.py b/build/android/gyp/proguard.py
index 9ab674c6a83db13d9982bd5f2924d191df3b053e..8de677139f20d19a362a6de6e20c2628a6773fbe 100755
--- a/build/android/gyp/proguard.py
+++ b/build/android/gyp/proguard.py
@@ -275,6 +275,7 @@ def _OptimizeWithR8(options,
         '--pg-map-output',
         tmp_mapping_path,
     ]
+    cmd.remove('--map-id-template'); cmd.remove(f'{options.source_file} ({options.package_name})') # required to omit package check for mapping file
 
     if options.disable_checks:
       cmd += ['--map-diagnostics:CheckDiscardDiagnostic', 'error', 'none']
@@ -592,7 +593,7 @@ def _DoTraceReferencesChecks(options, split_contexts_by_name):
 def main():
   build_utils.InitLogging('PROGUARD_DEBUG')
   options = _ParseOptions()
-
+  options.warnings_as_errors = False
   # ProGuard configs that are derived from flags.
   logging.debug('Preparing configs')
   dynamic_config_data = _CreateDynamicConfig(options)
