diff --git a/build/config/BUILDCONFIG.gn b/build/config/BUILDCONFIG.gn
index c737e53c2ce7237974f6c749eae60ba1de7ec2c1..349364caada7d3f9b32bae5410632bbd8be8c588 100644
--- a/build/config/BUILDCONFIG.gn
+++ b/build/config/BUILDCONFIG.gn
@@ -325,6 +325,7 @@ is_posix = !is_win && !is_fuchsia
 
 # Holds all configs used for running the compiler.
 default_compiler_configs = [
+  "//mises/build:compiler",
   "//build/config:feature_flags",
   "//build/config/compiler:afdo",
   "//build/config/compiler:afdo_optimize_size",
