diff --git a/build/android/gyp/dex.py b/build/android/gyp/dex.py
index ae87291066f3f9782ab087993ab130d70105ffa2..43934a9366e0807e8f3190606da7a1a2984d9451 100755
--- a/build/android/gyp/dex.py
+++ b/build/android/gyp/dex.py
@@ -30,6 +30,7 @@ DEFAULT_IGNORE_WARNINGS = (
     # META-INF/. This means that all rules with a '-upto-' qualifier will be
     # excluded and all rules with a -from- qualifier will be included.
     r'Running R8 version main', )
+DEFAULT_IGNORE_WARNINGS += ( r'Expected stack map table for method with non-linear control flow.', )
 
 INTERFACE_DESUGARING_WARNINGS = (r'default or static interface methods', )
 
