diff --git a/build/util/lastchange.py b/build/util/lastchange.py
index 98a6360b46872dbadb53aa5c832419c6f804fce5..e06ecb7196ed26235cc3226876521a083b286627 100755
--- a/build/util/lastchange.py
+++ b/build/util/lastchange.py
@@ -123,14 +123,14 @@ def FetchGitRevision(directory, commit_filter, start_commit="HEAD"):
   """
   hash_ = ''
 
-  git_args = ['log', '-1', '--format=%H %ct']
-  if commit_filter is not None:
-    git_args.append('--grep=' + commit_filter)
+  git_args = ['log', '-1', '--format=%H,%ct']
+  #if commit_filter is not None:
+  #  git_args.append('--grep=' + commit_filter)
 
   git_args.append(start_commit)
 
   output = _RunGitCommand(directory, git_args)
-  hash_, commit_timestamp = output.split()
+  hash_, commit_timestamp = output.split(',')
   if not hash_:
     return VersionInfo('0', '0', 0)
 
