diff --git a/build/android/gyp/proguard.py b/build/android/gyp/proguard.py
index 5dd433ae0bd92209bd0275e19e710c310e6e8793..2bab2011dbc6036bb67c065e6f01293c6e0e3c02 100755
--- a/build/android/gyp/proguard.py
+++ b/build/android/gyp/proguard.py
@@ -80,6 +80,7 @@ _IGNORE_WARNINGS = (
     # We are following up in b/290389974
     r'AppSearchDocumentClassMap\.class:.*Could not inline ServiceLoader\.load',
 )
+_IGNORE_WARNINGS += ( r'InnerClasses attribute has entries missing a corresponding EnclosingMethod attribute. Such InnerClasses attribute entries are ignored.', r'google_play_services_ads_java', r'google_play_services_ads_lite_java', r'google_play_services_measurement_java', r'Missing class com.android.org.conscrypt.SSLParametersImpl', r'Missing class org.apache.harmony.xnet.provider.jsse.SSLParametersImpl')
 
 _BLOCKLISTED_EXPECTATION_PATHS = [
     # A separate expectation file is created for these files.
@@ -376,6 +377,7 @@ def _OptimizeWithR8(options, config_paths, libraries, dynamic_config_data):
         '--pg-map-output',
         tmp_mapping_path,
     ]
+    cmd.remove('--map-id-template'); cmd.remove(f'{options.source_file} ({options.package_name})') # required to omit package check for mapping file
 
     if options.uses_split:
       cmd += ['--isolated-splits']
@@ -541,6 +543,7 @@ def _CheckForMissingSymbols(options, dex_files, error_title):
         # KotlinDetector: com.google.firebase.platforminfo.KotlinDetector.
         'kotlin.KotlinVersion',
     ]
+    ignored_lines += ['android.adservices.measurement', 'android.adservices.topics', 'com.android.org.conscrypt', 'org.apache.harmony.xnet.provider.jsse']
 
     had_unfiltered_items = '  ' in stderr
     stderr = build_utils.FilterLines(
