diff --git a/build/config/android/config.gni b/build/config/android/config.gni
index 8cf49d09545e00e9378bc1cb1c200da4d1c7c26d..b3004a646311c7c4d50be1157ef5b27a566e8239 100644
--- a/build/config/android/config.gni
+++ b/build/config/android/config.gni
@@ -29,6 +29,7 @@ assert(!(incremental_install && !is_java_debug),
 robolectric_toolchain = "//build/toolchain/android:robolectric_$host_cpu"
 
 if (is_android) {
+  import("//mises/build/android/config.gni")
   import("//build/config/android/channel.gni")
   import("//build/config/clang/clang.gni")
   import("//build/config/dcheck_always_on.gni")
@@ -135,7 +136,7 @@ if (is_android) {
   public_android_sdk_root = "//third_party/android_sdk/public"
   public_android_sdk_build_tools =
       "${public_android_sdk_root}/build-tools/35.0.0"
-  public_android_sdk_version = "34"
+  public_android_sdk_version = "35"
   if (android_sdk_release == "v") {
     default_android_sdk_root = public_android_sdk_root
     default_android_sdk_version = public_android_sdk_version
