diff --git a/build/config/compiler/BUILD.gn b/build/config/compiler/BUILD.gn
index 9afb4425840b92a5c16e5ce5e11086a778cf8451..7a206632a1918a4ad2de9a5a9e0c64fb1326c307 100644
--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -771,7 +771,7 @@ config("compiler") {
   # Pass flag to LLD so Android builds can allow debuggerd to properly symbolize
   # stack crashes (http://crbug.com/919499).
   if (use_lld && is_android) {
-    ldflags += [ "-Wl,--no-rosegment" ]
+    ldflags += [ "-Wl,--no-rosegment", "-Wl,--error-limit=0" ]
   }
 
   # TODO(crbug.com/1374347): Cleanup undefined symbol errors caught by
