diff --git a/build/config/android/rules.gni b/build/config/android/rules.gni
index 9bc8939ee101caa4eb14cf79d142c5936c22cc99..a18e8ba389e38d0d7c8e53c9a90f3e8025778b89 100644
--- a/build/config/android/rules.gni
+++ b/build/config/android/rules.gni
@@ -213,6 +213,7 @@ if (enable_java_templates && is_android) {
   #     ]
   #   }
   template("java_cpp_enum") {
+    foreach(source, invoker.sources) { foreach(filter, mises_java_cpp_enum_filter) { if(source == filter) { invoker.sources -= [filter] }}}
     action_with_pydeps(target_name) {
       forward_variables_from(invoker, TESTONLY_AND_VISIBILITY + [ "sources" ])
 
@@ -388,6 +389,7 @@ if (enable_java_templates && is_android) {
   #     output = "$target_gen_dir/AndroidManifest.xml"
   #   }
   template("jinja_template") {
+    if (defined(invoker.includes)) { invoker.includes += mises_android_manifest_includes } else { invoker.includes = mises_android_manifest_includes }
     action_with_pydeps(target_name) {
       forward_variables_from(invoker, TESTONLY_AND_VISIBILITY + [ "deps" ])
       inputs = [ invoker.input ]
@@ -939,7 +941,7 @@ if (enable_java_templates && is_android) {
       action(target_name) {
         forward_variables_from(invoker, [ "deps" ])
         public_deps = [ ":$_build_config_target_name" ]
-
+        if (defined(invoker.public_deps)) { public_deps += invoker.public_deps }
         script = "//build/android/gyp/validate_inputs.py"
         inputs = _all_sources
         outputs = [ "$target_gen_dir/$target_name.stamp" ]
