diff --git a/build/toolchain/apple/toolchain.gni b/build/toolchain/apple/toolchain.gni
index 93e65289fcc48ec897e887fc7f4d17db2108239f..3559e8f3a773bde2f028c403461b5366353c3787 100644
--- a/build/toolchain/apple/toolchain.gni
+++ b/build/toolchain/apple/toolchain.gni
@@ -655,7 +655,6 @@ template("single_apple_toolchain") {
         if (defined(toolchain_args.xcode_build)) {
           _extra_flags += " -xcode-version ${toolchain_args.xcode_build}"
         }
-
         if (swift_toolchain_path != "") {
           _extra_flags += " -swift-toolchain-path " +
                           rebase_path(swift_toolchain_path, root_build_dir)
@@ -696,7 +695,7 @@ template("single_apple_toolchain") {
             "-depfile {{target_out_dir}}/{{module_name}}.d " +
             "-depfile-filter {{target_gen_dir}}/{{module_name}}.swiftmodule " +
             "-bridge-header {{bridge_header}} $_extra_flags " +
-            "{{swiftflags}} {{include_dirs}} {{module_dirs}} {{inputs}}"
+            "{{swiftflags}} {{include_dirs}} -I../../mises -F../../mises/ios/third_party/mises/Sim/Frameworks {{module_dirs}} {{inputs}}"
       }
     }
 
