diff --git a/build/android/gyp/proguard.py b/build/android/gyp/proguard.py
index f89f7c5fb97bf2bb9e5a141a8d4a88ac1c7014f0..bc4a6930e6589c41ec3d3978371eee15883cd06a 100755
--- a/build/android/gyp/proguard.py
+++ b/build/android/gyp/proguard.py
@@ -331,6 +331,7 @@ def _OptimizeWithR8(options, config_paths, libraries, dynamic_config_data):
         '--pg-map-output',
         tmp_mapping_path,
     ]
+    cmd.remove('--map-id-template'); cmd.remove(f'{options.source_file} ({options.package_name})') # required to omit package check for mapping file
 
     if options.disable_checks:
       cmd += ['--map-diagnostics:CheckDiscardDiagnostic', 'error', 'none']
@@ -467,6 +468,8 @@ def _CheckForMissingSymbols(r8_path, dex_files, classpath, warnings_as_errors,
 
         # Found in: com/facebook/fbui/textlayoutbuilder/StaticLayoutHelper
         'android.text.StaticLayout.<init>',
+        'android.adservices.measurement',
+        'android.adservices.topics',
         # TODO(crbug/1426964): Remove once chrome builds with Android U SDK.
         ' android.',
 
@@ -719,7 +722,7 @@ def _Run(options):
 def main():
   build_utils.InitLogging('PROGUARD_DEBUG')
   options = _ParseOptions()
-
+  options.warnings_as_errors = False
   if options.dump_inputs:
     # Dumping inputs causes output to be emitted, avoid failing due to stdout.
     options.warnings_as_errors = False
