diff --git a/build/config/compiler/BUILD.gn b/build/config/compiler/BUILD.gn
index 80bc79e5fb350f5cc7664f1b5be3dc61142f9e55..b5c3b380c011ffdd314042db4476ab61112495bd 100644
--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -860,7 +860,7 @@ config("compiler") {
   # Pass flag to LLD so Android builds can allow debuggerd to properly symbolize
   # stack crashes (http://crbug.com/919499).
   if (use_lld && is_android) {
-    ldflags += [ "-Wl,--no-rosegment" ]
+    ldflags += [ "-Wl,--no-rosegment", "-Wl,--error-limit=0" ]
   }
 
   # TODO(crbug.com/1374347): Cleanup undefined symbol errors caught by
@@ -1142,6 +1142,10 @@ config("compiler_cpu_abi") {
   if (is_chromeos) {
     configs += [ "//build/config/chromeos:compiler_cpu_abi" ]
   }
+  if (is_apple && current_cpu == "arm64") {
+    cflags += [ "-fno-global-isel" , "-fobjc-legacy-dispatch", "-fobjc-abi-version=2"]
+    ldflags += [ "-fno-global-isel", "-Wl,-ObjC", "-fobjc-legacy-dispatch", "-fobjc-abi-version=2"]
+  }
 
   if ((is_posix && !is_apple) || is_fuchsia) {
     # CPU architecture. We may or may not be doing a cross compile now, so for
@@ -2951,6 +2955,11 @@ if (is_apple) {
     cflags_objc = common_flags
     cflags_objcc = common_flags
   }
+  config("enable_legacy_dispatch") {
+    common_flags = [ "-fobjc-legacy-dispatch", "-fobjc-abi-version=2" ]
+    cflags_objc = common_flags
+    cflags_objcc = common_flags
+  }
 }
 
 if (is_android) {
