diff --git a/build/config/ios/BUILD.gn b/build/config/ios/BUILD.gn
index 295f36de0864a3a9591774d0eda2012bced72707..f42f744297e752f499960ac69eca9f6e88161205 100644
--- a/build/config/ios/BUILD.gn
+++ b/build/config/ios/BUILD.gn
@@ -147,7 +147,7 @@ config("ios_executable_flags") {
     ldflags += [ "-Wl,-rpath,@loader_path/../Frameworks" ]
   }
 
-  ldflags += [ "-Wl,-rpath,@executable_path/Frameworks" ]
+  ldflags += [ "-Wl,-rpath,/usr/lib/swift,-rpath,@executable_path/Frameworks" ]
 }
 
 config("ios_extension_executable_flags") {
