diff --git a/BUILD.gn b/BUILD.gn
index 7cfadb4724141363c2fb4895ed2333e1cfc303b9..b840db9256870da8b5c433297b3af034de068635 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -278,7 +278,7 @@ group("gn_all") {
     deps += [ "//printing:printing_unittests" ]
   }
 
-  if (enable_extensions) {
+  if (false && enable_extensions) {
     deps += [
       "//extensions:extensions_browsertests",
       "//extensions:extensions_unittests",
