diff --git a/chrome/android/java/src/org/chromium/chrome/browser/tab/TabStateBrowserControlsVisibilityDelegate.java b/chrome/android/java/src/org/chromium/chrome/browser/tab/TabStateBrowserControlsVisibilityDelegate.java
index 2dc222c71ed4310f8bb51962d6840de296d67fb0..84370990e870f6778824bafc7dc01e4011470f5c 100644
--- a/chrome/android/java/src/org/chromium/chrome/browser/tab/TabStateBrowserControlsVisibilityDelegate.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/tab/TabStateBrowserControlsVisibilityDelegate.java
@@ -24,6 +24,8 @@ import org.chromium.content_public.browser.WebContents;
 import org.chromium.ui.base.WindowAndroid;
 import org.chromium.url.GURL;
 
+import org.chromium.base.ContextUtils;
+
 /**
  * Determines the desired visibility of the browser controls based on the current state of a given
  * tab.
@@ -203,6 +205,18 @@ public class TabStateBrowserControlsVisibilityDelegate
         enableHidingBrowserControls &= !ChromeAccessibilityUtil.get().isAccessibilityEnabled();
         enableHidingBrowserControls &= DeviceClassManager.enableFullscreen();
 
+        String KeepToolbarSetting = ContextUtils.getAppSharedPreferences().getString("keep_toolbar_visible_configuration", "unknown");
+        if (KeepToolbarSetting.equals("unknown")) {
+          if (ChromeAccessibilityUtil.get().isAccessibilityEnabled())
+            enableHidingBrowserControls &= false;
+          else
+            enableHidingBrowserControls &= true;
+        } else if (KeepToolbarSetting.equals("on")) {
+            enableHidingBrowserControls &= false;
+        } else {
+            enableHidingBrowserControls &= true;
+        }
+
         return enableHidingBrowserControls;
     }
 
