diff --git a/chrome/test/base/test_browser_window.h b/chrome/test/base/test_browser_window.h
index 0c390855e3d660793c1a349b785efbe9e9d0c0da..6669d8e4756586e7893a57a29647f1479e631e2c 100644
--- a/chrome/test/base/test_browser_window.h
+++ b/chrome/test/base/test_browser_window.h
@@ -21,6 +21,7 @@
 #include "chrome/browser/ui/browser_window.h"
 #include "chrome/browser/ui/location_bar/location_bar.h"
 #include "chrome/browser/ui/translate/partial_translate_bubble_model.h"
+#include "chrome/browser/ui/browser_list_observer.h"
 #include "chrome/common/buildflags.h"
 #include "components/user_education/common/feature_promo_controller.h"
 #include "components/user_education/common/new_badge_controller.h"
