diff --git a/chrome/browser/ui/android/omnibox/java/src/org/chromium/chrome/browser/omnibox/UrlBarData.java b/chrome/browser/ui/android/omnibox/java/src/org/chromium/chrome/browser/omnibox/UrlBarData.java
index 4e5cd991524ebce8a2b8f7e0a325786578065880..31eb75b48824ea66ff5d1f7b8f5f4c4aa0ede86d 100644
--- a/chrome/browser/ui/android/omnibox/java/src/org/chromium/chrome/browser/omnibox/UrlBarData.java
+++ b/chrome/browser/ui/android/omnibox/java/src/org/chromium/chrome/browser/omnibox/UrlBarData.java
@@ -42,7 +42,7 @@ public class UrlBarData {
                     UrlConstants.HTTPS_SCHEME,
                     UrlConstants.INLINE_SCHEME,
                     UrlConstants.JAVASCRIPT_SCHEME,
-                    UrlConstants.CHROME_SCHEME);
+                    UrlConstants.CHROME_SCHEME, "mises");
 
     /** Represents an empty URL bar. */
     public static final UrlBarData EMPTY = forNonUrlText("");
@@ -75,6 +75,13 @@ public class UrlBarData {
             return forNonUrlText(displayTextStr, editingText);
         }
 
+        String originalScheme = Uri.parse(displayTextStr).getScheme();
+	    if (!TextUtils.isEmpty(originalScheme) && (originalScheme.equals("chrome") || originalScheme.equals("chrome-extension"))){
+            displayTextStr = displayTextStr.replace("chrome://", "mises://");
+            displayTextStr = displayTextStr.replace("chrome-extension://", "mises-extension://");  
+            displayText = displayTextStr;
+        }
+
         // The displayText may come with scheme stripped. In these cases, attempting to extract
         // scheme (e.g. via Uri.parse()) may return
         // - hostname, if the address includes port number
@@ -188,4 +195,9 @@ public class UrlBarData {
         this.originEndIndex = originEndIndex;
         this.editingText = editingText;
     }
+
+    public static String replaceOnce(
+		    String url, String displayText, String editingText) {
+        return url.replace(displayText, editingText);
+    }
 }
