diff --git a/chrome/browser/ui/android/theme/java/src/org/chromium/chrome/browser/theme/ThemeUtils.java b/chrome/browser/ui/android/theme/java/src/org/chromium/chrome/browser/theme/ThemeUtils.java
index f224a9925990f9529b0463a87c6370ff3232d90b..d5ece0e62129df9c43beeb242bbeaea315a4004d 100644
--- a/chrome/browser/ui/android/theme/java/src/org/chromium/chrome/browser/theme/ThemeUtils.java
+++ b/chrome/browser/ui/android/theme/java/src/org/chromium/chrome/browser/theme/ThemeUtils.java
@@ -39,6 +39,9 @@ public class ThemeUtils {
      */
     public static int getBackgroundColor(Tab tab) {
         if (tab.isNativePage()) return tab.getNativePage().getBackgroundColor();
+        boolean isIncognito = tab != null && tab.isIncognito();
+        boolean isLocalNTP = tab != null && tab.getUrl().getSpec().contains("/local-ntp/");
+        if (isIncognito && isLocalNTP) return Color.BLACK;
 
         WebContents tabWebContents = tab.getWebContents();
         RenderWidgetHostView rwhv =
