diff --git a/chrome/android/java/src/org/chromium/chrome/browser/tasks/ReturnToChromeUtil.java b/chrome/android/java/src/org/chromium/chrome/browser/tasks/ReturnToChromeUtil.java
index cc63931e0f44a7b4ec85d2ae0b3135fc51532418..cbc7a008f5affbd3837293fe64a56a723d9e2671 100644
--- a/chrome/android/java/src/org/chromium/chrome/browser/tasks/ReturnToChromeUtil.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/tasks/ReturnToChromeUtil.java
@@ -465,7 +465,8 @@ public final class ReturnToChromeUtil {
             if (!initialized && !sSkipInitializationCheckForTesting) {
                 return false;
             } else {
-                return useChromeHomepage();
+                boolean mises_enforce = false;
+                return mises_enforce && useChromeHomepage();
             }
         }
 
