diff --git a/chrome/browser/ui/android/appmenu/java/src/org/chromium/chrome/browser/ui/appmenu/AppMenuDelegate.java b/chrome/browser/ui/android/appmenu/java/src/org/chromium/chrome/browser/ui/appmenu/AppMenuDelegate.java
index 5bc73f359b486fdbaf1f926a3290c16e63358bbf..41aeb396732126aa7e8ca9a7db0f657b674fd668 100644
--- a/chrome/browser/ui/android/appmenu/java/src/org/chromium/chrome/browser/ui/appmenu/AppMenuDelegate.java
+++ b/chrome/browser/ui/android/appmenu/java/src/org/chromium/chrome/browser/ui/appmenu/AppMenuDelegate.java
@@ -24,4 +24,7 @@ public interface AppMenuDelegate {
      *         should be using.
      */
     AppMenuPropertiesDelegate createAppMenuPropertiesDelegate();
+    void setLastItemTitle(String itemTitle);
+    void setLastVisibleItemTitle(String itemTitle);
+    void setShowExtensionOnly();
 }
