diff --git a/chrome/android/BUILD.gn b/chrome/android/BUILD.gn
index e61d70856282ffbef8abc4398c8c98cab4122688..846b01250ffb8b9bd95027bb63a2ce1179a39e25 100644
--- a/chrome/android/BUILD.gn
+++ b/chrome/android/BUILD.gn
@@ -170,6 +170,7 @@ if (current_toolchain == default_toolchain) {
       "java/res_chromium_base/mipmap-xxxhdpi/layered_app_icon_background.png",
       "java/res_chromium_base/values/channel_constants.xml",
     ]
+    sources += mises_base_module_resources
   }
 
   android_resources("chrome_app_java_resources") {
@@ -237,6 +238,7 @@ if (current_toolchain == default_toolchain) {
       "//third_party/androidx:androidx_gridlayout_gridlayout_java",
       "//third_party/androidx:androidx_preference_preference_java",
     ]
+    deps += mises_chrome_app_java_resources_deps sources += mises_java_resources resource_overlay = true
   }
 
   android_library("app_hooks_java") {
@@ -665,6 +667,7 @@ if (current_toolchain == default_toolchain) {
     ]
 
     deps += feed_deps
+    deps += mises_chrome_java_deps
 
     srcjar_deps = [
       ":chrome_android_java_enums_srcjar",
@@ -683,6 +686,7 @@ if (current_toolchain == default_toolchain) {
       "//components/supervised_user/core/browser:supervised_user_url_filter_enum_javagen",
       "//net:effective_connection_type_java",
     ]
+    srcjar_deps += mises_chrome_java_srcjar_deps
 
     # From java_sources.gni.
     sources = chrome_java_sources + [ app_hooks_impl ]
@@ -824,6 +828,7 @@ if (current_toolchain == default_toolchain) {
       "//components/messages/android/internal:java",
       "//components/segmentation_platform/internal:internal_java",
     ]
+    deps += mises_chrome_all_java_deps
   }
 
   action_with_pydeps("chrome_android_java_google_api_keys_srcjar") {
@@ -2171,6 +2176,7 @@ if (current_toolchain == default_toolchain) {
       "java/res_chromium_base/mipmap-xxxhdpi/layered_app_icon_background.png",
       "java/res_chromium_base/values/channel_constants.xml",
     ]
+    sources += mises_base_module_resources
 
     # Dep needed to ensure override works properly.
     deps = [ ":chrome_base_module_resources" ]
@@ -2341,7 +2347,7 @@ if (current_toolchain == default_toolchain) {
 
       additional_extra_paks = []
       if (!dfmify_dev_ui || !_is_bundle_module) {
-        additional_extra_paks += [ "$root_gen_dir/chrome/dev_ui_resources.pak" ]
+        # additional_extra_paks += [ "$root_gen_dir/chrome/dev_ui_resources.pak" ]
         deps += [ "//chrome/browser/resources:dev_ui_paks" ]
       }
 
@@ -2383,6 +2389,7 @@ if (current_toolchain == default_toolchain) {
         ":${_variant}_locale_pak_assets",
         ":${_variant}_paks",
       ]
+      deps += [ "//mises:mises_pak_assets" ]
       if (_is_monochrome) {
         deps += [ "//android_webview:locale_pak_assets" ]
       }
@@ -2647,6 +2654,7 @@ if (current_toolchain == default_toolchain) {
       "//components/payments/content/android:service_java",
       "//third_party/androidx:androidx_browser_browser_java",
     ]
+    sources += mises_java_base_module_sources deps += mises_java_base_module_deps
 
     # More deps for DFMs.
     if (dfmify_dev_ui) {
@@ -2774,6 +2782,7 @@ if (current_toolchain == default_toolchain) {
               "expectations/$target_name.$target_cpu.libs_and_assets.expected"
         }
       }
+      deps += mises_monochrome_public_apk_or_module_tmpl_deps loadable_modules = mises_monochrome_public_apk_or_module_tmpl_loadable_modules secondary_abi_loadable_modules = mises_monochrome_public_apk_or_module_tmpl_secondary_abi_loadable_modules
     }
   }
 
