diff --git a/chrome/browser/ui/browser.h b/chrome/browser/ui/browser.h
index f6efb82abb4ff65ca617b2c88501fe91382105a9..a2fa4f36474f286001d923d8f63c63a8ab740481 100644
--- a/chrome/browser/ui/browser.h
+++ b/chrome/browser/ui/browser.h
@@ -48,7 +48,7 @@
 #include "ui/gfx/geometry/rect.h"
 #include "ui/shell_dialogs/select_file_dialog.h"
 
-#if BUILDFLAG(IS_ANDROID)
+#if false && BUILDFLAG(IS_ANDROID)
 #error This file should only be included on desktop.
 #endif
 
