diff --git a/chrome/browser/ui/webui/search_engine_choice/generated_icon_utils.cc b/chrome/browser/ui/webui/search_engine_choice/generated_icon_utils.cc
index efe45255122a43669d969a604852206e1c061b6b..52d32c02fe64d1ea2e7fb5216d91caa30b3b6678 100644
--- a/chrome/browser/ui/webui/search_engine_choice/generated_icon_utils.cc
+++ b/chrome/browser/ui/webui/search_engine_choice/generated_icon_utils.cc
@@ -18,6 +18,7 @@ void AddGeneratedIconResources(content::WebUIDataSource* source,
 #if BUILDFLAG(GOOGLE_CHROME_BRANDING)
   source->AddResourcePath(directory + "google_com.png", IDR_GOOGLE_COM_PNG);
 #endif
+#if !BUILDFLAG(IS_ANDROID)
   source->AddResourcePath(directory + "baidu_com.png", IDR_BAIDU_COM_PNG);
   source->AddResourcePath(directory + "bing_com.png", IDR_BING_COM_PNG);
   source->AddResourcePath(directory + "search_brave_com.png",
@@ -86,4 +87,5 @@ void AddGeneratedIconResources(content::WebUIDataSource* source,
                           IDR_OCEANHERO_TODAY_PNG);
   source->AddResourcePath(directory + "privacywall_org.png",
                           IDR_PRIVACYWALL_ORG_PNG);
+#endif
 }
