diff --git a/chrome/browser/ui/android/ephemeraltab/java/src/org/chromium/chrome/browser/ephemeraltab/EphemeralTabMediator.java b/chrome/browser/ui/android/ephemeraltab/java/src/org/chromium/chrome/browser/ephemeraltab/EphemeralTabMediator.java
index c4cbd2ccac157cf5a7664184f86d3a6f5f273f42..ccdd8d5a30dded53355271cf256fff629cdbf46e 100644
--- a/chrome/browser/ui/android/ephemeraltab/java/src/org/chromium/chrome/browser/ephemeraltab/EphemeralTabMediator.java
+++ b/chrome/browser/ui/android/ephemeraltab/java/src/org/chromium/chrome/browser/ephemeraltab/EphemeralTabMediator.java
@@ -28,6 +28,12 @@ import org.chromium.content_public.common.ResourceRequestBody;
 import org.chromium.ui.widget.Toast;
 import org.chromium.url.GURL;
 
+import org.chromium.base.supplier.Supplier;
+import org.chromium.chrome.browser.tab.Tab;
+import org.chromium.chrome.browser.tab.TabLaunchType;
+import org.chromium.chrome.browser.tabmodel.TabCreator;
+import org.chromium.ui.base.PageTransition;
+
 /** Mediator class for preview tab, responsible for communicating with other objects. */
 public class EphemeralTabMediator {
     /** The delay (four video frames) after which the hide progress will be hidden. */
@@ -44,15 +50,34 @@ public class EphemeralTabMediator {
     private WebContentsDelegateAndroid mWebContentsDelegate;
     private Profile mProfile;
 
+    private final Supplier<Tab> mTabProvider;
+    private final Supplier<TabCreator> mTabCreator;
+
+    private void openInNewTab(final GURL url) {   
+        mTabCreator
+                .get()
+                .createNewTab(
+                        new LoadUrlParams(url.getSpec(), PageTransition.LINK),
+                        TabLaunchType.FROM_LINK,
+                        mTabProvider.get());
+    }
+    private boolean isShowingExtension() {
+        return mWebContents != null && mWebContents.getVisibleUrl().getSpec().startsWith("chrome-extension://");
+    }
+
     /** Constructor. */
     public EphemeralTabMediator(
             BottomSheetController bottomSheetController,
             EphemeralTabCoordinator.FaviconLoader faviconLoader,
-            int topControlsHeightDp) {
+            int topControlsHeightDp,
+            Supplier<Tab> tabProvider,
+            Supplier<TabCreator> tabCreator) {
         mBottomSheetController = bottomSheetController;
         mFaviconLoader = faviconLoader;
         mTopControlsHeightDp = topControlsHeightDp;
         mObservers = new ObserverList<EphemeralTabObserver>();
+        mTabProvider = tabProvider;
+        mTabCreator = tabCreator;
     }
 
     /** Initializes various objects for a new tab. */
@@ -105,6 +130,14 @@ public class EphemeralTabMediator {
         }
     }
 
+    public void onNavigationFinished(WebContents webContents) {
+        RewindableIterator<EphemeralTabObserver> observersIterator =
+                mObservers.rewindableIterator();
+        while (observersIterator.hasNext()) {
+            observersIterator.next().onNavigationFinished(webContents);
+        }
+    }
+
     /** Notify observers on title set. */
     public void onTitleSet(EphemeralTabSheetContent sheetContent, String title) {
         RewindableIterator<EphemeralTabObserver> observersIterator =
@@ -174,6 +207,7 @@ public class EphemeralTabMediator {
                         if (navigation.hasCommitted()) {
                             mIsOnErrorPage = navigation.isErrorPage();
                             mSheetContent.updateURL(mWebContents.get().getVisibleUrl());
+                            onNavigationFinished(mWebContents.get());
                         } else if (navigation.isDownload()) {
                             // Not viewable contents such as download. Show a toast and close the
                             // tab.
@@ -212,12 +246,22 @@ public class EphemeralTabMediator {
                             ResourceRequestBody postData,
                             int disposition,
                             boolean isRendererInitiated) {
+                        if (isShowingExtension()) {
+                            //open new tab in tabs when preview tab is for extension
+                            openInNewTab(url);
+                            return;
+                        }  
                         // We never open a separate tab when navigating in a preview tab.
                         loadUrl(url);
                     }
 
                     @Override
                     public boolean shouldCreateWebContents(GURL targetUrl) {
+                        if (isShowingExtension()) {
+                            //open new url in tabs when preview tab is for extension
+                            openInNewTab(targetUrl);
+                            return false;
+                        }                        
                         loadUrl(targetUrl);
                         return false;
                     }
