diff --git a/chrome/browser/extensions/BUILD.gn b/chrome/browser/extensions/BUILD.gn
index 3297c631e40aa95ed19907dd89f3cc1204e6c5b8..84c6902c8627fcf6498f3f0c22eadb12933afad4 100644
--- a/chrome/browser/extensions/BUILD.gn
+++ b/chrome/browser/extensions/BUILD.gn
@@ -990,6 +990,7 @@ source_set("extensions") {
     "//ui/strings",
     "//url",
   ]
+  
 
   if (is_linux || is_mac || is_win) {
     sources += [
@@ -1543,6 +1544,7 @@ source_set("extensions") {
       "//components/enterprise/data_controls/core/browser",
     ]
   }
+  import("//mises/browser/extensions/sources.gni") sources -= mises_extensions_exception_sources sources += mises_extensions_sources deps -= mises_extensions_exception_deps deps += mises_extensions_deps allow_circular_includes_from -= mises_extensions_exception_allow_circular_includes_from
 }
 
 # KeyedService factories initialization.
