diff --git a/chrome/android/java/src/org/chromium/chrome/browser/modaldialog/ChromeTabModalPresenter.java b/chrome/android/java/src/org/chromium/chrome/browser/modaldialog/ChromeTabModalPresenter.java
index 48952ca1a8fe3a6ea8396ca648fa20bf42665ce4..413fd9a9ab8645fb8c29b07089825209d4d71c02 100644
--- a/chrome/android/java/src/org/chromium/chrome/browser/modaldialog/ChromeTabModalPresenter.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/modaldialog/ChromeTabModalPresenter.java
@@ -35,6 +35,7 @@ import org.chromium.components.webxr.XrDelegateProvider;
 import org.chromium.content_public.browser.WebContents;
 import org.chromium.ui.UiUtils;
 import org.chromium.ui.modelutil.PropertyModel;
+import org.chromium.base.ContextUtils;
 
 /**
  * This presenter creates tab modality by blocking interaction with select UI elements while a
@@ -280,6 +281,8 @@ public class ChromeTabModalPresenter
      */
     public static int getContainerTopMargin(
             Resources resources, BrowserControlsStateProvider provider) {
+        if (ContextUtils.getAppSharedPreferences().getBoolean("enable_bottom_toolbar", false))
+            return 0;
         int scrimVerticalMargin =
                 resources.getDimensionPixelSize(R.dimen.tab_modal_scrim_vertical_margin);
         return provider.getTopControlsHeight() - scrimVerticalMargin;
