diff --git a/chrome/browser/extensions/api/permissions/permissions_api.cc b/chrome/browser/extensions/api/permissions/permissions_api.cc
index ab894f7e846b36287ab0fef8aa70cab4b6c3ecae..91237482e7101ef5ae5b0151b0903e5e33983ca8 100644
--- a/chrome/browser/extensions/api/permissions/permissions_api.cc
+++ b/chrome/browser/extensions/api/permissions/permissions_api.cc
@@ -310,9 +310,11 @@ ExtensionFunction::ResponseAction PermissionsRequestFunction::Run() {
 
   gfx::NativeWindow native_window =
       ChromeExtensionFunctionDetails(this).GetNativeWindowForUI();
+#if !BUILDFLAG(IS_ANDROID)
   if (!native_window && g_dialog_action == DialogAction::kDefault) {
     return RespondNow(Error("Could not find an active window."));
   }
+#endif
 
   std::optional<api::permissions::Request::Params> params =
       api::permissions::Request::Params::Create(args());
