diff --git a/chrome/browser/ui/webui/access_code_cast/BUILD.gn b/chrome/browser/ui/webui/access_code_cast/BUILD.gn
index 95df3bdd81afd0d8d335269fae90e66864807e79..3ac0308bb84d37f13782abd2be8ede438f62694a 100644
--- a/chrome/browser/ui/webui/access_code_cast/BUILD.gn
+++ b/chrome/browser/ui/webui/access_code_cast/BUILD.gn
@@ -5,7 +5,7 @@
 import("//mojo/public/tools/bindings/mojom.gni")
 import("//third_party/protobuf/proto_library.gni")
 
-assert(!is_android, "AccessCodeCast is not for android.")
+#assert(!is_android, "AccessCodeCast is not for android.")
 
 mojom("mojo_bindings") {
   sources = [ "access_code_cast.mojom" ]
