diff --git a/chrome/browser/ui/web_applications/BUILD.gn b/chrome/browser/ui/web_applications/BUILD.gn
index a4d302480ed486cb045f2ba1beeda8b12afe68fd..4bb4981850f1577842b11b831863bb3f68b47d6c 100644
--- a/chrome/browser/ui/web_applications/BUILD.gn
+++ b/chrome/browser/ui/web_applications/BUILD.gn
@@ -87,6 +87,10 @@ source_set("browser_tests") {
       "//chrome/browser/ui/ash/system_web_apps",
     ]
   }
+
+  if (is_android) {
+    deps -= ["//chrome/test:test_support_ui"]
+  }
 }
 
 # On Lacros, these browser tests require Ash to be running, as they rely on
@@ -143,4 +147,7 @@ source_set("app_service_browser_tests") {
       "//chromeos/lacros:test_support",
     ]
   }
+  if (is_android) {
+    deps -= ["//chrome/test:test_support_ui"]
+  }
 }
