diff --git a/chrome/browser/tabmodel/android/java/src/org/chromium/chrome/browser/tabmodel/EmptyTabModel.java b/chrome/browser/tabmodel/android/java/src/org/chromium/chrome/browser/tabmodel/EmptyTabModel.java
index f3745077720cca536a9b1c0117d9b6f3af0c9e57..1ed9e7a9c88b90758706a11a86fff282ddc4ed56 100644
--- a/chrome/browser/tabmodel/android/java/src/org/chromium/chrome/browser/tabmodel/EmptyTabModel.java
+++ b/chrome/browser/tabmodel/android/java/src/org/chromium/chrome/browser/tabmodel/EmptyTabModel.java
@@ -91,6 +91,11 @@ public class EmptyTabModel implements TabModel {
     @Override
     public void setIndex(int i, @TabSelectionType int type, boolean skipLoadingTab) {}
 
+    @Override
+    public int getLastNonExtensionActiveIndex() {
+        return INVALID_TAB_INDEX;
+    }
+
     @Override
     public boolean isActiveModel() {
         return false;
