diff --git a/chrome/browser/web_applications/BUILD.gn b/chrome/browser/web_applications/BUILD.gn
index c85587554c0abd50889d59616a24de9479bf24e6..e95ee606195df3f4d15770e850c9613ba7aadff6 100644
--- a/chrome/browser/web_applications/BUILD.gn
+++ b/chrome/browser/web_applications/BUILD.gn
@@ -369,6 +369,8 @@ source_set("web_applications") {
     ]
   }
 
+  import("//mises/browser/web_applications/sources.gni") sources += mises_browser_web_applications_sources
+
   if (is_mac) {
     sources += [
       "app_shim_registry_mac.cc",
@@ -973,6 +975,9 @@ source_set("web_applications_browser_tests") {
       "//chromeos/startup",
     ]
   }
+  if (is_android) {
+    deps -= ["//chrome/test:test_support_ui"]
+  }
 }
 
 group("browser_tests") {
@@ -1019,6 +1024,9 @@ source_set("app_service_browser_tests") {
       "//chromeos/startup",
     ]
   }
+  if (is_android) {
+    deps -= ["//chrome/test:test_support_ui"]
+  }
 }
 
 source_set("interactive_ui_tests") {
@@ -1044,6 +1052,9 @@ source_set("interactive_ui_tests") {
     "//ui/base/idle:test_support",
     "//ui/events/devices:test_support",
   ]
+  if (is_android) {
+    deps -= ["//chrome/test:test_support_ui"]
+  }
 
   if (is_chromeos) {
     sources += [ "preinstalled_web_app_window_experiment_browsertest.cc" ]
