diff --git a/chrome/browser/password_edit_dialog/android/java/src/org/chromium/chrome/browser/password_edit_dialog/PasswordEditDialogCoordinator.java b/chrome/browser/password_edit_dialog/android/java/src/org/chromium/chrome/browser/password_edit_dialog/PasswordEditDialogCoordinator.java
index 03e204bada6b71f8b5a55a54885c91f9062d191f..2ee19a101a66a2dc344e5bb0e3f07ba27cd4148e 100644
--- a/chrome/browser/password_edit_dialog/android/java/src/org/chromium/chrome/browser/password_edit_dialog/PasswordEditDialogCoordinator.java
+++ b/chrome/browser/password_edit_dialog/android/java/src/org/chromium/chrome/browser/password_edit_dialog/PasswordEditDialogCoordinator.java
@@ -120,7 +120,7 @@ class PasswordEditDialogCoordinator {
         mDialogModel = createModalDialogModel(mIsDialogWithDetailsFeatureEnabled
                         ? R.string.password_update_dialog_title
                         : R.string.confirm_username_dialog_title,
-                R.string.password_manager_update_button);
+                R.string.password_manager_update_button_android);
         mDialogViewModel =
                 createDialogViewModel(usernames, selectedUsernameIndex, password, account);
 
@@ -202,4 +202,4 @@ class PasswordEditDialogCoordinator {
     PropertyModel getDialogViewModelForTesting() {
         return mDialogViewModel;
     }
-}
\ No newline at end of file
+}
