diff --git a/chrome/browser/resources/side_panel/customize_chrome/icons/BUILD.gn b/chrome/browser/resources/side_panel/customize_chrome/icons/BUILD.gn
index 392d0fee946f0b735cc3ef0050b7fd2015ad36b5..5dbb0631df4282a2558d383a986524da1a5cd82f 100644
--- a/chrome/browser/resources/side_panel/customize_chrome/icons/BUILD.gn
+++ b/chrome/browser/resources/side_panel/customize_chrome/icons/BUILD.gn
@@ -4,7 +4,7 @@
 
 import("//ui/webui/resources/tools/generate_grd.gni")
 
-assert(!is_android)
+#assert(!is_android)
 
 generate_grd("build_grdp") {
   grd_prefix = "side_panel_customize_chrome"
