diff --git a/chrome/browser/resources/browser_switch/BUILD.gn b/chrome/browser/resources/browser_switch/BUILD.gn
index 07bf39c21dc8d225823dbd479756f6257a606b55..f45691b8d9f43f62deb54184eea389d75128a027 100644
--- a/chrome/browser/resources/browser_switch/BUILD.gn
+++ b/chrome/browser/resources/browser_switch/BUILD.gn
@@ -5,7 +5,7 @@
 import("//chrome/common/features.gni")
 import("//ui/webui/resources/tools/build_webui.gni")
 
-assert(is_win || is_mac || is_linux || is_chromeos_lacros || is_fuchsia)
+#assert(is_win || is_mac || is_linux || is_chromeos_lacros || is_fuchsia)
 
 build_webui("build") {
   grd_prefix = "browser_switch"
