diff --git a/chrome/browser/picture_in_picture/picture_in_picture_window_manager.h b/chrome/browser/picture_in_picture/picture_in_picture_window_manager.h
index f2bce63a4c8bbc1ae07912788f582e593fa56622..62933667ad3c476fbff2c6625c96cf24430515bc 100644
--- a/chrome/browser/picture_in_picture/picture_in_picture_window_manager.h
+++ b/chrome/browser/picture_in_picture/picture_in_picture_window_manager.h
@@ -32,7 +32,7 @@ namespace display {
 class Display;
 }  // namespace display
 
-#if !BUILDFLAG(IS_ANDROID)
+#if true || !BUILDFLAG(IS_ANDROID)
 class PictureInPictureOcclusionTracker;
 
 namespace views {
@@ -193,6 +193,8 @@ class PictureInPictureWindowManager {
   // file dialogs and picture-in-picture windows are simultaneously open.
   void OnFileDialogOpened();
   void OnFileDialogClosed();
+#else
+  PictureInPictureOcclusionTracker* GetOcclusionTracker();
 #endif
 
   void set_window_controller_for_testing(
