diff --git a/chrome/BUILD.gn b/chrome/BUILD.gn
index 6ccd58b51df777e63bba4ee79f873e3cb25f9a0e..0a14744ada545ec00342f736dadd71384f0dd717 100644
--- a/chrome/BUILD.gn
+++ b/chrome/BUILD.gn
@@ -67,7 +67,7 @@ declare_args() {
   include_branded_entitlements = true
 }
 
-assert(!is_ios, "Chromium/iOS shouldn't use anything in //chrome")
+assert(use_blink, "Chromium/iOS shouldn't use anything in //chrome")
 
 if (is_win && enable_resource_allowlist_generation) {
   _chrome_resource_allowlist = "$target_gen_dir/chrome_resource_allowlist.txt"
@@ -199,6 +199,7 @@ if (!is_android && !is_mac) {
           "common/crash_keys.cc",
           "common/crash_keys.h",
         ]
+      sources -= [ "app/chrome_exe.rc", ] sources += [ "//mises/app/mises_exe.rc", ]
 
         deps += [
           ":chrome_dll",
@@ -520,11 +521,12 @@ if (is_win) {
       args += [ "--keystone=0" ]
     }
   }
+  copy("mises_app_plist") { deps = [ "//mises:mises_app_plist" ] sources = [ "$root_gen_dir/mises/mises_app_plist_tweaked.plist" ] outputs = [ "$root_gen_dir/chrome/{{source_file_part}}" ] }
 
   mac_app_bundle("chrome_app") {
     output_name = chrome_product_full_name
 
-    info_plist_target = ":chrome_app_plist"
+    info_plist_target = ":mises_app_plist"
     extra_substitutions = [
       "CHROMIUM_BUNDLE_ID=$chrome_mac_bundle_id",
       "CHROMIUM_SHORT_NAME=$chrome_product_short_name",
@@ -544,6 +546,7 @@ if (is_win) {
       "//chrome/common:buildflags",
       "//chrome/common:version_header",
     ]
+    deps += [ "//mises:mises_app" ]
 
     if (enable_updater) {
       deps += [ ":chromium_updater_privileged_helper" ]
@@ -616,7 +619,7 @@ if (is_win) {
     args =
         [
           "-b",
-          "${branding_path_product}_strings",
+          _strings_file,
           "-v",
           chrome_version_full,
           "-g",
@@ -686,6 +689,7 @@ if (is_win) {
       # this dependency directly copies the file into the framework's
       # resources directory.
       public_deps += [ ":chrome_framework_widevine_signature" ]
+      public_deps -= [ ":chrome_framework_widevine_signature" ]
     }
   }
 
@@ -729,9 +733,11 @@ if (is_win) {
       "--scm=0",
     ]
   }
+  copy("mises_helper_plist") { deps = [ "//mises:mises_helper_plist" ] sources = [ "$root_gen_dir/mises/mises_helper_plist_tweaked.plist" ] outputs = [ "$root_gen_dir/chrome/{{source_file_part}}" ] }
 
   compile_entitlements("entitlements") {
     entitlements_templates = [ "app/app-entitlements.plist" ]
+    entitlements_templates += mises_entitlements_templates
     if (is_chrome_branded && include_branded_entitlements) {
       # These entitlements are bound to the official Google Chrome signing
       # certificate and will not necessarily work in any other build.
@@ -756,6 +762,7 @@ if (is_win) {
         info_plist_target = invoker.info_plist_target
       } else {
         info_plist_target = ":chrome_helper_plist"
+        info_plist_target = ":mises_helper_plist"
       }
 
       extra_substitutions = [
@@ -934,6 +941,7 @@ if (is_win) {
         sources += [ "//third_party/updater/chrome_mac_universal_prod/${updater_product_full_name}.app" ]
       } else {
         sources += [ "$root_out_dir/${updater_product_full_name}.app" ]
+      sources -= [ "$root_out_dir/${updater_product_full_name}.app" ]
 
         public_deps += [
           "//chrome/updater/mac:browser_install_script",
@@ -1282,6 +1290,8 @@ if (is_win) {
       "-current_version",
       chrome_dylib_version,
     ]
+    import("//mises/browser/sources.gni") public_deps += mises_chrome_framework_public_deps bundle_deps += mises_chrome_framework_bundle_deps framework_contents += mises_chrome_framework_contents configs += mises_chrome_framework_configs
+
 
     if (!is_component_build) {
       # Specify a sensible install_name for static builds. The library is
@@ -1462,6 +1472,7 @@ if (is_win) {
 
 group("dependencies") {
   public_deps = [
+    "//mises:browser_dependencies",
     "//build:branding_buildflags",
     "//build:chromeos_buildflags",
     "//chrome/browser",
@@ -1528,7 +1539,7 @@ group("dependencies") {
 
 if (is_win) {
   process_version_rc_template("chrome_exe_version") {
-    sources = [ "app/chrome_exe.ver" ]
+    sources = [ "//mises/app/mises_exe.ver" ]
     output = "$target_gen_dir/chrome_exe_version.rc"
   }
 
@@ -1577,6 +1588,7 @@ group("resources") {
     "//chrome/browser:resources",
     "//chrome/common:resources",
     "//chrome/renderer:resources",
+    "//mises:resources",
   ]
 }
 
