diff --git a/chrome/browser/touch_to_fill/touch_to_fill_keyboard_suppressor.cc b/chrome/browser/touch_to_fill/touch_to_fill_keyboard_suppressor.cc
index e8bc1ffe1f99e672e82f663087e328b8d6892e24..59c5fdad9623cba6a40f9f630a02a06307e47a74 100644
--- a/chrome/browser/touch_to_fill/touch_to_fill_keyboard_suppressor.cc
+++ b/chrome/browser/touch_to_fill/touch_to_fill_keyboard_suppressor.cc
@@ -46,7 +46,6 @@ void TouchToFillKeyboardSuppressor::OnContentAutofillDriverCreated(
   // function already, the callback has been registered with `rwh` already iff
   // the parent frame's RWH is identical to `rwh`.
   content::RenderFrameHost* rfh = driver.render_frame_host();
-  DCHECK_EQ(rfh->GetParent() != nullptr, driver.GetParent() != nullptr);
   content::RenderWidgetHost* rwh = rfh->GetRenderWidgetHost();
   if (!rfh->GetParent() || rfh->GetParent()->GetRenderWidgetHost() != rwh) {
     // We don't need to call RWH::RemoveSuppressShowingImeCallback(): it's
