diff --git a/chrome/android/features/tab_ui/java/src/org/chromium/chrome/browser/tasks/tab_management/TabSwitcherMediator.java b/chrome/android/features/tab_ui/java/src/org/chromium/chrome/browser/tasks/tab_management/TabSwitcherMediator.java
index 3b59af7159ca0558dc75c85ca0cb9bde91177036..cb1b58822aced42ac4374a9fb47bc33accfcc5a5 100644
--- a/chrome/android/features/tab_ui/java/src/org/chromium/chrome/browser/tasks/tab_management/TabSwitcherMediator.java
+++ b/chrome/android/features/tab_ui/java/src/org/chromium/chrome/browser/tasks/tab_management/TabSwitcherMediator.java
@@ -68,6 +68,7 @@ import org.chromium.chrome.tab_ui.R;
 import org.chromium.content_public.browser.UiThreadTaskTraits;
 import org.chromium.ui.modelutil.PropertyModel;
 
+import org.chromium.base.ContextUtils;
 import java.util.List;
 
 /**
@@ -255,6 +256,11 @@ class TabSwitcherMediator implements TabSwitcher.Controller, TabListRecyclerView
         mTabModelSelector = tabModelSelector;
         mBrowserControlsStateProvider = browserControlsStateProvider;
         mMultiWindowModeStateDispatcher = multiWindowModeStateDispatcher;
+        final String active_tabswitcher = ContextUtils.getAppSharedPreferences().getString("active_tabswitcher", "default");
+        if (active_tabswitcher.equals("list"))
+          mode = TabListCoordinator.TabListMode.LIST;
+        if (active_tabswitcher.equals("default") || active_tabswitcher.equals("grid"))
+          mode = TabListCoordinator.TabListMode.GRID;
         mMode = mode;
         mContext = context;
 
@@ -697,6 +703,8 @@ class TabSwitcherMediator implements TabSwitcher.Controller, TabListRecyclerView
     private void setInitialScrollIndexOffset() {
         int offset = mMode == TabListMode.CAROUSEL ? INITIAL_SCROLL_INDEX_OFFSET_CAROUSEL
                                                    : INITIAL_SCROLL_INDEX_OFFSET_GTS;
+        if (ContextUtils.getAppSharedPreferences().getString("active_tabswitcher", "default").equals("classic"))
+          offset = 0;
         int initialPosition = Math.max(
                 mTabModelSelector.getTabModelFilterProvider().getCurrentTabModelFilter().index()
                         - offset,
