diff --git a/chrome/browser/feedback/android/java/src/org/chromium/chrome/browser/feedback/HelpAndFeedbackLauncherDelegate.java b/chrome/browser/feedback/android/java/src/org/chromium/chrome/browser/feedback/HelpAndFeedbackLauncherDelegate.java
index b4410a1abd95f4619ef798c9795218fde81c3552..4a6df14b5220312e5d79cc62455cad6ac0403599 100644
--- a/chrome/browser/feedback/android/java/src/org/chromium/chrome/browser/feedback/HelpAndFeedbackLauncherDelegate.java
+++ b/chrome/browser/feedback/android/java/src/org/chromium/chrome/browser/feedback/HelpAndFeedbackLauncherDelegate.java
@@ -14,7 +14,7 @@ import androidx.annotation.NonNull;
 
 /** Delegate that handles the display of the HelpAndFeedback flows. */
 public interface HelpAndFeedbackLauncherDelegate {
-    static final String FALLBACK_SUPPORT_URL = "https://support.google.com/chrome/topic/6069782";
+    static final String FALLBACK_SUPPORT_URL = "https://www.mises.site/guide";
 
     /**
      * Starts an activity showing a help page for the specified context ID.
