diff --git a/chrome/browser/resources/new_tab_page/modules/recipes_v2/icons/BUILD.gn b/chrome/browser/resources/new_tab_page/modules/recipes_v2/icons/BUILD.gn
index d3170b58e8ce92e0267883a5dc7c6d121be5c7c4..196b111c023158e9064b64c1f915e3c0c6f90faa 100644
--- a/chrome/browser/resources/new_tab_page/modules/recipes_v2/icons/BUILD.gn
+++ b/chrome/browser/resources/new_tab_page/modules/recipes_v2/icons/BUILD.gn
@@ -4,7 +4,7 @@
 
 import("//ui/webui/resources/tools/generate_grd.gni")
 
-assert(!is_android)
+#assert(!is_android)
 
 generate_grd("build_grdp") {
   grd_prefix = "recipes"
