diff --git a/chrome/browser/ui/android/omnibox/java/src/org/chromium/chrome/browser/omnibox/suggestions/OmniboxSuggestionsDropdown.java b/chrome/browser/ui/android/omnibox/java/src/org/chromium/chrome/browser/omnibox/suggestions/OmniboxSuggestionsDropdown.java
index 1754fba47f6a0d31b2a163f175ac01f534abcfca..034ba586e1839bea6721fc0df79de9c3ac972674 100644
--- a/chrome/browser/ui/android/omnibox/java/src/org/chromium/chrome/browser/omnibox/suggestions/OmniboxSuggestionsDropdown.java
+++ b/chrome/browser/ui/android/omnibox/java/src/org/chromium/chrome/browser/omnibox/suggestions/OmniboxSuggestionsDropdown.java
@@ -27,6 +27,7 @@ import androidx.core.view.ViewCompat;
 import androidx.recyclerview.widget.LinearLayoutManager;
 import androidx.recyclerview.widget.RecyclerView;
 
+import org.chromium.base.ContextUtils;
 import org.chromium.base.TraceEvent;
 import org.chromium.base.metrics.TimingMetric;
 import org.chromium.base.task.PostTask;
@@ -315,11 +316,37 @@ public class OmniboxSuggestionsDropdown extends RecyclerView {
         try (TraceEvent tracing = TraceEvent.scoped("OmniboxSuggestionsList.Measure");
                 TimingMetric metric = SuggestionsMetrics.recordSuggestionListMeasureTime()) {
             int anchorBottomRelativeToContent = calculateAnchorBottomRelativeToContent();
+            if (ContextUtils.getAppSharedPreferences().getBoolean("enable_bottom_toolbar", false))
+              anchorBottomRelativeToContent = 0;
             maybeUpdateLayoutParams(anchorBottomRelativeToContent);
 
             int availableViewportHeight =
                     calculateAvailableViewportHeight(anchorBottomRelativeToContent);
+            int viewportDivider = 1;
+            if (ContextUtils.getAppSharedPreferences().getBoolean("enable_bottom_toolbar", false))
+              viewportDivider = 2;
+            int displayableSuggestions = 0;
+            int suggestionHeightPx = getResources().getDimensionPixelSize(R.dimen.omnibox_suggestion_semicompact_height);
+            for (int spaceTakenBySuggestions = 0; (spaceTakenBySuggestions + suggestionHeightPx) < (availableViewportHeight / viewportDivider); spaceTakenBySuggestions += suggestionHeightPx, displayableSuggestions++);
+            if (mAdapter != null && displayableSuggestions > mAdapter.getItemCount())
+              displayableSuggestions = mAdapter.getItemCount();
+            if (displayableSuggestions == 0)
+              displayableSuggestions = 1;
+            // When bottom toolbar is enabled, having two suggestions displays as 1
+            if (displayableSuggestions == 1 && ContextUtils.getAppSharedPreferences().getBoolean("enable_bottom_toolbar", false))
+              displayableSuggestions = 2;
+            availableViewportHeight = displayableSuggestions * suggestionHeightPx;
+            availableViewportHeight += getResources().getDimensionPixelSize(R.dimen.omnibox_suggestion_list_padding_bottom) / 2;
+
             int desiredWidth = mAnchorView.getMeasuredWidth();
+
+            notifyObserversIfViewportHeightChanged(availableViewportHeight);
+            mWidthMeasureSpec = MeasureSpec.makeMeasureSpec(desiredWidth, MeasureSpec.EXACTLY);
+            mHeightMeasureSpec = MeasureSpec.makeMeasureSpec(availableViewportHeight,
+                    mEmbedder.isTablet() ? MeasureSpec.AT_MOST : MeasureSpec.EXACTLY);
+            super.onMeasure(mWidthMeasureSpec, mHeightMeasureSpec);
+            requestLayout();
+
             // Suppress the initial requests to shrink the viewport of the omnibox suggestion
             // dropdown. The viewport will decrease when the keyboard is triggered, but the request
             // to resize happens when the keyboard starts showing before it has had the chance to
