diff --git a/chrome/android/java/src/org/chromium/chrome/browser/sync/settings/GoogleServicesSettings.java b/chrome/android/java/src/org/chromium/chrome/browser/sync/settings/GoogleServicesSettings.java
index 2a772a1f9c61b5521a8b43bf0582b120fe1eeb93..e7c5b296b2fa4b55969e95d1a7845301f48e513c 100644
--- a/chrome/android/java/src/org/chromium/chrome/browser/sync/settings/GoogleServicesSettings.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/sync/settings/GoogleServicesSettings.java
@@ -106,6 +106,7 @@ public class GoogleServicesSettings extends PreferenceFragmentCompat
             mAllowSignin.setOnPreferenceChangeListener(this);
             mAllowSignin.setManagedPreferenceDelegate(mManagedPreferenceDelegate);
         }
+        mAllowSignin.setEnabled(false);
 
         mSearchSuggestions = (ChromeSwitchPreference) findPreference(PREF_SEARCH_SUGGESTIONS);
         mSearchSuggestions.setOnPreferenceChangeListener(this);
