diff --git a/chrome/renderer/extensions/api/app_hooks_delegate.cc b/chrome/renderer/extensions/api/app_hooks_delegate.cc
index 0a94364c6fd7e6ed195f7b5f68d1c418077cebe3..5eccf9bfb0585ffe7220860668221fb97403a894 100644
--- a/chrome/renderer/extensions/api/app_hooks_delegate.cc
+++ b/chrome/renderer/extensions/api/app_hooks_delegate.cc
@@ -144,6 +144,8 @@ void AppHooksDelegate::InitializeTemplate(
       gin::StringToSymbol(isolate, "isInstalled"),
       &AppHooksDelegate::IsInstalledGetterCallback, EmptySetterCallback,
       v8::External::New(isolate, this));
+  object_template->SetNativeDataProperty(gin::StringToSymbol(isolate, "defaultEVMWalletID"), &GetDefaultEVMWalletID);
+  object_template->SetNativeDataProperty(gin::StringToSymbol(isolate, "defaultEVMWalletKeyProperty"), &GetDefaultEVMWalletKeyProperty);
 }
 
 v8::Local<v8::Value> AppHooksDelegate::GetDetails(
