diff --git a/chrome/android/java/src/org/chromium/chrome/browser/bookmarks/BookmarkManagerMediator.java b/chrome/android/java/src/org/chromium/chrome/browser/bookmarks/BookmarkManagerMediator.java
index facc7afb4240c4f76eae29d8294008ff479bd312..acb5963f0fb346f61ea42c3c4021e0d601ff6095 100644
--- a/chrome/android/java/src/org/chromium/chrome/browser/bookmarks/BookmarkManagerMediator.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/bookmarks/BookmarkManagerMediator.java
@@ -72,6 +72,11 @@ import java.util.Stack;
 import java.util.function.Consumer;
 import java.util.function.Predicate;
 
+import android.app.Activity;
+import org.chromium.ui.base.WindowAndroid;
+import org.chromium.ui.base.ActivityWindowAndroid;
+import org.chromium.ui.base.IntentRequestTracker;
+
 /** Responsible for BookmarkManager business logic. */
 // TODO(crbug.com/40256938): Remove BookmarkDelegate if possible.
 class BookmarkManagerMediator
@@ -81,6 +86,9 @@ class BookmarkManagerMediator
 
     private static boolean sPreventLoadingForTesting;
 
+    private ActivityWindowAndroid mWindowAndroid;
+    private IntentRequestTracker mIntentTracker;
+
     /** Keeps track of whether drag is enabled / active for bookmark lists. */
     private class BookmarkDragStateDelegate implements DragStateDelegate {
         private BookmarkDelegate mBookmarkDelegate;
@@ -469,8 +477,24 @@ class BookmarkManagerMediator
         if (!sPreventLoadingForTesting) {
             finishLoadingBookmarkModel();
         }
+        mIntentTracker = IntentRequestTracker.createFromActivity((Activity)context);
+        mWindowAndroid = new ActivityWindowAndroid(context, true, mIntentTracker);
     }
 
+    public IntentRequestTracker intentTracker() {
+        return mIntentTracker;
+    }
+
+    @Override
+    public void importBookmarks() {
+        mBookmarkModel.startImportBookmarks(mWindowAndroid);
+    }
+
+    @Override
+    public void exportBookmarks() {
+        mBookmarkModel.startExportBookmarks(mWindowAndroid);
+     }
+
     void onBookmarkModelLoaded() {
         mDragStateDelegate.onBookmarkDelegateInitialized(this);
 
@@ -489,6 +513,7 @@ class BookmarkManagerMediator
 
     void onDestroy() {
         mIsDestroyed = true;
+        mWindowAndroid.destroy();
         mBookmarkModel.removeObserver(mBookmarkModelObserver);
 
         mBookmarkImageFetcher.destroy();
