diff --git a/chrome/android/features/start_surface/java/src/org/chromium/chrome/features/start_surface/StartSurfaceMediator.java b/chrome/android/features/start_surface/java/src/org/chromium/chrome/features/start_surface/StartSurfaceMediator.java
index 50c8ebd7f287f8564cafb74cb8181e307a0dbf32..1a1d1d7f2c7d7f95fd60d744207bc2297f5eab47 100644
--- a/chrome/android/features/start_surface/java/src/org/chromium/chrome/features/start_surface/StartSurfaceMediator.java
+++ b/chrome/android/features/start_surface/java/src/org/chromium/chrome/features/start_surface/StartSurfaceMediator.java
@@ -108,6 +108,7 @@ import org.chromium.content_public.browser.LoadUrlParams;
 import org.chromium.ui.modelutil.PropertyModel;
 import org.chromium.ui.text.EmptyTextWatcher;
 import org.chromium.ui.util.ColorUtils;
+import org.chromium.chrome.browser.feed.FeedFeatures;
 
 import java.util.List;
 
@@ -595,7 +596,7 @@ class StartSurfaceMediator implements TabSwitcher.TabSwitcherViewObserver, View.
         if (mTabSwitcherModule != null) {
             mTabSwitcherModule.initWithNative();
         }
-        mFeedVisibilityPrefOnStartUp = prefService.getBoolean(Pref.ARTICLES_LIST_VISIBLE);
+        mFeedVisibilityPrefOnStartUp = FeedFeatures.isFeedEnabled() && prefService.getBoolean(Pref.ARTICLES_LIST_VISIBLE);
 
         // Trigger the creation of spare tab for StartSurface after the native is initialized to
         // speed up navigation from start.
