diff --git a/chrome/common/google_url_loader_throttle.cc b/chrome/common/google_url_loader_throttle.cc
index 9b5813b900de1e923f49b9160be7825414b7439b..c42b457a462e4ecf3fcb92dc44fa0c934185a487 100644
--- a/chrome/common/google_url_loader_throttle.cc
+++ b/chrome/common/google_url_loader_throttle.cc
@@ -304,8 +304,8 @@ void GoogleURLLoaderThrottle::WillProcessResponse(
     // relying on implicit ordering between this check and the time at which
     // ParsedHeaders is created.
     CHECK(response_head);
-    CHECK(response_head->parsed_headers);
-    if (response_head->parsed_headers->xfo !=
+    //CHECK(response_head->parsed_headers);
+    if (response_head->parsed_headers && response_head->parsed_headers->xfo !=
         network::mojom::XFrameOptionsValue::kDeny) {
       response_head->headers->SetHeader("X-Frame-Options", "SAMEORIGIN");
       response_head->parsed_headers->xfo =
