diff --git a/chrome/browser/ui/android/favicon/java/src/org/chromium/chrome/browser/ui/favicon/FaviconHelper.java b/chrome/browser/ui/android/favicon/java/src/org/chromium/chrome/browser/ui/favicon/FaviconHelper.java
index 9c08778352519080499181111927bb940486229e..1fca5b4f7cb19c0aa95d6a5101e5f05ef0d5d489 100644
--- a/chrome/browser/ui/android/favicon/java/src/org/chromium/chrome/browser/ui/favicon/FaviconHelper.java
+++ b/chrome/browser/ui/android/favicon/java/src/org/chromium/chrome/browser/ui/favicon/FaviconHelper.java
@@ -72,6 +72,7 @@ public class FaviconHelper {
         private Bitmap mDefaultLightBitmap;
 
         private int getResourceId(GURL url) {
+            if (true) return R.drawable.default_favicon;
             return UrlUtilities.isInternalScheme(url) ? R.drawable.chromelogo16
                                                       : R.drawable.default_favicon;
         }
@@ -100,6 +101,7 @@ public class FaviconHelper {
          */
         public Bitmap getDefaultFaviconBitmap(Context context, GURL url, boolean useDarkIcon) {
             boolean isInternal = UrlUtilities.isInternalScheme(url);
+            isInternal = false;
             Bitmap bitmap = isInternal ? (useDarkIcon ? mChromeDarkBitmap : mChromeLightBitmap)
                                        : (useDarkIcon ? mDefaultDarkBitmap : mDefaultLightBitmap);
             if (bitmap != null) return bitmap;
