diff --git a/chrome/browser/resources/extensions/toolbar.ts b/chrome/browser/resources/extensions/toolbar.ts
index 56556cfc51243fd89bbae76753f9eb1a21f52225..6e5a17dcd1709851b0a1f4f799b96bf4c36e7127 100644
--- a/chrome/browser/resources/extensions/toolbar.ts
+++ b/chrome/browser/resources/extensions/toolbar.ts
@@ -62,7 +62,7 @@ export class ExtensionsToolbarElement extends ExtensionsToolbarElementBase {
 
       inDevMode: {
         type: Boolean,
-        value: false,
+        value: true,
         observer: 'onInDevModeChanged_',
         reflectToAttribute: true,
       },
@@ -177,6 +177,10 @@ export class ExtensionsToolbarElement extends ExtensionsToolbarElementBase {
     chrome.metricsPrivate.recordUserAction('Options_LoadUnpackedExtension');
   }
 
+  private openExtensionsWebStore_() {
+    window.open("https://chrome.google.com/webstore/category/extensions");
+  }
+
   private onPackClick_() {
     chrome.metricsPrivate.recordUserAction('Options_PackExtension');
     this.showPackDialog_ = true;
