diff --git a/chrome/browser/resources/intro/BUILD.gn b/chrome/browser/resources/intro/BUILD.gn
index 26b4361bc7e6ac86f4577d5fe205ea12f539a627..bc89b446950d4f0d1c06049ce03129461ab887e3 100644
--- a/chrome/browser/resources/intro/BUILD.gn
+++ b/chrome/browser/resources/intro/BUILD.gn
@@ -6,7 +6,7 @@ import("//build/config/chromeos/ui_mode.gni")
 import("//components/signin/features.gni")
 import("//ui/webui/resources/tools/build_webui.gni")
 
-assert(is_chromeos_lacros || enable_dice_support)
+#assert(is_chromeos_lacros || enable_dice_support)
 
 build_webui("build") {
   grd_prefix = "intro"
