diff --git a/chrome/browser/chrome_content_browser_client_receiver_bindings.cc b/chrome/browser/chrome_content_browser_client_receiver_bindings.cc
index 34485c7703ab8d17f7acc23ba809a4b0c425a3a2..d41db474cec5fa18fe5a6d63412369ea1a8d9382 100644
--- a/chrome/browser/chrome_content_browser_client_receiver_bindings.cc
+++ b/chrome/browser/chrome_content_browser_client_receiver_bindings.cc
@@ -169,7 +169,7 @@ void MaybeCreateSafeBrowsingForRenderer(
       std::move(receiver));
 }
 
-#if BUILDFLAG(ENABLE_EXTENSIONS)
+#if BUILDFLAG(ENABLE_EXTENSIONS) && BUILDFLAG(FULL_SAFE_BROWSING)
 void MaybeCreateExtensionWebRequestReporterForRenderer(
     int process_id,
     mojo::PendingReceiver<safe_browsing::mojom::ExtensionWebRequestReporter>
@@ -238,7 +238,7 @@ void ChromeContentBrowserClient::ExposeInterfacesToRenderer(
                 &ChromeContentBrowserClient::GetSafeBrowsingUrlCheckerDelegate,
                 base::Unretained(this))),
         ui_task_runner);
-#if BUILDFLAG(ENABLE_EXTENSIONS)
+#if BUILDFLAG(ENABLE_EXTENSIONS) && BUILDFLAG(FULL_SAFE_BROWSING)
     registry->AddInterface<safe_browsing::mojom::ExtensionWebRequestReporter>(
         base::BindRepeating(&MaybeCreateExtensionWebRequestReporterForRenderer,
                             render_process_host->GetID()),
