diff --git a/chrome/browser/ui/web_applications/BUILD.gn b/chrome/browser/ui/web_applications/BUILD.gn
index 5d2724f8f3fdfe80c904a6411f0287a6ab284e59..bf6a7e1c9238735f149dc1b1a5a5795fd850ce67 100644
--- a/chrome/browser/ui/web_applications/BUILD.gn
+++ b/chrome/browser/ui/web_applications/BUILD.gn
@@ -91,6 +91,10 @@ source_set("browser_tests") {
       "//chrome/browser/ui/ash/system_web_apps",
     ]
   }
+
+  if (is_android) {
+    deps -= ["//chrome/test:test_support_ui"]
+  }
 }
 
 # On Lacros, these browser tests require Ash to be running.
@@ -128,4 +132,7 @@ source_set("app_service_browser_tests") {
       "//chromeos/lacros:test_support",
     ]
   }
+  if (is_android) {
+    deps -= ["//chrome/test:test_support_ui"]
+  }
 }
