diff --git a/chrome/browser/resources/about_sys/BUILD.gn b/chrome/browser/resources/about_sys/BUILD.gn
index db626ac98ead4f7cb53b7d11f9f2106852bed90e..3ba9955bda1538e8bc1df10ea18f7e6e4eb39cf4 100644
--- a/chrome/browser/resources/about_sys/BUILD.gn
+++ b/chrome/browser/resources/about_sys/BUILD.gn
@@ -4,7 +4,7 @@
 
 import("//ui/webui/resources/tools/build_webui.gni")
 
-assert(!is_android)
+#assert(!is_android)
 
 build_webui("build") {
   grd_prefix = "about_sys"
