diff --git a/chrome/android/java/src/org/chromium/chrome/browser/feedback/HelpAndFeedbackLauncherImpl.java b/chrome/android/java/src/org/chromium/chrome/browser/feedback/HelpAndFeedbackLauncherImpl.java
index 223815ed97a000ff44b83802036ae10cbbc0e00f..f2b345c3feea3356caec0e7ae6fa26d0f9efd03c 100644
--- a/chrome/android/java/src/org/chromium/chrome/browser/feedback/HelpAndFeedbackLauncherImpl.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/feedback/HelpAndFeedbackLauncherImpl.java
@@ -33,7 +33,7 @@ import javax.annotation.Nonnull;
  */
 public class HelpAndFeedbackLauncherImpl implements HelpAndFeedbackLauncher {
     protected static final String FALLBACK_SUPPORT_URL =
-            "https://support.google.com/chrome/topic/6069782";
+            "https://www.mises.site/guide";
     private static final String TAG = "HelpAndFeedback";
 
     private static HelpAndFeedbackLauncher sInstance;
