diff --git a/chrome/android/features/tab_ui/java/src/org/chromium/chrome/browser/tasks/tab_management/TabGridDialogMediator.java b/chrome/android/features/tab_ui/java/src/org/chromium/chrome/browser/tasks/tab_management/TabGridDialogMediator.java
index ebea916076ac9a4ea82a101d1f830f48929638ed..8ca650652d4b869f0ba8db501366653060151259 100644
--- a/chrome/android/features/tab_ui/java/src/org/chromium/chrome/browser/tasks/tab_management/TabGridDialogMediator.java
+++ b/chrome/android/features/tab_ui/java/src/org/chromium/chrome/browser/tasks/tab_management/TabGridDialogMediator.java
@@ -57,6 +57,8 @@ import java.util.ArrayList;
 import java.util.List;
 import java.util.Locale;
 
+import org.chromium.chrome.browser.homepage.HomepageManager;
+
 /**
  * A mediator for the TabGridDialog component, responsible for communicating
  * with the components' coordinator as well as managing the business logic
@@ -634,8 +636,12 @@ public class TabGridDialogMediator
             Tab currentTab = mTabModelSelector.getTabById(mCurrentTabId);
             hideDialog(false);
             if (currentTab == null) {
-                mTabCreatorManager.getTabCreator(mTabModelSelector.isIncognitoSelected())
-                        .launchNTP();
+                if (mTabModelSelector.isIncognitoSelected())
+                    mTabCreatorManager.getTabCreator(mTabModelSelector.isIncognitoSelected())
+                            .launchNTP();
+                else
+                    mTabCreatorManager.getTabCreator(mTabModelSelector.isIncognitoSelected())
+                            .launchUrl(HomepageManager.getInstance().getHomepageUriIgnoringEnabledState(), TabLaunchType.FROM_CHROME_UI);
                 return;
             }
             List<Tab> relatedTabs = getRelatedTabs(currentTab.getId());
