diff --git a/chrome/android/java/src/org/chromium/chrome/browser/app/bookmarks/BookmarkActivity.java b/chrome/android/java/src/org/chromium/chrome/browser/app/bookmarks/BookmarkActivity.java
index f20de9eceb32ac65c94d390a3a4b0e480dafddc8..4d0bb02d5f9b6a04d3482a41425e7647ff9d2d65 100644
--- a/chrome/android/java/src/org/chromium/chrome/browser/app/bookmarks/BookmarkActivity.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/app/bookmarks/BookmarkActivity.java
@@ -22,6 +22,8 @@ import org.chromium.chrome.browser.profiles.ProfileProvider;
 import org.chromium.components.bookmarks.BookmarkId;
 import org.chromium.components.embedder_support.util.UrlConstants;
 
+import org.chromium.ui.base.IntentRequestTracker;
+
 /**
  * The activity that displays the bookmark UI on the phone. It keeps a {@link
  * BookmarkManagerCoordinator} inside of it and creates a snackbar manager. This activity should
@@ -74,7 +76,9 @@ public class BookmarkActivity extends SnackbarActivity {
                     BookmarkId.getBookmarkIdFromString(
                             data.getStringExtra(INTENT_VISIT_BOOKMARK_ID));
             mBookmarkManagerCoordinator.openBookmark(bookmarkId);
-        }
+        } else {
+	  mBookmarkManagerCoordinator.intentTracker().onActivityResult(requestCode, resultCode, data);
+	}
     }
 
     /**
