diff --git a/chrome/browser/browser_controls/android/java/src/org/chromium/chrome/browser/browser_controls/BrowserControlsUtils.java b/chrome/browser/browser_controls/android/java/src/org/chromium/chrome/browser/browser_controls/BrowserControlsUtils.java
index 67f713284931d43f01bd2798f93c45d2694824c4..e4dcf44f5ef39eb71626fe328d145c16ce86688f 100644
--- a/chrome/browser/browser_controls/android/java/src/org/chromium/chrome/browser/browser_controls/BrowserControlsUtils.java
+++ b/chrome/browser/browser_controls/android/java/src/org/chromium/chrome/browser/browser_controls/BrowserControlsUtils.java
@@ -3,6 +3,7 @@
 // found in the LICENSE file.
 
 package org.chromium.chrome.browser.browser_controls;
+import org.chromium.base.ContextUtils;
 
 /** Static utilities related to browser controls interfaces. */
 public class BrowserControlsUtils {
@@ -25,6 +26,9 @@ public class BrowserControlsUtils {
      * @return Whether the browser controls should be drawn as a texture.
      */
     public static boolean drawControlsAsTexture(BrowserControlsStateProvider stateProvider) {
+        if (ContextUtils.getAppSharedPreferences().getBoolean("enable_bottom_toolbar", false) ) {
+            return false;
+        }
         return stateProvider.getBrowserControlHiddenRatio() > 0;
     }
 
