diff --git a/chrome/renderer/BUILD.gn b/chrome/renderer/BUILD.gn
index 3a4b21b9b90ab77b3a71d2905febac21eb54ecd1..ab9a3d84d127debae3f27c49dbd98a6d78c55e6f 100644
--- a/chrome/renderer/BUILD.gn
+++ b/chrome/renderer/BUILD.gn
@@ -381,7 +381,8 @@ static_library("renderer") {
       "//chrome/common/wallet:mojo_bindings",
       "//components/feed/content/renderer:feed_renderer",
     ]
-  } else {
+  }
+  if (!is_android || is_android) {
     sources += [
       "accessibility/ax_tree_distiller.cc",
       "accessibility/ax_tree_distiller.h",
@@ -402,6 +403,12 @@ static_library("renderer") {
       "searchbox/searchbox_extension.cc",
       "searchbox/searchbox_extension.h",
     ]
+    if (is_android) {
+      sources -= [
+        "media/chrome_speech_recognition_client.cc",
+        "media/chrome_speech_recognition_client.h",
+      ]
+    }
 
     deps += [
       "//chrome/common/accessibility:mojo_bindings",
@@ -470,4 +477,5 @@ static_library("renderer") {
       ]
     }
   }
+  import("//mises/renderer/sources.gni") sources += mises_chrome_renderer_sources deps += mises_chrome_renderer_deps public_deps += mises_chrome_renderer_public_deps deps -= mises_chrome_renderer_exception_deps
 }
