diff --git a/chrome/browser/new_tab_page/modules/photos/BUILD.gn b/chrome/browser/new_tab_page/modules/photos/BUILD.gn
index c2dfa316e15b2301ee730f62bafc77ecfceba796..d08f8cdc2fd5248a5baecfcb2815464cbd0810f6 100644
--- a/chrome/browser/new_tab_page/modules/photos/BUILD.gn
+++ b/chrome/browser/new_tab_page/modules/photos/BUILD.gn
@@ -4,7 +4,7 @@
 
 import("//mojo/public/tools/bindings/mojom.gni")
 
-assert(!is_android)
+#assert(!is_android)
 
 mojom("mojo_bindings") {
   sources = [ "photos.mojom" ]
