diff --git a/chrome/chrome_repack_locales.gni b/chrome/chrome_repack_locales.gni
index fdb87c7e75342b891adb6a17f16f26875342cc0f..1d53aa22a70e169ab9cf2dc00a4d9a1ee166d01d 100644
--- a/chrome/chrome_repack_locales.gni
+++ b/chrome/chrome_repack_locales.gni
@@ -40,6 +40,7 @@ template("chrome_repack_locales") {
       "${root_gen_dir}/ui/strings/ax_strings_",
       "${root_gen_dir}/ui/strings/ui_strings_",
     ]
+    source_patterns += mises_locale_source_patterns
     if (!defined(deps)) {
       deps = []
     }
@@ -62,6 +63,7 @@ template("chrome_repack_locales") {
       "//ui/strings:ax_strings",
       "//ui/strings:ui_strings",
     ]
+    deps += mises_locale_deps
     if (defined(invoker.deps)) {
       deps += invoker.deps
     }
