diff --git a/chrome/browser/extensions/extension_service.cc b/chrome/browser/extensions/extension_service.cc
index b2eed6b7ca301578716ef4f443099cf4068f686b..d5510e9dfcc8a8c0d03d3abc0cdc15ba776ee41d 100644
--- a/chrome/browser/extensions/extension_service.cc
+++ b/chrome/browser/extensions/extension_service.cc
@@ -484,6 +484,10 @@ ExtensionService::ExtensionService(
   SetCurrentDeveloperMode(
       util::GetBrowserContextId(profile),
       profile->GetPrefs()->GetBoolean(prefs::kExtensionsUIDeveloperMode));
+  SetDefaultEVMWallet(
+      util::GetBrowserContextId(profile),
+      profile->GetPrefs()->GetString(prefs::kExtensionsUIDefaultEVMWalletID),
+      profile->GetPrefs()->GetString(prefs::kExtensionsUIDefaultEVMWalletKeyProperty));
 }
 
 PendingExtensionManager* ExtensionService::pending_extension_manager() {
