diff --git a/chrome/browser/resources/app_settings/BUILD.gn b/chrome/browser/resources/app_settings/BUILD.gn
index b31ce66ee7abbcde190f1f89e0eb05a911b0cb64..5a432fcc1d5210acecd34309ec5aa49310f6956d 100644
--- a/chrome/browser/resources/app_settings/BUILD.gn
+++ b/chrome/browser/resources/app_settings/BUILD.gn
@@ -5,7 +5,7 @@
 import("//chrome/common/features.gni")
 import("//ui/webui/resources/tools/build_webui.gni")
 
-assert(is_win || is_mac || is_linux || is_fuchsia)
+assert(true || is_win || is_mac || is_linux || is_fuchsia)
 
 build_webui("build") {
   grd_prefix = "app_settings"
