diff --git a/chrome/browser/ui/views/extensions/extensions_toolbar_container.cc b/chrome/browser/ui/views/extensions/extensions_toolbar_container.cc
index 8dd4464a47b6d96f4620846ded0f5b1957e7bf63..4e0351b68f4b524bdfd0e87ba902a694ade3c2fe 100644
--- a/chrome/browser/ui/views/extensions/extensions_toolbar_container.cc
+++ b/chrome/browser/ui/views/extensions/extensions_toolbar_container.cc
@@ -57,6 +57,9 @@ base::OnceClosure& GetOnVisibleCallbackForTesting() {
 // overlapping with the extension popup. The media picker dialog has been
 // identified to be susceptible. See crbug.com/1300006.
 bool HasPossiblyOverlappingSecurityUI(Browser* browser) {
+#if BUILDFLAG(IS_ANDROID)
+  return false;
+#else
   views::ElementTrackerViews::ViewList media_picker_dialogs =
       views::ElementTrackerViews::GetInstance()->GetAllMatchingViews(
           DesktopMediaPickerDialogView::kDesktopMediaPickerDialogViewIdentifier,
@@ -67,6 +70,7 @@ bool HasPossiblyOverlappingSecurityUI(Browser* browser) {
                        views::Widget* dialog_widget = dialog_view->GetWidget();
                        return dialog_widget && dialog_widget->IsVisible();
                      });
+#endif
 }
 
 }  // namespace
