diff --git a/chrome/test/BUILD.gn b/chrome/test/BUILD.gn
index c38205909ab5e41f299e17fef3388f166b9fda61..eb7b584da204c10faec32b703081420c69423157 100644
--- a/chrome/test/BUILD.gn
+++ b/chrome/test/BUILD.gn
@@ -317,6 +317,7 @@ static_library("test_support") {
     "//ui/gl",
   ]
 
+  import("//mises/chromium_src/chrome/test/sources.gni") deps += mises_chromium_src_chrome_test_test_support_deps sources += mises_chromium_src_chrome_test_test_support_sources
   data_deps = []
 
   if (enable_pdf) {
