diff --git a/chrome/android/java/src/org/chromium/chrome/browser/gesturenav/NavigationHandler.java b/chrome/android/java/src/org/chromium/chrome/browser/gesturenav/NavigationHandler.java
index 46405c1d3a271c03c160148d58f8317a34306503..d83a60b60535d6858e906922e9733b3536c643a4 100644
--- a/chrome/android/java/src/org/chromium/chrome/browser/gesturenav/NavigationHandler.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/gesturenav/NavigationHandler.java
@@ -43,7 +43,7 @@ class NavigationHandler implements TouchEventObserver {
 
     // Weighted value to determine when to trigger an edge swipe. Initial scroll
     // vector should form 30 deg or below to initiate swipe action.
-    private static final float WEIGTHED_TRIGGER_THRESHOLD = 1.73f;
+    private static final float WEIGTHED_TRIGGER_THRESHOLD = 0.43f;
 
     // |EDGE_WIDTH_DP| in physical pixel.
     private final float mEdgeWidthPx;
