diff --git a/chrome/browser/ui/webui/access_code_cast/BUILD.gn b/chrome/browser/ui/webui/access_code_cast/BUILD.gn
index 3ecb6afc70df6d0bb69ec6c4fba2cfd0d34cc7fc..76c086263879118e2a871c7a61c227aa40f394e5 100644
--- a/chrome/browser/ui/webui/access_code_cast/BUILD.gn
+++ b/chrome/browser/ui/webui/access_code_cast/BUILD.gn
@@ -5,7 +5,7 @@
 import("//mojo/public/tools/bindings/mojom.gni")
 import("//third_party/protobuf/proto_library.gni")
 
-assert(!is_android, "AccessCodeCast is not for android.")
+#assert(!is_android, "AccessCodeCast is not for android.")
 
 mojom("mojo_bindings") {
   sources = [ "access_code_cast.mojom" ]
