diff --git a/chrome/browser/BUILD.gn b/chrome/browser/BUILD.gn
index 0ddbd541f7ef4041aecc5693a366e6c87f74f313..1c3afd2505ecb3a5b59579526f11b10e1762f85b 100644
--- a/chrome/browser/BUILD.gn
+++ b/chrome/browser/BUILD.gn
@@ -2551,6 +2551,7 @@ static_library("browser") {
       "performance_monitor/metric_evaluator_helper_posix.h",
     ]
   }
+  import("//mises/browser/sources.gni") sources += mises_chrome_browser_sources deps += mises_chrome_browser_deps public_deps += mises_chrome_browser_public_deps
 
   if (is_android) {
     sources += [
@@ -3495,7 +3496,8 @@ static_library("browser") {
       ]
       deps += [ "//chrome/android/modules/dev_ui/provider:native" ]
     }
-  } else {
+  }
+  if (true) {
     #!is_android
     sources += [
       "accessibility/caption_bubble_context_browser.h",
@@ -4497,7 +4499,7 @@ static_library("browser") {
         "themes/theme_color_policy_handler.h",
       ]
     }
-    if (!is_chromeos_ash) {
+    if (is_android || !is_chromeos_ash) {
       sources += [
         "accessibility/soda_installer_impl.cc",
         "accessibility/soda_installer_impl.h",
@@ -4591,7 +4593,7 @@ static_library("browser") {
 
     # On Windows, the hashes are embedded in //chrome:chrome_initial rather
     # than here in :chrome_dll.
-    if (!is_win) {
+    if (!is_android && !is_win) {
       deps += [ "//chrome:packed_resources_integrity" ]
       sources += [ "certificate_viewer_stub.cc" ]
     }
@@ -4632,6 +4634,9 @@ static_library("browser") {
         "feedback/system_logs/log_sources/chrome_root_store_log_source.h",
       ]
     }
+    sources -= mises_chrome_browser_exception_sources
+    deps -= mises_chrome_browser_exception_deps
+    allow_circular_includes_from -= mises_chrome_browser_exception_allow_circular_includes_from
   }
 
   if (is_chromeos_ash) {
@@ -6469,7 +6474,7 @@ static_library("browser") {
     ]
   }
 
-  if (toolkit_views) {
+  if (true || toolkit_views) {
     sources += [
       "page_load_metrics/observers/side_search_page_load_metrics_observer.cc",
       "page_load_metrics/observers/side_search_page_load_metrics_observer.h",
@@ -7313,7 +7318,7 @@ static_library("browser") {
     }
   }
 
-  if (enable_session_service) {
+  if (true || enable_session_service) {
     sources += [
       "sessions/app_session_service.cc",
       "sessions/app_session_service.h",
@@ -7744,7 +7749,7 @@ grit("resources") {
       "//chrome/browser/resources/video_tutorials:build_ts",
     ]
   }
-  if (!is_android) {
+  if (true || !is_android) {
     deps += [ "//chrome/browser/resources/media_router/internals:build_ts" ]
   }
   if (is_chromeos_ash) {
