diff --git a/chrome/browser/ui/android/toolbar/java/src/org/chromium/chrome/browser/toolbar/LocationBarModel.java b/chrome/browser/ui/android/toolbar/java/src/org/chromium/chrome/browser/toolbar/LocationBarModel.java
index e6c19a577ea8b9d1ae9676dfd8830ae37db4d1b4..56e8705fa06ba5b8b10a0f5ad4f108d5ccf5abd0 100644
--- a/chrome/browser/ui/android/toolbar/java/src/org/chromium/chrome/browser/toolbar/LocationBarModel.java
+++ b/chrome/browser/ui/android/toolbar/java/src/org/chromium/chrome/browser/toolbar/LocationBarModel.java
@@ -383,6 +383,10 @@ public class LocationBarModel implements ToolbarDataProvider, LocationBarDataPro
                 return buildUrlBarData(originalUrl, isOfflinePage);
             }
 
+            if (url.startsWith("devtools://")) {
+                return buildUrlBarData(gurl, false, mTab.getTitle(), "");
+            }
+
             if (isOfflinePage) {
                 GURL originalUrl = mTab.getOriginalUrl();
                 formattedUrl = UrlUtilities.stripScheme(mUrlFormatter.format(originalUrl));
@@ -593,7 +597,7 @@ public class LocationBarModel implements ToolbarDataProvider, LocationBarDataPro
 
     @Override
     public int getPageClassification(boolean isPrefetch) {
-        if (mNativeLocationBarModelAndroid == 0) return PageClassification.INVALID_SPEC_VALUE;
+        if (mNativeLocationBarModelAndroid == 0) return PageClassification.NTP_VALUE;
 
         return LocationBarModelJni.get()
                 .getPageClassification(mNativeLocationBarModelAndroid, isPrefetch);
@@ -622,6 +626,17 @@ public class LocationBarModel implements ToolbarDataProvider, LocationBarDataPro
             return ConnectionSecurityLevel.NONE;
         }
 
+        final String url = tab.getUrl().getSpec();
+        boolean mises_force_safe = !TextUtils.isEmpty(url) && (
+            url.startsWith("chrome-extension") || 
+            url.startsWith("chrome") || 
+            url.startsWith("devtools") || 
+            url.startsWith("mises") || 
+            url.startsWith("mises-extension"));
+        if (mises_force_safe) {
+            return ConnectionSecurityLevel.NONE;
+        }
+
         @Nullable GURL publisherUrl = TrustedCdn.getPublisherUrl(tab);
 
         if (publisherUrl != null) {
@@ -669,6 +684,10 @@ public class LocationBarModel implements ToolbarDataProvider, LocationBarDataPro
             return R.drawable.omnibox_info;
         }
 
+        if (mNativeLocationBarModelAndroid == 0) {  
+             return R.drawable.omnibox_info;
+        }
+
         // Return early if native initialization hasn't been done yet.
         if ((securityLevel == ConnectionSecurityLevel.NONE
                         || securityLevel == ConnectionSecurityLevel.WARNING)
@@ -682,6 +701,15 @@ public class LocationBarModel implements ToolbarDataProvider, LocationBarDataPro
                                         .shouldShowSearchEngineLogo())
                         || mNtpDelegate.isCurrentlyVisible();
 
+        String currentUrl = getCurrentUrl();
+        if (currentUrl != null && 
+            (currentUrl.startsWith("chrome://") ||
+            currentUrl.startsWith("mises://") ||
+            currentUrl.startsWith("chrome-search://") ||
+            currentUrl.startsWith("mises-search://") )
+        ){
+            skipIconForNeutralState = true;
+        }
         return SecurityStatusIcon.getSecurityIconResource(
                 securityLevel,
                 isSmallDevice,
