diff --git a/chrome/browser/extensions/BUILD.gn b/chrome/browser/extensions/BUILD.gn
index 9a61d5cc1a1c33017cad6fd9e0d4617fc824ecf8..04c00a8022276110888acccce0eae5ffdaa8fef0 100644
--- a/chrome/browser/extensions/BUILD.gn
+++ b/chrome/browser/extensions/BUILD.gn
@@ -953,6 +953,16 @@ static_library("extensions") {
     "//ui/strings",
     "//url",
   ]
+  if (is_android) {
+    sources += [
+      "//mises/browser/extensions/api/mises_private/mises_private_api.cc",
+      "//mises/browser/extensions/api/mises_private/mises_private_api.h",
+      "global_shortcut_listener_android.cc",
+      "global_shortcut_listener_android.h",
+      "system_display/display_info_provider_aura.cc",
+      "system_display/display_info_provider_aura.h",
+    ]
+  }
 
   if (is_linux || is_mac || is_win) {
     sources += [
@@ -1261,7 +1271,7 @@ static_library("extensions") {
     ]
   }
 
-  if (is_linux) {
+  if (is_linux || true) {
     sources +=
         [ "api/image_writer_private/removable_storage_provider_linux.cc" ]
   }
@@ -1362,6 +1372,7 @@ static_library("extensions") {
   } else {
     sources += [ "api/braille_display_private/braille_controller_stub.cc" ]
   }
+  deps += mises_extensions_resources
 }
 
 static_library("test_support") {
