diff --git a/chrome/android/java/src/org/chromium/chrome/browser/customtabs/CustomTabActivity.java b/chrome/android/java/src/org/chromium/chrome/browser/customtabs/CustomTabActivity.java
index 11e08f96b445c8b8c64863a27d47eabfaf854b0d..b9efb2eff4f20badadca02a20761df907860cdb7 100644
--- a/chrome/android/java/src/org/chromium/chrome/browser/customtabs/CustomTabActivity.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/customtabs/CustomTabActivity.java
@@ -267,6 +267,11 @@ public class CustomTabActivity extends BaseCustomTabActivity {
         return super.getPackageName();
     }
 
+
+    @Override
+    public void setLastVisibleItemTitle(String itemTitle) {
+    }
+
     @Override
     public boolean onOptionsItemSelected(int itemId, @Nullable Bundle menuItemData) {
         int menuIndex =
