diff --git a/chrome/android/features/tab_ui/java/src/org/chromium/chrome/browser/tasks/tab_management/TabSwitcherCoordinator.java b/chrome/android/features/tab_ui/java/src/org/chromium/chrome/browser/tasks/tab_management/TabSwitcherCoordinator.java
index 155b68a6d518e11265cfe87f947ce53305af412b..7f332c63dadd1cd0a7bbb8913dc203e3f20a9548 100644
--- a/chrome/android/features/tab_ui/java/src/org/chromium/chrome/browser/tasks/tab_management/TabSwitcherCoordinator.java
+++ b/chrome/android/features/tab_ui/java/src/org/chromium/chrome/browser/tasks/tab_management/TabSwitcherCoordinator.java
@@ -76,6 +76,8 @@ import org.chromium.ui.resources.dynamics.DynamicResourceLoader;
 import java.util.ArrayList;
 import java.util.List;
 
+import org.chromium.base.ContextUtils;
+
 /**
  * Parent coordinator that is responsible for showing a grid or carousel of tabs for the main
  * TabSwitcher UI.
@@ -169,6 +171,11 @@ public class TabSwitcherCoordinator
             @Nullable OneshotSupplier<LayoutStateProvider> layoutStateProviderSupplier) {
         try (TraceEvent e = TraceEvent.scoped("TabSwitcherCoordinator.constructor")) {
             mActivity = activity;
+	    final String active_tabswitcher = ContextUtils.getAppSharedPreferences().getString("active_tabswitcher", "default");
+            if (active_tabswitcher.equals("list"))
+              mode = TabListCoordinator.TabListMode.LIST;
+            if (active_tabswitcher.equals("default") || active_tabswitcher.equals("grid"))
+              mode = TabListCoordinator.TabListMode.GRID;
             mMode = mode;
             mBrowserControlsStateProvider = browserControls;
             mTabModelSelector = tabModelSelector;
