diff --git a/chrome/android/java/src/org/chromium/chrome/browser/about_settings/AboutChromeSettings.java b/chrome/android/java/src/org/chromium/chrome/browser/about_settings/AboutChromeSettings.java
index fe807929ef2ab743d2cfea50bee4bed10054663c..6972133eaba7a479537fc484b10effd4ef722a8d 100644
--- a/chrome/android/java/src/org/chromium/chrome/browser/about_settings/AboutChromeSettings.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/about_settings/AboutChromeSettings.java
@@ -64,7 +64,8 @@ public class AboutChromeSettings extends PreferenceFragmentCompat
      * versions are more useful.
      */
     public static String getApplicationVersion(Context context, String version) {
-        if (VersionInfo.isOfficialBuild()) {
+        boolean mises_enforce = false;
+	if (mises_enforce && VersionInfo.isOfficialBuild()) {
             return version;
         }
 
@@ -78,7 +79,7 @@ public class AboutChromeSettings extends PreferenceFragmentCompat
         CharSequence updateTimeString =
                 DateUtils.getRelativeTimeSpanString(
                         info.lastUpdateTime, System.currentTimeMillis(), 0);
-        return context.getString(R.string.version_with_update_time, version, updateTimeString);
+        return context.getString(R.string.version_with_update_time, info.versionName, updateTimeString);
     }
 
     @Override
