diff --git a/chrome/browser/ui/webui/side_panel/reading_list/BUILD.gn b/chrome/browser/ui/webui/side_panel/reading_list/BUILD.gn
index 23d480332b082daded175dd706306725e7131f5c..ab419b7dd2ab657c5e4620abb1967618bdb1c647 100644
--- a/chrome/browser/ui/webui/side_panel/reading_list/BUILD.gn
+++ b/chrome/browser/ui/webui/side_panel/reading_list/BUILD.gn
@@ -4,7 +4,7 @@
 
 import("//mojo/public/tools/bindings/mojom.gni")
 
-assert(!is_android)
+#assert(!is_android)
 
 mojom("mojo_bindings") {
   sources = [ "reading_list.mojom" ]
