diff --git a/chrome/browser/resources/tab_search/BUILD.gn b/chrome/browser/resources/tab_search/BUILD.gn
index 8e9a9fd72790b2c117dbbe14dbbf1164fac171ef..ee7c5d6275fde6b4d5e2a2dd106b3c38a6b37913 100644
--- a/chrome/browser/resources/tab_search/BUILD.gn
+++ b/chrome/browser/resources/tab_search/BUILD.gn
@@ -7,7 +7,7 @@ import("//chrome/common/features.gni")
 import("//ui/webui/resources/tools/generate_grd.gni")
 import("//ui/webui/webui_features.gni")
 
-assert(is_linux || is_chromeos || is_win || is_mac || is_fuchsia)
+# assert(is_linux || is_chromeos || is_win || is_mac || is_fuchsia)
 
 preprocess_folder = "preprocessed"
 
