diff --git a/chrome/android/java/src/org/chromium/chrome/browser/toolbar/ToolbarManager.java b/chrome/android/java/src/org/chromium/chrome/browser/toolbar/ToolbarManager.java
index 11fa20c56723ac6a9f265f484c6df258338b2e6d..000f1df54fff6ee4aeae65d4a47e5890926aff5a 100644
--- a/chrome/android/java/src/org/chromium/chrome/browser/toolbar/ToolbarManager.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/toolbar/ToolbarManager.java
@@ -196,6 +196,10 @@ import org.chromium.url.GURL;
 
 import java.util.List;
 
+import org.chromium.chrome.browser.toolbar.top.TabSwitcherActionMenuCoordinator;
+import android.view.View.OnLongClickListener;
+import org.chromium.base.ContextUtils;
+
 /**
  * Contains logic for managing the toolbar visual component. This class manages the interactions
  * with the rest of the application to ensure the toolbar is always visually up to date.
@@ -953,6 +957,7 @@ public class ToolbarManager
 
                     @Override
                     public void onUrlUpdated(Tab tab) {
+                        mLocationBarModel.notifySecurityStateChanged();
                         // Update the SSL security state as a result of this notification as it will
                         // sometimes be the only update we receive.
                         updateTabLoadingState(true);
