diff --git a/chrome/browser/prefs/browser_prefs.cc b/chrome/browser/prefs/browser_prefs.cc
index 921f99a4ee22d230e0a035d15c8800d15ef8223f..7cdbb3583067635d288e594017dd80923b23f637 100644
--- a/chrome/browser/prefs/browser_prefs.cc
+++ b/chrome/browser/prefs/browser_prefs.cc
@@ -1825,6 +1825,8 @@ void RegisterLocalState(PrefRegistrySimple* registry) {
   DeviceOAuth2TokenStoreDesktop::RegisterPrefs(registry);
 #endif
 
+  MisesRegisterLocalStatePrefs(registry);
+
 #if !BUILDFLAG(IS_ANDROID)
   screen_ai::RegisterLocalStatePrefs(registry);
 #endif  // !BUILDFLAG(IS_ANDROID)
@@ -2081,6 +2083,7 @@ void RegisterProfilePrefs(user_prefs::PrefRegistrySyncable* registry,
   captions::LiveCaptionController::RegisterProfilePrefs(registry);
 #endif  // !BUILDFLAG(IS_CHROMEOS_LACROS)
 #endif  // BUILDFLAG(IS_ANDROID)
+  MisesRegisterProfilePrefs(registry);
 
 #if BUILDFLAG(IS_CHROMEOS)
   extensions::DocumentScanAPIHandler::RegisterProfilePrefs(registry);
