diff --git a/chrome/browser/resources/side_panel/history_clusters/BUILD.gn b/chrome/browser/resources/side_panel/history_clusters/BUILD.gn
index dc654a49f91b4baccb2b4cbc32125bfcd6dcfaa6..ae2872d03b90e891d1341daf0d0852a94893cb49 100644
--- a/chrome/browser/resources/side_panel/history_clusters/BUILD.gn
+++ b/chrome/browser/resources/side_panel/history_clusters/BUILD.gn
@@ -4,7 +4,7 @@
 
 import("//ui/webui/resources/tools/build_webui.gni")
 
-assert(!is_android)
+#assert(!is_android)
 
 build_webui("build") {
   grd_prefix = "side_panel_history_clusters"
