diff --git a/chrome/browser/ui/browser.h b/chrome/browser/ui/browser.h
index 10105dd979aa4718ea06088961c8e47272feb658..9400433b47aa1a1c8b19d2fd1b608154d16f235e 100644
--- a/chrome/browser/ui/browser.h
+++ b/chrome/browser/ui/browser.h
@@ -48,7 +48,7 @@
 #include "ui/gfx/geometry/rect.h"
 #include "ui/shell_dialogs/select_file_dialog.h"
 
-#if BUILDFLAG(IS_ANDROID)
+#if false && BUILDFLAG(IS_ANDROID)
 #error This file should only be included on desktop.
 #endif
 
