diff --git a/chrome/android/java/src/org/chromium/chrome/browser/TabbedModeTabDelegateFactory.java b/chrome/android/java/src/org/chromium/chrome/browser/TabbedModeTabDelegateFactory.java
index 46793aad8a33caa5a8e128f907c6a27cdcf3eed6..9da2f1fd01edb4a973c4b55fd21e49750b013814 100644
--- a/chrome/android/java/src/org/chromium/chrome/browser/TabbedModeTabDelegateFactory.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/TabbedModeTabDelegateFactory.java
@@ -156,7 +156,7 @@ public class TabbedModeTabDelegateFactory implements TabDelegateFactory {
 
     @Override
     public ExternalNavigationHandler createExternalNavigationHandler(Tab tab) {
-        return new ExternalNavigationHandler(new ExternalNavigationDelegateImpl(tab));
+        return new org.chromium.chrome.browser.externalnav.MisesExternalNavigationHandler(new ExternalNavigationDelegateImpl(tab));
     }
 
     @Override
@@ -213,4 +213,8 @@ public class TabbedModeTabDelegateFactory implements TabDelegateFactory {
     public void destroy() {
         if (mNativePageFactory != null) mNativePageFactory.destroy();
     }
+
+    public EphemeralTabCoordinator getEphemeralTabCoordinator() {
+        return mEphemeralTabCoordinatorSupplier.get();
+    }
 }
