diff --git a/chrome/android/java/src/org/chromium/chrome/browser/suggestions/tile/MostVisitedTilesCoordinator.java b/chrome/android/java/src/org/chromium/chrome/browser/suggestions/tile/MostVisitedTilesCoordinator.java
index 16a500b6d26712bb0f7e3b8a2d8399b8e5b5f9ef..083285f58f4a3961ec1a14ac16dae660f55eea45 100644
--- a/chrome/android/java/src/org/chromium/chrome/browser/suggestions/tile/MostVisitedTilesCoordinator.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/suggestions/tile/MostVisitedTilesCoordinator.java
@@ -41,7 +41,7 @@ public class MostVisitedTilesCoordinator implements ConfigurationChangedObserver
      * enough space to fit all of them.
      */
     @VisibleForTesting
-    public static final int MAX_TILE_COLUMNS_FOR_GRID = 4;
+    public static final int MAX_TILE_COLUMNS_FOR_GRID = 6;
 
     private final Activity mActivity;
     private final ActivityLifecycleDispatcher mActivityLifecycleDispatcher;
