diff --git a/chrome/browser/ui/android/toolbar/BUILD.gn b/chrome/browser/ui/android/toolbar/BUILD.gn
index 9a3ee5650101af465f6479c3cbb17830e5b3525c..01b81f75ed80ef513806436fb29fdb5bfe8a2c1d 100644
--- a/chrome/browser/ui/android/toolbar/BUILD.gn
+++ b/chrome/browser/ui/android/toolbar/BUILD.gn
@@ -188,6 +188,15 @@ android_library("java") {
 
   annotation_processor_deps = [ "//base/android/jni_generator:jni_processor" ]
   resources_package = "org.chromium.chrome.browser.toolbar"
+  sources += [
+      "//mises/android/java/org/chromium/chrome/browser/toolbar/top/HandButton.java",
+  ]
+  deps += [
+    "//third_party/androidx:androidx_fragment_fragment_java",
+    "//mises/android/features/mises:public_java",
+    "//mises/android/features/mises:java_resources",
+    "//mises/third_party/android_deps:com_github_bumptech_glide_java",
+  ]
 }
 
 generate_jni("jni_headers") {
@@ -283,7 +292,7 @@ android_resources("java_resources") {
     "java/res/values/values.xml",
     "java/res/xml/adaptive_toolbar_preference.xml",
   ]
-
+  import("//mises/browser/ui/android/toolbar/sources.gni") sources += mises_browser_ui_android_toolbar_java_resources_sources
   deps = [
     "//chrome/browser/ui/android/strings:ui_strings_grd",
     "//chrome/browser/ui/android/theme:java_resources",
