diff --git a/chrome/browser/feed/android/feed_service_bridge.cc b/chrome/browser/feed/android/feed_service_bridge.cc
index 715ce66e444eb76d85b042ccef7918ee63984835..df8d1dd82f037a547015d99cb1c790a2415fab4b 100644
--- a/chrome/browser/feed/android/feed_service_bridge.cc
+++ b/chrome/browser/feed/android/feed_service_bridge.cc
@@ -177,6 +177,8 @@ uint64_t FeedServiceBridge::GetReliabilityLoggingId() {
 
 // static
 bool FeedServiceBridge::IsSignedIn() {
+  if (!GetFeedService())
+    return false;
   return GetFeedService()->IsSignedIn();
 }
 
