diff --git a/chrome/browser/resources/new_tab_page/modules/cart/icons/BUILD.gn b/chrome/browser/resources/new_tab_page/modules/cart/icons/BUILD.gn
index 78c9b0d25d3414ccabdbeab882b64c83ae7f61ed..8cb1fcbefd4fc93243584c22be72e5ca588d0602 100644
--- a/chrome/browser/resources/new_tab_page/modules/cart/icons/BUILD.gn
+++ b/chrome/browser/resources/new_tab_page/modules/cart/icons/BUILD.gn
@@ -4,7 +4,7 @@
 
 import("//ui/webui/resources/tools/generate_grd.gni")
 
-assert(!is_android)
+# assert(!is_android)
 
 generate_grd("build_grdp") {
   grd_prefix = "cart"
