diff --git a/chrome/browser/ui/android/toolbar/BUILD.gn b/chrome/browser/ui/android/toolbar/BUILD.gn
index 1f40d76224d521b64894ec51167ee6d2f9eea30e..de3180ca10da2fa0cd54299417873b83d6c4448d 100644
--- a/chrome/browser/ui/android/toolbar/BUILD.gn
+++ b/chrome/browser/ui/android/toolbar/BUILD.gn
@@ -107,6 +107,7 @@ android_library("java") {
     "java/src/org/chromium/chrome/browser/toolbar/top/tab_strip/HeightTransitionHandler.java",
     "java/src/org/chromium/chrome/browser/toolbar/top/tab_strip/TabStripTransitionCoordinator.java",
   ]
+  sources += mises_browser_ui_android_toolbar_sources
 
   deps = [
     ":java_resources",
@@ -193,6 +194,15 @@ android_library("java") {
   ]
 
   resources_package = "org.chromium.chrome.browser.toolbar"
+  sources += [
+      "//mises/android/java/org/chromium/chrome/browser/toolbar/top/HandButton.java",
+  ]
+  deps += [
+    "//third_party/androidx:androidx_fragment_fragment_java",
+    "//mises/android/features/mises:public_java",
+    "//mises/android/features/mises:java_resources",
+    "//mises/third_party/android_deps:com_github_bumptech_glide_java",
+  ]
 }
 
 generate_jni("jni_headers") {
@@ -289,7 +299,7 @@ android_resources("java_resources") {
     "java/res/xml/adaptive_toolbar_preference.xml",
     "java/res/xml/address_bar_settings.xml",
   ]
-
+  import("//mises/browser/ui/android/toolbar/sources.gni") sources += mises_browser_ui_android_toolbar_java_resources_sources
   deps = [
     "//chrome/browser/ui/android/strings:ui_strings_grd",
     "//chrome/browser/ui/android/theme:java_resources",
