diff --git a/chrome/android/java/src/org/chromium/chrome/browser/gesturenav/SideSlideLayout.java b/chrome/android/java/src/org/chromium/chrome/browser/gesturenav/SideSlideLayout.java
index 52d5498585140beef378f1e3eb39aca26c719c45..43162c85b272e75f23deb95e44915a1535413876 100644
--- a/chrome/android/java/src/org/chromium/chrome/browser/gesturenav/SideSlideLayout.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/gesturenav/SideSlideLayout.java
@@ -62,7 +62,7 @@ public class SideSlideLayout extends ViewGroup {
     private static final int ANIMATE_TO_START_DURATION_MS = 500;
 
     // Minimum number of pull updates necessary to trigger a side nav.
-    private static final int MIN_PULLS_TO_ACTIVATE = 3;
+    private static final int MIN_PULLS_TO_ACTIVATE = 1;
 
     // Time threshold to detect navigation reversal - i.e. user navigating
     // forward after navigating back (or back after forward) within a short
