diff --git a/chrome/android/features/tab_ui/java/src/org/chromium/chrome/browser/tasks/tab_management/TabManagementDelegateImpl.java b/chrome/android/features/tab_ui/java/src/org/chromium/chrome/browser/tasks/tab_management/TabManagementDelegateImpl.java
index 2ff14b0a87a337d569699dbbfed06ad0dd0c7b4d..c3db134c394631aafb6e975b08d45d9633c14e70 100644
--- a/chrome/android/features/tab_ui/java/src/org/chromium/chrome/browser/tasks/tab_management/TabManagementDelegateImpl.java
+++ b/chrome/android/features/tab_ui/java/src/org/chromium/chrome/browser/tasks/tab_management/TabManagementDelegateImpl.java
@@ -38,6 +38,8 @@ import org.chromium.components.browser_ui.widget.scrim.ScrimCoordinator;
 import org.chromium.ui.modaldialog.ModalDialogManager;
 import org.chromium.ui.resources.dynamics.DynamicResourceLoader;
 
+import org.chromium.base.ContextUtils;
+
 /**
  * Impl class that will resolve components for tab management.
  */
@@ -69,13 +71,17 @@ public class TabManagementDelegateImpl implements TabManagementDelegate {
             @NonNull OneshotSupplier<IncognitoReauthController> incognitoReauthControllerSupplier,
             @Nullable BackPressManager backPressManager,
             @Nullable OneshotSupplier<LayoutStateProvider> layoutStateProviderSupplier) {
+        int mode = TabListCoordinator.TabListMode.GRID;
+        if (TabUiFeatureUtilities.shouldUseListMode(activity))
+            mode = TabListCoordinator.TabListMode.LIST;
+        final String active_tabswitcher = ContextUtils.getAppSharedPreferences().getString("active_tabswitcher", "default");
+	       if (active_tabswitcher.equals("default") || active_tabswitcher.equals("grid"))
+	           mode = TabListCoordinator.TabListMode.GRID;
         return new TabSwitcherCoordinator(activity, activityLifecycleDispatcher, tabModelSelector,
                 tabContentManager, browserControlsStateProvider, tabCreatorManager,
                 menuOrKeyboardActionController, containerView, multiWindowModeStateDispatcher,
                 scrimCoordinator,
-                TabUiFeatureUtilities.shouldUseListMode(activity)
-                        ? TabListCoordinator.TabListMode.LIST
-                        : TabListCoordinator.TabListMode.GRID,
+                mode,
                 rootView, dynamicResourceLoaderSupplier, snackbarManager, modalDialogManager,
                 incognitoReauthControllerSupplier, backPressManager, layoutStateProviderSupplier);
     }
