diff --git a/chrome/browser/resources/settings/settings_routes.ts b/chrome/browser/resources/settings/settings_routes.ts
index 0fdf17e243bb5c7ffadec33598a0d22fc0cb2f91..c17d157d96429b761e8befa1f07a5d5aead0d402 100644
--- a/chrome/browser/resources/settings/settings_routes.ts
+++ b/chrome/browser/resources/settings/settings_routes.ts
@@ -89,4 +89,5 @@ export type SettingsRoutes = {
   SYNC_ADVANCED: Route,
   SYSTEM: Route,
   TRIGGERED_RESET_DIALOG: Route,
+  GET_STARTED: Route; SHIELDS: Route; SOCIAL_BLOCKING: Route; REWARDS: Route; EXTENSIONS: Route; BRAVE_SYNC: Route; BRAVE_IPFS: Route; BRAVE_WALLET: Route; BRAVE_HELP_TIPS: Route; BRAVE_NEW_TAB: Route; THEMES: Route;
 };
