diff --git a/chrome/android/java/src/org/chromium/chrome/browser/tab/TabStateBrowserControlsVisibilityDelegate.java b/chrome/android/java/src/org/chromium/chrome/browser/tab/TabStateBrowserControlsVisibilityDelegate.java
index ba6186e6b4e14573476046c335109493482ac868..417c30cbbae4cd78e9fdca40c5692c94438bbf24 100644
--- a/chrome/android/java/src/org/chromium/chrome/browser/tab/TabStateBrowserControlsVisibilityDelegate.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/tab/TabStateBrowserControlsVisibilityDelegate.java
@@ -24,6 +24,8 @@ import org.chromium.content_public.browser.WebContents;
 import org.chromium.ui.base.WindowAndroid;
 import org.chromium.url.GURL;
 
+import org.chromium.base.ContextUtils;
+
 /**
  * Determines the desired visibility of the browser controls based on the current state of a given
  * tab.
@@ -208,6 +210,18 @@ public class TabStateBrowserControlsVisibilityDelegate extends BrowserControlsVi
         enableHidingBrowserControls &= !ChromeAccessibilityUtil.get().isAccessibilityEnabled();
         enableHidingBrowserControls &= DeviceClassManager.enableFullscreen();
 
+        String KeepToolbarSetting = ContextUtils.getAppSharedPreferences().getString("keep_toolbar_visible_configuration", "unknown");
+        if (KeepToolbarSetting.equals("unknown")) {
+          if (ChromeAccessibilityUtil.get().isAccessibilityEnabled())
+            enableHidingBrowserControls &= false;
+          else
+            enableHidingBrowserControls &= true;
+        } else if (KeepToolbarSetting.equals("on")) {
+            enableHidingBrowserControls &= false;
+        } else {
+            enableHidingBrowserControls &= true;
+        }
+
         return enableHidingBrowserControls;
     }
 
