diff --git a/chrome/browser/android/resource_id.h b/chrome/browser/android/resource_id.h
index 9c14836493c6e888baf0f778ec541c237b0c3ce4..653ca4c71a57791f7f89d31961528b1d897a51cf 100644
--- a/chrome/browser/android/resource_id.h
+++ b/chrome/browser/android/resource_id.h
@@ -96,3 +96,5 @@ DECLARE_RESOURCE_ID(IDR_ANDROID_ABOUT_THIS_SITE_LOGO_24DP,
 DECLARE_RESOURCE_ID(IDR_ANDROID_ABOUT_THIS_SITE_LOGO_24DP,
                     R.drawable.ic_info_outline_grey_24dp)
 #endif  // BUILDFLAG(GOOGLE_CHROME_BRANDING) && BUILDFLAG(IS_ANDROID)
+
+DECLARE_RESOURCE_ID(IDR_EXTENSIONS, R.drawable.ic_extensions)
