diff --git a/chrome/browser/ui/android/toolbar/java/src/org/chromium/chrome/browser/toolbar/top/TabSwitcherActionMenuCoordinator.java b/chrome/browser/ui/android/toolbar/java/src/org/chromium/chrome/browser/toolbar/top/TabSwitcherActionMenuCoordinator.java
index a6d85b9a7c12b6a0bf6b7d3b6cf56b8a8b089e0c..13578ec5f35af037efcce94011e1d1fd592ce054 100644
--- a/chrome/browser/ui/android/toolbar/java/src/org/chromium/chrome/browser/toolbar/top/TabSwitcherActionMenuCoordinator.java
+++ b/chrome/browser/ui/android/toolbar/java/src/org/chromium/chrome/browser/toolbar/top/TabSwitcherActionMenuCoordinator.java
@@ -40,6 +40,9 @@ import org.chromium.ui.widget.RectProvider;
 import java.lang.annotation.Retention;
 import java.lang.annotation.RetentionPolicy;
 
+import org.chromium.base.ContextUtils;
+import android.view.View.OnClickListener;
+
 /**
  * The main coordinator for the Tab Switcher Action Menu, responsible for creating the popup menu
  * (popup window) in general and building a list of menu items.
@@ -73,18 +76,29 @@ public class TabSwitcherActionMenuCoordinator {
     public static OnLongClickListener createOnLongClickListener(
             Callback<Integer> onItemClicked,
             Profile profile,
-            ObservableSupplier<TabModelSelector> tabModelSelectorSupplier) {
+            ObservableSupplier<TabModelSelector> tabModelSelectorSupplier, OnClickListener newTabClickHandler) {
         return createOnLongClickListener(
                 new TabSwitcherActionMenuCoordinator(profile, tabModelSelectorSupplier),
                 profile,
-                onItemClicked);
+                onItemClicked, newTabClickHandler);
     }
 
     // internal helper function to create a long click listener.
     protected static OnLongClickListener createOnLongClickListener(
             TabSwitcherActionMenuCoordinator menu,
             Profile profile,
-            Callback<Integer> onItemClicked) {
+            Callback<Integer> onItemClicked, OnClickListener newTabClickHandler) {
+        if (!ContextUtils.getAppSharedPreferences().getBoolean("tabswitcher_opens_contextual_menu", false)) {
+            return new OnLongClickListener() {
+                @Override
+                public boolean onLongClick(View view) {
+                    recordUserActions(R.id.new_tab_menu_id);
+                    newTabClickHandler.onClick(view);
+                    return true;
+                }
+            };
+        }
+
         return (view) -> {
             Context context = view.getContext();
             menu.displayMenu(
