diff --git a/chrome/browser/resources/extensions/service.ts b/chrome/browser/resources/extensions/service.ts
index 1081d384b412d6ec6ea2c8c6f8d72031854b7e62..eb5854e5422dfbb8263205eccbd575c30dafc1b0 100644
--- a/chrome/browser/resources/extensions/service.ts
+++ b/chrome/browser/resources/extensions/service.ts
@@ -254,6 +254,11 @@ export class Service implements ServiceInterface {
         {inDeveloperMode: inDevMode});
   }
 
+  setProfileDefaultEVMWallet(id: string) {
+    chrome.developerPrivate.updateProfileConfiguration(
+        {defaultEVMWallet: id});    
+  }  
+
   loadUnpacked(): Promise<boolean> {
     return this.loadUnpackedHelper_();
   }
