diff --git a/chrome/android/chrome_common_shared_library.gni b/chrome/android/chrome_common_shared_library.gni
index 2a73ab3c8699460c0beb3c00cabf9fa96c0f7735..bc45f870fd8d8072a5c6099424fc8fe3ad00f1ba 100644
--- a/chrome/android/chrome_common_shared_library.gni
+++ b/chrome/android/chrome_common_shared_library.gni
@@ -116,6 +116,10 @@ template("chrome_common_shared_library") {
     }
 
     # Handle VR JNI registration and dependencies.
+    if (_is_webview) {
+      sources = [ "//chrome/browser/android/vr/register_jni_monochrome.cc" ]
+      deps += [ "//chrome/browser/android/vr:module_factory" ]
+    }
     if (!_is_webview && enable_vr) {
       if (_export_java_symbols) {
         # NOTE: While this file is named *_monochrome.cc, it just contains an
