diff --git a/chrome/browser/ui/webui/tab_search/BUILD.gn b/chrome/browser/ui/webui/tab_search/BUILD.gn
index 48539b48e5b5851c179430ce13ed71ddd5fb9ac7..ccee587e714ba29164912142c45f1eb8d0665860 100644
--- a/chrome/browser/ui/webui/tab_search/BUILD.gn
+++ b/chrome/browser/ui/webui/tab_search/BUILD.gn
@@ -4,7 +4,7 @@
 
 import("//mojo/public/tools/bindings/mojom.gni")
 
-assert(is_linux || is_chromeos || is_win || is_mac)
+#assert(is_linux || is_chromeos || is_win || is_mac)
 assert(!is_fuchsia, "Fuchsia shouldn't use anything in //chrome")
 
 mojom("mojo_bindings") {
