diff --git a/chrome/browser/ui/android/toolbar/java/src/org/chromium/chrome/browser/toolbar/TabSwitcherDrawable.java b/chrome/browser/ui/android/toolbar/java/src/org/chromium/chrome/browser/toolbar/TabSwitcherDrawable.java
index 9c2d283ae6946bc7db29a7bde250f10cdd4d5352..582f6984b0100e9089b27cab0a99e7ea598c49ee 100644
--- a/chrome/browser/ui/android/toolbar/java/src/org/chromium/chrome/browser/toolbar/TabSwitcherDrawable.java
+++ b/chrome/browser/ui/android/toolbar/java/src/org/chromium/chrome/browser/toolbar/TabSwitcherDrawable.java
@@ -47,6 +47,7 @@ public class TabSwitcherDrawable extends TintedDrawable {
 
     private final float mSingleDigitTextSize;
     private final float mDoubleDigitTextSize;
+    private final float mTripleDigitTextSize;
 
     private final Rect mTextBounds = new Rect();
     private final Paint mNotificationPaint;
@@ -76,7 +77,7 @@ public class TabSwitcherDrawable extends TintedDrawable {
             @TabSwitcherDrawableLocation int tabSwitcherDrawableLocation) {
         Bitmap icon =
                 BitmapFactory.decodeResource(
-                        context.getResources(), R.drawable.btn_tabswitcher_modern);
+                        context.getResources(), R.drawable.btn_tabswitcher_mises);
         return new TabSwitcherDrawable(
                 context, brandedColorScheme, icon, tabSwitcherDrawableLocation);
     }
@@ -92,6 +93,8 @@ public class TabSwitcherDrawable extends TintedDrawable {
                 context.getResources().getDimension(R.dimen.toolbar_tab_count_text_size_1_digit);
         mDoubleDigitTextSize =
                 context.getResources().getDimension(R.dimen.toolbar_tab_count_text_size_2_digit);
+        mTripleDigitTextSize =
+                context.getResources().getDimension(R.dimen.toolbar_tab_count_text_size_3_digit);
 
         mTextPaint = new TextPaint();
         mTextPaint.setAntiAlias(true);
@@ -190,6 +193,7 @@ public class TabSwitcherDrawable extends TintedDrawable {
         mTabCount = tabCount;
         mIncognito = incognito;
         float textSizePx = mTabCount > 9 ? mDoubleDigitTextSize : mSingleDigitTextSize;
+        textSizePx = mTabCount > 99 ? mTripleDigitTextSize : textSizePx;
         mTextPaint.setTextSize(textSizePx);
         invalidateSelf();
     }
@@ -197,7 +201,7 @@ public class TabSwitcherDrawable extends TintedDrawable {
     private String getTabCountString() {
         if (mTabCount <= 0) {
             return "";
-        } else if (mTabCount > 99) {
+        } else if (mTabCount > 999) {
             return mIncognito ? ";)" : ":D";
         } else {
             return String.format(Locale.getDefault(), "%d", mTabCount);
