diff --git a/chrome/browser/resources/new_tab_page/BUILD.gn b/chrome/browser/resources/new_tab_page/BUILD.gn
index 4f862b1b59d17b42bbd85aae745819f398f4f70c..c6458b60d387d7c7dbd61203d4dbb8b85fae46f2 100644
--- a/chrome/browser/resources/new_tab_page/BUILD.gn
+++ b/chrome/browser/resources/new_tab_page/BUILD.gn
@@ -7,7 +7,7 @@ import("./new_tab_page.gni")
 
 import("//ui/webui/resources/tools/build_webui.gni")
 
-assert(!is_android)
+#assert(!is_android)
 
 grd_prefix = "new_tab_page"
 
