diff --git a/chrome/browser/resources/side_panel/commerce/BUILD.gn b/chrome/browser/resources/side_panel/commerce/BUILD.gn
index e3a03d8ee2b40ccf87b4b31e28e39367e83a6682..306c6f153ac2938248050c67ca3f021c37a5adf6 100644
--- a/chrome/browser/resources/side_panel/commerce/BUILD.gn
+++ b/chrome/browser/resources/side_panel/commerce/BUILD.gn
@@ -4,7 +4,7 @@
 
 import("//ui/webui/resources/tools/build_webui.gni")
 
-assert(!is_android)
+#assert(!is_android)
 
 build_webui("build") {
   grd_prefix = "side_panel_commerce"
