diff --git a/chrome/browser/safe_browsing/BUILD.gn b/chrome/browser/safe_browsing/BUILD.gn
index 75934762c580cd6c3f321fa705db159831737c01..1c11dcb7e062d51c15b71185d299d92c0a9099de 100644
--- a/chrome/browser/safe_browsing/BUILD.gn
+++ b/chrome/browser/safe_browsing/BUILD.gn
@@ -452,6 +452,16 @@ static_library("safe_browsing") {
           "tailored_security/unconsented_message_android.h",
           "telemetry/android/android_telemetry_service.cc",
           "telemetry/android/android_telemetry_service.h",
+          "cloud_content_scanning/deep_scanning_utils.cc",
+          "cloud_content_scanning/deep_scanning_utils.h",
+          "extension_telemetry/declarative_net_request_signal.cc",
+          "extension_telemetry/declarative_net_request_signal.h",
+          "extension_telemetry/tabs_api_signal.cc",
+          "extension_telemetry/tabs_api_signal.h",
+          "extension_telemetry/extension_web_request_reporter_impl.cc",
+          "extension_telemetry/extension_web_request_reporter_impl.h",
+          "extension_telemetry/extension_signal_util.h",
+          "extension_telemetry/extension_signal_util.cc",
         ]
         deps += [
           "//chrome/app/vector_icons:vector_icons",
