diff --git a/chrome/common/BUILD.gn b/chrome/common/BUILD.gn
index e81374ea2d0b8c56e516a118c4e1db10d369bdd1..622a8944861aeee8bfd29e7a7c0a1411c3ec0d38 100644
--- a/chrome/common/BUILD.gn
+++ b/chrome/common/BUILD.gn
@@ -90,6 +90,7 @@ source_set("channel_info") {
   } else if (is_fuchsia) {
     sources += [ "channel_info_fuchsia.cc" ]
   }
+  public_deps += [ "//mises/common:channel_info" ]
 }
 
 source_set("ini_parser") {
@@ -403,7 +404,8 @@ static_library("common_lib") {
       "media/chrome_media_drm_bridge_client.cc",
       "media/chrome_media_drm_bridge_client.h",
     ]
-  } else {
+  }
+  if (true) {
     # Non-Android.
     public_deps += [ "//chrome/common/importer" ]
   }
@@ -535,6 +537,7 @@ static_library("non_code_constants") {
     "//printing/buildflags",
     "//ui/base:buildflags",
   ]
+  configs += [ "//mises/common:constants_configs" ]
 }
 
 component("chrome_features") {
