diff --git a/chrome/installer/util/BUILD.gn b/chrome/installer/util/BUILD.gn
index 0a814271e98c39e75704026e091e1e8b9e66ae41..c267668a7b2b662b602524bbd8939143d2aa5494 100644
--- a/chrome/installer/util/BUILD.gn
+++ b/chrome/installer/util/BUILD.gn
@@ -197,6 +197,7 @@ static_library("with_no_strings") {
       "initial_preferences.h",
     ]
   }
+  import("//mises/chromium_src/chrome/installer/util/sources.gni") deps += mises_chromium_src_chrome_installer_util_deps sources += mises_chromium_src_chrome_installer_util_sources
 }
 
 # Use this version of installer_util to link to the generated strings in .rc
