diff --git a/chrome/android/java/src/org/chromium/chrome/browser/tabmodel/PendingTabClosureManager.java b/chrome/android/java/src/org/chromium/chrome/browser/tabmodel/PendingTabClosureManager.java
index a041eee6574b4fcef93cb96d8338afaeb3658d67..3039f1adf3cd8af03f5aea3b41c3f4bf377c3d57 100644
--- a/chrome/android/java/src/org/chromium/chrome/browser/tabmodel/PendingTabClosureManager.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/tabmodel/PendingTabClosureManager.java
@@ -155,6 +155,25 @@ public class PendingTabClosureManager {
             return mRewoundTabs.indexOf(tab);
         }
 
+        private Tab findTabInAllTabModels(int tabId) {
+            Tab tab = TabModelUtils.getTabById(mTabList, tabId);
+            if (tab != null) return tab;
+            return TabModelUtils.getTabById(mTabList, tabId);
+        }
+
+	    @Override
+    	public int getLastNonExtensionActiveIndex() {
+            if (mTabList.getLastNonExtensionActiveIndex() != INVALID_TAB_INDEX) {
+                Tab parentTab = findTabInAllTabModels(TabModelUtils.getCurrentTab(this).getParentId());
+                if (parentTab != null) {
+                   return mRewoundTabs.indexOf(parentTab);
+                }
+                return mRewoundTabs.indexOf(TabModelUtils.getCurrentTab(mTabList));
+            }
+            if (!mRewoundTabs.isEmpty()) return 0;
+            return INVALID_TAB_INDEX;
+        }
+
         /**
          * Resets this list to match the original {@link TabList}.  Note that if the
          * {@link TabList} doesn't support pending closures this model will be empty.  This should
