diff --git a/chrome/browser/ui/global_error/global_error.cc b/chrome/browser/ui/global_error/global_error.cc
index b6a21a379e37917ec22aefac5c905e4410c0ac22..d781f06292da7fdbfc9fbcf27259e61c6be02efc 100644
--- a/chrome/browser/ui/global_error/global_error.cc
+++ b/chrome/browser/ui/global_error/global_error.cc
@@ -33,9 +33,11 @@ bool GlobalErrorWithStandardBubble::HasShownBubbleView() {
 }
 
 void GlobalErrorWithStandardBubble::ShowBubbleView(Browser* browser) {
+#if !BUILDFLAG(IS_ANDROID)
   has_shown_bubble_view_ = true;
   bubble_view_ =
       GlobalErrorBubbleViewBase::ShowStandardBubbleView(browser, AsWeakPtr());
+#endif
 }
 
 GlobalErrorBubbleViewBase* GlobalErrorWithStandardBubble::GetBubbleView() {
