diff --git a/chrome/chrome_paks.gni b/chrome/chrome_paks.gni
index 54efa2d231a3bb9218a4cd57b8cab1c88dbc30b6..799c6b2c77a3b23063aa3be22e7ecc22c87401b8 100644
--- a/chrome/chrome_paks.gni
+++ b/chrome/chrome_paks.gni
@@ -147,7 +147,7 @@ template("chrome_extra_paks") {
       sources += invoker.additional_paks
     }
 
-    if (!is_android) {
+    if (!is_android || is_android) {
       # New paks should be added here by default.
       sources += [
         "$root_gen_dir/chrome/about_sys_resources.pak",
@@ -379,7 +379,7 @@ template("chrome_extra_paks") {
       sources += [ "$root_gen_dir/chrome/webui_js_error_resources.pak" ]
       deps += [ "//chrome/browser/resources/webui_js_error:resources" ]
     }
-    if (!is_android && !is_chromeos_ash) {
+    if (!is_android && !is_chromeos_ash || is_android) {
       sources += [
         "$root_gen_dir/chrome/intro_resources.pak",
         "$root_gen_dir/chrome/profile_picker_resources.pak",
@@ -417,12 +417,12 @@ template("chrome_extra_paks") {
       deps += [ "//chrome/browser/resources/dlp_internals:resources" ]
     }
 
-    if (is_win || is_mac || is_linux || is_chromeos_lacros || is_fuchsia) {
+    if (is_win || is_mac || is_linux || is_chromeos_lacros || is_fuchsia || is_android) {
       sources += [ "$root_gen_dir/chrome/browser_switch_resources.pak" ]
       deps += [ "//chrome/browser/resources/browser_switch:resources" ]
     }
 
-    if (is_win || is_mac || is_linux || is_chromeos || is_fuchsia) {
+    if (is_win || is_mac || is_linux || is_chromeos || is_fuchsia || is_android) {
       sources += [ "$root_gen_dir/chrome/discards_resources.pak" ]
       deps += [ "//chrome/browser/resources/discards:resources" ]
     }
