diff --git a/chrome/browser/ui/android/night_mode/java/src/org/chromium/chrome/browser/night_mode/WebContentsDarkModeMessageController.java b/chrome/browser/ui/android/night_mode/java/src/org/chromium/chrome/browser/night_mode/WebContentsDarkModeMessageController.java
index 986d1d5b9391b2470f6fed58ac549603d91fbf25..4a2d547e04a2f6be90a9028540450f6291511393 100644
--- a/chrome/browser/ui/android/night_mode/java/src/org/chromium/chrome/browser/night_mode/WebContentsDarkModeMessageController.java
+++ b/chrome/browser/ui/android/night_mode/java/src/org/chromium/chrome/browser/night_mode/WebContentsDarkModeMessageController.java
@@ -62,6 +62,8 @@ public class WebContentsDarkModeMessageController {
      * @return Whether or not the user education message should be shown.
      */
     private static boolean shouldSendMessage(Profile profile, Context context) {
+        if (true)
+            return false;
         // Only send message if the feature is enabled and the message has not yet been shown.
         Tracker tracker = TrackerFactory.getTrackerForProfile(profile);
         boolean featureEnabled = WebContentsDarkModeController.isFeatureEnabled(context, profile);
