diff --git a/chrome/browser/tabmodel/android/java/src/org/chromium/chrome/browser/tabmodel/IncognitoTabModelImpl.java b/chrome/browser/tabmodel/android/java/src/org/chromium/chrome/browser/tabmodel/IncognitoTabModelImpl.java
index d2fe32e01cc930fff226c4e25f877e3a40f72e9d..38dccef6cb405b0390f4aade3b80e49382917ea8 100644
--- a/chrome/browser/tabmodel/android/java/src/org/chromium/chrome/browser/tabmodel/IncognitoTabModelImpl.java
+++ b/chrome/browser/tabmodel/android/java/src/org/chromium/chrome/browser/tabmodel/IncognitoTabModelImpl.java
@@ -184,6 +184,11 @@ class IncognitoTabModelImpl implements IncognitoTabModel {
         mDelegateModel.setIndex(i, type);
     }
 
+    @Override
+    public int getLastNonExtensionActiveIndex() {
+        return mDelegateModel.getLastNonExtensionActiveIndex();
+    }
+
     @Override
     public boolean isActiveModel() {
         return mActive;
