diff --git a/chrome/browser/new_tab_page/modules/recipes/BUILD.gn b/chrome/browser/new_tab_page/modules/recipes/BUILD.gn
index f0b8c910b5cab2e4907680575c5390f0b94ca8d6..4a9e194e1f4a13cf8918b6f88fd26be4882837fe 100644
--- a/chrome/browser/new_tab_page/modules/recipes/BUILD.gn
+++ b/chrome/browser/new_tab_page/modules/recipes/BUILD.gn
@@ -4,7 +4,7 @@
 
 import("//mojo/public/tools/bindings/mojom.gni")
 
-assert(!is_android)
+#assert(!is_android)
 
 mojom("mojo_bindings") {
   sources = [ "recipes.mojom" ]
