diff --git a/chrome/browser/resources/side_panel/companion/BUILD.gn b/chrome/browser/resources/side_panel/companion/BUILD.gn
index faa482f0eee59aa083ccf18a9f845eec6b34121b..870896210fadc9c360a95d1ea8f131c043e9aa88 100644
--- a/chrome/browser/resources/side_panel/companion/BUILD.gn
+++ b/chrome/browser/resources/side_panel/companion/BUILD.gn
@@ -4,7 +4,7 @@
 
 import("//ui/webui/resources/tools/build_webui.gni")
 
-assert(!is_android)
+#assert(!is_android)
 
 build_webui("build") {
   grd_prefix = "side_panel_companion"
