diff --git a/chrome/renderer/extensions/chrome_extensions_dispatcher_delegate.cc b/chrome/renderer/extensions/chrome_extensions_dispatcher_delegate.cc
index a15389567a935ca9fbd2f75aa0726d284d5ea0a8..5cc4c3907a7445f77c495e93ca4c64c508fe28a3 100644
--- a/chrome/renderer/extensions/chrome_extensions_dispatcher_delegate.cc
+++ b/chrome/renderer/extensions/chrome_extensions_dispatcher_delegate.cc
@@ -256,4 +256,6 @@ void ChromeExtensionsDispatcherDelegate::InitializeBindingsSystem(
   bindings->RegisterHooksDelegate(
       "printing", std::make_unique<extensions::PrintingHooksDelegate>());
 #endif
+    bindings->RegisterHooksDelegate(
+      "misesPrivate", std::make_unique<extensions::MisesHooksDelegate>());
 }
