diff --git a/chrome/browser/ui/android/toolbar/java/src/org/chromium/chrome/browser/toolbar/top/OptionalBrowsingModeButtonController.java b/chrome/browser/ui/android/toolbar/java/src/org/chromium/chrome/browser/toolbar/top/OptionalBrowsingModeButtonController.java
index a472a83ad885b14b0badee3bbde6c9c144482d93..5a7a1fcf0348c54b9ab972bac57bf789d74bab24 100644
--- a/chrome/browser/ui/android/toolbar/java/src/org/chromium/chrome/browser/toolbar/top/OptionalBrowsingModeButtonController.java
+++ b/chrome/browser/ui/android/toolbar/java/src/org/chromium/chrome/browser/toolbar/top/OptionalBrowsingModeButtonController.java
@@ -138,6 +138,7 @@ public class OptionalBrowsingModeButtonController {
             mUserEducationHelper.requestShowIPH(
                     buttonData.getButtonSpec().getIPHCommandBuilder().build());
         }
+        hideCurrentOptionalButton();
     }
 
     private void hideCurrentOptionalButton() {
