diff --git a/chrome/android/java/src/org/chromium/chrome/browser/download/DownloadManagerService.java b/chrome/android/java/src/org/chromium/chrome/browser/download/DownloadManagerService.java
index 7f1bc2339c2767d631d97d4a5e59f073dcbfa023..935113f11fecd6c38f13f1337549765373b89abc 100644
--- a/chrome/android/java/src/org/chromium/chrome/browser/download/DownloadManagerService.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/download/DownloadManagerService.java
@@ -473,6 +473,10 @@ public class DownloadManagerService implements DownloadServiceDelegate, ProfileM
                                     isSupportedMimeType);
                             broadcastDownloadSuccessful(info);
                         } else {
+		                     if (info.getMimeType().equalsIgnoreCase("application/x-chrome-extension")){
+		                         mDownloadNotifier.notifyDownloadCanceled(item.getContentId());
+		                         return;
+		                     }
                             info =
                                     DownloadInfo.Builder.fromDownloadInfo(info)
                                             .setFailState(FailState.CANNOT_DOWNLOAD)
