diff --git a/chrome/android/java/src/org/chromium/chrome/browser/tabbed_mode/TabbedNavigationBarColorController.java b/chrome/android/java/src/org/chromium/chrome/browser/tabbed_mode/TabbedNavigationBarColorController.java
index 45e431d1f294ddd76bd75f0c52075a44713560e3..6e8f574d4a13ac0639ec2970bffc581d054b9ad5 100644
--- a/chrome/android/java/src/org/chromium/chrome/browser/tabbed_mode/TabbedNavigationBarColorController.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/tabbed_mode/TabbedNavigationBarColorController.java
@@ -57,6 +57,11 @@ import org.chromium.ui.util.ColorUtils;
 
 import java.util.Optional;
 
+import org.chromium.base.ContextUtils;
+import android.graphics.Color;
+
+import org.chromium.chrome.browser.night_mode.GlobalNightModeStateProviderHolder;
+
 /** Controls the bottom system navigation bar color for the provided {@link Window}. */
 @RequiresApi(Build.VERSION_CODES.O_MR1)
 class TabbedNavigationBarColorController
@@ -354,6 +359,11 @@ class TabbedNavigationBarColorController
         forceDarkNavigation |= mIsInFullscreen;
         mForceDarkNavigationBarColor = forceDarkNavigation;
 
+        if (ContextUtils.getAppSharedPreferences().getBoolean("darken_websites_enabled", false) || ContextUtils.getAppSharedPreferences().getInt("ui_theme_setting", 0) == 2 || (ContextUtils.getAppSharedPreferences().getInt("ui_theme_setting", 0) == 0 && GlobalNightModeStateProviderHolder.getInstance().isInNightMode()))
+            forceDarkNavigation = true;
+
+        if (mForceDarkNavigationBarColor == forceDarkNavigation) return;
+
         // 2. Calculate colors and store update states.
         final @ColorInt int newNavigationBarColor = getNavigationBarColor(forceDarkNavigation);
         final @ColorInt int dividerColor =
