diff --git a/chrome/android/java/src/org/chromium/chrome/browser/init/AsyncInitializationActivity.java b/chrome/android/java/src/org/chromium/chrome/browser/init/AsyncInitializationActivity.java
index 3c12c5a5f064f8a373d380a4b1c6e7ab4865a6aa..81084001ed53b17bfe83bd6a2998ffc093957d83 100644
--- a/chrome/android/java/src/org/chromium/chrome/browser/init/AsyncInitializationActivity.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/init/AsyncInitializationActivity.java
@@ -459,7 +459,7 @@ public abstract class AsyncInitializationActivity extends ChromeBaseAppCompatAct
      * @return Whether or not the user needs to go through First Run before using this Activity.
      */
     protected boolean requiresFirstRunToBeCompleted(Intent intent) {
-        return true;
+        return false;
     }
 
     /**
