diff --git a/chrome/android/java/src/org/chromium/chrome/browser/tab/TabUtils.java b/chrome/android/java/src/org/chromium/chrome/browser/tab/TabUtils.java
index cd44b4be3ed4f9edc07581506779c7aa5a72d9e8..6071fdfdabae812a5b560ea0d2e47cf784d832ef 100644
--- a/chrome/android/java/src/org/chromium/chrome/browser/tab/TabUtils.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/tab/TabUtils.java
@@ -44,6 +44,8 @@ import org.chromium.url.GURL;
 import java.lang.annotation.Retention;
 import java.lang.annotation.RetentionPolicy;
 
+import org.chromium.base.ContextUtils;
+
 /** Collection of utility methods that operates on Tab. */
 public class TabUtils {
     @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
@@ -160,6 +162,8 @@ public class TabUtils {
                 tabUserAgent = TabUserAgent.DESKTOP;
             } else {
                 tabUserAgent = TabUserAgent.DEFAULT;
+                if (ContextUtils.getAppSharedPreferences().getBoolean("desktop_mode", false))
+                     tabUserAgent = TabUserAgent.DESKTOP;
             }
             tab.setUserAgent(tabUserAgent);
         }
@@ -200,6 +204,8 @@ public class TabUtils {
      * @return Whether the desktop site should be requested.
      */
     public static boolean isDesktopSiteGlobalEnabled(Profile profile) {
+        if (ContextUtils.getAppSharedPreferences().getBoolean("desktop_mode", false))
+            return true;
         return WebsitePreferenceBridge.isCategoryEnabled(
                 profile, ContentSettingsType.REQUEST_DESKTOP_SITE);
     }
