diff --git a/chrome/chrome_repack_locales.gni b/chrome/chrome_repack_locales.gni
index e8f61e2fbf6f21602e31d76c99ea544bed9c188d..b88b8ca2a9759c807c10ba34dc8e1bbd7d717460 100644
--- a/chrome/chrome_repack_locales.gni
+++ b/chrome/chrome_repack_locales.gni
@@ -41,6 +41,7 @@ template("chrome_repack_locales") {
       "${root_gen_dir}/ui/strings/ax_strings_",
       "${root_gen_dir}/ui/strings/ui_strings_",
     ]
+    source_patterns += mises_locale_source_patterns
     if (!defined(deps)) {
       deps = []
     }
@@ -64,6 +65,7 @@ template("chrome_repack_locales") {
       "//ui/strings:ax_strings",
       "//ui/strings:ui_strings",
     ]
+    deps += mises_locale_deps
     if (defined(invoker.deps)) {
       deps += invoker.deps
     }
