diff --git a/chrome/android/java/src/org/chromium/chrome/browser/tab/TabThemeColorHelper.java b/chrome/android/java/src/org/chromium/chrome/browser/tab/TabThemeColorHelper.java
index 9a19d40d776c223b4ef97d319719afc4fe006000..d4cd142e2f20ed0e2ee73d1468b4322b12cb2e17 100644
--- a/chrome/android/java/src/org/chromium/chrome/browser/tab/TabThemeColorHelper.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/tab/TabThemeColorHelper.java
@@ -11,6 +11,8 @@ import org.chromium.content_public.browser.NavigationHandle;
 import org.chromium.net.NetError;
 import org.chromium.ui.base.WindowAndroid;
 
+import org.chromium.components.embedder_support.util.UrlUtilities;
+
 /**
  * Monitor changes that indicate a theme color change may be needed from tab contents.
  */
@@ -43,6 +45,8 @@ public class TabThemeColorHelper extends EmptyTabObserver {
     @Override
     public void onUrlUpdated(Tab tab) {
         updateIfNeeded(tab, false);
+        if (tab != null && UrlUtilities.isNTPUrl(tab.getUrl().getSpec()))
+            updateIfNeeded(tab, true);
     }
 
     @Override
