diff --git a/chrome/browser/resources/new_tab_page/modules/photos/images/BUILD.gn b/chrome/browser/resources/new_tab_page/modules/photos/images/BUILD.gn
index 182bfc1f27fd23f95a7d265746e730080c44b4b6..0f4e477fc58c0314d6ed64fabe0a9f6db44bfad8 100644
--- a/chrome/browser/resources/new_tab_page/modules/photos/images/BUILD.gn
+++ b/chrome/browser/resources/new_tab_page/modules/photos/images/BUILD.gn
@@ -4,7 +4,7 @@
 
 import("//ui/webui/resources/tools/generate_grd.gni")
 
-assert(!is_android)
+#assert(!is_android)
 
 generate_grd("build_grdp") {
   grd_prefix = "photos"
