diff --git a/chrome/browser/ui/tab_helpers.cc b/chrome/browser/ui/tab_helpers.cc
index bc81c5a0b9088a86a70a1dd7fd5fc6b049614384..3218fd6d2619546f3145738e04e1d5724418e21c 100644
--- a/chrome/browser/ui/tab_helpers.cc
+++ b/chrome/browser/ui/tab_helpers.cc
@@ -355,6 +355,7 @@ void TabHelpers::AttachTabHelpers(WebContents* web_contents) {
   Profile* profile =
       Profile::FromBrowserContext(web_contents->GetBrowserContext());
 
+  MISES_TAB_HELPERS
   // --- Section 1: Common tab helpers ---
   if (page_info::IsAboutThisSiteAsyncFetchingEnabled()
 #if defined(TOOLKIT_VIEWS)
