diff --git a/chrome/browser/resources/media_router/internals/BUILD.gn b/chrome/browser/resources/media_router/internals/BUILD.gn
index 987996685d4a8898cf2372878462494f0fc40a2c..29473bc7f6a4409f1375cc8bae1c215ecde5bb5e 100644
--- a/chrome/browser/resources/media_router/internals/BUILD.gn
+++ b/chrome/browser/resources/media_router/internals/BUILD.gn
@@ -4,7 +4,7 @@
 
 import("//ui/webui/resources/tools/build_webui.gni")
 
-assert(!is_android)
+# assert(!is_android)
 
 build_webui("build") {
   grd_prefix = "media_router_internals"
