diff --git a/chrome/browser/background/background_contents_service.cc b/chrome/browser/background/background_contents_service.cc
index a7df191ad3ef001af5decfeb1f8f6891957d014c..e6c680b93ddb750863bd05e6517bad3255406eaa 100644
--- a/chrome/browser/background/background_contents_service.cc
+++ b/chrome/browser/background/background_contents_service.cc
@@ -729,9 +729,13 @@ void BackgroundContentsService::HandleExtensionCrashed(
   if (!extension)
     return;
 
+#if BUILDFLAG(IS_ANDROID)
+  const bool force_installed = true;
+#else
   const bool force_installed =
       extensions::Manifest::IsComponentLocation(extension->location()) ||
       extensions::Manifest::IsPolicyLocation(extension->location());
+#endif
   if (!force_installed) {
     ShowBalloon(extension);
   } else {
