diff --git a/chrome/android/java/src/org/chromium/chrome/browser/suggestions/tile/MostVisitedTilesMediator.java b/chrome/android/java/src/org/chromium/chrome/browser/suggestions/tile/MostVisitedTilesMediator.java
index cc84b2fe1b4cc1815e338ca01501251906c12d6c..0cbd8533bca527deec305341741d8291a95e2c62 100644
--- a/chrome/android/java/src/org/chromium/chrome/browser/suggestions/tile/MostVisitedTilesMediator.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/suggestions/tile/MostVisitedTilesMediator.java
@@ -78,8 +78,8 @@ public class MostVisitedTilesMediator implements TileGroup.Observer, TemplateUrl
 
         mTileViewLandscapePadding =
                 mResources.getDimensionPixelSize(R.dimen.tile_view_padding_landscape);
-        mTileViewPortraitEdgePadding =
-                mResources.getDimensionPixelSize(R.dimen.tile_view_padding_edge_portrait);
+        mTileViewPortraitEdgePadding = 0;
+                // mResources.getDimensionPixelSize(R.dimen.tile_view_padding_edge_portrait);
         mLateralMarginSum =
                 mResources.getDimensionPixelSize(R.dimen.mvt_container_lateral_margin) * 2;
 
