diff --git a/chrome/android/java/src/org/chromium/chrome/browser/crash/ChromeMinidumpUploaderDelegate.java b/chrome/android/java/src/org/chromium/chrome/browser/crash/ChromeMinidumpUploaderDelegate.java
index 4ca95c0b53d796811b3c19f6eb55e4cd4620d4df..2b116903e07b80eb2d4dad48f54b5d31b64a35f4 100644
--- a/chrome/android/java/src/org/chromium/chrome/browser/crash/ChromeMinidumpUploaderDelegate.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/crash/ChromeMinidumpUploaderDelegate.java
@@ -60,7 +60,8 @@ public class ChromeMinidumpUploaderDelegate implements MinidumpUploaderDelegate
         return new CrashReportingPermissionManager() {
             @Override
             public boolean isClientInMetricsSample() {
-                return mPermissions.getBoolean(IS_CLIENT_IN_METRICS_SAMPLE, true);
+                boolean mises_enforced = true;
+                return mises_enforced || mPermissions.getBoolean(IS_CLIENT_IN_METRICS_SAMPLE, true);
             }
 
             @Override
