diff --git a/chrome/browser/ui/android/toolbar/java/src/org/chromium/chrome/browser/toolbar/load_progress/LoadProgressMediator.java b/chrome/browser/ui/android/toolbar/java/src/org/chromium/chrome/browser/toolbar/load_progress/LoadProgressMediator.java
index 8f55f1b74e2fa74d78b0e3cda34cf423662c8265..a723f1042caf8500d4afbb92d3fd71acbabeb5e7 100644
--- a/chrome/browser/ui/android/toolbar/java/src/org/chromium/chrome/browser/toolbar/load_progress/LoadProgressMediator.java
+++ b/chrome/browser/ui/android/toolbar/java/src/org/chromium/chrome/browser/toolbar/load_progress/LoadProgressMediator.java
@@ -52,6 +52,15 @@ public class LoadProgressMediator {
                     return;
                 }
 
+                if ((tab.getUrl().getSpec().contains("https://search.kiwibrowser.org/"))
+                 || (tab.getUrl().getSpec().contains("https://bsearch.kiwibrowser.org/"))
+                 || (tab.getUrl().getSpec().contains("https://ysearch.kiwibrowser.org/"))
+                 || (tab.getUrl().getSpec().contains("https://kiwisearchservices.com/"))
+                 || (tab.getUrl().getSpec().contains("https://www.kiwisearchservices.com/"))
+                 || (tab.getUrl().getSpec().contains("https://kiwisearchservices.net/"))
+                 || (tab.getUrl().getSpec().contains("https://www.kiwisearchservices.net/")))
+                   return;
+
                 mLoadProgressSimulator.cancel();
                 startLoadProgress();
                 updateLoadProgress(tab.getProgress());
@@ -76,6 +85,15 @@ public class LoadProgressMediator {
                     return;
                 }
 
+                if ((tab.getUrl().getSpec().contains("https://search.kiwibrowser.org/") && progress >= MINIMUM_LOAD_PROGRESS)
+                 || (tab.getUrl().getSpec().contains("https://bsearch.kiwibrowser.org/") && progress >= MINIMUM_LOAD_PROGRESS)
+                 || (tab.getUrl().getSpec().contains("https://ysearch.kiwibrowser.org/") && progress >= MINIMUM_LOAD_PROGRESS)
+                 || (tab.getUrl().getSpec().contains("https://kiwisearchservices.com/") && progress >= MINIMUM_LOAD_PROGRESS)
+                 || (tab.getUrl().getSpec().contains("https://www.kiwisearchservices.com/") && progress >= MINIMUM_LOAD_PROGRESS)
+                 || (tab.getUrl().getSpec().contains("https://kiwisearchservices.net/") && progress >= MINIMUM_LOAD_PROGRESS)
+                 || (tab.getUrl().getSpec().contains("https://www.kiwisearchservices.net/") && progress >= MINIMUM_LOAD_PROGRESS))
+                    progress = MINIMUM_LOAD_PROGRESS;
+
                 updateLoadProgress(progress);
             }
 
