diff --git a/chrome/common/BUILD.gn b/chrome/common/BUILD.gn
index 20b82925438f560c26e1bd32f3bd1c9a0a6345ac..00ef6fd06849e45385290ba55f3f3b14624f6761 100644
--- a/chrome/common/BUILD.gn
+++ b/chrome/common/BUILD.gn
@@ -89,6 +89,7 @@ source_set("channel_info") {
   } else if (is_fuchsia) {
     sources += [ "channel_info_fuchsia.cc" ]
   }
+  public_deps += [ "//mises/common:channel_info" ]
 }
 
 source_set("ini_parser") {
@@ -404,7 +405,8 @@ static_library("common_lib") {
       "media/chrome_media_drm_bridge_client.cc",
       "media/chrome_media_drm_bridge_client.h",
     ]
-  } else {
+  }
+  if (true) {
     # Non-Android.
     public_deps += [ "//chrome/common/importer" ]
   }
@@ -535,6 +537,7 @@ static_library("non_code_constants") {
     "//printing/buildflags",
     "//ui/base:buildflags",
   ]
+  configs += [ "//mises/common:constants_configs" ]
 }
 
 component("chrome_features") {
