diff --git a/chrome/browser/resources/new_tab_page/modules/cart/icons/BUILD.gn b/chrome/browser/resources/new_tab_page/modules/cart/icons/BUILD.gn
index 843695cf037ea5a67e16952840aedf545d3d4ea9..708666be1db407ea806dbbc9263d06dd7089c5a3 100644
--- a/chrome/browser/resources/new_tab_page/modules/cart/icons/BUILD.gn
+++ b/chrome/browser/resources/new_tab_page/modules/cart/icons/BUILD.gn
@@ -4,7 +4,7 @@
 
 import("//ui/webui/resources/tools/generate_grd.gni")
 
-assert(!is_android)
+# assert(!is_android)
 
 generate_grd("build_grdp") {
   grd_prefix = "cart"
