diff --git a/chrome/browser/extensions/api/cookies/cookies_api.cc b/chrome/browser/extensions/api/cookies/cookies_api.cc
index 87631bc5c4475e10a23dcbf5a18afd073bb510a7..7c210312141c6f75e5ef2c61d969d376d040135e 100644
--- a/chrome/browser/extensions/api/cookies/cookies_api.cc
+++ b/chrome/browser/extensions/api/cookies/cookies_api.cc
@@ -37,6 +37,7 @@
 #include "net/cookies/canonical_cookie.h"
 #include "net/cookies/cookie_constants.h"
 #include "services/network/public/mojom/network_service.mojom.h"
+#include "components/safe_browsing/buildflags.h"
 
 using content::BrowserThread;
 
@@ -318,6 +319,7 @@ void CookiesGetFunction::GetCookieListCallback(
 }
 
 void CookiesGetFunction::NotifyExtensionTelemetry() {
+#if BUILDFLAG(FULL_SAFE_BROWSING)
   auto* telemetry_service =
       safe_browsing::ExtensionTelemetryServiceFactory::GetForProfile(
           Profile::FromBrowserContext(browser_context()));
@@ -331,6 +333,7 @@ void CookiesGetFunction::NotifyExtensionTelemetry() {
       parsed_args_->details.store_id.value_or(std::string()),
       parsed_args_->details.url, js_callstack().value_or(StackTrace()));
   telemetry_service->AddSignal(std::move(cookies_get_signal));
+#endif
 }
 
 CookiesGetAllFunction::CookiesGetAllFunction() {
@@ -423,6 +426,7 @@ void CookiesGetAllFunction::GetCookieListCallback(
 }
 
 void CookiesGetAllFunction::NotifyExtensionTelemetry() {
+#if BUILDFLAG(FULL_SAFE_BROWSING)
   auto* telemetry_service =
       safe_browsing::ExtensionTelemetryServiceFactory::GetForProfile(
           Profile::FromBrowserContext(browser_context()));
@@ -441,6 +445,7 @@ void CookiesGetAllFunction::NotifyExtensionTelemetry() {
           parsed_args_->details.url.value_or(std::string()),
           parsed_args_->details.session, js_callstack().value_or(StackTrace()));
   telemetry_service->AddSignal(std::move(cookies_get_all_signal));
+#endif
 }
 
 CookiesSetFunction::CookiesSetFunction()
