diff --git a/chrome/utility/BUILD.gn b/chrome/utility/BUILD.gn
index 54946c415c57c5ba1df3f643e8d79b216ac0d560..eef7ed7add6cb01c96dd7ea8ec60466a35b4ff76 100644
--- a/chrome/utility/BUILD.gn
+++ b/chrome/utility/BUILD.gn
@@ -264,6 +264,15 @@ static_library("utility") {
     deps += [ "//chrome/services/media_gallery_util:lib" ]
   }
 
+  if (is_android) {
+    sources += [
+      "importer/bookmark_html_reader.cc",
+      "importer/bookmark_html_reader.h",
+      "importer/favicon_reencode.cc",
+      "importer/favicon_reencode.h",
+    ]
+  }
+
   if (enable_browser_speech_service) {
     deps += [ "//chrome/services/speech:lib" ]
   }
