diff --git a/chrome/android/java/src/org/chromium/chrome/browser/tasks/ReturnToChromeUtil.java b/chrome/android/java/src/org/chromium/chrome/browser/tasks/ReturnToChromeUtil.java
index e6d19c9e881126562c93983454a10674eff4a555..58bebd2b3297bce4e7734e037053786d35ded90a 100644
--- a/chrome/android/java/src/org/chromium/chrome/browser/tasks/ReturnToChromeUtil.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/tasks/ReturnToChromeUtil.java
@@ -617,7 +617,8 @@ public final class ReturnToChromeUtil {
             if (!initialized && !sSkipInitializationCheckForTesting) {
                 return false;
             } else {
-                return useChromeHomepage();
+                boolean mises_enforce = false;
+                return mises_enforce && useChromeHomepage();
             }
         }
 
