diff --git a/chrome/browser/apps/app_service/BUILD.gn b/chrome/browser/apps/app_service/BUILD.gn
index 0ee96ae9072eaabe9e0d800e8844046e63dc9c31..67a710db6c7e74748665f4857f575ef217e4e344 100644
--- a/chrome/browser/apps/app_service/BUILD.gn
+++ b/chrome/browser/apps/app_service/BUILD.gn
@@ -4,7 +4,7 @@
 
 import("//build/config/chromeos/ui_mode.gni")
 
-assert(!is_android)
+#assert(!is_android)
 
 source_set("app_service") {
   sources = [
