diff --git a/chrome/browser/extensions/chrome_content_browser_client_extensions_part.cc b/chrome/browser/extensions/chrome_content_browser_client_extensions_part.cc
index 94fc2c99daeb44a1fd0d841e9e8a174e882057a8..7ac6d7e46697f19f69ed8e2814c30ee09ecd0dad 100644
--- a/chrome/browser/extensions/chrome_content_browser_client_extensions_part.cc
+++ b/chrome/browser/extensions/chrome_content_browser_client_extensions_part.cc
@@ -336,6 +336,11 @@ bool ChromeContentBrowserClientExtensionsPart::DoesSiteRequireDedicatedProcess(
   const Extension* extension = ExtensionRegistry::Get(browser_context)
                                    ->enabled_extensions()
                                    .GetExtensionOrAppByURL(effective_site_url);
+  if (!ProcessMap::IsIsolateExtensionsEnabled()) {
+    if (extension != nullptr && extension->is_extension()) {
+      return false; 
+    }
+  }
   // Isolate all extensions.
   return extension != nullptr;
 }
@@ -403,10 +408,17 @@ bool ChromeContentBrowserClientExtensionsPart::CanCommitURL(
   // commit. This accounts for cases where an extension might have multiple
   // processes, such as incognito split mode.
   ProcessMap* process_map = ProcessMap::Get(process_host->GetBrowserContext());
-  if (process_map->Contains(extension->id(), process_host->GetID())) {
-    return true;
+  if (ProcessMap::IsIsolateExtensionsEnabled() || (extension && process_map->ContainsExtension(extension->id()))) {
+    if (process_map->Contains(extension->id(), process_host->GetID())) {
+      return true;
+    }
+  } else {
+    if (process_map->Contains(process_host->GetID())) {
+      return true;
+    }
   }
 
+
   // If an extension URL is listed as sandboxed in the manifest, its process
   // won't be in the process map. Instead, allow it here and rely on the
   // ChildProcessSecurityPolicy::CanAccessDataForOrigin check (which occurs
@@ -473,11 +485,19 @@ bool ChromeContentBrowserClientExtensionsPart::IsSuitableHost(
   // SiteInstances for both extensions and hosted apps.
   const Extension* extension =
       GetEnabledExtensionFromSiteURL(profile, site_url);
-  if (extension &&
-      !process_map->Contains(extension->id(), process_host->GetID())) {
-    return false;
+  if (ProcessMap::IsIsolateExtensionsEnabled() || (extension && process_map->ContainsExtension(extension->id()))) {
+    if (extension &&
+        !process_map->Contains(extension->id(), process_host->GetID())) {
+      return false;
+    }
+  } else {
+    if (extension &&
+        !process_map->Contains(process_host->GetID())) {
+      return false;
+    }
   }
 
+
   // Conversely, don't use an extension process for a site URL that does not
   // map to an enabled extension. For example, this prevents a reload of an
   // extension or app that has just been disabled from staying in the
