diff --git a/chrome/browser/ui/browser.h b/chrome/browser/ui/browser.h
index 9081a0ed32b64202ead5495f188a97cedf1a6d35..0d4d9831448163a5495a619a2643cb949f457889 100644
--- a/chrome/browser/ui/browser.h
+++ b/chrome/browser/ui/browser.h
@@ -53,7 +53,7 @@
 #include "ui/gfx/geometry/rect.h"
 #include "ui/shell_dialogs/select_file_dialog.h"
 
-#if BUILDFLAG(IS_ANDROID)
+#if false && BUILDFLAG(IS_ANDROID)
 #error This file should only be included on desktop.
 #endif
 
