diff --git a/chrome/browser/resource_coordinator/tab_manager.cc b/chrome/browser/resource_coordinator/tab_manager.cc
index c9665181f979ffaa0d795b4d68b27f1d32549548..319a9c2f3591675e91ae8072ec4b26af1af3c25e 100644
--- a/chrome/browser/resource_coordinator/tab_manager.cc
+++ b/chrome/browser/resource_coordinator/tab_manager.cc
@@ -150,7 +150,7 @@ WebContents* TabManager::DiscardTabByExtension(content::WebContents* contents) {
     TabLifecycleUnitExternal* tab_lifecycle_unit_external =
         TabLifecycleUnitExternal::FromWebContents(contents);
     DCHECK(tab_lifecycle_unit_external);
-    if (tab_lifecycle_unit_external->DiscardTab(
+    if (tab_lifecycle_unit_external && tab_lifecycle_unit_external->DiscardTab(
             LifecycleUnitDiscardReason::EXTERNAL)) {
       return tab_lifecycle_unit_external->GetWebContents();
     }
