diff --git a/chrome/browser/resources/webui_gallery/BUILD.gn b/chrome/browser/resources/webui_gallery/BUILD.gn
index df2bc2ff78e7f035174cb9b6fee5a7332e69714e..7b4415d2768f621a62faf1dc8c9180baa0f84beb 100644
--- a/chrome/browser/resources/webui_gallery/BUILD.gn
+++ b/chrome/browser/resources/webui_gallery/BUILD.gn
@@ -4,7 +4,7 @@
 
 import("//chrome/browser/resources/tools/build_webui.gni")
 
-assert(!is_android)
+# assert(!is_android)
 
 build_webui("build") {
   grd_prefix = "webui_gallery"
