diff --git a/chrome/browser/ui/android/appmenu/internal/BUILD.gn b/chrome/browser/ui/android/appmenu/internal/BUILD.gn
index 00ca9a9e250635402df26382f9d7b4b6bf36084b..b376c03af9520284712589685e43d42fe31b0a4c 100644
--- a/chrome/browser/ui/android/appmenu/internal/BUILD.gn
+++ b/chrome/browser/ui/android/appmenu/internal/BUILD.gn
@@ -35,6 +35,12 @@ android_library("java") {
     "//third_party/androidx:androidx_core_core_java",
     "//ui/android:ui_java",
   ]
+  deps += [  
+    "//chrome/browser/tab:java",
+    "//chrome/browser/profiles/android:java",
+    "//chrome/android:chrome_java",
+    "//content/public/android:content_full_java",
+  ]
   resources_package = "org.chromium.chrome.browser.ui.appmenu.internal"
 }
 
