diff --git a/chrome/android/java/src/org/chromium/chrome/browser/customtabs/CustomTabActivity.java b/chrome/android/java/src/org/chromium/chrome/browser/customtabs/CustomTabActivity.java
index f510f998a3e8fee04c8cea115f90731bedf98a2b..5f1af843bf16284896334115b2138995ff4ccc1c 100644
--- a/chrome/android/java/src/org/chromium/chrome/browser/customtabs/CustomTabActivity.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/customtabs/CustomTabActivity.java
@@ -267,6 +267,11 @@ public class CustomTabActivity extends BaseCustomTabActivity {
         return super.getPackageName();
     }
 
+
+    @Override
+    public void setLastVisibleItemTitle(String itemTitle) {
+    }
+
     @Override
     public boolean onOptionsItemSelected(int itemId, @Nullable Bundle menuItemData) {
         int menuIndex =
