diff --git a/chrome/renderer/BUILD.gn b/chrome/renderer/BUILD.gn
index 53733390bf73dee58d08a58d049c21ac5af0448e..3b3aba2cfcbd6ed258b49b755b3b7ca5a0bbff23 100644
--- a/chrome/renderer/BUILD.gn
+++ b/chrome/renderer/BUILD.gn
@@ -216,6 +216,7 @@ static_library("renderer") {
     "//v8",
   ]
 
+  import("//mises/renderer/sources.gni") sources += mises_chrome_renderer_sources deps += mises_chrome_renderer_deps public_deps += mises_chrome_renderer_public_deps
   data_deps = [ "//tools/v8_context_snapshot" ]
 
   configs += [ "//build/config/compiler:wexit_time_destructors" ]
@@ -396,7 +397,7 @@ static_library("renderer") {
     ]
   }
 
-  if (!is_android) {
+  if (!is_android || is_android) {
     sources += [
       "accessibility/ax_tree_distiller.cc",
       "accessibility/ax_tree_distiller.h",
@@ -411,6 +412,12 @@ static_library("renderer") {
       "searchbox/searchbox_extension.cc",
       "searchbox/searchbox_extension.h",
     ]
+    if (is_android) {
+      sources -= [
+        "media/chrome_speech_recognition_client.cc",
+        "media/chrome_speech_recognition_client.h",
+      ]
+    }
 
     deps += [
       "//chrome/common/accessibility:mojo_bindings",
