diff --git a/chrome/browser/tab/BUILD.gn b/chrome/browser/tab/BUILD.gn
index 8b0495ca6bd16fbdeaae99194fb610620e3deb7f..54b502cbc4101cf710e06f0380bf8bf42126b1ba 100644
--- a/chrome/browser/tab/BUILD.gn
+++ b/chrome/browser/tab/BUILD.gn
@@ -72,7 +72,7 @@ android_library("java") {
     "//build/android:build_java",
     "//chrome/browser/android/crypto:java",
     "//chrome/browser/commerce/android:java",
-    "//chrome/browser/commerce/price_tracking/android:java",
+    #"//chrome/browser/commerce/price_tracking/android:java",
     "//chrome/browser/contextmenu:java",
     "//chrome/browser/endpoint_fetcher:java",
     "//chrome/browser/flags:java",
