diff --git a/chrome/browser/ui/android/appmenu/internal/BUILD.gn b/chrome/browser/ui/android/appmenu/internal/BUILD.gn
index d1f2ef120348297ebd98810dd0e6218c5df628bb..a7f637816792af3cba83c2f6c1df02e828bde990 100644
--- a/chrome/browser/ui/android/appmenu/internal/BUILD.gn
+++ b/chrome/browser/ui/android/appmenu/internal/BUILD.gn
@@ -35,6 +35,12 @@ android_library("java") {
     "//third_party/androidx:androidx_core_core_java",
     "//ui/android:ui_java",
   ]
+  deps += [  
+    "//chrome/browser/tab:java",
+    "//chrome/browser/profiles/android:java",
+    "//chrome/android:chrome_java",
+    "//content/public/android:content_full_java",
+  ]
   resources_package = "org.chromium.chrome.browser.ui.appmenu.internal"
 }
 
