diff --git a/chrome/android/features/tab_ui/java/src/org/chromium/chrome/browser/tasks/tab_management/TabSwitcherCoordinator.java b/chrome/android/features/tab_ui/java/src/org/chromium/chrome/browser/tasks/tab_management/TabSwitcherCoordinator.java
index d2c78ecfe19e91b6923d8e48c6ae8e4f2068ca14..9c89b8eaf4d77177f4e7d9239fb2890c4ba610de 100644
--- a/chrome/android/features/tab_ui/java/src/org/chromium/chrome/browser/tasks/tab_management/TabSwitcherCoordinator.java
+++ b/chrome/android/features/tab_ui/java/src/org/chromium/chrome/browser/tasks/tab_management/TabSwitcherCoordinator.java
@@ -72,6 +72,8 @@ import org.chromium.ui.resources.dynamics.DynamicResourceLoader;
 import java.util.ArrayList;
 import java.util.List;
 
+import org.chromium.base.ContextUtils;
+
 /**
  * Parent coordinator that is responsible for showing a grid or carousel of tabs for the main
  * TabSwitcher UI.
@@ -181,6 +183,11 @@ public class TabSwitcherCoordinator
             @Nullable BackPressManager backPressManager) {
         try (TraceEvent e = TraceEvent.scoped("TabSwitcherCoordinator.constructor")) {
             mActivity = activity;
+	    final String active_tabswitcher = ContextUtils.getAppSharedPreferences().getString("active_tabswitcher", "default");
+            if (active_tabswitcher.equals("list"))
+              mode = TabListCoordinator.TabListMode.LIST;
+            if (active_tabswitcher.equals("default") || active_tabswitcher.equals("grid"))
+              mode = TabListCoordinator.TabListMode.GRID;
             mMode = mode;
             mTabModelSelector = tabModelSelector;
             mContainer = container;
