diff --git a/chrome/browser/extensions/api/tabs/windows_event_router.cc b/chrome/browser/extensions/api/tabs/windows_event_router.cc
index 663cab095f535c0205954eebe9b1ed5f1270ef12..024f833ba32a95b1bd007a8570f32327dee2acac 100644
--- a/chrome/browser/extensions/api/tabs/windows_event_router.cc
+++ b/chrome/browser/extensions/api/tabs/windows_event_router.cc
@@ -169,18 +169,25 @@ WindowsEventRouter::WindowsEventRouter(Profile* profile)
 #elif defined(TOOLKIT_VIEWS)
   views::WidgetFocusManager::GetInstance()->AddFocusChangeListener(this);
 #else
-#error Unsupported
+// #error Unsupported
 #endif
 
   AppWindowRegistry* registry = AppWindowRegistry::Get(profile_);
   for (AppWindow* app_window : registry->app_windows())
     AddAppWindow(app_window);
+
+#if BUILDFLAG(IS_ANDROID)
+  TabModelList::AddObserver(this);
+#endif
 }
 
 WindowsEventRouter::~WindowsEventRouter() {
 #if defined(TOOLKIT_VIEWS) && !BUILDFLAG(IS_MAC)
   views::WidgetFocusManager::GetInstance()->RemoveFocusChangeListener(this);
 #endif
+#if BUILDFLAG(IS_ANDROID)
+  TabModelList::RemoveObserver(this);
+#endif
 }
 
 void WindowsEventRouter::OnAppWindowAdded(extensions::AppWindow* app_window) {
