diff --git a/chrome/browser/resources/settings/BUILD.gn b/chrome/browser/resources/settings/BUILD.gn
index 16e6d00fd3b5d2588164bf26e8aff462cabaced4..648ad880d03bd9cc0d020154079e196e0c2bf4d8 100644
--- a/chrome/browser/resources/settings/BUILD.gn
+++ b/chrome/browser/resources/settings/BUILD.gn
@@ -17,7 +17,7 @@ import("//ui/webui/webui_features.gni")
 import("//ui/webui/webui_features.gni")
 import("settings.gni")
 
-assert(!is_android, "Android does not use these settings")
+#assert(!is_android, "Android does not use these settings")
 
 preprocess_folder = "preprocessed"
 
