diff --git a/chrome/browser/ui/webui/realbox/BUILD.gn b/chrome/browser/ui/webui/realbox/BUILD.gn
index 6b66b6b558adc7a1eb1bd2c5be6f3e0881253314..ece3816664964d9813cb44559f34b6b3d554fd9c 100644
--- a/chrome/browser/ui/webui/realbox/BUILD.gn
+++ b/chrome/browser/ui/webui/realbox/BUILD.gn
@@ -4,7 +4,7 @@
 
 import("//mojo/public/tools/bindings/mojom.gni")
 
-assert(!is_android)
+#assert(!is_android)
 
 mojom("mojo_bindings") {
   sources = [ "realbox.mojom" ]
