diff --git a/chrome/browser/android/resource_id.h b/chrome/browser/android/resource_id.h
index 91f023065938368a5e059608137b7121437889ff..79f9882f428a8a5639f5814bff317f7fae0de276 100644
--- a/chrome/browser/android/resource_id.h
+++ b/chrome/browser/android/resource_id.h
@@ -98,3 +98,5 @@ DECLARE_RESOURCE_ID(IDR_ANDROID_ABOUT_THIS_SITE_LOGO_24DP,
 DECLARE_RESOURCE_ID(IDR_ANDROID_ABOUT_THIS_SITE_LOGO_24DP,
                     R.drawable.ic_info_outline_grey_24dp)
 #endif  // BUILDFLAG(GOOGLE_CHROME_BRANDING) && BUILDFLAG(IS_ANDROID)
+
+DECLARE_RESOURCE_ID(IDR_EXTENSIONS, R.drawable.ic_extensions)
