diff --git a/chrome/android/java/src/org/chromium/chrome/browser/ChromeTabbedActivity.java b/chrome/android/java/src/org/chromium/chrome/browser/ChromeTabbedActivity.java
index a9fff1726ba833b66537bb0988c51debee4a44d8..0841aca880685908ae1bae58bbb4c64c25991244 100644
--- a/chrome/android/java/src/org/chromium/chrome/browser/ChromeTabbedActivity.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/ChromeTabbedActivity.java
@@ -242,6 +242,21 @@ import java.util.Locale;
 import java.util.Set;
 import java.util.concurrent.atomic.AtomicBoolean;
 
+import org.chromium.base.ContextUtils;
+import android.net.Uri;
+import org.chromium.components.url_formatter.UrlFormatter;
+import org.chromium.chrome.browser.init.ChromeBrowserReferrer;
+import org.chromium.chrome.browser.init.InAppUpdater;
+import org.chromium.chrome.browser.homepage.HomepageManager;
+import android.animation.ValueAnimator;
+import org.chromium.chrome.browser.mises.MisesController;
+import org.chromium.chrome.browser.mises.MisesShareWin;
+import org.chromium.chrome.browser.mises.UIUtil;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.chromium.chrome.browser.feed.FeedFeatures;
+import org.chromium.base.MisesSysUtils;
+
 /**
  * This is the main activity for ChromeMobile when not running in document mode.  All the tabs
  * are accessible via a chrome specific tab switching UI.
@@ -260,7 +275,7 @@ public class ChromeTabbedActivity extends ChromeActivity<ChromeActivityComponent
     // Maximum delay for initial tab creation. This is for homepage and NTP, not previous tabs
     // restore. This is needed because we do not know when reading PartnerBrowserCustomizations
     // provider will be finished.
-    private static final int INITIAL_TAB_CREATION_TIMEOUT_MS = 500;
+    private static final int INITIAL_TAB_CREATION_TIMEOUT_MS = 1;
 
     /**
      * Sending an intent with this action to Chrome will cause it to close all tabs
@@ -412,6 +427,8 @@ public class ChromeTabbedActivity extends ChromeActivity<ChromeActivityComponent
 
     private ExperimentalStartupMetricsTracker mExperimentalStartupMetricsTracker;
 
+    private InAppUpdater mInAppUpdater = new InAppUpdater();
+
     private final IncognitoTabHost mIncognitoTabHost = new IncognitoTabHost() {
         @Override
         public boolean hasIncognitoTabs() {
@@ -827,7 +844,7 @@ public class ChromeTabbedActivity extends ChromeActivity<ChromeActivityComponent
                         (getLayoutManager().getActiveLayoutType() == LayoutType.TAB_SWITCHER)
                         ? TabLaunchType.FROM_TAB_SWITCHER_UI
                         : TabLaunchType.FROM_CHROME_UI;
-                getCurrentTabCreator().launchNTP(tabLaunchType);
+                launchNTP();
                 mLocaleManager.showSearchEnginePromoIfNeeded(ChromeTabbedActivity.this, null);
                 if (getTabModelSelector().isIncognitoSelected()) {
                     RecordUserAction.record("MobileToolbarStackViewNewIncognitoTab");
@@ -894,6 +911,12 @@ public class ChromeTabbedActivity extends ChromeActivity<ChromeActivityComponent
                         this, mLayoutManager, mTabModelSelector, getLifecycleDispatcher());
             }
 
+            try {
+                ValueAnimator.class.getMethod("setDurationScale", float.class).invoke(null, 0.60f);
+            } catch (Throwable t) {
+
+            }
+
             mUIWithNativeInitialized = true;
 
             // The dataset has already been created, we need to initialize our state.
@@ -942,6 +965,8 @@ public class ChromeTabbedActivity extends ChromeActivity<ChromeActivityComponent
                 // TODO(crbug.com/1344639): Enable split compositor task on tablets.
                 initializeCompositorContent();
             }
+            ChromeBrowserReferrer.handleInstallReferrer(this);
+            mInAppUpdater.startCheck(this);
 
             // All this initialization can be expensive so it's split into multiple tasks.
             PostTask.postTask(
@@ -985,6 +1010,8 @@ public class ChromeTabbedActivity extends ChromeActivity<ChromeActivityComponent
             }
 
             mInactivityTracker.setLastVisibleTimeMsAndRecord(System.currentTimeMillis());
+
+            MisesSysUtils.init(this);
         }
     }
 
@@ -1026,6 +1053,8 @@ public class ChromeTabbedActivity extends ChromeActivity<ChromeActivityComponent
         }
 
         FeatureNotificationUtils.handleIntentIfApplicable(getIntent());
+
+        mInAppUpdater.onResume(this);
     }
 
     @Override
@@ -1090,6 +1119,18 @@ public class ChromeTabbedActivity extends ChromeActivity<ChromeActivityComponent
                 return;
             }
 
+            if (intent != null) {
+                if (intent.getAction() != null && intent.getAction().equals(Intent.ACTION_MAIN)) {
+                    String url = intent.getStringExtra("mises_url");
+                    if (url != null && !url.isEmpty()) {
+                        url = UrlFormatter.fixupUrl(url).getSpec();
+                        Intent newintent = new Intent(Intent.ACTION_VIEW, Uri.parse(url));
+                        newintent.setPackage(getPackageName());
+                        startActivity(newintent);
+                    }
+                }
+            }
+
             // The intent to use in maybeDispatchExplicitMainViewIntent(). We're explicitly
             // adding NEW_TASK flag to make sure backing from CCT brings up the caller activity,
             // and not Chrome
@@ -1324,6 +1365,7 @@ public class ChromeTabbedActivity extends ChromeActivity<ChromeActivityComponent
 
             boolean noRestoreState =
                     CommandLine.getInstance().hasSwitch(ChromeSwitches.NO_RESTORE_STATE);
+            noRestoreState = noRestoreState || ContextUtils.getAppSharedPreferences().getBoolean("close_tabs_on_exit", false);
             boolean shouldShowNtpAsHomeSurfaceAtStartup = false;
             final AtomicBoolean isActiveUrlNTP = new AtomicBoolean(false);
             if (noRestoreState) {
@@ -1442,7 +1484,8 @@ public class ChromeTabbedActivity extends ChromeActivity<ChromeActivityComponent
             if (!mPendingInitialTabCreation) {
                 Tab currentTab = getActivityTab();
                 boolean isTabNtp = isTabRegularNtp(currentTab);
-                if (isTabNtp && !currentTab.isNativePage()) {
+                boolean isTabNativeUrl = currentTab != null && currentTab.getUrl().getScheme().equals(UrlConstants.CHROME_NATIVE_SCHEME);
+                if (isTabNativeUrl && isTabNtp && !currentTab.isNativePage()) {
                     // This will be a NTP, but the native page hasn't been created yet. Need to wait
                     // for this to be created before allowing the toolbar to draw.
                     currentTab.addObserver(new EmptyTabObserver() {
@@ -2371,6 +2414,19 @@ public class ChromeTabbedActivity extends ChromeActivity<ChromeActivityComponent
             TabModel currentModel = mTabModelSelector.getCurrentModel();
             if (!currentModel.isIncognito()) currentModel.openMostRecentlyClosedEntry();
             RecordUserAction.record("MobileTabClosedUndoShortCut");
+        } else if (id == R.id.mises_forward_menu_id) {
+            handleForward(currentTab);
+        } else if (id == R.id.new_home_tab_menu_id) {
+            getTabModelSelector().getModel(false).commitAllTabClosures();
+            RecordUserAction.record("MobileMenuNewHomeTab");
+            RecordUserAction.record("MobileNewHomeTabOpened");
+            ReturnToChromeUtil.onNewTabOpened();
+            reportNewTabShortcutUsed(false);
+            if (fromMenu) RecordUserAction.record("MobileMenuNewHomeTab.AppMenu");
+
+            getTabCreator(false).launchUrl(HomepageManager.getInstance().getHomepageUriIgnoringEnabledStateChromium(), TabLaunchType.FROM_CHROME_UI);
+
+            mLocaleManager.showSearchEnginePromoIfNeeded(this, null);
         } else if (id == R.id.quick_delete_menu_id
                 && QuickDeleteController.isQuickDeleteEnabled()) {
             assert !mTabModelSelector.getCurrentModel().isIncognito()
@@ -2789,7 +2845,7 @@ public class ChromeTabbedActivity extends ChromeActivity<ChromeActivityComponent
             // See crbug.com/1473947.
             if (mStartSurfaceSupplier.get() == null) {
                 if (getTabModelSelector().getCurrentModel().getCount() == 0) {
-                    getCurrentTabCreator().launchNTP();
+                    launchNTP();
                 } else {
                     mLayoutManager.showLayout(LayoutType.BROWSING, true);
                 }
@@ -3137,6 +3193,7 @@ public class ChromeTabbedActivity extends ChromeActivity<ChromeActivityComponent
     public void onActivityResult(int requestCode, int resultCode, Intent data) {
         try (TraceEvent e = TraceEvent.scoped("ChromeTabbedActivity.onActivityResult")) {
             super.onActivityResult(requestCode, resultCode, data);
+            mInAppUpdater.onActivityResult(requestCode, resultCode, data);
         }
     }
 
@@ -3294,4 +3351,55 @@ public class ChromeTabbedActivity extends ChromeActivity<ChromeActivityComponent
     private boolean skipSavingNonActiveNtps() {
         return StartSurfaceConfiguration.isNtpAsHomeSurfaceEnabled(isTablet());
     }
+
+    private void launchNTP() {
+        if (getTabModelSelector().isIncognitoSelected()) {
+            getCurrentTabCreator().launchIncognitoNTP();
+        } else {
+            getCurrentTabCreator().launchNTP(TabLaunchType.FROM_CHROME_UI);
+        }
+    }
+    private void handleForward(final Tab currentTab) {
+        String SCRIPT = "if(window.misesModule && window.misesModule.getWindowInformation){window.misesModule.getWindowInformation()} else {console.log('window.misesModule or window.misesModule.getWindowInformation is null')}";
+        final Context context = ChromeTabbedActivity.this;
+        if (currentTab == null || currentTab.getWebContents() == null)
+            return;
+        if (currentTab.isNativePage() || currentTab.isClosing()
+                || currentTab.isShowingErrorPage() ) {
+            Log.e("mises","share currentTab.isNativePage() || currentTab.isClosing() || currentTab.isShowingErrorPage() || currentTab.isShowingSadTab()");
+            Toast.makeText(context, context.getString(R.string.lbl_can_not_share_tip), Toast.LENGTH_SHORT).show();
+            return;
+        }
+        currentTab.getWebContents().evaluateJavaScript(SCRIPT, jsonResult -> {
+            Log.e("mises share msg : ", jsonResult);
+            if (jsonResult != null && !jsonResult.isEmpty()) {
+                try {
+                    JSONObject ob = new JSONObject(jsonResult);
+                    String icon = "";
+                    if (ob.has("icon"))
+                        icon = ob.getString("icon");
+                    String title = ob.getString("title");
+                    String url = ob.getString("url");
+                    if (MisesController.getInstance().getMisesToken().isEmpty()) {
+                        UIUtil.showAlertDialog(context, context.getString(R.string.lbl_login_tip), v1 -> {
+                            ChromeTabCreator tabCreator = ChromeTabbedActivity.this.getTabCreator(false);
+                            if (tabCreator != null) {
+                                tabCreator.openSinglePage(MisesController.MISES_HOME_BASE_URL + "/home/me");
+                            }
+                        });
+                        return;
+                    }
+                    MisesShareWin shareWin = MisesShareWin.newInstance(ChromeTabbedActivity.this, icon, title, url);
+                    shareWin.show((ChromeTabbedActivity.this).getSupportFragmentManager(), "MisesShareWin");
+                } catch (JSONException e) {
+                    Log.e("mises", "share is not json" + e.toString());
+                    Toast.makeText(context, context.getString(R.string.lbl_can_not_share_tip), Toast.LENGTH_SHORT).show();
+                    e.printStackTrace();
+                }
+            } else {
+                Log.e("mises", "share json is null");
+                Toast.makeText(context, context.getString(R.string.lbl_can_not_share_tip), Toast.LENGTH_SHORT).show();
+            }
+        });
+    }
 }
