diff --git a/chrome/android/java/src/org/chromium/chrome/browser/gesturenav/NavigationHandler.java b/chrome/android/java/src/org/chromium/chrome/browser/gesturenav/NavigationHandler.java
index 7d9efda292348f336ceb524599a14645b89c82f7..974c832aaf71f325f6f9102b5e12436609498ffa 100644
--- a/chrome/android/java/src/org/chromium/chrome/browser/gesturenav/NavigationHandler.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/gesturenav/NavigationHandler.java
@@ -50,7 +50,7 @@ class NavigationHandler implements TouchEventObserver {
 
     // Weighted value to determine when to trigger an edge swipe. Initial scroll
     // vector should form 30 deg or below to initiate swipe action.
-    private static final float WEIGTHED_TRIGGER_THRESHOLD = 1.73f;
+    private static final float WEIGTHED_TRIGGER_THRESHOLD = 0.43f;
 
     // |EDGE_WIDTH_DP| in physical pixel.
     private final float mEdgeWidthPx;
