diff --git a/chrome/android/java/src/org/chromium/chrome/browser/tabmodel/TabModelImpl.java b/chrome/android/java/src/org/chromium/chrome/browser/tabmodel/TabModelImpl.java
index 2d9d434575553fd364ddfa136c69a00b91be4255..0d097139b12983eea5572c0ffacc6fe786c1cb7d 100644
--- a/chrome/android/java/src/org/chromium/chrome/browser/tabmodel/TabModelImpl.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/tabmodel/TabModelImpl.java
@@ -850,14 +850,14 @@ public class TabModelImpl extends TabModelJniBridge {
     /** Used to restore tabs from native. */
     @Override
     protected boolean createTabWithWebContents(
-            Tab parent, Profile profile, WebContents webContents, boolean select) {
+            Tab parent, Profile profile, WebContents webContents, boolean select, @TabLaunchType int type) {
         return getTabCreator(profile.isOffTheRecord())
                 .createTabWithWebContents(
                         parent,
                         webContents,
                         select
                                 ? TabLaunchType.FROM_RECENT_TABS_FOREGROUND
-                                : TabLaunchType.FROM_RECENT_TABS);
+                                : type);
     }
 
     @Override
