diff --git a/chrome/browser/ui/views/extensions/extensions_toolbar_container.cc b/chrome/browser/ui/views/extensions/extensions_toolbar_container.cc
index b23cfc336a2ee6cdd952018bf1eccd72481c3cbe..2aad49ed05133366c907fc1126748fe8ccf6c804 100644
--- a/chrome/browser/ui/views/extensions/extensions_toolbar_container.cc
+++ b/chrome/browser/ui/views/extensions/extensions_toolbar_container.cc
@@ -61,6 +61,9 @@ base::OnceClosure& GetOnVisibleCallbackForTesting() {
 // overlapping with the extension popup. The media picker dialog has been
 // identified to be susceptible. See crbug.com/1300006.
 bool HasPossiblyOverlappingSecurityUI(Browser* browser) {
+#if BUILDFLAG(IS_ANDROID)
+  return false;
+#else
   views::ElementTrackerViews::ViewList media_picker_dialogs =
       views::ElementTrackerViews::GetInstance()->GetAllMatchingViews(
           DesktopMediaPickerDialogView::kDesktopMediaPickerDialogViewIdentifier,
@@ -71,6 +74,7 @@ bool HasPossiblyOverlappingSecurityUI(Browser* browser) {
                        views::Widget* dialog_widget = dialog_view->GetWidget();
                        return dialog_widget && dialog_widget->IsVisible();
                      });
+#endif
 }
 
 }  // namespace
