diff --git a/chrome/browser/ui/android/omnibox/java/src/org/chromium/chrome/browser/omnibox/status/StatusMediator.java b/chrome/browser/ui/android/omnibox/java/src/org/chromium/chrome/browser/omnibox/status/StatusMediator.java
index c13027d1f5e64d32d0b7eea88fb5574289351dc9..2a528838bc7007d03441131f092c6bfa8f5fd9a5 100644
--- a/chrome/browser/ui/android/omnibox/java/src/org/chromium/chrome/browser/omnibox/status/StatusMediator.java
+++ b/chrome/browser/ui/android/omnibox/java/src/org/chromium/chrome/browser/omnibox/status/StatusMediator.java
@@ -541,6 +541,8 @@ public class StatusMediator
             return true;
         }
 
+        boolean mises_enforced = true;
+        if (mises_enforced) return false;
         return (mUrlHasFocus || mUrlFocusPercent > 0)
                 && isNtpVisible()
                 && mProfileSupplier.hasValue();
