diff --git a/chrome/browser/resources/extensions/toolbar.ts b/chrome/browser/resources/extensions/toolbar.ts
index 5aeeef76f868bf6c849e10fc50de5d06b44cbc9c..420ee55c50bea554f27799b2263383c9f8de7b43 100644
--- a/chrome/browser/resources/extensions/toolbar.ts
+++ b/chrome/browser/resources/extensions/toolbar.ts
@@ -67,7 +67,7 @@ export class ExtensionsToolbarElement extends ExtensionsToolbarElementBase {
 
       inDevMode: {
         type: Boolean,
-        value: false,
+        value: true,
         observer: 'onInDevModeChanged_',
         reflectToAttribute: true,
       },
@@ -190,6 +190,10 @@ export class ExtensionsToolbarElement extends ExtensionsToolbarElementBase {
     chrome.metricsPrivate.recordUserAction('Options_LoadUnpackedExtension');
   }
 
+  private openExtensionsWebStore_() {
+    window.open("https://chrome.google.com/webstore/category/extensions");
+  }
+
   private onPackClick_() {
     chrome.metricsPrivate.recordUserAction('Options_PackExtension');
     this.showPackDialog_ = true;
