diff --git a/chrome/browser/resources/web_app_internals/BUILD.gn b/chrome/browser/resources/web_app_internals/BUILD.gn
index 206009f2925ca4fdccea396240c615d32d805806..80170c86769c1397ee47c2356766a6ec573bc7e3 100644
--- a/chrome/browser/resources/web_app_internals/BUILD.gn
+++ b/chrome/browser/resources/web_app_internals/BUILD.gn
@@ -4,7 +4,7 @@
 
 import("//ui/webui/resources/tools/build_webui.gni")
 
-assert(!is_android)
+#assert(!is_android)
 
 build_webui("build") {
   grd_prefix = "web_app_internals"
