diff --git a/chrome/browser/ui/web_applications/BUILD.gn b/chrome/browser/ui/web_applications/BUILD.gn
index 016835c9b612f05317c6199a652e03abf24ede84..a1efb41c8da7aac5c46da31c9ea3b54ab0874731 100644
--- a/chrome/browser/ui/web_applications/BUILD.gn
+++ b/chrome/browser/ui/web_applications/BUILD.gn
@@ -114,6 +114,10 @@ source_set("browser_tests") {
       "//components/metrics/structured:structured_events",
     ]
   }
+
+  if (is_android) {
+    deps -= ["//chrome/test:test_support_ui"]
+  }
 }
 
 # On Lacros, these browser tests require Ash to be running, as they rely on
@@ -175,4 +179,7 @@ source_set("app_service_browser_tests") {
       "//chromeos/lacros:test_support",
     ]
   }
+  if (is_android) {
+    deps -= ["//chrome/test:test_support_ui"]
+  }
 }
