diff --git a/chrome/browser/resources/new_tab_page/modules/cart/icons/BUILD.gn b/chrome/browser/resources/new_tab_page/modules/cart/icons/BUILD.gn
index fdf8deaab02c832e99f4176b26d5705915d8f6b6..fe1d6cc11d5c03941e778cfcdf226e25befeee06 100644
--- a/chrome/browser/resources/new_tab_page/modules/cart/icons/BUILD.gn
+++ b/chrome/browser/resources/new_tab_page/modules/cart/icons/BUILD.gn
@@ -4,7 +4,7 @@
 
 import("//ui/webui/resources/tools/generate_grd.gni")
 
-assert(!is_android)
+# assert(!is_android)
 
 generate_grd("build_grdp") {
   grd_prefix = "cart"
