diff --git a/chrome/browser/resources/side_panel/search_companion/BUILD.gn b/chrome/browser/resources/side_panel/search_companion/BUILD.gn
index 1158c241e4b33406ae5d2373f40e01ae2ad83609..cd060b1af54425b6edf634b79e0c2e3cdee7b714 100644
--- a/chrome/browser/resources/side_panel/search_companion/BUILD.gn
+++ b/chrome/browser/resources/side_panel/search_companion/BUILD.gn
@@ -4,7 +4,7 @@
 
 import("//ui/webui/resources/tools/build_webui.gni")
 
-assert(!is_android)
+# assert(!is_android)
 
 build_webui("build") {
   grd_prefix = "side_panel_search_companion"
