diff --git a/chrome/browser/profiles/profile_impl.h b/chrome/browser/profiles/profile_impl.h
index 7db7fd820a931f0d3560c83c53ab5990ccf860cc..7dad73d1d02b02a72b0287947e6dd6e43db22399 100644
--- a/chrome/browser/profiles/profile_impl.h
+++ b/chrome/browser/profiles/profile_impl.h
@@ -155,7 +155,7 @@ class ProfileImpl : public Profile {
   void OnLogin() override;
   void InitChromeOSPreferences() override;
 #endif  // BUILDFLAG(IS_CHROMEOS_ASH)
-
+  friend class BraveProfileImpl;
   bool IsNewProfile() const override;
 
   void SetCreationTimeForTesting(base::Time creation_time) override;
