diff --git a/chrome/android/java/src/org/chromium/chrome/browser/tabmodel/TabModelJniBridge.java b/chrome/android/java/src/org/chromium/chrome/browser/tabmodel/TabModelJniBridge.java
index 8297746bf51ca28a72cf062dde00a2f606c77cb6..4d6600a28419acb621464e0e264a2a5ff9515ec6 100644
--- a/chrome/android/java/src/org/chromium/chrome/browser/tabmodel/TabModelJniBridge.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/tabmodel/TabModelJniBridge.java
@@ -16,12 +16,14 @@ import org.chromium.chrome.browser.flags.ActivityType;
 import org.chromium.chrome.browser.profiles.Profile;
 import org.chromium.chrome.browser.tab.Tab;
 import org.chromium.chrome.browser.tab.TabLaunchType;
+import org.chromium.chrome.browser.tabmodel.MisesTabCreator;
 import org.chromium.chrome.browser.tasks.tab_groups.TabGroupModelFilter;
 import org.chromium.content_public.browser.LoadUrlParams;
 import org.chromium.content_public.browser.WebContents;
 import org.chromium.content_public.common.ResourceRequestBody;
 import org.chromium.url.GURL;
 import org.chromium.url.Origin;
+import org.chromium.ui.base.PageTransition;
 
 /** Bridges between the C++ and Java {@link TabModel} interfaces. */
 public abstract class TabModelJniBridge implements TabModel {
@@ -153,7 +155,7 @@ public abstract class TabModelJniBridge implements TabModel {
      */
     @CalledByNative
     protected abstract boolean createTabWithWebContents(
-            Tab parent, Profile profile, WebContents webContents, boolean select);
+            Tab parent, Profile profile, WebContents webContents, boolean select, @TabLaunchType int type);
 
     @CalledByNative
     protected abstract void openNewTab(
@@ -173,7 +175,20 @@ public abstract class TabModelJniBridge implements TabModel {
     @CalledByNative
     protected Tab createNewTabForDevTools(GURL url) {
         return getTabCreator(/* incognito= */ false)
-                .createNewTab(new LoadUrlParams(url), TabLaunchType.FROM_CHROME_UI, null);
+                .createNewTab(new LoadUrlParams(url), TabLaunchType.FROM_CHROME_UI, getTabAt(index()));
+    }
+
+    @CalledByNative
+    protected WebContents createNewTabForExtensions(GURL url) {
+        final LoadUrlParams param = new LoadUrlParams(url, PageTransition.AUTO_TOPLEVEL);
+        return ((MisesTabCreator)getTabCreator(/*incognito=*/false))
+                .createExtensionPopup(param, getTabAt(index()));
+    }
+
+    @CalledByNative
+    protected void closeTabForExtension(final String extensionId) {
+        ((MisesTabCreator)getTabCreator(/*incognito=*/false))
+                .closeExtensionPopup(extensionId);
     }
 
     /** Returns whether supplied Tab instance has been grouped together with other Tabs. */
