diff --git a/chrome/android/java/src/org/chromium/chrome/browser/bookmarks/BookmarkDelegate.java b/chrome/android/java/src/org/chromium/chrome/browser/bookmarks/BookmarkDelegate.java
index cf198756074615ecafaf234b7c3bda8f28b0d8dd..6fe2cc5ea9e86374f74bbb0867ec73d001b1ec8c 100644
--- a/chrome/android/java/src/org/chromium/chrome/browser/bookmarks/BookmarkDelegate.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/bookmarks/BookmarkDelegate.java
@@ -77,6 +77,9 @@ public interface BookmarkDelegate {
      */
     BookmarkModel getModel();
 
+    void importBookmarks();
+    void exportBookmarks();
+
     /**
      * Returns current mode of bookmark main UI. If no mode is stored, {@link
      * BookmarkUiMode.LOADING} is returned.
