diff --git a/chrome/browser/tabmodel/android/java/src/org/chromium/chrome/browser/tabmodel/EmptyTabModel.java b/chrome/browser/tabmodel/android/java/src/org/chromium/chrome/browser/tabmodel/EmptyTabModel.java
index 6142a504821fb1396b54f35a68326b4b88f501f5..d2175290c2ebd4cf106e5ca4a9d74920860570f2 100644
--- a/chrome/browser/tabmodel/android/java/src/org/chromium/chrome/browser/tabmodel/EmptyTabModel.java
+++ b/chrome/browser/tabmodel/android/java/src/org/chromium/chrome/browser/tabmodel/EmptyTabModel.java
@@ -91,6 +91,11 @@ public class EmptyTabModel implements TabModel {
     @Override
     public void setIndex(int i, @TabSelectionType int type, boolean skipLoadingTab) {}
 
+    @Override
+    public int getLastNonExtensionActiveIndex() {
+        return INVALID_TAB_INDEX;
+    }
+
     @Override
     public boolean isActiveModel() {
         return false;
