diff --git a/chrome/browser/BUILD.gn b/chrome/browser/BUILD.gn
index 44d3b5e543101ce7ff5fed778fc430532e7ae9fb..7fc9efa2595adc4dfee2df8a2a0497891f83fffe 100644
--- a/chrome/browser/BUILD.gn
+++ b/chrome/browser/BUILD.gn
@@ -2730,6 +2730,7 @@ static_library("browser") {
       "signin/wait_for_network_callback_helper_chrome.h",
     ]
   }
+  import("//mises/browser/sources.gni") sources += mises_chrome_browser_sources deps += mises_chrome_browser_deps public_deps += mises_chrome_browser_public_deps
 
   if (is_android) {
     sources += [
@@ -3676,7 +3677,8 @@ static_library("browser") {
       ]
       deps += [ "//chrome/android/modules/dev_ui/provider:native" ]
     }
-  } else {
+  }
+  if (true) {
     #!is_android
     sources += [
       "accessibility/caption_bubble_context_browser.h",
@@ -4734,7 +4736,7 @@ static_library("browser") {
         "themes/theme_color_policy_handler.h",
       ]
     }
-    if (!is_chromeos_ash) {
+    if (is_android || !is_chromeos_ash) {
       sources += [
         "accessibility/soda_installer_impl.cc",
         "accessibility/soda_installer_impl.h",
@@ -4821,7 +4823,7 @@ static_library("browser") {
 
     # On Windows, the hashes are embedded in //chrome:chrome_initial rather
     # than here in :chrome_dll.
-    if (!is_win) {
+    if (!is_android && !is_win) {
       deps += [ "//chrome:packed_resources_integrity" ]
       sources += [ "certificate_viewer_stub.cc" ]
     }
@@ -4862,6 +4864,9 @@ static_library("browser") {
         "feedback/system_logs/log_sources/chrome_root_store_log_source.h",
       ]
     }
+    sources -= mises_chrome_browser_exception_sources
+    deps -= mises_chrome_browser_exception_deps
+    allow_circular_includes_from -= mises_chrome_browser_exception_allow_circular_includes_from
   }
   if (is_chromeos_lacros) {
     assert(enable_system_notifications)
@@ -6806,7 +6811,7 @@ static_library("browser") {
     ]
   }
 
-  if (toolkit_views) {
+  if (is_android || toolkit_views) {
     sources += [
       "bookmarks/bookmark_expanded_state_tracker.cc",
       "bookmarks/bookmark_expanded_state_tracker.h",
@@ -7690,7 +7695,7 @@ static_library("browser") {
     }
   }
 
-  if (enable_session_service) {
+  if (true || enable_session_service) {
     sources += [
       "sessions/app_session_service.cc",
       "sessions/app_session_service.h",
