diff --git a/chrome/browser/extensions/BUILD.gn b/chrome/browser/extensions/BUILD.gn
index 9a61d5cc1a1c33017cad6fd9e0d4617fc824ecf8..86fd3f3932e0f48627d60e475eebbf2333d89091 100644
--- a/chrome/browser/extensions/BUILD.gn
+++ b/chrome/browser/extensions/BUILD.gn
@@ -953,6 +953,7 @@ static_library("extensions") {
     "//ui/strings",
     "//url",
   ]
+  import("//mises/browser/extensions/sources.gni") sources += mises_extensions_sources
 
   if (is_linux || is_mac || is_win) {
     sources += [
@@ -1261,7 +1262,7 @@ static_library("extensions") {
     ]
   }
 
-  if (is_linux) {
+  if (is_linux || true) {
     sources +=
         [ "api/image_writer_private/removable_storage_provider_linux.cc" ]
   }
