diff --git a/chrome/android/features/tab_ui/java/src/org/chromium/chrome/browser/tasks/tab_management/TabGridPanelViewBinder.java b/chrome/android/features/tab_ui/java/src/org/chromium/chrome/browser/tasks/tab_management/TabGridPanelViewBinder.java
index 18dbeabaf24cf0b22754f6dd4d56312c0be3039a..112157c5fe664ffcddc9b3f6bf5a35a007010f6f 100644
--- a/chrome/android/features/tab_ui/java/src/org/chromium/chrome/browser/tasks/tab_management/TabGridPanelViewBinder.java
+++ b/chrome/android/features/tab_ui/java/src/org/chromium/chrome/browser/tasks/tab_management/TabGridPanelViewBinder.java
@@ -43,6 +43,9 @@ import org.chromium.ui.base.ViewUtils;
 import org.chromium.ui.modelutil.PropertyKey;
 import org.chromium.ui.modelutil.PropertyModel;
 
+import org.chromium.base.ContextUtils;
+import android.content.res.Resources;
+
 /**
  * ViewBinder for TabGridDialog.
  */
@@ -176,6 +179,11 @@ class TabGridPanelViewBinder {
     private static void setScrollIndex(RecyclerView view, int index) {
         LinearLayoutManager layoutManager = (LinearLayoutManager) view.getLayoutManager();
         int offset = computeOffset(view, layoutManager);
+        if (ContextUtils.getAppSharedPreferences().getString("active_tabswitcher", "default").equals("classic")) {
+            offset = (int) Math.ceil(75 * Resources.getSystem().getDisplayMetrics().density);
+            if (index == 0)
+                offset = 0;
+        }
         layoutManager.scrollToPositionWithOffset(index, offset);
     }
 
