diff --git a/chrome/browser/safe_browsing/BUILD.gn b/chrome/browser/safe_browsing/BUILD.gn
index 038a280213b6a45dcef35833b7e741cde8aae5b1..00194ece6ffa13f5a9e4b783ad84b3d50da55e3d 100644
--- a/chrome/browser/safe_browsing/BUILD.gn
+++ b/chrome/browser/safe_browsing/BUILD.gn
@@ -426,6 +426,8 @@ static_library("safe_browsing") {
           "tailored_security/unconsented_message_android.h",
           "telemetry/android/android_telemetry_service.cc",
           "telemetry/android/android_telemetry_service.h",
+          "cloud_content_scanning/deep_scanning_utils.cc",
+          "cloud_content_scanning/deep_scanning_utils.h",
         ]
         deps += [
           "//chrome/app/vector_icons:vector_icons",
