diff --git a/chrome/android/java/src/org/chromium/chrome/browser/tabbed_mode/TabbedNavigationBarColorController.java b/chrome/android/java/src/org/chromium/chrome/browser/tabbed_mode/TabbedNavigationBarColorController.java
index cae1f4d5848b60e752c24a39cbb964ee203a1853..e0ea66bba6423f82caf09022d037c3eeb88fe58d 100644
--- a/chrome/android/java/src/org/chromium/chrome/browser/tabbed_mode/TabbedNavigationBarColorController.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/tabbed_mode/TabbedNavigationBarColorController.java
@@ -57,6 +57,11 @@ import org.chromium.ui.util.ColorUtils;
 
 import java.util.Optional;
 
+import org.chromium.base.ContextUtils;
+import android.graphics.Color;
+
+import org.chromium.chrome.browser.night_mode.GlobalNightModeStateProviderHolder;
+
 /** Controls the bottom system navigation bar color for the provided {@link Window}. */
 @RequiresApi(Build.VERSION_CODES.O_MR1)
 class TabbedNavigationBarColorController
@@ -354,6 +359,11 @@ class TabbedNavigationBarColorController
         forceDarkNavigation |= mIsInFullscreen;
         mForceDarkNavigationBarColor = forceDarkNavigation;
 
+        if (ContextUtils.getAppSharedPreferences().getBoolean("darken_websites_enabled", false) || ContextUtils.getAppSharedPreferences().getInt("ui_theme_setting", 0) == 2 || (ContextUtils.getAppSharedPreferences().getInt("ui_theme_setting", 0) == 0 && GlobalNightModeStateProviderHolder.getInstance().isInNightMode()))
+            forceDarkNavigation = true;
+
+        if (mForceDarkNavigationBarColor == forceDarkNavigation) return;
+
         // 2. Calculate colors and store update states.
         final @ColorInt int newNavigationBarColor = getNavigationBarColor(forceDarkNavigation);
         final @ColorInt int dividerColor =
