diff --git a/chrome/utility/BUILD.gn b/chrome/utility/BUILD.gn
index 63a0c23f068f15f7cbc8f41a80a3af93568f8486..8093a0af3aa458d342eeeb65413aca90ade348df 100644
--- a/chrome/utility/BUILD.gn
+++ b/chrome/utility/BUILD.gn
@@ -148,6 +148,15 @@ static_library("utility") {
     ]
   }
 
+  if (is_android) {
+    sources += [
+      "importer/bookmark_html_reader.cc",
+      "importer/bookmark_html_reader.h",
+      "importer/favicon_reencode.cc",
+      "importer/favicon_reencode.h",
+    ]
+  }
+
   if (enable_screen_ai_service) {
     deps += [ "//components/services/screen_ai" ]
   }
