diff --git a/chrome/android/features/tab_ui/java/src/org/chromium/chrome/browser/tasks/tab_management/TabGridDialogCoordinator.java b/chrome/android/features/tab_ui/java/src/org/chromium/chrome/browser/tasks/tab_management/TabGridDialogCoordinator.java
index a019ec3ad5441d63eec47370e9aa648a9b55df90..6219d63d6b4c8e7a7e109cdbb76cb42a2e9bfe79 100644
--- a/chrome/android/features/tab_ui/java/src/org/chromium/chrome/browser/tasks/tab_management/TabGridDialogCoordinator.java
+++ b/chrome/android/features/tab_ui/java/src/org/chromium/chrome/browser/tasks/tab_management/TabGridDialogCoordinator.java
@@ -59,6 +59,10 @@ import org.chromium.ui.widget.ViewRectProvider;
 
 import java.util.List;
 
+import org.chromium.base.ContextUtils;
+
+import org.chromium.base.ContextUtils;
+
 /**
  * A coordinator for TabGridDialog component. Manages the communication with {@link
  * TabListCoordinator} as well as the life-cycle of shared component objects.
@@ -184,13 +188,18 @@ public class TabGridDialogCoordinator implements TabGridDialogMediator.DialogCon
                             showColorPickerPopupRunnable,
                             actionConfirmationManager);
 
+            int mode = TabUiFeatureUtilities.shouldUseListMode()
+                            ? TabListCoordinator.TabListMode.LIST
+                            : TabListCoordinator.TabListMode.GRID;
+            final String active_tabswitcher = ContextUtils.getAppSharedPreferences().getString("active_tabswitcher", "default");
+            if (active_tabswitcher.equals("default") || active_tabswitcher.equals("grid"))
+                mode = TabListCoordinator.TabListMode.GRID;
+
             // TODO(crbug.com/40662311) : Remove the inline mode logic here, make the constructor to
             // take in a mode parameter instead.
             mTabListCoordinator =
                     new TabListCoordinator(
-                            TabUiFeatureUtilities.shouldUseListMode()
-                                    ? TabListCoordinator.TabListMode.LIST
-                                    : TabListCoordinator.TabListMode.GRID,
+                            mode,
                             activity,
                             mBrowserControlsStateProvider,
                             mModalDialogManager,
@@ -300,6 +309,10 @@ public class TabGridDialogCoordinator implements TabGridDialogMediator.DialogCon
                     TabUiFeatureUtilities.shouldUseListMode()
                             ? TabListCoordinator.TabListMode.LIST
                             : TabListCoordinator.TabListMode.GRID;
+            final String active_tabswitcher = ContextUtils.getAppSharedPreferences().getString("active_tabswitcher", "default");
+            if (active_tabswitcher.equals("default") || active_tabswitcher.equals("grid"))
+                mode = TabListCoordinator.TabListMode.GRID;
+
             ViewGroup container = mDialogView.findViewById(R.id.dialog_container_view);
             mTabListEditorCoordinator =
                     new TabListEditorCoordinator(
