diff --git a/chrome/browser/download/download_item_model.h b/chrome/browser/download/download_item_model.h
index e3570aac9e7a99da6dd54c09bb5a7f523e415421..56680a696a4f505fd396477d01ea0d048f007710 100644
--- a/chrome/browser/download/download_item_model.h
+++ b/chrome/browser/download/download_item_model.h
@@ -27,7 +27,7 @@ class WebContents;
 class DownloadItemModel : public DownloadUIModel,
                           public download::DownloadItem::Observer {
  public:
-#if !BUILDFLAG(IS_ANDROID)
+#if true || !BUILDFLAG(IS_ANDROID)
   // How long an ephemeral warning is displayed on the download bubble.
   static constexpr base::TimeDelta kEphemeralWarningLifetimeOnBubble =
       base::Minutes(5);
