diff --git a/chrome/browser/tabmodel/android/java/src/org/chromium/chrome/browser/tabmodel/EmptyTabModel.java b/chrome/browser/tabmodel/android/java/src/org/chromium/chrome/browser/tabmodel/EmptyTabModel.java
index 99dc3fd4a06373864dd8a118902c7e7eb82c548f..e283b3e075bf7a112a79259482cf863364fcb28d 100644
--- a/chrome/browser/tabmodel/android/java/src/org/chromium/chrome/browser/tabmodel/EmptyTabModel.java
+++ b/chrome/browser/tabmodel/android/java/src/org/chromium/chrome/browser/tabmodel/EmptyTabModel.java
@@ -92,6 +92,11 @@ public class EmptyTabModel implements IncognitoTabModel {
         return null;
     }
 
+    @Override
+    public int getLastNonExtensionActiveIndex() {
+        return INVALID_TAB_INDEX;
+    }
+
     @Override
     public int indexOf(Tab tab) {
         return INVALID_TAB_INDEX;
