diff --git a/chrome/browser/signin/primary_account_policy_manager.cc b/chrome/browser/signin/primary_account_policy_manager.cc
index 64370293de9d246d6edd81d2437e1d160775eece..8c435d3ad0e6652260bd6bc52607537176384a6b 100644
--- a/chrome/browser/signin/primary_account_policy_manager.cc
+++ b/chrome/browser/signin/primary_account_policy_manager.cc
@@ -304,7 +304,7 @@ void PrimaryAccountPolicyManager::OnUserConfirmedProfileDeletion(
   delete_profile_dialog_manager_.reset();
 
   DCHECK(profiles::IsMultipleProfilesEnabled());
-
+#if !BUILDFLAG(IS_ANDROID)
   g_browser_process->profile_manager()
       ->GetDeleteProfileHelper()
       .MaybeScheduleProfileForDeletion(
@@ -313,5 +313,6 @@ void PrimaryAccountPolicyManager::OnUserConfirmedProfileDeletion(
               ? base::DoNothing()
               : base::BindOnce(&webui::OpenNewWindowForProfile),
           ProfileMetrics::DELETE_PROFILE_PRIMARY_ACCOUNT_NOT_ALLOWED);
+#endif
 }
 #endif  // defined(TOOLKIT_VIEWS) && !BUILDFLAG(IS_CHROMEOS)
