diff --git a/chrome/browser/resources/intro/BUILD.gn b/chrome/browser/resources/intro/BUILD.gn
index f7a93df1c61df04d5fe98f2fd06f5f07988f7655..fde5bb7dfa7dcb505968ac1961568351aa8e1437 100644
--- a/chrome/browser/resources/intro/BUILD.gn
+++ b/chrome/browser/resources/intro/BUILD.gn
@@ -6,7 +6,7 @@ import("//build/config/chromeos/ui_mode.gni")
 import("//components/signin/features.gni")
 import("//ui/webui/resources/tools/build_webui.gni")
 
-assert(is_chromeos_lacros || enable_dice_support)
+#assert(is_chromeos_lacros || enable_dice_support)
 
 build_webui("build") {
   grd_prefix = "intro"
