diff --git a/chrome/android/features/tab_ui/java/src/org/chromium/chrome/browser/tasks/tab_management/TabGridDialogMediator.java b/chrome/android/features/tab_ui/java/src/org/chromium/chrome/browser/tasks/tab_management/TabGridDialogMediator.java
index aa390c2f9c71eb791a6991f55945e9300e34ba62..7430a5295c823ca4fde673f744cfe19710f58a17 100644
--- a/chrome/android/features/tab_ui/java/src/org/chromium/chrome/browser/tasks/tab_management/TabGridDialogMediator.java
+++ b/chrome/android/features/tab_ui/java/src/org/chromium/chrome/browser/tasks/tab_management/TabGridDialogMediator.java
@@ -78,6 +78,8 @@ import java.util.List;
 import java.util.Locale;
 import java.util.Objects;
 
+import org.chromium.chrome.browser.homepage.HomepageManager;
+
 /**
  * A mediator for the TabGridDialog component, responsible for communicating with the components'
  * coordinator as well as managing the business logic for dialog show/hide.
@@ -830,7 +832,12 @@ public class TabGridDialogMediator
             }
 
             if (currentTab == null) {
-                mTabCreatorManager.getTabCreator(filter.isIncognito()).launchNtp();
+                //mTabCreatorManager.getTabCreator(filter.isIncognito()).launchNtp();
+                if (filter.isIncognito())
+                    mTabCreatorManager.getTabCreator(filter.isIncognito()).launchNtp();
+                else
+                    mTabCreatorManager.getTabCreator(filter.isIncognito())
+                            .launchUrl(HomepageManager.getInstance().getHomepageGurlIgnoringEnabledState().getSpec(), TabLaunchType.FROM_CHROME_UI);
                 return;
             }
 
