diff --git a/chrome/browser/ui/android/omnibox/java/src/org/chromium/chrome/browser/omnibox/suggestions/OmniboxSuggestionsDropdown.java b/chrome/browser/ui/android/omnibox/java/src/org/chromium/chrome/browser/omnibox/suggestions/OmniboxSuggestionsDropdown.java
index bac2a418532ea26e9d9551c2737c38ca6a33a789..2c0637637aea40d9a6c48a920b34e9061332f750 100644
--- a/chrome/browser/ui/android/omnibox/java/src/org/chromium/chrome/browser/omnibox/suggestions/OmniboxSuggestionsDropdown.java
+++ b/chrome/browser/ui/android/omnibox/java/src/org/chromium/chrome/browser/omnibox/suggestions/OmniboxSuggestionsDropdown.java
@@ -39,6 +39,9 @@ import org.chromium.ui.KeyboardVisibilityDelegate;
 import org.chromium.ui.base.DeviceFormFactor;
 import org.chromium.ui.base.ViewUtils;
 
+import android.graphics.Rect;
+import org.chromium.base.ContextUtils;
+
 /** A widget for showing a list of omnibox suggestions. */
 public class OmniboxSuggestionsDropdown extends RecyclerView {
     /**
@@ -382,11 +385,29 @@ public class OmniboxSuggestionsDropdown extends RecyclerView {
         mEmbedder.removeAlignmentObserver(mOmniboxAlignmentObserver);
     }
 
+    private void onMeasureBottomToolbar(int widthMeasureSpec, int heightMeasureSpec) {
+        maybeUpdateLayoutParams(0);
+        final Rect mTempRect = new Rect();
+        mEmbedder.getWindowDelegate().getWindowVisibleDisplayFrame(mTempRect);
+        int availableViewportHeight = mTempRect.height();
+        OmniboxAlignment omniboxAlignment = mEmbedder.getCurrentAlignment();
+        int desiredWidth = omniboxAlignment.width;
+        adjustHorizontalPosition();
+        notifyObserversIfViewportHeightChanged(availableViewportHeight);
+        widthMeasureSpec = MeasureSpec.makeMeasureSpec(desiredWidth, MeasureSpec.EXACTLY);
+        heightMeasureSpec = MeasureSpec.makeMeasureSpec(availableViewportHeight,
+                mEmbedder.isTablet() ? MeasureSpec.AT_MOST : MeasureSpec.EXACTLY);
+        super.onMeasure(widthMeasureSpec, heightMeasureSpec);
+    }
     @Override
     protected void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {
         try (TraceEvent tracing = TraceEvent.scoped("OmniboxSuggestionsList.Measure");
                 TimingMetric metric = OmniboxMetrics.recordSuggestionListMeasureTime();
                 TimingMetric metric2 = OmniboxMetrics.recordSuggestionListMeasureWallTime()) {
+            if (ContextUtils.getAppSharedPreferences().getBoolean("enable_bottom_toolbar", false)) {
+                onMeasureBottomToolbar(widthMeasureSpec, heightMeasureSpec);
+                return;
+            }
             OmniboxAlignment omniboxAlignment = mEmbedder.getCurrentAlignment();
             maybeUpdateLayoutParams(omniboxAlignment.top);
             int availableViewportHeight = omniboxAlignment.height;
