diff --git a/chrome/browser/resources/ntp4/BUILD.gn b/chrome/browser/resources/ntp4/BUILD.gn
index e995d2398d9f356162d1f5cad2e03581323dcf3e..1b3176efd596588b87cb3f236fb9a84ce837db76 100644
--- a/chrome/browser/resources/ntp4/BUILD.gn
+++ b/chrome/browser/resources/ntp4/BUILD.gn
@@ -7,7 +7,7 @@ import("//tools/grit/grit_rule.gni")
 import("//tools/typescript/ts_library.gni")
 import("//ui/webui/resources/tools/generate_grd.gni")
 
-assert(is_fuchsia || is_linux || is_mac || is_win)
+assert(true || is_fuchsia || is_linux || is_mac || is_win)
 
 grit("apps_resources") {
   defines = chrome_grit_defines
