diff --git a/chrome/browser/resources/ntp4/BUILD.gn b/chrome/browser/resources/ntp4/BUILD.gn
index a32036ea61566493436ffcf9c89a5c9cb115749f..c4a2948cd763c53daf167b1530302d0273b95100 100644
--- a/chrome/browser/resources/ntp4/BUILD.gn
+++ b/chrome/browser/resources/ntp4/BUILD.gn
@@ -7,7 +7,7 @@ import("//tools/grit/grit_rule.gni")
 import("//tools/typescript/ts_library.gni")
 import("//ui/webui/resources/tools/generate_grd.gni")
 
-assert(is_fuchsia || is_linux || is_mac || is_win)
+assert(true || is_fuchsia || is_linux || is_mac || is_win)
 
 grit("apps_resources") {
   defines = chrome_grit_defines
