diff --git a/chrome/android/features/tab_ui/java/src/org/chromium/chrome/browser/tasks/tab_management/TabGridDialogCoordinator.java b/chrome/android/features/tab_ui/java/src/org/chromium/chrome/browser/tasks/tab_management/TabGridDialogCoordinator.java
index 2595dc5e9dcb49d2fbba2861d262cf56014f812b..c5d0fe5a471ee28c9376799765cf0ea82e479131 100644
--- a/chrome/android/features/tab_ui/java/src/org/chromium/chrome/browser/tasks/tab_management/TabGridDialogCoordinator.java
+++ b/chrome/android/features/tab_ui/java/src/org/chromium/chrome/browser/tasks/tab_management/TabGridDialogCoordinator.java
@@ -35,6 +35,8 @@ import org.chromium.ui.modelutil.PropertyModelChangeProcessor;
 
 import java.util.List;
 
+import org.chromium.base.ContextUtils;
+
 /**
  * A coordinator for TabGridDialog component. Manages the communication with
  * {@link TabListCoordinator} as well as the life-cycle of shared component
@@ -86,13 +88,18 @@ public class TabGridDialogCoordinator implements TabGridDialogMediator.DialogCon
                     animationSourceViewProvider, shareDelegateSupplier, mSnackbarManager,
                     mComponentName);
 
+            int mode = TabUiFeatureUtilities.isTabGroupsAndroidContinuationEnabled(activity)
+                                    && SysUtils.isLowEndDevice()
+                            ? TabListCoordinator.TabListMode.LIST
+                            : TabListCoordinator.TabListMode.GRID;
+            final String active_tabswitcher = ContextUtils.getAppSharedPreferences().getString("active_tabswitcher", "default");
+            if (active_tabswitcher.equals("default") || active_tabswitcher.equals("grid"))
+                mode = TabListCoordinator.TabListMode.GRID;
+
             // TODO(crbug.com/1031349) : Remove the inline mode logic here, make the constructor to
             // take in a mode parameter instead.
             mTabListCoordinator = new TabListCoordinator(
-                    TabUiFeatureUtilities.isTabGroupsAndroidContinuationEnabled(activity)
-                                    && SysUtils.isLowEndDevice()
-                            ? TabListCoordinator.TabListMode.LIST
-                            : TabListCoordinator.TabListMode.GRID,
+                    mode,
                     activity, tabModelSelector,
                     (tabId, thumbnailSize, callback, forceUpdate, writeBack, isSelected)
                             -> {
@@ -150,6 +157,10 @@ public class TabGridDialogCoordinator implements TabGridDialogMediator.DialogCon
             @TabListCoordinator.TabListMode
             int mode = SysUtils.isLowEndDevice() ? TabListCoordinator.TabListMode.LIST
                                                  : TabListCoordinator.TabListMode.GRID;
+            final String active_tabswitcher = ContextUtils.getAppSharedPreferences().getString("active_tabswitcher", "default");
+            if (active_tabswitcher.equals("default") || active_tabswitcher.equals("grid"))
+                mode = TabListCoordinator.TabListMode.GRID;
+
             mTabSelectionEditorCoordinator = new TabSelectionEditorCoordinator(mActivity,
                     mDialogView.findViewById(R.id.dialog_container_view), mTabModelSelector,
                     mTabContentManager, mTabListCoordinator::setRecyclerViewPosition, mode,
