diff --git a/chrome/browser/resources/media_router/internals/BUILD.gn b/chrome/browser/resources/media_router/internals/BUILD.gn
index 4315ed31a40b7e8d7efe164ebd4a6aea095422f4..b0c46b6ba586f2647262441dcda4bd880e850746 100644
--- a/chrome/browser/resources/media_router/internals/BUILD.gn
+++ b/chrome/browser/resources/media_router/internals/BUILD.gn
@@ -4,7 +4,7 @@
 
 import("//tools/typescript/ts_library.gni")
 
-assert(!is_android)
+# assert(!is_android)
 
 ts_library("build_ts") {
   root_dir = "."
