diff --git a/chrome/browser/resources/media_router/internals/BUILD.gn b/chrome/browser/resources/media_router/internals/BUILD.gn
index ebd904ddf87cc4788a84fb7b68e6c221b88183dc..5a4c95f49ecb7973ac370c80e4f05d437447665b 100644
--- a/chrome/browser/resources/media_router/internals/BUILD.gn
+++ b/chrome/browser/resources/media_router/internals/BUILD.gn
@@ -4,7 +4,7 @@
 
 import("//tools/typescript/ts_library.gni")
 
-assert(!is_android)
+# assert(!is_android)
 
 ts_library("build_ts") {
   root_dir = "."
