diff --git a/chrome/browser/media/router/discovery/access_code/BUILD.gn b/chrome/browser/media/router/discovery/access_code/BUILD.gn
index 6364bc527d5d77aca6284f336f68196a11d2181a..1ef8d36d77382f78096eb680e227e8e1e1a22a9c 100644
--- a/chrome/browser/media/router/discovery/access_code/BUILD.gn
+++ b/chrome/browser/media/router/discovery/access_code/BUILD.gn
@@ -27,7 +27,7 @@ static_library("access_code_cast_feature") {
   ]
 }
 
-if (!is_android) {
+if (true || !is_android) {
   static_library("access_code_sink_service") {
     sources = [
       "access_code_cast_discovery_interface.cc",
