diff --git a/chrome/common/accessibility/BUILD.gn b/chrome/common/accessibility/BUILD.gn
index 67b77a72bf51e77f45be42b974be71e25c59e93f..35be9e2c84bc36e2a813a60351da61cdca94b4cd 100644
--- a/chrome/common/accessibility/BUILD.gn
+++ b/chrome/common/accessibility/BUILD.gn
@@ -4,7 +4,7 @@
 
 import("//mojo/public/tools/bindings/mojom.gni")
 
-assert(!is_android)
+#assert(!is_android)
 
 mojom("mojo_bindings") {
   sources = [ "read_anything.mojom" ]
