diff --git a/chrome/android/java/src/org/chromium/chrome/browser/gesturenav/NavigationHandler.java b/chrome/android/java/src/org/chromium/chrome/browser/gesturenav/NavigationHandler.java
index 1a966aea0c507039de620a9697e883b505e2d94b..60ddf0a44409d22ebbfbb0d5466b3b6f07032aae 100644
--- a/chrome/android/java/src/org/chromium/chrome/browser/gesturenav/NavigationHandler.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/gesturenav/NavigationHandler.java
@@ -55,7 +55,7 @@ class NavigationHandler implements TouchEventObserver {
 
     // Weighted value to determine when to trigger an edge swipe. Initial scroll
     // vector should form 30 deg or below to initiate swipe action.
-    private static final float WEIGTHED_TRIGGER_THRESHOLD = 1.73f;
+    private static final float WEIGTHED_TRIGGER_THRESHOLD = 0.43f;
 
     // |EDGE_WIDTH_DP| in physical pixel.
     private final float mEdgeWidthPx;
