diff --git a/chrome/browser/ui/android/omnibox/java/src/org/chromium/chrome/browser/omnibox/UrlBar.java b/chrome/browser/ui/android/omnibox/java/src/org/chromium/chrome/browser/omnibox/UrlBar.java
index 5c7d412da89383551837c8c869e707e41c3bb814..f52be0c688f0b0b51a732c827f7fb760ce4c7a48 100644
--- a/chrome/browser/ui/android/omnibox/java/src/org/chromium/chrome/browser/omnibox/UrlBar.java
+++ b/chrome/browser/ui/android/omnibox/java/src/org/chromium/chrome/browser/omnibox/UrlBar.java
@@ -1009,6 +1009,7 @@ public class UrlBar extends AutocompleteEditText {
                     break;
                 }
             }
+            if (rtlStartIndexForEndingRun < 0) { rtlStartIndexForEndingRun = 0; };
             float width =
                     textLayout
                             .getPaint()
