diff --git a/chrome/browser/ui/android/theme/BUILD.gn b/chrome/browser/ui/android/theme/BUILD.gn
index 4f9a492455591e12d96cd9322dbe9e764e59d1b1..a99d48c78099e2e38ba2b18fc39164c4448cf797 100644
--- a/chrome/browser/ui/android/theme/BUILD.gn
+++ b/chrome/browser/ui/android/theme/BUILD.gn
@@ -26,6 +26,7 @@ android_library("java") {
     "//third_party/androidx:androidx_appcompat_appcompat_resources_java",
     "//third_party/androidx:androidx_core_core_java",
     "//ui/android:ui_no_recycler_view_java",
+    "//url:gurl_java",
   ]
   resources_package = "org.chromium.chrome.browser.theme"
 }
