diff --git a/chrome/browser/ui/android/night_mode/java/src/org/chromium/chrome/browser/night_mode/WebContentsDarkModeMessageController.java b/chrome/browser/ui/android/night_mode/java/src/org/chromium/chrome/browser/night_mode/WebContentsDarkModeMessageController.java
index b6fc5f23ddc6f23eebd8324a25965bf8c8a81602..cbbeec0c012b9d2c55c1a8e5f2b2eb42df2977a2 100644
--- a/chrome/browser/ui/android/night_mode/java/src/org/chromium/chrome/browser/night_mode/WebContentsDarkModeMessageController.java
+++ b/chrome/browser/ui/android/night_mode/java/src/org/chromium/chrome/browser/night_mode/WebContentsDarkModeMessageController.java
@@ -60,6 +60,8 @@ public class WebContentsDarkModeMessageController {
      * @return Whether or not the user education message should be shown.
      */
     private static boolean shouldSendMessage(Profile profile, Context context) {
+        if (true)
+            return false;
         // Only send message if the feature is enabled and the message has not yet been shown.
         Tracker tracker = TrackerFactory.getTrackerForProfile(profile);
         boolean featureEnabled = WebContentsDarkModeController.isFeatureEnabled(context, profile);
