diff --git a/chrome/android/chrome_public_apk_tmpl.gni b/chrome/android/chrome_public_apk_tmpl.gni
index 56949f4bb7496f40204bfc72f74123104f5ea1ff..d2cd4ca2295fd188f36705ba1631aa8aa4a4592f 100644
--- a/chrome/android/chrome_public_apk_tmpl.gni
+++ b/chrome/android/chrome_public_apk_tmpl.gni
@@ -326,6 +326,17 @@ template("chrome_public_common_apk_or_module_tmpl") {
       strip_unused_resources = is_official_build
     }
 
+    if (_is_bundle) {
+        # Resources config for blocklisting resource names from obfuscation
+        resources_config_paths = [
+          "//android_webview/aapt2.config",
+          "//chrome/android/aapt2.config",
+        ]
+        if (defined(invoker.resources_config_paths)) {
+          resources_config_paths += invoker.resources_config_paths
+        }
+    }
+
     if (!defined(aapt_locale_allowlist)) {
       # Include resource strings files only for supported locales.
       aapt_locale_allowlist = platform_pak_locales
@@ -386,6 +397,7 @@ template("chrome_public_common_apk_or_module_tmpl") {
         proguard_configs = []
       }
       proguard_configs += [ "//chrome/android/proguard/main.flags" ]
+      proguard_configs += [ "//mises/android/java/proguard.flags" ]
     }
 
     if (use_chromium_linker) {
@@ -712,7 +724,7 @@ template("monochrome_public_common_apk_or_module_tmpl") {
       }
 
       # We only optimize resources in bundles.
-      if (_is_bundle_module) {
+      if (_is_bundle_module && false) {
         # Resources config for blocklisting resource names from obfuscation
         resources_config_paths = [
           "//android_webview/aapt2.config",
@@ -751,6 +763,7 @@ template("monochrome_public_common_apk_or_module_tmpl") {
         if (defined(invoker.proguard_configs)) {
           proguard_configs += invoker.proguard_configs
         }
+        proguard_configs += [ "//mises/android/java/proguard.flags" ]
         if (is_monochrome) {
           proguard_configs +=
               [ "//android_webview/nonembedded/java/proguard.flags" ]
