diff --git a/chrome/browser/enterprise/connectors/common.cc b/chrome/browser/enterprise/connectors/common.cc
index d230ecf5dc44264b279aabe4b0cedab3cc87fe34..30ef316d5a77f3a13476bf7aca6f6663f3e4ee14 100644
--- a/chrome/browser/enterprise/connectors/common.cc
+++ b/chrome/browser/enterprise/connectors/common.cc
@@ -222,6 +222,7 @@ void ShowDownloadReviewDialog(const std::u16string& filename,
                               content::WebContents* web_contents,
                               base::OnceClosure keep_closure,
                               base::OnceClosure discard_closure) {
+#if !BUILDFLAG(IS_ANDROID)
   auto state = FinalContentAnalysisResult::FAILURE;
   download::DownloadDangerType danger_type = download_item->GetDangerType();
 
@@ -266,6 +267,7 @@ void ShowDownloadReviewDialog(const std::u16string& filename,
       true,  // Downloads are always cloud-based for now.
       web_contents, safe_browsing::DeepScanAccessPoint::DOWNLOAD,
       /* file_count */ 1, state, download_item);
+#endif
 }
 
 bool IsResumableUpload(const BinaryUploadService::Request& request) {
