diff --git a/chrome/browser/lens/region_search/lens_region_search_controller.cc b/chrome/browser/lens/region_search/lens_region_search_controller.cc
index 163bb43fe1a22c3b5ecd58c9cf482bcf57950b14..2a154d21eb83a070828adfbac0a711cf0e884a00 100644
--- a/chrome/browser/lens/region_search/lens_region_search_controller.cc
+++ b/chrome/browser/lens/region_search/lens_region_search_controller.cc
@@ -71,6 +71,7 @@ void LensRegionSearchController::Start(
   if (use_fullscreen_capture) {
     screenshot_flow_->StartFullscreenCapture(std::move(callback));
   } else {
+#if !BUILDFLAG(IS_ANDROID)
     // Create user education bubble anchored to the toolbar container.
     // This is only done for non-fulllscreen capture.
     bubble_widget_ = lens::OpenLensRegionSearchInstructions(
@@ -80,6 +81,7 @@ void LensRegionSearchController::Start(
         base::BindOnce(&LensRegionSearchController::Escape,
                        base::Unretained(this)));
     bubble_widget_->Show();
+#endif
     screenshot_flow_->Start(std::move(callback));
   }
 }
