diff --git a/chrome/browser/resources/key_value_pair_viewer_shared/BUILD.gn b/chrome/browser/resources/key_value_pair_viewer_shared/BUILD.gn
index 589e984f4a988c92ff9f74992611f0a7b4d67ac2..118a5c6a5059cf39e0ab5f485f4b1823e8158de5 100644
--- a/chrome/browser/resources/key_value_pair_viewer_shared/BUILD.gn
+++ b/chrome/browser/resources/key_value_pair_viewer_shared/BUILD.gn
@@ -4,7 +4,7 @@
 
 import("//ui/webui/resources/tools/build_webui.gni")
 
-assert(!is_android)
+#assert(!is_android)
 
 build_webui("build") {
   grd_prefix = "key_value_pair_viewer_shared"
