diff --git a/chrome/android/java/src/org/chromium/chrome/browser/base/SplitChromeApplication.java b/chrome/android/java/src/org/chromium/chrome/browser/base/SplitChromeApplication.java
index c8dad9e6424d772b7fdbd55279c35bd4b7af3313..a891d65e6b7f10ac63dd3491b55966f23251e2f0 100644
--- a/chrome/android/java/src/org/chromium/chrome/browser/base/SplitChromeApplication.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/base/SplitChromeApplication.java
@@ -18,6 +18,7 @@ import org.chromium.base.JNIUtils;
 import org.chromium.base.TraceEvent;
 import org.chromium.base.metrics.RecordHistogram;
 import org.chromium.build.annotations.IdentifierNameString;
+import org.chromium.base.MisesVpnUtils;
 
 /**
  * Application class to use for Chrome when //chrome code is in an isolated split. This class will
@@ -50,6 +51,7 @@ public class SplitChromeApplication extends SplitCompatApplication {
     public void onCreate() {
         finishPreload(CHROME_SPLIT_NAME);
         super.onCreate();
+	MisesVpnUtils.initApplication(this);
     }
 
     @Override
