diff --git a/chrome/browser/ui/webui/tab_search/BUILD.gn b/chrome/browser/ui/webui/tab_search/BUILD.gn
index 48046ed8a910a82bc3d6656d8f067b7d2464518e..ab8d226bdd891b5307e0ace24948fcaee5c2380d 100644
--- a/chrome/browser/ui/webui/tab_search/BUILD.gn
+++ b/chrome/browser/ui/webui/tab_search/BUILD.gn
@@ -4,7 +4,7 @@
 
 import("//mojo/public/tools/bindings/mojom.gni")
 
-assert(is_linux || is_chromeos || is_win || is_mac || is_fuchsia)
+#assert(is_linux || is_chromeos || is_win || is_mac || is_fuchsia)
 
 mojom("mojo_bindings") {
   sources = [ "tab_search.mojom" ]
