diff --git a/chrome/browser/extensions/chrome_component_extension_resource_manager.cc b/chrome/browser/extensions/chrome_component_extension_resource_manager.cc
index 9515376e628370c83e99000dd85b275d459074ce..cac6ba4ea784a85e675a9154fc132487835cf805 100644
--- a/chrome/browser/extensions/chrome_component_extension_resource_manager.cc
+++ b/chrome/browser/extensions/chrome_component_extension_resource_manager.cc
@@ -158,6 +158,7 @@ ChromeComponentExtensionResourceManager::Data::Data() {
         std::move(pdf_viewer_replacements);
   }
 #endif
+  MISES_CHROME_COMPONENT_EXTENSION_RESOURCE_MANAGER_DATA_DATA
 }
 
 void ChromeComponentExtensionResourceManager::Data::AddComponentResourceEntries(
