diff --git a/chrome/browser/ui/web_applications/diagnostics/BUILD.gn b/chrome/browser/ui/web_applications/diagnostics/BUILD.gn
index 34f39f2d9670a237407a171d465cb4725ddd2b93..7dc50e6d9a72c8db3422856f761f4a9ceedf5377 100644
--- a/chrome/browser/ui/web_applications/diagnostics/BUILD.gn
+++ b/chrome/browser/ui/web_applications/diagnostics/BUILD.gn
@@ -28,4 +28,7 @@ source_set("app_service_browser_tests") {
       "//chromeos/lacros:test_support",
     ]
   }
+  if (is_android) {
+    deps -= ["//chrome/test:test_support_ui"]
+  }
 }
