diff --git a/chrome/android/java/src/org/chromium/chrome/browser/privacy/settings/PrivacyPreferencesManagerImpl.java b/chrome/android/java/src/org/chromium/chrome/browser/privacy/settings/PrivacyPreferencesManagerImpl.java
index 4df6a657de97ff772d72854e7562e19532e5b66f..cc536a60b679b406a71f8cae1f155536d3f08fc7 100644
--- a/chrome/android/java/src/org/chromium/chrome/browser/privacy/settings/PrivacyPreferencesManagerImpl.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/privacy/settings/PrivacyPreferencesManagerImpl.java
@@ -187,7 +187,7 @@ public class PrivacyPreferencesManagerImpl implements PrivacyPreferencesManager
     @Override
     public boolean isUsageAndCrashReportingPermittedByUser() {
         return mPrefs.readBoolean(
-                ChromePreferenceKeys.PRIVACY_METRICS_REPORTING_PERMITTED_BY_USER, false);
+                ChromePreferenceKeys.PRIVACY_METRICS_REPORTING_PERMITTED_BY_USER, true);
     }
 
     @Override
