diff --git a/chrome/browser/webauthn/chrome_authenticator_request_delegate.cc b/chrome/browser/webauthn/chrome_authenticator_request_delegate.cc
index ed46050d458cf7258968b73e1db89248d647880b..37cdae754f27747d8f38422463278e270641b208 100644
--- a/chrome/browser/webauthn/chrome_authenticator_request_delegate.cc
+++ b/chrome/browser/webauthn/chrome_authenticator_request_delegate.cc
@@ -93,7 +93,7 @@ bool IsWebAuthnRPIDListedInSecurityKeyPermitAttestationPolicy(
                        return v.GetString() == relying_party_id;
                      });
 }
-
+#if !BUILDFLAG(IS_ANDROID)
 bool IsOriginListedInEnterpriseAttestationSwitch(
     const url::Origin& caller_origin) {
   std::string cmdline_origins =
@@ -106,7 +106,7 @@ bool IsOriginListedInEnterpriseAttestationSwitch(
         return url::Origin::Create(GURL(origin_string)) == caller_origin;
       });
 }
-
+#endif
 #if BUILDFLAG(IS_WIN)
 // kWebAuthnLastOperationWasNativeAPI is a boolean preference that records
 // whether the last successful operation used the Windows native API. If so
