diff --git a/chrome/browser/ui/android/toolbar/java/src/org/chromium/chrome/browser/toolbar/top/TopToolbarOverlayProperties.java b/chrome/browser/ui/android/toolbar/java/src/org/chromium/chrome/browser/toolbar/top/TopToolbarOverlayProperties.java
index 668a85060c7823b080debb8c99fb9cb8600550e9..93a2c685589646b2422343d9663d09f313127c27 100644
--- a/chrome/browser/ui/android/toolbar/java/src/org/chromium/chrome/browser/toolbar/top/TopToolbarOverlayProperties.java
+++ b/chrome/browser/ui/android/toolbar/java/src/org/chromium/chrome/browser/toolbar/top/TopToolbarOverlayProperties.java
@@ -14,18 +14,14 @@ import org.chromium.ui.modelutil.PropertyModel.WritableObjectPropertyKey;
 
 /** Properties associated with the top toolbar's composited layer. */
 public class TopToolbarOverlayProperties {
-    /** Whether the URL should be hidden when rendered. */
-    public static final WritableBooleanPropertyKey ANONYMIZE = new WritableBooleanPropertyKey();
 
-    /** A "struct" for progress bar drawing info. */
-    public static final WritableObjectPropertyKey<DrawingInfo> PROGRESS_BAR_INFO =
-            new WritableObjectPropertyKey<>(true);
+    public static final WritableBooleanPropertyKey ANONYMIZE = new WritableBooleanPropertyKey();
 
     /** The ID of the toolbar's texture resource. */
     public static final ReadableIntPropertyKey RESOURCE_ID = new ReadableIntPropertyKey();
 
-    /** Whether the shadow under the toolbar should be visible. */
-    public static final WritableBooleanPropertyKey SHOW_SHADOW = new WritableBooleanPropertyKey();
+    /** The texture resource used to draw the location/URL bar. */
+    public static final ReadableIntPropertyKey URL_BAR_RESOURCE_ID = new ReadableIntPropertyKey();
 
     /** The background color of the toolbar. */
     public static final WritableIntPropertyKey TOOLBAR_BACKGROUND_COLOR =
@@ -33,21 +29,24 @@ public class TopToolbarOverlayProperties {
 
     /** The color of the location/URL bar. */
     public static final WritableIntPropertyKey URL_BAR_COLOR = new WritableIntPropertyKey();
+    
+    /** The current x offset of the top toolbar. */
+    public static final WritableFloatPropertyKey X_OFFSET = new WritableFloatPropertyKey();
 
-    /** The texture resource used to draw the location/URL bar. */
-    public static final ReadableIntPropertyKey URL_BAR_RESOURCE_ID = new ReadableIntPropertyKey();
+    /** The current offset of the top toolbar. */
+    public static final WritableFloatPropertyKey Y_OFFSET = new WritableFloatPropertyKey();
+
+    /** Whether the shadow under the toolbar should be visible. */
+    public static final WritableBooleanPropertyKey SHOW_SHADOW = new WritableBooleanPropertyKey();
 
     /** Whether the layer should be visible. */
     public static final WritableBooleanPropertyKey VISIBLE = new WritableBooleanPropertyKey();
 
-    /** The current x offset of the top toolbar. */
-    public static final WritableFloatPropertyKey X_OFFSET = new WritableFloatPropertyKey();
-
-    /** The current y offset of the top toolbar. */
-    public static final WritableFloatPropertyKey Y_OFFSET = new WritableFloatPropertyKey();
+    /** A "struct" for progress bar drawing info. */
+    public static final WritableObjectPropertyKey<DrawingInfo> PROGRESS_BAR_INFO =
+            new WritableObjectPropertyKey<>(true);
 
     public static final PropertyKey[] ALL_KEYS =
-            new PropertyKey[] {ANONYMIZE, PROGRESS_BAR_INFO, RESOURCE_ID, SHOW_SHADOW,
-                    TOOLBAR_BACKGROUND_COLOR, URL_BAR_COLOR, URL_BAR_RESOURCE_ID, VISIBLE,
-                    X_OFFSET, Y_OFFSET};
+            new PropertyKey[] {RESOURCE_ID, URL_BAR_RESOURCE_ID, TOOLBAR_BACKGROUND_COLOR,
+                    URL_BAR_COLOR, X_OFFSET, Y_OFFSET, SHOW_SHADOW, VISIBLE, PROGRESS_BAR_INFO};
 }
