diff --git a/chrome/browser/ui/webui/side_panel/user_notes/BUILD.gn b/chrome/browser/ui/webui/side_panel/user_notes/BUILD.gn
index f7c09c2fb7dc09074d2f8ec74ecc0b509a3337b1..3789a1b3936a8034ade4b777b0b25447e6e7a7e5 100644
--- a/chrome/browser/ui/webui/side_panel/user_notes/BUILD.gn
+++ b/chrome/browser/ui/webui/side_panel/user_notes/BUILD.gn
@@ -4,7 +4,7 @@
 
 import("//mojo/public/tools/bindings/mojom.gni")
 
-assert(!is_android)
+#assert(!is_android)
 
 mojom("mojo_bindings") {
   sources = [ "user_notes.mojom" ]
