diff --git a/chrome/test/base/test_browser_window.h b/chrome/test/base/test_browser_window.h
index 6444ae54bfbe52853350809549c6da74bc50a209..199d642d8efbe36e6f24e2bd0d050a85d8b7a85d 100644
--- a/chrome/test/base/test_browser_window.h
+++ b/chrome/test/base/test_browser_window.h
@@ -21,6 +21,7 @@
 #include "chrome/browser/ui/browser_window.h"
 #include "chrome/browser/ui/location_bar/location_bar.h"
 #include "chrome/browser/ui/translate/partial_translate_bubble_model.h"
+#include "chrome/browser/ui/browser_list_observer.h"
 #include "chrome/common/buildflags.h"
 #include "components/user_education/common/feature_promo_controller.h"
 #include "components/user_education/common/new_badge_controller.h"
