diff --git a/chrome/browser/ui/android/device_dialog/usb_chooser_dialog_android.cc b/chrome/browser/ui/android/device_dialog/usb_chooser_dialog_android.cc
index 6946406d977f169667c76be801d32d54618ef788..c14b5f2ff8df8908f70730454625b6a5bbcfb13c 100644
--- a/chrome/browser/ui/android/device_dialog/usb_chooser_dialog_android.cc
+++ b/chrome/browser/ui/android/device_dialog/usb_chooser_dialog_android.cc
@@ -71,8 +71,15 @@ UsbChooserDialogAndroid::CreateInternal(
       content::WebContents::FromRenderFrameHost(render_frame_host);
 
   // Create (and show) the UsbChooser dialog.
+  auto *native_window = web_contents->GetNativeView()->GetWindowAndroid();
+  if (native_window == nullptr) {
+      native_window = web_contents->GetTopLevelNativeWindow();
+  }
+  if (native_window == nullptr) {
+      return nullptr;
+  }
   base::android::ScopedJavaLocalRef<jobject> window_android =
-      web_contents->GetNativeView()->GetWindowAndroid()->GetJavaObject();
+      native_window->GetJavaObject();
   JNIEnv* env = base::android::AttachCurrentThread();
   // Permission delegation means the permission request should be
   // attributed to the main frame.
