diff --git a/chrome/browser/ui/android/omnibox/java/src/org/chromium/chrome/browser/omnibox/AutocompleteEditText.java b/chrome/browser/ui/android/omnibox/java/src/org/chromium/chrome/browser/omnibox/AutocompleteEditText.java
index 540c97ada8406a167164cf71b7639476e58fd4ff..2eae1da6c79b7dd95e15bd41f68907648c536c02 100644
--- a/chrome/browser/ui/android/omnibox/java/src/org/chromium/chrome/browser/omnibox/AutocompleteEditText.java
+++ b/chrome/browser/ui/android/omnibox/java/src/org/chromium/chrome/browser/omnibox/AutocompleteEditText.java
@@ -249,7 +249,11 @@ public class AutocompleteEditText extends VerticallyFixedEditText
             if (DEBUG) Log.i(TAG, "Ignoring accessibility event from autocomplete.");
             return;
         }
-        super.sendAccessibilityEventUnchecked(event);
+        try {
+            super.sendAccessibilityEventUnchecked(event);
+        } catch (IllegalStateException e) {
+            // accessibility is off.
+        }
     }
 
     @Override
