diff --git a/chrome/browser/ui/views/extensions/extensions_toolbar_container.cc b/chrome/browser/ui/views/extensions/extensions_toolbar_container.cc
index 187ed3e88d294b5fbb41d4aa0d89d70d53c362c9..a31223b657fed9033289077ef798d4d7a5e5e908 100644
--- a/chrome/browser/ui/views/extensions/extensions_toolbar_container.cc
+++ b/chrome/browser/ui/views/extensions/extensions_toolbar_container.cc
@@ -56,6 +56,9 @@ base::OnceClosure& GetOnVisibleCallbackForTesting() {
 // overlapping with the extension popup. The media picker dialog has been
 // identified to be susceptible. See crbug.com/1300006.
 bool HasPossiblyOverlappingSecurityUI(Browser* browser) {
+#if BUILDFLAG(IS_ANDROID)
+  return false;
+#else
   views::ElementTrackerViews::ViewList media_picker_dialogs =
       views::ElementTrackerViews::GetInstance()->GetAllMatchingViews(
           DesktopMediaPickerDialogView::kDesktopMediaPickerDialogViewIdentifier,
@@ -66,6 +69,7 @@ bool HasPossiblyOverlappingSecurityUI(Browser* browser) {
                        views::Widget* dialog_widget = dialog_view->GetWidget();
                        return dialog_widget && dialog_widget->IsVisible();
                      });
+#endif
 }
 
 }  // namespace
