diff --git a/chrome/browser/extensions/api/permissions/permissions_api.cc b/chrome/browser/extensions/api/permissions/permissions_api.cc
index b5da189432e23255316f04cb7339532e83e012d2..2c3693e193e856ce10f6ae639e1d1657501198de 100644
--- a/chrome/browser/extensions/api/permissions/permissions_api.cc
+++ b/chrome/browser/extensions/api/permissions/permissions_api.cc
@@ -311,9 +311,11 @@ ExtensionFunction::ResponseAction PermissionsRequestFunction::Run() {
 
   gfx::NativeWindow native_window =
       ChromeExtensionFunctionDetails(this).GetNativeWindowForUI();
+#if !BUILDFLAG(IS_ANDROID)
   if (!native_window && g_dialog_action == DialogAction::kDefault) {
     return RespondNow(Error("Could not find an active window."));
   }
+#endif
 
   std::optional<api::permissions::Request::Params> params =
       api::permissions::Request::Params::Create(args());
