diff --git a/chrome/browser/BUILD.gn b/chrome/browser/BUILD.gn
index 7f1c26990d8d43e92615bd3a5f4046d121d6ac6b..c03f0aabab4241e2609c31e4ef249392726df333 100644
--- a/chrome/browser/BUILD.gn
+++ b/chrome/browser/BUILD.gn
@@ -2523,6 +2523,7 @@ static_library("browser") {
       "signin/wait_for_network_callback_helper_chrome.h",
     ]
   }
+  import("//mises/browser/sources.gni") sources += mises_chrome_browser_sources deps += mises_chrome_browser_deps public_deps += mises_chrome_browser_public_deps
 
   if (is_android) {
     sources += [
@@ -3475,7 +3476,8 @@ static_library("browser") {
       ]
       deps += [ "//chrome/android/modules/dev_ui/provider:native" ]
     }
-  } else {
+  }
+  if (true) {
     #!is_android
     sources += [
       "accessibility/ax_main_node_annotator_controller.cc",
@@ -4435,7 +4437,7 @@ static_library("browser") {
         ]
       }
     }
-    if (!is_chromeos_ash) {
+    if (is_android || !is_chromeos_ash) {
       sources += [
         "accessibility/soda_installer_impl.cc",
         "accessibility/soda_installer_impl.h",
@@ -4486,7 +4488,7 @@ static_library("browser") {
       ]
     }
 
-    if (!is_win) {
+    if (!is_android && !is_win) {
       # On Windows, the hashes are embedded in //chrome:chrome_initial rather
       # than here in :chrome_dll.
       deps += [ "//chrome:packed_resources_integrity_header" ]
@@ -4511,6 +4513,9 @@ static_library("browser") {
         "component_updater/hyphenation_component_installer.h",
       ]
     }
+    sources -= mises_chrome_browser_exception_sources
+    deps -= mises_chrome_browser_exception_deps
+    allow_circular_includes_from -= mises_chrome_browser_exception_allow_circular_includes_from
   }
 
   if (is_chromeos_ash) {
@@ -8061,7 +8066,7 @@ static_library("browser") {
     }
   }
 
-  if (enable_session_service) {
+  if (true || enable_session_service) {
     sources += [
       "sessions/app_session_service.cc",
       "sessions/app_session_service.h",
@@ -8554,7 +8559,8 @@ static_library("browser_generated_files") {
       "//chrome/browser/android/webapk/proto",
       "//chrome/browser/ui/webui/feed_internals:mojo_bindings",
     ]
-  } else {
+  }# else {
+  if (true) {
     public_deps += [
       "//chrome/browser/cart:mojo_bindings",
       "//chrome/browser/companion/core/mojom:mojo_bindings",
