diff --git a/chrome/browser/preferences/android/java/src/org/chromium/chrome/browser/preferences/LegacyChromePreferenceKeys.java b/chrome/browser/preferences/android/java/src/org/chromium/chrome/browser/preferences/LegacyChromePreferenceKeys.java
index cbeb37f164f7aa1f116e190ea0bac5e13be72884..0278a2661b415295effe97f1ff3c4764733f573f 100644
--- a/chrome/browser/preferences/android/java/src/org/chromium/chrome/browser/preferences/LegacyChromePreferenceKeys.java
+++ b/chrome/browser/preferences/android/java/src/org/chromium/chrome/browser/preferences/LegacyChromePreferenceKeys.java
@@ -142,6 +142,9 @@ public class LegacyChromePreferenceKeys {
                 ChromePreferenceKeys.UI_THEME_SETTING,
                 ChromePreferenceKeys.VERIFIED_DIGITAL_ASSET_LINKS,
                 ChromePreferenceKeys.WEBAPK_UNINSTALLED_PACKAGES,
+                ChromePreferenceKeys.DARKEN_WEBSITES_ENABLED,
+                ChromePreferenceKeys.MISES_USER_INFO,
+                ChromePreferenceKeys.MISES_FIRST_RUN,
                 ChromePreferenceKeys.KEY_ZERO_SUGGEST_LIST_SIZE
         );
         // clang-format on
