diff --git a/chrome/browser/ui/web_applications/diagnostics/BUILD.gn b/chrome/browser/ui/web_applications/diagnostics/BUILD.gn
index bcf19eb320b8d4e7d8f08190558fb40cc4a6650a..8aba3dd6df77f31249b1538d41686bcc3c280934 100644
--- a/chrome/browser/ui/web_applications/diagnostics/BUILD.gn
+++ b/chrome/browser/ui/web_applications/diagnostics/BUILD.gn
@@ -30,4 +30,7 @@ source_set("app_service_browser_tests") {
       "//chromeos/lacros:test_support",
     ]
   }
+  if (is_android) {
+    deps -= ["//chrome/test:test_support_ui"]
+  }
 }
