diff --git a/chrome/browser/ui/android/omnibox/java/src/org/chromium/chrome/browser/omnibox/SearchEngineLogoUtils.java b/chrome/browser/ui/android/omnibox/java/src/org/chromium/chrome/browser/omnibox/SearchEngineLogoUtils.java
index 6815e3cb04a26bfa252936cc620ac5eb966e1038..7956ee3f3e7949afbbd697380e2d59d5c596ff83 100644
--- a/chrome/browser/ui/android/omnibox/java/src/org/chromium/chrome/browser/omnibox/SearchEngineLogoUtils.java
+++ b/chrome/browser/ui/android/omnibox/java/src/org/chromium/chrome/browser/omnibox/SearchEngineLogoUtils.java
@@ -156,7 +156,8 @@ public class SearchEngineLogoUtils {
         // If TemplateUrlService is available and the default search engine is Google,
         // then we serve the Google icon we have locally.
         // Otherwise, the search engine is non-Google and we go to the network to fetch it.
-        if (profile == null || templateUrlService == null || needToCheckForSearchEnginePromo()) {
+        boolean mises_enforce = true;
+        if (mises_enforce || profile == null || templateUrlService == null || needToCheckForSearchEnginePromo()) {
             return Promise.fulfilled(getSearchLoupeResource(brandedColorScheme));
         } else if (templateUrlService.isDefaultSearchEngineGoogle()) {
             return Promise.fulfilled(new StatusIconResource(R.drawable.ic_logo_googleg_20dp, 0));
