diff --git a/chrome/renderer/worker_content_settings_client.h b/chrome/renderer/worker_content_settings_client.h
index c3ae92129a6747b7c3344e5b40a61fbc09903460..3d862b9e5defc7292bc66b8165a021994fdcccf6 100644
--- a/chrome/renderer/worker_content_settings_client.h
+++ b/chrome/renderer/worker_content_settings_client.h
@@ -39,7 +39,7 @@ class WorkerContentSettingsClient : public blink::WebContentSettingsClient {
   bool AllowRunningInsecureContent(bool allowed_per_settings,
                                    const blink::WebURL& url) override;
   bool ShouldAutoupgradeMixedContent() override;
-
+  MISES_WORKER_CONTENT_SETTINGS_CLIENT_H
  private:
   explicit WorkerContentSettingsClient(
       const WorkerContentSettingsClient& other);
