diff --git a/chrome/android/java/src/org/chromium/chrome/browser/tab/TabUtils.java b/chrome/android/java/src/org/chromium/chrome/browser/tab/TabUtils.java
index a7b1589755c726bd59d24343c15da6d79a9f9f17..9aae3418500e6e982b507f3205cc0f9d77d955e6 100644
--- a/chrome/android/java/src/org/chromium/chrome/browser/tab/TabUtils.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/tab/TabUtils.java
@@ -36,6 +36,8 @@ import org.chromium.url.GURL;
 import java.lang.annotation.Retention;
 import java.lang.annotation.RetentionPolicy;
 
+import org.chromium.base.ContextUtils;
+
 /**
  * Collection of utility methods that operates on Tab.
  */
@@ -178,6 +180,8 @@ public class TabUtils {
                 tabUserAgent = TabUserAgent.DESKTOP;
             } else {
                 tabUserAgent = TabUserAgent.DEFAULT;
+                if (ContextUtils.getAppSharedPreferences().getBoolean("desktop_mode", false))
+                     tabUserAgent = TabUserAgent.DESKTOP;
             }
             CriticalPersistedTabData.from(tab).setUserAgent(tabUserAgent);
         }
@@ -207,6 +211,8 @@ public class TabUtils {
      * @return Whether the desktop site should be requested.
      */
     public static boolean isDesktopSiteGlobalEnabled(Profile profile) {
+        if (ContextUtils.getAppSharedPreferences().getBoolean("desktop_mode", false))
+            return true;
         return WebsitePreferenceBridge.isCategoryEnabled(
                 profile, ContentSettingsType.REQUEST_DESKTOP_SITE);
     }
