diff --git a/chrome/chrome_paks.gni b/chrome/chrome_paks.gni
index f98782561d861e169ed5e515bd45c174d718d7a2..26165c1a56aafad49416a5908ae416aa22475ca6 100644
--- a/chrome/chrome_paks.gni
+++ b/chrome/chrome_paks.gni
@@ -154,7 +154,7 @@ template("chrome_extra_paks") {
       sources += invoker.additional_paks
     }
 
-    if (!is_android) {
+    if (!is_android || is_android) {
       # New paks should be added here by default.
       sources += [
         "$root_gen_dir/chrome/about_sys_resources.pak",
@@ -209,8 +209,8 @@ template("chrome_extra_paks") {
         "//chrome/browser/resources:dev_ui_paks",
         "//chrome/browser/resources/commerce/product_specifications:resources",
         "//chrome/browser/resources/data_sharing:resources",
-        "//chrome/browser/resources/lens/overlay:resources",
-        "//chrome/browser/resources/lens/overlay/search_bubble:resources",
+        #"//chrome/browser/resources/lens/overlay:resources",
+        #"//chrome/browser/resources/lens/overlay/search_bubble:resources",
         "//chrome/browser/resources/search_engine_choice:resources",
         "//content/browser/devtools:devtools_resources",
         "//content/browser/tracing:resources",
@@ -416,7 +416,7 @@ template("chrome_extra_paks") {
       sources += [ "$root_gen_dir/chrome/webui_js_error_resources.pak" ]
       deps += [ "//chrome/browser/resources/webui_js_error:resources" ]
     }
-    if (!is_android && !is_chromeos_ash) {
+    if (!is_android && !is_chromeos_ash || is_android) {
       sources += [
         "$root_gen_dir/chrome/intro_resources.pak",
         "$root_gen_dir/chrome/profile_picker_resources.pak",
@@ -449,12 +449,12 @@ template("chrome_extra_paks") {
       deps += [ "//chrome/browser/resources/dlp_internals:resources" ]
     }
 
-    if (is_win || is_mac || is_linux || is_chromeos_lacros) {
+    if (is_win || is_mac || is_linux || is_chromeos_lacros || is_android) {
       sources += [ "$root_gen_dir/chrome/browser_switch_resources.pak" ]
       deps += [ "//chrome/browser/resources/browser_switch:resources" ]
     }
 
-    if (is_win || is_mac || is_linux || is_chromeos) {
+    if (is_win || is_mac || is_linux || is_chromeos || is_android) {
       sources += [ "$root_gen_dir/chrome/discards_resources.pak" ]
       deps += [ "//chrome/browser/resources/discards:resources" ]
     }
