diff --git a/chrome/browser/ui/android/appmenu/java/src/org/chromium/chrome/browser/ui/appmenu/AppMenuPropertiesDelegate.java b/chrome/browser/ui/android/appmenu/java/src/org/chromium/chrome/browser/ui/appmenu/AppMenuPropertiesDelegate.java
index fb2164f698eba8c8dbc64e8a2aa986612812b5e1..590991ac8d29de31db82fbd855f4beaeb362fd1c 100644
--- a/chrome/browser/ui/android/appmenu/java/src/org/chromium/chrome/browser/ui/appmenu/AppMenuPropertiesDelegate.java
+++ b/chrome/browser/ui/android/appmenu/java/src/org/chromium/chrome/browser/ui/appmenu/AppMenuPropertiesDelegate.java
@@ -135,4 +135,6 @@ public interface AppMenuPropertiesDelegate {
 
     /** Returns whether the menu icon is positioned at the start. */
     boolean isMenuIconAtStart();
+
+    void showExtensionOnly(boolean show);
 }
