diff --git a/chrome/browser/tpcd/heuristics/redirect_heuristic_tab_helper.cc b/chrome/browser/tpcd/heuristics/redirect_heuristic_tab_helper.cc
index ae965929177190bc593de78f3736d3412079a940..3f05156837e9d34933d316a2c4ec6bfe5f375cdd 100644
--- a/chrome/browser/tpcd/heuristics/redirect_heuristic_tab_helper.cc
+++ b/chrome/browser/tpcd/heuristics/redirect_heuristic_tab_helper.cc
@@ -96,6 +96,8 @@ void RedirectHeuristicTabHelper::MaybeRecordRedirectHeuristic(
   const size_t sites_passed_count =
       first_party_site_index - third_party_site_index;
 
+  CHECK(dips_service_);
+  CHECK(!dips_service_->storage()->is_null());
   dips_service_->storage()
       ->AsyncCall(&DIPSStorage::LastInteractionTime)
       .WithArgs(details.url)
