diff --git a/chrome/browser/extensions/api/tabs/windows_event_router.h b/chrome/browser/extensions/api/tabs/windows_event_router.h
index f5cadaf2e34556770818a8b4a3cb941fed8bdcca..263ae7a9e64047cbc1b7e18ccb2003e2ea8931e9 100644
--- a/chrome/browser/extensions/api/tabs/windows_event_router.h
+++ b/chrome/browser/extensions/api/tabs/windows_event_router.h
@@ -42,6 +42,10 @@ class WindowsEventRouter : public AppWindowRegistry::Observer,
                            public KeyWindowNotifier::Observer,
 #elif defined(TOOLKIT_VIEWS)
                            public views::WidgetFocusChangeListener,
+#endif
+#if BUILDFLAG(IS_ANDROID)
+                           public TabModelListObserver,
+                           public TabModelObserver,
 #endif
                            public WindowControllerListObserver {
  public:
@@ -55,6 +59,15 @@ class WindowsEventRouter : public AppWindowRegistry::Observer,
   // |window_controller| is NULL to indicate a focused window has lost focus.
   void OnActiveWindowChanged(WindowController* window_controller);
 
+#if BUILDFLAG(IS_ANDROID)
+  // TabModelListObserver implementation.
+  void OnTabModelAdded() override;
+  void OnTabModelRemoved() override;
+  void WillCloseTab(TabAndroid* tab) override;
+  void DidAddTab(TabAndroid* tab, TabModel::TabLaunchType type) override;
+  raw_ptr<TabModel> observed_tab_model_ = nullptr;
+#endif
+
  private:
   // extensions::AppWindowRegistry::Observer:
   void OnAppWindowAdded(extensions::AppWindow* app_window) override;
