diff --git a/chrome/browser/extensions/extension_install_prompt.cc b/chrome/browser/extensions/extension_install_prompt.cc
index 8929e1c3e1bafe8aa7dc678cab8ad2c0fbdc9dfb..0bdfd46f62c7b922256ccb67cecb32c0b9ebb7ef 100644
--- a/chrome/browser/extensions/extension_install_prompt.cc
+++ b/chrome/browser/extensions/extension_install_prompt.cc
@@ -429,6 +429,7 @@ ExtensionInstallPrompt::ExtensionInstallPrompt(content::WebContents* contents)
       extension_(nullptr),
       install_ui_(std::make_unique<ExtensionInstallUI>(profile_)),
       show_params_(new ExtensionInstallPromptShowParams(contents)),
+      contents_(contents),
       did_call_show_dialog_(false) {}
 
 ExtensionInstallPrompt::ExtensionInstallPrompt(Profile* profile,
@@ -582,6 +583,7 @@ void ExtensionInstallPrompt::ShowConfirmation() {
 
   // Notify observers.
   prompt_->OnDialogOpened();
+  MISES_EXTENSION_INSTALL_PROMPT_SHOW_CONFIRMATION
 
   // If true, auto confirm is enabled and already handled the result.
   if (AutoConfirmPromptIfEnabled())
