diff --git a/chrome/browser/ui/android/toolbar/java/src/org/chromium/chrome/browser/toolbar/top/ToolbarPhone.java b/chrome/browser/ui/android/toolbar/java/src/org/chromium/chrome/browser/toolbar/top/ToolbarPhone.java
index 404ff70b970ed28221ace433cd36c62c5eb9fa10..dff104a65c4aa9e3d908df3a705724ff9220457c 100644
--- a/chrome/browser/ui/android/toolbar/java/src/org/chromium/chrome/browser/toolbar/top/ToolbarPhone.java
+++ b/chrome/browser/ui/android/toolbar/java/src/org/chromium/chrome/browser/toolbar/top/ToolbarPhone.java
@@ -101,13 +101,53 @@ import java.util.ArrayList;
 import java.util.List;
 import java.util.function.BooleanSupplier;
 
+import android.view.ViewTreeObserver;
+import android.widget.LinearLayout;
+import android.animation.ValueAnimator;
+import org.chromium.base.ContextUtils;
+import org.chromium.chrome.browser.toolbar.top.HandButton;
+import android.widget.Toast;
+import android.animation.ValueAnimator;
+
+import org.chromium.ui.display.DisplayAndroid;
+import android.os.Bundle;
+import android.widget.ImageButton;
+import android.widget.PopupWindow;
+import android.widget.Toast;
+import org.chromium.base.Log;
+import org.chromium.ui.widget.ChromeImageButton;
+import org.chromium.base.MisesController;
+import org.chromium.chrome.browser.mises.MisesShareWin;
+import org.chromium.chrome.browser.mises.MisesWeb3SafeAlert;
+import org.chromium.chrome.browser.mises.MisesUserInfoMenu;
+import org.chromium.chrome.browser.tab.TabLaunchType;
+import org.chromium.chrome.browser.tabmodel.TabCreatorManager;
+import org.chromium.chrome.browser.tabmodel.TabCreator;
+import org.chromium.chrome.browser.tabmodel.TabModel;
+import org.chromium.content_public.browser.JavaScriptCallback;
+import org.chromium.ui.base.Clipboard;
+import org.json.JSONException;
+import org.json.JSONObject;
+import androidx.fragment.app.FragmentActivity;
+//import com.google.firebase.analytics.FirebaseAnalytics;
+import com.bumptech.glide.Glide;
+import com.bumptech.glide.load.resource.bitmap.CircleCrop;
+import com.bumptech.glide.request.RequestOptions;
+import org.chromium.chrome.browser.ui.appmenu.AppMenuDelegate;
+import android.graphics.drawable.BitmapDrawable;
+import android.graphics.Bitmap;
+import android.graphics.BitmapFactory;
+import android.text.TextUtils;
+import android.util.Base64;
+import android.graphics.PorterDuff.Mode;
+
 /**
  * Phone specific toolbar implementation.
  */
 public class ToolbarPhone extends ToolbarLayout
-        implements OnClickListener, TabCountObserver, OmniboxSuggestionsDropdownScrollListener {
+        implements OnClickListener, TabCountObserver, OmniboxSuggestionsDropdownScrollListener, MisesController.MisesControllerObserver {
     /** The amount of time transitioning from one theme color to another should take in ms. */
-    public static final long THEME_COLOR_TRANSITION_DURATION = 250;
+    public static final long THEME_COLOR_TRANSITION_DURATION = 50;
 
     public static final int URL_FOCUS_CHANGE_ANIMATION_DURATION_MS = 225;
     private static final int URL_FOCUS_TOOLBAR_BUTTONS_DURATION_MS = 100;
@@ -141,6 +181,10 @@ public class ToolbarPhone extends ToolbarLayout
     protected @Nullable ToggleTabStackButton mToggleTabStackButton;
     // Non-null after inflation occurs.
     protected @NonNull ImageView mHomeButton;
+    
+    protected @Nullable HandButton mHandButton;
+    protected @Nullable ImageButton mMisesMainButton;
+
     private TextView mUrlBar;
     protected View mUrlActionContainer;
     protected ImageView mToolbarShadow;
@@ -346,6 +390,7 @@ public class ToolbarPhone extends ToolbarLayout
 
             mToolbarButtonsContainer = (ViewGroup) findViewById(R.id.toolbar_buttons);
             mHomeButton = findViewById(R.id.home_button);
+            mHandButton = findViewById(R.id.hand_button);
             mUrlBar = (TextView) findViewById(R.id.url_bar);
             mUrlActionContainer = findViewById(R.id.url_action_container);
             mToolbarBackground =
@@ -362,6 +407,7 @@ public class ToolbarPhone extends ToolbarLayout
             setWillNotDraw(false);
             mUrlFocusTranslationX =
                     getResources().getDimensionPixelSize(R.dimen.toolbar_url_focus_translation_x);
+            setupMisesButton();
 
             // Set hover tooltip texts for toolbar buttons shared between phones and tablets.
             super.setTooltipTextForToolbarButtons();
@@ -383,6 +429,7 @@ public class ToolbarPhone extends ToolbarLayout
 
     @Override
     public void destroy() {
+        MisesController.getInstance().RemoveObserver(this);
         cancelAnimations();
         Handler handler = getHandler();
         if (handler != null) {
@@ -522,6 +569,12 @@ public class ToolbarPhone extends ToolbarLayout
         enableTabSwitchingResources();
         mHomeButton.setOnClickListener(this);
 
+        if (mHandButton != null) {
+            mHandButton.setOnClickListener(this);
+        }
+
+        setupMisesButtonClickListener();
+
         getMenuButtonCoordinator().setOnKeyListener(new KeyboardNavigationListener() {
             @Override
             public View getNextFocusForward() {
@@ -596,6 +649,11 @@ public class ToolbarPhone extends ToolbarLayout
                         EventConstants.PARTNER_HOME_PAGE_BUTTON_PRESSED);
             }
         }
+
+        if (mHandButton != null && mHandButton == v) {
+            openOverscroll();
+        }
+        onClickMisesButton(v);
     }
 
     @Override
@@ -775,6 +833,7 @@ public class ToolbarPhone extends ToolbarLayout
      */
     private int getBoundsAfterAccountingForLeftButton() {
         int padding = mToolbarSidePaddingForRealOmnibox;
+        padding = getMisesButtonPadding();
 
         // If home button is visible, or it's now in overview and toolbar is not shown (url bar
         // shouldn't be focused), mHomeButton.getMeasuredWidth() should be returned as the left
@@ -890,6 +949,7 @@ public class ToolbarPhone extends ToolbarLayout
 
     @Override
     protected void dispatchDraw(Canvas canvas) {
+        setDurationScale(0.0f);
         if (!mTextureCaptureMode && mToolbarBackground.getColor() != Color.TRANSPARENT) {
             // Update to compensate for orientation changes.
             mToolbarBackground.setBounds(0, 0, getWidth(), getHeight());
@@ -907,6 +967,7 @@ public class ToolbarPhone extends ToolbarLayout
         } else {
             super.dispatchDraw(canvas);
         }
+        setDurationScale(0.6f);
     }
 
     @Override
@@ -1059,6 +1120,10 @@ public class ToolbarPhone extends ToolbarLayout
         if (mHomeButton.getVisibility() != GONE) {
             mHomeButton.setVisibility(toolbarButtonVisibility);
         }
+        if (mHandButton != null && mHandButton.getVisibility() != GONE) {
+            mHandButton.setVisibility(toolbarButtonVisibility);
+        }
+        updateMisesButtonVisibility(toolbarButtonVisibility);
 
         updateLocationBarLayoutForExpansionAnimation();
     }
@@ -1313,7 +1378,8 @@ public class ToolbarPhone extends ToolbarLayout
         mLocationBar.getPhoneCoordinator().setTranslationX(0);
         if (!mUrlFocusChangeInProgress) {
             mToolbarButtonsContainer.setTranslationY(0);
-            mHomeButton.setTranslationY(0);
+            if (mHomeButton != null) mHomeButton.setTranslationY(0);
+            if (mHandButton != null) mHandButton.setTranslationY(0);
         }
 
         if (!mUrlFocusChangeInProgress && mToolbarShadow != null) {
@@ -1446,7 +1512,8 @@ public class ToolbarPhone extends ToolbarLayout
         int transY = mTabSwitcherState == STATIC_TAB ? Math.min(mNtpSearchBoxTranslation.y, 0) : 0;
 
         mToolbarButtonsContainer.setTranslationY(transY);
-        mHomeButton.setTranslationY(transY);
+        if (mHomeButton != null) mHomeButton.setTranslationY(transY);
+        if (mHandButton != null) mHandButton.setTranslationY(transY);
     }
 
     private void setAncestorsShouldClipChildren(boolean clip) {
@@ -1473,8 +1540,12 @@ public class ToolbarPhone extends ToolbarLayout
             mHomeButton.setAlpha(previousAlpha * floatAlpha);
             drawChild(canvas, mHomeButton, SystemClock.uptimeMillis());
             mHomeButton.setAlpha(previousAlpha);
+        } else {
+            previousAlpha = 1.0f;
         }
 
+        previousAlpha = drawMisesMainButton(canvas, previousAlpha, floatAlpha);
+
         // Draw the location/URL bar.
         previousAlpha = mLocationBar.getPhoneCoordinator().getAlpha();
         mLocationBar.getPhoneCoordinator().setAlpha(previousAlpha * floatAlpha);
@@ -1501,6 +1572,33 @@ public class ToolbarPhone extends ToolbarLayout
             canvas.restore();
         }
 
+        if (mHandButton != null && mHandButton.getVisibility() != View.GONE) {
+            canvas.save();
+            Drawable oversButton = mHandButton.getDrawable();
+
+            ViewUtils.translateCanvasToView(mToolbarButtonsContainer, mHandButton, canvas);
+
+            int backgroundWidth = mHandButton.getDrawable().getIntrinsicWidth();
+            int backgroundHeight = mHandButton.getDrawable().getIntrinsicHeight();
+            int backgroundLeft =
+                    (mHandButton.getWidth() - mHandButton.getPaddingLeft()
+                            - mHandButton.getPaddingRight() - backgroundWidth)
+                    / 2;
+            backgroundLeft += mHandButton.getPaddingLeft();
+            int backgroundTop =
+                    (mHandButton.getHeight() - mHandButton.getPaddingTop()
+                            - mHandButton.getPaddingBottom() - backgroundHeight)
+                    / 2;
+            backgroundTop += mHandButton.getPaddingTop();
+            canvas.translate(backgroundLeft, backgroundTop);
+
+            oversButton.setAlpha(rgbAlpha);
+            oversButton.draw(canvas);
+            canvas.restore();
+        }
+
+        previousAlpha = drawMisesExtensionButton(canvas, previousAlpha, rgbAlpha);
+
         // Draw the tab stack button and associated text if necessary.
         if (mTabSwitcherAnimationTabStackDrawable != null && mToggleTabStackButton != null
                 && mUrlExpansionFraction != 1f) {
@@ -1586,7 +1684,8 @@ public class ToolbarPhone extends ToolbarLayout
     }
 
     private boolean isChildLeft(View child) {
-        return child == mHomeButton ^ LocalizationUtils.isLayoutRtl();
+        boolean is_main = (mMisesMainButton != null && child == mMisesMainButton);
+        return (child == mHomeButton || is_main) ^ LocalizationUtils.isLayoutRtl();
     }
 
     /**
@@ -1735,11 +1834,12 @@ public class ToolbarPhone extends ToolbarLayout
 
     @Override
     public CaptureReadinessResult isReadyForTextureCapture() {
+        boolean mises_enforce = true;
         if (ToolbarFeatures.shouldBlockCapturesForAblation()) {
             return CaptureReadinessResult.notReady(TopToolbarBlockCaptureReason.SCROLL_ABLATION);
         } else if (mForceTextureCapture) {
             return CaptureReadinessResult.readyForced();
-        } else if (ToolbarFeatures.shouldSuppressCaptures()) {
+        } else if (mises_enforce || ToolbarFeatures.shouldSuppressCaptures()) {
             return getReadinessStateWithSuppression();
         } else {
             return CaptureReadinessResult.unknown(!(urlHasFocus() || mUrlFocusChangeInProgress));
@@ -1858,8 +1958,9 @@ public class ToolbarPhone extends ToolbarLayout
 
     @Override
     public void updateButtonVisibility() {
+        boolean mises_enforce = true;
         boolean hideHomeButton =
-                !mIsHomeButtonEnabled || getToolbarDataProvider().isInOverviewAndShowingOmnibox();
+                mises_enforce || !mIsHomeButtonEnabled || getToolbarDataProvider().isInOverviewAndShowingOmnibox();
         if (hideHomeButton) {
             mHomeButton.setVisibility(View.GONE);
         } else {
@@ -1886,6 +1987,15 @@ public class ToolbarPhone extends ToolbarLayout
         if (mLocationBar != null) mLocationBar.updateVisualsForState();
 
         if (mLayoutUpdater != null) mLayoutUpdater.run();
+        
+        if (mHomeButton != null) {
+            ImageViewCompat.setImageTintList(mHomeButton, tint);
+        }
+        if (mHandButton != null) {
+            ImageViewCompat.setImageTintList(mHandButton, tint);
+        }
+        if (mMisesExtensionButton != null && mExtensionButtonDrawableOveride == null) 
+            ImageViewCompat.setImageTintList(mMisesExtensionButton, tint);
     }
 
     @Override
@@ -2031,6 +2141,11 @@ public class ToolbarPhone extends ToolbarLayout
                 ((RippleDrawable) drawable).jumpToCurrentState();
             }
         }
+        if (mHandButton != null) {
+            boolean mises_enforce = true;
+            if (mises_enforce || !ContextUtils.getAppSharedPreferences().getBoolean("enable_overscroll_button", true))
+            mHandButton.setVisibility(GONE);
+        }
     }
 
     @Override
@@ -2350,7 +2465,8 @@ public class ToolbarPhone extends ToolbarLayout
 
     @Override
     public void onTabCountChanged(int numberOfTabs, boolean isIncognito) {
-        mHomeButton.setEnabled(true);
+        if (mHomeButton != null) mHomeButton.setEnabled(true);
+	    if (mHandButton != null) mHandButton.setEnabled(true);
         if (mToggleTabStackButton == null) return;
 
         @BrandedColorScheme
@@ -2522,7 +2638,8 @@ public class ToolbarPhone extends ToolbarLayout
      * Update the visibility of the toolbar shadow.
      */
     private void updateShadowVisibility() {
-        boolean shouldDrawShadow = shouldDrawShadow();
+        boolean mises_enforce = false;
+        boolean shouldDrawShadow = mises_enforce && shouldDrawShadow();
         int shadowVisibility = shouldDrawShadow ? View.VISIBLE : View.INVISIBLE;
 
         if (mToolbarShadow != null && mToolbarShadow.getVisibility() != shadowVisibility) {
@@ -3110,4 +3227,254 @@ public class ToolbarPhone extends ToolbarLayout
     void setNtpSearchBoxScrollFractionForTesting(float ntpSearchBoxScrollFraction) {
         mNtpSearchBoxScrollFraction = ntpSearchBoxScrollFraction;
     }
+
+
+    public void openOverscroll() {
+        Tab currentTab = getToolbarDataProvider().getTab();
+        if (currentTab == null) return;
+
+        String SCRIPT = "var _kbOverscroll;"
++"(function (d) {"
++   " if (typeof _kbOverscroll == 'undefined' || _kbOverscroll == false) {"
++        "d.getElementsByTagName('html')[0].style.transition = '0.5s ease-in-out';"
++        "d.getElementsByTagName('html')[0].style.transform = 'translate(0px, 98vw)';"
++        "d.getElementsByTagName('html')[0].style.overflowY = 'initial';"
++        "d.getElementsByTagName('body')[0].style.display='block';"
++        "d.getElementsByTagName('body')[0].style.position='fixed';"
++        "d.getElementsByTagName('body')[0].style.overflowY='scroll';"
++        "d.getElementsByTagName('body')[0].style.height='98vw';"
++        "window.scrollTo({top: 0,left: 0,behavior: 'smooth' });"
++        "_kbOverscroll = true;"
++    "} else {"
++        "d.getElementsByTagName('html')[0].style.transition = '0.5s ease-in-out';"
++        "d.getElementsByTagName('html')[0].style.transform = '';"
++        "d.getElementsByTagName('html')[0].style.overflowY = 'initial';"
++        "d.getElementsByTagName('body')[0].style.display='';"
++        "d.getElementsByTagName('body')[0].style.position='initial';"
++        "d.getElementsByTagName('body')[0].style.overflowY='auto';"
++        "d.getElementsByTagName('body')[0].style.height='';"
++        "_kbOverscroll = false;}}(document));";
+        currentTab.getWebContents().evaluateJavaScript(SCRIPT, null);
+        // matching chrome-native://newtab and chrome://newtab and kiwi://newtab
+        if (currentTab.isNativePage() && currentTab.getUrl().getSpec().contains("/newtab")) {
+            View nativeView = currentTab.getView();
+            DisplayAndroid display =
+                    DisplayAndroid.getNonMultiDisplay(ContextUtils.getApplicationContext());
+            int screenHeight = display.getDisplayHeight();
+            int targetHeight = (int)Math.round(screenHeight * 0.42);
+            final View handSpacer = nativeView.findViewWithTag("hand_button_spacer");
+            if (handSpacer != null && handSpacer.getHeight() > 0) {
+                ValueAnimator animator = ValueAnimator.ofInt(handSpacer.getHeight(), 0);
+                animator.addUpdateListener(new ValueAnimator.AnimatorUpdateListener() {
+                    @Override
+                    public void onAnimationUpdate(ValueAnimator valueAnimator) {
+                        LinearLayout.LayoutParams params = (LinearLayout.LayoutParams) handSpacer.getLayoutParams();
+                        params.height = ((Integer) valueAnimator.getAnimatedValue());
+                        handSpacer.setLayoutParams(params);
+                    }
+                });
+                animator.setDuration(500);
+                animator.start();
+            } else if (handSpacer != null) {
+                ValueAnimator animator = ValueAnimator.ofInt(handSpacer.getHeight(), targetHeight);
+                animator.addUpdateListener(new ValueAnimator.AnimatorUpdateListener() {
+                    @Override
+                    public void onAnimationUpdate(ValueAnimator valueAnimator) {
+                        LinearLayout.LayoutParams params = (LinearLayout.LayoutParams) handSpacer.getLayoutParams();
+                        params.height = ((Integer) valueAnimator.getAnimatedValue());
+                        handSpacer.setLayoutParams(params);
+                    }
+                });
+                animator.setDuration(500);
+                animator.start();
+            }
+        }
+    }
+    private void setDurationScale(float scale) {
+        try {
+            ValueAnimator.class.getMethod("setDurationScale", float.class).invoke(null, scale);
+        } catch (Throwable t) {
+
+        }
+    }
+    
+    public void updateAvatarBtn() {
+        if (mMisesMainButton == null)
+            return;
+        if (!MisesController.getInstance().getMisesAvatar().isEmpty()) {
+            Glide.with(getContext()).load(MisesController.getInstance().getMisesAvatar()).apply(RequestOptions.bitmapTransform(new CircleCrop()))
+                     .error(R.mipmap.head_small).placeholder(R.mipmap.head_small).into(mMisesMainButton);
+        } else {
+            mMisesMainButton.setImageResource(R.mipmap.head_small);
+        }
+    }
+
+    @Override
+    public void OnMisesUserInfoChanged() {
+        if (mMisesMainButton == null)
+            return;
+        Context context = getContext();
+        if (!(context instanceof FragmentActivity) || !(context instanceof TabCreatorManager))
+            return;
+        FragmentActivity chromeTabbedActivity = (FragmentActivity) context;
+        updateAvatarBtn();
+        if (!MisesController.getInstance().getLastShareUrl().isEmpty()) {
+            MisesShareWin shareWin = MisesShareWin.newInstance((TabCreatorManager) context, MisesController.getInstance().getLastShareIcon(),
+                    MisesController.getInstance().getLastShareTitle(),
+                    MisesController.getInstance().getLastShareUrl());
+            MisesController.getInstance().clearLastShareInfo();
+            shareWin.show(chromeTabbedActivity.getSupportFragmentManager(), "MisesShareWin");
+        }
+    }
+
+    @Override
+    protected void updateReloadButtonVisibility(boolean isReloading) {
+    }
+
+
+    private float drawMisesExtensionButton(Canvas canvas,float previousAlpha, int rgbAlpha) {
+        if (mMisesExtensionButton != null && mMisesExtensionButton.getVisibility() != View.GONE) {
+            Log.e("Mises","drawMisesExtensionButton");
+	    canvas.save();
+	    ChromeImageButton imageButton = mMisesExtensionButton;
+	    Drawable buttonDrawable = mMisesExtensionButton.getDrawable();
+            ViewUtils.translateCanvasToView( mToolbarButtonsContainer, imageButton, canvas);
+            int backgroundWidth = buttonDrawable.getIntrinsicWidth();
+            int backgroundHeight = buttonDrawable.getIntrinsicHeight();
+            int backgroundLeft =
+                    (imageButton.getWidth() - imageButton.getPaddingLeft()
+                            - imageButton.getPaddingRight() - backgroundWidth)
+                            / 2;
+            backgroundLeft += imageButton.getPaddingLeft();
+            int backgroundTop =
+                    (imageButton.getHeight() - imageButton.getPaddingTop()
+                            - imageButton.getPaddingBottom() - backgroundHeight)
+                            / 2;
+            backgroundTop += imageButton.getPaddingTop();
+            canvas.translate(backgroundLeft, backgroundTop);
+            buttonDrawable.setAlpha(rgbAlpha);
+            buttonDrawable.draw(canvas);
+
+            canvas.restore();
+        }
+	    return previousAlpha;
+    }
+    private float drawMisesMainButton(Canvas canvas, float previousAlpha, float floatAlpha) {
+        if (mMisesMainButton != null && mMisesMainButton.getVisibility() != View.GONE) {
+            // Draw the New Tab button used in the URL view.
+            previousAlpha = mMisesMainButton.getAlpha();
+            mMisesMainButton.setAlpha(previousAlpha * floatAlpha);
+            drawChild(canvas, mMisesMainButton, SystemClock.uptimeMillis());
+            mMisesMainButton.setAlpha(previousAlpha);
+        }
+	    return previousAlpha;
+    }
+    private int getMisesButtonPadding() {
+        int padding = mToolbarSidePadding;
+        if (mMisesMainButton != null && mMisesMainButton.getVisibility() != GONE) {
+            padding = mMisesMainButton.getMeasuredWidth();
+        }
+        return padding;
+    }
+    private void updateMisesButtonVisibility(int toolbarButtonVisibility) {
+
+        if (mMisesMainButton != null && mMisesMainButton.getVisibility() != GONE) {
+            mMisesMainButton.setVisibility(toolbarButtonVisibility);
+        }
+        if (mMisesExtensionButton != null && mMisesExtensionButton.getVisibility() != GONE) {
+            mMisesExtensionButton.setVisibility(toolbarButtonVisibility);
+        }
+    }
+    private void setupMisesButton() {
+        mMisesMainButton =  (ImageButton) findViewById(R.id.mises_main_button);
+        updateAvatarBtn();
+        MisesController.getInstance().AddObserver(this);
+
+        mMisesExtensionButton =  (ChromeImageButton) findViewById(R.id.mises_extension_button);
+    }
+    private void setupMisesButtonClickListener() {
+        if (mMisesMainButton != null) mMisesMainButton.setOnClickListener(this);
+	    if (mMisesExtensionButton != null) mMisesExtensionButton.setOnClickListener(this);
+    }
+
+    @Override
+    protected void onClickMisesButton(View v) {
+        super.onClickMisesButton(v);
+        if (mMisesMainButton != null && mMisesMainButton == v) {
+            String id = MisesController.getInstance().getMisesId();
+            String username = MisesController.getInstance().getMisesNickname();
+            String avatar = MisesController.getInstance().getMisesAvatar();
+            MisesUserInfoMenu misesUserInfoMenu = new MisesUserInfoMenu(getContext(), id, username, avatar);
+            misesUserInfoMenu.setOnClickListener(new OnClickListener() {
+                @Override
+                public void onClick(View v) {
+                    Context context = getContext();
+                    TabCreator tabCreator = null;
+                    if (context instanceof TabCreatorManager) {
+                        TabCreatorManager chromeTabbedActivity = (TabCreatorManager) context;
+                        tabCreator = chromeTabbedActivity.getTabCreator(false);
+                    }
+                    if (v.getId() == R.id.tv_my_data) {
+                        if (tabCreator != null) {
+                            tabCreator.openSinglePage(MisesController.MISES_HOME_BASE_URL + "/userInfo");
+                        }
+                    } else if (v.getId() == R.id.tv_my_rewards) {
+                        if (tabCreator != null) {
+                            tabCreator.openSinglePage(MisesController.MISES_MINING_BASE_URL + "/bonuses");
+                        }
+                    } else if (v.getId() == R.id.tv_my_rewards_action_mine) {
+                        if (tabCreator != null) {
+                            tabCreator.openSinglePage(MisesController.MISES_MINING_BASE_URL + "/mining");
+                        }
+                    } else if (v.getId() == R.id.tv_my_rewards_action_redeem) {
+                        if (tabCreator != null) {
+                            tabCreator.openSinglePage(MisesController.MISES_MINING_BASE_URL + "/bonuses");
+                        }
+                    } else if (v.getId() == R.id.tv_mises_discover) {
+                        if (tabCreator != null) {
+                            tabCreator.openSinglePage(MisesController.MISES_HOME_BASE_URL + "/home/discover");
+                        }
+                    } else if (v.getId() == R.id.tv_wallet) {
+                        if (tabCreator != null) {
+                            tabCreator.openSinglePage(MisesController.MISES_WALLET_BASE_URL);
+                        }
+                    } else if (v.getId() == R.id.tv_portal) {
+                        if (tabCreator != null) {
+                            tabCreator.openSinglePage("https://portal.mises.site");
+                        }
+                    } else if (v.getId() == R.id.tv_nft) {
+                        if (tabCreator != null) {
+                            tabCreator.openSinglePage(MisesController.MISES_HOME_BASE_URL + "/nft");
+                        }
+                    } else if (v.getId() == R.id.tv_invite) {  
+                        if (tabCreator != null) {
+                            tabCreator.openSinglePage(MisesController.MISES_HOME_BASE_URL + "/myInvitation?misesId=" + id);
+                        }
+                    } else if (v.getId() == R.id.tv_login) {
+                        if (tabCreator != null) {
+                            tabCreator.openSinglePage(MisesController.MISES_HOME_BASE_URL + "/home/me");
+                        }
+                    } else if (v.getId() == R.id.tv_create_mises) {
+                        if (tabCreator != null) {
+                            tabCreator.openSinglePage(MisesController.MISES_WALLET_BASE_URL);
+                        }
+                    } else if (v.getId() == R.id.tv_website || v.getId() == R.id.tv_website1) {
+                        if (tabCreator != null) {
+                            tabCreator.launchUrl("https://www.mises.site", TabLaunchType.FROM_CHROME_UI);
+                        }
+                    }  else if (v.getId() == R.id.tv_twitter || v.getId() == R.id.tv_twitter1) {
+                        if (tabCreator != null) {
+                            tabCreator.launchUrl("https://twitter.com/Mises001", TabLaunchType.FROM_CHROME_UI);
+                        }
+                    } else if (v.getId() == R.id.tv_id) {
+                        Clipboard.getInstance().setText(id);
+			            Toast.makeText(getContext(), getContext().getString(R.string.lbl_id_copied_tip), Toast.LENGTH_SHORT).show();
+			            return;
+   	                } 
+                    misesUserInfoMenu.dismiss();
+                }
+            });
+            misesUserInfoMenu.showAtLocation(mMisesMainButton, Gravity.START | Gravity.TOP, 0, 0);
+        }
+    }
 }
