diff --git a/chrome/android/java/src/org/chromium/chrome/browser/about_settings/AboutChromeSettings.java b/chrome/android/java/src/org/chromium/chrome/browser/about_settings/AboutChromeSettings.java
index 5d33049d737f23e74626ea815657f65d8b34b083..3e93f97d961af6bfe2477b7ddf7514f0b5b979d2 100644
--- a/chrome/android/java/src/org/chromium/chrome/browser/about_settings/AboutChromeSettings.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/about_settings/AboutChromeSettings.java
@@ -73,7 +73,8 @@ public class AboutChromeSettings extends PreferenceFragmentCompat
      * versions are more useful.
      */
     public static String getApplicationVersion(Context context, String version) {
-        if (VersionInfo.isOfficialBuild()) {
+        boolean mises_enforce = false;
+	if (mises_enforce && VersionInfo.isOfficialBuild()) {
             return version;
         }
 
@@ -87,7 +88,7 @@ public class AboutChromeSettings extends PreferenceFragmentCompat
         CharSequence updateTimeString =
                 DateUtils.getRelativeTimeSpanString(
                         info.lastUpdateTime, System.currentTimeMillis(), 0);
-        return context.getString(R.string.version_with_update_time, version, updateTimeString);
+        return context.getString(R.string.version_with_update_time, info.versionName, updateTimeString);
     }
 
     @Override
