diff --git a/chrome/browser/resources/hats/BUILD.gn b/chrome/browser/resources/hats/BUILD.gn
index 7309121fd0df942060780b851ecd115f7397f8c6..036e4fc5165ea7c1084f6f707eb3ddc573077161 100644
--- a/chrome/browser/resources/hats/BUILD.gn
+++ b/chrome/browser/resources/hats/BUILD.gn
@@ -4,7 +4,7 @@
 
 import("//ui/webui/resources/tools/build_webui.gni")
 
-assert(!is_android)
+#assert(!is_android)
 
 build_webui("build") {
   grd_prefix = "hats"
