diff --git a/chrome/android/java/src/org/chromium/chrome/browser/homepage/HomepageManager.java b/chrome/android/java/src/org/chromium/chrome/browser/homepage/HomepageManager.java
index 7cd3d2bb13dc7e5db18a2039b4975cbb095ec116..cc38158b167dd94476535d70f9167051df362868 100644
--- a/chrome/android/java/src/org/chromium/chrome/browser/homepage/HomepageManager.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/homepage/HomepageManager.java
@@ -31,6 +31,8 @@ import org.chromium.components.embedder_support.util.UrlConstants;
 import org.chromium.components.embedder_support.util.UrlUtilities;
 import org.chromium.url.GURL;
 
+import org.chromium.base.ContextUtils;
+
 /**
  * Provides information regarding homepage enabled states and URI.
  *
@@ -124,8 +126,7 @@ public class HomepageManager implements HomepagePolicyManager.HomepagePolicyStat
      * @return Whether to close the app when the user has zero tabs.
      */
     public static boolean shouldCloseAppWithZeroTabs() {
-        return HomepageManager.isHomepageEnabled()
-                && !UrlUtilities.isNTPUrl(HomepageManager.getHomepageGurl());
+        return ContextUtils.getAppSharedPreferences().getBoolean("close_browser_after_last_tab", false);
     }
 
     /**
@@ -220,7 +221,7 @@ public class HomepageManager implements HomepagePolicyManager.HomepagePolicyStat
      * Get homepage URI without checking if the homepage is enabled.
      * @return Homepage GURL based on policy and shared preference settings.
      */
-    private @NonNull GURL getHomepageGurlIgnoringEnabledState() {
+    public @NonNull GURL getHomepageGurlIgnoringEnabledState() {
         if (HomepagePolicyManager.isHomepageManagedByPolicy()) {
             return HomepagePolicyManager.getHomepageUrl();
         }
@@ -232,6 +233,9 @@ public class HomepageManager implements HomepagePolicyManager.HomepagePolicyStat
         }
         return getPrefHomepageCustomGurl();
     }
+    public boolean usingCustomHomepageUri() {
+        return getHomepageGurlIgnoringEnabledState().equals(getPrefHomepageCustomGurl());
+    }
 
     /**
      * Returns the user preference for whether the homepage is enabled. This doesn't take into
