diff --git a/chrome/android/java/src/org/chromium/chrome/browser/download/DownloadManagerService.java b/chrome/android/java/src/org/chromium/chrome/browser/download/DownloadManagerService.java
index 8d7f9c1c4f1fdae1361ab0cad71bb52de11d08a6..8b1b7a9d8a2a2e7ee89d2c48577fcb1bcc014ab7 100644
--- a/chrome/android/java/src/org/chromium/chrome/browser/download/DownloadManagerService.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/download/DownloadManagerService.java
@@ -516,6 +516,10 @@ public class DownloadManagerService implements DownloadController.Observer,
                             info, item.getSystemDownloadId(), result.second, isSupportedMimeType);
                     broadcastDownloadSuccessful(info);
                 } else {
+		             if (info.getMimeType().equalsIgnoreCase("application/x-chrome-extension")){
+		               mDownloadNotifier.notifyDownloadCanceled(item.getContentId());
+		               return;
+		             }
                     info = DownloadInfo.Builder.fromDownloadInfo(info)
                                    .setFailState(FailState.CANNOT_DOWNLOAD)
                                    .build();
