diff --git a/chrome/android/java/src/org/chromium/chrome/browser/tabmodel/TabModelJniBridge.java b/chrome/android/java/src/org/chromium/chrome/browser/tabmodel/TabModelJniBridge.java
index 8297746bf51ca28a72cf062dde00a2f606c77cb6..ebd13933d716c70bdb1b449356f731fab8c004c1 100644
--- a/chrome/android/java/src/org/chromium/chrome/browser/tabmodel/TabModelJniBridge.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/tabmodel/TabModelJniBridge.java
@@ -22,6 +22,7 @@ import org.chromium.content_public.browser.WebContents;
 import org.chromium.content_public.common.ResourceRequestBody;
 import org.chromium.url.GURL;
 import org.chromium.url.Origin;
+import org.chromium.ui.base.PageTransition;
 
 /** Bridges between the C++ and Java {@link TabModel} interfaces. */
 public abstract class TabModelJniBridge implements TabModel {
@@ -153,7 +154,7 @@ public abstract class TabModelJniBridge implements TabModel {
      */
     @CalledByNative
     protected abstract boolean createTabWithWebContents(
-            Tab parent, Profile profile, WebContents webContents, boolean select);
+            Tab parent, Profile profile, WebContents webContents, boolean select, @TabLaunchType int type);
 
     @CalledByNative
     protected abstract void openNewTab(
@@ -173,7 +174,14 @@ public abstract class TabModelJniBridge implements TabModel {
     @CalledByNative
     protected Tab createNewTabForDevTools(GURL url) {
         return getTabCreator(/* incognito= */ false)
-                .createNewTab(new LoadUrlParams(url), TabLaunchType.FROM_CHROME_UI, null);
+                .createNewTab(new LoadUrlParams(url), TabLaunchType.FROM_CHROME_UI, getTabAt(index()));
+    }
+
+    @CalledByNative
+    protected Tab createNewTabForExtensions(GURL url) {
+        final LoadUrlParams param = new LoadUrlParams(url, PageTransition.AUTO_TOPLEVEL);
+        return getTabCreator(/*incognito=*/false)
+                .createNewTab(param, TabLaunchType.FROM_LINK, getTabAt(index()));
     }
 
     /** Returns whether supplied Tab instance has been grouped together with other Tabs. */
@@ -197,6 +205,10 @@ public abstract class TabModelJniBridge implements TabModel {
     @CalledByNative
     public abstract int index();
 
+    @Override
+    @CalledByNative
+    public abstract int getLastNonExtensionActiveIndex();
+
     /** Returns whether or not a sync session is currently being restored. */
     @CalledByNative
     protected abstract boolean isSessionRestoreInProgress();
