diff --git a/chrome/browser/resources/support_tool/BUILD.gn b/chrome/browser/resources/support_tool/BUILD.gn
index 2d27a5ee181e31f05eda79cc74a9f99002eb4e68..70c72b0503b1ede3ae976f5c270ea773fe2cad4e 100644
--- a/chrome/browser/resources/support_tool/BUILD.gn
+++ b/chrome/browser/resources/support_tool/BUILD.gn
@@ -4,7 +4,7 @@
 
 import("//ui/webui/resources/tools/build_webui.gni")
 
-assert(!is_android)
+#assert(!is_android)
 
 build_webui("build") {
   grd_prefix = "support_tool"
