diff --git a/chrome/android/java/src/org/chromium/chrome/browser/homepage/HomepageManager.java b/chrome/android/java/src/org/chromium/chrome/browser/homepage/HomepageManager.java
index 06593f15483f890dc56ba8fcccec47c8e90caaea..17ffbd1d4b115f3880c624c911910f89a5ff3b3d 100644
--- a/chrome/android/java/src/org/chromium/chrome/browser/homepage/HomepageManager.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/homepage/HomepageManager.java
@@ -29,6 +29,8 @@ import org.chromium.components.embedder_support.util.UrlConstants;
 import org.chromium.components.embedder_support.util.UrlUtilities;
 import org.chromium.url.GURL;
 
+import org.chromium.base.ContextUtils;
+
 /**
  * Provides information regarding homepage enabled states and URI.
  *
@@ -110,7 +112,7 @@ public class HomepageManager
      * @return Whether to close the app when the user has zero tabs.
      */
     public boolean shouldCloseAppWithZeroTabs() {
-        return isHomepageEnabled() && !UrlUtilities.isNtpUrl(getHomepageGurl());
+        return ContextUtils.getAppSharedPreferences().getBoolean("close_browser_after_last_tab", false);
     }
 
     /**
@@ -212,7 +214,7 @@ public class HomepageManager
      * Get homepage URI without checking if the homepage is enabled.
      * @return Homepage GURL based on policy and shared preference settings.
      */
-    private @NonNull GURL getHomepageGurlIgnoringEnabledState() {
+    public @NonNull GURL getHomepageGurlIgnoringEnabledState() {
         if (HomepagePolicyManager.isHomepageManagedByPolicy()) {
             return HomepagePolicyManager.getHomepageUrl();
         }
@@ -224,6 +226,9 @@ public class HomepageManager
         }
         return getPrefHomepageCustomGurl();
     }
+    public boolean usingCustomHomepageUri() {
+        return getHomepageGurlIgnoringEnabledState().equals(getPrefHomepageCustomGurl());
+    }
 
     /**
      * Returns the user preference for whether the homepage is enabled. This doesn't take into
