diff --git a/chrome/browser/resources/tab_search/BUILD.gn b/chrome/browser/resources/tab_search/BUILD.gn
index b592786d9fa4ca061e6b8d9a71912d47a1cd353d..febb37430481ce3c502cb08b86863ba4615c6329 100644
--- a/chrome/browser/resources/tab_search/BUILD.gn
+++ b/chrome/browser/resources/tab_search/BUILD.gn
@@ -7,7 +7,7 @@ import("//ui/webui/resources/tools/build_webui.gni")
 import("//ui/webui/resources/tools/generate_grd.gni")
 import("//ui/webui/webui_features.gni")
 
-assert(is_linux || is_chromeos || is_win || is_mac)
+# assert(is_linux || is_chromeos || is_win || is_mac)
 
 build_webui("build") {
   grd_prefix = "tab_search"
