diff --git a/chrome/browser/ui/android/omnibox/java/src/org/chromium/chrome/browser/omnibox/suggestions/OmniboxSuggestionsDropdown.java b/chrome/browser/ui/android/omnibox/java/src/org/chromium/chrome/browser/omnibox/suggestions/OmniboxSuggestionsDropdown.java
index f7eb18f34d323d47524d382ad7dab2c5623ea94b..ea2274e87b0da95e72f173ee53b5bf97f14e1028 100644
--- a/chrome/browser/ui/android/omnibox/java/src/org/chromium/chrome/browser/omnibox/suggestions/OmniboxSuggestionsDropdown.java
+++ b/chrome/browser/ui/android/omnibox/java/src/org/chromium/chrome/browser/omnibox/suggestions/OmniboxSuggestionsDropdown.java
@@ -43,6 +43,9 @@ import org.chromium.ui.base.ViewUtils;
 
 import java.util.Optional;
 
+import android.graphics.Rect;
+import org.chromium.base.ContextUtils;
+
 /** A widget for showing a list of omnibox suggestions. */
 public class OmniboxSuggestionsDropdown extends RecyclerView {
     /**
@@ -410,10 +413,28 @@ public class OmniboxSuggestionsDropdown extends RecyclerView {
         super.setAdapter(mAdapter);
     }
 
+    private void onMeasureBottomToolbar(int widthMeasureSpec, int heightMeasureSpec) {
+        maybeUpdateLayoutParams(0);
+        final Rect mTempRect = new Rect();
+        mEmbedder.getWindowDelegate().getWindowVisibleDisplayFrame(mTempRect);
+        int availableViewportHeight = mTempRect.height();
+        OmniboxAlignment omniboxAlignment = mEmbedder.getCurrentAlignment();
+        int desiredWidth = omniboxAlignment.width;
+        adjustHorizontalPosition();
+        notifyObserversIfViewportHeightChanged(availableViewportHeight);
+        widthMeasureSpec = MeasureSpec.makeMeasureSpec(desiredWidth, MeasureSpec.EXACTLY);
+        heightMeasureSpec = MeasureSpec.makeMeasureSpec(availableViewportHeight,
+                mEmbedder.isTablet() ? MeasureSpec.AT_MOST : MeasureSpec.EXACTLY);
+        super.onMeasure(widthMeasureSpec, heightMeasureSpec);
+    }
     @Override
     protected void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {
         boolean isTablet = mEmbedder.map(e -> e.isTablet()).orElse(false);
 
+        if (ContextUtils.getAppSharedPreferences().getBoolean("enable_bottom_toolbar", false)) {
+            onMeasureBottomToolbar(widthMeasureSpec, heightMeasureSpec);
+            return;
+        }
         try (TraceEvent tracing = TraceEvent.scoped("OmniboxSuggestionsList.Measure");
                 TimingMetric metric = OmniboxMetrics.recordSuggestionListMeasureTime();
                 TimingMetric metric2 = OmniboxMetrics.recordSuggestionListMeasureWallTime()) {
