diff --git a/chrome/browser/extensions/api/tabs/tabs_event_router.cc b/chrome/browser/extensions/api/tabs/tabs_event_router.cc
index c72256066c8085bac1a0492fe6b7a25c98f88f54..c96a7a6ba0c3d8184f411f2fe49b9104bdb53e91 100644
--- a/chrome/browser/extensions/api/tabs/tabs_event_router.cc
+++ b/chrome/browser/extensions/api/tabs/tabs_event_router.cc
@@ -173,10 +173,21 @@ TabsEventRouter::TabsEventRouter(Profile* profile)
   browser_tab_strip_tracker_.Init();
 
   tab_manager_scoped_observation_.Observe(g_browser_process->GetTabManager());
+
+#if BUILDFLAG(IS_ANDROID)
+  TabModelList::AddObserver(this);
+  if (!TabModelList::models().empty()) {
+    OnTabModelAdded();
+  }
+#endif
 }
 
 TabsEventRouter::~TabsEventRouter() {
   BrowserList::RemoveObserver(this);
+
+#if BUILDFLAG(IS_ANDROID)
+  TabModelList::RemoveObserver(this);
+#endif
 }
 
 bool TabsEventRouter::ShouldTrackBrowser(Browser* browser) {
@@ -380,7 +391,7 @@ void TabsEventRouter::DispatchTabClosingAt(TabStripModel* tab_strip_model,
   object_args.Set(tabs_constants::kWindowIdKey,
                   ExtensionTabUtil::GetWindowIdOfTab(contents));
   object_args.Set(tabs_constants::kWindowClosing,
-                  tab_strip_model->closing_all());
+                  tab_strip_model?tab_strip_model->closing_all():false);
   args.Append(std::move(object_args));
 
   Profile* profile = Profile::FromBrowserContext(contents->GetBrowserContext());
@@ -601,10 +612,12 @@ void TabsEventRouter::DispatchTabUpdatedEvent(
 }
 
 void TabsEventRouter::RegisterForTabNotifications(WebContents* contents) {
+#if !BUILDFLAG(IS_ANDROID)
   favicon_scoped_observations_.AddObservation(
       favicon::ContentFaviconDriver::FromWebContents(contents));
   zoom_scoped_observations_.AddObservation(
       ZoomController::FromWebContents(contents));
+#endif
 
   int tab_id = ExtensionTabUtil::GetTabId(contents);
   DCHECK(tab_entries_.find(tab_id) == tab_entries_.end());
@@ -612,12 +625,14 @@ void TabsEventRouter::RegisterForTabNotifications(WebContents* contents) {
 }
 
 void TabsEventRouter::UnregisterForTabNotifications(WebContents* contents) {
+#if !BUILDFLAG(IS_ANDROID)
   if (auto* zoom_controller = ZoomController::FromWebContents(contents);
       zoom_scoped_observations_.IsObservingSource(zoom_controller)) {
     zoom_scoped_observations_.RemoveObservation(zoom_controller);
   }
   favicon_scoped_observations_.RemoveObservation(
       favicon::ContentFaviconDriver::FromWebContents(contents));
+#endif
 
   int tab_id = ExtensionTabUtil::GetTabId(contents);
   int removed_count = tab_entries_.erase(tab_id);
