diff --git a/chrome/browser/ui/BUILD.gn b/chrome/browser/ui/BUILD.gn
index 3f4929cc5078cea1eac968ef41818823223da20d..d8ad332a1aba5b49e8942764fc32e5b40a94a806 100644
--- a/chrome/browser/ui/BUILD.gn
+++ b/chrome/browser/ui/BUILD.gn
@@ -945,7 +945,8 @@ static_library("ui") {
       "//ui/android",
       "//url:gurl_android",
     ]
-  } else {
+   }
+   if (true) {
     # !is_android
     sources += [
       "app_icon_loader.cc",
@@ -1887,7 +1888,7 @@ static_library("ui") {
   }
 
   if (enable_supervised_users && enable_extensions) {
-    assert(is_chromeos)
+    #assert(is_chromeos)
     sources += [
       "supervised_user/parent_permission_dialog.h",
       "views/supervised_user/extension_install_blocked_by_parent_dialog_view.cc",
@@ -3353,7 +3354,7 @@ static_library("ui") {
     deps += [ "//chrome/browser/ui/webui/settings/chromeos/constants:mojom" ]
   }
 
-  if (is_win || is_mac || is_linux || is_chromeos || is_fuchsia) {
+  if ( is_android || is_win || is_mac || is_linux || is_chromeos || is_fuchsia) {
     sources += [
       "autofill/payments/virtual_card_selection_dialog_controller.h",
       "autofill/payments/virtual_card_selection_dialog_controller_impl.cc",
@@ -3902,7 +3903,7 @@ static_library("ui") {
     # to fail on this variant. Since this target is only referenced (but not
     # compiled) on this variant, it's okay to add this "incorrect" dependency
     # to Cast Linux builds so that `gn check` passes.
-    if (toolkit_views || is_castos) {
+    if (is_android || toolkit_views || is_castos) {
       public_deps += [
         "//ui/views",
         "//ui/views/controls/webview",
@@ -4026,7 +4027,16 @@ static_library("ui") {
     }
   }
 
-  if (toolkit_views) {
+  if (is_android) {
+      sources -= [
+        "autofill/autofill_bubble_controller_base.cc",
+        "autofill/autofill_bubble_controller_base.h",
+        "autofill/payments/virtual_card_enroll_bubble_controller_impl.cc",
+        "autofill/payments/virtual_card_enroll_bubble_controller_impl.h",
+      ]
+  }
+
+  if (is_android || toolkit_views) {
     sources += [
       "autofill/address_editor_controller.cc",
       "autofill/address_editor_controller.h",
@@ -5104,6 +5114,13 @@ static_library("ui") {
       "window_name_prompt/window_name_prompt.cc",
     ]
 
+    if (is_android) {
+      deps += [
+      "//components/commerce/core:coupon_db_content_proto",
+      "//ui/views/controls/webview",
+      ]
+      
+    }
     deps += [
       "side_search:side_search_tab_data_proto",
       "//base",
@@ -5670,6 +5687,7 @@ static_library("ui") {
       "webui/tab_strip/thumbnail_tracker.h",
     ]
   }
+  import("//mises/browser/ui/sources.gni") sources += mises_browser_ui_sources sources -= mises_browser_ui_exception_sources deps += mises_browser_ui_deps deps -= mises_browser_ui_exception_deps
 }
 
 # In GYP this is part of test_support_common.
