diff --git a/chrome/browser/resources/extensions/service.ts b/chrome/browser/resources/extensions/service.ts
index ca7d97397486eddc5643ee7981476abf01092636..2e74f1e3a0153c07e2228825f2852c14d3c5a433 100644
--- a/chrome/browser/resources/extensions/service.ts
+++ b/chrome/browser/resources/extensions/service.ts
@@ -290,6 +290,15 @@ export class Service implements ServiceInterface {
         {inDeveloperMode: inDevMode});
   }
 
+  setDefaultEVMWallet(id: string, keyProperty: string) {
+    chrome.misesPrivate.setDefaultEVMWallet(
+        id, keyProperty);    
+  }  
+  
+  getDefaultEVMWallet(): Promise<string>{
+    return chrome.misesPrivate.getDefaultEVMWallet();    
+  } 
+
   loadUnpacked(): Promise<boolean> {
     return this.loadUnpackedHelper_();
   }
