diff --git a/chrome/browser/safe_browsing/BUILD.gn b/chrome/browser/safe_browsing/BUILD.gn
index 023d643451c168a07c4a4efd4a3c0b617d0d9d35..7703f4c1c5477a21aff48cb2423d07a4503a55f9 100644
--- a/chrome/browser/safe_browsing/BUILD.gn
+++ b/chrome/browser/safe_browsing/BUILD.gn
@@ -432,6 +432,8 @@ static_library("safe_browsing") {
           "tailored_security/unconsented_message_android.h",
           "telemetry/android/android_telemetry_service.cc",
           "telemetry/android/android_telemetry_service.h",
+          "cloud_content_scanning/deep_scanning_utils.cc",
+          "cloud_content_scanning/deep_scanning_utils.h",
         ]
         deps += [
           "//chrome/app/vector_icons:vector_icons",
