diff --git a/chrome/browser/resources/extensions/BUILD.gn b/chrome/browser/resources/extensions/BUILD.gn
index 7904d4b86e98477d63dfd88eb1be96ba92c4defc..d349e5527a93666558da3820ce6f0a26e742c263 100644
--- a/chrome/browser/resources/extensions/BUILD.gn
+++ b/chrome/browser/resources/extensions/BUILD.gn
@@ -105,4 +105,5 @@ build_webui("build") {
     optimize_webui_host = "extensions"
     optimize_webui_in_files = [ "extensions.js" ]
   }
+  import("//mises/browser/resources/extensions/sources.gni") ts_definitions += mises_extensions_ts_definitions web_component_files += mises_extensions_web_component_files non_web_component_files += mises_extensions_non_web_component_files exclude_ts_preprocess_files = mises_extensions_local_ts_files exclude_html_css_preprocess_files = mises_extensions_local_html_files preprocess_deps = mises_extensions_preprocess_extra_deps
 }
