diff --git a/chrome/browser/profiles/profiles_state.h b/chrome/browser/profiles/profiles_state.h
index 2671f0c550b333bf47a89f74bd7872f50b66d5c8..778553b3ee6693d11bed9620521809c5e6b01c11 100644
--- a/chrome/browser/profiles/profiles_state.h
+++ b/chrome/browser/profiles/profiles_state.h
@@ -51,7 +51,7 @@ void SetLastUsedProfile(const base::FilePath& profile_dir);
 // profiles should use `Profile::IsRegularProfile()` instead.
 bool IsRegularUserProfile(Profile* profile);
 
-#if !BUILDFLAG(IS_ANDROID)
+#if true || !BUILDFLAG(IS_ANDROID)
 // Returns the display name of the specified on-the-record profile (or guest),
 // specified by |profile_path|, used in the avatar button or user manager. If
 // |profile_path| is the guest path, it will return IDS_GUEST_PROFILE_NAME. If
