diff --git a/chrome/utility/BUILD.gn b/chrome/utility/BUILD.gn
index 69dedb71cc24e08c66580189393790254dd0c879..3ba0cf79b0a4b5a86e8b40af132b373c2e38d7ef 100644
--- a/chrome/utility/BUILD.gn
+++ b/chrome/utility/BUILD.gn
@@ -144,6 +144,15 @@ static_library("utility") {
     ]
   }
 
+  if (is_android) {
+    sources += [
+      "importer/bookmark_html_reader.cc",
+      "importer/bookmark_html_reader.h",
+      "importer/favicon_reencode.cc",
+      "importer/favicon_reencode.h",
+    ]
+  }
+
   if (enable_screen_ai_service) {
     deps += [ "//components/services/screen_ai" ]
   }
