diff --git a/chrome/browser/ui/android/omnibox/java/src/org/chromium/chrome/browser/omnibox/AutocompleteEditText.java b/chrome/browser/ui/android/omnibox/java/src/org/chromium/chrome/browser/omnibox/AutocompleteEditText.java
index 082f9b7ac6782007703c9ab73ca0109dbdbd2a17..b8b7a9f421d474f751f902fa0ba2f74d6717e618 100644
--- a/chrome/browser/ui/android/omnibox/java/src/org/chromium/chrome/browser/omnibox/AutocompleteEditText.java
+++ b/chrome/browser/ui/android/omnibox/java/src/org/chromium/chrome/browser/omnibox/AutocompleteEditText.java
@@ -262,7 +262,11 @@ public class AutocompleteEditText extends VerticallyFixedEditText
             if (DEBUG) Log.i(TAG, "Ignoring accessibility event from autocomplete.");
             return;
         }
-        super.sendAccessibilityEventUnchecked(event);
+        try {
+            super.sendAccessibilityEventUnchecked(event);
+        } catch (IllegalStateException e) {
+            // accessibility is off.
+        }
     }
 
     @Override
