diff --git a/chrome/browser/tabmodel/android/java/src/org/chromium/chrome/browser/tabmodel/EmptyTabModelFilter.java b/chrome/browser/tabmodel/android/java/src/org/chromium/chrome/browser/tabmodel/EmptyTabModelFilter.java
index 7dc09bea688ddfda9eb92e35c68c36b228630573..4a169835bfa21705a2841ffc66ddc3393d783024 100644
--- a/chrome/browser/tabmodel/android/java/src/org/chromium/chrome/browser/tabmodel/EmptyTabModelFilter.java
+++ b/chrome/browser/tabmodel/android/java/src/org/chromium/chrome/browser/tabmodel/EmptyTabModelFilter.java
@@ -50,6 +50,11 @@ public class EmptyTabModelFilter extends TabModelFilter {
         return getTabModel().index();
     }
 
+    @Override
+    public int getLastNonExtensionActiveIndex() {
+        return getTabModel().getLastNonExtensionActiveIndex();
+    }
+
     @Override
     public int getCount() {
         return getTabModel().getCount();
