diff --git a/chrome/renderer/BUILD.gn b/chrome/renderer/BUILD.gn
index 6a646ba3e19753001ff4f1278a543b52a03587c4..8aa11daed66603767c95b9c951da286d43955422 100644
--- a/chrome/renderer/BUILD.gn
+++ b/chrome/renderer/BUILD.gn
@@ -390,7 +390,7 @@ static_library("renderer") {
     ]
   }
 
-  if (!is_android) {
+  if (!is_android || is_android) {
     sources += [
       "accessibility/ax_tree_distiller.cc",
       "accessibility/ax_tree_distiller.h",
@@ -403,6 +403,12 @@ static_library("renderer") {
       "searchbox/searchbox_extension.cc",
       "searchbox/searchbox_extension.h",
     ]
+    if (is_android) {
+      sources -= [
+        "media/chrome_speech_recognition_client.cc",
+        "media/chrome_speech_recognition_client.h",
+      ]
+    }
 
     deps += [
       "//chrome/common/accessibility:mojo_bindings",
