diff --git a/chrome/browser/resources/new_tab_page/BUILD.gn b/chrome/browser/resources/new_tab_page/BUILD.gn
index 445a8a6e7a95798709cc7563da237dee56efd34a..2741bf1b4e1e0c0c0087eda9c2b90eab81bf56fb 100644
--- a/chrome/browser/resources/new_tab_page/BUILD.gn
+++ b/chrome/browser/resources/new_tab_page/BUILD.gn
@@ -8,7 +8,7 @@ import("./new_tab_page.gni")
 
 import("//ui/webui/resources/tools/build_webui.gni")
 
-assert(!is_android)
+#assert(!is_android)
 
 grd_prefix = "new_tab_page"
 
