diff --git a/chrome/browser/new_tab_page/modules/drive/BUILD.gn b/chrome/browser/new_tab_page/modules/drive/BUILD.gn
index 362f0861a1934923456dda88a1fd8051be16b89a..c2f081e7927eced5941a57d1a4800c1ba32700ef 100644
--- a/chrome/browser/new_tab_page/modules/drive/BUILD.gn
+++ b/chrome/browser/new_tab_page/modules/drive/BUILD.gn
@@ -4,7 +4,7 @@
 
 import("//mojo/public/tools/bindings/mojom.gni")
 
-assert(!is_android)
+#assert(!is_android)
 
 mojom("mojo_bindings") {
   sources = [ "drive.mojom" ]
