diff --git a/chrome/browser/resources/welcome/BUILD.gn b/chrome/browser/resources/welcome/BUILD.gn
index 800ec9173188f4ac25b99d6f366d3f277a2abb3c..f6b6ef977eeceb8bbe8bb12bc7253507d5bbf841 100644
--- a/chrome/browser/resources/welcome/BUILD.gn
+++ b/chrome/browser/resources/welcome/BUILD.gn
@@ -6,7 +6,7 @@ import("//chrome/browser/resources/tools/build_webui.gni")
 import("//chrome/common/features.gni")
 import("//ui/webui/resources/tools/generate_grd.gni")
 
-assert(!is_chromeos_ash && !is_android)
+# assert(!is_chromeos_ash && !is_android)
 
 if (is_chrome_branded) {
   generate_grd("build_icons_grdp") {
