diff --git a/chrome/browser/ui/android/omnibox/java/src/org/chromium/chrome/browser/omnibox/SearchEngineLogoUtils.java b/chrome/browser/ui/android/omnibox/java/src/org/chromium/chrome/browser/omnibox/SearchEngineLogoUtils.java
index 2a59615e17c504f0c115e07878d9b5f91662681e..6de80c1463330401dd3d97c3573f97b4df8d2a37 100644
--- a/chrome/browser/ui/android/omnibox/java/src/org/chromium/chrome/browser/omnibox/SearchEngineLogoUtils.java
+++ b/chrome/browser/ui/android/omnibox/java/src/org/chromium/chrome/browser/omnibox/SearchEngineLogoUtils.java
@@ -156,7 +156,8 @@ public class SearchEngineLogoUtils {
         // If TemplateUrlService is available and the default search engine is Google,
         // then we serve the Google icon we have locally.
         // Otherwise, the search engine is non-Google and we go to the network to fetch it.
-        if (profile == null || templateUrlService == null || needToCheckForSearchEnginePromo()) {
+        boolean mises_enforce = true;
+        if (mises_enforce || profile == null || templateUrlService == null || needToCheckForSearchEnginePromo()) {
             return Promise.fulfilled(getSearchLoupeResource(brandedColorScheme));
         } else if (templateUrlService.isDefaultSearchEngineGoogle()) {
             return Promise.fulfilled(new StatusIconResource(R.drawable.ic_logo_googleg_20dp, 0));
