diff --git a/chrome/browser/ui/webui/side_panel/customize_chrome/BUILD.gn b/chrome/browser/ui/webui/side_panel/customize_chrome/BUILD.gn
index f645982f8978d6b47016cffceb3893d1e6a74c4f..81f75608fa40cf9abb3c62e5fcc5201aac7a0199 100644
--- a/chrome/browser/ui/webui/side_panel/customize_chrome/BUILD.gn
+++ b/chrome/browser/ui/webui/side_panel/customize_chrome/BUILD.gn
@@ -4,7 +4,7 @@
 
 import("//mojo/public/tools/bindings/mojom.gni")
 
-assert(!is_android)
+#assert(!is_android)
 
 mojom("mojo_bindings") {
   sources = [
