diff --git a/chrome/browser/resource_coordinator/tab_manager.cc b/chrome/browser/resource_coordinator/tab_manager.cc
index 07911e97b05158047ffe6b1cec9583bb75e955bd..aa8219fb9c03c46d064c84f05a72986d6a24d30f 100644
--- a/chrome/browser/resource_coordinator/tab_manager.cc
+++ b/chrome/browser/resource_coordinator/tab_manager.cc
@@ -197,7 +197,7 @@ WebContents* TabManager::DiscardTabByExtension(content::WebContents* contents) {
     TabLifecycleUnitExternal* tab_lifecycle_unit_external =
         TabLifecycleUnitExternal::FromWebContents(contents);
     DCHECK(tab_lifecycle_unit_external);
-    if (tab_lifecycle_unit_external->DiscardTab(
+    if (tab_lifecycle_unit_external && tab_lifecycle_unit_external->DiscardTab(
             LifecycleUnitDiscardReason::EXTERNAL)) {
       return tab_lifecycle_unit_external->GetWebContents();
     }
