diff --git a/chrome/android/chrome_public_apk_tmpl.gni b/chrome/android/chrome_public_apk_tmpl.gni
index 43a9301530e6ee92c79ea5b375f2fcecfadcb34b..05c2dbffeac5b34b4a4261f91eba950a3abef95c 100644
--- a/chrome/android/chrome_public_apk_tmpl.gni
+++ b/chrome/android/chrome_public_apk_tmpl.gni
@@ -337,6 +337,7 @@ template("chrome_common_apk_or_module_tmpl") {
 
       proguard_enabled = true
       proguard_configs = [ "//chrome/android/proguard/main.flags" ]
+      proguard_configs += [ "//mises/android/java/proguard.flags" ]
       if (_is_monochrome) {
         proguard_configs +=
             [ "//android_webview/nonembedded/java/proguard.flags" ]
@@ -748,6 +749,7 @@ template("chrome_common_apk_or_module_tmpl") {
                                  "version_code",
                                  "version_name",
                                ])
+    _wireguard_target = "//mises/third_party/android_deps:com_wireguard_android_J__unpack_aar" deps += [ _wireguard_target] _libwireguard_dir = get_label_info("//mises/third_party/android_deps:com_wireguard_android($default_toolchain)","target_out_dir") + "/com_wireguard_android_java/jni" loadable_modules +=[ "$_libwireguard_dir/$android_app_abi/libwg-go.so" ]
   }
 }
 
@@ -776,6 +778,7 @@ template("monochrome_public_common_apk_or_module_tmpl") {
     if (!defined(is_monochrome)) {
       is_trichrome = true
     }
+    deps += mises_monochrome_public_apk_or_module_tmpl_deps loadable_modules = mises_monochrome_public_apk_or_module_tmpl_loadable_modules secondary_abi_loadable_modules = mises_monochrome_public_apk_or_module_tmpl_secondary_abi_loadable_modules
   }
 }
 template("chrome_public_common_apk_or_module_tmpl") {
