diff --git a/chrome/test/BUILD.gn b/chrome/test/BUILD.gn
index 26702ebe305dae944d7f425ec610058e51c43656..b77a8f7a70d1209151f97e2b3e52bada92c457a0 100644
--- a/chrome/test/BUILD.gn
+++ b/chrome/test/BUILD.gn
@@ -578,6 +578,7 @@ static_library("test_support") {
   ]
 
   data = []
+  import("//mises/chromium_src/chrome/test/sources.gni") deps += mises_chromium_src_chrome_test_test_support_deps deps -= mises_chromium_src_chrome_test_test_support_exception_deps sources += mises_chromium_src_chrome_test_test_support_sources
   data_deps = []
 
   if (enable_pdf) {
@@ -9356,6 +9357,10 @@ test("unit_tests") {
       "//tools/json_schema_compiler/test:unit_tests",
     ]
 
+    if (is_android) {
+      deps -= ["//extensions:extensions_resources",]
+    }
+
     if (enable_service_discovery) {
       sources += [ "../browser/extensions/api/mdns/mdns_api_unittest.cc" ]
     }
