diff --git a/chrome/browser/tab_group/java/src/org/chromium/chrome/browser/tasks/tab_groups/TabGroupModelFilter.java b/chrome/browser/tab_group/java/src/org/chromium/chrome/browser/tasks/tab_groups/TabGroupModelFilter.java
index b1396ab7d6f14189e3c88d1e2cc65d9148c09b45..8691870f3ed811959b84f2916643e36fc33aac06 100644
--- a/chrome/browser/tab_group/java/src/org/chromium/chrome/browser/tasks/tab_groups/TabGroupModelFilter.java
+++ b/chrome/browser/tab_group/java/src/org/chromium/chrome/browser/tasks/tab_groups/TabGroupModelFilter.java
@@ -850,6 +850,11 @@ public class TabGroupModelFilter extends TabModelFilter {
         return mAbsentSelectedTab == null;
     }
 
+    @Override
+    public int getLastNonExtensionActiveIndex() {
+        return INVALID_TAB_INDEX;
+    }
+
     @Override
     @VisibleForTesting(otherwise = VisibleForTesting.PACKAGE_PRIVATE)
     public void markTabStateInitialized() {
