diff --git a/chrome/browser/resources/new_tab_page/icons/BUILD.gn b/chrome/browser/resources/new_tab_page/icons/BUILD.gn
index 4a5453986d037709fe2bd108cd1dcaee66f7137d..0384242ab74c13f29314dec787bacb68c20f8940 100644
--- a/chrome/browser/resources/new_tab_page/icons/BUILD.gn
+++ b/chrome/browser/resources/new_tab_page/icons/BUILD.gn
@@ -4,7 +4,7 @@
 
 import("//ui/webui/resources/tools/generate_grd.gni")
 
-assert(!is_android)
+#assert(!is_android)
 
 generate_grd("build_grdp") {
   grd_prefix = "new_tab_page"
