diff --git a/chrome/browser/resources/access_code_cast/BUILD.gn b/chrome/browser/resources/access_code_cast/BUILD.gn
index 479a3a70d3abe16b0de5f62864107b3137fe6574..b00f13a90c4e33033a576da80c6ad45653982a21 100644
--- a/chrome/browser/resources/access_code_cast/BUILD.gn
+++ b/chrome/browser/resources/access_code_cast/BUILD.gn
@@ -4,7 +4,7 @@
 
 import("//ui/webui/resources/tools/build_webui.gni")
 
-assert(!is_android, "access_code_cast is not for android.")
+# assert(!is_android, "access_code_cast is not for android.")
 
 build_webui("build") {
   grd_prefix = "access_code_cast"
