diff --git a/chrome/browser/resources/BUILD.gn b/chrome/browser/resources/BUILD.gn
index 745ab36b7f834eb8a0d0274c6393bcaaf1534faa..f25d46dfa353e8038f9d048a67c1406bf5f03613 100644
--- a/chrome/browser/resources/BUILD.gn
+++ b/chrome/browser/resources/BUILD.gn
@@ -21,7 +21,7 @@ group("resources") {
     "segmentation_internals:resources",
   ]
 
-  if (!is_android) {
+  if (true || !is_android) {
     public_deps += [
       ":app_icon_resources",
       ":component_extension_resources",
@@ -124,12 +124,20 @@ group("resources") {
       "app_settings:resources",
     ]
   }
+  if (is_android) {
+    public_deps -= [
+      ":dev_ui_paks",
+      "lens/shared:resources",
+      "lens/overlay:resources",
+      "lens/overlay/search_bubble:resources"
+    ]
+  }
 
   if (is_linux || is_chromeos) {
     public_deps += [ "webui_js_error:resources" ]
   }
 
-  if (!is_android && !is_chromeos_ash) {
+  if (is_android || !is_android && !is_chromeos_ash) {
     public_deps += [
       "inline_login:resources",
       "intro:resources",
@@ -180,7 +188,7 @@ if (is_chromeos_ash && enable_js_type_check) {
   }
 }
 
-if (!is_android) {
+if (true || !is_android) {
   grit("component_extension_resources") {
     source = "component_extension_resources.grd"
 
@@ -297,7 +305,8 @@ group("dev_ui_resources") {
       "//chrome/browser/resources/feed_internals:resources",
       "//chrome/browser/resources/webapks:resources",
     ]
-  } else {
+  }
+  if (true) {
     public_deps += [
       "media_router/internals:resources",
       "on_device_internals:resources",
@@ -355,7 +364,8 @@ repack("dev_ui_paks") {
       "$root_gen_dir/chrome/feed_internals_resources.pak",
       "$root_gen_dir/chrome/webapks_resources.pak",
     ]
-  } else {
+  } 
+  if (true) {
     sources += [
       "$root_gen_dir/chrome/media_router_internals_resources.pak",
       "$root_gen_dir/chrome/on_device_internals_resources.pak",
