diff --git a/chrome/browser/resources/settings/settings.gni b/chrome/browser/resources/settings/settings.gni
index 4952b09b0e7a6b8ba558c0edc84e640ba4bc3971..755474b2fe91bce67d4edd818d9c054a5df8890b 100644
--- a/chrome/browser/resources/settings/settings.gni
+++ b/chrome/browser/resources/settings/settings.gni
@@ -4,7 +4,7 @@
 
 import("//third_party/closure_compiler/compile_js.gni")
 
-assert(!is_android, "Android does not use these settings")
+#assert(!is_android, "Android does not use these settings")
 
 settings_closure_flags =
     default_closure_args + [
