diff --git a/chrome/android/java/src/org/chromium/chrome/browser/bookmarks/BookmarkBridge.java b/chrome/android/java/src/org/chromium/chrome/browser/bookmarks/BookmarkBridge.java
index 9064b9b6ef98aeb75de231eb36033fa810f7e6ca..138946e62eb83a3e37f16238fea26f7d52a0bd6d 100644
--- a/chrome/android/java/src/org/chromium/chrome/browser/bookmarks/BookmarkBridge.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/bookmarks/BookmarkBridge.java
@@ -46,11 +46,22 @@ import java.util.ArrayList;
 import java.util.HashSet;
 import java.util.List;
 
+import android.content.Intent;
+import android.net.Uri;
+import android.widget.Toast;
+import android.content.Context;
+import org.chromium.chrome.browser.document.ChromeLauncherActivity;
+import android.provider.Browser;
+import org.chromium.chrome.browser.IntentHandler;
+import org.chromium.ui.base.PageTransition;
+
+import org.chromium.ui.base.WindowAndroid;
+
 /**
  * Provides the communication channel for Android to fetch and manipulate the
  * bookmark model stored in native.
  */
-class BookmarkBridge {
+public class BookmarkBridge {
     private long mNativeBookmarkBridge;
     private boolean mIsDestroyed;
     private boolean mIsDoingExtensiveChanges;
@@ -190,6 +201,48 @@ class BookmarkBridge {
         return mIsNativeBookmarkModelLoaded;
     }
 
+
+    public void startImportBookmarks(final WindowAndroid window) {
+        assert mIsNativeBookmarkModelLoaded;
+
+        BookmarkBridgeJni.get().importBookmarks(mNativeBookmarkBridge, window);
+    }
+
+    public void startExportBookmarks() {
+        assert mIsNativeBookmarkModelLoaded;
+
+        BookmarkBridgeJni.get().exportBookmarks(mNativeBookmarkBridge);
+    }
+
+    @CalledByNative
+    public void bookmarksImported(String message) {
+        Context context = ContextUtils.getApplicationContext();
+
+        Toast.makeText(ContextUtils.getApplicationContext(), message, Toast.LENGTH_LONG).show();
+    }
+
+    @CalledByNative
+    public void bookmarksExported(String bookmarksPath) {
+        String url = bookmarksPath;
+
+        Context context = ContextUtils.getApplicationContext();
+
+        url = "file://" + url;
+
+        Intent intent = new Intent(Intent.ACTION_VIEW, Uri.parse(url));
+        intent.putExtra(Browser.EXTRA_APPLICATION_ID,
+                        context.getPackageName());
+        intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
+        intent.putExtra(IntentHandler.EXTRA_PAGE_TRANSITION_TYPE, PageTransition.AUTO_BOOKMARK);
+
+        // If the bookmark manager is shown in a tab on a phone (rather than in a separate
+        // activity) the component name may be null. Send the intent through
+        // ChromeLauncherActivity instead to avoid crashing. See crbug.com/615012.
+        intent.setClass(context, ChromeLauncherActivity.class);
+
+        IntentHandler.startActivityForTrustedIntent(intent);
+    }
+
     /**
      * Schedules a runnable to run after the bookmark model is loaded. If the
      * model is already loaded, executes the runnable immediately. If not, also
@@ -1084,5 +1137,8 @@ class BookmarkBridge {
         void reorderChildren(long nativeBookmarkBridge, BookmarkBridge caller, BookmarkId parent,
                 long[] orderedNodes);
         boolean isBookmarked(long nativeBookmarkBridge, GURL url);
+
+	void importBookmarks(long nativeBookmarkBridge, WindowAndroid window);
+	void exportBookmarks(long nativeBookmarkBridge);
     }
 }
