diff --git a/chrome/android/features/tab_ui/java/src/org/chromium/chrome/browser/tasks/tab_management/TabSwitcherCoordinator.java b/chrome/android/features/tab_ui/java/src/org/chromium/chrome/browser/tasks/tab_management/TabSwitcherCoordinator.java
index a9974b5800c68ad3999b938fd091f4ad81550e81..194f8f2c4401ddcbbb57c0a055849c92676f5e62 100644
--- a/chrome/android/features/tab_ui/java/src/org/chromium/chrome/browser/tasks/tab_management/TabSwitcherCoordinator.java
+++ b/chrome/android/features/tab_ui/java/src/org/chromium/chrome/browser/tasks/tab_management/TabSwitcherCoordinator.java
@@ -70,6 +70,8 @@ import org.chromium.ui.resources.dynamics.DynamicResourceLoader;
 import java.util.ArrayList;
 import java.util.List;
 
+import org.chromium.base.ContextUtils;
+
 /**
  * Parent coordinator that is responsible for showing a grid or carousel of tabs for the main
  * TabSwitcher UI.
@@ -177,6 +179,11 @@ public class TabSwitcherCoordinator
             @Nullable BackPressManager backPressManager) {
         try (TraceEvent e = TraceEvent.scoped("TabSwitcherCoordinator.constructor")) {
             mActivity = activity;
+	    final String active_tabswitcher = ContextUtils.getAppSharedPreferences().getString("active_tabswitcher", "default");
+            if (active_tabswitcher.equals("list"))
+              mode = TabListCoordinator.TabListMode.LIST;
+            if (active_tabswitcher.equals("default") || active_tabswitcher.equals("grid"))
+              mode = TabListCoordinator.TabListMode.GRID;
             mMode = mode;
             mTabModelSelector = tabModelSelector;
             mContainer = container;
