diff --git a/chrome/browser/about_flags.cc b/chrome/browser/about_flags.cc
index 3dd02e4175a3884266dd55c648976ea64837f794..ed474096d28eeb84e5d306a799c76d27171652c6 100644
--- a/chrome/browser/about_flags.cc
+++ b/chrome/browser/about_flags.cc
@@ -4231,6 +4231,7 @@ const FeatureEntry::FeatureVariation
 //
 // When adding a new choice, add it to the end of the list.
 const FeatureEntry kFeatureEntries[] = {
+MISES_ABOUT_FLAGS_FEATURE_ENTRIES
 // Include generated flags for flag unexpiry; see //docs/flag_expiry.md and
 // //tools/flags/generate_unexpire_flags.py.
 #include "build/chromeos_buildflags.h"
