diff --git a/chrome/browser/resources/access_code_cast/BUILD.gn b/chrome/browser/resources/access_code_cast/BUILD.gn
index b08dc72503ca86ab583fdccc5c0d5bf50041169b..ddb694d4a63f23c076a28146d2d65338bcd9f519 100644
--- a/chrome/browser/resources/access_code_cast/BUILD.gn
+++ b/chrome/browser/resources/access_code_cast/BUILD.gn
@@ -4,7 +4,7 @@
 
 import("//ui/webui/resources/tools/build_webui.gni")
 
-assert(!is_android, "access_code_cast is not for android.")
+# assert(!is_android, "access_code_cast is not for android.")
 
 build_webui("build") {
   grd_prefix = "access_code_cast"
