diff --git a/chrome/browser/resources/new_tab_page/BUILD.gn b/chrome/browser/resources/new_tab_page/BUILD.gn
index c08ddefe4a4e64e0ae7e4c6086da4efe591768b2..0a128bbd349ad11809e9e366106842270fc8ab98 100644
--- a/chrome/browser/resources/new_tab_page/BUILD.gn
+++ b/chrome/browser/resources/new_tab_page/BUILD.gn
@@ -7,7 +7,7 @@ import("./new_tab_page.gni")
 
 import("//ui/webui/resources/tools/build_webui.gni")
 
-assert(!is_android)
+#assert(!is_android)
 
 grd_prefix = "new_tab_page"
 
