diff --git a/chrome/browser/ui/android/omnibox/java/src/org/chromium/chrome/browser/omnibox/UrlBarMediator.java b/chrome/browser/ui/android/omnibox/java/src/org/chromium/chrome/browser/omnibox/UrlBarMediator.java
index 2b794f1590f6626289ac0060a1eacaec49e9c4aa..abea919ca8a69cbde115617ddfec3083538c43b7 100644
--- a/chrome/browser/ui/android/omnibox/java/src/org/chromium/chrome/browser/omnibox/UrlBarMediator.java
+++ b/chrome/browser/ui/android/omnibox/java/src/org/chromium/chrome/browser/omnibox/UrlBarMediator.java
@@ -93,12 +93,12 @@ class UrlBarMediator implements UrlBar.UrlBarTextContextMenuDelegate, UrlBar.Url
      */
     public boolean setUrlBarData(
             UrlBarData data, @ScrollType int scrollType, @SelectionState int selectionState) {
-        if (data.originEndIndex == data.originStartIndex) {
+        if (data != null && data.url != null && data.originEndIndex == data.originStartIndex) {
             scrollType = UrlBar.ScrollType.SCROLL_TO_BEGINNING;
         }
 
         // Do not scroll to the end of the host for URLs such as data:, javascript:, etc...
-        if (data.url != null
+        if (data != null && data.url != null
                 && data.displayText != null
                 && data.originEndIndex == data.displayText.length()) {
             String scheme = data.url.getScheme();
@@ -125,6 +125,8 @@ class UrlBarMediator implements UrlBar.UrlBarTextContextMenuDelegate, UrlBar.Url
     }
 
     private void pushTextToModel() {
+        if (mUrlBarData == null)
+          return;
         CharSequence text =
                 !mHasFocus ? mUrlBarData.displayText : mUrlBarData.getEditingOrDisplayText();
         CharSequence textForAutofillServices = text;
