diff --git a/chrome/android/java/src/org/chromium/chrome/browser/toolbar/ToolbarManager.java b/chrome/android/java/src/org/chromium/chrome/browser/toolbar/ToolbarManager.java
index 5a17dfba35a99d4029b50279ccf1de0e3cd3f278..2e3086bbf3c9267abf2dcd85648a70629dfdf0bd 100644
--- a/chrome/android/java/src/org/chromium/chrome/browser/toolbar/ToolbarManager.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/toolbar/ToolbarManager.java
@@ -195,6 +195,10 @@ import org.chromium.url.GURL;
 
 import java.util.List;
 
+import org.chromium.chrome.browser.toolbar.top.TabSwitcherActionMenuCoordinator;
+import android.view.View.OnLongClickListener;
+import org.chromium.base.ContextUtils;
+
 /**
  * Contains logic for managing the toolbar visual component. This class manages the interactions
  * with the rest of the application to ensure the toolbar is always visually up to date.
@@ -950,6 +954,7 @@ public class ToolbarManager
 
                     @Override
                     public void onUrlUpdated(Tab tab) {
+                        mLocationBarModel.notifySecurityStateChanged();
                         // Update the SSL security state as a result of this notification as it will
                         // sometimes be the only update we receive.
                         updateTabLoadingState(true);
@@ -1631,7 +1636,7 @@ public class ToolbarManager
                     TabSwitcherActionMenuCoordinator.createOnLongClickListener(
                             menuItemId -> mAppMenuDelegate.onOptionsItemSelected(menuItemId, null),
                             profile,
-                            mTabModelSelectorSupplier);
+                            mTabModelSelectorSupplier, newTabClickHandler);
             mTabSwitcherButtonCoordinator.initializeWithNative(
                     newTabClickHandler,
                     tabSwitcherLongClickListener,
