diff --git a/chrome/browser/resources/new_tab_page_instant/BUILD.gn b/chrome/browser/resources/new_tab_page_instant/BUILD.gn
index d63f6317c5053f093d0a40dd50f0623b3cc44796..2c4e1c7ecee1e40559da7e96255d7407a03b197f 100644
--- a/chrome/browser/resources/new_tab_page_instant/BUILD.gn
+++ b/chrome/browser/resources/new_tab_page_instant/BUILD.gn
@@ -4,7 +4,7 @@
 
 import("//ui/webui/resources/tools/build_webui.gni")
 
-assert(is_linux || is_chromeos || is_win || is_mac)
+# assert(is_linux || is_chromeos || is_win || is_mac)
 assert(!is_fuchsia, "Fuchsia shouldn't use anything in //chrome")
 
 build_webui("build") {
