diff --git a/chrome/browser/signin/primary_account_policy_manager.cc b/chrome/browser/signin/primary_account_policy_manager.cc
index b2ac2f696f3c80aef0df8c592d4ca84144300036..0488312ef1836c861805eab6b66adbff2224ef82 100644
--- a/chrome/browser/signin/primary_account_policy_manager.cc
+++ b/chrome/browser/signin/primary_account_policy_manager.cc
@@ -301,7 +301,7 @@ void PrimaryAccountPolicyManager::OnUserConfirmedProfileDeletion(
   delete_profile_dialog_manager_.reset();
 
   DCHECK(profiles::IsMultipleProfilesEnabled());
-
+#if !BUILDFLAG(IS_ANDROID)
   g_browser_process->profile_manager()
       ->GetDeleteProfileHelper()
       .MaybeScheduleProfileForDeletion(
@@ -310,5 +310,6 @@ void PrimaryAccountPolicyManager::OnUserConfirmedProfileDeletion(
               ? base::DoNothing()
               : base::BindOnce(&webui::OpenNewWindowForProfile),
           ProfileMetrics::DELETE_PROFILE_PRIMARY_ACCOUNT_NOT_ALLOWED);
+#endif
 }
 #endif  // defined(TOOLKIT_VIEWS) && !BUILDFLAG(IS_CHROMEOS)
