diff --git a/chrome/browser/resources/management/BUILD.gn b/chrome/browser/resources/management/BUILD.gn
index 8b3f1b048a1c1e6f22fa46fd5caf8d133d4837a9..d6be4e547af5fa3e1c945fe6e4e74ce731599d65 100644
--- a/chrome/browser/resources/management/BUILD.gn
+++ b/chrome/browser/resources/management/BUILD.gn
@@ -5,7 +5,7 @@
 import("//chrome/common/features.gni")
 import("//ui/webui/resources/tools/build_webui.gni")
 
-assert(!is_android, "!is_android check failed")
+# assert(!is_android, "!is_android check failed")
 
 build_webui("build") {
   grd_prefix = "management"
