diff --git a/chrome/browser/resources/app_settings/BUILD.gn b/chrome/browser/resources/app_settings/BUILD.gn
index 5600a51b43ad11e65d2c5929cb96cf7f7b6e8a15..00b7ba1d666ccb6e0c40047a1b83eddfea3b08c8 100644
--- a/chrome/browser/resources/app_settings/BUILD.gn
+++ b/chrome/browser/resources/app_settings/BUILD.gn
@@ -5,7 +5,7 @@
 import("//chrome/browser/resources/tools/build_webui.gni")
 import("//chrome/common/features.gni")
 
-assert(is_win || is_mac || is_linux || is_fuchsia)
+assert(true || is_win || is_mac || is_linux || is_fuchsia)
 
 build_webui("build") {
   grd_prefix = "app_settings"
