diff --git a/chrome/browser/resources/extensions/service.ts b/chrome/browser/resources/extensions/service.ts
index 1081d384b412d6ec6ea2c8c6f8d72031854b7e62..38ad75dc593759ba658977edf5d20e3be973fa60 100644
--- a/chrome/browser/resources/extensions/service.ts
+++ b/chrome/browser/resources/extensions/service.ts
@@ -254,6 +254,11 @@ export class Service implements ServiceInterface {
         {inDeveloperMode: inDevMode});
   }
 
+  setDefaultEVMWallet(id: string, keyProperty: string) {
+    chrome.misesPrivate.setDefaultEVMWallet(
+        id, keyProperty);    
+  }  
+
   loadUnpacked(): Promise<boolean> {
     return this.loadUnpackedHelper_();
   }
