diff --git a/chrome/android/java/src/org/chromium/chrome/browser/crash/ChromeMinidumpUploaderDelegate.java b/chrome/android/java/src/org/chromium/chrome/browser/crash/ChromeMinidumpUploaderDelegate.java
index af005d5f955c0930a4b64e449e08b1a4c6cd4a75..93f2a9029766cc9df5052c4da655021a1b48b12b 100644
--- a/chrome/android/java/src/org/chromium/chrome/browser/crash/ChromeMinidumpUploaderDelegate.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/crash/ChromeMinidumpUploaderDelegate.java
@@ -12,6 +12,8 @@ import org.chromium.components.minidump_uploader.MinidumpUploaderDelegate;
 import org.chromium.components.minidump_uploader.util.CrashReportingPermissionManager;
 import org.chromium.components.minidump_uploader.util.NetworkPermissionUtil;
 
+import org.chromium.chrome.browser.MisesConfig;
+
 import java.io.File;
 
 /** Chrome-specific implementations for minidump uploading logic. */
@@ -54,7 +56,8 @@ public class ChromeMinidumpUploaderDelegate implements MinidumpUploaderDelegate
         return new CrashReportingPermissionManager() {
             @Override
             public boolean isClientInSampleForCrashes() {
-                return mPermissions.getBoolean(IS_CLIENT_IN_SAMPLE_FOR_CRASHES, true);
+                boolean mises_enforced = MisesConfig.FORCE_CRASH_REPORT;
+                return mises_enforced || mPermissions.getBoolean(IS_CLIENT_IN_SAMPLE_FOR_CRASHES, true);
             }
 
             @Override
@@ -64,13 +67,15 @@ public class ChromeMinidumpUploaderDelegate implements MinidumpUploaderDelegate
 
             @Override
             public boolean isUsageAndCrashReportingPermittedByPolicy() {
-                return PrivacyPreferencesManagerImpl.getInstance()
+                boolean mises_enforced = MisesConfig.FORCE_CRASH_REPORT;
+                return mises_enforced || PrivacyPreferencesManagerImpl.getInstance()
                         .isUsageAndCrashReportingPermittedByPolicy();
             }
 
             @Override
             public boolean isUsageAndCrashReportingPermittedByUser() {
-                return PrivacyPreferencesManagerImpl.getInstance()
+                boolean mises_enforced = MisesConfig.FORCE_CRASH_REPORT;
+                return mises_enforced || PrivacyPreferencesManagerImpl.getInstance()
                         .isUsageAndCrashReportingPermittedByUser();
             }
 
