diff --git a/chrome/android/features/tab_ui/java/src/org/chromium/chrome/browser/tasks/tab_management/TabSwitcherMediator.java b/chrome/android/features/tab_ui/java/src/org/chromium/chrome/browser/tasks/tab_management/TabSwitcherMediator.java
index 7ea24aa1bd7b8245136af864432fb0927fa0bac3..3725b37a364aae7e51240f343415454ad40423a7 100644
--- a/chrome/android/features/tab_ui/java/src/org/chromium/chrome/browser/tasks/tab_management/TabSwitcherMediator.java
+++ b/chrome/android/features/tab_ui/java/src/org/chromium/chrome/browser/tasks/tab_management/TabSwitcherMediator.java
@@ -74,6 +74,7 @@ import org.chromium.components.browser_ui.widget.gesture.BackPressHandler;
 import org.chromium.ui.base.DeviceFormFactor;
 import org.chromium.ui.modelutil.PropertyModel;
 
+import org.chromium.base.ContextUtils;
 import java.util.List;
 
 /**
@@ -311,6 +312,11 @@ class TabSwitcherMediator implements TabSwitcher.Controller, TabListRecyclerView
         mTabModelSelector = tabModelSelector;
         mBrowserControlsStateProvider = browserControlsStateProvider;
         mMultiWindowModeStateDispatcher = multiWindowModeStateDispatcher;
+        final String active_tabswitcher = ContextUtils.getAppSharedPreferences().getString("active_tabswitcher", "default");
+        if (active_tabswitcher.equals("list"))
+          mode = TabListCoordinator.TabListMode.LIST;
+        if (active_tabswitcher.equals("default") || active_tabswitcher.equals("grid"))
+          mode = TabListCoordinator.TabListMode.GRID;
         mMode = mode;
         mContainerViewModel.set(MODE, mode);
         mContext = context;
