diff --git a/chrome/android/java/src/org/chromium/chrome/browser/tabmodel/TabModelJniBridge.java b/chrome/android/java/src/org/chromium/chrome/browser/tabmodel/TabModelJniBridge.java
index aa403bd3743ad45b8e1783f14e955f9e4bc76fc6..e82bde763a37513352cb6f66a9f45a8472a67f51 100644
--- a/chrome/android/java/src/org/chromium/chrome/browser/tabmodel/TabModelJniBridge.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/tabmodel/TabModelJniBridge.java
@@ -22,6 +22,7 @@ import org.chromium.content_public.common.ResourceRequestBody;
 import org.chromium.ui.base.WindowAndroid;
 import org.chromium.url.GURL;
 import org.chromium.url.Origin;
+import org.chromium.ui.base.PageTransition;
 
 /**
  * Bridges between the C++ and Java {@link TabModel} interfaces.
@@ -133,7 +134,7 @@ public abstract class TabModelJniBridge implements TabModel {
      */
     @CalledByNative
     protected abstract boolean createTabWithWebContents(
-            Tab parent, Profile profile, WebContents webContents);
+            Tab parent, Profile profile, WebContents webContents, @TabLaunchType int type);
 
     @CalledByNative
     protected abstract void openNewTab(Tab parent, GURL url, @Nullable Origin initiatorOrigin,
@@ -147,7 +148,14 @@ public abstract class TabModelJniBridge implements TabModel {
     @CalledByNative
     protected Tab createNewTabForDevTools(GURL url) {
         return getTabCreator(/*incognito=*/false)
-                .createNewTab(new LoadUrlParams(url), TabLaunchType.FROM_CHROME_UI, null);
+                .createNewTab(new LoadUrlParams(url), TabLaunchType.FROM_CHROME_UI, getTabAt(index()));
+    }
+
+    @CalledByNative
+    protected Tab createNewTabForExtensions(GURL url) {
+        final LoadUrlParams param = new LoadUrlParams(url, PageTransition.AUTO_TOPLEVEL);
+        return getTabCreator(/*incognito=*/false)
+                .createNewTab(param, TabLaunchType.FROM_LINK, getTabAt(index()));
     }
 
     /** Returns whether supplied Tab instance has been grouped together with other Tabs. */
@@ -181,6 +189,10 @@ public abstract class TabModelJniBridge implements TabModel {
     @CalledByNative
     public abstract int index();
 
+    @Override
+    @CalledByNative
+    public abstract int getLastNonExtensionActiveIndex();
+
     /** @return Whether or not a sync session is currently being restored. */
     @CalledByNative
     protected abstract boolean isSessionRestoreInProgress();
