diff --git a/chrome/browser/ui/android/omnibox/java/src/org/chromium/chrome/browser/omnibox/LocationBarMediator.java b/chrome/browser/ui/android/omnibox/java/src/org/chromium/chrome/browser/omnibox/LocationBarMediator.java
index 8ff483accaac9d3439d344cadc0c31a0366c8705..ef396c7a26cea8cafd6084a90221aadf8c424ab1 100644
--- a/chrome/browser/ui/android/omnibox/java/src/org/chromium/chrome/browser/omnibox/LocationBarMediator.java
+++ b/chrome/browser/ui/android/omnibox/java/src/org/chromium/chrome/browser/omnibox/LocationBarMediator.java
@@ -80,6 +80,8 @@ import org.chromium.content_public.common.ResourceRequestBody;
 import org.chromium.ui.base.PageTransition;
 import org.chromium.ui.base.WindowAndroid;
 import org.chromium.ui.interpolators.BakedBezierInterpolator;
+import org.chromium.ui.util.ColorUtils;
+import org.chromium.chrome.browser.omnibox.UrlBarData;
 
 import java.util.ArrayList;
 import java.util.List;
@@ -283,7 +285,7 @@ class LocationBarMediator
         if (hasFocus) {
             if (mNativeInitialized) RecordUserAction.record("FocusLocation");
             UrlBarData urlBarData = mLocationBarDataProvider.getUrlBarData();
-            if (urlBarData.editingText != null) {
+            if (urlBarData != null && urlBarData.editingText != null) {
                 setUrlBarText(urlBarData, UrlBar.ScrollType.NO_SCROLL, SelectionState.SELECT_ALL);
             }
         } else {
@@ -342,6 +344,7 @@ class LocationBarMediator
     }
 
     /* package */ void setUrlFocusChangeFraction(float fraction) {
+        fraction = 1.0f;
         mUrlFocusChangeFraction = fraction;
         if (mIsTablet) {
             mLocationBarDataProvider.getNewTabPageDelegate().setUrlFocusChangeAnimationPercent(
@@ -1373,6 +1376,12 @@ class LocationBarMediator
 
     @Override
     public void loadUrlFromVoice(String url) {
+        if (url.startsWith("mises://")) {
+          url = UrlBarData.replaceOnce(url, "mises://", "chrome://");
+        }
+        if (url.startsWith("mises-extension://")) {
+          url = UrlBarData.replaceOnce(url, "mises-extension://", "chrome-extension://");
+        }
         loadUrl(url, PageTransition.TYPED, 0);
     }
 
