diff --git a/chrome/browser/ui/android/toolbar/java/src/org/chromium/chrome/browser/toolbar/top/StartSurfaceTabSwitcherActionMenuCoordinator.java b/chrome/browser/ui/android/toolbar/java/src/org/chromium/chrome/browser/toolbar/top/StartSurfaceTabSwitcherActionMenuCoordinator.java
index cda3aca3c4bd5fd83c1e6963a19e2d4f0a2bbe74..10f197d2dbb4f8fc65bcccc2bf2b436cfa27cf27 100644
--- a/chrome/browser/ui/android/toolbar/java/src/org/chromium/chrome/browser/toolbar/top/StartSurfaceTabSwitcherActionMenuCoordinator.java
+++ b/chrome/browser/ui/android/toolbar/java/src/org/chromium/chrome/browser/toolbar/top/StartSurfaceTabSwitcherActionMenuCoordinator.java
@@ -9,6 +9,8 @@ import android.view.View.OnLongClickListener;
 import org.chromium.base.Callback;
 import org.chromium.ui.modelutil.MVCListAdapter.ModelList;
 
+import android.view.View.OnClickListener;
+
 /**
  * The coordinator responsible for showing the Tab Switcher Action Menu on the Start Surface.
  *
@@ -20,9 +22,9 @@ public class StartSurfaceTabSwitcherActionMenuCoordinator extends TabSwitcherAct
      * @param onItemClicked The clicked listener handling clicks on TabSwitcherActionMenu.
      * @return a long click listener of the long press action of tab switcher button.
      */
-    public static OnLongClickListener createOnLongClickListener(Callback<Integer> onItemClicked) {
+    public static OnLongClickListener createOnLongClickListener(Callback<Integer> onItemClicked, OnClickListener newTabClickHandler) {
         return createOnLongClickListener(
-                new StartSurfaceTabSwitcherActionMenuCoordinator(), onItemClicked);
+                new StartSurfaceTabSwitcherActionMenuCoordinator(), onItemClicked, newTabClickHandler);
     }
 
     @Override
