diff --git a/chrome/renderer/BUILD.gn b/chrome/renderer/BUILD.gn
index 53733390bf73dee58d08a58d049c21ac5af0448e..33c8b0a9162a730dcde5bd63b5e62cbacff8a157 100644
--- a/chrome/renderer/BUILD.gn
+++ b/chrome/renderer/BUILD.gn
@@ -396,7 +396,7 @@ static_library("renderer") {
     ]
   }
 
-  if (!is_android) {
+  if (!is_android || is_android) {
     sources += [
       "accessibility/ax_tree_distiller.cc",
       "accessibility/ax_tree_distiller.h",
@@ -411,6 +411,12 @@ static_library("renderer") {
       "searchbox/searchbox_extension.cc",
       "searchbox/searchbox_extension.h",
     ]
+    if (is_android) {
+      sources -= [
+        "media/chrome_speech_recognition_client.cc",
+        "media/chrome_speech_recognition_client.h",
+      ]
+    }
 
     deps += [
       "//chrome/common/accessibility:mojo_bindings",
