diff --git a/chrome/android/java/src/org/chromium/chrome/browser/sync/settings/GoogleServicesSettings.java b/chrome/android/java/src/org/chromium/chrome/browser/sync/settings/GoogleServicesSettings.java
index ec7194176b91b87277f83674f11ff1a937ecd490..c419f83cfbf1d53de4394b50db799acfaf77da2d 100644
--- a/chrome/android/java/src/org/chromium/chrome/browser/sync/settings/GoogleServicesSettings.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/sync/settings/GoogleServicesSettings.java
@@ -127,6 +127,7 @@ public class GoogleServicesSettings extends ChromeBaseSettingsFragment
             mAllowSignin.setOnPreferenceChangeListener(this);
             mAllowSignin.setManagedPreferenceDelegate(mManagedPreferenceDelegate);
         }
+        mAllowSignin.setEnabled(false);
 
         mPasswordsAccountStorage =
                 (ChromeSwitchPreference) findPreference(PREF_PASSWORDS_ACCOUNT_STORAGE);
