diff --git a/chrome/browser/resources/hats/BUILD.gn b/chrome/browser/resources/hats/BUILD.gn
index e056c659f30027d6b1dfd55b6a50e6fe9c40fe54..3d972f586d26106eef43df3170282d27f1aba348 100644
--- a/chrome/browser/resources/hats/BUILD.gn
+++ b/chrome/browser/resources/hats/BUILD.gn
@@ -4,7 +4,7 @@
 
 import("//ui/webui/resources/tools/build_webui.gni")
 
-assert(!is_android)
+#assert(!is_android)
 
 build_webui("build") {
   grd_prefix = "hats"
