diff --git a/chrome/android/java/src/org/chromium/chrome/browser/bookmarks/BookmarkDelegate.java b/chrome/android/java/src/org/chromium/chrome/browser/bookmarks/BookmarkDelegate.java
index e9af53ce50300f9d34ddccaf226c73a3aa59f360..f1c4530056018c5d85870ea5d565d6bd9efaeccb 100644
--- a/chrome/android/java/src/org/chromium/chrome/browser/bookmarks/BookmarkDelegate.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/bookmarks/BookmarkDelegate.java
@@ -80,6 +80,9 @@ public interface BookmarkDelegate {
      */
     void closeSearchUI();
 
+    void importBookmarks();
+    void exportBookmarks();
+
     /**
      * Add an observer to bookmark UI changes.
      */
