diff --git a/chrome/browser/web_applications/BUILD.gn b/chrome/browser/web_applications/BUILD.gn
index 73bc56eed05aaf63fd36cf1e87f14c177cadd207..a1360055d1c276aac3f33b21c09e62eb062359c4 100644
--- a/chrome/browser/web_applications/BUILD.gn
+++ b/chrome/browser/web_applications/BUILD.gn
@@ -290,6 +290,8 @@ source_set("web_applications") {
     ]
   }
 
+  import("//mises/browser/web_applications/sources.gni") sources += mises_browser_web_applications_sources
+
   if (is_mac) {
     sources += [
       "app_shim_registry_mac.cc",
@@ -844,6 +846,9 @@ source_set("web_applications_browser_tests") {
       "//chromeos/startup",
     ]
   }
+  if (is_android) {
+    deps -= ["//chrome/test:test_support_ui"]
+  }
 }
 
 group("browser_tests") {
@@ -853,6 +858,9 @@ group("browser_tests") {
     ":web_applications_browser_tests",
     "adjustments:adjustments_browser_tests",
   ]
+  if (is_android) {
+    deps -= ["//chrome/test:test_support_ui"]
+  }
 }
 
 # On Lacros, these browser tests require Ash to be running.
@@ -900,4 +908,7 @@ source_set("interactive_ui_tests") {
     "//ui/base/idle:test_support",
     "//ui/events/devices:test_support",
   ]
+  if (is_android) {
+    deps -= ["//chrome/test:test_support_ui"]
+  }
 }
