diff --git a/chrome/browser/resources/feedback/BUILD.gn b/chrome/browser/resources/feedback/BUILD.gn
index 8dbde1c62633ad5eaf39eeef33d3b8b2c289a8b7..8a16a220ef57c2e3939a096a2d478b7a8ec3ed3b 100644
--- a/chrome/browser/resources/feedback/BUILD.gn
+++ b/chrome/browser/resources/feedback/BUILD.gn
@@ -6,7 +6,7 @@ import("//build/config/chromeos/ui_mode.gni")
 import("//chrome/common/features.gni")
 import("//ui/webui/resources/tools/build_webui.gni")
 
-assert(!is_android)
+#assert(!is_android)
 
 build_webui("build") {
   grd_prefix = "feedback"
