diff --git a/chrome/common/net/BUILD.gn b/chrome/common/net/BUILD.gn
index c563d5d31d8fe72d3356534a72c2e3786ee9861c..5dfb07ac868dbd27abd900c2a06dbacadd959163 100644
--- a/chrome/common/net/BUILD.gn
+++ b/chrome/common/net/BUILD.gn
@@ -5,7 +5,7 @@
 import("//chrome/common/features.gni")
 import("//crypto/features.gni")
 
-assert(!is_ios, "Chromium/iOS shouldn't use anything in //chrome")
+assert(use_blink, "Chromium/iOS shouldn't use anything in //chrome")
 
 static_library("net") {
   sources = [
