diff --git a/chrome/android/chrome_public_apk_tmpl.gni b/chrome/android/chrome_public_apk_tmpl.gni
index 35cc9620e875d8481db03650226caf81316926bf..794f4e7543c9bdf850627c1d9b86a027b67bc857 100644
--- a/chrome/android/chrome_public_apk_tmpl.gni
+++ b/chrome/android/chrome_public_apk_tmpl.gni
@@ -142,6 +142,17 @@ template("chrome_public_common_apk_or_module_tmpl") {
       strip_unused_resources = is_official_build
     }
 
+    if (_is_bundle) {
+        # Resources config for blocklisting resource names from obfuscation
+        resources_config_paths = [
+          "//android_webview/aapt2.config",
+          "//chrome/android/aapt2.config",
+        ]
+        if (defined(invoker.resources_config_paths)) {
+          resources_config_paths += invoker.resources_config_paths
+        }
+    }
+
     if (!defined(aapt_locale_allowlist)) {
       # Include resource strings files only for supported locales.
       aapt_locale_allowlist = platform_pak_locales
@@ -540,7 +551,7 @@ template("monochrome_public_common_apk_or_module_tmpl") {
       }
 
       # We only optimize resources in bundles.
-      if (_is_bundle_module) {
+      if (_is_bundle_module && false) {
         # Resources config for blocklisting resource names from obfuscation
         resources_config_paths = [
           "//android_webview/aapt2.config",
