diff --git a/chrome/browser/signin/primary_account_policy_manager.cc b/chrome/browser/signin/primary_account_policy_manager.cc
index 55da6e622e52c84a9c96cf513ac25822d55bbfe2..f20ace01d9d5e76c676e76421696ae5b7fcc1218 100644
--- a/chrome/browser/signin/primary_account_policy_manager.cc
+++ b/chrome/browser/signin/primary_account_policy_manager.cc
@@ -303,7 +303,7 @@ void PrimaryAccountPolicyManager::OnUserConfirmedProfileDeletion(
   delete_profile_dialog_manager_.reset();
 
   DCHECK(profiles::IsMultipleProfilesEnabled());
-
+#if !BUILDFLAG(IS_ANDROID)
   g_browser_process->profile_manager()
       ->GetDeleteProfileHelper()
       .MaybeScheduleProfileForDeletion(
@@ -312,5 +312,6 @@ void PrimaryAccountPolicyManager::OnUserConfirmedProfileDeletion(
               ? base::DoNothing()
               : base::BindOnce(&webui::OpenNewWindowForProfile),
           ProfileMetrics::DELETE_PROFILE_PRIMARY_ACCOUNT_NOT_ALLOWED);
+#endif
 }
 #endif  // defined(TOOLKIT_VIEWS) && !BUILDFLAG(IS_CHROMEOS)
