diff --git a/chrome/browser/safe_browsing/BUILD.gn b/chrome/browser/safe_browsing/BUILD.gn
index fd47cd395ad8e531da57dc4ca0666162f5b28caa..0673ca1cff24e88f4393359a603fb559b4dcb5b3 100644
--- a/chrome/browser/safe_browsing/BUILD.gn
+++ b/chrome/browser/safe_browsing/BUILD.gn
@@ -457,6 +457,16 @@ static_library("safe_browsing") {
           "tailored_security/unconsented_message_android.h",
           "telemetry/android/android_telemetry_service.cc",
           "telemetry/android/android_telemetry_service.h",
+          "cloud_content_scanning/deep_scanning_utils.cc",
+          "cloud_content_scanning/deep_scanning_utils.h",
+          "extension_telemetry/declarative_net_request_signal.cc",
+          "extension_telemetry/declarative_net_request_signal.h",
+          "extension_telemetry/tabs_api_signal.cc",
+          "extension_telemetry/tabs_api_signal.h",
+          "extension_telemetry/extension_web_request_reporter_impl.cc",
+          "extension_telemetry/extension_web_request_reporter_impl.h",
+          "extension_telemetry/extension_signal_util.h",
+          "extension_telemetry/extension_signal_util.cc",
         ]
         deps += [
           "//chrome/app/vector_icons:vector_icons",
