diff --git a/chrome/android/features/tab_ui/java/src/org/chromium/chrome/browser/tasks/tab_management/TabGridDialogMediator.java b/chrome/android/features/tab_ui/java/src/org/chromium/chrome/browser/tasks/tab_management/TabGridDialogMediator.java
index 19dd96359b6eae139eec7a94fbf49e542d799da4..37f6b0023095a120300bc2a3797520e8c02962b1 100644
--- a/chrome/android/features/tab_ui/java/src/org/chromium/chrome/browser/tasks/tab_management/TabGridDialogMediator.java
+++ b/chrome/android/features/tab_ui/java/src/org/chromium/chrome/browser/tasks/tab_management/TabGridDialogMediator.java
@@ -56,6 +56,8 @@ import java.util.ArrayList;
 import java.util.List;
 import java.util.Locale;
 
+import org.chromium.chrome.browser.homepage.HomepageManager;
+
 /**
  * A mediator for the TabGridDialog component, responsible for communicating
  * with the components' coordinator as well as managing the business logic
@@ -629,8 +631,12 @@ public class TabGridDialogMediator
             Tab currentTab = mTabModelSelector.getTabById(mCurrentTabId);
             hideDialog(false);
             if (currentTab == null) {
-                mTabCreatorManager.getTabCreator(mTabModelSelector.isIncognitoSelected())
-                        .launchNTP();
+                if (mTabModelSelector.isIncognitoSelected())
+                    mTabCreatorManager.getTabCreator(mTabModelSelector.isIncognitoSelected())
+                            .launchNTP();
+                else
+                    mTabCreatorManager.getTabCreator(mTabModelSelector.isIncognitoSelected())
+                            .launchUrl(HomepageManager.getInstance().getHomepageUriIgnoringEnabledState(), TabLaunchType.FROM_CHROME_UI);
                 return;
             }
             List<Tab> relatedTabs = getRelatedTabs(currentTab.getId());
