diff --git a/chrome/browser/ui/autofill_assistant/password_change/apc_utils.cc b/chrome/browser/ui/autofill_assistant/password_change/apc_utils.cc
index 029495f7d937cccddd2d30b0b333a4e315dd4621..88487b4193e11811ce2a3ec822201683c2701ace 100644
--- a/chrome/browser/ui/autofill_assistant/password_change/apc_utils.cc
+++ b/chrome/browser/ui/autofill_assistant/password_change/apc_utils.cc
@@ -8,13 +8,14 @@
 #include "chrome/app/vector_icons/vector_icons.h"
 #include "chrome/browser/autofill_assistant/password_change/vector_icons/vector_icons.h"
 #include "components/autofill_assistant/browser/public/password_change/proto/actions.pb.h"
+#include "components/vector_icons/vector_icons.h"
 
 const gfx::VectorIcon& GetAssistantIconOrFallback() {
 #if BUILDFLAG(GOOGLE_CHROME_BRANDING)
   return kAssistantIcon;
 #else
   // Only developer builds will ever use this branch.
-  return kProductIcon;
+  return vector_icons::kProductIcon;
 #endif
 }
 
