diff --git a/chrome/browser/resources/settings/BUILD.gn b/chrome/browser/resources/settings/BUILD.gn
index b6d81b8457a70ea46f38af743bffa855622cc261..d90cd8ac5ac5f5df54041c2d7479244aea83c79f 100644
--- a/chrome/browser/resources/settings/BUILD.gn
+++ b/chrome/browser/resources/settings/BUILD.gn
@@ -17,7 +17,7 @@ import("//ui/webui/webui_features.gni")
 import("../tools/optimize_webui.gni")
 import("settings.gni")
 
-assert(!is_android, "Android does not use these settings")
+#assert(!is_android, "Android does not use these settings")
 
 preprocess_folder = "preprocessed"
 
