diff --git a/chrome/browser/resources/media_router/internals/BUILD.gn b/chrome/browser/resources/media_router/internals/BUILD.gn
index 80bdb1b139c352eb556dab0f2a2dfbc44fbad4bc..da3fe6d236274d33b04daf70dc750d15737386c2 100644
--- a/chrome/browser/resources/media_router/internals/BUILD.gn
+++ b/chrome/browser/resources/media_router/internals/BUILD.gn
@@ -4,7 +4,7 @@
 
 import("//ui/webui/resources/tools/build_webui.gni")
 
-assert(!is_android)
+# assert(!is_android)
 
 build_webui("build") {
   grd_prefix = "media_router_internals"
