diff --git a/chrome/browser/ui/android/toolbar/java/src/org/chromium/chrome/browser/toolbar/menu_button/MenuButtonCoordinator.java b/chrome/browser/ui/android/toolbar/java/src/org/chromium/chrome/browser/toolbar/menu_button/MenuButtonCoordinator.java
index 1aae770439750337618ac014f2ea91bedf2cc48d..9615246ce42b9a0167f151a65715312c0c3b6568 100644
--- a/chrome/browser/ui/android/toolbar/java/src/org/chromium/chrome/browser/toolbar/menu_button/MenuButtonCoordinator.java
+++ b/chrome/browser/ui/android/toolbar/java/src/org/chromium/chrome/browser/toolbar/menu_button/MenuButtonCoordinator.java
@@ -258,4 +258,7 @@ public class MenuButtonCoordinator {
     public boolean isShowingUpdateBadge() {
         return mPropertyModel.get(MenuButtonProperties.SHOW_UPDATE_BADGE).mShowUpdateBadge;
     }
+    public AppMenuButtonHelper getHelper() {
+	return mPropertyModel.get(MenuButtonProperties.APP_MENU_BUTTON_HELPER);
+    }
 }
