diff --git a/chrome/browser/ui/android/toolbar/java/src/org/chromium/chrome/browser/toolbar/adaptive/OptionalNewTabButtonController.java b/chrome/browser/ui/android/toolbar/java/src/org/chromium/chrome/browser/toolbar/adaptive/OptionalNewTabButtonController.java
index 678139258d760234021d6f27a1f71d89bb7a1af0..4eb0082f0f45056c1a5ee8b1c0eebd21a1574acd 100644
--- a/chrome/browser/ui/android/toolbar/java/src/org/chromium/chrome/browser/toolbar/adaptive/OptionalNewTabButtonController.java
+++ b/chrome/browser/ui/android/toolbar/java/src/org/chromium/chrome/browser/toolbar/adaptive/OptionalNewTabButtonController.java
@@ -122,6 +122,9 @@ public class OptionalNewTabButtonController
 
         boolean isIncognito = activeTabSupplier.get().isIncognito();
         RecordUserAction.record("MobileTopToolbarOptionalButtonNewTab");
+        if (isIncognito)
+        tabCreatorManager.getTabCreator(isIncognito).launchIncognitoNTP();
+        else
         tabCreatorManager.getTabCreator(isIncognito).launchNTP();
 
         if (mTrackerSupplier.hasValue()) {
