diff --git a/chrome/browser/ui/views/side_panel/BUILD.gn b/chrome/browser/ui/views/side_panel/BUILD.gn
index b25954bac4b1fc404d91a0ca486cede3b87c082d..88ab75ae43dfc87ede9017bbdc163609b0575cab 100644
--- a/chrome/browser/ui/views/side_panel/BUILD.gn
+++ b/chrome/browser/ui/views/side_panel/BUILD.gn
@@ -2,7 +2,7 @@
 # Use of this source code is governed by a BSD - style license that can be
 # found in the LICENSE file.
 
-assert(is_win || is_mac || is_linux || is_chromeos)
+#assert(is_win || is_mac || is_linux || is_chromeos)
 
 import("//components/lens/features.gni")
 import("//extensions/buildflags/buildflags.gni")
