diff --git a/chrome/browser/ui/webui/tab_search/BUILD.gn b/chrome/browser/ui/webui/tab_search/BUILD.gn
index 5d39383f4d4af4c5e31261f16e3685060ffc8c5c..391d5d68d79800f974a4e8fc3f2c7bf8796afd16 100644
--- a/chrome/browser/ui/webui/tab_search/BUILD.gn
+++ b/chrome/browser/ui/webui/tab_search/BUILD.gn
@@ -4,7 +4,7 @@
 
 import("//mojo/public/tools/bindings/mojom.gni")
 
-assert(is_linux || is_chromeos || is_win || is_mac || is_fuchsia)
+#assert(is_linux || is_chromeos || is_win || is_mac || is_fuchsia)
 
 mojom("mojo_bindings") {
   sources = [ "tab_search.mojom" ]
