diff --git a/chrome/browser/resources/extensions/manager.ts b/chrome/browser/resources/extensions/manager.ts
index 81fe22a532ca53b3c344ac73659d8da6350afe75..fd4fdede8366d82caa2e0a11cc491bddb0f7b385 100644
--- a/chrome/browser/resources/extensions/manager.ts
+++ b/chrome/browser/resources/extensions/manager.ts
@@ -27,6 +27,8 @@ import './toolbar.js';
 import './kiosk_dialog.js';
 
 // </if>
+import './mises_open_web_store_footer.js'
+import './mises_default_extension_setting.js';
 
 import {CrViewManagerElement} from 'chrome://resources/cr_elements/cr_view_manager/cr_view_manager.js';
 import {assert, assertNotReached} from 'chrome://resources/js/assert_ts.js';
