diff --git a/chrome/android/java/src/org/chromium/chrome/browser/tabmodel/PendingTabClosureManager.java b/chrome/android/java/src/org/chromium/chrome/browser/tabmodel/PendingTabClosureManager.java
index a041eee6574b4fcef93cb96d8338afaeb3658d67..5aa0f54347691191065e6462371844a4c64989ff 100644
--- a/chrome/android/java/src/org/chromium/chrome/browser/tabmodel/PendingTabClosureManager.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/tabmodel/PendingTabClosureManager.java
@@ -8,6 +8,7 @@ import androidx.annotation.NonNull;
 
 import org.chromium.base.ThreadUtils.ThreadChecker;
 import org.chromium.chrome.browser.tab.Tab;
+import org.chromium.chrome.browser.tab.state.CriticalPersistedTabData;
 
 import java.util.ArrayList;
 import java.util.HashSet;
@@ -155,6 +156,25 @@ public class PendingTabClosureManager {
             return mRewoundTabs.indexOf(tab);
         }
 
+        private Tab findTabInAllTabModels(int tabId) {
+            Tab tab = TabModelUtils.getTabById(mTabList, tabId);
+            if (tab != null) return tab;
+            return TabModelUtils.getTabById(mTabList, tabId);
+        }
+
+	    @Override
+    	public int getLastNonExtensionActiveIndex() {
+            if (mTabList.getLastNonExtensionActiveIndex() != INVALID_TAB_INDEX) {
+                Tab parentTab = findTabInAllTabModels(CriticalPersistedTabData.from(TabModelUtils.getCurrentTab(this)).getParentId());
+                if (parentTab != null) {
+                   return mRewoundTabs.indexOf(parentTab);
+                }
+                return mRewoundTabs.indexOf(TabModelUtils.getCurrentTab(mTabList));
+            }
+            if (!mRewoundTabs.isEmpty()) return 0;
+            return INVALID_TAB_INDEX;
+        }
+
         /**
          * Resets this list to match the original {@link TabList}.  Note that if the
          * {@link TabList} doesn't support pending closures this model will be empty.  This should
