diff --git a/chrome/browser/ui/views/permissions/permission_prompt_bubble_base_view.cc b/chrome/browser/ui/views/permissions/permission_prompt_bubble_base_view.cc
index 7bb9017febd5ea1959b3869f267105488641b018..b3f97d178ceaa06308486e97d20b14364b16aa05 100644
--- a/chrome/browser/ui/views/permissions/permission_prompt_bubble_base_view.cc
+++ b/chrome/browser/ui/views/permissions/permission_prompt_bubble_base_view.cc
@@ -66,6 +66,7 @@ PermissionPromptBubbleBaseView::PermissionPromptBubbleBaseView(
       views::DISTANCE_BUBBLE_PREFERRED_WIDTH));
 
   SetProperty(views::kElementIdentifierKey, kMainViewId);
+  MISES_PERMISSION_PROMPT_BUBBLE_BASE_VIEW
 }
 
 PermissionPromptBubbleBaseView::~PermissionPromptBubbleBaseView() = default;
