diff --git a/chrome/browser/resources/new_tab_page/modules/photos/icons/BUILD.gn b/chrome/browser/resources/new_tab_page/modules/photos/icons/BUILD.gn
index 25dc9b94101c560be325c17db05575e66c62e59c..67bcaa099dfd1c2f5e09e0131bcc7790ee048585 100644
--- a/chrome/browser/resources/new_tab_page/modules/photos/icons/BUILD.gn
+++ b/chrome/browser/resources/new_tab_page/modules/photos/icons/BUILD.gn
@@ -4,7 +4,7 @@
 
 import("//ui/webui/resources/tools/generate_grd.gni")
 
-assert(!is_android)
+#assert(!is_android)
 
 generate_grd("build_grdp") {
   grd_prefix = "photos"
