diff --git a/chrome/browser/resources/side_panel/shared/BUILD.gn b/chrome/browser/resources/side_panel/shared/BUILD.gn
index d09f43a0a3aa79abba8f6006c1becbfebf099991..e53a1bb69664b38049a6a1fafa093692128e378a 100644
--- a/chrome/browser/resources/side_panel/shared/BUILD.gn
+++ b/chrome/browser/resources/side_panel/shared/BUILD.gn
@@ -4,7 +4,7 @@
 
 import("//ui/webui/resources/tools/build_webui.gni")
 
-assert(!is_android)
+#assert(!is_android)
 
 build_webui("build") {
   grd_prefix = "side_panel_shared"
