diff --git a/chrome/renderer/searchbox/searchbox.cc b/chrome/renderer/searchbox/searchbox.cc
index eb914dbc0179925ebaec99373e59cb90c03d6010..fec6b99ea690ba7d38a3700d528235701addbbdc 100644
--- a/chrome/renderer/searchbox/searchbox.cc
+++ b/chrome/renderer/searchbox/searchbox.cc
@@ -34,7 +34,7 @@
 namespace {
 
 // The size of the InstantMostVisitedItem cache.
-const size_t kMaxInstantMostVisitedItemCacheSize = 100;
+const size_t kMaxInstantMostVisitedItemCacheSize = 500;
 
 // Returns true if items stored in |old_item_id_pairs| and |new_items| are
 // equal.
