diff --git a/chrome/browser/tab_group/java/src/org/chromium/chrome/browser/tasks/tab_groups/TabGroupModelFilter.java b/chrome/browser/tab_group/java/src/org/chromium/chrome/browser/tasks/tab_groups/TabGroupModelFilter.java
index 2d0dd5a2a26506ddd837a52a4195a484dce046d5..4384d30998bbe9c42f15d6e20ebd290e8dd6107d 100644
--- a/chrome/browser/tab_group/java/src/org/chromium/chrome/browser/tasks/tab_groups/TabGroupModelFilter.java
+++ b/chrome/browser/tab_group/java/src/org/chromium/chrome/browser/tasks/tab_groups/TabGroupModelFilter.java
@@ -1032,6 +1032,11 @@ public class TabGroupModelFilter extends TabModelFilter {
         return mAbsentSelectedTab == null;
     }
 
+    @Override
+    public int getLastNonExtensionActiveIndex() {
+        return INVALID_TAB_INDEX;
+    }
+
     @Override
     @VisibleForTesting(otherwise = VisibleForTesting.PACKAGE_PRIVATE)
     public void markTabStateInitialized() {
