diff --git a/chrome/browser/safe_browsing/BUILD.gn b/chrome/browser/safe_browsing/BUILD.gn
index da80c70f42a61a32c5e9b3209eaba80ec433dd1e..78ff3fa2af982eee64ce7acde510cb979f79a6de 100644
--- a/chrome/browser/safe_browsing/BUILD.gn
+++ b/chrome/browser/safe_browsing/BUILD.gn
@@ -425,6 +425,8 @@ static_library("safe_browsing") {
           "tailored_security/unconsented_message_android.h",
           "telemetry/android/android_telemetry_service.cc",
           "telemetry/android/android_telemetry_service.h",
+          "cloud_content_scanning/deep_scanning_utils.cc",
+          "cloud_content_scanning/deep_scanning_utils.h",
         ]
         deps += [
           "//chrome/app/vector_icons:vector_icons",
