diff --git a/chrome/android/java/src/org/chromium/chrome/browser/sync/settings/GoogleServicesSettings.java b/chrome/android/java/src/org/chromium/chrome/browser/sync/settings/GoogleServicesSettings.java
index 92fc963fec6c4cf3b7d77952b3f1aa0fa896a8da..e9c23f0272832cce38f08b7bc89167943b28142b 100644
--- a/chrome/android/java/src/org/chromium/chrome/browser/sync/settings/GoogleServicesSettings.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/sync/settings/GoogleServicesSettings.java
@@ -123,6 +123,7 @@ public class GoogleServicesSettings extends ChromeBaseSettingsFragment
             mAllowSignin.setOnPreferenceChangeListener(this);
             mAllowSignin.setManagedPreferenceDelegate(mManagedPreferenceDelegate);
         }
+        mAllowSignin.setEnabled(false);
 
         mPasswordsAccountStorage =
                 (ChromeSwitchPreference) findPreference(PREF_PASSWORDS_ACCOUNT_STORAGE);
