diff --git a/chrome/browser/chrome_content_browser_client_receiver_bindings.cc b/chrome/browser/chrome_content_browser_client_receiver_bindings.cc
index 3ad28f1eee437bd4447c49cd93444fd4842d8630..5e34dee825459dc19b5c3a3095fa845ccb7db712 100644
--- a/chrome/browser/chrome_content_browser_client_receiver_bindings.cc
+++ b/chrome/browser/chrome_content_browser_client_receiver_bindings.cc
@@ -169,7 +169,7 @@ void MaybeCreateSafeBrowsingForRenderer(
       std::move(receiver));
 }
 
-#if BUILDFLAG(ENABLE_EXTENSIONS)
+#if BUILDFLAG(ENABLE_EXTENSIONS) && BUILDFLAG(FULL_SAFE_BROWSING)
 void MaybeCreateExtensionWebRequestReporterForRenderer(
     int process_id,
     mojo::PendingReceiver<safe_browsing::mojom::ExtensionWebRequestReporter>
@@ -238,7 +238,7 @@ void ChromeContentBrowserClient::ExposeInterfacesToRenderer(
                 &ChromeContentBrowserClient::GetSafeBrowsingUrlCheckerDelegate,
                 base::Unretained(this))),
         ui_task_runner);
-#if BUILDFLAG(ENABLE_EXTENSIONS)
+#if BUILDFLAG(ENABLE_EXTENSIONS) && BUILDFLAG(FULL_SAFE_BROWSING)
     registry->AddInterface<safe_browsing::mojom::ExtensionWebRequestReporter>(
         base::BindRepeating(&MaybeCreateExtensionWebRequestReporterForRenderer,
                             render_process_host->GetID()),
