diff --git a/chrome/android/java/src/org/chromium/chrome/browser/bookmarks/BookmarkToolbarMediator.java b/chrome/android/java/src/org/chromium/chrome/browser/bookmarks/BookmarkToolbarMediator.java
index a80cc1761d9da2a7fc757ce496bbcd2ca90fd9d5..ae5d38b25c6be893d42237335fe4540b4f59a87d 100644
--- a/chrome/android/java/src/org/chromium/chrome/browser/bookmarks/BookmarkToolbarMediator.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/bookmarks/BookmarkToolbarMediator.java
@@ -188,7 +188,13 @@ class BookmarkToolbarMediator
         } else if (id == R.id.close_menu_id) {
             BookmarkUtils.finishActivityOnPhone(mContext);
             return true;
-        } else if (id == R.id.selection_mode_edit_menu_id) {
+        } else if (id == R.id.import_bookmarks) {
+            mBookmarkDelegate.importBookmarks();
+            return true;
+        } else if (id == R.id.export_bookmarks) {
+            mBookmarkDelegate.exportBookmarks();
+            return true;
+        }  else if (id == R.id.selection_mode_edit_menu_id) {
             List<BookmarkId> list = mSelectionDelegate.getSelectedItemsAsList();
             assert list.size() == 1;
             BookmarkItem item = mBookmarkModel.getBookmarkById(list.get(0));
