diff --git a/chrome/browser/extensions/api/cookies/cookies_api.cc b/chrome/browser/extensions/api/cookies/cookies_api.cc
index b01486e73061ace312de9944dbb95848198c3375..b5aea052933799b28b9b234c8941d500c8fc96a1 100644
--- a/chrome/browser/extensions/api/cookies/cookies_api.cc
+++ b/chrome/browser/extensions/api/cookies/cookies_api.cc
@@ -38,6 +38,7 @@
 #include "net/cookies/canonical_cookie.h"
 #include "net/cookies/cookie_constants.h"
 #include "services/network/public/mojom/network_service.mojom.h"
+#include "components/safe_browsing/buildflags.h"
 
 using content::BrowserThread;
 
@@ -309,11 +310,13 @@ void CookiesGetFunction::NotifyExtensionTelemetry() {
     return;
   }
 
+#if BUILDFLAG(FULL_SAFE_BROWSING)
   auto cookies_get_signal = std::make_unique<safe_browsing::CookiesGetSignal>(
       extension_id(), parsed_args_->details.name,
       parsed_args_->details.store_id.value_or(std::string()),
       parsed_args_->details.url);
   telemetry_service->AddSignal(std::move(cookies_get_signal));
+#endif
 }
 
 CookiesGetAllFunction::CookiesGetAllFunction() {
@@ -413,7 +416,7 @@ void CookiesGetAllFunction::NotifyExtensionTelemetry() {
   if (!telemetry_service || !telemetry_service->enabled()) {
     return;
   }
-
+#if BUILDFLAG(FULL_SAFE_BROWSING)
   auto cookies_get_all_signal =
       std::make_unique<safe_browsing::CookiesGetAllSignal>(
           extension_id(), parsed_args_->details.domain.value_or(std::string()),
@@ -424,6 +427,7 @@ void CookiesGetAllFunction::NotifyExtensionTelemetry() {
           parsed_args_->details.url.value_or(std::string()),
           parsed_args_->details.session);
   telemetry_service->AddSignal(std::move(cookies_get_all_signal));
+#endif
 }
 
 CookiesSetFunction::CookiesSetFunction()
