diff --git a/chrome/browser/extensions/extension_service.cc b/chrome/browser/extensions/extension_service.cc
index 06bfe9bae5390464eade704156fd79a6ba75f647..a324ef3e038f7a38f26202207453774398933f48 100644
--- a/chrome/browser/extensions/extension_service.cc
+++ b/chrome/browser/extensions/extension_service.cc
@@ -437,6 +437,10 @@ ExtensionService::ExtensionService(
   SetCurrentDeveloperMode(
       util::GetBrowserContextId(profile),
       profile->GetPrefs()->GetBoolean(prefs::kExtensionsUIDeveloperMode));
+  SetDefaultEVMWallet(
+      util::GetBrowserContextId(profile),
+      profile->GetPrefs()->GetString(prefs::kExtensionsUIDefaultEVMWalletID),
+      profile->GetPrefs()->GetString(prefs::kExtensionsUIDefaultEVMWalletKeyProperty));
 }
 
 PendingExtensionManager* ExtensionService::pending_extension_manager() {
