diff --git a/chrome/common/accessibility/BUILD.gn b/chrome/common/accessibility/BUILD.gn
index 5dd333ac1814be33bc5f5eb2fa817ac6a2a663e4..abea5ab2b54b82ccfa62b888c637929ca2679d56 100644
--- a/chrome/common/accessibility/BUILD.gn
+++ b/chrome/common/accessibility/BUILD.gn
@@ -4,7 +4,7 @@
 
 import("//mojo/public/tools/bindings/mojom.gni")
 
-assert(!is_android)
+#assert(!is_android)
 
 mojom("mojo_bindings") {
   sources = [ "read_anything.mojom" ]
