diff --git a/chrome/browser/BUILD.gn b/chrome/browser/BUILD.gn
index 59773d17657448a508a8cea31ca3a1f72f3c5abc..6a39dca5239a518c7c38a42b22f35afed8fce9b3 100644
--- a/chrome/browser/BUILD.gn
+++ b/chrome/browser/BUILD.gn
@@ -2513,6 +2513,7 @@ static_library("browser") {
       "signin/wait_for_network_callback_helper_chrome.h",
     ]
   }
+  import("//mises/browser/sources.gni") sources += mises_chrome_browser_sources deps += mises_chrome_browser_deps public_deps += mises_chrome_browser_public_deps
 
   if (is_android) {
     sources += [
@@ -3457,7 +3458,8 @@ static_library("browser") {
       ]
       deps += [ "//chrome/android/modules/dev_ui/provider:native" ]
     }
-  } else {
+  }
+  if (true) {
     #!is_android
     sources += [
       "accessibility/ax_main_node_annotator_controller.cc",
@@ -4415,7 +4417,7 @@ static_library("browser") {
         ]
       }
     }
-    if (!is_chromeos_ash) {
+    if (is_android || !is_chromeos_ash) {
       sources += [
         "accessibility/soda_installer_impl.cc",
         "accessibility/soda_installer_impl.h",
@@ -4474,7 +4476,7 @@ static_library("browser") {
       ]
     }
 
-    if (!is_win) {
+    if (!is_android && !is_win) {
       # On Windows, the hashes are embedded in //chrome:chrome_initial rather
       # than here in :chrome_dll.
       deps += [ "//chrome:packed_resources_integrity_header" ]
@@ -4499,6 +4501,9 @@ static_library("browser") {
         "component_updater/hyphenation_component_installer.h",
       ]
     }
+    sources -= mises_chrome_browser_exception_sources
+    deps -= mises_chrome_browser_exception_deps
+    allow_circular_includes_from -= mises_chrome_browser_exception_allow_circular_includes_from
   }
 
   if (is_chromeos_ash) {
@@ -7958,7 +7963,7 @@ static_library("browser") {
     }
   }
 
-  if (enable_session_service) {
+  if (true || enable_session_service) {
     sources += [
       "sessions/app_session_service.cc",
       "sessions/app_session_service.h",
@@ -8163,7 +8168,7 @@ static_library("browser") {
       "navigation_predictor/preloading_model_handler.cc",
       "navigation_predictor/preloading_model_handler.h",
     ]
-    if (!is_android) {
+    if (is_android || !is_android) {
       sources += [
         "new_tab_page/modules/history_clusters/ranking/history_clusters_module_ranking_model_executor.cc",
         "new_tab_page/modules/history_clusters/ranking/history_clusters_module_ranking_model_executor.h",
@@ -8449,7 +8454,8 @@ static_library("browser_generated_files") {
       "//chrome/browser/android/webapk/proto",
       "//chrome/browser/ui/webui/feed_internals:mojo_bindings",
     ]
-  } else {
+  }# else {
+  if (true) {
     public_deps += [
       "//chrome/browser/cart:mojo_bindings",
       "//chrome/browser/companion/core/mojom:mojo_bindings",
