diff --git a/chrome/browser/extensions/api/chrome_extensions_api_client.cc b/chrome/browser/extensions/api/chrome_extensions_api_client.cc
index 4007e26f780c3dc8564c3f9e152e7ead160b8974..d4992c57de03f3710771985a13087ea2da1c8f89 100644
--- a/chrome/browser/extensions/api/chrome_extensions_api_client.cc
+++ b/chrome/browser/extensions/api/chrome_extensions_api_client.cc
@@ -502,7 +502,9 @@ std::vector<KeyedServiceBaseFactory*>
 ChromeExtensionsAPIClient::GetFactoryDependencies() {
   // clang-format off
   return {
+#if !BUILDFLAG(IS_ANDROID)
       InstantServiceFactory::GetInstance(),
+#endif
       SupervisedUserServiceFactory::GetInstance(),
   };
   // clang-format on
