diff --git a/chrome/android/features/tab_ui/java/src/org/chromium/chrome/browser/tasks/tab_management/TabSwitcherMediator.java b/chrome/android/features/tab_ui/java/src/org/chromium/chrome/browser/tasks/tab_management/TabSwitcherMediator.java
index df98be4d6d26200815cf5612c85ad6c2020ead8f..317ff056efae1da2028721b908178e51e1fcd81a 100644
--- a/chrome/android/features/tab_ui/java/src/org/chromium/chrome/browser/tasks/tab_management/TabSwitcherMediator.java
+++ b/chrome/android/features/tab_ui/java/src/org/chromium/chrome/browser/tasks/tab_management/TabSwitcherMediator.java
@@ -72,6 +72,7 @@ import org.chromium.components.browser_ui.widget.gesture.BackPressHandler;
 import org.chromium.ui.base.DeviceFormFactor;
 import org.chromium.ui.modelutil.PropertyModel;
 
+import org.chromium.base.ContextUtils;
 import java.util.List;
 
 /**
@@ -299,6 +300,11 @@ class TabSwitcherMediator implements TabSwitcher.Controller, TabListRecyclerView
         mTabModelSelector = tabModelSelector;
         mBrowserControlsStateProvider = browserControlsStateProvider;
         mMultiWindowModeStateDispatcher = multiWindowModeStateDispatcher;
+        final String active_tabswitcher = ContextUtils.getAppSharedPreferences().getString("active_tabswitcher", "default");
+        if (active_tabswitcher.equals("list"))
+          mode = TabListCoordinator.TabListMode.LIST;
+        if (active_tabswitcher.equals("default") || active_tabswitcher.equals("grid"))
+          mode = TabListCoordinator.TabListMode.GRID;
         mMode = mode;
         mContainerViewModel.set(MODE, mode);
         mContext = context;
