diff --git a/chrome/browser/ui/webui/side_panel/customize_chrome/BUILD.gn b/chrome/browser/ui/webui/side_panel/customize_chrome/BUILD.gn
index 80504b429ee9c6a8dc215f28abb15e3db392e6c8..86d03b97d6e72fab44ad7edf8f00466cb49dd676 100644
--- a/chrome/browser/ui/webui/side_panel/customize_chrome/BUILD.gn
+++ b/chrome/browser/ui/webui/side_panel/customize_chrome/BUILD.gn
@@ -4,7 +4,7 @@
 
 import("//mojo/public/tools/bindings/mojom.gni")
 
-assert(!is_android)
+#assert(!is_android)
 
 mojom("mojo_bindings") {
   sources = [
