diff --git a/chrome/android/java/src/org/chromium/chrome/browser/bookmarks/BookmarkManager.java b/chrome/android/java/src/org/chromium/chrome/browser/bookmarks/BookmarkManager.java
index afb4b6eeb880a3c03aba5c58a8a0de9fcb165ac9..ded4740a24664fa101001a6377e73bcf9ec6980d 100644
--- a/chrome/android/java/src/org/chromium/chrome/browser/bookmarks/BookmarkManager.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/bookmarks/BookmarkManager.java
@@ -45,6 +45,11 @@ import org.chromium.url.GURL;
 import java.util.List;
 import java.util.Stack;
 
+import android.app.Activity;
+import org.chromium.ui.base.WindowAndroid;
+import org.chromium.ui.base.ActivityWindowAndroid;
+import org.chromium.ui.base.IntentRequestTracker;
+
 /**
  * The new bookmark manager that is planned to replace the existing bookmark manager. It holds all
  * views and shared logics between tablet and phone. For tablet/phone specific logics, see
@@ -58,6 +63,9 @@ public class BookmarkManager implements BookmarkDelegate, SearchDelegate,
 
     private static boolean sPreventLoadingForTesting;
 
+    private ActivityWindowAndroid mWindowAndroid;
+    private IntentRequestTracker mIntentTracker;
+
     private Context mContext;
     private ViewGroup mMainView;
     private BookmarkModel mBookmarkModel;
@@ -301,6 +309,22 @@ public class BookmarkManager implements BookmarkDelegate, SearchDelegate,
         if (!isDialogUi) {
             RecordUserAction.record("MobileBookmarkManagerPageOpen");
         }
+        mIntentTracker = IntentRequestTracker.createFromActivity((Activity)context);
+        mWindowAndroid = new ActivityWindowAndroid(context, true, mIntentTracker);
+    }
+
+    public IntentRequestTracker intentTracker() {
+        return mIntentTracker;
+    }
+
+    @Override
+    public void importBookmarks() {
+        mBookmarkModel.startImportBookmarks(mWindowAndroid);
+    }
+
+    @Override
+    public void exportBookmarks() {
+        mBookmarkModel.startExportBookmarks(mWindowAndroid);
     }
 
     @Override
