diff --git a/chrome/android/features/tab_ui/java/src/org/chromium/chrome/browser/tasks/tab_management/TabGroupUiMediator.java b/chrome/android/features/tab_ui/java/src/org/chromium/chrome/browser/tasks/tab_management/TabGroupUiMediator.java
index d972e414bcaf087074ea6d519750e9dd5f19c356..22f9e5531a8ddd8fc51dba405758da968cc577c2 100644
--- a/chrome/android/features/tab_ui/java/src/org/chromium/chrome/browser/tasks/tab_management/TabGroupUiMediator.java
+++ b/chrome/android/features/tab_ui/java/src/org/chromium/chrome/browser/tasks/tab_management/TabGroupUiMediator.java
@@ -48,6 +48,8 @@ import org.chromium.url.GURL;
 
 import java.util.List;
 
+import org.chromium.base.ContextUtils;
+
 /**
  * A mediator for the TabGroupUi. Responsible for managing the internal state of the component.
  */
@@ -343,7 +345,7 @@ public class TabGroupUiMediator implements BackPressHandler {
                 parentTabToAttach = relatedTabs.get(relatedTabs.size() - 1);
             }
             mTabCreatorManager.getTabCreator(currentTab.isIncognito())
-                    .createNewTab(new LoadUrlParams(UrlConstants.NTP_URL),
+                    .createNewTab(new LoadUrlParams("chrome-search://local-ntp/local-ntp.html"),
                             TabLaunchType.FROM_TAB_GROUP_UI, parentTabToAttach);
             RecordUserAction.record("MobileNewTabOpened." + TabGroupUiCoordinator.COMPONENT_NAME);
             if (!currentTab.isIncognito()) {
@@ -367,6 +369,9 @@ public class TabGroupUiMediator implements BackPressHandler {
      *            not, associated tabs from #getTabsToShowForID will be showing in the tab strip.
      */
     private void resetTabStripWithRelatedTabsForId(int id) {
+        if (ContextUtils.getAppSharedPreferences().getBoolean("enable_bottom_toolbar", false))
+            id = Tab.INVALID_TAB_ID;
+
         // TODO(crbug.com/1090655): We should be able to guard this call behind some checks so that
         // we can assert here that 1) mIsShowingOverViewMode is false 2) mIsTabGroupUiVisible with
         // valid id is false.
