diff --git a/chrome/android/features/tab_ui/java/src/org/chromium/chrome/browser/tasks/tab_management/TabGridDialogCoordinator.java b/chrome/android/features/tab_ui/java/src/org/chromium/chrome/browser/tasks/tab_management/TabGridDialogCoordinator.java
index 1469f4e9f320bce344a7a2b75316c0e345cab4f9..1282d37a4efd97261a05e0ee7c8c174fcb3488f7 100644
--- a/chrome/android/features/tab_ui/java/src/org/chromium/chrome/browser/tasks/tab_management/TabGridDialogCoordinator.java
+++ b/chrome/android/features/tab_ui/java/src/org/chromium/chrome/browser/tasks/tab_management/TabGridDialogCoordinator.java
@@ -57,6 +57,10 @@ import org.chromium.ui.widget.ViewRectProvider;
 
 import java.util.List;
 
+import org.chromium.base.ContextUtils;
+
+import org.chromium.base.ContextUtils;
+
 /**
  * A coordinator for TabGridDialog component. Manages the communication with {@link
  * TabListCoordinator} as well as the life-cycle of shared component objects.
@@ -186,13 +190,18 @@ public class TabGridDialogCoordinator implements TabGridDialogMediator.DialogCon
                             showColorPickerPopupRunnable,
                             actionConfirmationManager);
 
+            int mode = TabUiFeatureUtilities.shouldUseListMode()
+                            ? TabListCoordinator.TabListMode.LIST
+                            : TabListCoordinator.TabListMode.GRID;
+            final String active_tabswitcher = ContextUtils.getAppSharedPreferences().getString("active_tabswitcher", "default");
+            if (active_tabswitcher.equals("default") || active_tabswitcher.equals("grid"))
+                mode = TabListCoordinator.TabListMode.GRID;
+
             // TODO(crbug.com/40662311) : Remove the inline mode logic here, make the constructor to
             // take in a mode parameter instead.
             mTabListCoordinator =
                     new TabListCoordinator(
-                            TabUiFeatureUtilities.shouldUseListMode()
-                                    ? TabListCoordinator.TabListMode.LIST
-                                    : TabListCoordinator.TabListMode.GRID,
+                            mode,
                             activity,
                             mBrowserControlsStateProvider,
                             mModalDialogManager,
@@ -285,6 +294,10 @@ public class TabGridDialogCoordinator implements TabGridDialogMediator.DialogCon
                     TabUiFeatureUtilities.shouldUseListMode()
                             ? TabListCoordinator.TabListMode.LIST
                             : TabListCoordinator.TabListMode.GRID;
+            final String active_tabswitcher = ContextUtils.getAppSharedPreferences().getString("active_tabswitcher", "default");
+            if (active_tabswitcher.equals("default") || active_tabswitcher.equals("grid"))
+                mode = TabListCoordinator.TabListMode.GRID;
+
             ViewGroup container = mDialogView.findViewById(R.id.dialog_container_view);
             mTabListEditorCoordinator =
                     new TabListEditorCoordinator(
