diff --git a/chrome/browser/ui/web_applications/BUILD.gn b/chrome/browser/ui/web_applications/BUILD.gn
index 2f15bf6f5b3c586387ba75893205eb196e18c390..cd599a8ffb56f5d3b2d01123e93e13f5d70da454 100644
--- a/chrome/browser/ui/web_applications/BUILD.gn
+++ b/chrome/browser/ui/web_applications/BUILD.gn
@@ -130,6 +130,10 @@ source_set("browser_tests") {
   if (is_chromeos) {
     deps += [ "//chromeos/constants" ]
   }
+
+  if (is_android) {
+    deps -= ["//chrome/test:test_support_ui", "//chrome/browser/apps/link_capturing", "//chrome/browser/apps/link_capturing:test_support", "//chrome/browser/picture_in_picture:test_support_ui"]
+  }
 }
 
 # On Lacros, these browser tests require Ash to be running, as they rely on web
@@ -199,4 +203,7 @@ source_set("app_service_browser_tests") {
       "//chromeos/startup",
     ]
   }
+  if (is_android) {
+    deps -= ["//chrome/test:test_support_ui", "//chrome/browser/apps/link_capturing", "//chrome/browser/apps/link_capturing:test_support"]
+  }
 }
