diff --git a/chrome/android/java/src/org/chromium/chrome/browser/tabmodel/TabModelImpl.java b/chrome/android/java/src/org/chromium/chrome/browser/tabmodel/TabModelImpl.java
index ea4afd0e46918faa4e41e529ebbccd4fdd9ba26a..181d8be9483a32cc0784dfb4a3d01addc0555a65 100644
--- a/chrome/android/java/src/org/chromium/chrome/browser/tabmodel/TabModelImpl.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/tabmodel/TabModelImpl.java
@@ -779,9 +779,9 @@ public class TabModelImpl extends TabModelJniBridge {
      */
     @Override
     protected boolean createTabWithWebContents(
-            Tab parent, Profile profile, WebContents webContents) {
+            Tab parent, Profile profile, WebContents webContents, @TabLaunchType int type) {
         return getTabCreator(profile.isOffTheRecord())
-                .createTabWithWebContents(parent, webContents, TabLaunchType.FROM_RECENT_TABS);
+                .createTabWithWebContents(parent, webContents, type);
     }
 
     @Override
@@ -828,6 +828,24 @@ public class TabModelImpl extends TabModelJniBridge {
         return mIndex;
     }
 
+    @Override
+    public int getLastNonExtensionActiveIndex() {
+        int result = mIndex;
+        Tab currentTab = getTabAt(mIndex);
+        // We get the adjacent tab in case we are currently on a chrome-extension page
+        if (currentTab != null && currentTab.getUrl() != null && (currentTab.getUrl().getSpec().contains("chrome-extension://"))) {
+          Tab parentTab = findTabInAllTabModels(currentTab.getParentId());
+            if (parentTab != null) {
+                result =  indexOf(parentTab);
+            } else {
+                    result = mIndex - 1;
+            }	
+	    }
+        if (result < 0)
+          return 0;
+        return result;    
+    }
+
     @Override
     protected boolean isSessionRestoreInProgress() {
         return mModelDelegate.isSessionRestoreInProgress();
