diff --git a/chrome/browser/resources/settings/BUILD.gn b/chrome/browser/resources/settings/BUILD.gn
index 22fdc428177889b68d4a5ccb15ba1cdd39d4302f..c42bfb0a809e32cebdb481ce651ee13960f17661 100644
--- a/chrome/browser/resources/settings/BUILD.gn
+++ b/chrome/browser/resources/settings/BUILD.gn
@@ -8,7 +8,7 @@ import("//crypto/features.gni")
 import("//ui/webui/resources/tools/build_webui.gni")
 import("./settings.gni")
 
-assert(!is_android, "Android does not use these settings")
+#assert(!is_android, "Android does not use these settings")
 
 build_webui("build") {
   grd_prefix = "settings"
