diff --git a/chrome/browser/resources/privacy_sandbox/BUILD.gn b/chrome/browser/resources/privacy_sandbox/BUILD.gn
index d03c15d8d0c8817c160377c048c53ac6c72cfc0e..010e2393cea3b413e5243e4b5c3a4dad260fc7f8 100644
--- a/chrome/browser/resources/privacy_sandbox/BUILD.gn
+++ b/chrome/browser/resources/privacy_sandbox/BUILD.gn
@@ -5,7 +5,7 @@
 import("//chrome/common/features.gni")
 import("//ui/webui/resources/tools/build_webui.gni")
 
-assert(!is_android, "!is_android check failed")
+#assert(!is_android, "!is_android check failed")
 
 build_webui("build") {
   grd_prefix = "privacy_sandbox"
