diff --git a/chrome/browser/ui/android/toolbar/BUILD.gn b/chrome/browser/ui/android/toolbar/BUILD.gn
index ce1b3b257a3fbee98ced527222d25f05d5c9051c..4bf8d16dd2ad36d017c35699b7b0d54fe4aeb47b 100644
--- a/chrome/browser/ui/android/toolbar/BUILD.gn
+++ b/chrome/browser/ui/android/toolbar/BUILD.gn
@@ -191,6 +191,15 @@ android_library("java") {
   ]
 
   resources_package = "org.chromium.chrome.browser.toolbar"
+  sources += [
+      "//mises/android/java/org/chromium/chrome/browser/toolbar/top/HandButton.java",
+  ]
+  deps += [
+    "//third_party/androidx:androidx_fragment_fragment_java",
+    "//mises/android/features/mises:public_java",
+    "//mises/android/features/mises:java_resources",
+    "//mises/third_party/android_deps:com_github_bumptech_glide_java",
+  ]
 }
 
 generate_jni("jni_headers") {
@@ -284,7 +293,7 @@ android_resources("java_resources") {
     "java/res/values/values.xml",
     "java/res/xml/adaptive_toolbar_preference.xml",
   ]
-
+  import("//mises/browser/ui/android/toolbar/sources.gni") sources += mises_browser_ui_android_toolbar_java_resources_sources
   deps = [
     "//chrome/browser/ui/android/strings:ui_strings_grd",
     "//chrome/browser/ui/android/theme:java_resources",
