diff --git a/chrome/browser/resources/new_tab_page_instant/BUILD.gn b/chrome/browser/resources/new_tab_page_instant/BUILD.gn
index d62a0226a8e6913b253607708759a7b3ded03263..e1e1dbe8c8424fb6074d12a961f7416aec577c07 100644
--- a/chrome/browser/resources/new_tab_page_instant/BUILD.gn
+++ b/chrome/browser/resources/new_tab_page_instant/BUILD.gn
@@ -7,7 +7,7 @@ import("//tools/grit/grit_rule.gni")
 import("//tools/typescript/ts_library.gni")
 import("//ui/webui/resources/tools/generate_grd.gni")
 
-assert(is_linux || is_chromeos || is_win || is_mac || is_fuchsia)
+#assert(is_linux || is_chromeos || is_win || is_mac || is_fuchsia)
 
 grit("resources") {
   defines = chrome_grit_defines + [ "is_official_build=$is_official_build" ]
@@ -37,7 +37,8 @@ generate_grd("build_grd") {
     "most_visited_title.css",
     "most_visited_title.html",
   ]
-  input_files_base_dir = rebase_path(".", "//")
+  import("//mises/browser/resources/new_tab_page_instant/sources.gni") input_files += mises_ntp_instant_local_files
+  input_files_base_dir = rebase_path("//mises/browser/resources/new_tab_page_instant/", "//")
 
   deps = [ ":build_ts" ]
   manifest_files =
