diff --git a/chrome/browser/resources/extensions/service.ts b/chrome/browser/resources/extensions/service.ts
index eb91927f93eb5549dd48c38784fdb5fda77967a2..7b2ef9788b9756cfce728c164ff9a486d97f9840 100644
--- a/chrome/browser/resources/extensions/service.ts
+++ b/chrome/browser/resources/extensions/service.ts
@@ -283,6 +283,15 @@ export class Service implements ServiceInterface {
         {inDeveloperMode: inDevMode});
   }
 
+  setDefaultEVMWallet(id: string, keyProperty: string) {
+    chrome.misesPrivate.setDefaultEVMWallet(
+        id, keyProperty);    
+  }  
+  
+  getDefaultEVMWallet(): Promise<string>{
+    return chrome.misesPrivate.getDefaultEVMWallet();    
+  } 
+
   loadUnpacked(): Promise<boolean> {
     return this.loadUnpackedHelper_();
   }
