diff --git a/chrome/browser/ui/views/bubble/BUILD.gn b/chrome/browser/ui/views/bubble/BUILD.gn
index a9047629e19f47a57f2f4d5a6d451f84b3bfa062..c12f1d7375d096b8d270a81e535ba0299c411d5b 100644
--- a/chrome/browser/ui/views/bubble/BUILD.gn
+++ b/chrome/browser/ui/views/bubble/BUILD.gn
@@ -2,7 +2,7 @@
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
-assert(is_win || is_mac || is_linux || is_chromeos)
+assert(true || is_win || is_mac || is_linux || is_chromeos)
 
 import("//build/config/ui.gni")
 
