diff --git a/chrome/browser/resources/lens/overlay/search_bubble/BUILD.gn b/chrome/browser/resources/lens/overlay/search_bubble/BUILD.gn
index 38bef05e3e017201b64f63523efd43aa3bebc935..06b2bfaebd93ad7940dc5d2f51362c1e5cb860f3 100644
--- a/chrome/browser/resources/lens/overlay/search_bubble/BUILD.gn
+++ b/chrome/browser/resources/lens/overlay/search_bubble/BUILD.gn
@@ -5,7 +5,7 @@
 import("//build/config/chrome_build.gni")
 import("//ui/webui/resources/tools/build_webui.gni")
 
-assert(toolkit_views)
+#assert(toolkit_views)
 
 build_webui("build") {
   grd_prefix = "lens_search_bubble"
