diff --git a/chrome/browser/resources/new_tab_page/icons/BUILD.gn b/chrome/browser/resources/new_tab_page/icons/BUILD.gn
index ba3ca425bc417c6a2ccfb5f77123c2d25b282c75..6a0fcf55220662e820d92633bf55ddbfc52734d5 100644
--- a/chrome/browser/resources/new_tab_page/icons/BUILD.gn
+++ b/chrome/browser/resources/new_tab_page/icons/BUILD.gn
@@ -4,7 +4,7 @@
 
 import("//ui/webui/resources/tools/generate_grd.gni")
 
-assert(!is_android)
+#assert(!is_android)
 
 generate_grd("build_grdp") {
   grd_prefix = "new_tab_page"
