diff --git a/chrome/android/java/src/org/chromium/chrome/browser/customtabs/content/CustomTabActivityNavigationController.java b/chrome/android/java/src/org/chromium/chrome/browser/customtabs/content/CustomTabActivityNavigationController.java
index 7000df22af600c401854815636eb1626ee2d82ee..170e1644a2c68910384bc319e3fbf3ef886325e2 100644
--- a/chrome/android/java/src/org/chromium/chrome/browser/customtabs/content/CustomTabActivityNavigationController.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/customtabs/content/CustomTabActivityNavigationController.java
@@ -407,7 +407,11 @@ public class CustomTabActivityNavigationController
             if (mIntentDataProvider.isInfoPage()) {
                 IntentHandler.startChromeLauncherActivityForTrustedIntent(intent);
             } else {
-                mActivity.startActivity(intent, startActivityOptions);
+                try {
+                    mActivity.startActivity(intent, startActivityOptions);
+                } catch (android.content.ActivityNotFoundException ex) {
+                    // avoid crashing.
+                }
                 finish(FinishReason.OPEN_IN_BROWSER);
             }
         }
