diff --git a/chrome/android/java/src/org/chromium/chrome/browser/WarmupManager.java b/chrome/android/java/src/org/chromium/chrome/browser/WarmupManager.java
index 8763230e5b289d38831e3e87fdc16f4cae11050b..ff4f66a8668c3c0c35c3d5f71dfec7d3838f35e1 100644
--- a/chrome/android/java/src/org/chromium/chrome/browser/WarmupManager.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/WarmupManager.java
@@ -42,6 +42,10 @@ import java.util.HashSet;
 import java.util.Map;
 import java.util.Set;
 
+import org.chromium.base.ContextUtils;
+import android.view.Gravity;
+import org.chromium.components.browser_ui.widget.CoordinatorLayoutForPointer;
+
 /**
  * This class is a singleton that holds utilities for warming up Chrome and prerendering urls
  * without creating the Activity.
@@ -120,6 +124,13 @@ public class WarmupManager {
                     layoutInflater, R.layout.main, contentHolder);
             if (toolbarContainerId != ActivityUtils.NO_RESOURCE_ID) {
                 ViewStub stub = (ViewStub) mainView.findViewById(R.id.control_container_stub);
+                if (ContextUtils.getAppSharedPreferences().getBoolean("enable_bottom_toolbar", false)) {
+                      CoordinatorLayoutForPointer.LayoutParams params = new CoordinatorLayoutForPointer.LayoutParams(
+                        CoordinatorLayoutForPointer.LayoutParams.MATCH_PARENT, CoordinatorLayoutForPointer.LayoutParams.WRAP_CONTENT);
+                      params.gravity = Gravity.START | Gravity.BOTTOM;
+                      stub.setLayoutParams(params);
+                }
+
                 stub.setLayoutResource(toolbarContainerId);
                 stub.inflate();
             }
