diff --git a/chrome/browser/resources/new_tab_page/modules/recipes_v2/icons/BUILD.gn b/chrome/browser/resources/new_tab_page/modules/recipes_v2/icons/BUILD.gn
index a84f8ba2065fa87873f1a0e290597303135c05d9..75ffe943779c8c32a790abc735fd332e02346032 100644
--- a/chrome/browser/resources/new_tab_page/modules/recipes_v2/icons/BUILD.gn
+++ b/chrome/browser/resources/new_tab_page/modules/recipes_v2/icons/BUILD.gn
@@ -4,7 +4,7 @@
 
 import("//ui/webui/resources/tools/generate_grd.gni")
 
-assert(!is_android)
+#assert(!is_android)
 
 generate_grd("build_grdp") {
   grd_prefix = "recipes"
