diff --git a/chrome/android/java/src/org/chromium/chrome/browser/webapps/WebappActivity.java b/chrome/android/java/src/org/chromium/chrome/browser/webapps/WebappActivity.java
index ddad2dd419d9f133af5fcfc8b9e90cea1d539aaf..2bf21c6640ff86eef70db5c74e5953ac96249b39 100644
--- a/chrome/android/java/src/org/chromium/chrome/browser/webapps/WebappActivity.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/webapps/WebappActivity.java
@@ -96,4 +96,7 @@ public class WebappActivity extends BaseCustomTabActivity {
                         ? null
                         : mWebappActivityCoordinator.getWebappInfo());
     }
+
+    @Override
+    public void setLastVisibleItemTitle(String itemTitle) {}
 }
