diff --git a/chrome/browser/ui/content_settings/BUILD.gn b/chrome/browser/ui/content_settings/BUILD.gn
index c097f9ab151a27259989ebda7d240da557f5ebd9..9038b54b094283595c58a1483ec3bc8b00822be7 100644
--- a/chrome/browser/ui/content_settings/BUILD.gn
+++ b/chrome/browser/ui/content_settings/BUILD.gn
@@ -2,7 +2,7 @@
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
-assert(is_win || is_mac || is_linux || is_chromeos)
+#assert(is_win || is_mac || is_linux || is_chromeos)
 
 source_set("content_settings") {
   sources = []
@@ -31,7 +31,7 @@ source_set("content_settings") {
 source_set("impl") {
   sources = []
 
-  if (!is_android) {
+  if (true || !is_android) {
     sources += [
       "content_setting_bubble_model.cc",
       "content_setting_image_model.cc",
