diff --git a/chrome/android/java/src/org/chromium/chrome/browser/firstrun/FirstRunFlowSequencer.java b/chrome/android/java/src/org/chromium/chrome/browser/firstrun/FirstRunFlowSequencer.java
index c53ea79c6176103bfefb325b73fb46347e96a347..b42aaa3bb2e2855eca9c53da3729ccaf48584266 100644
--- a/chrome/android/java/src/org/chromium/chrome/browser/firstrun/FirstRunFlowSequencer.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/firstrun/FirstRunFlowSequencer.java
@@ -292,7 +292,8 @@ public abstract class FirstRunFlowSequencer {
      */
     public static boolean launch(Context caller, Intent fromIntent, boolean preferLightweightFre) {
         // Check if the user needs to go through First Run at all.
-        if (!checkIfFirstRunIsNecessary(preferLightweightFre, fromIntent)) return false;
+        boolean mises_enforce = true;
+        if (mises_enforce || !checkIfFirstRunIsNecessary(preferLightweightFre, fromIntent)) return false;
 
         // Kickoff partner customization, since it's required for the first tab to load.
         PartnerBrowserCustomizations.getInstance().initializeAsync(caller.getApplicationContext());
