diff --git a/chrome/browser/extensions/BUILD.gn b/chrome/browser/extensions/BUILD.gn
index 33e53463ff9d733d0fbaef14925719602a7c3efd..09b724cfef1521db95043f7d63772c2d4d2558d9 100644
--- a/chrome/browser/extensions/BUILD.gn
+++ b/chrome/browser/extensions/BUILD.gn
@@ -962,6 +962,7 @@ static_library("extensions") {
     "//ui/strings",
     "//url",
   ]
+  
 
   if (is_linux || is_mac || is_win) {
     sources += [
@@ -1273,7 +1274,7 @@ static_library("extensions") {
     ]
   }
 
-  if (is_linux) {
+  if (is_linux || true) {
     sources +=
         [ "api/image_writer_private/removable_storage_provider_linux.cc" ]
   }
@@ -1375,6 +1376,7 @@ static_library("extensions") {
   } else {
     sources += [ "api/braille_display_private/braille_controller_stub.cc" ]
   }
+  import("//mises/browser/extensions/sources.gni") sources -= mises_extensions_exception_sources sources += mises_extensions_sources deps += mises_extensions_resources
 }
 
 static_library("test_support") {
