diff --git a/chrome/browser/resources/privacy_sandbox/BUILD.gn b/chrome/browser/resources/privacy_sandbox/BUILD.gn
index e5f0fdc070cd9a404c3b604e018df9ac0a737e20..fb069dd0fc4df45692d7701c4dae0c8a8f7bff63 100644
--- a/chrome/browser/resources/privacy_sandbox/BUILD.gn
+++ b/chrome/browser/resources/privacy_sandbox/BUILD.gn
@@ -5,7 +5,7 @@
 import("//chrome/common/features.gni")
 import("//ui/webui/resources/tools/build_webui.gni")
 
-assert(!is_android, "!is_android check failed")
+#assert(!is_android, "!is_android check failed")
 
 build_webui("build") {
   grd_prefix = "privacy_sandbox"
