diff --git a/chrome/browser/tabmodel/android/java/src/org/chromium/chrome/browser/tabmodel/TabList.java b/chrome/browser/tabmodel/android/java/src/org/chromium/chrome/browser/tabmodel/TabList.java
index 4d1e2f0534c41f06da4a474437175eb56276732e..bb4d3b2c2ae3251a23a4b2805b17795fa2a66b2a 100644
--- a/chrome/browser/tabmodel/android/java/src/org/chromium/chrome/browser/tabmodel/TabList.java
+++ b/chrome/browser/tabmodel/android/java/src/org/chromium/chrome/browser/tabmodel/TabList.java
@@ -26,6 +26,8 @@ public interface TabList {
     @Deprecated
     boolean isIncognito();
 
+    int getLastNonExtensionActiveIndex();
+
     /**
      * Returns whether this tab model contains only off-the-record tabs or only normal tabs.
      *
