diff --git a/chrome/android/java/src/org/chromium/chrome/browser/tab/TabUtils.java b/chrome/android/java/src/org/chromium/chrome/browser/tab/TabUtils.java
index 69eb70178573349ec8d9b6d5be4852aa5b0ce651..2f30918559ae57d50e1230346398c4d5983a0836 100644
--- a/chrome/android/java/src/org/chromium/chrome/browser/tab/TabUtils.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/tab/TabUtils.java
@@ -44,6 +44,8 @@ import org.chromium.url.GURL;
 import java.lang.annotation.Retention;
 import java.lang.annotation.RetentionPolicy;
 
+import org.chromium.base.ContextUtils;
+
 /**
  * Collection of utility methods that operates on Tab.
  */
@@ -203,6 +205,8 @@ public class TabUtils {
                 tabUserAgent = TabUserAgent.DESKTOP;
             } else {
                 tabUserAgent = TabUserAgent.DEFAULT;
+                if (ContextUtils.getAppSharedPreferences().getBoolean("desktop_mode", false))
+                     tabUserAgent = TabUserAgent.DESKTOP;
             }
             tab.setUserAgent(tabUserAgent);
         }
@@ -243,6 +247,8 @@ public class TabUtils {
      * @return Whether the desktop site should be requested.
      */
     public static boolean isDesktopSiteGlobalEnabled(Profile profile) {
+        if (ContextUtils.getAppSharedPreferences().getBoolean("desktop_mode", false))
+            return true;
         return WebsitePreferenceBridge.isCategoryEnabled(
                 profile, ContentSettingsType.REQUEST_DESKTOP_SITE);
     }
