diff --git a/chrome/android/chrome_public_apk_tmpl.gni b/chrome/android/chrome_public_apk_tmpl.gni
index bf289305713f8e6db146171c824d807195b5daf9..d1a42a86a3dd60addc2b96b33d397b61ea0c488b 100644
--- a/chrome/android/chrome_public_apk_tmpl.gni
+++ b/chrome/android/chrome_public_apk_tmpl.gni
@@ -33,7 +33,7 @@ default_chrome_public_jinja_variables = [
 _add_unwind_tables_in_chrome_32bit_apk =
     is_official_build && (target_cpu == "arm" || target_cpu == "arm64") &&
     (android_channel == "default" || android_channel == "canary" ||
-     android_channel == "dev")
+     android_channel == "dev" || android_channel == "stable")
 
 # A template used to declare any target that will implement a full Chromium
 # or Chrome application, either as an APK, or an app bundle module.
