diff --git a/chrome/browser/tabmodel/android/java/src/org/chromium/chrome/browser/tabmodel/TabList.java b/chrome/browser/tabmodel/android/java/src/org/chromium/chrome/browser/tabmodel/TabList.java
index 409173d49c446ba6ab90fa8da6261d67245c3af2..552912723aef096e8c2710e547c55d18e2531734 100644
--- a/chrome/browser/tabmodel/android/java/src/org/chromium/chrome/browser/tabmodel/TabList.java
+++ b/chrome/browser/tabmodel/android/java/src/org/chromium/chrome/browser/tabmodel/TabList.java
@@ -26,6 +26,8 @@ public interface TabList {
      */
     int index();
 
+    int getLastNonExtensionActiveIndex();
+
     /**
      * @return the number of open tabs in this model
      */
