diff --git a/chrome/browser/ui/android/toolbar/java/src/org/chromium/chrome/browser/toolbar/top/ToolbarTablet.java b/chrome/browser/ui/android/toolbar/java/src/org/chromium/chrome/browser/toolbar/top/ToolbarTablet.java
index 13fd6a6109749d18c5691f040950954c5b6c5879..c583502fd37b404713a44eaa692fae9589059cdb 100644
--- a/chrome/browser/ui/android/toolbar/java/src/org/chromium/chrome/browser/toolbar/top/ToolbarTablet.java
+++ b/chrome/browser/ui/android/toolbar/java/src/org/chromium/chrome/browser/toolbar/top/ToolbarTablet.java
@@ -62,6 +62,14 @@ import java.util.ArrayList;
 import java.util.Collection;
 import java.util.function.BooleanSupplier;
 
+
+import org.chromium.ui.display.DisplayAndroid;
+import android.os.Bundle;
+import android.widget.ImageButton;
+import android.widget.PopupWindow;
+import org.chromium.base.Log;
+import org.chromium.ui.widget.ChromeImageButton;
+
 /**
  * The Toolbar object for Tablet screens.
  */
@@ -91,6 +99,7 @@ public class ToolbarTablet
     private ImageButton mSaveOfflineButton;
     private ToggleTabStackButton mSwitcherButton;
 
+
     private OnClickListener mBookmarkListener;
 
     private boolean mIsInTabSwitcherMode;
@@ -175,6 +184,7 @@ public class ToolbarTablet
         mShouldAnimateButtonVisibilityChange = false;
         mToolbarButtonsVisible = true;
         mToolbarButtons = new ImageButton[] {mBackButton, mForwardButton, mReloadButton};
+        setupMisesButton();
 
         setTooltipTextForToolbarButtons();
     }
@@ -335,6 +345,7 @@ public class ToolbarTablet
 
         mSaveOfflineButton.setOnClickListener(this);
         mSaveOfflineButton.setOnLongClickListener(this);
+        setupMisesButtonClickListener();
     }
 
     @Override
@@ -402,6 +413,7 @@ public class ToolbarTablet
             mOfflineDownloader.downloadPage(getContext(), getToolbarDataProvider().getTab());
             RecordUserAction.record("MobileToolbarDownloadPage");
         }
+        onClickMisesButton(v);
     }
 
     @Override
@@ -500,6 +512,8 @@ public class ToolbarTablet
         if (mOptionalButton != null && mOptionalButtonUsesTint) {
             ImageViewCompat.setImageTintList(mOptionalButton, tint);
         }
+        if (mMisesExtensionButton != null && mExtensionButtonDrawableOveride == null) 
+            ImageViewCompat.setImageTintList(mMisesExtensionButton, tint);
     }
 
     @Override
@@ -882,4 +896,12 @@ public class ToolbarTablet
     void setToolbarButtonsVisibleForTesting(boolean value) {
         mToolbarButtonsVisible = value;
     }
+
+    private void setupMisesButton() {
+        mMisesExtensionButton =  (ChromeImageButton) findViewById(R.id.mises_extension_button);
+    }
+    private void setupMisesButtonClickListener() {
+	    if (mMisesExtensionButton != null) mMisesExtensionButton.setOnClickListener(this);
+    }
+
 }
