diff --git a/chrome/android/chrome_public_apk_tmpl.gni b/chrome/android/chrome_public_apk_tmpl.gni
index 56949f4bb7496f40204bfc72f74123104f5ea1ff..a8c9a6e8f63e107ccf411efc4d32ddd71f6a97f3 100644
--- a/chrome/android/chrome_public_apk_tmpl.gni
+++ b/chrome/android/chrome_public_apk_tmpl.gni
@@ -326,6 +326,17 @@ template("chrome_public_common_apk_or_module_tmpl") {
       strip_unused_resources = is_official_build
     }
 
+    if (_is_bundle) {
+        # Resources config for blocklisting resource names from obfuscation
+        resources_config_paths = [
+          "//android_webview/aapt2.config",
+          "//chrome/android/aapt2.config",
+        ]
+        if (defined(invoker.resources_config_paths)) {
+          resources_config_paths += invoker.resources_config_paths
+        }
+    }
+
     if (!defined(aapt_locale_allowlist)) {
       # Include resource strings files only for supported locales.
       aapt_locale_allowlist = platform_pak_locales
@@ -712,7 +723,7 @@ template("monochrome_public_common_apk_or_module_tmpl") {
       }
 
       # We only optimize resources in bundles.
-      if (_is_bundle_module) {
+      if (_is_bundle_module && false) {
         # Resources config for blocklisting resource names from obfuscation
         resources_config_paths = [
           "//android_webview/aapt2.config",
