diff --git a/chrome/android/java/src/org/chromium/chrome/browser/settings/MainSettings.java b/chrome/android/java/src/org/chromium/chrome/browser/settings/MainSettings.java
index 6a8e28d741e7e99ce20f5f6c56d0fd2a81f7a240..40b984081360f69cf3b407bb3c4b25c7f438534b 100644
--- a/chrome/android/java/src/org/chromium/chrome/browser/settings/MainSettings.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/settings/MainSettings.java
@@ -181,6 +181,12 @@ public class MainSettings extends PreferenceFragmentCompat
             setManagedPreferenceDelegateForPreference(PREF_PASSWORDS);
         }
 
+        Preference safetyCheck = findPreference(PREF_SAFETY_CHECK);
+        if (safetyCheck != null)
+            getPreferenceScreen().removePreference(safetyCheck);
+	    removePreferenceIfPresent("toolbar_options");
+	    removePreferenceIfPresent("nightmode");
+
         setManagedPreferenceDelegateForPreference(PREF_SEARCH_ENGINE);
 
         if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
@@ -193,6 +199,7 @@ public class MainSettings extends PreferenceFragmentCompat
             PackageManager pm = getActivity().getPackageManager();
             if (intent.resolveActivity(pm) != null) {
                 Preference notifications = findPreference(PREF_NOTIFICATIONS);
+                if (notifications != null)
                 notifications.setOnPreferenceClickListener(preference -> {
                     startActivity(intent);
                     // We handle the click so the default action isn't triggered.
@@ -215,7 +222,7 @@ public class MainSettings extends PreferenceFragmentCompat
         new AdaptiveToolbarStatePredictor(null).recomputeUiState(uiState -> {
             // We don't show the toolbar shortcut settings page if disabled from finch.
             if (uiState.canShowUi) return;
-            getPreferenceScreen().removePreference(findPreference(PREF_TOOLBAR_SHORTCUT));
+            removePreferenceIfPresent(PREF_TOOLBAR_SHORTCUT);
         });
     }
 
