diff --git a/chrome/browser/resources/side_panel/reading_list/BUILD.gn b/chrome/browser/resources/side_panel/reading_list/BUILD.gn
index 2d0a7eb4c28b22d4d5821a9a7b704ecf681b4d14..184c36f7f4ebe6623e1f09f367bbae6ca0088954 100644
--- a/chrome/browser/resources/side_panel/reading_list/BUILD.gn
+++ b/chrome/browser/resources/side_panel/reading_list/BUILD.gn
@@ -4,7 +4,7 @@
 
 import("//ui/webui/resources/tools/build_webui.gni")
 
-assert(!is_android)
+#assert(!is_android)
 
 build_webui("build") {
   grd_prefix = "side_panel_reading_list"
