diff --git a/chrome/browser/preferences/android/java/src/org/chromium/chrome/browser/preferences/ChromePreferenceKeys.java b/chrome/browser/preferences/android/java/src/org/chromium/chrome/browser/preferences/ChromePreferenceKeys.java
index 48712bdbe58487b97e48aab253354a3f6b03a907..9bc80fa7ea43463c9e21396c67255d9735564481 100644
--- a/chrome/browser/preferences/android/java/src/org/chromium/chrome/browser/preferences/ChromePreferenceKeys.java
+++ b/chrome/browser/preferences/android/java/src/org/chromium/chrome/browser/preferences/ChromePreferenceKeys.java
@@ -907,6 +907,11 @@ public final class ChromePreferenceKeys {
     public static final String BLUETOOTH_NOTIFICATION_IDS = "Chrome.Bluetooth.NotificationIds";
     public static final String USB_NOTIFICATION_IDS = "Chrome.USB.NotificationIds";
 
+    public static final String DARKEN_WEBSITES_ENABLED = "darken_websites_enabled";
+
+    public static final String MISES_USER_INFO = "mises_user_info";
+    public static final String MISES_FIRST_RUN = "mises_first_run";
+
     /**
      * These values are currently used as SharedPreferences keys, along with the keys in
      * {@link LegacyChromePreferenceKeys#getKeysInUse()}. Add new SharedPreferences keys
