diff --git a/chrome/browser/ui/android/toolbar/java/src/org/chromium/chrome/browser/toolbar/top/ToolbarTablet.java b/chrome/browser/ui/android/toolbar/java/src/org/chromium/chrome/browser/toolbar/top/ToolbarTablet.java
index a158d0d3652ade968a504548fa220176bea8299b..f4f52cfec3e2e770019ddf26830a20e9beb39ee7 100644
--- a/chrome/browser/ui/android/toolbar/java/src/org/chromium/chrome/browser/toolbar/top/ToolbarTablet.java
+++ b/chrome/browser/ui/android/toolbar/java/src/org/chromium/chrome/browser/toolbar/top/ToolbarTablet.java
@@ -67,6 +67,14 @@ import java.util.ArrayList;
 import java.util.Collection;
 import java.util.function.BooleanSupplier;
 
+
+import org.chromium.ui.display.DisplayAndroid;
+import android.os.Bundle;
+import android.widget.ImageButton;
+import android.widget.PopupWindow;
+import org.chromium.base.Log;
+import org.chromium.ui.widget.ChromeImageButton;
+
 /** The Toolbar object for Tablet screens. */
 @SuppressLint("Instantiatable")
 public class ToolbarTablet extends ToolbarLayout
@@ -92,6 +100,7 @@ public class ToolbarTablet extends ToolbarLayout
     private ImageButton mSaveOfflineButton;
     private View mIncognitoIndicator;
 
+
     private OnClickListener mBookmarkListener;
 
     private boolean mIsInTabSwitcherMode;
@@ -182,6 +191,7 @@ public class ToolbarTablet extends ToolbarLayout
         mShouldAnimateButtonVisibilityChange = false;
         mToolbarButtonsVisible = true;
         mToolbarButtons = new ImageButton[] {mBackButton, mForwardButton, mReloadButton};
+        setupMisesButton();
 
         setTooltipTextForToolbarButtons();
     }
@@ -350,6 +360,7 @@ public class ToolbarTablet extends ToolbarLayout
 
         mSaveOfflineButton.setOnClickListener(this);
         mSaveOfflineButton.setOnLongClickListener(this);
+        setupMisesButtonClickListener();
     }
 
     /**
@@ -439,6 +450,7 @@ public class ToolbarTablet extends ToolbarLayout
             mOfflineDownloader.downloadPage(getContext(), getToolbarDataProvider().getTab());
             RecordUserAction.record("MobileToolbarDownloadPage");
         }
+        onClickMisesButton(v);
     }
 
     @Override
@@ -571,6 +583,8 @@ public class ToolbarTablet extends ToolbarLayout
         if (mOptionalButton != null && mOptionalButtonUsesTint) {
             ImageViewCompat.setImageTintList(mOptionalButton, activityFocusTint);
         }
+        if (mMisesExtensionButton != null && mExtensionButtonDrawableOveride == null) 
+            ImageViewCompat.setImageTintList(mMisesExtensionButton, tint);
     }
 
     @Override
@@ -1004,4 +1018,12 @@ public class ToolbarTablet extends ToolbarLayout
     OnTouchListener getReloadButtonTouchListenerForTest() {
         return mReloadButtonTouchListener;
     }
+
+    private void setupMisesButton() {
+        mMisesExtensionButton =  (ChromeImageButton) findViewById(R.id.mises_extension_button);
+    }
+    private void setupMisesButtonClickListener() {
+	    if (mMisesExtensionButton != null) mMisesExtensionButton.setOnClickListener(this);
+    }
+
 }
