diff --git a/chrome/browser/resources/management/BUILD.gn b/chrome/browser/resources/management/BUILD.gn
index 386eaf976bf67530f9d5c6bfe9adce12b7ab7e72..aac0531ad73a936c76150a672b81babc711a15ca 100644
--- a/chrome/browser/resources/management/BUILD.gn
+++ b/chrome/browser/resources/management/BUILD.gn
@@ -5,7 +5,7 @@
 import("//chrome/browser/resources/tools/build_webui.gni")
 import("//chrome/common/features.gni")
 
-assert(!is_android, "!is_android check failed")
+# assert(!is_android, "!is_android check failed")
 
 build_webui("build") {
   grd_prefix = "management"
