diff --git a/chrome/browser/preferences/android/java/src/org/chromium/chrome/browser/preferences/ChromePreferenceKeys.java b/chrome/browser/preferences/android/java/src/org/chromium/chrome/browser/preferences/ChromePreferenceKeys.java
index 7e18844fcc3f4b3abce2b819b39fd302be13d7f7..5198b3d0513d945fef32b890a3c7ba87649e4198 100644
--- a/chrome/browser/preferences/android/java/src/org/chromium/chrome/browser/preferences/ChromePreferenceKeys.java
+++ b/chrome/browser/preferences/android/java/src/org/chromium/chrome/browser/preferences/ChromePreferenceKeys.java
@@ -992,6 +992,11 @@ public final class ChromePreferenceKeys {
     public static final String BLUETOOTH_NOTIFICATION_IDS = "Chrome.Bluetooth.NotificationIds";
     public static final String USB_NOTIFICATION_IDS = "Chrome.USB.NotificationIds";
 
+    public static final String DARKEN_WEBSITES_ENABLED = "darken_websites_enabled";
+
+    public static final String MISES_USER_INFO = "mises_user_info";
+    public static final String MISES_FIRST_RUN = "mises_first_run";
+
     /**
      * These values are currently used as SharedPreferences keys, along with the keys in
      * {@link LegacyChromePreferenceKeys#getKeysInUse()}. Add new SharedPreferences keys
