diff --git a/chrome/test/BUILD.gn b/chrome/test/BUILD.gn
index a5a8685b9627dc85f96a00eb87115604f239d4ee..285085a74e2e44f6785ce56bcbdabe03e62ff48d 100644
--- a/chrome/test/BUILD.gn
+++ b/chrome/test/BUILD.gn
@@ -343,6 +343,7 @@ static_library("test_support") {
     "//ui/gl",
   ]
 
+  import("//mises/chromium_src/chrome/test/sources.gni") deps += mises_chromium_src_chrome_test_test_support_deps sources += mises_chromium_src_chrome_test_test_support_sources
   data_deps = []
 
   if (enable_pdf) {
@@ -8107,6 +8108,10 @@ test("unit_tests") {
       "//tools/json_schema_compiler/test:unit_tests",
     ]
 
+    if (is_android) {
+      deps -= ["//extensions:extensions_resources",]
+    }
+
     if (enable_service_discovery) {
       sources += [ "../browser/extensions/api/mdns/mdns_api_unittest.cc" ]
     }
