diff --git a/chrome/common/features.gni b/chrome/common/features.gni
index 74b29b9907c6ad7f2b28f806398aea5ad3e980ab..6648cb843496b5e35bd2cb533b1cad095c629562 100644
--- a/chrome/common/features.gni
+++ b/chrome/common/features.gni
@@ -42,8 +42,8 @@ declare_args() {
   chrome_root_store_policy_supported = is_android
 
   # Enables support for background apps.
-  enable_background_contents = !is_android
-  enable_background_mode = !is_android && !is_chromeos
+  enable_background_contents = is_android || !is_android
+  enable_background_mode = is_android || !is_android && !is_chromeos
 
   # Enables usage of notifications via Chrome's MessageCenter.
   enable_chrome_notifications = is_mac || is_win || is_linux || is_fuchsia
@@ -73,7 +73,7 @@ declare_args() {
   enable_wayland_server = is_chromeos_ash
 
   # Enables the webui certificate viewer dialog.
-  enable_webui_certificate_viewer = toolkit_views
+  enable_webui_certificate_viewer = is_android || toolkit_views
 
   # optimize_webui was moved to ui/base/ui_features.gni
 }
