diff --git a/chrome/browser/ui/webui/side_panel/read_anything/read_anything_untrusted_page_handler.cc b/chrome/browser/ui/webui/side_panel/read_anything/read_anything_untrusted_page_handler.cc
index 6ebdfb4be6dd5a3d3586084230905b2bdd919268..645d263168afc3374a9b1120fae498d03fab9b4b 100644
--- a/chrome/browser/ui/webui/side_panel/read_anything/read_anything_untrusted_page_handler.cc
+++ b/chrome/browser/ui/webui/side_panel/read_anything/read_anything_untrusted_page_handler.cc
@@ -319,7 +319,7 @@ ReadAnythingUntrustedPageHandler::ReadAnythingUntrustedPageHandler(
   std::string prefs_lang = language_model->GetLanguages().front().lang_code;
   prefs_lang = language::ExtractBaseLanguage(prefs_lang);
   SetDefaultLanguageCode(prefs_lang);
-
+#if !BUILDFLAG(IS_ANDROID)
   if (features::IsReadAnythingWithScreen2xEnabled()) {
     screen_ai::ScreenAIServiceRouterFactory::GetForBrowserContext(profile_)
         ->GetServiceStateAsync(
@@ -333,7 +333,7 @@ ReadAnythingUntrustedPageHandler::ReadAnythingUntrustedPageHandler(
         ->GetServiceStateAsync(screen_ai::ScreenAIServiceRouter::Service::kOCR,
                                base::DoNothing());
   }
-
+#endif
   // Enable accessibility for the top level render frame and all descendants.
   // This causes AXTreeSerializer to reset and send accessibility events of
   // the AXTree when it is re-serialized.
