diff --git a/chrome/browser/safe_browsing/cloud_content_scanning/file_analysis_request.h b/chrome/browser/safe_browsing/cloud_content_scanning/file_analysis_request.h
index 09ed45d3b59c7651da03fab0fe0807df9720d634..13911bad7264c5a5c5d43e6eeaa09c3b7743f961 100644
--- a/chrome/browser/safe_browsing/cloud_content_scanning/file_analysis_request.h
+++ b/chrome/browser/safe_browsing/cloud_content_scanning/file_analysis_request.h
@@ -11,8 +11,10 @@
 #include "chrome/browser/enterprise/connectors/service_provider_config.h"
 #include "chrome/browser/safe_browsing/cloud_content_scanning/binary_upload_service.h"
 #include "chrome/common/safe_browsing/archive_analyzer_results.h"
+#if BUILDFLAG(FULL_SAFE_BROWSING)
 #include "chrome/services/file_util/public/cpp/sandboxed_rar_analyzer.h"
 #include "chrome/services/file_util/public/cpp/sandboxed_zip_analyzer.h"
+#endif
 
 namespace safe_browsing {
 
@@ -81,11 +83,13 @@ class FileAnalysisRequest : public BinaryUploadService::Request {
   // GetRequestData call.
   bool delay_opening_file_;
 
+#if BUILDFLAG(FULL_SAFE_BROWSING)
   // Used to unpack and analyze archives in a sandbox.
   std::unique_ptr<SandboxedZipAnalyzer, base::OnTaskRunnerDeleter>
       zip_analyzer_{nullptr, base::OnTaskRunnerDeleter(nullptr)};
   std::unique_ptr<SandboxedRarAnalyzer, base::OnTaskRunnerDeleter>
       rar_analyzer_{nullptr, base::OnTaskRunnerDeleter(nullptr)};
+#endif
 
   base::WeakPtrFactory<FileAnalysisRequest> weakptr_factory_{this};
 };
