diff --git a/chrome/browser/extensions/extension_service.cc b/chrome/browser/extensions/extension_service.cc
index 51d09735dee68145483f5c9c475b683c2c04229f..d7687ba3c778a15247a950ce5694ffc5404dc106 100644
--- a/chrome/browser/extensions/extension_service.cc
+++ b/chrome/browser/extensions/extension_service.cc
@@ -486,6 +486,10 @@ ExtensionService::ExtensionService(
   SetCurrentDeveloperMode(
       util::GetBrowserContextId(profile),
       profile->GetPrefs()->GetBoolean(prefs::kExtensionsUIDeveloperMode));
+  SetDefaultEVMWallet(
+      util::GetBrowserContextId(profile),
+      profile->GetPrefs()->GetString(prefs::kExtensionsUIDefaultEVMWalletID),
+      profile->GetPrefs()->GetString(prefs::kExtensionsUIDefaultEVMWalletKeyProperty));
 }
 
 PendingExtensionManager* ExtensionService::pending_extension_manager() {
