diff --git a/chrome/browser/ui/android/toolbar/java/src/org/chromium/chrome/browser/toolbar/top/ToolbarTablet.java b/chrome/browser/ui/android/toolbar/java/src/org/chromium/chrome/browser/toolbar/top/ToolbarTablet.java
index a78210e6c74c4c6a9d7f1cf484ab2fd3ad77df64..c966706934f31b93698508d36a895499ce3db9f8 100644
--- a/chrome/browser/ui/android/toolbar/java/src/org/chromium/chrome/browser/toolbar/top/ToolbarTablet.java
+++ b/chrome/browser/ui/android/toolbar/java/src/org/chromium/chrome/browser/toolbar/top/ToolbarTablet.java
@@ -65,6 +65,14 @@ import java.util.ArrayList;
 import java.util.Collection;
 import java.util.function.BooleanSupplier;
 
+
+import org.chromium.ui.display.DisplayAndroid;
+import android.os.Bundle;
+import android.widget.ImageButton;
+import android.widget.PopupWindow;
+import org.chromium.base.Log;
+import org.chromium.ui.widget.ChromeImageButton;
+
 /** The Toolbar object for Tablet screens. */
 @SuppressLint("Instantiatable")
 public class ToolbarTablet extends ToolbarLayout
@@ -89,6 +97,7 @@ public class ToolbarTablet extends ToolbarLayout
     private ImageButton mBookmarkButton;
     private ImageButton mSaveOfflineButton;
 
+
     private OnClickListener mBookmarkListener;
 
     private boolean mIsInTabSwitcherMode;
@@ -176,6 +185,7 @@ public class ToolbarTablet extends ToolbarLayout
         mShouldAnimateButtonVisibilityChange = false;
         mToolbarButtonsVisible = true;
         mToolbarButtons = new ImageButton[] {mBackButton, mForwardButton, mReloadButton};
+        setupMisesButton();
 
         setTooltipTextForToolbarButtons();
     }
@@ -343,6 +353,7 @@ public class ToolbarTablet extends ToolbarLayout
 
         mSaveOfflineButton.setOnClickListener(this);
         mSaveOfflineButton.setOnLongClickListener(this);
+        setupMisesButtonClickListener();
     }
 
     @Override
@@ -410,6 +421,7 @@ public class ToolbarTablet extends ToolbarLayout
             mOfflineDownloader.downloadPage(getContext(), getToolbarDataProvider().getTab());
             RecordUserAction.record("MobileToolbarDownloadPage");
         }
+        onClickMisesButton(v);
     }
 
     @Override
@@ -541,6 +553,8 @@ public class ToolbarTablet extends ToolbarLayout
         if (mOptionalButton != null && mOptionalButtonUsesTint) {
             ImageViewCompat.setImageTintList(mOptionalButton, activityFocusTint);
         }
+        if (mMisesExtensionButton != null && mExtensionButtonDrawableOveride == null) 
+            ImageViewCompat.setImageTintList(mMisesExtensionButton, tint);
     }
 
     @Override
@@ -953,4 +967,12 @@ public class ToolbarTablet extends ToolbarLayout
     public ImageButton getBookmarkButtonForTesting() {
         return mBookmarkButton;
     }
+
+    private void setupMisesButton() {
+        mMisesExtensionButton =  (ChromeImageButton) findViewById(R.id.mises_extension_button);
+    }
+    private void setupMisesButtonClickListener() {
+	    if (mMisesExtensionButton != null) mMisesExtensionButton.setOnClickListener(this);
+    }
+
 }
