diff --git a/chrome/common/BUILD.gn b/chrome/common/BUILD.gn
index f21f98e89fb254dc9d7865365ece3f50da5ef585..3b9e84c0f19dbb5527dd2c0170d8f880d3e8e738 100644
--- a/chrome/common/BUILD.gn
+++ b/chrome/common/BUILD.gn
@@ -89,6 +89,7 @@ source_set("channel_info") {
   } else if (is_fuchsia) {
     sources += [ "channel_info_fuchsia.cc" ]
   }
+  public_deps += [ "//mises/common:channel_info" ]
 }
 
 source_set("ini_parser") {
@@ -404,7 +405,8 @@ static_library("common_lib") {
       "media/chrome_media_drm_bridge_client.cc",
       "media/chrome_media_drm_bridge_client.h",
     ]
-  } else {
+  }
+  if (true) {
     # Non-Android.
     public_deps += [ "//chrome/common/importer" ]
     sources += [ "accessibility/read_anything_constants.h" ]
@@ -537,6 +539,7 @@ static_library("non_code_constants") {
     "//printing/buildflags",
     "//ui/base:buildflags",
   ]
+  configs += [ "//mises/common:constants_configs" ]
 }
 
 component("chrome_features") {
