diff --git a/chrome/browser/apps/app_service/BUILD.gn b/chrome/browser/apps/app_service/BUILD.gn
index 40b02cbffd0d3ed922efe95e56ddb3a74128520d..d6761cd28b6ea59b662bc5126e8ef30991710484 100644
--- a/chrome/browser/apps/app_service/BUILD.gn
+++ b/chrome/browser/apps/app_service/BUILD.gn
@@ -5,7 +5,7 @@
 import("//build/config/chromeos/ui_mode.gni")
 import("//extensions/buildflags/buildflags.gni")
 
-assert(!is_android)
+#assert(!is_android)
 
 source_set("app_service") {
   sources = [
