diff --git a/chrome/android/chrome_public_apk_tmpl.gni b/chrome/android/chrome_public_apk_tmpl.gni
index 43a9301530e6ee92c79ea5b375f2fcecfadcb34b..f3cf23ac65b3c82b8bd87b58f14297d7f04c7b51 100644
--- a/chrome/android/chrome_public_apk_tmpl.gni
+++ b/chrome/android/chrome_public_apk_tmpl.gni
@@ -337,6 +337,7 @@ template("chrome_common_apk_or_module_tmpl") {
 
       proguard_enabled = true
       proguard_configs = [ "//chrome/android/proguard/main.flags" ]
+      proguard_configs += [ "//mises/android/java/proguard.flags" ]
       if (_is_monochrome) {
         proguard_configs +=
             [ "//android_webview/nonembedded/java/proguard.flags" ]
@@ -748,6 +749,7 @@ template("chrome_common_apk_or_module_tmpl") {
                                  "version_code",
                                  "version_name",
                                ])
+    deps += mises_public_apk_or_module_tmpl_deps loadable_modules += mises_public_apk_or_module_tmpl_loadable_modules
   }
 }
 
@@ -776,6 +778,7 @@ template("monochrome_public_common_apk_or_module_tmpl") {
     if (!defined(is_monochrome)) {
       is_trichrome = true
     }
+    deps += mises_monochrome_public_apk_or_module_tmpl_deps loadable_modules = mises_monochrome_public_apk_or_module_tmpl_loadable_modules secondary_abi_loadable_modules = mises_monochrome_public_apk_or_module_tmpl_secondary_abi_loadable_modules
   }
 }
 template("chrome_public_common_apk_or_module_tmpl") {
