diff --git a/chrome/browser/resources/settings/BUILD.gn b/chrome/browser/resources/settings/BUILD.gn
index 16e6d00fd3b5d2588164bf26e8aff462cabaced4..dfe357cc56dfbb494cd4b0547a82fb9e138b9a10 100644
--- a/chrome/browser/resources/settings/BUILD.gn
+++ b/chrome/browser/resources/settings/BUILD.gn
@@ -10,7 +10,7 @@ import("//third_party/closure_compiler/compile_js.gni")
 import("//ui/webui/webui_features.gni")
 import("./settings.gni")
 
-assert(!is_android, "Android does not use these settings")
+#assert(!is_android, "Android does not use these settings")
 
 if (is_chromeos_ash) {
   group("closure_compile") {
