diff --git a/chrome/common/google_url_loader_throttle.cc b/chrome/common/google_url_loader_throttle.cc
index 546ec3a4dcd104390d9f863150bb13b8efd30376..4f5d8e09713d6571ede6c0c6e96882cec5b94f0b 100644
--- a/chrome/common/google_url_loader_throttle.cc
+++ b/chrome/common/google_url_loader_throttle.cc
@@ -147,8 +147,8 @@ void GoogleURLLoaderThrottle::WillProcessResponse(
     // relying on implicit ordering between this check and the time at which
     // ParsedHeaders is created.
     CHECK(response_head);
-    CHECK(response_head->parsed_headers);
-    if (response_head->parsed_headers->xfo !=
+    //CHECK(response_head->parsed_headers);
+    if (response_head->parsed_headers && response_head->parsed_headers->xfo !=
         network::mojom::XFrameOptionsValue::kDeny) {
       response_head->headers->SetHeader("X-Frame-Options", "SAMEORIGIN");
       response_head->parsed_headers->xfo =
