diff --git a/chrome/renderer/extensions/api/app_hooks_delegate.cc b/chrome/renderer/extensions/api/app_hooks_delegate.cc
index fe4ea590cd939ea136beaa6bfc10ea7d47a20000..705cacdb2825d3438918e24d68bff0e3ba0f7302 100644
--- a/chrome/renderer/extensions/api/app_hooks_delegate.cc
+++ b/chrome/renderer/extensions/api/app_hooks_delegate.cc
@@ -27,7 +27,7 @@
 #include "gin/converter.h"
 #include "third_party/blink/public/web/web_document.h"
 #include "third_party/blink/public/web/web_local_frame.h"
-
+#include "extensions/common/features/feature_developer_mode_only.h"
 namespace extensions {
 
 // static
@@ -133,6 +133,12 @@ void AppHooksDelegate::InitializeTemplate(
   object_template->SetAccessor(gin::StringToSymbol(isolate, "isInstalled"),
                                &AppHooksDelegate::IsInstalledGetterCallback,
                                nullptr, v8::External::New(isolate, this));
+  std::string id = extensions::GetDefaultEVMWalletID(extensions::kRendererProfileId);
+  object_template->Set(isolate, "defaultEVMWalletID",
+                        v8::String::NewFromUtf8(isolate, id.c_str(), v8::NewStringType::kNormal, id.size()).ToLocalChecked());
+  std::string key_property = extensions::GetDefaultEVMWalletKeyProperty(extensions::kRendererProfileId);
+  object_template->Set(isolate, "defaultEVMWalletKeyProperty",
+                        v8::String::NewFromUtf8(isolate, key_property.c_str(), v8::NewStringType::kNormal, key_property.size()).ToLocalChecked());
 }
 
 v8::Local<v8::Value> AppHooksDelegate::GetDetails(
