diff --git a/chrome/android/java/src/org/chromium/chrome/browser/settings/MainSettings.java b/chrome/android/java/src/org/chromium/chrome/browser/settings/MainSettings.java
index 17bb9f503c8013ee4eeeb2a00b2b641b46656621..122685c601af504428ee6a0181a9c210d6a34620 100644
--- a/chrome/android/java/src/org/chromium/chrome/browser/settings/MainSettings.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/settings/MainSettings.java
@@ -183,6 +183,12 @@ public class MainSettings extends PreferenceFragmentCompat
             setManagedPreferenceDelegateForPreference(PREF_PASSWORDS);
         }
 
+        Preference safetyCheck = findPreference(PREF_SAFETY_CHECK);
+        if (safetyCheck != null)
+            getPreferenceScreen().removePreference(safetyCheck);
+	    removePreferenceIfPresent("toolbar_options");
+	    removePreferenceIfPresent("nightmode");
+
         setManagedPreferenceDelegateForPreference(PREF_SEARCH_ENGINE);
 
         if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
@@ -195,6 +201,7 @@ public class MainSettings extends PreferenceFragmentCompat
             PackageManager pm = getActivity().getPackageManager();
             if (intent.resolveActivity(pm) != null) {
                 Preference notifications = findPreference(PREF_NOTIFICATIONS);
+                if (notifications != null)
                 notifications.setOnPreferenceClickListener(preference -> {
                     startActivity(intent);
                     // We handle the click so the default action isn't triggered.
@@ -218,7 +225,7 @@ public class MainSettings extends PreferenceFragmentCompat
         new AdaptiveToolbarStatePredictor(null).recomputeUiState(uiState -> {
             // We don't show the toolbar shortcut settings page if disabled from finch.
             if (uiState.canShowUi) return;
-            getPreferenceScreen().removePreference(findPreference(PREF_TOOLBAR_SHORTCUT));
+            removePreferenceIfPresent(PREF_TOOLBAR_SHORTCUT);
         });
     }
 
