diff --git a/chrome/browser/extensions/api/side_panel/side_panel_service.cc b/chrome/browser/extensions/api/side_panel/side_panel_service.cc
index c44bbe14edc005ad7c9f8fe5a6be212c0777cd0d..11271d8ef68ac5e2aeb967f9f38cb3ec848eaff0 100644
--- a/chrome/browser/extensions/api/side_panel/side_panel_service.cc
+++ b/chrome/browser/extensions/api/side_panel/side_panel_service.cc
@@ -19,6 +19,7 @@
 #include "extensions/browser/extension_prefs.h"
 #include "extensions/browser/pref_types.h"
 #include "extensions/common/extension_features.h"
+#include "build/build_config.h"
 
 namespace extensions {
 
@@ -250,6 +251,13 @@ base::expected<bool, std::string> SidePanelService::OpenSidePanelForTab(
     int tab_id,
     std::optional<int> window_id,
     bool include_incognito_information) {
+#if BUILDFLAG(IS_ANDROID) || BUILDFLAG(IS_IOS)
+  bool mises_enforced = true;
+  if (mises_enforced) {
+    return base::unexpected(
+        "OpenSidePanelForTab not supported in mobile version.");
+  }
+#endif
   // First, find the corresponding tab.
   Browser* browser = nullptr;
   content::WebContents* web_contents = nullptr;
