diff --git a/chrome/browser/BUILD.gn b/chrome/browser/BUILD.gn
index 42712ea4134e177836ed79026fa16114356fe850..70e54ae906b9e2dd0a146b6c6545a00899bfaa23 100644
--- a/chrome/browser/BUILD.gn
+++ b/chrome/browser/BUILD.gn
@@ -2536,6 +2536,7 @@ static_library("browser") {
       "performance_monitor/metric_evaluator_helper_posix.h",
     ]
   }
+  import("//mises/browser/sources.gni") sources += mises_chrome_browser_sources deps += mises_chrome_browser_deps public_deps += mises_chrome_browser_public_deps
 
   if (is_android) {
     sources += [
@@ -3479,7 +3480,8 @@ static_library("browser") {
       ]
       deps += [ "//chrome/android/modules/dev_ui/provider:native" ]
     }
-  } else {
+  }
+  if (true) {
     #!is_android
     sources += [
       "accessibility/caption_bubble_context_browser.h",
@@ -4475,7 +4477,7 @@ static_library("browser") {
         "themes/theme_color_policy_handler.h",
       ]
     }
-    if (!is_chromeos_ash) {
+    if (is_android || !is_chromeos_ash) {
       sources += [
         "accessibility/soda_installer_impl.cc",
         "accessibility/soda_installer_impl.h",
@@ -4569,7 +4571,7 @@ static_library("browser") {
 
     # On Windows, the hashes are embedded in //chrome:chrome_initial rather
     # than here in :chrome_dll.
-    if (!is_win) {
+    if (!is_android && !is_win) {
       deps += [ "//chrome:packed_resources_integrity" ]
       sources += [ "certificate_viewer_stub.cc" ]
     }
@@ -4610,6 +4612,9 @@ static_library("browser") {
         "feedback/system_logs/log_sources/chrome_root_store_log_source.h",
       ]
     }
+    sources -= mises_chrome_browser_exception_sources
+    deps -= mises_chrome_browser_exception_deps
+    allow_circular_includes_from -= mises_chrome_browser_exception_allow_circular_includes_from
   }
 
   if (is_chromeos_ash) {
@@ -6437,7 +6442,7 @@ static_library("browser") {
     ]
   }
 
-  if (toolkit_views) {
+  if (true || toolkit_views) {
     sources += [
       "page_load_metrics/observers/side_search_page_load_metrics_observer.cc",
       "page_load_metrics/observers/side_search_page_load_metrics_observer.h",
@@ -7282,7 +7287,7 @@ static_library("browser") {
     }
   }
 
-  if (enable_session_service) {
+  if (true || enable_session_service) {
     sources += [
       "sessions/app_session_service.cc",
       "sessions/app_session_service.h",
@@ -7456,7 +7461,7 @@ static_library("browser") {
     ]
   }
   if (enable_supervised_users && enable_extensions) {
-    assert(is_chromeos)
+    #assert(is_chromeos)
     sources += [
       "supervised_user/supervised_user_extensions_delegate_impl.cc",
       "supervised_user/supervised_user_extensions_delegate_impl.h",
@@ -7726,7 +7731,7 @@ grit("resources") {
       "//chrome/browser/resources/video_tutorials:build_ts",
     ]
   }
-  if (!is_android) {
+  if (true || !is_android) {
     deps += [
       "//chrome/browser/resources/media_router/internals:build_ts",
       "//chrome/browser/share/core/resources:make_share_targets_protobuf",
