diff --git a/chrome/browser/resources/tab_search/BUILD.gn b/chrome/browser/resources/tab_search/BUILD.gn
index 7585e08cf5bbb10703cecb9eca8e9e92b5bc0ca1..c82da6ebb9969a0dee7517027015ca0478e8925a 100644
--- a/chrome/browser/resources/tab_search/BUILD.gn
+++ b/chrome/browser/resources/tab_search/BUILD.gn
@@ -7,7 +7,7 @@ import("//ui/webui/resources/tools/build_webui.gni")
 import("//ui/webui/resources/tools/generate_grd.gni")
 import("//ui/webui/webui_features.gni")
 
-assert(is_linux || is_chromeos || is_win || is_mac)
+# assert(is_linux || is_chromeos || is_win || is_mac)
 
 build_webui("build") {
   grd_prefix = "tab_search"
