diff --git a/chrome/android/java/src/org/chromium/chrome/browser/tabmodel/ChromeTabCreator.java b/chrome/android/java/src/org/chromium/chrome/browser/tabmodel/ChromeTabCreator.java
index fcd7a43635e1ca5262379701fe37fd271f1d42e3..95ac962654617bcf0c7f2cef66b125429413284d 100644
--- a/chrome/android/java/src/org/chromium/chrome/browser/tabmodel/ChromeTabCreator.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/tabmodel/ChromeTabCreator.java
@@ -208,6 +208,7 @@ public class ChromeTabCreator extends TabCreator {
                         NewTabPageUtils.decodeOriginFromNtpUrl(loadUrlParams.getUrl()))) {
             return null;
         }
+        setDurationScale(0.0f);
         // Measure tab creation duration for different launch types to understand tab creation
         // performance.
         try (TraceEvent te = TraceEvent.scoped("ChromeTabCreator.createNewTab");
@@ -307,6 +308,7 @@ public class ChromeTabCreator extends TabCreator {
                 creationState = TabCreationState.LIVE_IN_BACKGROUND;
             }
             mTabModel.addTab(tab, position, type, creationState);
+            setDurationScale(0.6f);
             return tab;
         }
     }
@@ -317,6 +319,7 @@ public class ChromeTabCreator extends TabCreator {
         // The parent tab was already closed.  Do not open child tabs.
         int parentId = parent != null ? parent.getId() : Tab.INVALID_TAB_ID;
         if (mTabModel.isClosurePending(parentId)) return false;
+        setDurationScale(0.0f);
 
         // Measure tab creation duration for different launch types to understand tab creation
         // performance using an existing WebContents.
@@ -347,6 +350,7 @@ public class ChromeTabCreator extends TabCreator {
                                                      ? TabCreationState.FROZEN_FOR_LAZY_LOAD
                                                      : TabCreationState.LIVE_IN_BACKGROUND);
             mTabModel.addTab(tab, position, type, creationState);
+            setDurationScale(0.6f);
             return true;
         }
     }
@@ -567,4 +571,29 @@ public class ChromeTabCreator extends TabCreator {
     public void setWindowAndroid(WindowAndroid window) {
         mNativeWindow = window;
     }
+
+    private void setDurationScale(float scale) {
+        try {
+            ValueAnimator.class.getMethod("setDurationScale", float.class).invoke(null, scale);
+        } catch (Throwable t) {
+
+        }
+    }
+    @Override
+    public void openSinglePage(String url) {
+        try {
+            TraceEvent.begin("TabCreator.openSinglePage");
+
+            for (int i=0; i<mTabModel.getCount(); i++) {
+                String pageurl = mTabModel.getTabAt(i).getUrl().getSpec();
+                if (pageurl.indexOf(url) != -1) {
+                    mTabModel.setIndex(i, TabSelectionType.FROM_USER, false);
+                    return;
+                }
+            }
+            launchUrl(url, TabLaunchType.FROM_CHROME_UI);
+        } finally {
+            TraceEvent.end("TabCreator.openSinglePage");
+        }
+    }
 }
