diff --git a/chrome/browser/resources/privacy_sandbox/BUILD.gn b/chrome/browser/resources/privacy_sandbox/BUILD.gn
index beb7c1028438b0cbdf572199449df69b988b8ebe..9a217b3ba290a8a5bc1d7fa826fd5832f315414a 100644
--- a/chrome/browser/resources/privacy_sandbox/BUILD.gn
+++ b/chrome/browser/resources/privacy_sandbox/BUILD.gn
@@ -5,7 +5,7 @@
 import("//chrome/common/features.gni")
 import("//ui/webui/resources/tools/build_webui.gni")
 
-assert(!is_android, "!is_android check failed")
+#assert(!is_android, "!is_android check failed")
 
 build_webui("build") {
   grd_prefix = "privacy_sandbox"
