diff --git a/chrome/browser/ui/webui/side_panel/bookmarks/BUILD.gn b/chrome/browser/ui/webui/side_panel/bookmarks/BUILD.gn
index 124b66d54af6d681041177f838544deef77a1702..d4d21cb3b0cc5c6e0828e02cd4894ed355955593 100644
--- a/chrome/browser/ui/webui/side_panel/bookmarks/BUILD.gn
+++ b/chrome/browser/ui/webui/side_panel/bookmarks/BUILD.gn
@@ -4,7 +4,7 @@
 
 import("//mojo/public/tools/bindings/mojom.gni")
 
-assert(!is_android)
+#assert(!is_android)
 
 mojom("mojo_bindings") {
   sources = [ "bookmarks.mojom" ]
