diff --git a/chrome/browser/tabmodel/android/java/src/org/chromium/chrome/browser/tabmodel/EmptyTabModelFilter.java b/chrome/browser/tabmodel/android/java/src/org/chromium/chrome/browser/tabmodel/EmptyTabModelFilter.java
index ff3efa07346c4d62d59624f2a082adc8cb2611bb..416084cc7d75325cbf188431c5e84366ddf68abf 100644
--- a/chrome/browser/tabmodel/android/java/src/org/chromium/chrome/browser/tabmodel/EmptyTabModelFilter.java
+++ b/chrome/browser/tabmodel/android/java/src/org/chromium/chrome/browser/tabmodel/EmptyTabModelFilter.java
@@ -45,6 +45,11 @@ public class EmptyTabModelFilter extends TabModelFilter {
         return getTabModel().index();
     }
 
+    @Override
+    public int getLastNonExtensionActiveIndex() {
+        return getTabModel().getLastNonExtensionActiveIndex();
+    }
+
     @Override
     public int getCount() {
         return getTabModel().getCount();
