diff --git a/chrome/browser/resources/on_device_internals/BUILD.gn b/chrome/browser/resources/on_device_internals/BUILD.gn
index 3ba37cba5800478ed4f3b59a20fc6dd77225d56d..2931f2e1aad7af237b3214e31df595fba00f8c85 100644
--- a/chrome/browser/resources/on_device_internals/BUILD.gn
+++ b/chrome/browser/resources/on_device_internals/BUILD.gn
@@ -4,7 +4,7 @@
 
 import("//ui/webui/resources/tools/build_webui.gni")
 
-assert(!is_android)
+#assert(!is_android)
 
 build_webui("build") {
   grd_prefix = "on_device_internals"
