diff --git a/chrome/browser/resources/BUILD.gn b/chrome/browser/resources/BUILD.gn
index cafe4607b5b9c2f08f0f8f5cb737525a32108f3d..2f6e99f5ed7decba2fc40cae46bb22ef8b1d2654 100644
--- a/chrome/browser/resources/BUILD.gn
+++ b/chrome/browser/resources/BUILD.gn
@@ -19,7 +19,7 @@ assert(!is_ios, "Chromium/iOS shouldn't use anything in //chrome")
 group("resources") {
   public_deps = [ "segmentation_internals:resources" ]
 
-  if (!is_android) {
+  if (true || !is_android) {
     public_deps += [
       ":component_extension_resources",
       ":dev_ui_paks",
@@ -99,6 +99,11 @@ group("resources") {
       "ntp4:apps_resources",
     ]
   }
+  if (is_android) {
+    public_deps -= [
+      ":dev_ui_paks",
+    ]
+  }
 
   if (is_linux || is_chromeos) {
     public_deps += [ "webui_js_error:resources" ]
@@ -146,7 +151,7 @@ if (is_chromeos_ash && enable_js_type_check) {
   }
 }
 
-if (!is_android) {
+if (true || !is_android) {
   grit("component_extension_resources") {
     source = "component_extension_resources.grd"
 
