diff --git a/chrome/android/java/src/org/chromium/chrome/browser/modaldialog/ChromeTabModalPresenter.java b/chrome/android/java/src/org/chromium/chrome/browser/modaldialog/ChromeTabModalPresenter.java
index a6cd8942f4ba6f486a4dea8968e1a170ad26236d..dc009985fe1d29426cfcb140cd8906c06529cc94 100644
--- a/chrome/android/java/src/org/chromium/chrome/browser/modaldialog/ChromeTabModalPresenter.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/modaldialog/ChromeTabModalPresenter.java
@@ -33,6 +33,7 @@ import org.chromium.components.webxr.XrDelegateProvider;
 import org.chromium.content_public.browser.WebContents;
 import org.chromium.ui.UiUtils;
 import org.chromium.ui.modelutil.PropertyModel;
+import org.chromium.base.ContextUtils;
 
 /**
  * This presenter creates tab modality by blocking interaction with select UI elements while a
@@ -285,6 +286,8 @@ public class ChromeTabModalPresenter extends TabModalPresenter
      */
     public static int getContainerTopMargin(
             Resources resources, BrowserControlsStateProvider provider) {
+        if (ContextUtils.getAppSharedPreferences().getBoolean("enable_bottom_toolbar", false))
+            return 0;
         int scrimVerticalMargin =
                 resources.getDimensionPixelSize(R.dimen.tab_modal_scrim_vertical_margin);
         return provider.getTopControlsHeight() - scrimVerticalMargin;
