diff --git a/chrome/browser/resources/side_panel/customize_chrome/BUILD.gn b/chrome/browser/resources/side_panel/customize_chrome/BUILD.gn
index 8022dd5334119c4942344d77f90a2fd6d316b0c7..5fc899887f50befd84b8666b9ce80fd23d90e613 100644
--- a/chrome/browser/resources/side_panel/customize_chrome/BUILD.gn
+++ b/chrome/browser/resources/side_panel/customize_chrome/BUILD.gn
@@ -4,7 +4,7 @@
 
 import("//ui/webui/resources/tools/build_webui.gni")
 
-assert(!is_android)
+# assert(!is_android)
 
 build_webui("build") {
   grd_prefix = "side_panel_customize_chrome"
