diff --git a/chrome/browser/extensions/api/tabs/windows_util.cc b/chrome/browser/extensions/api/tabs/windows_util.cc
index 3504a6f268e64709ff369ef11ff22be4c1cc0aff..ad7671702672111d0aba373fc7a3ee23476f3584 100644
--- a/chrome/browser/extensions/api/tabs/windows_util.cc
+++ b/chrome/browser/extensions/api/tabs/windows_util.cc
@@ -24,6 +24,8 @@
 #include "extensions/common/extension.h"
 #include "url/gurl.h"
 
+#include "chrome/browser/extensions/extension_tab_util.h"
+
 namespace windows_util {
 
 bool GetControllerFromWindowID(ExtensionFunction* function,
