diff --git a/chrome/browser/resources/new_tab_page/BUILD.gn b/chrome/browser/resources/new_tab_page/BUILD.gn
index db4708a51c522cffe34634ddc1f7c1a6a96cbc89..08d5dbe396f2b7f3e45af0b1035b05bf42792488 100644
--- a/chrome/browser/resources/new_tab_page/BUILD.gn
+++ b/chrome/browser/resources/new_tab_page/BUILD.gn
@@ -7,7 +7,7 @@ import("./new_tab_page.gni")
 
 import("//ui/webui/resources/tools/build_webui.gni")
 
-assert(!is_android)
+#assert(!is_android)
 
 grd_prefix = "new_tab_page"
 
