diff --git a/chrome/browser/resources/signin/BUILD.gn b/chrome/browser/resources/signin/BUILD.gn
index 8d466f1abee6dc522e65fb54115cca833f5b295f..e582de9dcff5f75602f3654ea30cbc186879bb34 100644
--- a/chrome/browser/resources/signin/BUILD.gn
+++ b/chrome/browser/resources/signin/BUILD.gn
@@ -7,7 +7,7 @@ import("//chrome/common/features.gni")
 import("//components/signin/features.gni")
 import("//ui/webui/resources/tools/build_webui.gni")
 
-assert(!is_android)
+#assert(!is_android)
 
 build_webui("build") {
   grd_prefix = "signin"
