diff --git a/chrome/browser/safe_browsing/cloud_content_scanning/deep_scanning_utils.cc b/chrome/browser/safe_browsing/cloud_content_scanning/deep_scanning_utils.cc
index e55744659641b71abd542e9fd8c708aeab313b49..4db2189e07c619fd7fbdc6a782f813f55de7443c 100644
--- a/chrome/browser/safe_browsing/cloud_content_scanning/deep_scanning_utils.cc
+++ b/chrome/browser/safe_browsing/cloud_content_scanning/deep_scanning_utils.cc
@@ -459,7 +459,11 @@ bool IsConsumerScanRequest(const BinaryUploadService::Request& request) {
       return false;
     }
   }
+#if BUILDFLAG(IS_ANDROID)
+  return false;
+#else
   return request.device_token().empty();
+#endif
 }
 
 }  // namespace safe_browsing
