diff --git a/chrome/browser/ui/web_applications/BUILD.gn b/chrome/browser/ui/web_applications/BUILD.gn
index 269bc69a2fda90355db9e8f22e27934d75d9a453..256d16bf6da2b2ca1a8863901e9c4e6b9d918971 100644
--- a/chrome/browser/ui/web_applications/BUILD.gn
+++ b/chrome/browser/ui/web_applications/BUILD.gn
@@ -126,6 +126,10 @@ source_set("browser_tests") {
   if (is_chromeos) {
     deps += [ "//chromeos/constants" ]
   }
+
+  if (is_android) {
+    deps -= ["//chrome/test:test_support_ui"]
+  }
 }
 
 # On Lacros, these browser tests require Ash to be running, as they rely on web
@@ -191,4 +195,7 @@ source_set("app_service_browser_tests") {
       "//chromeos/startup",
     ]
   }
+  if (is_android) {
+    deps -= ["//chrome/test:test_support_ui"]
+  }
 }
