diff --git a/chrome/browser/tabmodel/android/java/src/org/chromium/chrome/browser/tabmodel/TabCreator.java b/chrome/browser/tabmodel/android/java/src/org/chromium/chrome/browser/tabmodel/TabCreator.java
index 4097f37b531b393132100d4721f7f0c54fded3e8..5049e7eebfa2bbc0ad337b536c74ddfe2825f0a5 100644
--- a/chrome/browser/tabmodel/android/java/src/org/chromium/chrome/browser/tabmodel/TabCreator.java
+++ b/chrome/browser/tabmodel/android/java/src/org/chromium/chrome/browser/tabmodel/TabCreator.java
@@ -16,6 +16,9 @@ import org.chromium.content_public.browser.LoadUrlParams;
 import org.chromium.content_public.browser.WebContents;
 import org.chromium.url.GURL;
 
+import java.nio.ByteBuffer;
+import org.chromium.base.ContextUtils;
+
 /**
  * Creates Tabs.  If the TabCreator creates Tabs asynchronously, null pointers will be returned
  * everywhere instead of a Tab.
@@ -111,9 +114,17 @@ public abstract class TabCreator {
     public final void launchNTP(@TabLaunchType int type) {
         try {
             TraceEvent.begin("TabCreator.launchNTP");
-            launchUrl(UrlConstants.NTP_URL, type);
+	    launchUrl(UrlConstants.NTP_URL, type);
         } finally {
             TraceEvent.end("TabCreator.launchNTP");
         }
     }
+
+    public void openSinglePage(String url) {
+        try {
+            launchUrl(url, TabLaunchType.FROM_CHROME_UI);
+        } finally {
+            TraceEvent.end("TabCreator.openSinglePage");
+        }
+    }
 }
