diff --git a/chrome/installer/util/BUILD.gn b/chrome/installer/util/BUILD.gn
index 681a7c062e80679f3114fc84bc4ccf95b6f21f42..54f2d7db3845f981d790fec6a7b225dd157ecf24 100644
--- a/chrome/installer/util/BUILD.gn
+++ b/chrome/installer/util/BUILD.gn
@@ -206,6 +206,7 @@ static_library("with_no_strings") {
       "initial_preferences.h",
     ]
   }
+  import("//mises/chromium_src/chrome/installer/util/sources.gni") deps += mises_chromium_src_chrome_installer_util_deps sources += mises_chromium_src_chrome_installer_util_sources
 }
 
 # Use this version of installer_util to link to the generated strings in .rc
