diff --git a/chrome/common/accessibility/BUILD.gn b/chrome/common/accessibility/BUILD.gn
index d18c1271e418306c26aa4d4258613897c41f9896..2263a76a86c25ef8167382cb517d045efc72d2ed 100644
--- a/chrome/common/accessibility/BUILD.gn
+++ b/chrome/common/accessibility/BUILD.gn
@@ -5,7 +5,7 @@
 import("//components/services/screen_ai/buildflags/features.gni")
 import("//mojo/public/tools/bindings/mojom.gni")
 
-assert(!is_android)
+#assert(!is_android)
 
 mojom("mojo_bindings") {
   sources = [ "read_anything.mojom" ]
