diff --git a/chrome/common/extensions/api/api_sources.gni b/chrome/common/extensions/api/api_sources.gni
index cd69077aacd08ac6f0251fbbfd1db7a2f2fee291..6cacd5c84de3267e7540259843c249ce45f53196 100644
--- a/chrome/common/extensions/api/api_sources.gni
+++ b/chrome/common/extensions/api/api_sources.gni
@@ -79,7 +79,7 @@ if (enable_extensions) {
     "windows.json",
   ]
 
-  if (!is_android) {
+  if (true || !is_android) {
     schema_sources_ += [ "processes.idl" ]
   }
 
