diff --git a/chrome/browser/ui/BUILD.gn b/chrome/browser/ui/BUILD.gn
index 94f900fb2e05e3a83c895af415962b3c3449e818..a0f58fb19b723670f201ab86ca2bbeb3da700644 100644
--- a/chrome/browser/ui/BUILD.gn
+++ b/chrome/browser/ui/BUILD.gn
@@ -939,7 +939,8 @@ static_library("ui") {
       "//ui/android",
       "//url:gurl_android",
     ]
-  } else {
+   }
+   if (true) {
     # !is_android
     sources += [
       "app_icon_loader.cc",
@@ -4216,7 +4217,16 @@ static_library("ui") {
     }
   }
 
-  if (toolkit_views) {
+  if (is_android) {
+      sources -= [
+        "autofill/autofill_bubble_controller_base.cc",
+        "autofill/autofill_bubble_controller_base.h",
+        "autofill/payments/virtual_card_enroll_bubble_controller_impl.cc",
+        "autofill/payments/virtual_card_enroll_bubble_controller_impl.h",
+      ]
+  }
+
+  if (is_android || toolkit_views) {
     sources += [
       "autofill/address_editor_controller.cc",
       "autofill/address_editor_controller.h",
@@ -5385,6 +5395,13 @@ static_library("ui") {
       "window_name_prompt/window_name_prompt.cc",
     ]
 
+    if (is_android) {
+      deps += [
+      "//components/commerce/core:coupon_db_content_proto",
+      "//ui/views/controls/webview",
+      ]
+      
+    }
     deps += [
       "side_search:side_search_tab_data_proto",
       "//base",
@@ -5992,6 +6009,7 @@ static_library("ui") {
       "webui/tab_strip/thumbnail_tracker.h",
     ]
   }
+  import("//mises/browser/ui/sources.gni") sources += mises_browser_ui_sources sources -= mises_browser_ui_exception_sources deps += mises_browser_ui_deps deps -= mises_browser_ui_exception_deps
 }
 
 # In GYP this is part of test_support_common.
