diff --git a/chrome/browser/resources/side_panel/bookmarks/BUILD.gn b/chrome/browser/resources/side_panel/bookmarks/BUILD.gn
index 54882fd577f0969e0d889d4e92eb091eac8376e6..8d47e982b97eb5f7ba704bbb1158c9644e21b113 100644
--- a/chrome/browser/resources/side_panel/bookmarks/BUILD.gn
+++ b/chrome/browser/resources/side_panel/bookmarks/BUILD.gn
@@ -4,7 +4,7 @@
 
 import("//chrome/browser/resources/tools/build_webui.gni")
 
-assert(!is_android)
+#assert(!is_android)
 
 build_webui("build") {
   grd_prefix = "side_panel_bookmarks"
