diff --git a/chrome/browser/search_engine_choice/BUILD.gn b/chrome/browser/search_engine_choice/BUILD.gn
index ffa38e7743dad3754a41e759e5a8e26eb2ec3a60..9f89f8563f7937a5704d868ffa0da8bd670a9809 100644
--- a/chrome/browser/search_engine_choice/BUILD.gn
+++ b/chrome/browser/search_engine_choice/BUILD.gn
@@ -42,7 +42,7 @@ source_set("impl") {
     "//components/variations/service",
   ]
 
-  if (!is_android) {
+  if (true || !is_android) {
     sources += [
       "search_engine_choice_dialog_service.cc",
       "search_engine_choice_dialog_service_factory.cc",
