diff --git a/chrome/browser/resources/side_panel/history_clusters/BUILD.gn b/chrome/browser/resources/side_panel/history_clusters/BUILD.gn
index 3d1dccd925b3e23101e54496e8413a3c57b08501..45c19467d8be52361adafe558256e606f305e5a0 100644
--- a/chrome/browser/resources/side_panel/history_clusters/BUILD.gn
+++ b/chrome/browser/resources/side_panel/history_clusters/BUILD.gn
@@ -4,7 +4,7 @@
 
 import("//ui/webui/resources/tools/build_webui.gni")
 
-assert(!is_android)
+#assert(!is_android)
 
 build_webui("build") {
   grd_prefix = "side_panel_history_clusters"
