diff --git a/chrome/browser/resources/browser_switch/BUILD.gn b/chrome/browser/resources/browser_switch/BUILD.gn
index 28286c25300a56c43f781b3c5c0a7ce647c4cbee..09ef86b542759843a1c0d6b46648cf87c7ad86ce 100644
--- a/chrome/browser/resources/browser_switch/BUILD.gn
+++ b/chrome/browser/resources/browser_switch/BUILD.gn
@@ -5,7 +5,7 @@
 import("//chrome/browser/resources/tools/build_webui.gni")
 import("//chrome/common/features.gni")
 
-assert(is_win || is_mac || is_linux || is_chromeos_lacros || is_fuchsia)
+#assert(is_win || is_mac || is_linux || is_chromeos_lacros || is_fuchsia)
 
 build_webui("build") {
   grd_prefix = "browser_switch"
