diff --git a/chrome/browser/image_editor/BUILD.gn b/chrome/browser/image_editor/BUILD.gn
index 3acbdae3dd12e4aba644d74f718fd32541d4f838..eb32837368bccd175765346d8ad6114f5d4b7a72 100644
--- a/chrome/browser/image_editor/BUILD.gn
+++ b/chrome/browser/image_editor/BUILD.gn
@@ -3,8 +3,8 @@
 # found in the LICENSE file.
 import("//build/config/ui.gni")
 
-assert(is_chromeos || is_linux || is_mac || is_win || is_fuchsia,
-       "Image Editor is for desktop plaforms only")
+#assert(is_chromeos || is_linux || is_mac || is_win || is_fuchsia,
+#       "Image Editor is for desktop plaforms only")
 
 static_library("image_editor_component_util") {
   sources = [
