diff --git a/chrome/browser/ui/webui/access_code_cast/BUILD.gn b/chrome/browser/ui/webui/access_code_cast/BUILD.gn
index 89cd7af99b26ba3dde40ec2c33ff1ad049c37a3c..b6fd04725e1598e0503f53bdf827a2745b6d9b42 100644
--- a/chrome/browser/ui/webui/access_code_cast/BUILD.gn
+++ b/chrome/browser/ui/webui/access_code_cast/BUILD.gn
@@ -5,7 +5,7 @@
 import("//mojo/public/tools/bindings/mojom.gni")
 import("//third_party/protobuf/proto_library.gni")
 
-assert(!is_android, "AccessCodeCast is not for android.")
+#assert(!is_android, "AccessCodeCast is not for android.")
 
 mojom("mojo_bindings") {
   sources = [ "access_code_cast.mojom" ]
