diff --git a/chrome/browser/ui/webui/top_chrome/BUILD.gn b/chrome/browser/ui/webui/top_chrome/BUILD.gn
index 97a83d8f5be2da93c8f6272a5498d30f84e5977b..c4d9e34bebecf02dc04bf85eee70f9dbd4faa747 100644
--- a/chrome/browser/ui/webui/top_chrome/BUILD.gn
+++ b/chrome/browser/ui/webui/top_chrome/BUILD.gn
@@ -2,7 +2,7 @@
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
-assert(!is_android)
+#assert(!is_android)
 
 source_set("top_chrome") {
   sources = [
