diff --git a/chrome/browser/web_applications/BUILD.gn b/chrome/browser/web_applications/BUILD.gn
index caed81a2a79294a4b129720103bba54142956ce9..61368e1273558b40862b0a0fd6188c55a74ede25 100644
--- a/chrome/browser/web_applications/BUILD.gn
+++ b/chrome/browser/web_applications/BUILD.gn
@@ -395,6 +395,8 @@ source_set("web_applications") {
     ]
   }
 
+  import("//mises/browser/web_applications/sources.gni") sources += mises_browser_web_applications_sources
+
   if (is_mac) {
     sources += [
       "os_integration/mac/app_shim_launch.h",
@@ -1123,6 +1125,9 @@ source_set("web_applications_browser_tests") {
       "//chromeos/startup",
     ]
   }
+  if (is_android) {
+    deps -= ["//chrome/test:test_support_ui"]
+  }
 }
 
 group("browser_tests") {
@@ -1174,6 +1179,9 @@ source_set("app_service_browser_tests") {
   if (is_chromeos_ash) {
     deps += [ "//chrome/browser/ui/ash/shelf" ]
   }
+  if (is_android) {
+    deps -= ["//chrome/test:test_support_ui"]
+  }
 }
 
 source_set("interactive_ui_tests") {
@@ -1199,6 +1207,9 @@ source_set("interactive_ui_tests") {
     "//ui/base/idle:test_support",
     "//ui/events/devices:test_support",
   ]
+  if (is_android) {
+    deps -= ["//chrome/test:test_support_ui"]
+  }
 
   if (is_mac) {
     deps += [ "//chrome/browser/apps/app_shim" ]
