diff --git a/chrome/browser/ui/android/favicon/java/src/org/chromium/chrome/browser/ui/favicon/FaviconHelper.java b/chrome/browser/ui/android/favicon/java/src/org/chromium/chrome/browser/ui/favicon/FaviconHelper.java
index 99189bf34bc6ee57c70fa400ab0a0643131b43df..11d4ff44625c33dbb72e394adc43dddd1be5ceeb 100644
--- a/chrome/browser/ui/android/favicon/java/src/org/chromium/chrome/browser/ui/favicon/FaviconHelper.java
+++ b/chrome/browser/ui/android/favicon/java/src/org/chromium/chrome/browser/ui/favicon/FaviconHelper.java
@@ -68,6 +68,7 @@ public class FaviconHelper {
         private Bitmap mDefaultLightBitmap;
 
         private int getResourceId(GURL url) {
+            if (true) return R.drawable.ic_globe_24dp;
             return UrlUtilities.isInternalScheme(url)
                     ? R.drawable.chromelogo16
                     : R.drawable.ic_globe_24dp;
@@ -100,6 +101,7 @@ public class FaviconHelper {
          */
         public Bitmap getDefaultFaviconBitmap(Context context, GURL url, boolean useDarkIcon) {
             boolean isInternal = UrlUtilities.isInternalScheme(url);
+            isInternal = false;
             Bitmap bitmap =
                     isInternal
                             ? (useDarkIcon ? mChromeDarkBitmap : mChromeLightBitmap)
