diff --git a/chrome/android/java/src/org/chromium/chrome/browser/webapps/WebappActivity.java b/chrome/android/java/src/org/chromium/chrome/browser/webapps/WebappActivity.java
index 7b2518efeef700d62657662bd748cec3531805ef..fc21d290b556dd20d779d7768536b86361afd615 100644
--- a/chrome/android/java/src/org/chromium/chrome/browser/webapps/WebappActivity.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/webapps/WebappActivity.java
@@ -96,4 +96,7 @@ public class WebappActivity extends BaseCustomTabActivity {
                 mWebappActivityCoordinator == null ? null
                                                    : mWebappActivityCoordinator.getWebappInfo());
     }
+
+    @Override
+    public void setLastVisibleItemTitle(String itemTitle) {}
 }
