diff --git a/chrome/android/chrome_common_shared_library.gni b/chrome/android/chrome_common_shared_library.gni
index f23200549c9b57f1b45ed4268c6705657d57ae7b..4ca403ea928f8978c4a0945aa0623aca54d6a8c9 100644
--- a/chrome/android/chrome_common_shared_library.gni
+++ b/chrome/android/chrome_common_shared_library.gni
@@ -119,6 +119,10 @@ template("chrome_common_shared_library") {
     }
 
     # Handle VR JNI registration and dependencies.
+    if (_is_webview) {
+      sources = [ "//chrome/browser/android/vr/register_jni_monochrome.cc" ]
+      deps += [ "//chrome/browser/android/vr:module_factory" ]
+    }
     if (!_is_webview && enable_vr) {
       if (_export_java_symbols) {
         # NOTE: While this file is named *_monochrome.cc, it just contains an
