diff --git a/chrome/browser/ui/webui/searchbox/BUILD.gn b/chrome/browser/ui/webui/searchbox/BUILD.gn
index 4ccb3686f288ed86e6c35b8d6390d6a67aaa1021..b60a2bd4b117ff1373396ac3667d500850828769 100644
--- a/chrome/browser/ui/webui/searchbox/BUILD.gn
+++ b/chrome/browser/ui/webui/searchbox/BUILD.gn
@@ -2,7 +2,7 @@
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
-assert(is_win || is_mac || is_linux || is_chromeos)
+#assert(is_win || is_mac || is_linux || is_chromeos)
 
 source_set("searchbox") {
   sources = [
