diff --git a/chrome/browser/tabmodel/android/java/src/org/chromium/chrome/browser/tabmodel/TabCreator.java b/chrome/browser/tabmodel/android/java/src/org/chromium/chrome/browser/tabmodel/TabCreator.java
index d66ed8aebc7313f5a3ebe5845d217cdf77290454..9b883d8c59cf981c6138b1fe7844480ffbeb4596 100644
--- a/chrome/browser/tabmodel/android/java/src/org/chromium/chrome/browser/tabmodel/TabCreator.java
+++ b/chrome/browser/tabmodel/android/java/src/org/chromium/chrome/browser/tabmodel/TabCreator.java
@@ -17,6 +17,9 @@ import org.chromium.content_public.browser.LoadUrlParams;
 import org.chromium.content_public.browser.WebContents;
 import org.chromium.url.GURL;
 
+import java.nio.ByteBuffer;
+import org.chromium.base.ContextUtils;
+
 /**
  * Creates Tabs.  If the TabCreator creates Tabs asynchronously, null pointers will be returned
  * everywhere instead of a Tab.
@@ -113,9 +116,26 @@ public abstract class TabCreator {
     public final void launchNTP(@TabLaunchType int type) {
         try {
             TraceEvent.begin("TabCreator.launchNTP");
-            launchUrl(UrlConstants.NTP_URL, type);
+	    launchUrl(UrlConstants.NTP_URL, type);
         } finally {
             TraceEvent.end("TabCreator.launchNTP");
         }
     }
+
+    public void openSinglePage(String url) {
+        try {
+            launchUrl(url, TabLaunchType.FROM_CHROME_UI);
+        } finally {
+            TraceEvent.end("TabCreator.openSinglePage");
+        }
+    }
+
+    public final void launchIncognitoNTP() {
+        try {
+            TraceEvent.begin("TabCreator.launchIncognitoNTP");
+            launchUrl("chrome-search://local-ntp/incognito-ntp.html", TabLaunchType.FROM_CHROME_UI);
+         } finally {
+             TraceEvent.end("TabCreator.launchIncognitoNTP");
+         }
+    }
 }
