diff --git a/chrome/browser/extensions/api/passwords_private/password_check_delegate.cc b/chrome/browser/extensions/api/passwords_private/password_check_delegate.cc
index 6bdc83a1a7b1ca3ba755eeacad9b1e55f7fa229c..44a12fc54f9ac53d331a1c47775e176026421b92 100644
--- a/chrome/browser/extensions/api/passwords_private/password_check_delegate.cc
+++ b/chrome/browser/extensions/api/passwords_private/password_check_delegate.cc
@@ -344,12 +344,14 @@ void PasswordCheckDelegate::StartPasswordAnalyses(
   insecure_credentials_manager_.StartWeakCheck(base::BindOnce(
       &PasswordCheckDelegate::RecordAndNotifyAboutCompletedWeakPasswordCheck,
       weak_ptr_factory_.GetWeakPtr()));
+#if !BUILDFLAG(IS_ANDROID)
   if (base::FeatureList::IsEnabled(
           password_manager::features::kPasswordManagerRedesign)) {
     insecure_credentials_manager_.StartReuseCheck(
         base::BindOnce(&PasswordCheckDelegate::NotifyPasswordCheckStatusChanged,
                        weak_ptr_factory_.GetWeakPtr()));
   }
+#endif
   auto progress = base::MakeRefCounted<PasswordCheckProgress>();
   for (const auto& password : saved_passwords_presenter_->GetSavedPasswords())
     progress->IncrementCounts(password);
