diff --git a/chrome/browser/ui/android/toolbar/java/src/org/chromium/chrome/browser/toolbar/menu_button/MenuButtonCoordinator.java b/chrome/browser/ui/android/toolbar/java/src/org/chromium/chrome/browser/toolbar/menu_button/MenuButtonCoordinator.java
index ba0d02ba3dbb645cb67f93c35d77ba901ffa57d4..a0a9ad1af697b77233341ef92091e0f30846159c 100644
--- a/chrome/browser/ui/android/toolbar/java/src/org/chromium/chrome/browser/toolbar/menu_button/MenuButtonCoordinator.java
+++ b/chrome/browser/ui/android/toolbar/java/src/org/chromium/chrome/browser/toolbar/menu_button/MenuButtonCoordinator.java
@@ -247,4 +247,7 @@ public class MenuButtonCoordinator {
     public boolean isShowingUpdateBadge() {
         return mPropertyModel.get(MenuButtonProperties.SHOW_UPDATE_BADGE).mShowUpdateBadge;
     }
+    public AppMenuButtonHelper getHelper() {
+	return mPropertyModel.get(MenuButtonProperties.APP_MENU_BUTTON_HELPER);
+    }
 }
