diff --git a/chrome/browser/offline_pages/offline_page_request_handler.cc b/chrome/browser/offline_pages/offline_page_request_handler.cc
index effa889e1a0088ace4056542061589322b9fa3d1..f174f25f1004a82103b36750c2dd59e69493e23d 100644
--- a/chrome/browser/offline_pages/offline_page_request_handler.cc
+++ b/chrome/browser/offline_pages/offline_page_request_handler.cc
@@ -507,6 +507,9 @@ void OfflinePageRequestHandler::OpenFile(
 
   int flags = base::File::FLAG_OPEN | base::File::FLAG_READ |
               base::File::FLAG_ASYNC | base::File::FLAG_WIN_EXCLUSIVE_READ;
+#if BUILDFLAG(IS_ANDROID)
+  flags &= ~base::File::FLAG_WIN_EXCLUSIVE_READ;
+#endif
   int result = stream_->Open(file_path, flags, callback);
   if (result != net::ERR_IO_PENDING)
     callback.Run(result);
