diff --git a/chrome/browser/resources/webui_gallery/BUILD.gn b/chrome/browser/resources/webui_gallery/BUILD.gn
index fdb7cf6da64a6a00a86bb0d09a51e977354d3d12..1c520123bdda8a004c7b2a1ebecd6669f848c1eb 100644
--- a/chrome/browser/resources/webui_gallery/BUILD.gn
+++ b/chrome/browser/resources/webui_gallery/BUILD.gn
@@ -4,7 +4,7 @@
 
 import("//ui/webui/resources/tools/build_webui.gni")
 
-assert(!is_android)
+# assert(!is_android)
 
 build_webui("build") {
   grd_prefix = "webui_gallery"
