diff --git a/chrome/common/extensions/api/api_sources.gni b/chrome/common/extensions/api/api_sources.gni
index 1e0f1779d08107ffa552df995403487638a165e9..2a151bf5c5f91812d76888f71129d294c50b16b6 100644
--- a/chrome/common/extensions/api/api_sources.gni
+++ b/chrome/common/extensions/api/api_sources.gni
@@ -69,7 +69,7 @@ schema_sources_ = [
   "windows.json",
 ]
 
-if (!is_android) {
+if (true || !is_android) {
   schema_sources_ += [ "processes.idl" ]
 }
 
@@ -156,7 +156,6 @@ if (is_chromeos_ash) {
 }
 
 chrome_extensions_api_schema_sources = get_path_info(schema_sources_, "abspath")
-
 chrome_extensions_manifest_only_schema_sources =
     get_path_info([
                     "manifest_types.json",
