diff --git a/chrome/browser/ui/android/toolbar/java/src/org/chromium/chrome/browser/toolbar/LocationBarModel.java b/chrome/browser/ui/android/toolbar/java/src/org/chromium/chrome/browser/toolbar/LocationBarModel.java
index f4670792368dcc9280a780a0602172609c5f3d2a..7247dba1d4f2f618c670ec2339121eac4b1b4d47 100644
--- a/chrome/browser/ui/android/toolbar/java/src/org/chromium/chrome/browser/toolbar/LocationBarModel.java
+++ b/chrome/browser/ui/android/toolbar/java/src/org/chromium/chrome/browser/toolbar/LocationBarModel.java
@@ -386,6 +386,10 @@ public class LocationBarModel implements ToolbarDataProvider, LocationBarDataPro
                 return buildUrlBarData(originalUrl, isOfflinePage);
             }
 
+            if (url.startsWith("devtools://")) {
+                return buildUrlBarData(url, false, mTab.getTitle(), "");
+            }
+
             if (isOfflinePage) {
                 GURL originalUrl = mTab.getOriginalUrl();
                 formattedUrl = UrlUtilities.stripScheme(mUrlFormatter.format(originalUrl));
@@ -626,7 +630,7 @@ public class LocationBarModel implements ToolbarDataProvider, LocationBarDataPro
 
     @Override
     public int getPageClassification(boolean isFocusedFromFakebox, boolean isPrefetch) {
-        if (mNativeLocationBarModelAndroid == 0) return PageClassification.INVALID_SPEC_VALUE;
+        if (mNativeLocationBarModelAndroid == 0) return PageClassification.NTP_VALUE;
 
         // Provide NTP as page class in overview mode (when Start Surface is enabled). No call
         // to the backend necessary or possible, since there is no tab or navigation entry.
@@ -655,6 +659,16 @@ public class LocationBarModel implements ToolbarDataProvider, LocationBarDataPro
             return ConnectionSecurityLevel.NONE;
         }
 
+        final String url = tab.getUrl().getSpec();
+        boolean mises_force_safe = !TextUtils.isEmpty(url) && (
+            url.startsWith("chrome-extension") || 
+            url.startsWith("chrome") || 
+            url.startsWith("devtools") || 
+            url.startsWith("mises"));
+        if (mises_force_safe) {
+            return ConnectionSecurityLevel.NONE;
+        }
+
         @Nullable
         GURL publisherUrl = TrustedCdn.getPublisherUrl(tab);
 
@@ -689,6 +703,10 @@ public class LocationBarModel implements ToolbarDataProvider, LocationBarDataPro
             return R.drawable.ic_offline_pin_24dp;
         }
 
+        if (mNativeLocationBarModelAndroid == 0) {  
+             return R.drawable.omnibox_info;
+        }
+
         // Return early if native initialization hasn't been done yet.
         if ((securityLevel == ConnectionSecurityLevel.NONE
                     || securityLevel == ConnectionSecurityLevel.WARNING)
@@ -700,6 +718,16 @@ public class LocationBarModel implements ToolbarDataProvider, LocationBarDataPro
                 !mSearchEngineLogoUtils.shouldShowSearchEngineLogo(isIncognito())
                 || mNtpDelegate.isCurrentlyVisible() || isInOverviewAndShowingOmnibox();
 
+        String currentUrl = getCurrentUrl();
+        if (currentUrl != null && currentUrl.startsWith("chrome://"))
+            skipIconForNeutralState = true;
+        if (currentUrl != null && currentUrl.startsWith("mises://"))
+            skipIconForNeutralState = true;
+        if (currentUrl != null && currentUrl.startsWith("chrome-search://"))
+            skipIconForNeutralState = true;
+        if (currentUrl != null && currentUrl.startsWith("mises-search://"))
+            skipIconForNeutralState = true;
+
         boolean useUpdatedConnectionSecurityIndicators =
                 mOmniboxUpdatedConnectionSecurityIndicatorsEnabled
                 && !(hasTab() && mTab.isCustomTab());
