diff --git a/chrome/browser/extensions/extension_install_prompt.cc b/chrome/browser/extensions/extension_install_prompt.cc
index 58a265a2f2e4496af4d01677058406d9b4e77291..7b6856a6d5d8821c25454e7ee9a229eeb81fb6ce 100644
--- a/chrome/browser/extensions/extension_install_prompt.cc
+++ b/chrome/browser/extensions/extension_install_prompt.cc
@@ -440,6 +440,7 @@ ExtensionInstallPrompt::ExtensionInstallPrompt(content::WebContents* contents)
       extension_(nullptr),
       install_ui_(extensions::CreateExtensionInstallUI(profile_)),
       show_params_(new ExtensionInstallPromptShowParams(contents)),
+      contents_(contents),
       did_call_show_dialog_(false) {}
 
 ExtensionInstallPrompt::ExtensionInstallPrompt(Profile* profile,
@@ -597,6 +598,7 @@ void ExtensionInstallPrompt::ShowConfirmation() {
 
   // Notify observers.
   prompt_->OnDialogOpened();
+  MISES_EXTENSION_INSTALL_PROMPT_SHOW_CONFIRMATION
 
   // If true, auto confirm is enabled and already handled the result.
   if (AutoConfirmPromptIfEnabled())
