diff --git a/chrome/browser/resources/identity_internals/BUILD.gn b/chrome/browser/resources/identity_internals/BUILD.gn
index 729d461a5867398169680ba70a187fdaca6c274e..090d3fe97b4297b578f53779d10296da3d4db862 100644
--- a/chrome/browser/resources/identity_internals/BUILD.gn
+++ b/chrome/browser/resources/identity_internals/BUILD.gn
@@ -4,7 +4,7 @@
 
 import("//ui/webui/resources/tools/build_webui.gni")
 
-assert(!is_android)
+#assert(!is_android)
 
 build_webui("build") {
   grd_prefix = "identity_internals"
