diff --git a/chrome/browser/resources/side_panel/shared/BUILD.gn b/chrome/browser/resources/side_panel/shared/BUILD.gn
index 4bde9cbe0af954dc998e533ab071970609154dd0..9cdd7b46447f92b37ebe17b66ef1e7cddddcf2bf 100644
--- a/chrome/browser/resources/side_panel/shared/BUILD.gn
+++ b/chrome/browser/resources/side_panel/shared/BUILD.gn
@@ -4,7 +4,7 @@
 
 import("//ui/webui/resources/tools/build_webui.gni")
 
-assert(!is_android)
+#assert(!is_android)
 
 build_webui("build") {
   grd_prefix = "side_panel_shared"
