diff --git a/chrome/browser/resources/signin/BUILD.gn b/chrome/browser/resources/signin/BUILD.gn
index cde6595ba0df1bfc048ad6c41a2b0b29f64670a4..b5cb580add40bffe76dcd1322e466eec961423ad 100644
--- a/chrome/browser/resources/signin/BUILD.gn
+++ b/chrome/browser/resources/signin/BUILD.gn
@@ -7,7 +7,7 @@ import("//chrome/browser/resources/tools/build_webui.gni")
 import("//chrome/common/features.gni")
 import("//components/signin/features.gni")
 
-assert(!is_android)
+#assert(!is_android)
 
 build_webui("build") {
   grd_prefix = "signin"
