diff --git a/chrome/browser/background/background_contents_service.cc b/chrome/browser/background/background_contents_service.cc
index 4199417a65ab09e2b29a3939f9628f4ca722696b..777563fc74998356e541c0308010671c51457849 100644
--- a/chrome/browser/background/background_contents_service.cc
+++ b/chrome/browser/background/background_contents_service.cc
@@ -727,9 +727,13 @@ void BackgroundContentsService::HandleExtensionCrashed(
   if (!extension)
     return;
 
+#if BUILDFLAG(IS_ANDROID)
+  const bool force_installed = true;
+#else
   const bool force_installed =
       extensions::Manifest::IsComponentLocation(extension->location()) ||
       extensions::Manifest::IsPolicyLocation(extension->location());
+#endif
   if (!force_installed) {
     ShowBalloon(extension);
   } else {
