diff --git a/chrome/browser/ui/android/omnibox/java/src/org/chromium/chrome/browser/omnibox/status/StatusMediator.java b/chrome/browser/ui/android/omnibox/java/src/org/chromium/chrome/browser/omnibox/status/StatusMediator.java
index 4468f903aba749822db38d38e6c2d7778736072a..231fa84051416dd38c342b8fd93b839474b10e4c 100644
--- a/chrome/browser/ui/android/omnibox/java/src/org/chromium/chrome/browser/omnibox/status/StatusMediator.java
+++ b/chrome/browser/ui/android/omnibox/java/src/org/chromium/chrome/browser/omnibox/status/StatusMediator.java
@@ -564,8 +564,8 @@ public class StatusMediator
             return true;
         }
 
-        return (mUrlHasFocus || mUrlFocusPercent > 0)
-                && isNTPOrStartSurfaceVisible()
+        boolean mises_enforced = false;
+        return mises_enforced && (mUrlHasFocus || mUrlFocusPercent > 0) && isNTPOrStartSurfaceVisible()
                 && mProfileSupplier.hasValue();
     }
 
