diff --git a/chrome/browser/ui/webui/side_panel/bookmarks/BUILD.gn b/chrome/browser/ui/webui/side_panel/bookmarks/BUILD.gn
index b92ce452fc8b00f7e9adeb1e69eff25882b14ba7..05f98cafefa135518673a3f0751c60c0064eb7cd 100644
--- a/chrome/browser/ui/webui/side_panel/bookmarks/BUILD.gn
+++ b/chrome/browser/ui/webui/side_panel/bookmarks/BUILD.gn
@@ -4,7 +4,7 @@
 
 import("//mojo/public/tools/bindings/mojom.gni")
 
-assert(!is_android)
+#assert(!is_android)
 
 mojom("mojo_bindings") {
   sources = [ "bookmarks.mojom" ]
