diff --git a/chrome/browser/resources/extensions/manager.ts b/chrome/browser/resources/extensions/manager.ts
index a8ae24bb7fb91d82ea032a0ac6aea4c5736c3058..82bd1df05bdcd0903ce0e174a614b708c01d75f6 100644
--- a/chrome/browser/resources/extensions/manager.ts
+++ b/chrome/browser/resources/extensions/manager.ts
@@ -28,6 +28,8 @@ import './toolbar.js';
 import './kiosk_dialog.js';
 
 // </if>
+import './mises_open_web_store_footer.js'
+import './mises_default_extension_setting.js';
 
 import {CrContainerShadowMixin} from 'chrome://resources/cr_elements/cr_container_shadow_mixin.js';
 import {CrViewManagerElement} from 'chrome://resources/cr_elements/cr_view_manager/cr_view_manager.js';
