diff --git a/chrome/browser/resources/side_panel/history_clusters/BUILD.gn b/chrome/browser/resources/side_panel/history_clusters/BUILD.gn
index 3915ec58da686fe74ac39e8c7197f147984b5fb8..f70d9abc2fd71164deed7a32633ed9eec8bf2bb2 100644
--- a/chrome/browser/resources/side_panel/history_clusters/BUILD.gn
+++ b/chrome/browser/resources/side_panel/history_clusters/BUILD.gn
@@ -4,7 +4,7 @@
 
 import("//ui/webui/resources/tools/build_webui.gni")
 
-assert(!is_android)
+#assert(!is_android)
 
 build_webui("build") {
   grd_prefix = "side_panel_history_clusters"
