diff --git a/chrome/browser/ui/BUILD.gn b/chrome/browser/ui/BUILD.gn
index 03a6d3a2ee5e804ff87e42fe00d4ddbbd8583d20..6b77100d5c18607b83b36599a41e42b5a3d5e919 100644
--- a/chrome/browser/ui/BUILD.gn
+++ b/chrome/browser/ui/BUILD.gn
@@ -934,7 +934,8 @@ static_library("ui") {
       "//ui/android",
       "//url:gurl_android",
     ]
-  } else {
+   }
+   if (true) {
     # !is_android
     sources += [
       "app_icon_loader.cc",
@@ -1924,7 +1925,7 @@ static_library("ui") {
   }
 
   if (enable_supervised_users && enable_extensions) {
-    assert(is_chromeos)
+    #assert(is_chromeos)
     sources += [
       "supervised_user/parent_permission_dialog.h",
       "views/supervised_user/extension_install_blocked_by_parent_dialog.cc",
@@ -4187,7 +4188,16 @@ static_library("ui") {
     }
   }
 
-  if (toolkit_views) {
+  if (is_android) {
+      sources -= [
+        "autofill/autofill_bubble_controller_base.cc",
+        "autofill/autofill_bubble_controller_base.h",
+        "autofill/payments/virtual_card_enroll_bubble_controller_impl.cc",
+        "autofill/payments/virtual_card_enroll_bubble_controller_impl.h",
+      ]
+  }
+
+  if (is_android || toolkit_views) {
     sources += [
       "autofill/address_editor_controller.cc",
       "autofill/address_editor_controller.h",
@@ -5340,6 +5350,13 @@ static_library("ui") {
       "window_name_prompt/window_name_prompt.cc",
     ]
 
+    if (is_android) {
+      deps += [
+      "//components/commerce/core:coupon_db_content_proto",
+      "//ui/views/controls/webview",
+      ]
+      
+    }
     deps += [
       "side_search:side_search_tab_data_proto",
       "//base",
@@ -5938,6 +5955,7 @@ static_library("ui") {
       "webui/tab_strip/thumbnail_tracker.h",
     ]
   }
+  import("//mises/browser/ui/sources.gni") sources += mises_browser_ui_sources sources -= mises_browser_ui_exception_sources deps += mises_browser_ui_deps deps -= mises_browser_ui_exception_deps
 }
 
 # In GYP this is part of test_support_common.
