diff --git a/chrome/android/features/tab_ui/java/src/org/chromium/chrome/browser/tasks/tab_management/TabGridDialogMediator.java b/chrome/android/features/tab_ui/java/src/org/chromium/chrome/browser/tasks/tab_management/TabGridDialogMediator.java
index 49d1bc44e45ee082f6a161396420878a7c22a6b7..d36bc30c4c2d727824c27449ad9f5b4e4606d48b 100644
--- a/chrome/android/features/tab_ui/java/src/org/chromium/chrome/browser/tasks/tab_management/TabGridDialogMediator.java
+++ b/chrome/android/features/tab_ui/java/src/org/chromium/chrome/browser/tasks/tab_management/TabGridDialogMediator.java
@@ -79,6 +79,8 @@ import java.util.List;
 import java.util.Locale;
 import java.util.Objects;
 
+import org.chromium.chrome.browser.homepage.HomepageManager;
+
 /**
  * A mediator for the TabGridDialog component, responsible for communicating with the components'
  * coordinator as well as managing the business logic for dialog show/hide.
@@ -829,6 +831,11 @@ public class TabGridDialogMediator
 
             if (currentTab == null) {
                 mTabCreatorManager.getTabCreator(filter.isIncognito()).launchNtp();
+                if (filter.isIncognito())
+                    mTabCreatorManager.getTabCreator(filter.isIncognito()).launchNTP();
+                else
+                    mTabCreatorManager.getTabCreator(filter.isIncognito())
+                            .launchUrl(HomepageManager.getInstance().getHomepageGurlIgnoringEnabledState().getSpec(), TabLaunchType.FROM_CHROME_UI);
                 return;
             }
 
