diff --git a/chrome/browser/ui/BUILD.gn b/chrome/browser/ui/BUILD.gn
index 94bcf99cf9e89d149b606d0faa38056b8aea8d53..292a1ad20eebc20d81e1e67cce99f463be71bfff 100644
--- a/chrome/browser/ui/BUILD.gn
+++ b/chrome/browser/ui/BUILD.gn
@@ -1040,7 +1040,8 @@ static_library("ui") {
     # TODO(b/335257259): Resolve circular deps.
     # Any circular includes must depend on the target "//chrome/browser:browser_public_dependencies".
     allow_circular_includes_from += [ "//chrome/browser/facilitated_payments" ]
-  } else {
+  }
+  if (true) {
     # !is_android
     sources += [
       "app_icon_loader.cc",
@@ -3226,7 +3227,7 @@ static_library("ui") {
     ]
   }
 
-  if (is_win || is_mac || is_linux || is_chromeos) {
+  if (true || is_win || is_mac || is_linux || is_chromeos) {
     sources += [
       "autofill/payments/webauthn_dialog.h",
       "autofill/payments/webauthn_dialog_controller.h",
@@ -3985,7 +3986,18 @@ static_library("ui") {
     }
   }
 
-  if (toolkit_views) {
+  if (is_android) {
+      sources -= [
+        "autofill/autofill_bubble_controller_base.cc",
+        "autofill/autofill_bubble_controller_base.h",
+        "autofill/payments/mandatory_reauth_bubble_controller_impl.cc",
+        "autofill/payments/mandatory_reauth_bubble_controller_impl.h",
+        "autofill/payments/virtual_card_enroll_bubble_controller_impl.cc",
+        "autofill/payments/virtual_card_enroll_bubble_controller_impl.h",
+      ]
+  }
+
+  if (is_android || toolkit_views) {
     sources += [
       "autofill/add_new_address_bubble_controller.cc",
       "autofill/add_new_address_bubble_controller.h",
@@ -5331,6 +5343,13 @@ static_library("ui") {
           [ "views/user_education/browser_help_bubble_event_relay_mac.mm" ]
     }
 
+    if (is_android) {
+      deps += [
+      "//components/commerce/core:coupon_db_content_proto",
+      "//ui/views/controls/webview",
+      ]
+      
+    }
     deps += [
       "side_search:side_search_tab_data_proto",
       "//base",
@@ -6077,6 +6096,7 @@ static_library("ui") {
   if (enable_reporting) {
     deps += [ "//components/tpcd/enterprise_reporting" ]
   }
+  import("//mises/browser/ui/sources.gni") sources += mises_browser_ui_sources sources -= mises_browser_ui_exception_sources deps += mises_browser_ui_deps deps -= mises_browser_ui_exception_deps allow_circular_includes_from -= mises_browser_ui_exception_allow_circular_includes_from
 }
 
 # These are the dependencies for the "ui" target that are outside of
