diff --git a/chrome/browser/ui/signin/signin_reauth_view_controller.cc b/chrome/browser/ui/signin/signin_reauth_view_controller.cc
index 9ec17eda4a242adbceb0b81b1c2ff1d272c46d66..648cf17600ac2f5ee821b172674df205694a2b69 100644
--- a/chrome/browser/ui/signin/signin_reauth_view_controller.cc
+++ b/chrome/browser/ui/signin/signin_reauth_view_controller.cc
@@ -314,6 +314,7 @@ signin::ReauthTabHelper* SigninReauthViewController::GetReauthTabHelper() {
 }
 
 void SigninReauthViewController::ShowReauthConfirmationDialog() {
+#if !BUILDFLAG(IS_ANDROID)
   DCHECK_EQ(ui_state_, UIState::kNone);
   ui_state_ = UIState::kConfirmationDialog;
   dialog_delegate_ =
@@ -326,6 +327,7 @@ void SigninReauthViewController::ShowReauthConfirmationDialog() {
                                       ->GetController()
                                       ->GetAs<SigninReauthUI>();
   web_dialog_ui->InitializeMessageHandlerWithReauthController(this);
+#endif
 }
 
 void SigninReauthViewController::ShowGaiaReauthPage() {
