diff --git a/chrome/browser/ui/android/toolbar/java/src/org/chromium/chrome/browser/toolbar/adaptive/OptionalNewTabButtonController.java b/chrome/browser/ui/android/toolbar/java/src/org/chromium/chrome/browser/toolbar/adaptive/OptionalNewTabButtonController.java
index ab202623c40c1372f41349d8d18e1d67f4bbc29a..3e54b254e7d8ff2258c0e785d75d965c021a388d 100644
--- a/chrome/browser/ui/android/toolbar/java/src/org/chromium/chrome/browser/toolbar/adaptive/OptionalNewTabButtonController.java
+++ b/chrome/browser/ui/android/toolbar/java/src/org/chromium/chrome/browser/toolbar/adaptive/OptionalNewTabButtonController.java
@@ -123,6 +123,9 @@ public class OptionalNewTabButtonController
 
         boolean isIncognito = activeTabSupplier.get().isIncognito();
         RecordUserAction.record("MobileTopToolbarOptionalButtonNewTab");
+        if (isIncognito)
+        tabCreatorManager.getTabCreator(isIncognito).launchIncognitoNTP();
+        else
         tabCreatorManager.getTabCreator(isIncognito).launchNTP();
 
         if (mTrackerSupplier.hasValue()) {
