diff --git a/chrome/browser/resources/web_app_internals/BUILD.gn b/chrome/browser/resources/web_app_internals/BUILD.gn
index 642538782f2c1ae9fdcee6f8ea153dce3c49efdd..1c779943ccb6c8cf1354df72112e4eb47e2a91ff 100644
--- a/chrome/browser/resources/web_app_internals/BUILD.gn
+++ b/chrome/browser/resources/web_app_internals/BUILD.gn
@@ -4,7 +4,7 @@
 
 import("//ui/webui/resources/tools/build_webui.gni")
 
-assert(!is_android)
+#assert(!is_android)
 
 build_webui("build") {
   grd_prefix = "web_app_internals"
