diff --git a/components/omnibox/browser/shortcuts_backend.cc b/components/omnibox/browser/shortcuts_backend.cc
index 1533544825bc26b31981c2975ba5b296b3e33b12..b264ff7bc19a71cfaa9d41aa4b9ae57c98161a85 100644
--- a/components/omnibox/browser/shortcuts_backend.cc
+++ b/components/omnibox/browser/shortcuts_backend.cc
@@ -393,7 +393,7 @@ ShortcutsDatabase::Shortcut::MatchCore ShortcutsBackend::MatchToMatchCore(
   const AutocompleteMatch* normalized_match = &match;
   AutocompleteMatch temp;
 
-  if (AutocompleteMatch::IsSpecializedSearchType(match.type)) {
+  if (match.search_terms_args && AutocompleteMatch::IsSpecializedSearchType(match.type)) {
     DCHECK(match.search_terms_args);
     temp = BaseSearchProvider::CreateShortcutSearchSuggestion(
         match.search_terms_args->search_terms, match_type,
