diff --git a/components/search_engines/search_engine_choice/search_engine_choice_utils.h b/components/search_engines/search_engine_choice/search_engine_choice_utils.h
index 20aacc4b870d941a5b2046e797f3de4ee3305a4b..e13fafe47f93c34a1ed8f2a39a17dc57d5e6067a 100644
--- a/components/search_engines/search_engine_choice/search_engine_choice_utils.h
+++ b/components/search_engines/search_engine_choice/search_engine_choice_utils.h
@@ -298,7 +298,7 @@ void RecordUnexpectedSearchProvider(const TemplateURLData& data);
 void WipeSearchEngineChoicePrefs(PrefService& profile_prefs,
                                  WipeSearchEngineChoiceReason reason);
 
-#if !BUILDFLAG(IS_ANDROID)
+#if true || !BUILDFLAG(IS_ANDROID)
 // Returns the engine marketing snippet string resource id or -1 if the snippet
 // was not found.
 // The function definition is generated in `generated_marketing_snippets.cc`.
