diff --git a/components/ui_devtools/views/overlay_agent_views.cc b/components/ui_devtools/views/overlay_agent_views.cc
index 213852ef61e8077ed24006cfffd2934336df514b..f5a14b7d3be70404faeadea392159c6287987d7b 100644
--- a/components/ui_devtools/views/overlay_agent_views.cc
+++ b/components/ui_devtools/views/overlay_agent_views.cc
@@ -725,7 +725,7 @@ bool OverlayAgentViews::UpdateHighlight(
   root_layer = widget->GetLayer();
   layer_for_highlighting_screen_offset_ =
       widget->GetContentsView()->GetBoundsInScreen().OffsetFromOrigin();
-#else
+#elif defined(USE_AURA)
   gfx::NativeWindow root = window_and_bounds.first->GetRootWindow();
   root_layer = root->layer();
   layer_for_highlighting_screen_offset_ =
