diff --git a/components/keyed_service/ios/browser_state_keyed_service_factory.cc b/components/keyed_service/ios/browser_state_keyed_service_factory.cc
index 33c888e0f688c0bf3f488153477b1fb52d828ef5..e65eff353968f8bf15a3a06c79144bbefa7a64e4 100644
--- a/components/keyed_service/ios/browser_state_keyed_service_factory.cc
+++ b/components/keyed_service/ios/browser_state_keyed_service_factory.cc
@@ -9,6 +9,7 @@
 #include "components/keyed_service/core/keyed_service.h"
 #include "components/keyed_service/ios/browser_state_dependency_manager.h"
 #include "ios/web/public/browser_state.h"
+#include "base/logging.h"
 
 namespace {
 
@@ -39,7 +40,7 @@ void BrowserStateKeyedServiceFactory::SetTestingFactory(
 BrowserStateKeyedServiceFactory::BrowserStateKeyedServiceFactory(
     const char* name,
     BrowserStateDependencyManager* manager)
-    : KeyedServiceFactory(name, manager, BROWSER_STATE) {}
+    : KeyedServiceFactory(name, manager, BROWSER_STATE) { LOG(INFO) << "BrowserStateKeyedServiceFactory " << name;}
 
 BrowserStateKeyedServiceFactory::~BrowserStateKeyedServiceFactory() {
 }
@@ -102,6 +103,7 @@ void BrowserStateKeyedServiceFactory::ContextDestroyed(void* context) {
 
 void BrowserStateKeyedServiceFactory::RegisterPrefs(
     user_prefs::PrefRegistrySyncable* registry) {
+  LOG(INFO) << "BrowserStateKeyedServiceFactory::RegisterPrefs " << name();
   RegisterBrowserStatePrefs(registry);
 }
 
