diff --git a/components/permissions/request_type.h b/components/permissions/request_type.h
index 1a5bd7e8239decc49b4eb355b994aa604962b932..539ff2fa2ecca23c1e172a6f29eb578274025474 100644
--- a/components/permissions/request_type.h
+++ b/components/permissions/request_type.h
@@ -22,14 +22,14 @@ namespace permissions {
 enum class RequestType {
   kAccessibilityEvents,
   kArSession,
-#if !BUILDFLAG(IS_ANDROID)
+#if true || !BUILDFLAG(IS_ANDROID)
   kCameraPanTiltZoom,
 #endif
   kCameraStream,
   kClipboard,
   kTopLevelStorageAccess,
   kDiskQuota,
-#if !BUILDFLAG(IS_ANDROID)
+#if true || !BUILDFLAG(IS_ANDROID)
   kLocalFonts,
 #endif
   kGeolocation,
@@ -44,16 +44,16 @@ enum class RequestType {
 #if BUILDFLAG(IS_ANDROID) || BUILDFLAG(IS_CHROMEOS) || BUILDFLAG(IS_WIN)
   kProtectedMediaIdentifier,
 #endif
-#if !BUILDFLAG(IS_ANDROID)
+#if true || !BUILDFLAG(IS_ANDROID)
   kRegisterProtocolHandler,
   kSecurityAttestation,
 #endif
   kStorageAccess,
-#if !BUILDFLAG(IS_ANDROID)
+#if true || !BUILDFLAG(IS_ANDROID)
   kU2fApiRequest,
 #endif
   kVrSession,
-#if !BUILDFLAG(IS_ANDROID)
+#if true || !BUILDFLAG(IS_ANDROID)
   kWindowManagement,
   kMaxValue = kWindowManagement
 #else
@@ -83,7 +83,7 @@ bool IsConfirmationChipSupported(RequestType for_request_type);
 // Returns the icon to display.
 IconId GetIconId(RequestType type);
 
-#if !BUILDFLAG(IS_ANDROID)
+#if true || !BUILDFLAG(IS_ANDROID)
 // Returns the blocked icon to display.
 IconId GetBlockedIconId(RequestType type);
 #endif
