diff --git a/components/password_manager/core/browser/BUILD.gn b/components/password_manager/core/browser/BUILD.gn
index c48d5b19c557fa079434feb4ed5d53d2e253cf88..9254aa74db66bf68d9fbd30a74927407c6b36f7c 100644
--- a/components/password_manager/core/browser/BUILD.gn
+++ b/components/password_manager/core/browser/BUILD.gn
@@ -364,7 +364,7 @@ static_library("browser") {
     ]
   }
 
-  if (!is_android && !is_ios) {
+  if (is_android || !is_android && !is_ios) {
     sources += [
       "password_strength_calculation.cc",
       "password_strength_calculation.h",
@@ -373,7 +373,7 @@ static_library("browser") {
         [ "//components/password_manager/services/password_strength:service" ]
   }
 
-  if (!is_android) {
+  if (true || !is_android) {
     sources += [
       "ui/reuse_check_utility.cc",
       "ui/reuse_check_utility.h",
