diff --git a/components/omnibox/browser/BUILD.gn b/components/omnibox/browser/BUILD.gn
index 469daca0ec5296f95e59c4eaf80059cfc2f9314a..8001499b3d490b1eb9a5d2331fd7c273c5d0ca89 100644
--- a/components/omnibox/browser/BUILD.gn
+++ b/components/omnibox/browser/BUILD.gn
@@ -370,6 +370,8 @@ static_library("browser") {
     "//ui/gfx",
   ]
 
+  sources += mises_components_omnibox_browser_sources deps += mises_components_omnibox_browser_deps
+
   configs += [ "//build/config/compiler:wexit_time_destructors" ]
 
   if (is_android) {
@@ -911,6 +913,7 @@ source_set("unit_tests") {
     "//ui/gfx:test_support",
     "//url",
   ]
+  
 
   if ((!is_android || enable_vr) && !is_ios) {
     deps += [
