diff --git a/components/embedder_support/user_agent_utils.cc b/components/embedder_support/user_agent_utils.cc
index 0d9ed98759c4bc39ae06b1c554079b6e12d3463d..eaa4d8360c21a995f8e0779aec85e004ab11b671 100644
--- a/components/embedder_support/user_agent_utils.cc
+++ b/components/embedder_support/user_agent_utils.cc
@@ -205,7 +205,7 @@ const blink::UserAgentBrandList GetUserAgentBrandList(
       output_version_type == blink::UserAgentBrandVersionType::kFullVersion
           ? full_version
           : major_version;
-
+  MISES_GET_USER_AGENT_BRAND_LIST
   return GenerateBrandVersionList(major_version_number, brand, brand_version,
                                   maybe_brand_override, maybe_version_override,
                                   enable_updated_grease_by_policy,
@@ -266,7 +266,7 @@ std::string GetUserAgentInternal(
     UserAgentReductionEnterprisePolicyState user_agent_reduction) {
   std::string product = GetProductAndVersion(user_agent_reduction);
   if (base::CommandLine::ForCurrentProcess()->HasSwitch(kHeadless)) {
-    product.insert(0, "Headless");
+    //product.insert(0, "Headless");
   }
 
 #if BUILDFLAG(IS_ANDROID)
