diff --git a/components/media_router/common/BUILD.gn b/components/media_router/common/BUILD.gn
index 2241640103bab3568415e976413313b28014a51b..c16af7946f54f98855eea9f82aac30b4abf12fdd 100644
--- a/components/media_router/common/BUILD.gn
+++ b/components/media_router/common/BUILD.gn
@@ -40,7 +40,7 @@ static_library("common") {
 
   configs += [ "//build/config/compiler:wexit_time_destructors" ]
 
-  if (!is_android) {
+  if (true || !is_android) {
     sources += [
       "discovery/media_sink_service_base.cc",
       "discovery/media_sink_service_base.h",
