diff --git a/components/embedder_support/android/java/src/org/chromium/components/embedder_support/util/UrlUtilities.java b/components/embedder_support/android/java/src/org/chromium/components/embedder_support/util/UrlUtilities.java
index ef4b1c7b35e140f3f5bf5399ef5bc42bbb50a54a..06ffcf5c33ea6e3daa3b03640e2495d6320017dc 100644
--- a/components/embedder_support/android/java/src/org/chromium/components/embedder_support/util/UrlUtilities.java
+++ b/components/embedder_support/android/java/src/org/chromium/components/embedder_support/util/UrlUtilities.java
@@ -55,7 +55,7 @@ public class UrlUtilities {
      */
     private static final HashSet<String> INTERNAL_SCHEMES =
             CollectionUtil.newHashSet(UrlConstants.CHROME_SCHEME, UrlConstants.CHROME_NATIVE_SCHEME,
-                    ContentUrlConstants.ABOUT_SCHEME);
+                    ContentUrlConstants.ABOUT_SCHEME, "chrome-search", "mises", "mises-search", "devtools", "chrome-extension");
 
     private static final String TEL_SCHEME = "tel";
 
@@ -117,6 +117,12 @@ public class UrlUtilities {
      * @return Whether the URL's scheme is for a internal chrome page.
      */
     public static boolean isInternalScheme(@NonNull GURL gurl) {
+        if ("local-ntp".equals(gurl.getHost()))
+          return true;
+        if (gurl.getSpec().startsWith("chrome-search://"))
+          return true;
+        if (gurl.getSpec().startsWith("mises-search://"))
+          return true;
         return INTERNAL_SCHEMES.contains(gurl.getScheme());
     }
 
@@ -270,6 +276,12 @@ public class UrlUtilities {
      */
     public static boolean isNTPUrl(GURL gurl) {
         if (!gurl.isValid() || !isInternalScheme(gurl)) return false;
+        if ("local-ntp".equals(gurl.getHost()))
+          return true;
+        if (gurl.getSpec().startsWith("chrome-search://"))
+          return true;
+        if (gurl.getSpec().startsWith("mises-search://"))
+          return true;
         return UrlConstants.NTP_HOST.equals(gurl.getHost());
     }
 
@@ -308,7 +320,13 @@ public class UrlUtilities {
         if (LibraryLoader.getInstance().isInitialized()) return isNTPUrl(url);
         return TextUtils.equals(url, UrlConstants.NTP_URL)
                 || TextUtils.equals(url, UrlConstants.NTP_NON_NATIVE_URL)
-                || TextUtils.equals(url, UrlConstants.NTP_ABOUT_URL);
+                || TextUtils.equals(url, UrlConstants.NTP_ABOUT_URL)
+                || TextUtils.equals(url, "chrome-search://local-ntp/local-ntp.html")
+                || TextUtils.equals(url, "chrome-search://local-ntp/incognito-ntp.html")
+                || TextUtils.equals(url, "chrome-search://local-ntp/new-ntp.html")
+                || TextUtils.equals(url, "mises-search://local-ntp/local-ntp.html")
+                || TextUtils.equals(url, "mises-search://local-ntp/incognito-ntp.html")
+                || TextUtils.equals(url, "mises-search://local-ntp/new-ntp.html");
     }
 
     public static String extractPublisherFromPublisherUrl(String publisherUrl) {
