diff --git a/components/commerce/content/browser/BUILD.gn b/components/commerce/content/browser/BUILD.gn
index 795fd8bc8122cfe6cf99c0ada0d107622675e446..425303108eae239f193d53e89d21c029c28d2167 100644
--- a/components/commerce/content/browser/BUILD.gn
+++ b/components/commerce/content/browser/BUILD.gn
@@ -45,7 +45,7 @@ source_set("unit_tests") {
   ]
 }
 
-if (!is_android) {
+if (true || !is_android) {
   static_library("hint") {
     sources = [
       "hint/commerce_hint_tab_helper.cc",
