diff --git a/components/ntp_tiles/most_visited_sites.h b/components/ntp_tiles/most_visited_sites.h
index 86c98b818f8dc5179c2ad15f046f489f85df8d45..4624ce797a293ea4e25f4a03b8cc1f46afb5f4bc 100644
--- a/components/ntp_tiles/most_visited_sites.h
+++ b/components/ntp_tiles/most_visited_sites.h
@@ -203,6 +203,7 @@ class MostVisitedSites :
   void OnURLFilterChanged() override;
 #endif
 
+  void FetchPopularSitesIcons(const PopularSites::SitesVector& sites_vector);
   static void RegisterProfilePrefs(user_prefs::PrefRegistrySyncable* registry);
   static void ResetProfilePrefs(PrefService* prefs);
 
