diff --git a/components/keep_alive_registry/BUILD.gn b/components/keep_alive_registry/BUILD.gn
index f1df4c7e3932bdfa52a1286243029b767f938c16..b80278c900b4257d58155e7ba7716e6f369b0066 100644
--- a/components/keep_alive_registry/BUILD.gn
+++ b/components/keep_alive_registry/BUILD.gn
@@ -2,7 +2,7 @@
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
-assert(!is_android)
+#assert(!is_android)
 
 source_set("keep_alive_registry") {
   sources = [
