diff --git a/components/omnibox/browser/BUILD.gn b/components/omnibox/browser/BUILD.gn
index 4a6febc44e44f31c19f783680029d066a701579e..b3d393338262b213f5f7951e1ab865aab612b521 100644
--- a/components/omnibox/browser/BUILD.gn
+++ b/components/omnibox/browser/BUILD.gn
@@ -334,7 +334,9 @@ static_library("browser") {
     "//ui/base",
     "//ui/gfx",
   ]
-
+  
+  sources += mises_components_omnibox_browser_sources deps += mises_components_omnibox_browser_deps
+ 
   if ((!is_android || enable_vr) && !is_ios) {
     deps += [ ":vector_icons" ]
   }
