diff --git a/components/strings/BUILD.gn b/components/strings/BUILD.gn
index 62efcab467a0048760068c28b7d0be1236311746..aeeb1aa9bde605dc5a5c14c5e23e28d8a31831a7 100644
--- a/components/strings/BUILD.gn
+++ b/components/strings/BUILD.gn
@@ -52,6 +52,7 @@ grit("components_strings") {
   if (is_android) {
     outputs += android_components_strings_java_resources
   }
+  deps = [ "//mises/components/resources:strings" ]
 }
 
 if (is_android) {
