diff --git a/components/commerce/core/BUILD.gn b/components/commerce/core/BUILD.gn
index 349dadb3e82fda568600eff373608e655a3e2c0a..97440c8bb34a9e8159003921a6c54d02479bff41 100644
--- a/components/commerce/core/BUILD.gn
+++ b/components/commerce/core/BUILD.gn
@@ -127,7 +127,8 @@ if (is_android) {
     proto_in_dir = "//"
     sources = [ "proto/merchant_signal_db_content.proto" ]
   }
-} else {
+}
+if (true) {
   proto_library("cart_db_content_proto") {
     proto_in_dir = "//"
     sources = [ "proto/cart_db_content.proto" ]
