diff --git a/components/guest_view/renderer/BUILD.gn b/components/guest_view/renderer/BUILD.gn
index d12cd543e3b53995975c9811e647ad12c0dce28c..331bd677dd0bb1d005517fe0ed9666cac6dc23fe 100644
--- a/components/guest_view/renderer/BUILD.gn
+++ b/components/guest_view/renderer/BUILD.gn
@@ -6,7 +6,7 @@
 # assert to prevent the accidental building of GuestViews on mobile
 # platforms. If you're now using GuestViews on mobile, go ahead and
 # remove this assert.
-assert(!is_android && !is_ios)
+#assert(!is_android && !is_ios)
 
 static_library("renderer") {
   sources = [
