diff --git a/components/crash/android/java/src/org/chromium/components/crash/MinidumpLogcatPrepender.java b/components/crash/android/java/src/org/chromium/components/crash/MinidumpLogcatPrepender.java
index f825e634374999265caf6760a6a14607e8a190ac..dea702c2bb7eda4d7fbae9bc3f247ad2a232d7f8 100644
--- a/components/crash/android/java/src/org/chromium/components/crash/MinidumpLogcatPrepender.java
+++ b/components/crash/android/java/src/org/chromium/components/crash/MinidumpLogcatPrepender.java
@@ -71,7 +71,7 @@ public class MinidumpLogcatPrepender {
         try {
             writer = new BufferedWriter(new FileWriter(targetFile, false));
             writer.write(boundary);
-            writer.newLine();
+            writer.write("\r\n");
             // Next we write the logcat data in a MIME block.
             writer.write(LOGCAT_CONTENT_DISPOSITION);
             writer.newLine();
