diff --git a/components/payments/content/BUILD.gn b/components/payments/content/BUILD.gn
index bb89ee6622555d40e262273c93a8dead7e9fe05f..930e0aa40def3e504d7434f25947230dc77ce210 100644
--- a/components/payments/content/BUILD.gn
+++ b/components/payments/content/BUILD.gn
@@ -122,7 +122,8 @@ static_library("content") {
       "secure_payment_confirmation_no_creds_view_stub.cc",
       "secure_payment_confirmation_view_stub.cc",
     ]
-  } else {
+  } #} else {
+  if (true) {
     sources += [
       "content_payment_request_delegate.cc",
       "content_payment_request_delegate.h",
