diff --git a/components/media_router/browser/BUILD.gn b/components/media_router/browser/BUILD.gn
index 5b8464e01730ccf9605b173162e809e071f71b01..656c1147784e8279d5780b4153652c237c8672f9 100644
--- a/components/media_router/browser/BUILD.gn
+++ b/components/media_router/browser/BUILD.gn
@@ -77,7 +77,9 @@ source_set("browser") {
       "android/media_router_dialog_controller_android.h",
     ]
     deps += [ "android:jni_headers" ]
-  } else {
+#  } else {
+  }
+  if (true) {
     sources += [
       "issue_manager.cc",
       "issue_manager.h",
