diff --git a/components/permissions/permission_manager.cc b/components/permissions/permission_manager.cc
index 96d7af0319c885e883669516fc05de60417fa3a1..67482c0108211890b91100389c706f4e15ed0f4c 100644
--- a/components/permissions/permission_manager.cc
+++ b/components/permissions/permission_manager.cc
@@ -678,6 +678,10 @@ PermissionResult PermissionManager::GetPermissionStatusHelper(
   if (status != CONTENT_SETTING_DEFAULT)
     return PermissionResult(status, PermissionStatusSource::UNSPECIFIED);
   PermissionContextBase* context = GetPermissionContext(permission);
+  //DCHECK(context) << "for:" << (int)permission;
+  if (!context) 
+    return PermissionResult(CONTENT_SETTING_DEFAULT, PermissionStatusSource::UNSPECIFIED);
+
   PermissionResult result = context->GetPermissionStatus(
       render_frame_host, canonical_requesting_origin.DeprecatedGetOriginAsURL(),
       embedding_origin.DeprecatedGetOriginAsURL());
