diff --git a/components/password_manager/services/password_strength/BUILD.gn b/components/password_manager/services/password_strength/BUILD.gn
index 586c091832d787fb5038a50dd90b4ff78dc767f8..83bf6025f9bcb82148c42605b1a0ade1c6345d6e 100644
--- a/components/password_manager/services/password_strength/BUILD.gn
+++ b/components/password_manager/services/password_strength/BUILD.gn
@@ -2,7 +2,7 @@
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
-assert(!is_android && !is_ios)
+#assert(!is_android && !is_ios)
 
 source_set("lib") {
   sources = [
