diff --git a/components/policy/tools/generate_policy_source.py b/components/policy/tools/generate_policy_source.py
index be1553785d5b53b05e218c18acf980f59c89fa2a..720208097369c57003a1ed12bc0e61aabc32e1eb 100755
--- a/components/policy/tools/generate_policy_source.py
+++ b/components/policy/tools/generate_policy_source.py
@@ -44,6 +44,7 @@ PLATFORM_STRINGS = {
     'chrome.win7': ['win'],
 }
 
+from policy_source_helper import AddMisesPolicies, CHROMIUM_POLICY_KEY
 
 class PolicyDetails:
   """Parses a policy template and caches all its details."""
@@ -333,6 +334,7 @@ def main():
     chrome_major_version = args.chrome_version_major
 
   template_file_contents = _LoadJSONFile(template_file_name)
+  AddMisesPolicies(template_file_contents)
   risk_tags = RiskTags(template_file_contents)
   policy_details = [
       PolicyDetails(policy, chrome_major_version, target_platform,
