diff --git a/components/cdm/renderer/key_system_support_update.cc b/components/cdm/renderer/key_system_support_update.cc
index d949bb47e1289df14474d84376e42dcd15b78f4d..9bfe57faaac818d2ce5aeccebcf430331bc8ebc9 100644
--- a/components/cdm/renderer/key_system_support_update.cc
+++ b/components/cdm/renderer/key_system_support_update.cc
@@ -512,7 +512,9 @@ void AddAndroidPlatformKeySystem(
     const media::mojom::KeySystemCapabilityPtr& capability,
     bool can_persist_data,
     KeySystemInfos* key_systems) {
+#if BUILDFLAG(ENABLE_WIDEVINE)
   DCHECK_NE(key_system, kWidevineKeySystem);
+#endif
 
   // When using MediaDrm, we assume it'll always try to persist some data.
   // If we don't allow data to be persisted, then skip this key system.
