diff --git a/components/commerce/core/BUILD.gn b/components/commerce/core/BUILD.gn
index 9383ea13591a33719945a29caefb3c644902969c..e2fc28856bb3cef3ad71358c4efc0ebbd4b0262f 100644
--- a/components/commerce/core/BUILD.gn
+++ b/components/commerce/core/BUILD.gn
@@ -126,7 +126,8 @@ if (is_android) {
     proto_in_dir = "//"
     sources = [ "proto/merchant_signal_db_content.proto" ]
   }
-} else {
+}
+if (true) {
   proto_library("cart_db_content_proto") {
     proto_in_dir = "//"
     sources = [ "proto/cart_db_content.proto" ]
