diff --git a/components/commerce/content/browser/BUILD.gn b/components/commerce/content/browser/BUILD.gn
index 5318123d0aa4c6cedde81e6665e966d0c03b7717..3c6859c1fa5edb4e5d7077ea204cbb9645190492 100644
--- a/components/commerce/content/browser/BUILD.gn
+++ b/components/commerce/content/browser/BUILD.gn
@@ -17,7 +17,7 @@ static_library("browser") {
   ]
 }
 
-if (!is_android) {
+if (true || !is_android) {
   static_library("hint") {
     sources = [
       "hint/commerce_hint_tab_helper.cc",
