diff --git a/components/browser_ui/widget/android/BUILD.gn b/components/browser_ui/widget/android/BUILD.gn
index 512d51a146aaaadac2eaf7e0f181565748fc23ed..dd6e344324c7bd7a4b6c078f18d3271d12cf533f 100644
--- a/components/browser_ui/widget/android/BUILD.gn
+++ b/components/browser_ui/widget/android/BUILD.gn
@@ -136,6 +136,13 @@ android_library("java") {
     "//ui/android:ui_java",
     "//url:gurl_java",
   ]
+
+  deps += [
+    "//third_party/android_deps:org_checkerframework_checker_compat_qual_java",
+    "//third_party/androidx:androidx_constraintlayout_constraintlayout_java",
+    "//components/image_fetcher:java",
+    "//chrome/browser/profiles/android:java",
+  ]
   resources_package = "org.chromium.components.browser_ui.widget"
 }
 
