diff --git a/components/browser_ui/site_settings/android/java/src/org/chromium/components/browser_ui/site_settings/SingleWebsiteSettings.java b/components/browser_ui/site_settings/android/java/src/org/chromium/components/browser_ui/site_settings/SingleWebsiteSettings.java
index c14d4fa7bd4a8fbb2fcee7cb81121b49fc6edc7a..e0c9c105e7457aaac2d57e969c83f09a4fbd4e97 100644
--- a/components/browser_ui/site_settings/android/java/src/org/chromium/components/browser_ui/site_settings/SingleWebsiteSettings.java
+++ b/components/browser_ui/site_settings/android/java/src/org/chromium/components/browser_ui/site_settings/SingleWebsiteSettings.java
@@ -257,7 +257,13 @@ public class SingleWebsiteSettings extends SiteSettingsPreferenceFragment
      */
     public static Bundle createFragmentArgsForSite(String url) {
         Bundle fragmentArgs = new Bundle();
-        String origin = Origin.createOrThrow(url).toString();
+        String origin;
+        try {
+            origin = Origin.createOrThrow(url).toString();
+        }
+        catch (IllegalArgumentException e) {
+            origin = "localhost";
+        }    
         fragmentArgs.putSerializable(EXTRA_SITE_ADDRESS, WebsiteAddress.create(origin));
         return fragmentArgs;
     }
