diff --git a/components/bookmarks/browser/BUILD.gn b/components/bookmarks/browser/BUILD.gn
index 95a01fe95dc7e6af3804deb9df46dc495aa6594c..344decdeceb6a34a5007766d68d1cb8fb04f5eaa 100644
--- a/components/bookmarks/browser/BUILD.gn
+++ b/components/bookmarks/browser/BUILD.gn
@@ -79,7 +79,7 @@ static_library("browser") {
     all_dependent_configs = [ "//ui/views:flags" ]
   }
 
-  if (toolkit_views && !is_mac) {
+  if (is_android || toolkit_views && !is_mac) {
     sources += [ "bookmark_node_data_views.cc" ]
   }
 
