diff --git a/components/omnibox/browser/BUILD.gn b/components/omnibox/browser/BUILD.gn
index fa8f859e5204cbf3ce7f241613c5e9f9c978a2d3..f6319910f4b91540d91bacfbd0d97f6b8745de06 100644
--- a/components/omnibox/browser/BUILD.gn
+++ b/components/omnibox/browser/BUILD.gn
@@ -373,6 +373,8 @@ static_library("browser") {
     "//ui/gfx",
   ]
 
+  sources += mises_components_omnibox_browser_sources deps += mises_components_omnibox_browser_deps
+
   configs += [ "//build/config/compiler:wexit_time_destructors" ]
 
   if (is_android) {
@@ -920,6 +922,7 @@ source_set("unit_tests") {
     "//ui/gfx:test_support",
     "//url",
   ]
+  
 
   if ((!is_android || enable_vr) && !is_ios) {
     deps += [
