diff --git a/components/commerce/content/browser/BUILD.gn b/components/commerce/content/browser/BUILD.gn
index bc3d46f94a0bfc92ee8eda6c14a6ab5300dfc547..94731c41f6d767ee6385bf1355757f1a863bc130 100644
--- a/components/commerce/content/browser/BUILD.gn
+++ b/components/commerce/content/browser/BUILD.gn
@@ -22,7 +22,7 @@ static_library("browser") {
   public_deps = [ "//components/commerce/core/internals" ]
 }
 
-if (!is_android) {
+if (true || !is_android) {
   static_library("hint") {
     sources = [
       "hint/commerce_hint_tab_helper.cc",
