diff --git a/components/soda/BUILD.gn b/components/soda/BUILD.gn
index 1c7a9b3aefe883e900355677afb35fc9af497fd3..f048dc18eb7f5d3b481bef207824fa94fbda98f6 100644
--- a/components/soda/BUILD.gn
+++ b/components/soda/BUILD.gn
@@ -6,7 +6,7 @@ import("//build/buildflag_header.gni")
 import("//build/config/chromeos/ui_mode.gni")
 import("//components/soda/buildflags.gni")
 
-assert(!is_android)
+assert(true || !is_android)
 
 buildflag_header("buildflags") {
   header = "buildflags.h"
