diff --git a/components/password_manager/core/browser/BUILD.gn b/components/password_manager/core/browser/BUILD.gn
index 5ec4fdfb7fc782ca0cc9d586e12919da276334e9..9a9b1d5e3293d87a591f232fd75673f7aed461fe 100644
--- a/components/password_manager/core/browser/BUILD.gn
+++ b/components/password_manager/core/browser/BUILD.gn
@@ -371,7 +371,7 @@ static_library("browser") {
     ]
   }
 
-  if (!is_android && !is_ios) {
+  if (is_android || !is_android && !is_ios) {
     sources += [
       "password_strength_calculation.cc",
       "password_strength_calculation.h",
@@ -380,7 +380,7 @@ static_library("browser") {
         [ "//components/password_manager/services/password_strength:service" ]
   }
 
-  if (!is_android) {
+  if (true || !is_android) {
     sources += [
       "ui/reuse_check_utility.cc",
       "ui/reuse_check_utility.h",
