diff --git a/components/permissions/permission_request.cc b/components/permissions/permission_request.cc
index f7a062ce39e14c6c8e94ef49aa4339cf795956c5..6a70252a79b70443a5de020f6664f4606c9d23d1 100644
--- a/components/permissions/permission_request.cc
+++ b/components/permissions/permission_request.cc
@@ -124,7 +124,7 @@ bool PermissionRequest::IsEmbeddedPermissionElementInitiated() const {
   return data_.embedded_permission_element_initiated;
 }
 
-#if !BUILDFLAG(IS_ANDROID)
+#if true || !BUILDFLAG(IS_ANDROID)
 
 bool PermissionRequest::IsConfirmationChipSupported() {
   return permissions::IsConfirmationChipSupported(request_type());
