diff --git a/components/browser_ui/widget/android/BUILD.gn b/components/browser_ui/widget/android/BUILD.gn
index 72eca4feb541e2a6e42777434bffc8a78e3f6012..6bb281e508bceeec3d66f853ddd4dcb637864688 100644
--- a/components/browser_ui/widget/android/BUILD.gn
+++ b/components/browser_ui/widget/android/BUILD.gn
@@ -146,6 +146,12 @@ android_library("java") {
     "//ui/android:ui_java",
     "//url:gurl_java",
   ]
+
+  deps += [
+    "//third_party/android_deps:org_checkerframework_checker_compat_qual_java",
+    "//third_party/androidx:androidx_constraintlayout_constraintlayout_java",
+    "//components/image_fetcher:java",
+  ]
   resources_package = "org.chromium.components.browser_ui.widget"
 }
 
