diff --git a/components/keep_alive_registry/BUILD.gn b/components/keep_alive_registry/BUILD.gn
index bf3144a94926f3350c8765fccafac1d93d9ee93b..2111b4a8687b1ea435dc4f9a4faeee1f407acd94 100644
--- a/components/keep_alive_registry/BUILD.gn
+++ b/components/keep_alive_registry/BUILD.gn
@@ -2,7 +2,7 @@
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
-assert(!is_android)
+#assert(!is_android)
 
 source_set("keep_alive_registry") {
   sources = [
