diff --git a/components/manta/BUILD.gn b/components/manta/BUILD.gn
index 313c6443eb536e41b04af9f2ef88359657b14a9e..081fbbde953fa061d3f9a1966432aa522ef9a4a8 100644
--- a/components/manta/BUILD.gn
+++ b/components/manta/BUILD.gn
@@ -2,7 +2,7 @@
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
-assert(!is_android)
+#assert(!is_android)
 
 component("manta") {
   sources = [
