diff --git a/components/media_router/browser/BUILD.gn b/components/media_router/browser/BUILD.gn
index c7cbbc8d6e95d28e9e3ffd62e9d109b24c2741ee..7cb0d891232adeb8457854fc15de3bafd1a48f04 100644
--- a/components/media_router/browser/BUILD.gn
+++ b/components/media_router/browser/BUILD.gn
@@ -80,7 +80,9 @@ source_set("browser") {
       "android/media_router_dialog_controller_android.h",
     ]
     deps += [ "android:jni_headers" ]
-  } else {
+#  } else {
+  }
+  if (true) {
     sources += [
       "issue_manager.cc",
       "issue_manager.h",
