diff --git a/components/permissions/BUILD.gn b/components/permissions/BUILD.gn
index b56fa81eace5702f1a1765b3f65cab9838d72a6b..e6e2fbb3fe64c52699a971ad18ff32c5a85d0712 100644
--- a/components/permissions/BUILD.gn
+++ b/components/permissions/BUILD.gn
@@ -142,6 +142,7 @@ source_set("permissions") {
   if (!is_android && !is_ios) {
     deps += [ "//components/guest_view/browser" ]
   }
+  import("//mises/components/permissions/sources.gni") sources += mises_components_permissions_sources deps += mises_components_permissions_deps
   if (is_android) {
     sources += [
       "android/android_permission_util.cc",
