diff --git a/components/stylus_handwriting/android/java/src/org/chromium/components/stylus_handwriting/DirectWritingSettingsHelper.java b/components/stylus_handwriting/android/java/src/org/chromium/components/stylus_handwriting/DirectWritingSettingsHelper.java
index 2d24ce9acf54b957c6805026f73af961b9362e21..ccd438a18e087722b80a179e486d8317d5e0a9a2 100644
--- a/components/stylus_handwriting/android/java/src/org/chromium/components/stylus_handwriting/DirectWritingSettingsHelper.java
+++ b/components/stylus_handwriting/android/java/src/org/chromium/components/stylus_handwriting/DirectWritingSettingsHelper.java
@@ -25,6 +25,7 @@ public class DirectWritingSettingsHelper {
     public static boolean isEnabled(Context context) {
         // Samsung keyboard supports handwriting in Chrome and Webview from Android S onwards.
         if (Build.VERSION.SDK_INT < Build.VERSION_CODES.S) return false;
+        if (!isDirectWritingServiceCallbackAvailable()) return false;
         return isHoneyboardDefault(context) && isFeatureEnabled(context);
     }
 
@@ -64,4 +65,24 @@ public class DirectWritingSettingsHelper {
         }
         return false;
     }
+    private static @Nullable Boolean sDirectWritingServiceCallbackAvailable;
+    private static boolean isDirectWritingServiceCallbackAvailable() {
+        if (sDirectWritingServiceCallbackAvailable == null) {
+            try {
+                Class dwCallbackClass =
+                        Class.forName(
+                                "org.chromium.components.stylus_handwriting.DirectWritingServiceCallback");
+                // On some devices, the DirectWritingServiceCallback constructor is not available
+                // so this throws a NoSuchMethodException.
+                dwCallbackClass.getConstructor().isAccessible();
+                Class dwCallbackStubClass = Class.forName(
+                        "android.widget.directwriting.IDirectWritingServiceCallback.Stub");
+                dwCallbackStubClass.getConstructor().isAccessible();
+                sDirectWritingServiceCallbackAvailable = true;
+            } catch (ClassNotFoundException | NoSuchMethodException e | NoSuchMethodError e) {
+                sDirectWritingServiceCallbackAvailable = false;
+            }
+        }
+        return sDirectWritingServiceCallbackAvailable;
+    }
 }
