diff --git a/components/browser_ui/site_settings/android/java/src/org/chromium/components/browser_ui/site_settings/SingleWebsiteSettings.java b/components/browser_ui/site_settings/android/java/src/org/chromium/components/browser_ui/site_settings/SingleWebsiteSettings.java
index a404f12fb9bd2c2ea1b09fb70dfcf389428afbb2..252ccc9327abd8aa0efb9b6f151a56eccf376e62 100644
--- a/components/browser_ui/site_settings/android/java/src/org/chromium/components/browser_ui/site_settings/SingleWebsiteSettings.java
+++ b/components/browser_ui/site_settings/android/java/src/org/chromium/components/browser_ui/site_settings/SingleWebsiteSettings.java
@@ -263,7 +263,13 @@ public class SingleWebsiteSettings extends BaseSiteSettingsFragment
      */
     public static Bundle createFragmentArgsForSite(String url) {
         Bundle fragmentArgs = new Bundle();
-        String origin = Origin.createOrThrow(url).toString();
+        String origin;
+        try {
+            origin = Origin.createOrThrow(url).toString();
+        }
+        catch (IllegalArgumentException e) {
+            origin = "localhost";
+        }    
         fragmentArgs.putSerializable(EXTRA_SITE_ADDRESS, WebsiteAddress.create(origin));
         return fragmentArgs;
     }
