diff --git a/components/component_updater/component_updater_service.h b/components/component_updater/component_updater_service.h
index 36215051767a1b1ff4d924e977b438047ccc1527..a3a3417ffdb49e37fff1e9005eb2d643062b3aa2 100644
--- a/components/component_updater/component_updater_service.h
+++ b/components/component_updater/component_updater_service.h
@@ -186,6 +186,7 @@ class ComponentUpdateService {
 
   friend class speech::SodaInstallerImpl;
   friend class ::ComponentsHandler;
+  MISES_COMPONENT_UPDATER_SERVICE_H_
   FRIEND_TEST_ALL_PREFIXES(ComponentInstallerTest, RegisterComponent);
 };
 
@@ -199,7 +200,6 @@ class OnDemandUpdater {
   enum class Priority { BACKGROUND = 0, FOREGROUND = 1 };
 
   virtual ~OnDemandUpdater() = default;
-
  private:
   friend class OnDemandTester;
   friend class policy::ComponentUpdaterPolicyTest;
@@ -212,7 +212,7 @@ class OnDemandUpdater {
 #if BUILDFLAG(IS_CHROMEOS_ASH)
   friend class CrOSComponentInstaller;
 #endif  // BUILDFLAG(IS_CHROMEOS_ASH)
-
+  MISES_COMPONENT_UPDATER_SERVICE_H_ON_DEMAND_UPDATER
   // Triggers an update check for a component. |id| is a value
   // returned by GetCrxComponentID(). If an update for this component is already
   // in progress, the function returns |kInProgress|. If an update is available,
