diff --git a/components/embedder_support/android/java/src/org/chromium/components/embedder_support/util/Origin.java b/components/embedder_support/android/java/src/org/chromium/components/embedder_support/util/Origin.java
index 5f278ecbc999b30a5fec21bc7cab9ba09ef4884d..909af99cf9ec92917ad1ecb32f533aa60f5fd7a2 100644
--- a/components/embedder_support/android/java/src/org/chromium/components/embedder_support/util/Origin.java
+++ b/components/embedder_support/android/java/src/org/chromium/components/embedder_support/util/Origin.java
@@ -50,7 +50,7 @@ public class Origin {
 
         // This class can only correctly handle certain origins, see https://crbug.com/1019244.
         String scheme = uri.getScheme();
-        if (!scheme.equals(UrlConstants.HTTP_SCHEME) && !scheme.equals(UrlConstants.HTTPS_SCHEME)) {
+        if (!scheme.equals(UrlConstants.HTTP_SCHEME) && !scheme.equals(UrlConstants.HTTPS_SCHEME) && !scheme.equals("chrome-extension") && !scheme.equals("mises-extension")) {
             return null;
         }
 
