diff --git a/components/commerce/core/BUILD.gn b/components/commerce/core/BUILD.gn
index 7d11bd541cfd9f9e62b37b69373d10c66d17089a..e9cd382245d0d991af11f80809cbacacefcdd85f 100644
--- a/components/commerce/core/BUILD.gn
+++ b/components/commerce/core/BUILD.gn
@@ -86,7 +86,8 @@ if (is_android) {
     proto_in_dir = "//"
     sources = [ "proto/merchant_signal_db_content.proto" ]
   }
-} else {
+}
+if (true) {
   proto_library("cart_db_content_proto") {
     proto_in_dir = "//"
     sources = [ "proto/cart_db_content.proto" ]
