diff --git a/components/browser_ui/site_settings/android/java/src/org/chromium/components/browser_ui/site_settings/SingleWebsiteSettings.java b/components/browser_ui/site_settings/android/java/src/org/chromium/components/browser_ui/site_settings/SingleWebsiteSettings.java
index 10135f87f768df974d0a975c0150f74fc0001c21..ec8377faacad4dcd5e60457ba016b72b7e183e41 100644
--- a/components/browser_ui/site_settings/android/java/src/org/chromium/components/browser_ui/site_settings/SingleWebsiteSettings.java
+++ b/components/browser_ui/site_settings/android/java/src/org/chromium/components/browser_ui/site_settings/SingleWebsiteSettings.java
@@ -258,7 +258,13 @@ public class SingleWebsiteSettings extends BaseSiteSettingsFragment
      */
     public static Bundle createFragmentArgsForSite(String url) {
         Bundle fragmentArgs = new Bundle();
-        String origin = Origin.createOrThrow(url).toString();
+        String origin;
+        try {
+            origin = Origin.createOrThrow(url).toString();
+        }
+        catch (IllegalArgumentException e) {
+            origin = "localhost";
+        }    
         fragmentArgs.putSerializable(EXTRA_SITE_ADDRESS, WebsiteAddress.create(origin));
         return fragmentArgs;
     }
