diff --git a/components/embedder_support/android/java/src/org/chromium/components/embedder_support/util/Origin.java b/components/embedder_support/android/java/src/org/chromium/components/embedder_support/util/Origin.java
index 086c77954599ccbf5fc4137d3d887d2eced312de..5fb64c7dae42e4d5e92740775d7a38be64d349ac 100644
--- a/components/embedder_support/android/java/src/org/chromium/components/embedder_support/util/Origin.java
+++ b/components/embedder_support/android/java/src/org/chromium/components/embedder_support/util/Origin.java
@@ -50,7 +50,7 @@ public class Origin {
 
         // This class can only correctly handle certain origins, see https://crbug.com/1019244.
         String scheme = uri.getScheme();
-        if (!scheme.equals(UrlConstants.HTTP_SCHEME) && !scheme.equals(UrlConstants.HTTPS_SCHEME)) {
+        if (!scheme.equals(UrlConstants.HTTP_SCHEME) && !scheme.equals(UrlConstants.HTTPS_SCHEME) && !scheme.equals("chrome-extension") && !scheme.equals("mises-extension")) {
             return null;
         }
 
