diff --git a/components/stylus_handwriting/android/java/src/org/chromium/components/stylus_handwriting/DirectWritingSettingsHelper.java b/components/stylus_handwriting/android/java/src/org/chromium/components/stylus_handwriting/DirectWritingSettingsHelper.java
index c766ea32b6400b53b972067d282ad6802d182375..ba4c9ee1ba96a721d617e565332f7a6a16351e8e 100644
--- a/components/stylus_handwriting/android/java/src/org/chromium/components/stylus_handwriting/DirectWritingSettingsHelper.java
+++ b/components/stylus_handwriting/android/java/src/org/chromium/components/stylus_handwriting/DirectWritingSettingsHelper.java
@@ -80,9 +80,12 @@ public class DirectWritingSettingsHelper {
                 // On some devices, the DirectWritingServiceCallback constructor is not available
                 // so this throws a NoSuchMethodException.
                 dwCallbackClass.getConstructor().isAccessible();
+                Class dwCallbackStubClass = Class.forName(
+                        "android.widget.directwriting.IDirectWritingServiceCallback.Stub");
+                dwCallbackStubClass.getConstructor().isAccessible();
                 sDirectWritingServiceCallbackAvailable = true;
                 logDWServiceCallbackFailed(false);
-            } catch (ClassNotFoundException | NoSuchMethodException e) {
+            } catch (ClassNotFoundException | NoSuchMethodException | NoSuchMethodError e) {
                 logDWServiceCallbackFailed(true);
                 sDirectWritingServiceCallbackAvailable = false;
             }
