diff --git a/components/webauthn/core/browser/BUILD.gn b/components/webauthn/core/browser/BUILD.gn
index 206e0fd6848a40b7237393b99ac63f4656d4bc97..f22577bccd147432df1877d65040e28edf2ed8ff 100644
--- a/components/webauthn/core/browser/BUILD.gn
+++ b/components/webauthn/core/browser/BUILD.gn
@@ -15,7 +15,7 @@ source_set("browser") {
     ]
   }
 
-  if (!is_ios && !is_android) {
+  if (is_android || !is_ios && !is_android) {
     sources += [
       "passkey_model.h",
       "passkey_model_utils.cc",
