diff --git a/components/permissions/request_type.h b/components/permissions/request_type.h
index 6443d5fbe8377998f93f28794d02da963f45d7e4..8a8eb13b80b3fc635b1f7085d796bb898d322c46 100644
--- a/components/permissions/request_type.h
+++ b/components/permissions/request_type.h
@@ -23,23 +23,23 @@ namespace permissions {
 enum class RequestType {
   kAccessibilityEvents,
   kArSession,
-#if !BUILDFLAG(IS_ANDROID)
+#if true || !BUILDFLAG(IS_ANDROID)
   kCameraPanTiltZoom,
 #endif
   kCameraStream,
-#if !BUILDFLAG(IS_ANDROID)
+#if true || !BUILDFLAG(IS_ANDROID)
   kCapturedSurfaceControl,
 #endif
   kClipboard,
   kTopLevelStorageAccess,
   kDiskQuota,
-#if !BUILDFLAG(IS_ANDROID)
+#if true || !BUILDFLAG(IS_ANDROID)
   kFileSystemAccess,
 #endif
   kGeolocation,
   kIdentityProvider,
   kIdleDetection,
-#if !BUILDFLAG(IS_ANDROID)
+#if true || !BUILDFLAG(IS_ANDROID)
   kLocalFonts,
 #endif
   kMicStream,
@@ -56,7 +56,7 @@ enum class RequestType {
 #if BUILDFLAG(IS_ANDROID) || BUILDFLAG(IS_CHROMEOS) || BUILDFLAG(IS_WIN)
   kProtectedMediaIdentifier,
 #endif
-#if !BUILDFLAG(IS_ANDROID)
+#if true || !BUILDFLAG(IS_ANDROID)
   kRegisterProtocolHandler,
 #endif
 #if BUILDFLAG(IS_CHROMEOS)
@@ -67,7 +67,7 @@ enum class RequestType {
 #if BUILDFLAG(IS_CHROMEOS) && BUILDFLAG(USE_CUPS)
   kWebPrinting,
 #endif
-#if !BUILDFLAG(IS_ANDROID)
+#if true || !BUILDFLAG(IS_ANDROID)
   kWindowManagement,
   kMaxValue = kWindowManagement
 #else
@@ -100,7 +100,7 @@ bool IsConfirmationChipSupported(RequestType for_request_type);
 // Returns the icon to display.
 IconId GetIconId(RequestType type);
 
-#if !BUILDFLAG(IS_ANDROID)
+#if true || !BUILDFLAG(IS_ANDROID)
 // Returns the blocked icon to display.
 IconId GetBlockedIconId(RequestType type);
 #endif
