diff --git a/components/embedder_support/android/java/src/org/chromium/components/embedder_support/util/UrlUtilities.java b/components/embedder_support/android/java/src/org/chromium/components/embedder_support/util/UrlUtilities.java
index 91b83c79ce98899ec7eb8901428276bc98391392..7dd9287325383ef768a84c5b7d42d9cf8041d508 100644
--- a/components/embedder_support/android/java/src/org/chromium/components/embedder_support/util/UrlUtilities.java
+++ b/components/embedder_support/android/java/src/org/chromium/components/embedder_support/util/UrlUtilities.java
@@ -65,7 +65,7 @@ public class UrlUtilities {
             CollectionUtil.newHashSet(
                     UrlConstants.CHROME_SCHEME,
                     UrlConstants.CHROME_NATIVE_SCHEME,
-                    ContentUrlConstants.ABOUT_SCHEME);
+                    ContentUrlConstants.ABOUT_SCHEME, "mises", "devtools", "chrome-extension");
 
     private static final String TEL_SCHEME = "tel";
 
