diff --git a/components/ui_devtools/views/overlay_agent_views.cc b/components/ui_devtools/views/overlay_agent_views.cc
index a1a26f88e19827e82a8b99a58b9aa55c3a6e6976..27a86bd53f19d7c0fea0761148f7f713664a487f 100644
--- a/components/ui_devtools/views/overlay_agent_views.cc
+++ b/components/ui_devtools/views/overlay_agent_views.cc
@@ -727,7 +727,7 @@ bool OverlayAgentViews::UpdateHighlight(
   root_layer = widget->GetLayer();
   layer_for_highlighting_screen_offset_ =
       widget->GetContentsView()->GetBoundsInScreen().OffsetFromOrigin();
-#else
+#elif defined(USE_AURA)
   gfx::NativeWindow root = window_and_bounds.first->GetRootWindow();
   root_layer = root->layer();
   layer_for_highlighting_screen_offset_ =
