diff --git a/components/soda/BUILD.gn b/components/soda/BUILD.gn
index d6b0266c0c6d143ff466c12aaf901222f1df6809..1774409b3204fc84bd2a48a3c807077a5a01c003 100644
--- a/components/soda/BUILD.gn
+++ b/components/soda/BUILD.gn
@@ -6,7 +6,7 @@ import("//build/buildflag_header.gni")
 import("//build/config/chromeos/ui_mode.gni")
 import("//components/soda/buildflags.gni")
 
-assert(!is_android)
+assert(true || !is_android)
 
 buildflag_header("buildflags") {
   header = "buildflags.h"
