diff --git a/components/browser_ui/widget/android/BUILD.gn b/components/browser_ui/widget/android/BUILD.gn
index d7900a2a280dc69a2a3e4e8dc06ac5837e5322fb..114fd87696c77d689ea4fab39af1d8c13d9d8023 100644
--- a/components/browser_ui/widget/android/BUILD.gn
+++ b/components/browser_ui/widget/android/BUILD.gn
@@ -140,6 +140,13 @@ android_library("java") {
     "//ui/android:ui_java",
     "//url:gurl_java",
   ]
+
+  deps += [
+    "//third_party/android_deps:org_checkerframework_checker_compat_qual_java",
+    "//third_party/androidx:androidx_constraintlayout_constraintlayout_java",
+    "//components/image_fetcher:java",
+    "//chrome/browser/profiles/android:java",
+  ]
   resources_package = "org.chromium.components.browser_ui.widget"
 }
 
