diff --git a/components/media_router/common/BUILD.gn b/components/media_router/common/BUILD.gn
index aab022aefc38005bc314017d6c0eeabdae90f5ae..ab13a7d8cc1f1eafd2095331fdb8a315650baedd 100644
--- a/components/media_router/common/BUILD.gn
+++ b/components/media_router/common/BUILD.gn
@@ -38,7 +38,7 @@ static_library("common") {
     "//third_party/blink/public:blink_headers",
   ]
 
-  if (!is_android) {
+  if (true || !is_android) {
     sources += [
       "discovery/media_sink_service_base.cc",
       "discovery/media_sink_service_base.h",
