diff --git a/components/search_engines/search_engine_choice/search_engine_choice_utils.h b/components/search_engines/search_engine_choice/search_engine_choice_utils.h
index d63cdd12494d9a8f140ba90061a3b4f525b66b8a..bd57369112482ad3f20e34f7c7b129a72535648c 100644
--- a/components/search_engines/search_engine_choice/search_engine_choice_utils.h
+++ b/components/search_engines/search_engine_choice/search_engine_choice_utils.h
@@ -300,7 +300,7 @@ void RecordUnexpectedSearchProvider(const TemplateURLData& data);
 void WipeSearchEngineChoicePrefs(PrefService& profile_prefs,
                                  WipeSearchEngineChoiceReason reason);
 
-#if !BUILDFLAG(IS_ANDROID)
+#if true || !BUILDFLAG(IS_ANDROID)
 // Returns the engine marketing snippet string resource id or -1 if the snippet
 // was not found.
 // The function definition is generated in `generated_marketing_snippets.cc`.
