diff --git a/components/zoom/BUILD.gn b/components/zoom/BUILD.gn
index c3c862ffe95d2cdf63b73a5ac25d1e1bdfc9c54a..00cbbe560e81e5b84d06efbee9d2db4aefaa515e 100644
--- a/components/zoom/BUILD.gn
+++ b/components/zoom/BUILD.gn
@@ -2,7 +2,7 @@
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
-assert(!is_ios, "Desktop zoom is not used on iOS platform.")
+#assert(!is_ios, "Desktop zoom is not used on iOS platform.")
 
 static_library("zoom") {
   sources = [
