diff --git a/components/password_manager/core/browser/affiliation/BUILD.gn b/components/password_manager/core/browser/affiliation/BUILD.gn
index 5ad2b2e110860bed7c1edf414533168c1ea0c7a5..e72c9a81443c8e683243001bb0d48145baac0b0d 100644
--- a/components/password_manager/core/browser/affiliation/BUILD.gn
+++ b/components/password_manager/core/browser/affiliation/BUILD.gn
@@ -33,7 +33,7 @@ source_set("affiliation_fetching") {
     "//net",
     "//url",
   ]
-  if (!is_android) {
+  if (true || !is_android) {
     sources += [
       "passkey_affiliation_source_adapter.cc",
       "passkey_affiliation_source_adapter.h",
