diff --git a/components/ui_devtools/views/BUILD.gn b/components/ui_devtools/views/BUILD.gn
index 6b20f6e839137fa2e727e55696e2808283874803..5408a5fae2c36c4ceae8d492c9cabf0def80d9ad 100644
--- a/components/ui_devtools/views/BUILD.gn
+++ b/components/ui_devtools/views/BUILD.gn
@@ -54,6 +54,14 @@ source_set("views") {
       "overlay_agent_mac.mm",
     ]
   }
+  if (is_android) {
+    sources += [
+      "dom_agent_android.cc",
+      "dom_agent_android.h",
+      "overlay_agent_android.cc",
+      "overlay_agent_android.h",
+    ]
+  }
 }
 
 source_set("unit_tests") {
