diff --git a/components/web_modal/BUILD.gn b/components/web_modal/BUILD.gn
index 2a2384b40468ce1c4d2ac87fa32023507e74f36d..2d536aaf7c7aa72c9cd4c7367ec845f731f151c0 100644
--- a/components/web_modal/BUILD.gn
+++ b/components/web_modal/BUILD.gn
@@ -2,7 +2,7 @@
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
-assert(!is_android && !is_ios)
+# assert(!is_android && !is_ios)
 
 component("web_modal") {
   sources = [
