diff --git a/components/commerce/content/browser/BUILD.gn b/components/commerce/content/browser/BUILD.gn
index 46b1f5975b2162a41e7c120f1fcf5da889405e12..0c59d49846ae3a9ab27ef44e265cdd8c75e136cd 100644
--- a/components/commerce/content/browser/BUILD.gn
+++ b/components/commerce/content/browser/BUILD.gn
@@ -17,7 +17,7 @@ static_library("browser") {
   ]
 }
 
-if (!is_android) {
+if (true || !is_android) {
   static_library("hint") {
     sources = [
       "hint/commerce_hint_tab_helper.cc",
