diff --git a/components/history/core/browser/history_types.h b/components/history/core/browser/history_types.h
index 7479a283c5b9993487aa0ea8342631e852726ad1..1012e339b2f433a509e96a392381298d21665d82 100644
--- a/components/history/core/browser/history_types.h
+++ b/components/history/core/browser/history_types.h
@@ -52,6 +52,8 @@ enum VisitSource {
   SOURCE_FIREFOX_IMPORTED = 3,
   SOURCE_IE_IMPORTED = 4,
   SOURCE_SAFARI_IMPORTED = 5,
+  SOURCE_CHROME_IMPORTED = 6,
+  SOURCE_MISES_IMPORTED = 7,
 };
 
 // Corresponds to the "id" column of the "visits" SQL table.
