diff --git a/components/webauthn/core/browser/BUILD.gn b/components/webauthn/core/browser/BUILD.gn
index 243ade7459383485f06c5b34f2e47e526e0072a6..998ea413e36fd1aa5cafff47179604e9dec1857e 100644
--- a/components/webauthn/core/browser/BUILD.gn
+++ b/components/webauthn/core/browser/BUILD.gn
@@ -20,7 +20,7 @@ source_set("browser") {
     ]
   }
 
-  if (!is_android) {
+  if (is_android || !is_android) {
     sources += [
       "passkey_sync_bridge.cc",
       "passkey_sync_bridge.h",
