diff --git a/components/bookmarks/browser/BUILD.gn b/components/bookmarks/browser/BUILD.gn
index fa0a550e020745e01bf824fe57bf06d3b8acc57d..80db16942c12edf1db36408db733013bbf5317c6 100644
--- a/components/bookmarks/browser/BUILD.gn
+++ b/components/bookmarks/browser/BUILD.gn
@@ -79,7 +79,7 @@ static_library("browser") {
     all_dependent_configs = [ "//ui/views:flags" ]
   }
 
-  if (toolkit_views && !is_mac) {
+  if (is_android || toolkit_views && !is_mac) {
     sources += [ "bookmark_node_data_views.cc" ]
   }
 
