diff --git a/components/search/search.cc b/components/search/search.cc
index a7167fae92de59a58aa4770073c8b06e87ae8c76..8d8432e3c5376c774f22d1e7efbc1bf2e022c611 100644
--- a/components/search/search.cc
+++ b/components/search/search.cc
@@ -13,7 +13,7 @@
 namespace search {
 
 bool IsInstantExtendedAPIEnabled() {
-#if BUILDFLAG(IS_IOS) || BUILDFLAG(IS_ANDROID)
+#if BUILDFLAG(IS_IOS) // || BUILDFLAG(IS_ANDROID)
   return false;
 #else
   return true;
