diff --git a/components/permissions/request_type.cc b/components/permissions/request_type.cc
index e90e8ce27cbdcc571eef7a4b315269e6395ab1de..dec9817901edfdca191a605d79689fe9bb211a4e 100644
--- a/components/permissions/request_type.cc
+++ b/components/permissions/request_type.cc
@@ -65,6 +65,8 @@ int GetIconIdAndroid(RequestType type) {
     case RequestType::kStorageAccess:
     case RequestType::kTopLevelStorageAccess:
       return IDR_ANDROID_STORAGE_ACCESS;
+    default:
+      NOTREACHED_IN_MIGRATION();
   }
   NOTREACHED_IN_MIGRATION();
   return 0;
@@ -470,6 +472,9 @@ const char* PermissionKeyForRequestType(permissions::RequestType request_type) {
 #endif
     case permissions::RequestType::kIdentityProvider:
       return "identity_provider";
+    MISES_PERMISSION_KEY_FOR_REQUEST_TYPE
+    default:
+      NOTREACHED();
   }
 
   return nullptr;
