diff --git a/components/content_settings/core/common/content_settings_types.h b/components/content_settings/core/common/content_settings_types.h
index 17c5e6aafb7047867306b8a9e9e715effdbb7c5b..03f6e203d8126b5fd43c1542b90ac64af37a7dd2 100644
--- a/components/content_settings/core/common/content_settings_types.h
+++ b/components/content_settings/core/common/content_settings_types.h
@@ -323,6 +323,8 @@ enum class ContentSettingsType : int32_t {
   // should be enabled.
   ANTI_ABUSE,
 
+  MISES_ETHEREUM,
+  MISES_SOLANA,
   NUM_TYPES,
 };
 
