diff --git a/components/update_client/request_sender.cc b/components/update_client/request_sender.cc
index 0dce4e24b2fd2abb1a01712747e965d65bf05f60..b8715d9db52f4733eed97d156e77abd64ff41be4 100644
--- a/components/update_client/request_sender.cc
+++ b/components/update_client/request_sender.cc
@@ -100,7 +100,7 @@ void RequestSender::SendInternal() {
     url = BuildUpdateUrl(url, request_query_string);
   }
 
-  VLOG(2) << "Sending Omaha request: " << request_body_;
+  LOG(INFO) << "Sending Omaha request: " << request_body_;
 
   network_fetcher_ = config_->GetNetworkFetcherFactory()->Create();
   if (!network_fetcher_) {
@@ -127,7 +127,7 @@ void RequestSender::SendInternalComplete(
     const std::string& response_etag,
     const std::string& response_cup_server_proof,
     int retry_after_sec) {
-  VLOG(2) << "Omaha response received: " << response_body;
+  LOG(INFO) << "Omaha response received: " << response_body;
   VLOG_IF(2, error) << "Omaha send error: " << error;
 
   if (!error) {
