diff --git a/components/media_router/common/BUILD.gn b/components/media_router/common/BUILD.gn
index 4a33857f6563c148d46c33aeaf757572f7ad5a86..24b3c5fcb9e1fd4b4fa4d817f3490476f83bfe2d 100644
--- a/components/media_router/common/BUILD.gn
+++ b/components/media_router/common/BUILD.gn
@@ -41,7 +41,7 @@ static_library("common") {
 
   configs += [ "//build/config/compiler:wexit_time_destructors" ]
 
-  if (!is_android) {
+  if (true || !is_android) {
     sources += [
       "discovery/media_sink_service_base.cc",
       "discovery/media_sink_service_base.h",
