diff --git a/components/commerce/core/BUILD.gn b/components/commerce/core/BUILD.gn
index a01ae9eb3c56fdf14f064c5d2ee49410ebba23c0..f29d7c0fed4655d7c6df439ae1f03ed08b953f5b 100644
--- a/components/commerce/core/BUILD.gn
+++ b/components/commerce/core/BUILD.gn
@@ -84,7 +84,8 @@ if (is_android) {
     proto_in_dir = "//"
     sources = [ "proto/commerce_subscription_db_content.proto" ]
   }
-} else {
+}
+if (true) {
   proto_library("cart_db_content_proto") {
     proto_in_dir = "//"
     sources = [ "proto/cart_db_content.proto" ]
