diff --git a/components/commerce/core/BUILD.gn b/components/commerce/core/BUILD.gn
index e024e19e95cace9156e3dbb15e2a7ff2058b897e..3b63d2e28475781be0e4d2323fb0c2d8187b71e2 100644
--- a/components/commerce/core/BUILD.gn
+++ b/components/commerce/core/BUILD.gn
@@ -86,7 +86,8 @@ if (is_android) {
     proto_in_dir = "//"
     sources = [ "proto/merchant_signal_db_content.proto" ]
   }
-} else {
+}
+if (true) {
   proto_library("cart_db_content_proto") {
     proto_in_dir = "//"
     sources = [ "proto/cart_db_content.proto" ]
