diff --git a/components/strings/BUILD.gn b/components/strings/BUILD.gn
index 8330fc3056b1a4962ea0ace8cbb255dac11510e7..071063b7906173d99ccb04f207b4253dadd6aed2 100644
--- a/components/strings/BUILD.gn
+++ b/components/strings/BUILD.gn
@@ -52,6 +52,7 @@ grit("components_strings") {
   if (is_android) {
     outputs += android_components_strings_java_resources
   }
+  deps = [ "//mises/components/resources:strings" ]
 }
 
 if (is_android) {
