diff --git a/components/content_settings/core/common/content_settings_types.h b/components/content_settings/core/common/content_settings_types.h
index 0a38dcddaf2b0bf50c3c2926b2e5dc29d5bae451..1a4742fefedbdb0a8a8d6b02f0c7b62692d2eaa0 100644
--- a/components/content_settings/core/common/content_settings_types.h
+++ b/components/content_settings/core/common/content_settings_types.h
@@ -369,6 +369,8 @@ enum class ContentSettingsType : int32_t {
   // heuristics.
   TPCD_HEURISTICS_GRANTS,
 
+  MISES_ETHEREUM,
+  MISES_SOLANA,
   NUM_TYPES,
 };
 
