diff --git a/components/browser_ui/widget/android/BUILD.gn b/components/browser_ui/widget/android/BUILD.gn
index 7ac2d9c2371907807a78bad2acd891ece41d7e37..dbae830d2bdf10eb47245449fe066676cdd5ab8f 100644
--- a/components/browser_ui/widget/android/BUILD.gn
+++ b/components/browser_ui/widget/android/BUILD.gn
@@ -145,6 +145,12 @@ android_library("java") {
     "//ui/android:ui_java",
     "//url:gurl_java",
   ]
+
+  deps += [
+    "//third_party/android_deps:org_checkerframework_checker_compat_qual_java",
+    "//third_party/androidx:androidx_constraintlayout_constraintlayout_java",
+    "//components/image_fetcher:java",
+  ]
   resources_package = "org.chromium.components.browser_ui.widget"
 }
 
