diff --git a/components/storage_monitor/BUILD.gn b/components/storage_monitor/BUILD.gn
index 546f194a24bb9d65d2e20eccc963da3dfa313a5a..7c1a884f1ee8c6b23780d8763fe648895b77b971 100644
--- a/components/storage_monitor/BUILD.gn
+++ b/components/storage_monitor/BUILD.gn
@@ -73,7 +73,7 @@ static_library("storage_monitor") {
       ]
       deps += [ "//services/device/public/mojom" ]
     }
-  } else if (is_linux || is_chromeos_lacros) {
+  } else if (is_android || is_linux || is_chromeos_lacros) {
     if (use_udev) {
       sources += [
         "mtab_watcher_linux.cc",
