diff --git a/components/permissions/request_type.h b/components/permissions/request_type.h
index 97aafc80d1013f5c6ec86de2519a27a568fcd8c7..388fcfbb61a7cf1f4de365afcb72d5714ea780ac 100644
--- a/components/permissions/request_type.h
+++ b/components/permissions/request_type.h
@@ -23,24 +23,24 @@ namespace permissions {
 enum class RequestType {
   kAccessibilityEvents,
   kArSession,
-#if !BUILDFLAG(IS_ANDROID)
+#if true || !BUILDFLAG(IS_ANDROID)
   kCameraPanTiltZoom,
 #endif
   kCameraStream,
-#if !BUILDFLAG(IS_ANDROID)
+#if true || !BUILDFLAG(IS_ANDROID)
   kCapturedSurfaceControl,
 #endif
   kClipboard,
   kTopLevelStorageAccess,
   kDiskQuota,
-#if !BUILDFLAG(IS_ANDROID)
+#if true || !BUILDFLAG(IS_ANDROID)
   kFileSystemAccess,
 #endif
   kGeolocation,
   kHandTracking,
   kIdentityProvider,
   kIdleDetection,
-#if !BUILDFLAG(IS_ANDROID)
+#if true || !BUILDFLAG(IS_ANDROID)
   kLocalFonts,
 #endif
   kMicStream,
@@ -50,14 +50,14 @@ enum class RequestType {
   kNfcDevice,
 #endif
   kNotifications,
-#if !BUILDFLAG(IS_ANDROID)
+#if !true || BUILDFLAG(IS_ANDROID)
   kKeyboardLock,
   kPointerLock,
 #endif
 #if BUILDFLAG(IS_ANDROID) || BUILDFLAG(IS_CHROMEOS) || BUILDFLAG(IS_WIN)
   kProtectedMediaIdentifier,
 #endif
-#if !BUILDFLAG(IS_ANDROID)
+#if true || !BUILDFLAG(IS_ANDROID)
   kRegisterProtocolHandler,
 #endif
 #if BUILDFLAG(IS_CHROMEOS)
@@ -65,13 +65,13 @@ enum class RequestType {
 #endif
   kStorageAccess,
   kVrSession,
-#if !BUILDFLAG(IS_ANDROID)
+#if true || !BUILDFLAG(IS_ANDROID)
   kWebAppInstallation,
 #endif
 #if BUILDFLAG(IS_CHROMEOS) && BUILDFLAG(USE_CUPS)
   kWebPrinting,
 #endif
-#if !BUILDFLAG(IS_ANDROID)
+#if true || !BUILDFLAG(IS_ANDROID)
   kWindowManagement,
   kMaxValue = kWindowManagement
 #else
@@ -104,7 +104,7 @@ bool IsConfirmationChipSupported(RequestType for_request_type);
 // Returns the icon to display.
 IconId GetIconId(RequestType type);
 
-#if !BUILDFLAG(IS_ANDROID)
+#if true || !BUILDFLAG(IS_ANDROID)
 // Returns the blocked icon to display.
 IconId GetBlockedIconId(RequestType type);
 #endif
