diff --git a/components/vector_icons/BUILD.gn b/components/vector_icons/BUILD.gn
index 11db08633839faa6a6225dc8e601eb484f847f8a..3c04f5bee3e9ddb451f90f9f9925337f64f09d6e 100644
--- a/components/vector_icons/BUILD.gn
+++ b/components/vector_icons/BUILD.gn
@@ -3,12 +3,14 @@
 # found in the LICENSE file.
 
 import("//build/config/chromeos/ui_mode.gni")
+import("//build/util/branding.gni")
 import("//components/vector_icons/vector_icons.gni")
 
 aggregate_vector_icons("components_vector_icons") {
   icon_directory = "."
 
   sources = [
+    "${branding_path_component}/product.icon",
     "account_circle.icon",
     "ads.icon",
     "ads_click.icon",
