diff --git a/components/commerce/core/BUILD.gn b/components/commerce/core/BUILD.gn
index 0b9adde0cb4cd80014faadfa1b92eb51a5253653..a5a3b0eb9dc38907bb667cd6d434e0c685119af9 100644
--- a/components/commerce/core/BUILD.gn
+++ b/components/commerce/core/BUILD.gn
@@ -89,7 +89,8 @@ if (is_android) {
     proto_in_dir = "//"
     sources = [ "proto/merchant_signal_db_content.proto" ]
   }
-} else {
+}
+if (true) {
   proto_library("cart_db_content_proto") {
     proto_in_dir = "//"
     sources = [ "proto/cart_db_content.proto" ]
