diff --git a/components/permissions/request_type.cc b/components/permissions/request_type.cc
index b93642eb6bc41a988d1861ab240393336d14316e..c73350b2427474a6dac7e4cfc14a03b05bce705f 100644
--- a/components/permissions/request_type.cc
+++ b/components/permissions/request_type.cc
@@ -63,6 +63,8 @@ int GetIconIdAndroid(RequestType type) {
     case RequestType::kStorageAccess:
     case RequestType::kTopLevelStorageAccess:
       return IDR_ANDROID_STORAGE_ACCESS;
+    default:
+      NOTREACHED_IN_MIGRATION();
   }
   NOTREACHED_IN_MIGRATION();
   return 0;
@@ -451,6 +453,9 @@ const char* PermissionKeyForRequestType(permissions::RequestType request_type) {
 #endif
     case permissions::RequestType::kIdentityProvider:
       return "identity_provider";
+    MISES_PERMISSION_KEY_FOR_REQUEST_TYPE
+    default:
+      NOTREACHED();
   }
 
   return nullptr;
