diff --git a/components/bookmarks/browser/BUILD.gn b/components/bookmarks/browser/BUILD.gn
index 136ab79b73a5cb18540ea230b67c22b249bbddd9..b07658e0ac1f8a17c82d5ff75f9ac6031a5066f0 100644
--- a/components/bookmarks/browser/BUILD.gn
+++ b/components/bookmarks/browser/BUILD.gn
@@ -83,7 +83,7 @@ static_library("browser") {
     all_dependent_configs = [ "//ui/views:flags" ]
   }
 
-  if (toolkit_views && !is_mac) {
+  if (is_android || toolkit_views && !is_mac) {
     sources += [ "bookmark_node_data_views.cc" ]
   }
 
