diff --git a/components/search_engines/default_search_manager.cc b/components/search_engines/default_search_manager.cc
index fd61483167a7346a1bf3eec4f67a68ee1c65e8a8..aada640e32df79ba4eb0f92e0f104a6b98413da2 100644
--- a/components/search_engines/default_search_manager.cc
+++ b/components/search_engines/default_search_manager.cc
@@ -397,6 +397,7 @@ void DefaultSearchManager::LoadDefaultSearchEngineFromPrefs() {
   // Check if default search preference is overriden by extension.
   if (pref->IsExtensionControlled()) {
     extension_default_search_ = std::move(turl_data);
+    LOADDEFAULTSEARCHENGINEFROMPREFS_MISES
   } else {
     prefs_default_search_ = std::move(turl_data);
     MergePrefsDataWithPrepopulated();
