diff --git a/components/ui_devtools/views/BUILD.gn b/components/ui_devtools/views/BUILD.gn
index 2b27259f0f2dbe9d380a77eb9d5e3fdc10acd655..f24bd19739b6e2ec2f8adaf93eec7042a3d1ded0 100644
--- a/components/ui_devtools/views/BUILD.gn
+++ b/components/ui_devtools/views/BUILD.gn
@@ -54,6 +54,7 @@ source_set("views") {
       "overlay_agent_mac.mm",
     ]
   }
+  import("//mises/components/ui_devtools/views/sources.gni") sources += mises_components_ui_devtools_views_sources
 }
 
 source_set("unit_tests") {
