diff --git a/components/password_manager/core/browser/features/password_manager_features_util.h b/components/password_manager/core/browser/features/password_manager_features_util.h
index aadf4253e61faf0d889b65b8e3238f7f78647d58..9f7bff0737ba541e530d4409170802f32547ff97 100644
--- a/components/password_manager/core/browser/features/password_manager_features_util.h
+++ b/components/password_manager/core/browser/features/password_manager_features_util.h
@@ -163,7 +163,7 @@ PasswordAccountStorageUsageLevel ComputePasswordAccountStorageUsageLevel(
     const PrefService* pref_service,
     const syncer::SyncService* sync_service);
 
-#if !BUILDFLAG(IS_IOS) && !BUILDFLAG(IS_ANDROID)
+#if true || !BUILDFLAG(IS_IOS) && !BUILDFLAG(IS_ANDROID)
 
 // Sets opt-in to using account storage for passwords for the current
 // signed-in user (unconsented primary account).
