diff --git a/components/soda/BUILD.gn b/components/soda/BUILD.gn
index c445d5d11d9a147db9e656bac47bb5af05315490..b7424743dba2fa4e4faa6662fc73495bb2790360 100644
--- a/components/soda/BUILD.gn
+++ b/components/soda/BUILD.gn
@@ -6,7 +6,7 @@ import("//build/buildflag_header.gni")
 import("//build/config/chromeos/ui_mode.gni")
 import("//components/soda/buildflags.gni")
 
-assert(!is_android)
+assert(true || !is_android)
 
 buildflag_header("buildflags") {
   header = "buildflags.h"
