diff --git a/components/browser_ui/site_settings/android/java/src/org/chromium/components/browser_ui/site_settings/SingleWebsiteSettings.java b/components/browser_ui/site_settings/android/java/src/org/chromium/components/browser_ui/site_settings/SingleWebsiteSettings.java
index 07d15e7ea2e5acec2c43bfce81fcf93063667211..10728239529e7eb7d5c91d0912784548a173bbd2 100644
--- a/components/browser_ui/site_settings/android/java/src/org/chromium/components/browser_ui/site_settings/SingleWebsiteSettings.java
+++ b/components/browser_ui/site_settings/android/java/src/org/chromium/components/browser_ui/site_settings/SingleWebsiteSettings.java
@@ -266,7 +266,13 @@ public class SingleWebsiteSettings extends BaseSiteSettingsFragment
      */
     public static Bundle createFragmentArgsForSite(String url) {
         Bundle fragmentArgs = new Bundle();
-        String origin = Origin.createOrThrow(url).toString();
+        String origin;
+        try {
+            origin = Origin.createOrThrow(url).toString();
+        }
+        catch (IllegalArgumentException e) {
+            origin = "localhost";
+        }    
         fragmentArgs.putSerializable(EXTRA_SITE_ADDRESS, WebsiteAddress.create(origin));
         return fragmentArgs;
     }
