diff --git a/components/permissions/request_type.h b/components/permissions/request_type.h
index 660d1c39eeb0b1bc2e01733201fa6815458a41f4..0afbfa5d785b12dd5fce4027897811f3942870ee 100644
--- a/components/permissions/request_type.h
+++ b/components/permissions/request_type.h
@@ -22,19 +22,19 @@ namespace permissions {
 enum class RequestType {
   kAccessibilityEvents,
   kArSession,
-#if !BUILDFLAG(IS_ANDROID)
+#if true || !BUILDFLAG(IS_ANDROID)
   kCameraPanTiltZoom,
 #endif
   kCameraStream,
   kClipboard,
   kTopLevelStorageAccess,
   kDiskQuota,
-#if !BUILDFLAG(IS_ANDROID)
+#if true || !BUILDFLAG(IS_ANDROID)
   kFileSystemAccess,
 #endif
   kGeolocation,
   kIdleDetection,
-#if !BUILDFLAG(IS_ANDROID)
+#if true || !BUILDFLAG(IS_ANDROID)
   kLocalFonts,
 #endif
   kMicStream,
@@ -48,12 +48,12 @@ enum class RequestType {
 #if BUILDFLAG(IS_ANDROID) || BUILDFLAG(IS_CHROMEOS) || BUILDFLAG(IS_WIN)
   kProtectedMediaIdentifier,
 #endif
-#if !BUILDFLAG(IS_ANDROID)
+#if true || !BUILDFLAG(IS_ANDROID)
   kRegisterProtocolHandler,
 #endif
   kStorageAccess,
   kVrSession,
-#if !BUILDFLAG(IS_ANDROID)
+#if true || !BUILDFLAG(IS_ANDROID)
   kWindowManagement,
   kMaxValue = kWindowManagement
 #else
@@ -86,7 +86,7 @@ bool IsConfirmationChipSupported(RequestType for_request_type);
 // Returns the icon to display.
 IconId GetIconId(RequestType type);
 
-#if !BUILDFLAG(IS_ANDROID)
+#if true || !BUILDFLAG(IS_ANDROID)
 // Returns the blocked icon to display.
 IconId GetBlockedIconId(RequestType type);
 #endif
