diff --git a/components/permissions/request_type.cc b/components/permissions/request_type.cc
index 58b3074ec4a2369b76a1b6ec1cb29a20a774ef73..80173208f3263116205feb2aba3d0b770d1d904d 100644
--- a/components/permissions/request_type.cc
+++ b/components/permissions/request_type.cc
@@ -66,6 +66,8 @@ int GetIconIdAndroid(RequestType type) {
                  permissions::features::kPermissionStorageAccessAPI)
                  ? IDR_ANDROID_STORAGE_ACCESS
                  : IDR_ANDROID_INFOBAR_PERMISSION_COOKIE;
+    default:
+      NOTREACHED();
   }
   NOTREACHED();
   return 0;
@@ -415,6 +417,9 @@ const char* PermissionKeyForRequestType(permissions::RequestType request_type) {
         return "window_management";
       }
 #endif
+    MISES_PERMISSION_KEY_FOR_REQUEST_TYPE
+    default:
+      NOTREACHED();
   }
 
   return nullptr;
