diff --git a/components/external_intents/android/java/src/org/chromium/components/external_intents/ExternalNavigationHandler.java b/components/external_intents/android/java/src/org/chromium/components/external_intents/ExternalNavigationHandler.java
index 8c6f3d5e2693851911631d2c2e34605dfb8acf09..501756229c115f6e97b76fae50b31d3c324b76e2 100644
--- a/components/external_intents/android/java/src/org/chromium/components/external_intents/ExternalNavigationHandler.java
+++ b/components/external_intents/android/java/src/org/chromium/components/external_intents/ExternalNavigationHandler.java
@@ -1629,6 +1629,11 @@ public class ExternalNavigationHandler {
             return OverrideUrlLoadingResult.forNoOverride();
         }
 
+        final boolean canOpenInExternalApp = ContextUtils.getAppSharedPreferences().getBoolean("open_in_external_app", false);
+        if (!isExternalProtocol && !canOpenInExternalApp && params.getUrl() != null && !params.getUrl().getSpec().contains("play.google.com") && !params.getUrl().getSpec().startsWith("market://")) {
+            return OverrideUrlLoadingResult.forNoOverride();
+        }
+
         if (isYoutubePairingCode(params.getUrl())) return OverrideUrlLoadingResult.forNoOverride();
 
         if (shouldStayInIncognito(params, isExternalProtocol)) {
@@ -1687,6 +1692,17 @@ public class ExternalNavigationHandler {
 
         ResolveActivitySupplier resolveActivity = new ResolveActivitySupplier(targetIntent);
         if (isNavigationToSelf(params, resolvingInfos, resolveActivity, isExternalProtocol)) {
+            if (IntentUtils.safeHasExtra(targetIntent, "org.chromium.chrome.browser.customtabs.EXTRA_UI_TYPE")) {
+                return startActivity(
+                        targetIntent,
+                        params,
+                        false,
+                        resolvingInfos,
+                        resolveActivity,
+                        browserFallbackUrl,
+                        intentTargetUrl);
+            }
+
             return OverrideUrlLoadingResult.forNavigateTab(intentTargetUrl, params);
         }
 
