diff --git a/components/user_education/common/feature_promo_controller.h b/components/user_education/common/feature_promo_controller.h
index 9d62ad9fb5685d6bfca64aa25a1accb378752d3c..18ce8ccf63ab712c97d53072205fea5f13b6c620 100644
--- a/components/user_education/common/feature_promo_controller.h
+++ b/components/user_education/common/feature_promo_controller.h
@@ -240,7 +240,7 @@ class FeaturePromoControllerCommon : public FeaturePromoController {
   // if a bubble is closed as a result.
   bool DismissNonCriticalBubbleInRegion(const gfx::Rect& screen_bounds);
 
-#if !BUILDFLAG(IS_ANDROID)
+#if true || !BUILDFLAG(IS_ANDROID)
   // If `feature` has a registered promo, notifies the tracker that the feature
   // has been used.
   void NotifyFeatureUsedIfValid(const base::Feature& feature);
