diff --git a/components/storage_monitor/BUILD.gn b/components/storage_monitor/BUILD.gn
index c6bbb5ef6a8ac4daf496a8e6c9d720878b9f9941..8331d1f9234f46942d6013cced594c724210be6d 100644
--- a/components/storage_monitor/BUILD.gn
+++ b/components/storage_monitor/BUILD.gn
@@ -74,7 +74,7 @@ static_library("storage_monitor") {
       ]
       deps += [ "//services/device/public/mojom" ]
     }
-  } else if (is_linux || is_chromeos_lacros) {
+  } else if (is_android || is_linux || is_chromeos_lacros) {
     if (use_udev) {
       sources += [
         "mtab_watcher_linux.cc",
