diff --git a/components/browser_ui/widget/android/java/src/org/chromium/components/browser_ui/widget/TintedDrawable.java b/components/browser_ui/widget/android/java/src/org/chromium/components/browser_ui/widget/TintedDrawable.java
index 4b4b11bf0aabe8f5809cf4110c080444c4cadfcb..bac76d7693cafb4e2da1aa52beb5a57e326b4122 100644
--- a/components/browser_ui/widget/android/java/src/org/chromium/components/browser_ui/widget/TintedDrawable.java
+++ b/components/browser_ui/widget/android/java/src/org/chromium/components/browser_ui/widget/TintedDrawable.java
@@ -56,7 +56,11 @@ public class TintedDrawable extends BitmapDrawable {
         if (bitmap != null && bitmap.isRecycled()) {
             Log.e(TAG, "Trying to draw with recycled BitmapDrawable.");
         }
-        super.draw(canvas);
+        try {
+            super.draw(canvas);
+        } catch (RuntimeException ex) {
+
+        }
     }
 
     /**
