diff --git a/components/password_manager/core/browser/BUILD.gn b/components/password_manager/core/browser/BUILD.gn
index 5c9fda008526f2b408d97389cac00fde625c3fc4..7d93b9cb7cfd9877d04ec41f5c4eea459ec07b04 100644
--- a/components/password_manager/core/browser/BUILD.gn
+++ b/components/password_manager/core/browser/BUILD.gn
@@ -367,7 +367,7 @@ static_library("browser") {
     ]
   }
 
-  if (!is_android && !is_ios) {
+  if (is_android || !is_android && !is_ios) {
     sources += [
       "ui/weak_check_utility.cc",
       "ui/weak_check_utility.h",
