diff --git a/components/saved_tab_groups/types.h b/components/saved_tab_groups/types.h
index dbdd5997760d50658cb7c7523fdf5b761789d27e..dee1073999e8f00bd3f5ed226e91d94eceb6b9ff 100644
--- a/components/saved_tab_groups/types.h
+++ b/components/saved_tab_groups/types.h
@@ -17,8 +17,8 @@ namespace tab_groups {
 
 // Types for tab and tab group IDs.
 #if BUILDFLAG(IS_ANDROID)
-using LocalTabID = int;
-using LocalTabGroupID = base::Token;
+using LocalTabID = base::Token;
+using LocalTabGroupID = tab_groups::TabGroupId;
 #elif BUILDFLAG(IS_IOS)
 using LocalTabID = int;
 using LocalTabGroupID = tab_groups::TabGroupId;
