diff --git a/components/embedder_support/android/java/src/org/chromium/components/embedder_support/util/UrlConstants.java b/components/embedder_support/android/java/src/org/chromium/components/embedder_support/util/UrlConstants.java
index 0b5a4df40f78c29f7f5a43712d8f0c58f2a43762..ade897dc258e98ec6fa581dd2e9d6c48e7e41d80 100644
--- a/components/embedder_support/android/java/src/org/chromium/components/embedder_support/util/UrlConstants.java
+++ b/components/embedder_support/android/java/src/org/chromium/components/embedder_support/util/UrlConstants.java
@@ -50,7 +50,7 @@ public class UrlConstants {
 
     public static final String NTP_HOST = "newtab";
     public static final String NTP_URL = "chrome-native://newtab/";
-    public static final String NTP_NON_NATIVE_URL = "chrome://newtab/";
+    public static final String NTP_NON_NATIVE_URL = "chrome-search://local-ntp/local-ntp.html";
     public static final String NTP_ABOUT_URL = "about:newtab";
 
     public static final String BOOKMARKS_HOST = "bookmarks";
