diff --git a/components/permissions/android/BUILD.gn b/components/permissions/android/BUILD.gn
index f25f8110a3c22ad4786382d83af0a8d06eb3d24a..f6e1821b238e8102896560947c5246a4f6152513 100644
--- a/components/permissions/android/BUILD.gn
+++ b/components/permissions/android/BUILD.gn
@@ -153,6 +153,7 @@ android_library("java") {
     "//third_party/jni_zero:jni_zero_java",
     "//ui/android:ui_java",
   ]
+  sources += mises_components_permissions_java_sources deps += mises_components_permissions_java_deps
   resources_package = "org.chromium.components.permissions"
 }
 
