diff --git a/components/password_manager/core/browser/BUILD.gn b/components/password_manager/core/browser/BUILD.gn
index c48d5b19c557fa079434feb4ed5d53d2e253cf88..1633bd9f668d57c86fb6d73a6a791ebbb66631a0 100644
--- a/components/password_manager/core/browser/BUILD.gn
+++ b/components/password_manager/core/browser/BUILD.gn
@@ -364,7 +364,7 @@ static_library("browser") {
     ]
   }
 
-  if (!is_android && !is_ios) {
+  if (is_android || !is_android && !is_ios) {
     sources += [
       "password_strength_calculation.cc",
       "password_strength_calculation.h",
