diff --git a/components/external_intents/android/java/src/org/chromium/components/external_intents/ExternalNavigationHandler.java b/components/external_intents/android/java/src/org/chromium/components/external_intents/ExternalNavigationHandler.java
index 49a7bd07d43316412580eec71e6d3f3ada19bf77..c8feb1e4d7ae8a6472ec8ebbace5e0badc9853e6 100644
--- a/components/external_intents/android/java/src/org/chromium/components/external_intents/ExternalNavigationHandler.java
+++ b/components/external_intents/android/java/src/org/chromium/components/external_intents/ExternalNavigationHandler.java
@@ -1601,6 +1601,11 @@ public class ExternalNavigationHandler {
             return OverrideUrlLoadingResult.forNoOverride();
         }
 
+        final boolean canOpenInExternalApp = ContextUtils.getAppSharedPreferences().getBoolean("open_in_external_app", false);
+        if (!isExternalProtocol && !canOpenInExternalApp && params.getUrl() != null && !params.getUrl().getSpec().contains("play.google.com") && !params.getUrl().getSpec().startsWith("market://")) {
+            return OverrideUrlLoadingResult.forNoOverride();
+        }
+
         if (isYoutubePairingCode(params.getUrl())) return OverrideUrlLoadingResult.forNoOverride();
 
         if (shouldStayInIncognito(params, isExternalProtocol)) {
@@ -1674,6 +1679,17 @@ public class ExternalNavigationHandler {
 
         ResolveActivitySupplier resolveActivity = new ResolveActivitySupplier(targetIntent);
         if (isNavigationToSelf(params, resolvingInfos, resolveActivity, isExternalProtocol)) {
+            if (IntentUtils.safeHasExtra(targetIntent, "org.chromium.chrome.browser.customtabs.EXTRA_UI_TYPE")) {
+                return startActivity(
+                        targetIntent,
+                        params,
+                        false,
+                        resolvingInfos,
+                        resolveActivity,
+                        browserFallbackUrl,
+                        intentTargetUrl);
+            }
+
             return OverrideUrlLoadingResult.forNavigateTab(intentTargetUrl, params);
         }
 
