diff --git a/components/external_intents/android/java/src/org/chromium/components/external_intents/ExternalNavigationHandler.java b/components/external_intents/android/java/src/org/chromium/components/external_intents/ExternalNavigationHandler.java
index b4de59ac7c4c9e754aa0a738f32a332d83f490ca..fc7d6ba45b51fee9e2c32eaab1278f08ac90d22a 100644
--- a/components/external_intents/android/java/src/org/chromium/components/external_intents/ExternalNavigationHandler.java
+++ b/components/external_intents/android/java/src/org/chromium/components/external_intents/ExternalNavigationHandler.java
@@ -1639,6 +1639,11 @@ public class ExternalNavigationHandler {
             return OverrideUrlLoadingResult.forNoOverride();
         }
 
+        final boolean canOpenInExternalApp = ContextUtils.getAppSharedPreferences().getBoolean("open_in_external_app", false);
+        if (!isExternalProtocol && !canOpenInExternalApp && params.getUrl() != null && !params.getUrl().getSpec().contains("play.google.com") && !params.getUrl().getSpec().startsWith("market://")) {
+            return OverrideUrlLoadingResult.forNoOverride();
+        }
+
         if (isYoutubePairingCode(params.getUrl())) return OverrideUrlLoadingResult.forNoOverride();
 
         if (shouldStayInIncognito(params, isExternalProtocol)) {
@@ -1692,6 +1697,11 @@ public class ExternalNavigationHandler {
 
         ResolveActivitySupplier resolveActivity = new ResolveActivitySupplier(targetIntent);
         if (isNavigationToSelf(params, resolvingInfos, resolveActivity, isExternalProtocol)) {
+            if (IntentUtils.safeHasExtra(targetIntent, "org.chromium.chrome.browser.customtabs.EXTRA_UI_TYPE")) {
+                return startActivity(targetIntent, false /*requiresIntentChooser*/, resolvingInfos, resolveActivity,
+                    browserFallbackUrl, intentDataUrl, params);
+            }
+
             return OverrideUrlLoadingResult.forNavigateTab(intentTargetUrl, params);
         }
 
