diff --git a/components/media_router/browser/BUILD.gn b/components/media_router/browser/BUILD.gn
index 057b13bb99a66485de35ab715837793177dc6408..eb50c027933bde90be06dbcdeac21ddbd7760118 100644
--- a/components/media_router/browser/BUILD.gn
+++ b/components/media_router/browser/BUILD.gn
@@ -76,7 +76,9 @@ source_set("browser") {
       "android/media_router_dialog_controller_android.h",
     ]
     deps += [ "android:jni_headers" ]
-  } else {
+#  } else {
+  }
+  if (true) {
     sources += [
       "issue_manager.cc",
       "issue_manager.h",
