diff --git a/components/BUILD.gn b/components/BUILD.gn
index 6b3872497da5af46c2967f64283c6acd2cc2253b..430440ad23a08b33bcc657191a4fac8b9b182b3b 100644
--- a/components/BUILD.gn
+++ b/components/BUILD.gn
@@ -698,8 +698,8 @@ test("components_unittests") {
   # Since //chrome depends on //extensions, we also only assert_no_deps on
   # extensions targets for non-lacros builds.
   if (!is_chromeos_lacros) {
-    assert_no_deps = [ "//chrome/*" ]
-    assert_no_deps += disallowed_extension_deps_
+#    assert_no_deps = [ "//chrome/*" ]
+#    assert_no_deps += disallowed_extension_deps_
   }
 
   if (is_ios) {
