diff --git a/components/browser_ui/widget/android/BUILD.gn b/components/browser_ui/widget/android/BUILD.gn
index 8ee2c87821adcba00cc11395a683de474ddb23df..c83bb58178b85e5d7408db770b01ee2d1331d6e7 100644
--- a/components/browser_ui/widget/android/BUILD.gn
+++ b/components/browser_ui/widget/android/BUILD.gn
@@ -139,6 +139,13 @@ android_library("java") {
     "//ui/android:ui_java",
     "//url:gurl_java",
   ]
+
+  deps += [
+    "//third_party/android_deps:org_checkerframework_checker_compat_qual_java",
+    "//third_party/androidx:androidx_constraintlayout_constraintlayout_java",
+    "//components/image_fetcher:java",
+    "//chrome/browser/profiles/android:java",
+  ]
   resources_package = "org.chromium.components.browser_ui.widget"
 }
 
