diff --git a/components/custom_handlers/BUILD.gn b/components/custom_handlers/BUILD.gn
index 475a2be6868df95527fb795fcfbbf6f28865f2f8..10047c4bbf0e6a19b4728c24276801a1eb093d28 100644
--- a/components/custom_handlers/BUILD.gn
+++ b/components/custom_handlers/BUILD.gn
@@ -29,7 +29,7 @@ static_library("custom_handlers") {
     "//third_party/blink/public/common:headers",
   ]
 
-  if (!is_android) {
+  if (true || !is_android) {
     sources += [
       "register_protocol_handler_permission_request.cc",
       "register_protocol_handler_permission_request.h",
