diff --git a/components/permissions/BUILD.gn b/components/permissions/BUILD.gn
index b598c95dc815e19225497736db25196a0f984b04..080ee69d46395a43515c3cadc2e0dfe99f2d5327 100644
--- a/components/permissions/BUILD.gn
+++ b/components/permissions/BUILD.gn
@@ -165,6 +165,7 @@ source_set("permissions") {
   if (!is_android) {
     deps += [ "//components/guest_view/browser" ]
   }
+  import("//mises/components/permissions/sources.gni") sources += mises_components_permissions_sources deps += mises_components_permissions_deps
   if (is_android) {
     sources += [
       "android/android_permission_util.cc",
