diff --git a/components/page_info/page_info.cc b/components/page_info/page_info.cc
index f8738806601f67ae917a13f0b663d64b2973312b..3fc93fdb8a40c32933a05e07463dd6e91bd6cdac 100644
--- a/components/page_info/page_info.cc
+++ b/components/page_info/page_info.cc
@@ -911,6 +911,7 @@ void PageInfo::ComputeUIInputs(const GURL& url) {
   bool is_chrome_ui_native_scheme = false;
 #if BUILDFLAG(IS_ANDROID)
   is_chrome_ui_native_scheme = url.SchemeIs(browser_ui::kChromeUINativeScheme);
+  MISES_COMPONENTS_PAGE_INFO_PAGE_INFO_INSERT
 #endif
 
   if (url.SchemeIs(url::kAboutScheme)) {
