diff --git a/components/soda/BUILD.gn b/components/soda/BUILD.gn
index 30eaba1a32fe3dc681dd9f1969e1192312e0a473..c5862efec082905785784d4b17656d92cc8cc6ae 100644
--- a/components/soda/BUILD.gn
+++ b/components/soda/BUILD.gn
@@ -6,7 +6,7 @@ import("//build/buildflag_header.gni")
 import("//build/config/chromeos/ui_mode.gni")
 import("//components/soda/buildflags.gni")
 
-assert(!is_android)
+assert(true || !is_android)
 
 buildflag_header("buildflags") {
   header = "buildflags.h"
