diff --git a/components/browser_ui/site_settings/android/java/src/org/chromium/components/browser_ui/site_settings/WebsitePreferenceBridge.java b/components/browser_ui/site_settings/android/java/src/org/chromium/components/browser_ui/site_settings/WebsitePreferenceBridge.java
index 06988ed338551022790e41f028b9889a9ae0f910..62ff0688d670c7b24dc57f1669ebcfa985fcee0e 100644
--- a/components/browser_ui/site_settings/android/java/src/org/chromium/components/browser_ui/site_settings/WebsitePreferenceBridge.java
+++ b/components/browser_ui/site_settings/android/java/src/org/chromium/components/browser_ui/site_settings/WebsitePreferenceBridge.java
@@ -396,6 +396,10 @@ public class WebsitePreferenceBridge {
                 .isContentSettingGlobal(
                         browserContextHandle, contentSettingType, primaryUrl, secondaryUrl);
     }
+    public static void setPopupSettingForOrigin(BrowserContextHandle browserContextHandle, String origin, int contentSettingsType, boolean isIncognito) {
+        WebsitePreferenceBridgeJni.get().setPopupSettingForOrigin(browserContextHandle, origin, contentSettingsType, isIncognito);
+    }
+
 
     /**
      * Sets the content setting for the default scope of the url that is appropriate for the given
@@ -623,5 +627,7 @@ public class WebsitePreferenceBridge {
         String toDomainWildcardPattern(String pattern);
 
         String toHostOnlyPattern(String pattern);
+        void setPopupSettingForOrigin(BrowserContextHandle browserContextHandle,
+            String origin, int value, boolean isIncognito);
     }
 }
