diff --git a/components/content_settings/core/browser/content_settings_uma_util.cc b/components/content_settings/core/browser/content_settings_uma_util.cc
index 4c80092a6c62e4943f42b7830ec6da62d3277677..5a925cac215fb4ba8b2f038b0a6fd32c7fb56c3b 100644
--- a/components/content_settings/core/browser/content_settings_uma_util.cc
+++ b/components/content_settings/core/browser/content_settings_uma_util.cc
@@ -148,6 +148,7 @@ constexpr auto kHistogramValue = base::MakeFixedFlatMap<ContentSettingsType,
 
     // As mentioned at the top, please don't forget to update ContentType in
     // enums.xml when you add entries here!
+    MISES_HISTOGRAM_VALUE_LIST
 });
 // LINT.ThenChange(//tools/metrics/histograms/enums.xml:ContentType)
 
