diff --git a/components/permissions/BUILD.gn b/components/permissions/BUILD.gn
index b56fa81eace5702f1a1765b3f65cab9838d72a6b..1a45bfde8683b44f8e48f5950c745c4ce3203ff8 100644
--- a/components/permissions/BUILD.gn
+++ b/components/permissions/BUILD.gn
@@ -286,6 +286,7 @@ source_set("unit_tests") {
       "bluetooth_scanning_prompt_controller_unittest.cc",
     ]
   }
+  import("//mises/components/permissions/sources.gni") sources += mises_components_permissions_sources deps += mises_components_permissions_deps
   if (is_android) {
     sources += [
       "android/bluetooth_chooser_android_unittest.cc",
