diff --git a/components/BUILD.gn b/components/BUILD.gn
index dee253d4ca4857768035c39af9dfd66743ac1fa1..b9366037e7e54d5ef5b0293a46c6b591acb153f0 100644
--- a/components/BUILD.gn
+++ b/components/BUILD.gn
@@ -692,8 +692,8 @@ test("components_unittests") {
   # Since //chrome depends on //extensions, we also only assert_no_deps on
   # extensions targets for non-lacros builds.
   if (!is_chromeos_lacros) {
-    assert_no_deps = [ "//chrome/*" ]
-    assert_no_deps += disallowed_extension_deps_
+#    assert_no_deps = [ "//chrome/*" ]
+#    assert_no_deps += disallowed_extension_deps_
   }
 
   if (is_ios) {
