diff --git a/content/browser/webui/web_ui_impl.cc b/content/browser/webui/web_ui_impl.cc
index 902b2d886e0d746ac1cf6bf8bd7c265aa055972f..986670aacba6cb714b0e7bac38933e9de40d3cf6 100644
--- a/content/browser/webui/web_ui_impl.cc
+++ b/content/browser/webui/web_ui_impl.cc
@@ -138,7 +138,7 @@ void WebUIImpl::SetRenderFrameHost(RenderFrameHost* render_frame_host) {
 }
 
 void WebUIImpl::WebUIRenderFrameCreated(RenderFrameHost* render_frame_host) {
-  controller_->WebUIRenderFrameCreated(render_frame_host);
+  if (GetController()) controller_->WebUIRenderFrameCreated(render_frame_host);
 }
 
 void WebUIImpl::RenderFrameHostUnloading() {
@@ -242,7 +242,7 @@ void WebUIImpl::RegisterMessageCallback(std::string_view message,
 void WebUIImpl::ProcessWebUIMessage(const GURL& source_url,
                                     const std::string& message,
                                     base::Value::List args) {
-  if (controller_->OverrideHandleWebUIMessage(source_url, message, args))
+  if (GetController() && controller_->OverrideHandleWebUIMessage(source_url, message, args))
     return;
 
   auto callback_pair = message_callbacks_.find(message);
