diff --git a/content/browser/BUILD.gn b/content/browser/BUILD.gn
index 4e163a30b7f987b0127cabf84a4610bfb81b9662..9214530a166616291c7d1bd5bc344546e2722f74 100644
--- a/content/browser/BUILD.gn
+++ b/content/browser/BUILD.gn
@@ -3291,6 +3291,8 @@ source_set("browser") {
     }
     libs += [ "jnigraphics" ]
   } else {
+  }
+  if (true) {
     # Not Android.
     sources += [
       # The WebAuthn devtools protocol API is not supported in Android yet.
@@ -3390,6 +3392,22 @@ source_set("browser") {
       deps += [ "//media/mojo/mojom:web_speech_recognition" ]
     }
 
+    if (is_android) {
+      sources -= [
+        "speech/soda_speech_recognition_engine_impl.cc",
+        "media/session/audio_focus_delegate_default.cc", 
+        "network_sandbox.cc",
+        "media/capture/sub_capture_target_id_web_contents_helper.cc",
+      ]
+      deps += [
+        "//mises/browser/android/preferences:features"
+      ]
+    } else {
+      deps += [
+        "//mises/components/ungoogled:ungoogled_switches"
+      ]
+    }
+
     deps += [
       "//components/device_event_log",
       "//components/soda:constants",
@@ -3404,7 +3422,7 @@ source_set("browser") {
     ]
   }
 
-  if (!is_android && !is_ios) {
+  if (is_android || !is_android && !is_ios) {
     sources += [
       # Devtools frontend not included in Android and iOS
       "devtools/devtools_frontend_host_impl.cc",
