diff --git a/content/browser/devtools/BUILD.gn b/content/browser/devtools/BUILD.gn
index fad8c1c626d46d02067ffa4672c2b1699fc66572..24c69ac449685dcbd883c3fd413fffdaffc16299 100644
--- a/content/browser/devtools/BUILD.gn
+++ b/content/browser/devtools/BUILD.gn
@@ -11,7 +11,7 @@ import("//third_party/protobuf/proto_library.gni")
 import("//tools/grit/grit_rule.gni")
 
 # Android and iOS don't support DevTools front-end.
-if (!is_android && !is_ios) {
+if (!is_ios) {
   source_set("devtools_resources_extern") {
     sources = [ "devtools_resources.cc" ]
 
