diff --git a/content/browser/picture_in_picture/document_picture_in_picture_window_controller_impl.cc b/content/browser/picture_in_picture/document_picture_in_picture_window_controller_impl.cc
index 69ab72d4afaab953309a545fbd592be4aba88914..b0561ba652b53c7887e5dbb7b8d48dbc06d71d3e 100644
--- a/content/browser/picture_in_picture/document_picture_in_picture_window_controller_impl.cc
+++ b/content/browser/picture_in_picture/document_picture_in_picture_window_controller_impl.cc
@@ -22,6 +22,7 @@
 
 namespace content {
 
+#if !BUILDFLAG(IS_ANDROID)
 // static
 DocumentPictureInPictureWindowController*
 PictureInPictureWindowController::GetOrCreateDocumentPictureInPictureController(
@@ -29,6 +30,7 @@ PictureInPictureWindowController::GetOrCreateDocumentPictureInPictureController(
   return DocumentPictureInPictureWindowControllerImpl::
       GetOrCreateForWebContents(web_contents);
 }
+#endif
 
 // static
 DocumentPictureInPictureWindowControllerImpl*
