diff --git a/content/public/android/java/src/org/chromium/content/browser/ChildProcessCreationParamsImpl.java b/content/public/android/java/src/org/chromium/content/browser/ChildProcessCreationParamsImpl.java
index 46804b0711a24104d012c641826087d6721149a0..e6fdcd303d7efdeda0f3e19677dbc4c46b77a0ed 100644
--- a/content/public/android/java/src/org/chromium/content/browser/ChildProcessCreationParamsImpl.java
+++ b/content/public/android/java/src/org/chromium/content/browser/ChildProcessCreationParamsImpl.java
@@ -18,7 +18,7 @@ public class ChildProcessCreationParamsImpl {
     private static final String PRIVILEGED_SERVICES_NAME =
             "org.chromium.content.app.PrivilegedProcessService";
     private static final String SANDBOXED_SERVICES_NAME =
-            "org.chromium.content.app.SandboxedProcessService";
+            "org.chromium.content.app.MisesSandboxedProcessService";
 
     // Members should all be immutable to avoid worrying about thread safety.
     private static String sPackageNameForPrivilegedService;
