diff --git a/content/browser/tracing/BUILD.gn b/content/browser/tracing/BUILD.gn
index 9d744a67cfd8f074f9676a1ef7760a9a765ce2a2..efbf9a88d00a830786d9141a5024cb18fd064168 100644
--- a/content/browser/tracing/BUILD.gn
+++ b/content/browser/tracing/BUILD.gn
@@ -5,7 +5,7 @@
 import("//tools/grit/grit_rule.gni")
 import("//ui/webui/resources/tools/generate_grd.gni")
 
-assert(!is_android)
+# assert(!is_android)
 
 # The script just writes filename with no dirs to the .grd, so we always need
 # this file to be in the same directory as the inputs.
