diff --git a/content/public/android/java/src/org/chromium/content/browser/ChildProcessCreationParamsImpl.java b/content/public/android/java/src/org/chromium/content/browser/ChildProcessCreationParamsImpl.java
index 7a9bd887b7ea560697a9dc6416a4400c7fc2d068..a97b6bb475ccd69a4d8b80f3c30cd78a49c9ff27 100644
--- a/content/public/android/java/src/org/chromium/content/browser/ChildProcessCreationParamsImpl.java
+++ b/content/public/android/java/src/org/chromium/content/browser/ChildProcessCreationParamsImpl.java
@@ -16,7 +16,7 @@ public class ChildProcessCreationParamsImpl {
     private static final String PRIVILEGED_SERVICES_NAME =
             "org.chromium.content.app.PrivilegedProcessService";
     private static final String SANDBOXED_SERVICES_NAME =
-            "org.chromium.content.app.SandboxedProcessService";
+            "org.chromium.content.app.MisesSandboxedProcessService";
 
     // Members should all be immutable to avoid worrying about thread safety.
     private static String sPackageNameForPrivilegedService;
