diff --git a/content/browser/webui/web_ui_impl.cc b/content/browser/webui/web_ui_impl.cc
index 371e45c0f4e0682658baee02687ece01ca9193ef..7bf92678697eca72859617d7fefe26bf173a373e 100644
--- a/content/browser/webui/web_ui_impl.cc
+++ b/content/browser/webui/web_ui_impl.cc
@@ -137,7 +137,7 @@ void WebUIImpl::SetRenderFrameHost(RenderFrameHost* render_frame_host) {
 }
 
 void WebUIImpl::WebUIRenderFrameCreated(RenderFrameHost* render_frame_host) {
-  controller_->WebUIRenderFrameCreated(render_frame_host);
+  if (GetController()) controller_->WebUIRenderFrameCreated(render_frame_host);
 }
 
 void WebUIImpl::RenderFrameHostUnloading() {
@@ -241,7 +241,7 @@ void WebUIImpl::RegisterMessageCallback(std::string_view message,
 void WebUIImpl::ProcessWebUIMessage(const GURL& source_url,
                                     const std::string& message,
                                     base::Value::List args) {
-  if (controller_->OverrideHandleWebUIMessage(source_url, message, args))
+  if (GetController() && controller_->OverrideHandleWebUIMessage(source_url, message, args))
     return;
 
   auto callback_pair = message_callbacks_.find(message);
