diff --git a/content/public/browser/web_contents_delegate.h b/content/public/browser/web_contents_delegate.h
index a080fb57f00c712eb8a3a9be669413e1082ce3b3..b142ff809a10506dcb654cc06fef7a2fa8e5c0db 100644
--- a/content/public/browser/web_contents_delegate.h
+++ b/content/public/browser/web_contents_delegate.h
@@ -794,7 +794,7 @@ class CONTENT_EXPORT WebContentsDelegate {
       const std::vector<blink::mojom::DraggableRegionPtr>& regions,
       WebContents* contents) {}
 
-#if !BUILDFLAG(IS_ANDROID)
+#if true || !BUILDFLAG(IS_ANDROID)
   // Whether the WebContents should use per PWA instanced
   // system media controls.
   virtual bool ShouldUseInstancedSystemMediaControls() const;
