diff --git a/content/public/android/generate_child_service.py b/content/public/android/generate_child_service.py
index fa08851c5d7a71911557e3ff218bd6e70991e084..a02fd1316631b13ca278be3949606a5318201dcd 100755
--- a/content/public/android/generate_child_service.py
+++ b/content/public/android/generate_child_service.py
@@ -26,7 +26,7 @@ package org.chromium.content.app;
  * This is needed to register multiple SandboxedProcess services so that we
  * can have more than one sandboxed process.
  */
-public class SandboxedProcessService{0} extends SandboxedProcessService {{
+public class MisesSandboxedProcessService{0} extends SandboxedProcessService {{
 }}"""
   return template.format(str(i))
 
@@ -42,7 +42,7 @@ def DoMain(argv):
   number, output = args
   number = int(number)
 
-  path_template = "org/chromium/content/app/SandboxedProcessService{0}.java"
+  path_template = "org/chromium/content/app/MisesSandboxedProcessService{0}.java"
   with action_helpers.atomic_output(output) as f:
     with zipfile.ZipFile(f, 'w', zipfile.ZIP_STORED) as srcjar:
       for i in range(number):
