diff --git a/content/browser/devtools/BUILD.gn b/content/browser/devtools/BUILD.gn
index e29f1f853c5bfd56045f16e2991936639b880a28..c13c26d1f02344c80c21356b47757cac00dd78df 100644
--- a/content/browser/devtools/BUILD.gn
+++ b/content/browser/devtools/BUILD.gn
@@ -10,7 +10,7 @@ import("//third_party/protobuf/proto_library.gni")
 import("//tools/grit/grit_rule.gni")
 
 # Android does not support DevTools front-end.
-if (!is_android) {
+if (true || !is_android) {
   source_set("devtools_resources_extern") {
     sources = [ "devtools_resources.cc" ]
 
