diff --git a/content/browser/BUILD.gn b/content/browser/BUILD.gn
index 62a5aa48fdf02da4951202f62c74305b1c279dc1..81a4d27bdeafb7fa62877d387248ee90345ffdfc 100644
--- a/content/browser/BUILD.gn
+++ b/content/browser/BUILD.gn
@@ -3095,6 +3095,8 @@ source_set("browser") {
     }
     libs += [ "jnigraphics" ]
   } else {
+  }
+  if (true) {
     # Not Android.
     sources += [
       # The WebAuthn devtools protocol API is not supported in Android yet.
@@ -3169,6 +3171,10 @@ source_set("browser") {
       "picture_in_picture/document_picture_in_picture_window_controller_impl.h",
     ]
 
+    if (is_android) {
+      sources -= ["media/session/audio_focus_delegate_default.cc", "media/capture/crop_id_web_contents_helper.cc","network_sandbox.cc",]
+    }
+
     deps += [
       "//components/speech:speech",
       "//components/vector_icons",
