diff --git a/content/browser/devtools/BUILD.gn b/content/browser/devtools/BUILD.gn
index 85be27d2909d6ea4e5ca5582498f5b6b5c9a6d70..807c74433cd34272c050f76f13174fd901f35acb 100644
--- a/content/browser/devtools/BUILD.gn
+++ b/content/browser/devtools/BUILD.gn
@@ -10,7 +10,7 @@ import("//third_party/protobuf/proto_library.gni")
 import("//tools/grit/grit_rule.gni")
 
 # Android does not support DevTools front-end.
-if (!is_android) {
+if (true || !is_android) {
   source_set("devtools_resources_extern") {
     sources = [ "devtools_resources.cc" ]
 
