diff --git a/content/renderer/render_thread_impl.cc b/content/renderer/render_thread_impl.cc
index 83daef460d5dd9b6b908485da1c807ec94a00f3f..23c134a906ed86d9a034c8c6141411c4b33b790b 100644
--- a/content/renderer/render_thread_impl.cc
+++ b/content/renderer/render_thread_impl.cc
@@ -907,6 +907,7 @@ void RenderThreadImpl::InitializeRenderer(
 }
 
 void RenderThreadImpl::RegisterSchemes() {
+  REGISTER_MISES_SCHEMES_DISPLAY_ISOLATED_AND_NO_JS
   // chrome:
   WebString chrome_scheme(WebString::FromASCII(kChromeUIScheme));
   WebSecurityPolicy::RegisterURLSchemeAsDisplayIsolated(chrome_scheme);
