diff --git a/crypto/apple_keychain_mac.cc b/crypto/apple_keychain_mac.cc
index 2571641c40173e2d1e6ff66f9cf39f0fe0acb0ca..58420a8d058abe069479c06be0f56c33a638ef98 100644
--- a/crypto/apple_keychain_mac.cc
+++ b/crypto/apple_keychain_mac.cc
@@ -41,6 +41,12 @@ class OptionalOutParameter {
 
 }  // namespace
 
+// Much of the Keychain API was marked deprecated as of the macOS 13 SDK.
+// Removal of its use is tracked in https://crbug.com/1348251 but deprecation
+// warnings are disabled in the meanwhile.
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wdeprecated-declarations"
+
 namespace crypto {
 
 AppleKeychain::AppleKeychain() = default;
@@ -109,4 +115,6 @@ ScopedKeychainUserInteractionAllowed::~ScopedKeychainUserInteractionAllowed() {
   }
 }
 
+#pragma clang diagnostic pop
+
 }  // namespace crypto
