diff --git a/DEPS b/DEPS
index d6b071667e8b14c4a2d078a3575ff38bca2f4384..6c37d09a77b26e7eb8b3a27c551c20898ef94dbc 100644
--- a/DEPS
+++ b/DEPS
@@ -1537,7 +1537,7 @@ deps = {
     Var('chromium_git') + '/webm/libwebm.git' + '@' + 'e4fbea0c9751ae8aa86629b197a28d8276a2b0da',
 
   'src/third_party/libwebp/src':
-    Var('chromium_git') + '/webm/libwebp.git' + '@' +  'fd7b5d48464475408d32d2611bdb6947d4246b97',
+    Var('chromium_git') + '/webm/libwebp.git' + '@' +  '4619a48fc3292743d7ce9658bee4245406734109',
 
   'src/third_party/libyuv':
     Var('chromium_git') + '/libyuv/libyuv.git' + '@' + '2bdc210be9eb11ded16bf3ef1f6cadb0d4dcb0c2',
