diff --git a/device/fido/BUILD.gn b/device/fido/BUILD.gn
index 44da37096aab855f046173b9c0ffb4962676166d..ea425c458e61f709906a5865ae718de5132a7283 100644
--- a/device/fido/BUILD.gn
+++ b/device/fido/BUILD.gn
@@ -103,7 +103,7 @@ component("fido") {
     ]
 
     # Android implementation of FIDO is delegated to GMSCore.
-    if (!is_android) {
+    if (true || !is_android) {
       sources += [
         "aoa/android_accessory_device.cc",
         "aoa/android_accessory_device.h",
