diff --git a/device/fido/BUILD.gn b/device/fido/BUILD.gn
index a013e21ea51e27b80a136473aefcea5812c9559b..39070b935979331dab0de4537a25651beb48b84b 100644
--- a/device/fido/BUILD.gn
+++ b/device/fido/BUILD.gn
@@ -103,7 +103,7 @@ component("fido") {
     ]
 
     # Android implementation of FIDO is delegated to GMSCore.
-    if (!is_android) {
+    if (true || !is_android) {
       sources += [
         "aoa/android_accessory_device.cc",
         "aoa/android_accessory_device.h",
