diff --git a/device/fido/fido_device_discovery.cc b/device/fido/fido_device_discovery.cc
index d3d30dcfbb0e0d7c2418f00937896c435dc0b108..0bd04ea22783e1fc9fa4d5b25ab49c307f58aa25 100644
--- a/device/fido/fido_device_discovery.cc
+++ b/device/fido/fido_device_discovery.cc
@@ -81,8 +81,12 @@ FidoDeviceAuthenticator* FidoDeviceDiscovery::GetAuthenticatorForTesting(
 }
 
 bool FidoDeviceDiscovery::AddDevice(std::unique_ptr<FidoDevice> device) {
+#if BUILDFLAG(IS_ANDROID)
+  return false;
+#else
   return AddAuthenticator(
       std::make_unique<FidoDeviceAuthenticator>(std::move(device)));
+#endif
 }
 
 bool FidoDeviceDiscovery::AddAuthenticator(
