diff --git a/extensions/BUILD.gn b/extensions/BUILD.gn
index bcdc39564584e16ed4fb29f29fcf6a5533e94ed9..57abf4c1ceb5512e4086b97ca2dd041f1eafc77c 100644
--- a/extensions/BUILD.gn
+++ b/extensions/BUILD.gn
@@ -10,6 +10,10 @@ import("//ui/base/ui_features.gni")
 
 assert(enable_extensions_core)
 
+if (is_android) {
+  import("//build/config/android/rules.gni")
+}
+
 if (is_component_build) {
   component("extensions") {
     visibility = [
@@ -260,6 +264,27 @@ if (enable_extensions) {
       "//ui/resources",
       "//ui/strings",
     ]
+
+    if (is_android) {
+      sources -= [
+        "$root_gen_dir/mojo/public/js/mojo_bindings_resources.pak",
+        "$root_gen_dir/third_party/blink/public/resources/blink_resources.pak",
+        "$root_gen_dir/third_party/blink/public/resources/blink_scaled_resources_100_percent.pak",
+        "$root_gen_dir/third_party/blink/public/strings/blink_strings_en-US.pak",
+        "$root_gen_dir/ui/resources/ui_resources_100_percent.pak",
+        "$root_gen_dir/ui/strings/app_locale_settings_en-US.pak",
+        "$root_gen_dir/ui/strings/ax_strings_en-US.pak",
+        "$root_gen_dir/ui/strings/ui_strings_en-US.pak",
+      ]
+      deps -= [
+        "//mojo/public/js:resources",
+        "//third_party/blink/public:resources",
+        "//third_party/blink/public:scaled_resources_100_percent",
+        "//third_party/blink/public/strings",
+        "//ui/resources",
+        "//ui/strings",
+      ]
+    }
   }
 
   test("extensions_unittests") {
@@ -303,6 +328,25 @@ if (enable_extensions) {
     if (is_chromeos) {
       deps += [ "//chrome/browser/chromeos/extensions/telemetry/api:extensions_unittests" ]
     }
+  if (is_android) {
+    deps -= [":extensions_resources",]
+    deps += [
+      "//content/public/android:content_java",
+      "//content/public/browser",
+      "//gin:v8_snapshot_assets",
+      "//net",
+      "//services/data_decoder/public/cpp/android:safe_json_java",
+      "//ui/android:ui_java",
+      "//ui/base",
+      "//ui/gfx",
+    ]
+    data_deps += [
+      ":shell_and_test_pak",
+      "//third_party/mesa_headers",
+      "//ui/resources:ui_test_pak_data",
+    ]
+  }
+
   }
 
   test("extensions_browsertests") {
