diff --git a/extensions/common/features/feature_provider_unittest.cc b/extensions/common/features/feature_provider_unittest.cc
index c7abdea5c94d5832ef046d913a6e41f73cf2fb54..bb211ea7afbaed8281682ae5d42d41016d442abc 100644
--- a/extensions/common/features/feature_provider_unittest.cc
+++ b/extensions/common/features/feature_provider_unittest.cc
@@ -127,7 +127,7 @@ TEST(FeatureProviderTest, PermissionFeatureAvailability) {
   // NOT_FOUND_IN_ALLOWLIST.
   // TODO(https://crbug.com/1251347): Port //device/bluetooth to Fuchsia to
   // enable bluetooth extensions.
-#if !BUILDFLAG(IS_FUCHSIA)
+#if !BUILDFLAG(IS_FUCHSIA) && !BUILDFLAG(IS_ANDROID)
   feature = provider->GetFeature("bluetoothPrivate");
   ASSERT_TRUE(feature);
   EXPECT_EQ(Feature::NOT_FOUND_IN_ALLOWLIST,
