diff --git a/extensions/browser/event_router.cc b/extensions/browser/event_router.cc
index a57d1337a443d4371597a311aa585009d8d6593b..ea93dea308f023767f4be4654f9afc355d41c9be 100644
--- a/extensions/browser/event_router.cc
+++ b/extensions/browser/event_router.cc
@@ -44,6 +44,7 @@
 #include "extensions/common/permissions/permissions_data.h"
 #include "ipc/ipc_channel_proxy.h"
 #include "third_party/abseil-cpp/absl/types/optional.h"
+#include "chrome/browser/extensions/api/tabs/tabs_windows_api.h"
 
 using content::BrowserContext;
 using content::BrowserThread;
@@ -901,6 +902,7 @@ void EventRouter::DispatchEventWithLazyListener(const std::string& extension_id,
 void EventRouter::DispatchEventImpl(const std::string& restrict_to_extension_id,
                                     const GURL& restrict_to_url,
                                     std::unique_ptr<Event> event) {
+  //LOG(INFO) << "[Mises] EventRouter::DispatchEventImpl: " << event->event_name;
   DCHECK(event);
   // We don't expect to get events from a completely different browser context.
   DCHECK(!event->restrict_to_browser_context ||
@@ -1297,6 +1299,7 @@ void EventRouter::AddFilterToEvent(const std::string& event_name,
 
 void EventRouter::OnExtensionLoaded(content::BrowserContext* browser_context,
                                     const Extension* extension) {
+  extensions::TabsWindowsAPI::Get(browser_context);                                    
   // TODO(richardzh): revisit here once we create separate lazy listeners for
   // regular and incognito(split) context. How do we ensure lazy listeners and
   // regular listeners are loaded for both browser context.
