diff --git a/extensions/browser/api/system_cpu/BUILD.gn b/extensions/browser/api/system_cpu/BUILD.gn
index 2f39c29f7fbb0ebf3657e811cf7e5dac7a880a33..84d9cee1356677e9822efa6f01452d652f1e2c60 100644
--- a/extensions/browser/api/system_cpu/BUILD.gn
+++ b/extensions/browser/api/system_cpu/BUILD.gn
@@ -40,6 +40,10 @@ source_set("system_cpu") {
     sources += [ "cpu_info_provider_win.cc" ]
   }
 
+  if (is_android) {
+    sources += [ "//mises/chromium_src/extensions/browser/api/system_cpu/cpu_info_provider_android.cc" ]
+  }
+
   if (is_chromeos) {
     deps += [ "//chromeos/system" ]
   }
