diff --git a/extensions/browser/sandboxed_unpacker.cc b/extensions/browser/sandboxed_unpacker.cc
index 6eb55b282d4d74b4b5480252a357bde4c0ac1511..1f9d06f7bbaf198ad999e0611403d75b3858fd71 100644
--- a/extensions/browser/sandboxed_unpacker.cc
+++ b/extensions/browser/sandboxed_unpacker.cc
@@ -441,6 +441,10 @@ void SandboxedUnpacker::UnzipDone(const base::FilePath& zip_file,
                   l10n_util::GetStringUTF16(IDS_EXTENSION_PACKAGE_UNZIP_ERROR));
     return;
   }
+#if BUILDFLAG(IS_ANDROID)
+  extension_root_ = unzip_dir;
+#endif
+  
   base::FilePath verified_contents_path =
       file_util::GetVerifiedContentsPath(extension_root_);
   // If the verified contents are already present in the _metadata folder, we
@@ -522,7 +526,7 @@ void SandboxedUnpacker::StoreVerifiedContentsInExtensionDir(
 
 void SandboxedUnpacker::Unpack(const base::FilePath& directory) {
   DCHECK(unpacker_io_task_runner_->RunsTasksInCurrentSequence());
-  DCHECK(directory.DirName() == temp_dir_.GetPath());
+  //DCHECK(directory.DirName() == temp_dir_.GetPath());
 
   base::FilePath manifest_path = extension_root_.Append(kManifestFilename);
 
