diff --git a/extensions/browser/renderer_startup_helper.cc b/extensions/browser/renderer_startup_helper.cc
index bafd0ab1168cfa2ebea7072e80535a3d7a948f22..630b7f50226b60f761af9fd0b10559843559e757 100644
--- a/extensions/browser/renderer_startup_helper.cc
+++ b/extensions/browser/renderer_startup_helper.cc
@@ -157,6 +157,11 @@ void RendererStartupHelper::InitializeProcess(
   renderer->SetDeveloperMode(
       GetCurrentDeveloperMode(util::GetBrowserContextId(browser_context_)));
 
+  renderer->SetDefaultEVMWallet(
+      GetDefaultEVMWalletID(util::GetBrowserContextId(browser_context_)), 
+      GetDefaultEVMWalletKeyProperty(util::GetBrowserContextId(browser_context_))
+  );
+
   // Extensions need to know the channel and the session type for API
   // restrictions. The values are sent to all renderers, as the non-extension
   // renderers may have content scripts.
