diff --git a/extensions/renderer/native_extension_bindings_system_unittest.cc b/extensions/renderer/native_extension_bindings_system_unittest.cc
index c5de3564518050682ec6e86003f0e1769711599b..560df1bcc2bebd4972914ae706486ad2f8870eab 100644
--- a/extensions/renderer/native_extension_bindings_system_unittest.cc
+++ b/extensions/renderer/native_extension_bindings_system_unittest.cc
@@ -917,6 +917,7 @@ TEST_F(NativeExtensionBindingsSystemUnittest, UnmanagedEvents) {
   ::testing::Mock::VerifyAndClearExpectations(ipc_message_sender());
 }
 
+#if !BUILDFLAG(IS_ANDROID)
 // Tests that a context having access to an aliased API (like networking.onc)
 // does not allow for accessing the source API (networkingPrivate) directly.
 TEST_F(NativeExtensionBindingsSystemUnittest,
@@ -1017,6 +1018,7 @@ TEST_F(NativeExtensionBindingsSystemUnittest, AliasedAPIsAreDifferentObjects) {
       &equal));
   EXPECT_FALSE(equal);
 }
+#endif
 
 // Tests that script can overwrite the value of an API.
 TEST_F(NativeExtensionBindingsSystemUnittest, CanOverwriteAPIs) {
