diff --git a/extensions/renderer/script_injection_manager.cc b/extensions/renderer/script_injection_manager.cc
index 4974008cd130101daa322eb7b6506b78343351c6..849a03c9041d3dd9ab9968d2d40de12c2912875f 100644
--- a/extensions/renderer/script_injection_manager.cc
+++ b/extensions/renderer/script_injection_manager.cc
@@ -417,6 +417,7 @@ void ScriptInjectionManager::TryToInject(
   // has blocked, add it to the list of running injections.
   // The Unretained below is safe because this object owns all the
   // ScriptInjections, so is guaranteed to outlive them.
+    LOG(INFO) << "ScriptInjectionManager::TryToInject " << injection->host_id().id;
   switch (injection->TryToInject(
       run_location, scripts_run_info,
       base::BindOnce(&ScriptInjectionManager::OnInjectionStatusUpdated,
