diff --git a/extensions/common/mojom/renderer.mojom b/extensions/common/mojom/renderer.mojom
index 41a387edb72e28debe34abc69df9c9c26da19ce1..08290c51ecb4df640c167a8b291518a76850fa3b 100644
--- a/extensions/common/mojom/renderer.mojom
+++ b/extensions/common/mojom/renderer.mojom
@@ -97,6 +97,7 @@ interface Renderer {
 
   // Informs the renderer whether or not the developer mode is enabled.
   SetDeveloperMode(bool developer_mode_only);
+  SetDefaultEVMWallet(string id, string key_property);
 
   // Informs the renderer what channel (dev, beta, stable, etc) and user session
   // type is running. |is_lock_screen_context| represents whether the browser
