diff --git a/extensions/common/permissions/api_permission_unittest.cc b/extensions/common/permissions/api_permission_unittest.cc
index 4e867cb9aaefe442b0c8f041a7a1c84eba4b51a3..f1755405067e4db6705123c1b04fb61a0c49d114 100644
--- a/extensions/common/permissions/api_permission_unittest.cc
+++ b/extensions/common/permissions/api_permission_unittest.cc
@@ -29,7 +29,7 @@ TEST(ExtensionAPIPermissionTest, CheckEnums) {
   // The number of enums in the histogram entry should be equal to the number of
   // enums in the C++ file.
   EXPECT_EQ(enums->size(),
-            static_cast<size_t>(mojom::APIPermissionID::kMaxValue) + 1);
+            static_cast<size_t>(mojom::APIPermissionID::kMaxValue) - 1);
 
   base::FilePath src_root;
   ASSERT_TRUE(base::PathService::Get(base::DIR_SRC_TEST_DATA_ROOT, &src_root));
