diff --git a/extensions/common/features/feature_provider_unittest.cc b/extensions/common/features/feature_provider_unittest.cc
index 848b57fe32731625762071d70c6af19f5106f3da..761ed976b0f560383a3e1c35b1639a02c09408d6 100644
--- a/extensions/common/features/feature_provider_unittest.cc
+++ b/extensions/common/features/feature_provider_unittest.cc
@@ -124,7 +124,7 @@ TEST(FeatureProviderTest, PermissionFeatureAvailability) {
   // NOT_FOUND_IN_ALLOWLIST.
   // TODO(https://crbug.com/1251347): Port //device/bluetooth to Fuchsia to
   // enable bluetooth extensions.
-#if !BUILDFLAG(IS_FUCHSIA)
+#if !BUILDFLAG(IS_FUCHSIA) && !BUILDFLAG(IS_ANDROID)
   feature = provider->GetFeature("bluetoothPrivate");
   ASSERT_TRUE(feature);
   EXPECT_EQ(Feature::NOT_FOUND_IN_ALLOWLIST,
