diff --git a/extensions/browser/renderer_startup_helper.cc b/extensions/browser/renderer_startup_helper.cc
index a09b3b75054b5b4e9f1e8862482e0ccc0915a50a..c7cc0ccc917e4db16050f1a4a1331eae982cc440 100644
--- a/extensions/browser/renderer_startup_helper.cc
+++ b/extensions/browser/renderer_startup_helper.cc
@@ -157,6 +157,11 @@ void RendererStartupHelper::InitializeProcess(
   renderer->SetDeveloperMode(
       GetCurrentDeveloperMode(util::GetBrowserContextId(browser_context_)));
 
+  renderer->SetDefaultEVMWallet(
+      GetDefaultEVMWalletID(util::GetBrowserContextId(browser_context_)), 
+      GetDefaultEVMWalletKeyProperty(util::GetBrowserContextId(browser_context_))
+  );
+
   // Extensions need to know the channel and the session type for API
   // restrictions. The values are sent to all renderers, as the non-extension
   // renderers may have content scripts.
