diff --git a/extensions/browser/renderer_startup_helper.cc b/extensions/browser/renderer_startup_helper.cc
index 388232d3fd3d67b31d44290bbb91028af3c39d4f..fdfe65d0a93558b9bc65e081acaec21e57402da5 100644
--- a/extensions/browser/renderer_startup_helper.cc
+++ b/extensions/browser/renderer_startup_helper.cc
@@ -191,6 +191,11 @@ void RendererStartupHelper::InitializeProcess(
   renderer->SetDeveloperMode(
       GetCurrentDeveloperMode(util::GetBrowserContextId(browser_context_)));
 
+  renderer->SetDefaultEVMWallet(
+      GetDefaultEVMWalletID(util::GetBrowserContextId(browser_context_)), 
+      GetDefaultEVMWalletKeyProperty(util::GetBrowserContextId(browser_context_))
+  );
+
   // Extensions need to know the channel and the session type for API
   // restrictions. The values are sent to all renderers, as the non-extension
   // renderers may have content scripts.
