diff --git a/extensions/common/command_unittest.cc b/extensions/common/command_unittest.cc
index 1e2316045752a6c6fcfaf68f817bafc59bb36f26..4d4c2767c5769411512b8d8c0e062b22a7e98b72 100644
--- a/extensions/common/command_unittest.cc
+++ b/extensions/common/command_unittest.cc
@@ -237,7 +237,7 @@ TEST(CommandTest, ExtensionCommandParsingFallback) {
 #elif BUILDFLAG(IS_LINUX)
   ui::Accelerator accelerator(ui::VKEY_L,
                               ui::EF_SHIFT_DOWN | ui::EF_CONTROL_DOWN);
-#elif BUILDFLAG(IS_FUCHSIA)
+#elif BUILDFLAG(IS_FUCHSIA) || BUILDFLAG(IS_ANDROID)
   // TODO(crbug.com/1312215): Change this once we decide on a unique platform
   // key for Fuchsia.
   ui::Accelerator accelerator(ui::VKEY_L,
