diff --git a/extensions/shell/test/shell_test_launcher_delegate.cc b/extensions/shell/test/shell_test_launcher_delegate.cc
index 913f75f8c27662447187aac072ad3842853c5a8c..6457832ea5cdbb22579f0a306896e020a8726944 100644
--- a/extensions/shell/test/shell_test_launcher_delegate.cc
+++ b/extensions/shell/test/shell_test_launcher_delegate.cc
@@ -31,10 +31,11 @@ std::string
 AppShellTestLauncherDelegate::GetUserDataDirectoryCommandLineSwitch() {
   return switches::kContentShellUserDataDir;
 }
-
+#if !BUILDFLAG(IS_ANDROID)
 content::ContentMainDelegate*
 AppShellTestLauncherDelegate::CreateContentMainDelegate() {
   return new TestShellMainDelegate();
 }
+#endif
 
 }  // namespace extensions
