diff --git a/extensions/BUILD.gn b/extensions/BUILD.gn
index 2b7dfb97986a0e5c3672d26de7f0a7dc5f64c0d2..6c9cf1df0ddd5b1c85b5cab8802160538d7021a8 100644
--- a/extensions/BUILD.gn
+++ b/extensions/BUILD.gn
@@ -10,6 +10,10 @@ import("//ui/base/ui_features.gni")
 
 assert(enable_extensions_core)
 
+if (is_android) {
+  import("//build/config/android/rules.gni")
+}
+
 if (is_component_build) {
   component("extensions") {
     visibility = [
@@ -265,6 +269,27 @@ repack("shell_and_test_pak") {
       "//device/bluetooth/strings",
       "//extensions/shell:resources",
     ]
+
+    if (is_android) {
+      sources -= [
+        "$root_gen_dir/mojo/public/js/mojo_bindings_resources.pak",
+        "$root_gen_dir/third_party/blink/public/resources/blink_resources.pak",
+        "$root_gen_dir/third_party/blink/public/resources/blink_scaled_resources_100_percent.pak",
+        "$root_gen_dir/third_party/blink/public/strings/blink_strings_en-US.pak",
+        "$root_gen_dir/ui/resources/ui_resources_100_percent.pak",
+        "$root_gen_dir/ui/strings/app_locale_settings_en-US.pak",
+        "$root_gen_dir/ui/strings/ax_strings_en-US.pak",
+        "$root_gen_dir/ui/strings/ui_strings_en-US.pak",
+      ]
+      deps -= [
+        "//mojo/public/js:resources",
+        "//third_party/blink/public:resources",
+        "//third_party/blink/public:scaled_resources_100_percent",
+        "//third_party/blink/public/strings",
+        "//ui/resources",
+        "//ui/strings",
+      ]
+    }
   }
 }
 
@@ -421,5 +446,6 @@ if (enable_extensions) {
     if (is_chromeos_ash) {
       deps += [ "//components/user_manager:test_support" ]
     }
+
   }
 }
