diff --git a/extensions/shell/test/shell_test_launcher_delegate.h b/extensions/shell/test/shell_test_launcher_delegate.h
index aa867e7e158885dac091ceb07cf851f08b35728b..c7a2bc6e427e9c613b3803b7d2785d88ec3749a4 100644
--- a/extensions/shell/test/shell_test_launcher_delegate.h
+++ b/extensions/shell/test/shell_test_launcher_delegate.h
@@ -13,7 +13,9 @@ class AppShellTestLauncherDelegate : public content::TestLauncherDelegate {
  public:
   int RunTestSuite(int argc, char** argv) override;
   std::string GetUserDataDirectoryCommandLineSwitch() override;
+#if !BUILDFLAG(IS_ANDROID)
   content::ContentMainDelegate* CreateContentMainDelegate() override;
+#endif
 };
 
 }  // namespace extensions
