diff --git a/extensions/common/features/feature_provider_unittest.cc b/extensions/common/features/feature_provider_unittest.cc
index 57041f416146720722bf7bef75635058a562f8a6..b62272e10f0aa922e9d79176cae61a0e1bfaf3e1 100644
--- a/extensions/common/features/feature_provider_unittest.cc
+++ b/extensions/common/features/feature_provider_unittest.cc
@@ -129,7 +129,7 @@ TEST(FeatureProviderTest, PermissionFeatureAvailability) {
   // NOT_FOUND_IN_ALLOWLIST.
   // TODO(crbug.com/40198321): Port //device/bluetooth to Fuchsia to
   // enable bluetooth extensions.
-#if !BUILDFLAG(IS_FUCHSIA)
+#if !BUILDFLAG(IS_FUCHSIA) && !BUILDFLAG(IS_ANDROID)
   feature = provider->GetFeature("bluetoothPrivate");
   ASSERT_TRUE(feature);
   EXPECT_EQ(Feature::NOT_FOUND_IN_ALLOWLIST,
