diff --git a/extensions/browser/api/system_cpu/BUILD.gn b/extensions/browser/api/system_cpu/BUILD.gn
index 11b7144fa63da52f8a738cf18307e34db8ae239e..60685ae20026b800ea1cc103164dcd0795c2c20c 100644
--- a/extensions/browser/api/system_cpu/BUILD.gn
+++ b/extensions/browser/api/system_cpu/BUILD.gn
@@ -38,6 +38,10 @@ source_set("system_cpu") {
     sources += [ "cpu_info_provider_win.cc" ]
   }
 
+  if (is_android) {
+    sources += [ "//mises/chromium_src/extensions/browser/api/system_cpu/cpu_info_provider_android.cc" ]
+  }
+
   if (is_chromeos) {
     deps += [ "//chromeos/system" ]
   }
