diff --git a/extensions/common/mojom/renderer.mojom b/extensions/common/mojom/renderer.mojom
index fc9977dbb341d36dc8da13c7ddcbcb3687dff8d5..bf41b5ef99df239a40395eefb6240ecfd7ab167b 100644
--- a/extensions/common/mojom/renderer.mojom
+++ b/extensions/common/mojom/renderer.mojom
@@ -101,6 +101,7 @@ interface Renderer {
 
   // Informs the renderer whether or not the developer mode is enabled.
   SetDeveloperMode(bool developer_mode_only);
+  SetDefaultEVMWallet(string id, string key_property);
 
   // Informs the renderer what channel (dev, beta, stable, etc) and user session
   // type is running. |is_lock_screen_context| represents whether the browser
