diff --git a/extensions/BUILD.gn b/extensions/BUILD.gn
index eb5b6dc41f187b7b82257aa5c7588f498861e771..180b052c2c12856ba6759585226960c5c7c5a2ba 100644
--- a/extensions/BUILD.gn
+++ b/extensions/BUILD.gn
@@ -240,6 +240,7 @@ repack("shell_and_test_pak") {
   ]
 }
 
+if (!is_android) {
 test("extensions_unittests") {
   use_xvfb = use_xvfb_in_this_config
 
@@ -290,6 +291,7 @@ test("extensions_unittests") {
     ]
   }
 }
+}
 
 test("extensions_browsertests") {
   use_xvfb = use_xvfb_in_this_config
