diff --git a/extensions/browser/zipfile_installer.cc b/extensions/browser/zipfile_installer.cc
index 8783c13d1b6b16669b6893a82bcb57b698d54e28..4650ff2ad6b47de512f5730ebca5b75f9c51faec 100644
--- a/extensions/browser/zipfile_installer.cc
+++ b/extensions/browser/zipfile_installer.cc
@@ -29,6 +29,7 @@
 namespace extensions {
 
 namespace {
+[[clang::no_destroy]]static base::FilePath additional_path;
 
 constexpr char kExtensionHandlerTempDirError[] =
     "Could not create temporary directory for zipped extension.";
@@ -200,10 +201,12 @@ void ZipFileInstaller::ManifestUnzipped(const base::FilePath& unzip_dir,
         : ReportFailure(kExtensionHandlerFileUnzipError);
     return;
   }
+  base::FilePath new_unzip_dir;
+  new_unzip_dir = unzip_dir.Append(additional_path);
 
   io_task_runner_->PostTaskAndReplyWithResult(
       FROM_HERE,
-      base::BindOnce(&ReadFileContent, unzip_dir.Append(kManifestFilename)),
+      base::BindOnce(&ReadFileContent, new_unzip_dir.Append(kManifestFilename)),
       base::BindOnce(&ZipFileInstaller::ManifestRead, this, unzip_dir));
 }
 
@@ -276,7 +279,9 @@ void ZipFileInstaller::UnzipDone(const base::FilePath& unzip_dir,
     return;
   }
 
-  std::move(done_callback_).Run(zip_file_, unzip_dir, std::string());
+  base::FilePath new_unzip_dir;
+  new_unzip_dir = unzip_dir.Append(additional_path);
+  std::move(done_callback_).Run(zip_file_, new_unzip_dir, std::string());
 }
 
 void ZipFileInstaller::ReportFailure(const std::string& error) {
@@ -304,6 +309,12 @@ bool ZipFileInstaller::ShouldExtractFile(bool is_theme,
 // static
 bool ZipFileInstaller::IsManifestFile(const base::FilePath& file_path) {
   CHECK(!file_path.IsAbsolute());
+#if BUILDFLAG(IS_ANDROID)
+  if (base::FilePath::CompareEqualIgnoreCase(file_path.BaseName().value(), kManifestFilename)) {
+    additional_path = file_path.DirName();
+    return true;
+  }
+#endif
   return base::FilePath::CompareEqualIgnoreCase(file_path.value(),
                                                 kManifestFilename);
 }
