diff --git a/ios/chrome/app/BUILD.gn b/ios/chrome/app/BUILD.gn
index 52fe2a5587fd536c0a250ffc00ca0d21ebada838..8a6d38318fa6a5e12ca1f1614de0dc06f5ec566a 100644
--- a/ios/chrome/app/BUILD.gn
+++ b/ios/chrome/app/BUILD.gn
@@ -147,6 +147,7 @@ if (!is_fat_secondary_toolchain) {
     info_plists = [
       "resources/Info.plist",
       "resources/ChromeAddition+Info.plist",
+      "//mises/ios/app/resources/Mises+Info.plist",
     ]
     if (ios_enable_background_mode) {
       info_plists += [ "resources/Background+Info.plist" ]
@@ -171,6 +172,10 @@ if (!is_fat_secondary_toolchain) {
     args = [
       "--branding=$chromium_short_name",
       "--add-gtm-metadata=1",
+      "--version-overrides=MAJOR=1",
+      "--version-overrides=MINOR=0",
+      "--version-overrides=BUILD=2",
+      "--version-overrides=PATCH=0",
     ]
   }
 
@@ -686,7 +691,8 @@ ios_app_bundle("chrome") {
 
   deps = [
     ":main",
-
+    "//mises/ios/app",
+    "//mises/ios/third_party/mises",
     # Use the provider API controlled by args.gn.
     ios_provider_target,
   ]
@@ -754,8 +760,7 @@ ios_app_bundle("chrome") {
   # (see https://crbug.com/1394553 for details of the build failure). This
   # is a stop-gap solution.
   if (target_environment == "catalyst" ||
-      ios_provider_target !=
-      "//ios/chrome/browser/providers:chromium_providers") {
+      !ios_chrome_links_with_lottie_framework) {
     assert_no_deps += [
       "//ios/third_party/lottie:lottie+bundle",
       "//ios/third_party/lottie:lottie+link",
