diff --git a/ios/chrome/common/app_group/app_group_helper.mm b/ios/chrome/common/app_group/app_group_helper.mm
index 04bb2f01054f73977063d6990dfe26b4c06614d5..d6df9c77a3691b3428b5a038b5165b2f911d0f72 100644
--- a/ios/chrome/common/app_group/app_group_helper.mm
+++ b/ios/chrome/common/app_group/app_group_helper.mm
@@ -14,7 +14,7 @@
   NSBundle* bundle = base::apple::FrameworkBundle();
   NSString* group = [bundle objectForInfoDictionaryKey:@"KSApplicationGroup"];
   if (![group length]) {
-    return [NSString stringWithFormat:@"group.%s.chrome",
+    return [NSString stringWithFormat:@"group.%s.browser",
                                       BUILDFLAG(IOS_APP_BUNDLE_ID_PREFIX), nil];
   }
   return group;
