diff --git a/ios/chrome/browser/ui/default_promo/default_browser_utils.mm b/ios/chrome/browser/ui/default_promo/default_browser_utils.mm
index 54b002ddfefb4b43aae608140de4c3025aa811c6..b8f40503aba2292b004e7fb4e1cc3b7f14cb855e 100644
--- a/ios/chrome/browser/ui/default_promo/default_browser_utils.mm
+++ b/ios/chrome/browser/ui/default_promo/default_browser_utils.mm
@@ -562,6 +562,9 @@ bool IsChromeLikelyDefaultBrowser() {
 }
 
 bool IsLikelyInterestedDefaultBrowserUser(DefaultPromoType promo_type) {
+  if (promo_type == DefaultPromoTypeGeneral) {
+    return NO;
+  }
   std::vector<base::Time> times = LoadTimestampsForPromoType(promo_type);
   return !times.empty();
 }
