diff --git a/ios/chrome/browser/ui/browser_view/browser_coordinator.mm b/ios/chrome/browser/ui/browser_view/browser_coordinator.mm
index e6828650f4ac81eeef41f7567fa78f4da537ee0d..c277961dc05d2f884c3c7e40a508535d8129293d 100644
--- a/ios/chrome/browser/ui/browser_view/browser_coordinator.mm
+++ b/ios/chrome/browser/ui/browser_view/browser_coordinator.mm
@@ -381,7 +381,7 @@ @interface BrowserCoordinator () <BrowserCoordinatorCommands,
     SafariDownloadCoordinator* SafariDownloadCoordinator;
 
 // Coordinator for Safe Browsing.
-@property(nonatomic, strong) SafeBrowsingCoordinator* safeBrowsingCoordinator;
+//@property(nonatomic, strong) SafeBrowsingCoordinator* safeBrowsingCoordinator;
 
 // Coordinator for sharing scenarios.
 @property(nonatomic, strong) SharingCoordinator* sharingCoordinator;
@@ -1021,10 +1021,10 @@ - (void)startChildCoordinators {
       initWithBaseViewController:self.viewController
                          browser:self.browser];
 
-  self.safeBrowsingCoordinator = [[SafeBrowsingCoordinator alloc]
-      initWithBaseViewController:self.viewController
-                         browser:self.browser];
-  [self.safeBrowsingCoordinator start];
+//  self.safeBrowsingCoordinator = [[SafeBrowsingCoordinator alloc]
+//      initWithBaseViewController:self.viewController
+//                         browser:self.browser];
+//  [self.safeBrowsingCoordinator start];
 
   self.textFragmentsCoordinator = [[TextFragmentsCoordinator alloc]
       initWithBaseViewController:self.viewController
@@ -1153,8 +1153,8 @@ - (void)stopChildCoordinators {
   [self.sadTabCoordinator disconnect];
   self.sadTabCoordinator = nil;
 
-  [self.safeBrowsingCoordinator stop];
-  self.safeBrowsingCoordinator = nil;
+//  [self.safeBrowsingCoordinator stop];
+//  self.safeBrowsingCoordinator = nil;
 
   [self.sharingCoordinator stop];
   self.sharingCoordinator = nil;
