diff --git a/ios/chrome/browser/ui/toolbar/public/toolbar_constants.mm b/ios/chrome/browser/ui/toolbar/public/toolbar_constants.mm
index 089a1af531426821d5c4652bcfbff981dec074da..4473838a3a7120b7a32cbf26eb3d72cf6129f434 100644
--- a/ios/chrome/browser/ui/toolbar/public/toolbar_constants.mm
+++ b/ios/chrome/browser/ui/toolbar/public/toolbar_constants.mm
@@ -13,7 +13,7 @@
 const CGFloat kTopButtonsBottomMargin = 3.0f;
 const CGFloat kBottomButtonsTopMargin = 0.0f;
 const CGFloat kAdaptiveToolbarMargin = 10.0f;
-const CGFloat kAdaptiveToolbarStackViewSpacing = 11.0f;
+const CGFloat kAdaptiveToolbarStackViewSpacing = 1.0f;
 
 const CGFloat kProgressBarHeight = 2.0f;
 
