diff --git a/ios/chrome/browser/ui/toolbar/public/toolbar_constants.h b/ios/chrome/browser/ui/toolbar/public/toolbar_constants.h
index 061c7c3402d9f404b48706e201760b0149213daf..e1a886ad2ebc5810453e27f85b3bfd7f7b7ce095 100644
--- a/ios/chrome/browser/ui/toolbar/public/toolbar_constants.h
+++ b/ios/chrome/browser/ui/toolbar/public/toolbar_constants.h
@@ -109,4 +109,8 @@ extern const CGFloat kLocationBarTintBlue;
 // Font sizes used in omnibox and location bar.
 extern const CGFloat kLocationBarFontSize;
 
+
+extern const CGFloat kAdaptiveToolbarIconSize;
+extern NSString* const kToolbarMisesButtonIdentifier;
+
 #endif  // IOS_CHROME_BROWSER_UI_TOOLBAR_PUBLIC_TOOLBAR_CONSTANTS_H_
