diff --git a/ios/chrome/browser/ui/popup_menu/cells/popup_menu_tools_item.mm b/ios/chrome/browser/ui/popup_menu/cells/popup_menu_tools_item.mm
index 82cc22ea25089a9b054fb009660d4b2980897cd6..fbd92e7e598b215262a24a4370f673321de91900 100644
--- a/ios/chrome/browser/ui/popup_menu/cells/popup_menu_tools_item.mm
+++ b/ios/chrome/browser/ui/popup_menu/cells/popup_menu_tools_item.mm
@@ -252,7 +252,7 @@ NSString* const kToolsMenuTextBadgeAccessibilityIdentifier =
 - (UIColor*)contentColor {
   if (self.destructiveAction)
     return [UIColor colorNamed:kRedColor];
-  return [UIColor colorNamed:kBlueColor];
+  return [UIColor colorNamed:kGrey900Color];
 }
 
 - (void)registerForContentSizeUpdates {
