diff --git a/ios/chrome/browser/ui/toolbar/secondary_toolbar_coordinator.mm b/ios/chrome/browser/ui/toolbar/secondary_toolbar_coordinator.mm
index 7c463af2703d432389bf21cc64564f8b43fef9b5..3d965a717f006d3c65308d13f616c4ad1279db0f 100644
--- a/ios/chrome/browser/ui/toolbar/secondary_toolbar_coordinator.mm
+++ b/ios/chrome/browser/ui/toolbar/secondary_toolbar_coordinator.mm
@@ -15,9 +15,12 @@
 #import "ios/chrome/browser/ui/toolbar/adaptive_toolbar_coordinator+subclassing.h"
 #import "ios/chrome/browser/ui/toolbar/secondary_toolbar_mediator.h"
 #import "ios/chrome/browser/ui/toolbar/secondary_toolbar_view_controller.h"
+#import "mises/ios/browser/ui/toolbar/mises_toolbar_coordinator.h"
+
 
 @interface SecondaryToolbarCoordinator ()
 @property(nonatomic, strong) SecondaryToolbarViewController* viewController;
+@property(nonatomic, strong) MisesToolbarCoordinator* misesToolbarCoordinator;
 @end
 
 @implementation SecondaryToolbarCoordinator {
@@ -47,6 +50,9 @@
   self.viewController.fullscreenController = controller;
   self.viewController.toolbarHeightDelegate = self.toolbarHeightDelegate;
 
+  self.misesToolbarCoordinator = [[MisesToolbarCoordinator alloc] initWithBrowser:self.browser];
+  self.misesToolbarCoordinator.delegate = self.viewController;
+  [self.misesToolbarCoordinator start];
   [super start];
 }
 
@@ -63,4 +69,14 @@
   [self.viewController triggerToolbarSlideInAnimationFromBelow:YES];
 }
 
+- (void)stop {
+  [super stop];
+  [self.misesToolbarCoordinator stop];
+}
+- (void) activate {
+
+  [self.misesToolbarCoordinator activate];
+
+}
+
 @end
