diff --git a/ios/web/web_state/ui/crw_web_controller.mm b/ios/web/web_state/ui/crw_web_controller.mm
index 09d35b85c67f4cf268d9f78fc65bfc434d767978..b00c39b13be73e3192442c990accbcc2e45d3b18 100644
--- a/ios/web/web_state/ui/crw_web_controller.mm
+++ b/ios/web/web_state/ui/crw_web_controller.mm
@@ -2265,4 +2265,7 @@ - (void)resetInjectedWebViewContentView {
   return self.navigationHandler.navigationState;
 }
 
+- (WKWebView *)wkWebView {
+  return self.webView;
+}
 @end
