diff --git a/ios/chrome/browser/ui/util/pasteboard_util.mm b/ios/chrome/browser/ui/util/pasteboard_util.mm
index a69ef743790f666d1b811c93911ffe1a5fd65ce8..26046de548e6cb1d3ba16ad5bf50706879a0ca55 100644
--- a/ios/chrome/browser/ui/util/pasteboard_util.mm
+++ b/ios/chrome/browser/ui/util/pasteboard_util.mm
@@ -77,3 +77,16 @@ void StoreInPasteboard(NSString* text, const GURL& URL) {
 void ClearPasteboard() {
   UIPasteboard.generalPasteboard.items = @[];
 }
+
+
+void StoreTextInPasteboard(NSString* text) {
+   if (!text) {
+    return;
+  }
+  NSDictionary* copiedText = @{
+    UTTypeText.identifier : text,
+    UTTypeUTF8PlainText.identifier : [text dataUsingEncoding:NSUTF8StringEncoding],
+  };
+
+  UIPasteboard.generalPasteboard.items = @[ copiedText ];
+}
