diff --git a/ios/web/web_state/ui/crw_web_controller.mm b/ios/web/web_state/ui/crw_web_controller.mm
index b0ce65cd4e763d17061dc159d7a9b051aac006c6..f3bf8ac709adf4a2774111c987c6a80876eaaf0c 100644
--- a/ios/web/web_state/ui/crw_web_controller.mm
+++ b/ios/web/web_state/ui/crw_web_controller.mm
@@ -1955,4 +1955,7 @@ CrFullscreenState CrFullscreenStateFromWKFullscreenState(
   return self.navigationHandler.navigationState;
 }
 
+- (WKWebView *)wkWebView {
+  return self.webView;
+}
 @end
