diff --git a/ios/chrome/browser/ui/commands/popup_menu_commands.h b/ios/chrome/browser/ui/commands/popup_menu_commands.h
index ecdbc1fc40b08fd9255964525b7a37865b017060..721fa3f6aa65fbe3f905beb57e96d1828bd32dcb 100644
--- a/ios/chrome/browser/ui/commands/popup_menu_commands.h
+++ b/ios/chrome/browser/ui/commands/popup_menu_commands.h
@@ -35,7 +35,7 @@ typedef NS_ENUM(NSInteger, PopupMenuCommandType) {
 // Shows a snackbar that allows the user to UNDO its pin/unpin action.
 - (void)showSnackbarForPinnedState:(BOOL)pinnedState
                           webState:(web::WebState*)webState;
-
+- (void)showMisesMenuPopup;
 @end
 
 #endif  // IOS_CHROME_BROWSER_UI_COMMANDS_POPUP_MENU_COMMANDS_H_
