diff --git a/ios/chrome/browser/ui/toolbar/secondary_toolbar_view_controller.mm b/ios/chrome/browser/ui/toolbar/secondary_toolbar_view_controller.mm
index de66115369be6fd959edb857a65f8f8bfbabfd50..13d6fa50ae36981577b805861c41dad5c817f5ab 100644
--- a/ios/chrome/browser/ui/toolbar/secondary_toolbar_view_controller.mm
+++ b/ios/chrome/browser/ui/toolbar/secondary_toolbar_view_controller.mm
@@ -166,4 +166,12 @@
   }
 }
 
+
+#pragma mark - MisesToolbarDelegate
+
+- (void)updateMisesAvatar:(UIImage*)image {
+    if ([self.view respondsToSelector:@selector(updateMisesAvatar:)]) {
+        [self.view performSelector:@selector(updateMisesAvatar:) withObject:image];
+    }
+}
 @end
