diff --git a/ios/chrome/browser/ui/popup_menu/popup_menu_coordinator.mm b/ios/chrome/browser/ui/popup_menu/popup_menu_coordinator.mm
index f633f3e792aaa015691016a30c68773f4fd06751..de4c08723285326e961fea91d83684777e096003 100644
--- a/ios/chrome/browser/ui/popup_menu/popup_menu_coordinator.mm
+++ b/ios/chrome/browser/ui/popup_menu/popup_menu_coordinator.mm
@@ -226,6 +226,7 @@ - (void)showToolsMenuPopup {
   tableViewController.baseViewController = self.baseViewController;
   tableViewController.tableView.accessibilityIdentifier =
       kPopupMenuToolsMenuTableViewId;
+  MISES_POPUP_MENU_COORDINATOR_INSERT_TYPE
 
   self.viewController = tableViewController;
 
