diff --git a/ios/chrome/browser/net/BUILD.gn b/ios/chrome/browser/net/BUILD.gn
index 3ea46f6bb7459496100c39a0a4e0635520ffff85..cbe102ea5bd606f9baf82f8a849ae49c245236b3 100644
--- a/ios/chrome/browser/net/BUILD.gn
+++ b/ios/chrome/browser/net/BUILD.gn
@@ -42,6 +42,7 @@ source_set("net") {
     ":crurl",
     ":net_types",
   ]
+  import("//mises/chromium_src/ios/chrome/browser/net/sources.gni") deps += mises_chromium_src_ios_chrome_browser_net_deps
 }
 
 source_set("net_types") {
