diff --git a/ios/chrome/browser/ui/toolbar/buttons/toolbar_button_actions_handler.h b/ios/chrome/browser/ui/toolbar/buttons/toolbar_button_actions_handler.h
index 7e67be79b548d139c8059ee68d026418c8d8cee3..f69b1d9dce6c091935d6c620d010566ff718452d 100644
--- a/ios/chrome/browser/ui/toolbar/buttons/toolbar_button_actions_handler.h
+++ b/ios/chrome/browser/ui/toolbar/buttons/toolbar_button_actions_handler.h
@@ -60,6 +60,8 @@ class WebNavigationBrowserAgent;
 // Action when the button to cancel the omnibox focus is tapped.
 - (void)cancelOmniboxFocusAction;
 
+- (void)misesMenuAction;
+
 @end
 
 #endif  // IOS_CHROME_BROWSER_UI_TOOLBAR_BUTTONS_TOOLBAR_BUTTON_ACTIONS_HANDLER_H_
