diff --git a/ios/chrome/browser/ui/content_suggestions/cells/content_suggestions_return_to_recent_tab_view.mm b/ios/chrome/browser/ui/content_suggestions/cells/content_suggestions_return_to_recent_tab_view.mm
index 9dcc63622a2ebf66cc722627bd54707bfabe185c..913806ac00defc102d87c7a202d0051897302793 100644
--- a/ios/chrome/browser/ui/content_suggestions/cells/content_suggestions_return_to_recent_tab_view.mm
+++ b/ios/chrome/browser/ui/content_suggestions/cells/content_suggestions_return_to_recent_tab_view.mm
@@ -42,7 +42,7 @@ const CGFloat kIconWidth = 32.0f;
     _titleLabel.isAccessibilityElement = NO;
     _titleLabel.font = [UIFont preferredFontForTextStyle:UIFontTextStyleBody];
     _titleLabel.adjustsFontForContentSizeCategory = YES;
-    _titleLabel.textColor = [UIColor colorNamed:kTextPrimaryColor];
+    _titleLabel.textColor = [UIColor whiteColor];
     _titleLabel.backgroundColor = UIColor.clearColor;
 
     _subtitleLabel = [[UILabel alloc] init];
@@ -50,7 +50,7 @@ const CGFloat kIconWidth = 32.0f;
         [UIFont preferredFontForTextStyle:UIFontTextStyleFootnote];
     _subtitleLabel.lineBreakMode = NSLineBreakByTruncatingMiddle;
     _subtitleLabel.adjustsFontForContentSizeCategory = YES;
-    _subtitleLabel.textColor = [UIColor colorNamed:kTextSecondaryColor];
+    _subtitleLabel.textColor = [UIColor grayColor];
     _subtitleLabel.backgroundColor = UIColor.clearColor;
 
     UIStackView* textStackView = [[UIStackView alloc]
