diff --git a/ios/chrome/browser/ui/browser_view/browser_coordinator.mm b/ios/chrome/browser/ui/browser_view/browser_coordinator.mm
index d7fb4f0d11b477f82b4595ddbf8bd5504be32e33..275db3bc10822aee6aa73fb2604d055df8f0c3ed 100644
--- a/ios/chrome/browser/ui/browser_view/browser_coordinator.mm
+++ b/ios/chrome/browser/ui/browser_view/browser_coordinator.mm
@@ -457,7 +457,7 @@ @interface BrowserCoordinator () <
     SafariDownloadCoordinator* SafariDownloadCoordinator;
 
 // Coordinator for Safe Browsing.
-@property(nonatomic, strong) SafeBrowsingCoordinator* safeBrowsingCoordinator;
+//@property(nonatomic, strong) SafeBrowsingCoordinator* safeBrowsingCoordinator;
 
 // Coordinator for sharing scenarios.
 @property(nonatomic, strong) SharingCoordinator* sharingCoordinator;
@@ -1209,10 +1209,10 @@ - (void)startChildCoordinators {
 
   /* SharingCoordinator is created and started by an ActivityServiceCommand */
 
-  self.safeBrowsingCoordinator = [[SafeBrowsingCoordinator alloc]
-      initWithBaseViewController:self.viewController
-                         browser:self.browser];
-  [self.safeBrowsingCoordinator start];
+//  self.safeBrowsingCoordinator = [[SafeBrowsingCoordinator alloc]
+//      initWithBaseViewController:self.viewController
+//                         browser:self.browser];
+//  [self.safeBrowsingCoordinator start];
 
   self.textFragmentsCoordinator = [[TextFragmentsCoordinator alloc]
       initWithBaseViewController:self.viewController
@@ -1337,8 +1337,8 @@ - (void)stopChildCoordinators {
   [self.sadTabCoordinator disconnect];
   self.sadTabCoordinator = nil;
 
-  [self.safeBrowsingCoordinator stop];
-  self.safeBrowsingCoordinator = nil;
+//  [self.safeBrowsingCoordinator stop];
+//  self.safeBrowsingCoordinator = nil;
 
   [self.sharingCoordinator stop];
   self.sharingCoordinator = nil;
@@ -3076,7 +3076,8 @@ - (CGFloat)headerInsetForOverscrollActionsController:
   OverscrollActionsTabHelper* activeTabHelper =
       OverscrollActionsTabHelper::FromWebState(activeWebState);
   if (controller == activeTabHelper->GetOverscrollActionsController()) {
-    return self.viewController.headerHeight;
+   // return self.viewController.headerHeight;
+    return 0.0;
   } else {
     return 0.0;
   }
