diff --git a/ios/chrome/browser/ui/content_suggestions/content_suggestions_mediator_util.mm b/ios/chrome/browser/ui/content_suggestions/content_suggestions_mediator_util.mm
index a4687f46d016823ffe6370cff4393a4e4295bcf5..22f580e85e22cc4322c08e3e3f10947968e26fcc 100644
--- a/ios/chrome/browser/ui/content_suggestions/content_suggestions_mediator_util.mm
+++ b/ios/chrome/browser/ui/content_suggestions/content_suggestions_mediator_util.mm
@@ -64,6 +64,8 @@
   suggestion.source = tile.source;
   suggestion.titleSource = tile.title_source;
   suggestion.accessibilityTraits = UIAccessibilityTraitButton;
+    
+  suggestion.iconURL = tile.favicon_url;
 
   return suggestion;
 }
@@ -92,3 +94,18 @@
   return [[ContentSuggestionsMostVisitedActionItem alloc]
       initWithCollectionShortcutType:NTPCollectionShortcutTypeWhatsNew];
 }
+
+
+
+ContentSuggestionsSectionInformation* MisesSectionInformation() {
+  return EmptySectionInfo(ContentSuggestionsSectionMises);
+}
+
+
+ContentSuggestionsSectionInformation* MisesBoxMarginInformation() {
+  return EmptySectionInfo(ContentSuggestionsSectionMisesBox);
+}
+
+ContentSuggestionsSectionInformation* MisesWeb3siteSectionInformation() {
+  return EmptySectionInfo(ContentSuggestionsSectionMisesWeb3site);
+}
