diff --git a/ios/chrome/browser/ui/popup_menu/overflow_menu/feature_flags.mm b/ios/chrome/browser/ui/popup_menu/overflow_menu/feature_flags.mm
index 7da5ded90a843cc3dbde6487d8a3329d4146da1d..402cb600cb6e97e971120bd6eb6c1cf8887e8206 100644
--- a/ios/chrome/browser/ui/popup_menu/overflow_menu/feature_flags.mm
+++ b/ios/chrome/browser/ui/popup_menu/overflow_menu/feature_flags.mm
@@ -8,7 +8,7 @@
 
 BASE_FEATURE(kNewOverflowMenu,
              "NewOverflowMenu",
-             base::FEATURE_ENABLED_BY_DEFAULT);
+             base::FEATURE_DISABLED_BY_DEFAULT);
 
 BASE_FEATURE(kOverflowMenuCustomization,
              "OverflowMenuCustomization",
