diff --git a/ios/chrome/browser/ui/content_suggestions/cells/content_suggestions_tile_constants.h b/ios/chrome/browser/ui/content_suggestions/cells/content_suggestions_tile_constants.h
index 5c54d50ff50aa99c1d5e0a09a6d5b9f439110495..812b2b59dc63bca3ba81aa769a244aa1cc7828f6 100644
--- a/ios/chrome/browser/ui/content_suggestions/cells/content_suggestions_tile_constants.h
+++ b/ios/chrome/browser/ui/content_suggestions/cells/content_suggestions_tile_constants.h
@@ -10,7 +10,7 @@
 // The minimum index value of the Bookmarks Shortcut content in the order behind
 // the four Most Visited tiles. NTPCollectionShortcutType is used as a proxy for
 // index value of the Shortcuts content.
-const int kShortcutMinimumIndex = 4;
+const int kShortcutMinimumIndex = 8;
 
 // Enum listing the collection shortcuts on NTP and similar surfaces.
 typedef NS_ENUM(NSInteger, NTPCollectionShortcutType) {
