diff --git a/ios/chrome/browser/ui/toolbar/primary_toolbar_coordinator.mm b/ios/chrome/browser/ui/toolbar/primary_toolbar_coordinator.mm
index 08d8ebf568a75642f431e8bc839c62747697c989..ddb7ae0ae6758ac1d9d981d0b6c3c0b635032108 100644
--- a/ios/chrome/browser/ui/toolbar/primary_toolbar_coordinator.mm
+++ b/ios/chrome/browser/ui/toolbar/primary_toolbar_coordinator.mm
@@ -38,6 +38,8 @@
 #import "ios/components/webui/web_ui_url_constants.h"
 #import "ios/web/public/navigation/referrer.h"
 
+#import "mises/ios/browser/ui/toolbar/mises_toolbar_coordinator.h"
+
 #if !defined(__has_feature) || !__has_feature(objc_arc)
 #error "This file requires ARC support."
 #endif
@@ -64,6 +66,8 @@
 // Whether the omnibox is currently focused.
 @property(nonatomic, assign) BOOL locationBarFocused;
 
+@property(nonatomic, strong) MisesToolbarCoordinator* misesToolbarCoordinator;
+
 @end
 
 @implementation PrimaryToolbarCoordinator
@@ -124,6 +128,11 @@
   _prerenderService = PrerenderServiceFactory::GetForBrowserState(
       self.browser->GetBrowserState());
 
+    
+      self.misesToolbarCoordinator = [[MisesToolbarCoordinator alloc] initWithBrowser:self.browser];
+      self.misesToolbarCoordinator.delegate = self.viewController;
+      [self.misesToolbarCoordinator start];
+    
   [super start];
   self.started = YES;
 }
@@ -331,4 +340,10 @@
   [self.locationBarCoordinator start];
 }
 
+- (void) activate {
+
+  [self.misesToolbarCoordinator activate];
+
+}
+
 @end
