diff --git a/ios/chrome/credential_provider_extension/entitlements/external/credential_provider_extension.appex.entitlements b/ios/chrome/credential_provider_extension/entitlements/external/credential_provider_extension.appex.entitlements
index 64e7177b8bda039c877f8dd1238b64b1f4944063..0e50681ab242d8b80997dc7f83d37bfd1747ce8d 100644
--- a/ios/chrome/credential_provider_extension/entitlements/external/credential_provider_extension.appex.entitlements
+++ b/ios/chrome/credential_provider_extension/entitlements/external/credential_provider_extension.appex.entitlements
@@ -8,7 +8,7 @@
 	<string>$(AppIdentifierPrefix)$(CFBundleIdentifier)</string>
 	<key>com.apple.security.application-groups</key>
 	<array>
-		<string>group.${IOS_BUNDLE_ID_PREFIX}.chrome</string>
+		<string>group.${IOS_BUNDLE_ID_PREFIX}.browser.ios</string>
 	</array>
 	<key>keychain-access-groups</key>
 	<array>
