diff --git a/ios/chrome/browser/ui/content_suggestions/content_suggestions_feature.mm b/ios/chrome/browser/ui/content_suggestions/content_suggestions_feature.mm
index 5bb45cc4754d06faf4df41be941f2dfad0bd5906..3580f8dc51be0c1f69a52931eda71a222c28326a 100644
--- a/ios/chrome/browser/ui/content_suggestions/content_suggestions_feature.mm
+++ b/ios/chrome/browser/ui/content_suggestions/content_suggestions_feature.mm
@@ -177,7 +177,7 @@ bool ShouldHideShortcutsForTrendingQueries() {
         false);
   }
   return base::GetFieldTrialParamByFeatureAsBool(
-      kTrendingQueriesModule, kTrendingQueriesHideShortcutsParam, false);
+      kTrendingQueriesModule, kTrendingQueriesHideShortcutsParam, true);
 }
 
 bool ShouldOnlyShowTrendingQueriesForDisabledFeed() {
