diff --git a/ios/web/js_messaging/java_script_feature.mm b/ios/web/js_messaging/java_script_feature.mm
index 481f65a6239fe33c4296384797fe520938eb8ba9..7a966c5dc8e231850677ca34bdcaa50ffcbf8778 100644
--- a/ios/web/js_messaging/java_script_feature.mm
+++ b/ios/web/js_messaging/java_script_feature.mm
@@ -207,6 +207,7 @@
 
   JavaScriptContentWorld* content_world =
       feature_manager->GetContentWorldForFeature(this);
+  if (!content_world) return false;
   DCHECK(content_world);
 
   return web_frame->GetWebFrameInternal()->CallJavaScriptFunctionInContentWorld(
