diff --git a/ios/chrome/browser/signin/signin_util.mm b/ios/chrome/browser/signin/signin_util.mm
index 3cfab9037943f2ab6cedb3bc8e5c2baa5e84d6e8..871b3be202586efcb444c7a38b8463c9db509e39 100644
--- a/ios/chrome/browser/signin/signin_util.mm
+++ b/ios/chrome/browser/signin/signin_util.mm
@@ -18,6 +18,7 @@
 #import "ios/chrome/browser/signin/signin_util_internal.h"
 #import "ios/chrome/browser/signin/system_identity.h"
 #import "ios/public/provider/chrome/browser/signin/signin_error_api.h"
+#include "base/threading/thread_restrictions.h"
 
 namespace {
 
@@ -97,6 +98,7 @@ CGSize GetSizeForIdentityAvatarSize(IdentityAvatarSize avatar_size) {
       signin::Tribool::kUnknown;
   static dispatch_once_t once;
   dispatch_once(&once, ^{
+      base::ScopedAllowBlockingForTesting allow_blocking;
     is_first_session_after_device_restore =
         IsFirstSessionAfterDeviceRestoreInternal();
   });
