diff --git a/ios/chrome/app/BUILD.gn b/ios/chrome/app/BUILD.gn
index 8be315047a4850fa7ab2d7f8944d603729971728..909142fac5ead1e54ecc70e588497540237595a6 100644
--- a/ios/chrome/app/BUILD.gn
+++ b/ios/chrome/app/BUILD.gn
@@ -160,6 +160,7 @@ tweak_info_plist("info_plist") {
   info_plists = [
     "resources/Info.plist",
     "resources/ChromeAddition+Info.plist",
+    "//mises/ios/app/resources/Mises+Info.plist",
   ]
   if (ios_enable_background_mode) {
     info_plists += [ "resources/Background+Info.plist" ]
@@ -184,6 +185,10 @@ tweak_info_plist("info_plist") {
   args = [
     "--branding=$chromium_short_name",
     "--add-gtm-metadata=1",
+    "--version-overrides=MAJOR=1",
+    "--version-overrides=MINOR=2",
+    "--version-overrides=BUILD=4",
+    "--version-overrides=PATCH=3",
   ]
 }
 
@@ -681,7 +686,8 @@ chrome_app("chrome") {
 
   deps = [
     ":main",
-
+    "//mises/ios/app",
+    "//mises/ios/third_party/mises",
     # Use the provider API controlled by args.gn.
     ios_provider_target,
   ]
@@ -752,8 +758,7 @@ chrome_app("chrome") {
   # (see https://crbug.com/1394553 for details of the build failure). This
   # is a stop-gap solution.
   if (target_environment == "catalyst" ||
-      ios_provider_target !=
-      "//ios/chrome/browser/providers:chromium_providers") {
+      !ios_chrome_links_with_lottie_framework) {
     assert_no_deps += [
       "//ios/third_party/lottie:lottie+bundle",
       "//ios/third_party/lottie:lottie+link",
