diff --git a/ios/chrome/browser/ui/toolbar/secondary_toolbar_view.h b/ios/chrome/browser/ui/toolbar/secondary_toolbar_view.h
index c2a444d597b2c415d2e48c38fd5b0ac7a55f512d..a2dbd7c9c3fa171fb306f63e25edd329aefaa287 100644
--- a/ios/chrome/browser/ui/toolbar/secondary_toolbar_view.h
+++ b/ios/chrome/browser/ui/toolbar/secondary_toolbar_view.h
@@ -34,6 +34,8 @@
 - (instancetype)initWithCoder:(NSCoder*)aDecoder NS_UNAVAILABLE;
 - (instancetype)init NS_UNAVAILABLE;
 
+- (void)updateMisesAvatar:(UIImage*)image;
+
 @end
 
 #endif  // IOS_CHROME_BROWSER_UI_TOOLBAR_SECONDARY_TOOLBAR_VIEW_H_
