diff --git a/ios/chrome/app/tests_fake_hook.mm b/ios/chrome/app/tests_fake_hook.mm
index 8c19562177b997e57446c2d847d7b23abe4ccef7..1fcb18644340f180b16543c7832390fb8451eaed 100644
--- a/ios/chrome/app/tests_fake_hook.mm
+++ b/ios/chrome/app/tests_fake_hook.mm
@@ -23,7 +23,7 @@ bool DisableDiscoverFeed() {
   return false;
 }
 bool DisableFirstRun() {
-  return false;
+  return true;
 }
 bool DisableGeolocation() {
   return false;
