diff --git a/ios/features.gni b/ios/features.gni
index 7ddf02f88a750cae7ee29f2f7c0201107b6c034f..bb5258f0780209eec5fb5412e3651a7b818f74ed 100644
--- a/ios/features.gni
+++ b/ios/features.gni
@@ -9,7 +9,7 @@ declare_args() {
 
   # Control whether iOS builds with PartitionAlloc.
   # Temporarily enable for rollout -- see crbug.com/1412190 for details.
-  ios_partition_alloc_enabled = true
+  ios_partition_alloc_enabled = false
 
   # Control whether //ios/chrome and //ios/web targets are build. This
   # is used by bots from projects based on top of Chromium that want to
