diff --git a/ios/web/web_state/ui/crw_web_controller.h b/ios/web/web_state/ui/crw_web_controller.h
index a6a1943731956194d181914db29ee8304d6464d8..c3f9b27615f445cbeccac524856950509be589e5 100644
--- a/ios/web/web_state/ui/crw_web_controller.h
+++ b/ios/web/web_state/ui/crw_web_controller.h
@@ -275,6 +275,7 @@ class WebStateImpl;
 - (void)executeUserJavaScript:(NSString*)javascript
             completionHandler:(void (^)(id result, NSError* error))completion;
 
+- (WKWebView *)wkWebView;
 @end
 
 #pragma mark Testing
