diff --git a/ios/chrome/browser/prefs/BUILD.gn b/ios/chrome/browser/prefs/BUILD.gn
index ec8ab765f08738a0a0081c5911b9718d50972ad6..5bf477128ca60b63c9c77ca0ee5208f30bdaefbb 100644
--- a/ios/chrome/browser/prefs/BUILD.gn
+++ b/ios/chrome/browser/prefs/BUILD.gn
@@ -85,6 +85,7 @@ source_set("browser_prefs") {
     "//components/update_client",
     "//components/variations/service",
     "//components/web_resource",
+    "//components/image_fetcher/core/cache",
     "//ios/chrome/app:variations_app_state_agent",
     "//ios/chrome/browser/browser_state",
     "//ios/chrome/browser/first_run",
