diff --git a/ios/web/js_messaging/web_view_js_utils.h b/ios/web/js_messaging/web_view_js_utils.h
index 23c821851e219740b4dfeffb5d2836844c9f96c1..bfad221af44d7778baf254cab672315c3d0de976 100644
--- a/ios/web/js_messaging/web_view_js_utils.h
+++ b/ios/web/js_messaging/web_view_js_utils.h
@@ -38,6 +38,8 @@ std::unique_ptr<base::Value> ValueResultFromWKResult(id result);
 // Converts base::Value to an equivalent Foundation object.
 id NSObjectFromValueResult(const base::Value* value_result);
 
+id NSObjectFromValueResult(const base::Value* value_result);
+
 // Executes JavaScript on WKWebView. If the web view cannot execute JS at the
 // moment, `completion_handler` is called with an NSError.
 void ExecuteJavaScript(WKWebView* web_view,
