diff --git a/ios/chrome/browser/ui/ntp/feed_wrapper_view_controller.mm b/ios/chrome/browser/ui/ntp/feed_wrapper_view_controller.mm
index 29c95b6f135a20e0f573a4b20507e036f276dc01..d47e5a12e77a2a58eb069dc551110a8479d39808 100644
--- a/ios/chrome/browser/ui/ntp/feed_wrapper_view_controller.mm
+++ b/ios/chrome/browser/ui/ntp/feed_wrapper_view_controller.mm
@@ -84,7 +84,7 @@ - (void)configureEmptyCollectionAsWrapper {
   UICollectionView* emptyCollectionView =
       [[UICollectionView alloc] initWithFrame:CGRectZero
                          collectionViewLayout:layout];
-  [emptyCollectionView setShowsVerticalScrollIndicator:NO];
+  //[emptyCollectionView setShowsVerticalScrollIndicator:NO];
   [self.view addSubview:emptyCollectionView];
   self.contentCollectionView = emptyCollectionView;
   self.contentCollectionView.backgroundColor =
