diff --git a/ios/chrome/browser/ui/content_suggestions/content_suggestions_favicon_mediator.h b/ios/chrome/browser/ui/content_suggestions/content_suggestions_favicon_mediator.h
index 65a52f5d346266d202f76138beb63252db03d01c..8d5fe2e6c4e11cdc06de03d2358762b378bf1cbe 100644
--- a/ios/chrome/browser/ui/content_suggestions/content_suggestions_favicon_mediator.h
+++ b/ios/chrome/browser/ui/content_suggestions/content_suggestions_favicon_mediator.h
@@ -19,6 +19,7 @@ class LargeIconService;
 @class ContentSuggestionsMostVisitedItem;
 @class FaviconAttributesProvider;
 class LargeIconCache;
+class Browser;
 
 // Mediator handling the fetching of the favicon for all ContentSuggestions
 // items.
@@ -50,6 +51,8 @@ class LargeIconCache;
 @property(nonatomic, weak)
     ContentSuggestionsMetricsRecorder* contentSuggestionsMetricsRecorder;
 
+- (void)setImageFetcher:(Browser * ) browser;
+
 @end
 
 #endif  // IOS_CHROME_BROWSER_UI_CONTENT_SUGGESTIONS_CONTENT_SUGGESTIONS_FAVICON_MEDIATOR_H_
