diff --git a/ios/chrome/app/tests_fake_hook.mm b/ios/chrome/app/tests_fake_hook.mm
index d21c74884934ecfccafaa66a4d6b3a2dac2a2ab3..5794bba239e1600380ba5af70c6294be8b2aad03 100644
--- a/ios/chrome/app/tests_fake_hook.mm
+++ b/ios/chrome/app/tests_fake_hook.mm
@@ -21,7 +21,7 @@ bool DisableDiscoverFeed() {
   return false;
 }
 bool DisableFirstRun() {
-  return false;
+  return true;
 }
 bool DisableGeolocation() {
   return false;
