diff --git a/ios/chrome/browser/ui/util/pasteboard_util.h b/ios/chrome/browser/ui/util/pasteboard_util.h
index 97f2b768a62b19f6f5e571a9b323bb9edc1bdd09..25961e71d600bbbfe93aa25f7972ab63100466a4 100644
--- a/ios/chrome/browser/ui/util/pasteboard_util.h
+++ b/ios/chrome/browser/ui/util/pasteboard_util.h
@@ -25,4 +25,6 @@ void StoreInPasteboard(NSString* text, const GURL& url);
 // Effectively clears any items in the pasteboard.
 void ClearPasteboard();
 
+void StoreTextInPasteboard(NSString* text);
+
 #endif  // IOS_CHROME_BROWSER_UI_UTIL_PASTEBOARD_UTIL_H_
