diff --git a/ios/chrome/browser/ui/toolbar/primary_toolbar_coordinator.mm b/ios/chrome/browser/ui/toolbar/primary_toolbar_coordinator.mm
index 3ba8dc900fa968c3f83e13c88c21b70321c3a197..923b1fb5392a73ed113a873a22a6b199c97c6e1d 100644
--- a/ios/chrome/browser/ui/toolbar/primary_toolbar_coordinator.mm
+++ b/ios/chrome/browser/ui/toolbar/primary_toolbar_coordinator.mm
@@ -24,6 +24,8 @@
 #import "ios/chrome/browser/ui/toolbar/adaptive_toolbar_coordinator+subclassing.h"
 #import "ios/chrome/browser/ui/toolbar/primary_toolbar_view_controller.h"
 
+#import "mises/ios/browser/ui/toolbar/mises_toolbar_coordinator.h"
+
 @interface PrimaryToolbarCoordinator ()
 
 // Whether the coordinator is started.
@@ -31,6 +33,8 @@ @interface PrimaryToolbarCoordinator ()
 // Redefined as PrimaryToolbarViewController.
 @property(nonatomic, strong) PrimaryToolbarViewController* viewController;
 
+@property(nonatomic, strong) MisesToolbarCoordinator* misesToolbarCoordinator;
+
 @end
 
 @implementation PrimaryToolbarCoordinator
@@ -61,6 +65,11 @@ - (void)start {
   self.viewController.buttonFactory =
       [self buttonFactoryWithType:ToolbarType::kPrimary];
 
+    
+      self.misesToolbarCoordinator = [[MisesToolbarCoordinator alloc] initWithBrowser:self.browser];
+      self.misesToolbarCoordinator.delegate = self.viewController;
+      [self.misesToolbarCoordinator start];
+    
   [super start];
   self.started = YES;
 }
@@ -90,4 +99,10 @@ - (void)triggerToolbarSlideInAnimation {
   [self.viewController triggerToolbarSlideInAnimationFromBelow:NO];
 }
 
+- (void) activate {
+
+  [self.misesToolbarCoordinator activate];
+
+}
+
 @end
