diff --git a/ios/web/web_state/web_view_internal_creation_util.mm b/ios/web/web_state/web_view_internal_creation_util.mm
index 6edb33de33701f2c04cd107baf42bdfef20b755f..673261ce4c6819b69bf8dbf42a597e989385e7fb 100644
--- a/ios/web/web_state/web_view_internal_creation_util.mm
+++ b/ios/web/web_state/web_view_internal_creation_util.mm
@@ -75,6 +75,10 @@ void VerifyWKWebViewCreationPreConditions(
     base::UmaHistogramBoolean("IOS.WebInspector.Enabled", enable_web_inspector);
   }
 
+  if (@available(macOS 13.3, iOS 16.4, tvOS 16.4, *)) {
+    web_view.inspectable = YES;
+  }
+
   return web_view;
 }
 
