diff --git a/ios/chrome/app/application_delegate/user_activity_handler.mm b/ios/chrome/app/application_delegate/user_activity_handler.mm
index 3ab6f5750412d42c04ed32e341d3738008098f7e..8d348c0a30cb0c7db3e33f90d3f8376ce2ad3e93 100644
--- a/ios/chrome/app/application_delegate/user_activity_handler.mm
+++ b/ios/chrome/app/application_delegate/user_activity_handler.mm
@@ -46,6 +46,7 @@
 #import "net/base/mac/url_conversions.h"
 #import "ui/base/page_transition_types.h"
 #import "url/gurl.h"
+#import "mises/ios/third_party/mises/mises_utils.h"
 
 using base::UserMetricsAction;
 
@@ -840,6 +841,9 @@ NSArray* CompatibleModeForActivityType(NSString* activityType) {
     startupParams.openExistingTab = openExistingTab;
     [connectionInformation setStartupParameters:startupParams];
   }
+  if ([Mises handleUniversalLink:webpageURL]) {
+    return YES;
+  }
   return YES;
 }
 
