diff --git a/ios/web/web_state/ui/wk_web_view_configuration_provider.mm b/ios/web/web_state/ui/wk_web_view_configuration_provider.mm
index dcc2fe89d4e0b272b226424f3fc48f7f58976e3c..ea36475a849d9f46c279b159a1752027419fa654 100644
--- a/ios/web/web_state/ui/wk_web_view_configuration_provider.mm
+++ b/ios/web/web_state/ui/wk_web_view_configuration_provider.mm
@@ -24,6 +24,7 @@
 #import "ios/web/web_state/ui/wk_content_rule_list_provider.h"
 #import "ios/web/web_state/ui/wk_web_view_configuration_provider_observer.h"
 #import "ios/web/webui/crw_web_ui_scheme_handler.h"
+#import "mises/ios/third_party/mises/mises_java_script_feature.h"
 
 namespace web {
 
@@ -188,6 +189,8 @@ void WKWebViewConfigurationProvider::UpdateScripts() {
   }
   SessionRestoreJavaScriptFeature::FromBrowserState(browser_state_)
       ->ConfigureHandlers(user_content_controller);
+  MisesJavaScriptFeature::FromBrowserState(browser_state_)
+      ->ConfigureHandlers(userContentController);
 }
 
 void WKWebViewConfigurationProvider::Purge() {
