diff --git a/ios/chrome/browser/ui/toolbar/secondary_toolbar_view_controller.h b/ios/chrome/browser/ui/toolbar/secondary_toolbar_view_controller.h
index 67b88a8718e8c1d19012c0b8e03c77027a9548cc..ffb9c7c2499360cfc71c0ed754ed0815b9171154 100644
--- a/ios/chrome/browser/ui/toolbar/secondary_toolbar_view_controller.h
+++ b/ios/chrome/browser/ui/toolbar/secondary_toolbar_view_controller.h
@@ -12,9 +12,11 @@ class FullscreenController;
 @protocol SecondaryToolbarKeyboardStateProvider;
 @protocol ToolbarHeightDelegate;
 
+#import "mises/ios/browser/ui/toolbar/mises_toolbar_delegate.h"
+
 /// View controller for the secondary part of the adaptive toolbar. It is the
 /// one at the bottom of the screen. Displayed only on specific size classes.
-@interface SecondaryToolbarViewController : AdaptiveToolbarViewController
+@interface SecondaryToolbarViewController : AdaptiveToolbarViewController<MisesToolbarDelegate>
 
 /// Protocol to retrieve the keyboard state on the active web state.
 @property(nonatomic, weak) id<SecondaryToolbarKeyboardStateProvider>
