diff --git a/ios/chrome/browser/ui/omnibox/omnibox_view_controller.mm b/ios/chrome/browser/ui/omnibox/omnibox_view_controller.mm
index 6a6e66990cd5814fb2fe9fb2106947f911393592..1cb3adc31dab6fd4e3ca182a6b12c363c9fdd820 100644
--- a/ios/chrome/browser/ui/omnibox/omnibox_view_controller.mm
+++ b/ios/chrome/browser/ui/omnibox/omnibox_view_controller.mm
@@ -513,6 +513,9 @@ const CGFloat kClearButtonImageSize = 17.0f;
   if (self.isUpdatingCachedClipboardState) {
     return;
   }
+  if (!base::SequencedTaskRunner::HasCurrentDefault()) {
+      return;
+  }
   self.isUpdatingCachedClipboardState = YES;
   self.hasCopiedContent = NO;
   ClipboardRecentContent* clipboardRecentContent =
