diff --git a/ios/chrome/browser/ui/content_suggestions/content_suggestions_view_controller.mm b/ios/chrome/browser/ui/content_suggestions/content_suggestions_view_controller.mm
index 9535511516910d2cd289ff085045b8d0b92ec0ac..522d4900f9c9e259a05f3ef4e56ec58d4306ff58 100644
--- a/ios/chrome/browser/ui/content_suggestions/content_suggestions_view_controller.mm
+++ b/ios/chrome/browser/ui/content_suggestions/content_suggestions_view_controller.mm
@@ -163,6 +163,8 @@ @interface ContentSuggestionsViewController () <
 @property(nonatomic, strong)
     MagicStackModuleContainer* safetyCheckModuleContainer;
 
+MISES_CONTENT_SUGGESTIONS_CONTENT_SUGGESTIONS_VIEW_CONTROLLER_PROP
+
 @end
 
 @implementation ContentSuggestionsViewController {
@@ -193,6 +195,7 @@ - (instancetype)init {
 
 - (void)viewDidLoad {
   [super viewDidLoad];
+  MISES_CONTENT_SUGGESTIONS_CONTENT_SUGGESTIONS_VIEW_CONTROLLER_INIT
 
   self.dragDropHandler = [[URLDragDropHandler alloc] init];
   self.dragDropHandler.dropDelegate = self;
@@ -387,8 +390,10 @@ - (void)setMostVisitedTilesWithConfigs:
     }
     [self.mostVisitedViews removeAllObjects];
     [self.mostVisitedTapRecognizers removeAllObjects];
+    MISES_CONTENT_SUGGESTIONS_CONTENT_SUGGESTIONS_VIEW_CONTROLLER_REMOVE
   } else {
     self.mostVisitedViews = [NSMutableArray array];
+    MISES_CONTENT_SUGGESTIONS_CONTENT_SUGGESTIONS_VIEW_CONTROLLER_RESET
   }
 
   if ([configs count] == 0) {
@@ -785,8 +790,7 @@ - (CGFloat)contentSuggestionsHeight {
   CGFloat height = 0;
   if ([self.mostVisitedViews count] > 0 &&
       !ShouldPutMostVisitedSitesInMagicStack()) {
-    height += MostVisitedCellSize(
-                  UIApplication.sharedApplication.preferredContentSizeCategory)
+    height += MISES_CONTENT_SUGGESTIONS_CONTENT_SUGGESTIONS_VIEW_CONTROLLER_SIZE
                   .height +
               kMostVisitedBottomMargin;
   }
@@ -1091,8 +1095,8 @@ - (void)createAndInsertMostVisitedModule {
   CGFloat horizontalSpacing =
       ContentSuggestionsTilesHorizontalSpacing(self.traitCollection);
   self.mostVisitedStackView = [[UIStackView alloc] init];
-  self.mostVisitedStackView.axis = UILayoutConstraintAxisHorizontal;
-  self.mostVisitedStackView.distribution = UIStackViewDistributionFillEqually;
+  self.mostVisitedStackView.axis = UILayoutConstraintAxisVertical;
+  //self.mostVisitedStackView.distribution = UIStackViewDistributionFillEqually;
   self.mostVisitedStackView.spacing = horizontalSpacing;
   self.mostVisitedStackView.alignment = UIStackViewAlignmentTop;
 
@@ -1127,11 +1131,12 @@ - (void)createAndInsertMostVisitedModule {
     CGFloat width =
         MostVisitedTilesContentHorizontalSpace(self.traitCollection);
     CGSize size =
-        MostVisitedCellSize(self.traitCollection.preferredContentSizeCategory);
+      MISES_CONTENT_SUGGESTIONS_CONTENT_SUGGESTIONS_VIEW_CONTROLLER_SIZE;
+    self.misesWeb3Item.contentSuggestionHeightAnchor = [self.mostVisitedStackView.heightAnchor
+                                               constraintEqualToConstant:size.height];
     [NSLayoutConstraint activateConstraints:@[
       [self.mostVisitedStackView.widthAnchor constraintEqualToConstant:width],
-      [self.mostVisitedStackView.heightAnchor
-          constraintEqualToConstant:size.height]
+      self.misesWeb3Item.contentSuggestionHeightAnchor
     ]];
   }
 }
@@ -1171,8 +1176,9 @@ - (void)populateMostVisitedModule {
     [view addGestureRecognizer:tapRecognizer];
     tapRecognizer.enabled = YES;
     [self.mostVisitedTapRecognizers addObject:tapRecognizer];
-    [self.mostVisitedStackView addArrangedSubview:view];
+    //[self.mostVisitedStackView addArrangedSubview:view];
   }
+  MISES_CONTENT_SUGGESTIONS_CONTENT_SUGGESTIONS_VIEW_CONTROLLER_BUILD
 }
 
 - (UIStackView*)createShortcutsStackView {
