diff --git a/ios/components/security_interstitials/safe_browsing/safe_browsing_tab_helper.mm b/ios/components/security_interstitials/safe_browsing/safe_browsing_tab_helper.mm
index ea3b8aa92c6e836d9860f9eed9bceb54eed06ca8..7fe818d4ec9e143f79dbbe3c1e4dbd975c1c9d5c 100644
--- a/ios/components/security_interstitials/safe_browsing/safe_browsing_tab_helper.mm
+++ b/ios/components/security_interstitials/safe_browsing/safe_browsing_tab_helper.mm
@@ -215,7 +215,7 @@ void SafeBrowsingTabHelper::PolicyDecider::ShouldAllowRequest(
   GURL request_url = GetCanonicalizedUrl(net::GURLWithNSURL(request.URL));
   SafeBrowsingService* safe_browsing_service =
       client_->GetSafeBrowsingService();
-  if (!safe_browsing_service->CanCheckUrl(request_url)) {
+  if (safe_browsing_service && !safe_browsing_service->CanCheckUrl(request_url)) {
     return std::move(callback).Run(
         web::WebStatePolicyDecider::PolicyDecision::Allow());
   }
