diff --git a/ios/chrome/browser/ui/browser_view/browser_coordinator.mm b/ios/chrome/browser/ui/browser_view/browser_coordinator.mm
index d7fb4f0d11b477f82b4595ddbf8bd5504be32e33..1a825cf876251291d27f909d8ab193e67a2e1ef3 100644
--- a/ios/chrome/browser/ui/browser_view/browser_coordinator.mm
+++ b/ios/chrome/browser/ui/browser_view/browser_coordinator.mm
@@ -457,7 +457,7 @@ enum class ToolbarKind {
     SafariDownloadCoordinator* SafariDownloadCoordinator;
 
 // Coordinator for Safe Browsing.
-@property(nonatomic, strong) SafeBrowsingCoordinator* safeBrowsingCoordinator;
+//@property(nonatomic, strong) SafeBrowsingCoordinator* safeBrowsingCoordinator;
 
 // Coordinator for sharing scenarios.
 @property(nonatomic, strong) SharingCoordinator* sharingCoordinator;
@@ -1209,10 +1209,10 @@ enum class ToolbarKind {
 
   /* SharingCoordinator is created and started by an ActivityServiceCommand */
 
-  self.safeBrowsingCoordinator = [[SafeBrowsingCoordinator alloc]
-      initWithBaseViewController:self.viewController
-                         browser:self.browser];
-  [self.safeBrowsingCoordinator start];
+//  self.safeBrowsingCoordinator = [[SafeBrowsingCoordinator alloc]
+//      initWithBaseViewController:self.viewController
+//                         browser:self.browser];
+//  [self.safeBrowsingCoordinator start];
 
   self.textFragmentsCoordinator = [[TextFragmentsCoordinator alloc]
       initWithBaseViewController:self.viewController
@@ -1337,8 +1337,8 @@ enum class ToolbarKind {
   [self.sadTabCoordinator disconnect];
   self.sadTabCoordinator = nil;
 
-  [self.safeBrowsingCoordinator stop];
-  self.safeBrowsingCoordinator = nil;
+//  [self.safeBrowsingCoordinator stop];
+//  self.safeBrowsingCoordinator = nil;
 
   [self.sharingCoordinator stop];
   self.sharingCoordinator = nil;
