diff --git a/ios/chrome/browser/ui/util/layout_guide_names.h b/ios/chrome/browser/ui/util/layout_guide_names.h
index 3b52e79dec16a5274349d13f1f166151961a66cd..f95c5d5eec6652b5c4369c47762bccb68051a9d2 100644
--- a/ios/chrome/browser/ui/util/layout_guide_names.h
+++ b/ios/chrome/browser/ui/util/layout_guide_names.h
@@ -53,4 +53,6 @@ extern GuideName* const kDiscoverFeedHeaderMenuGuide;
 // tab grid.
 extern GuideName* const kTabGridBottomToolbarGuide;
 
+extern GuideName* const kMisesButtonGuide;
+
 #endif  // IOS_CHROME_BROWSER_UI_UTIL_LAYOUT_GUIDE_NAMES_H_
