diff --git a/ios/chrome/browser/ui/popup_menu/public/cells/popup_menu_item.h b/ios/chrome/browser/ui/popup_menu/public/cells/popup_menu_item.h
index f6688a814fed7ba7aaffa47fca94377178689af0..6514671f2a3f038c9ebe3720ef4e60ac3cdb5bb8 100644
--- a/ios/chrome/browser/ui/popup_menu/public/cells/popup_menu_item.h
+++ b/ios/chrome/browser/ui/popup_menu/public/cells/popup_menu_item.h
@@ -37,6 +37,18 @@ typedef NS_ENUM(NSInteger, PopupMenuAction) {
   PopupMenuActionSettings,
   // Textual popup menu
   PopupMenuActionEnterpriseInfoMessage,
+  PopupMenuActionMises,
+  PopupMenuActionMisesAccount,
+  PopupMenuActionMisesMyData,
+  PopupMenuActionMisesDiscover,
+  PopupMenuActionMisesWallet,
+  PopupMenuActionMisesStakingPortal,
+  PopupMenuActionMisesNft,
+  PopupMenuActionMisesInvite,
+  PopupMenuActionMisesOfficialSite,
+  PopupMenuActionMisesOpenMetamask,
+  PopupMenuActionMisesOpenMisesWallet,
+  PopupMenuActionMisesOpenKeplr
 };
 
 // Protocol defining a popup item.
