diff --git a/ios/chrome/browser/ui/popup_menu/public/cells/popup_menu_item.h b/ios/chrome/browser/ui/popup_menu/public/cells/popup_menu_item.h
index 55372f94dd3485d28cfb1eba12e34e50c7ef66f5..7bfd1bf64970f47575d982cd5d8c941435b064a4 100644
--- a/ios/chrome/browser/ui/popup_menu/public/cells/popup_menu_item.h
+++ b/ios/chrome/browser/ui/popup_menu/public/cells/popup_menu_item.h
@@ -62,6 +62,18 @@ typedef NS_ENUM(NSInteger, PopupMenuAction) {
   PopupMenuActionShowPermissionsOptions,
   // Textual popup menu
   PopupMenuActionEnterpriseInfoMessage,
+  PopupMenuActionMises,
+  PopupMenuActionMisesAccount,
+  PopupMenuActionMisesMyData,
+  PopupMenuActionMisesDiscover,
+  PopupMenuActionMisesWallet,
+  PopupMenuActionMisesStakingPortal,
+  PopupMenuActionMisesNft,
+  PopupMenuActionMisesInvite,
+  PopupMenuActionMisesOfficialSite,
+  PopupMenuActionMisesOpenMetamask,
+  PopupMenuActionMisesOpenMisesWallet,
+  PopupMenuActionMisesOpenKeplr
 };
 
 // Protocol defining a popup item.
