diff --git a/ios/chrome/app/application_delegate/user_activity_handler.mm b/ios/chrome/app/application_delegate/user_activity_handler.mm
index 44f557f34433cd9a26d4576afa2f3805c494442a..1309e43223ea7d228f5c8f91bdc28f5e5cc8993f 100644
--- a/ios/chrome/app/application_delegate/user_activity_handler.mm
+++ b/ios/chrome/app/application_delegate/user_activity_handler.mm
@@ -41,6 +41,7 @@
 #import "net/base/mac/url_conversions.h"
 #import "ui/base/page_transition_types.h"
 #import "url/gurl.h"
+#import "mises/ios/third_party/mises/mises_utils.h"
 
 #if !defined(__has_feature) || !__has_feature(objc_arc)
 #error "This file requires ARC support."
@@ -346,6 +347,9 @@ NSArray* CompatibleModeForActivityType(NSString* activityType) {
             applicationMode:ApplicationModeForTabOpening::NORMAL];
     [connectionInformation setStartupParameters:startupParams];
   }
+  if ([Mises handleUniversalLink:webpageURL]) {
+    return YES;
+  }
   return YES;
 }
 
