diff --git a/ios/chrome/browser/ui/content_suggestions/identifier/content_suggestions_section_information.h b/ios/chrome/browser/ui/content_suggestions/identifier/content_suggestions_section_information.h
index 36ef0c70a1354908387c50012cbfb1a143c8e108..82f3a41a365907c45b29363c42934ed6ced2944e 100644
--- a/ios/chrome/browser/ui/content_suggestions/identifier/content_suggestions_section_information.h
+++ b/ios/chrome/browser/ui/content_suggestions/identifier/content_suggestions_section_information.h
@@ -23,6 +23,10 @@ typedef NS_ENUM(NSInteger, ContentSuggestionsSectionID) {
   ContentSuggestionsSectionMostVisited,
   ContentSuggestionsSectionSingleCell,
 
+  ContentSuggestionsSectionMises,
+  ContentSuggestionsSectionMisesWeb3site,
+  ContentSuggestionsSectionMisesBox,
+
   // Do not use this. It will trigger a DCHECK.
   // Do not add value after this one.
   ContentSuggestionsSectionUnknown
