diff --git a/ios/chrome/browser/ui/toolbar/adaptive_toolbar_view_controller.mm b/ios/chrome/browser/ui/toolbar/adaptive_toolbar_view_controller.mm
index b9ae37cfd99ff54ed632d79b6a73d02606c6b9d0..74a30b8280c78675ae94b43656f362128a66faa4 100644
--- a/ios/chrome/browser/ui/toolbar/adaptive_toolbar_view_controller.mm
+++ b/ios/chrome/browser/ui/toolbar/adaptive_toolbar_view_controller.mm
@@ -155,7 +155,9 @@ const CGFloat kFullscreenProgressFullyExpanded = 1.0;
   self.view.forwardButton.guideName = kForwardButtonGuide;
   self.view.backButton.guideName = kBackButtonGuide;
   self.view.shareButton.guideName = kShareButtonGuide;
-
+  self.view.misesButton.guideName = kMisesButtonGuide;
+  self.view.walletButton.guideName = kWalletButtonGuide;
+    
   [self addLayoutGuideCenterToButtons];
 
   // Add navigation popup menu triggers.
@@ -459,6 +461,8 @@ const CGFloat kFullscreenProgressFullyExpanded = 1.0;
   self.view.openNewTabButton.iphHighlighted = NO;
   self.view.tabGridButton.iphHighlighted = NO;
   self.view.toolsMenuButton.iphHighlighted = NO;
+  self.view.misesButton.iphHighlighted = NO;
+  self.view.walletButton.iphHighlighted = NO;
 }
 
 #pragma mark - Private
@@ -550,6 +554,10 @@ const CGFloat kFullscreenProgressFullyExpanded = 1.0;
   } else if (sender == self.view.openNewTabButton) {
     base::RecordAction(base::UserMetricsAction("MobileToolbarNewTabShortcut"));
     base::RecordAction(base::UserMetricsAction("MobileTabNewTab"));
+  } else if (sender == self.view.misesButton) {
+    base::RecordAction(base::UserMetricsAction("MobileToolbarMises"));
+  } else if (sender == self.view.walletButton) {
+    base::RecordAction(base::UserMetricsAction("MobileToolbarWallet"));
   } else {
     NOTREACHED();
   }
@@ -590,6 +598,8 @@ const CGFloat kFullscreenProgressFullyExpanded = 1.0;
   self.view.forwardButton.layoutGuideCenter = self.layoutGuideCenter;
   self.view.backButton.layoutGuideCenter = self.layoutGuideCenter;
   self.view.shareButton.layoutGuideCenter = self.layoutGuideCenter;
+  self.view.misesButton.layoutGuideCenter = self.layoutGuideCenter;
+  self.view.walletButton.layoutGuideCenter = self.layoutGuideCenter;
 }
 
 // Exits fullscreen.
