diff --git a/ios/build/chrome_build.gni b/ios/build/chrome_build.gni
index 35113412870bb445b93d30e691376b08d2603363..708e909130bb392731a679b7134b841888ca56c7 100644
--- a/ios/build/chrome_build.gni
+++ b/ios/build/chrome_build.gni
@@ -95,9 +95,9 @@ if (is_official_build && is_chrome_branded) {
   is_official_release = false
 }
 
-chromium_bundle_id = "$ios_app_bundle_id_prefix.chrome.ios.dev"
-chromium_handoff_id = "$ios_app_bundle_id_prefix.chrome.handoff"
-ios_move_tab_activity_type = "$ios_app_bundle_id_prefix.chrome.move-tab"
+chromium_bundle_id = "$ios_app_bundle_id_prefix.browser"
+chromium_handoff_id = "$ios_app_bundle_id_prefix.browser.handoff"
+ios_move_tab_activity_type = "$ios_app_bundle_id_prefix.browser.move-tab"
 
 if (is_chrome_branded) {
   chromium_short_name = "Chrome"
@@ -107,10 +107,10 @@ if (is_chrome_branded) {
   url_unsecure_scheme = "googlechrome"
   url_x_callback_scheme = "googlechrome-x-callback"
 } else {
-  chromium_short_name = "Chromium"
-  url_channel_scheme = "chromium-dev"
-  url_secure_scheme = "chromiums"
-  url_ssoauth_scheme = "$ios_app_bundle_id_prefix.sso.chromium"
-  url_unsecure_scheme = "chromium"
-  url_x_callback_scheme = "chromium-x-callback"
+  chromium_short_name = "Mises"
+  url_channel_scheme = "mises-dev"
+  url_secure_scheme = "mises"
+  url_ssoauth_scheme = "$ios_app_bundle_id_prefix.sso.mises"
+  url_unsecure_scheme = "mises"
+  url_x_callback_scheme = "mises-x-callback"
 }
