diff --git a/ios/chrome/browser/ui/ntp/feed_wrapper_view_controller.mm b/ios/chrome/browser/ui/ntp/feed_wrapper_view_controller.mm
index 20c58421d04fa5ebb5b7bfd37ea552d5b4b17a12..e1a4ea98928d2873a6df263339d6f92511d87bc8 100644
--- a/ios/chrome/browser/ui/ntp/feed_wrapper_view_controller.mm
+++ b/ios/chrome/browser/ui/ntp/feed_wrapper_view_controller.mm
@@ -94,7 +94,12 @@
   self.contentCollectionView.backgroundColor =
       IsContentSuggestionsUIModuleRefreshEnabled()
           ? [UIColor clearColor]
-          : ntp_home::NTPBackgroundColor();
+    : [UIColor clearColor];//ntp_home::NTPBackgroundColor();
+    UIGraphicsBeginImageContext(self.view.frame.size);
+    [[UIImage imageNamed:@"ntp_desktop"] drawInRect:self.view.bounds];
+    UIImage *image = UIGraphicsGetImageFromCurrentImageContext();
+    UIGraphicsEndImageContext();
+    self.view.backgroundColor = [UIColor colorWithPatternImage:image];
   self.contentCollectionView.translatesAutoresizingMaskIntoConstraints = NO;
   AddSameConstraints(self.contentCollectionView, self.view);
 }
