diff --git a/ios/web/web_state/ui/wk_web_view_configuration_provider.mm b/ios/web/web_state/ui/wk_web_view_configuration_provider.mm
index dcc2fe89d4e0b272b226424f3fc48f7f58976e3c..3b8ee4e45c496f04748584fed0d19448e038a56b 100644
--- a/ios/web/web_state/ui/wk_web_view_configuration_provider.mm
+++ b/ios/web/web_state/ui/wk_web_view_configuration_provider.mm
@@ -188,6 +188,7 @@
   }
   SessionRestoreJavaScriptFeature::FromBrowserState(browser_state_)
       ->ConfigureHandlers(user_content_controller);
+  MISES_WK_WEB_VIEW_CONFIGUTATION_PROVIDER_INSERT
 }
 
 void WKWebViewConfigurationProvider::Purge() {
