diff --git a/ios/chrome/browser/web/chrome_main_parts.mm b/ios/chrome/browser/web/chrome_main_parts.mm
index e9b717a8ebf6fcff01eb2b8dc2b6ff656a278c89..5c4da493801f1a22bc5041df2c5f211d1b171e2f 100644
--- a/ios/chrome/browser/web/chrome_main_parts.mm
+++ b/ios/chrome/browser/web/chrome_main_parts.mm
@@ -143,6 +143,11 @@ void IOSChromeMainParts::PreCreateMainMessageLoop() {
   base::PathService::Get(ios::FILE_RESOURCES_PACK, &resources_pack_path);
   ui::ResourceBundle::GetSharedInstance().AddDataPackFromPath(
       resources_pack_path, ui::k100Percent);
+
+    base::FilePath mises_pack_path;
+    base::PathService::Get(base::DIR_MODULE, &mises_pack_path);
+    mises_pack_path = mises_pack_path.AppendASCII("mises_resources.pak");
+    ui::ResourceBundle::GetSharedInstance().AddDataPackFromPath( mises_pack_path, ui::kScaleFactorNone);
 }
 
 void IOSChromeMainParts::PreCreateThreads() {
@@ -373,6 +378,7 @@ void IOSChromeMainParts::PreMainMessageLoopRun() {
   safe_browsing::SafeBrowsingMetricsCollector* safe_browsing_metrics_collector =
       SafeBrowsingMetricsCollectorFactory::GetForBrowserState(
           last_used_browser_state);
+  if (safe_browsing_service)
   safe_browsing_service->Initialize(last_used_browser_state->GetPrefs(),
                                     user_data_path,
                                     safe_browsing_metrics_collector);
