diff --git a/ios/chrome/browser/ui/toolbar/toolbar_coordinator.mm b/ios/chrome/browser/ui/toolbar/toolbar_coordinator.mm
index 56fcc5e9139825d6bae2a3d926d849a37f2d1b01..335f8a825c5a3af0d9addd6e48569afd8660d206 100644
--- a/ios/chrome/browser/ui/toolbar/toolbar_coordinator.mm
+++ b/ios/chrome/browser/ui/toolbar/toolbar_coordinator.mm
@@ -268,7 +268,7 @@ - (void)updateToolbar {
                      ![self showingOmniboxPopup] && !canShowTabStrip &&
                      IsSplitToolbarMode(self.traitEnvironment);
 
-  self.primaryToolbarViewController.view.hidden = hideToolbar;
+  self.primaryToolbarViewController.view.hidden = hideToolbar && FALSE;
 }
 
 - (BOOL)isLoadingPrerenderer {
@@ -404,7 +404,7 @@ - (void)onFakeboxBlur {
       self.browser->GetWebStateList()->GetActiveWebState();
   if (webState && IsVisibleURLNewTabPage(webState)) {
     self.primaryToolbarViewController.view.hidden =
-        IsSplitToolbarMode(self.traitEnvironment);
+        IsSplitToolbarMode(self.traitEnvironment) && FALSE;
   }
 }
 
