diff --git a/ios/chrome/browser/ui/content_suggestions/content_suggestions_header_view.mm b/ios/chrome/browser/ui/content_suggestions/content_suggestions_header_view.mm
index d360cb4f27584d11ccbb4e84c0fe920e903f073c..dac60fe85988662d45cdcee63a4671c0d5755382 100644
--- a/ios/chrome/browser/ui/content_suggestions/content_suggestions_header_view.mm
+++ b/ios/chrome/browser/ui/content_suggestions/content_suggestions_header_view.mm
@@ -389,6 +389,7 @@ CGFloat ToolbarHeight() {
     percent = base::clamp<CGFloat>(
         animatingOffset / ntp_header::kAnimationDistance, 0, 1);
   }
+  percent = 1.0;
   if (IsContentSuggestionsUIModuleRefreshEnabled()) {
     // Update background color of fake toolbar if stuck to top of NTP so that it
     // has a non-clear background. Otherwise, return to clear background.
