diff --git a/ios/chrome/browser/ui/toolbar/primary_toolbar_view_controller.h b/ios/chrome/browser/ui/toolbar/primary_toolbar_view_controller.h
index c515d00b7defcab500bde659fc4ebfc86b08d135..aa1194b056fab1a8e739fc95e7b6102aae42d981 100644
--- a/ios/chrome/browser/ui/toolbar/primary_toolbar_view_controller.h
+++ b/ios/chrome/browser/ui/toolbar/primary_toolbar_view_controller.h
@@ -9,6 +9,7 @@
 #import "ios/chrome/browser/ui/orchestrator/toolbar_animatee.h"
 #import "ios/chrome/browser/ui/sharing/sharing_positioner.h"
 #import "ios/chrome/browser/ui/toolbar/adaptive_toolbar_view_controller.h"
+#import "mises/ios/browser/ui/toolbar/mises_toolbar_delegate.h"
 
 @protocol PrimaryToolbarViewControllerDelegate;
 @class ViewRevealingVerticalPanHandler;
@@ -18,7 +19,7 @@
 @interface PrimaryToolbarViewController
     : AdaptiveToolbarViewController <SharingPositioner,
                                      KeyCommandActions,
-                                     ToolbarAnimatee>
+                                     ToolbarAnimatee, MisesToolbarDelegate>
 
 @property(nonatomic, weak) id<PrimaryToolbarViewControllerDelegate> delegate;
 
