diff --git a/ios/chrome/browser/ui/toolbar/adaptive_toolbar_coordinator.h b/ios/chrome/browser/ui/toolbar/adaptive_toolbar_coordinator.h
index f9508bd7897da2b62af0b197540dbb72fc018e9e..3e3ce263ddcddbf912671a1c46d4d2ef17cd8ee1 100644
--- a/ios/chrome/browser/ui/toolbar/adaptive_toolbar_coordinator.h
+++ b/ios/chrome/browser/ui/toolbar/adaptive_toolbar_coordinator.h
@@ -44,6 +44,8 @@ class WebState;
 // Shows the animation when transitioning to a prerendered page.
 - (void)showPrerenderingAnimation;
 
+- (void) activate;
+
 @end
 
 #endif  // IOS_CHROME_BROWSER_UI_TOOLBAR_ADAPTIVE_TOOLBAR_COORDINATOR_H_
