diff --git a/ios/chrome/features.gni b/ios/chrome/features.gni
index 8b9bdea4552788e47fb6ab7c21a0006c0ddfa9b1..d3098e6999a02d9c4fdcc82f9b36937c704352ec 100644
--- a/ios/chrome/features.gni
+++ b/ios/chrome/features.gni
@@ -11,7 +11,7 @@ declare_args() {
   # Controls whether Chrome links with Lottie.framework or gets the
   # implementation from its downstream provider. Ignored if Lottie is
   # not build as a framework.
-  ios_chrome_links_with_lottie_framework = true
+  ios_chrome_links_with_lottie_framework = false
 
   # Enable an experimental option to dump sandbox.
   ios_enable_sandbox_dump = false
