diff --git a/ios/chrome/share_extension/entitlements/external/share_extension.appex.entitlements b/ios/chrome/share_extension/entitlements/external/share_extension.appex.entitlements
index 96ad8751490c60a136bcdbc7aedc0ae6d3a48786..14d9052b93f15fafeef6f22792c047425ef18ff6 100644
--- a/ios/chrome/share_extension/entitlements/external/share_extension.appex.entitlements
+++ b/ios/chrome/share_extension/entitlements/external/share_extension.appex.entitlements
@@ -6,7 +6,7 @@
 	<string>$(AppIdentifierPrefix)$(CFBundleIdentifier)</string>
 	<key>com.apple.security.application-groups</key>
 	<array>
-		<string>group.${IOS_BUNDLE_ID_PREFIX}.chrome</string>
+		<string>group.${IOS_BUNDLE_ID_PREFIX}.browser</string>
 		<string>group.${IOS_BUNDLE_ID_PREFIX}.common</string>
 	</array>
 </dict>
