diff --git a/ios/chrome/browser/ui/popup_menu/overflow_menu/overflow_menu_mediator.mm b/ios/chrome/browser/ui/popup_menu/overflow_menu/overflow_menu_mediator.mm
index 11a2a1fc95c707180965bfc4adeb9ae845426fd7..1b1caeade6f397489d60119e2b7bb09ef1df77b4 100644
--- a/ios/chrome/browser/ui/popup_menu/overflow_menu/overflow_menu_mediator.mm
+++ b/ios/chrome/browser/ui/popup_menu/overflow_menu/overflow_menu_mediator.mm
@@ -96,6 +96,9 @@
 #import "ui/base/l10n/l10n_util.h"
 #import "ui/base/l10n/l10n_util_mac.h"
 
+#import "mises/ios/third_party/mises/mises_utils.h"
+#import "components/grit/mises_components_strings.h"
+
 using base::RecordAction;
 using base::UmaHistogramEnumeration;
 using base::UserMetricsAction;
@@ -2225,4 +2228,9 @@ - (void)onShownToggledForDestination:(OverflowMenuDestination*)destination {
                                      actionSubtitle:subtitle];
 }
 
+- (void)openMises {
+  RecordAction(UserMetricsAction("MobileMenuOpenMises"));
+  [Mises popupMisesWallet];
+}
+
 @end
