diff --git a/ios/chrome/browser/ui/content_suggestions/cells/content_suggestions_most_visited_tile_view.mm b/ios/chrome/browser/ui/content_suggestions/cells/content_suggestions_most_visited_tile_view.mm
index 4433e09a53e7507fc47a1e253aa208d5e47d1aad..ac33711a402985652ab252555b5ea8c4fb99af70 100644
--- a/ios/chrome/browser/ui/content_suggestions/cells/content_suggestions_most_visited_tile_view.mm
+++ b/ios/chrome/browser/ui/content_suggestions/cells/content_suggestions_most_visited_tile_view.mm
@@ -58,6 +58,7 @@
     } else {
       _config = config;
       self.titleLabel.text = config.title;
+      self.titleLabel.textColor = [UIColor whiteColor];
       self.accessibilityLabel = config.title;
       _incognitoAvailable = config.incognitoAvailable;
       [_faviconView configureWithAttributes:config.attributes];
