diff --git a/ios/chrome/browser/ui/toolbar/secondary_toolbar_view.mm b/ios/chrome/browser/ui/toolbar/secondary_toolbar_view.mm
index 6fbb2621b96a7a4943c662501a1120e28704bf93..3a47789752575a57f95ee16d34ca7894ce20aaaa 100644
--- a/ios/chrome/browser/ui/toolbar/secondary_toolbar_view.mm
+++ b/ios/chrome/browser/ui/toolbar/secondary_toolbar_view.mm
@@ -85,6 +85,9 @@ @interface SecondaryToolbarView ()
 // redefined as readwrite.
 @property(nonatomic, strong, readwrite) UIButton* collapsedToolbarButton;
 
+@property(nonatomic, strong, readwrite) ToolbarButton* misesButton;
+@property(nonatomic, strong, readwrite) ToolbarButton* walletButton;
+
 @end
 
 @implementation SecondaryToolbarView {
@@ -110,6 +113,9 @@ @implementation SecondaryToolbarView {
 @synthesize toolsMenuButton = _toolsMenuButton;
 @synthesize tabGridButton = _tabGridButton;
 
+@synthesize misesButton = _misesButton;
+@synthesize walletButton = _walletButton;
+
 #pragma mark - Public
 
 - (instancetype)initWithButtonFactory:(ToolbarButtonFactory*)factory {
@@ -166,6 +172,9 @@ - (void)setUp {
   self.openNewTabButton = [self.buttonFactory openNewTabButton];
   self.tabGridButton = [self.buttonFactory tabGridButton];
   self.toolsMenuButton = [self.buttonFactory toolsMenuButton];
+    
+  self.misesButton = [self.buttonFactory misesButton];
+  self.walletButton = [self.buttonFactory walletButton];
 
   // Move the tools menu button such as it looks visually balanced with the
   // button on the other side of the toolbar.
@@ -174,8 +183,8 @@ - (void)setUp {
       CGAffineTransformMakeTranslation(textDirection * kToolsMenuOffset, 0);
 
   self.allButtons = @[
-    self.backButton, self.forwardButton, self.openNewTabButton,
-    self.tabGridButton, self.toolsMenuButton
+    self.misesButton, self.backButton, self.forwardButton, //self.openNewTabButton,
+    self.tabGridButton, self.walletButton, self.toolsMenuButton
   ];
 
   // Separator.
@@ -348,4 +357,22 @@ - (void)updateButtonStackViewConstraint {
   }
 }
 
+
+- (void)updateMisesAvatar:(UIImage*)image {
+
+    if (image == nil) {
+        UIImage* icon = [[UIImage imageNamed:@"mises_user_default"] imageWithRenderingMode:UIImageRenderingModeAlwaysOriginal];
+        [self.misesButton setImage:icon forState:UIControlStateNormal];
+        return;
+    }
+
+    CGSize newSize = CGSizeMake(kAdaptiveToolbarIconSize, kAdaptiveToolbarIconSize);
+    UIGraphicsBeginImageContextWithOptions(newSize, NO, 0.0);
+    [image drawInRect:CGRectMake(0, 0, newSize.width, newSize.height)];
+    UIImage *resized_image = UIGraphicsGetImageFromCurrentImageContext();
+    UIGraphicsEndImageContext();
+    UIImage *icon = [resized_image imageWithRenderingMode:UIImageRenderingModeAlwaysOriginal];
+    [self.misesButton setImage:icon forState:UIControlStateNormal];
+}
+
 @end
