diff --git a/ios/web/js_messaging/web_view_js_utils.mm b/ios/web/js_messaging/web_view_js_utils.mm
index 59eb4efafada7f19d94f4dff03bca37c048bf1ca..096c7927d562ccbad96138f3d8ef46b2d53ecc88 100644
--- a/ios/web/js_messaging/web_view_js_utils.mm
+++ b/ios/web/js_messaging/web_view_js_utils.mm
@@ -73,6 +73,61 @@ std::unique_ptr<base::Value> ValueResultFromWKResult(id wk_result,
   return result;
 }
 
+// Converts base::Value to an equivalent Foundation object, parsing,
+// `value_result` up to a depth of `max_depth`.
+id NSObjectFromValueResult(const base::Value* value_result, int max_depth) {
+  if (!value_result) {
+    return nil;
+  }
+
+  id result = nil;
+
+  if (max_depth < 0) {
+    DLOG(WARNING) << "JS maximum recursion depth exceeded.";
+    return result;
+  }
+
+  if (value_result->is_string()) {
+    result = base::SysUTF8ToNSString(value_result->GetString());
+    DCHECK([result isKindOfClass:[NSString class]]);
+  } else if (value_result->is_int()) {
+    result = [NSNumber numberWithInt:value_result->GetInt()];
+    DCHECK([result isKindOfClass:[NSNumber class]]);
+  } else if (value_result->is_double()) {
+    result = [NSNumber numberWithDouble:value_result->GetDouble()];
+    DCHECK([result isKindOfClass:[NSNumber class]]);
+  } else if (value_result->is_bool()) {
+    result = [NSNumber numberWithBool:value_result->GetBool()];
+    DCHECK([result isKindOfClass:[NSNumber class]]);
+  } else if (value_result->is_none()) {
+    result = [NSNull null];
+    DCHECK([result isKindOfClass:[NSNull class]]);
+  } else if (value_result->is_dict()) {
+    NSMutableDictionary* dictionary = [[NSMutableDictionary alloc] init];
+    for (const auto pair : value_result->GetDict()) {
+      NSString* key = base::SysUTF8ToNSString(pair.first);
+      id wk_result = NSObjectFromValueResult(&pair.second, max_depth - 1);
+      if (wk_result) {
+        [dictionary setValue:wk_result forKey:key];
+      }
+    }
+    result = [dictionary copy];
+  } else if (value_result->is_list()) {
+    NSMutableArray* array = [[NSMutableArray alloc] init];
+    for (const base::Value& value : value_result->GetList()) {
+      id wk_result = NSObjectFromValueResult(&value, max_depth - 1);
+      if (wk_result) {
+        [array addObject:wk_result];
+      }
+    }
+    result = [array copy];
+  } else {
+    NOTREACHED();  // Convert other types as needed.
+  }
+  return result;
+}
+
+
 // Converts base::Value to an equivalent Foundation object, parsing,
 // `value_result` up to a depth of `max_depth`.
 id NSObjectFromValueResult(const base::Value* value_result, int max_depth) {
@@ -157,6 +212,10 @@ id NSObjectFromValueResult(const base::Value* value_result) {
   return ::NSObjectFromValueResult(value_result, kMaximumParsingRecursionDepth);
 }
 
+id NSObjectFromValueResult(const base::Value* value_result) {
+  return ::NSObjectFromValueResult(value_result, kMaximumParsingRecursionDepth);
+}
+
 void ExecuteJavaScript(WKWebView* web_view,
                        NSString* script,
                        void (^completion_handler)(id, NSError*)) {
