diff --git a/ios/chrome/browser/ui/content_suggestions/content_suggestions_favicon_mediator.mm b/ios/chrome/browser/ui/content_suggestions/content_suggestions_favicon_mediator.mm
index a7e9dce8b910c5d579e6cdeefd9d4f31aeddf099..ae2ec36ea5207a1cac7f7358807df98d1275bb95 100644
--- a/ios/chrome/browser/ui/content_suggestions/content_suggestions_favicon_mediator.mm
+++ b/ios/chrome/browser/ui/content_suggestions/content_suggestions_favicon_mediator.mm
@@ -17,6 +17,14 @@
 #import "ios/chrome/browser/ui/favicon/favicon_attributes_with_payload.h"
 #import "ui/gfx/image/image.h"
 
+#include "components/image_fetcher/core/image_fetcher_impl.h"
+#include "components/image_fetcher/core/cached_image_fetcher.h"
+#include "components/image_fetcher/ios/ios_image_decoder_impl.h"
+#include "ios/chrome/browser/shared/model/browser_state/chrome_browser_state.h"
+#include "base/time/default_clock.h"
+#include "services/network/public/cpp/shared_url_loader_factory.h"
+#import "ios/chrome/browser/shared/model/browser/browser.h"
+
 namespace {
 
 // Size of the favicon returned by the provider for the most visited items.
@@ -25,6 +33,8 @@
 const CGFloat kMostVisitedFaviconMinimalSize = 32;
 const CGFloat kMagicStackMostVisitedFaviconMinimalSize = 18;
 
+const char kImageFetcherUmaClient[] = "ContentSuggestionsFaviconMediator";
+
 }  // namespace
 
 @interface ContentSuggestionsFaviconMediator () {
@@ -41,6 +51,9 @@ @interface ContentSuggestionsFaviconMediator () {
 @end
 
 @implementation ContentSuggestionsFaviconMediator
+{
+    std::unique_ptr<image_fetcher::ImageFetcher> _imageFetcher;
+}
 
 #pragma mark - Public.
 
@@ -64,6 +77,10 @@ - (instancetype)initWithLargeIconService:
   return self;
 }
 
+- (void)setImageFetcher:(Browser*) browser {
+    ChromeBrowserState* browser_state = browser->GetBrowserState();
+    _imageFetcher = std::make_unique<image_fetcher::ImageFetcherImpl>(image_fetcher::CreateIOSImageDecoder(),browser_state->GetSharedURLLoaderFactory());
+}
 - (void)setMostVisitedDataForLogging:
     (const ntp_tiles::NTPTilesVector&)mostVisitedData {
   DCHECK(_mostVisitedDataForLogging.empty());
@@ -79,14 +96,52 @@ - (void)fetchFaviconForMostVisited:(ContentSuggestionsMostVisitedItem*)item {
     ContentSuggestionsMostVisitedItem* strongItem = weakItem;
     if (!strongSelf || !strongItem)
       return;
-
     strongItem.attributes = attributes;
     [strongSelf logFaviconFetchedForItem:strongItem];
+
     [strongSelf.consumer updateMostVisitedTileConfig:strongItem];
   };
-
+    if (item.itemType == ITEM_TYPE_MISES_FEATURE) {
+      auto callback =
+            base::BindOnce(^(const gfx::Image& gfximage,
+                             const image_fetcher::RequestMetadata& metadata) {
+            ContentSuggestionsFaviconMediator* strongSelf = weakSelf;
+            ContentSuggestionsMostVisitedItem* strongItem = weakItem;
+            if (!strongSelf || !strongItem || gfximage.IsEmpty()) {
+              return;
+            }
+            UIImage* image = [gfximage.ToUIImage() copy];
+            strongItem.attributes = [FaviconAttributesWithPayload attributesWithImage:image];
+            [strongSelf logFaviconFetchedForItem:strongItem];
+
+            [strongSelf.consumer updateMostVisitedTileConfig:strongItem];
+        });
+        net::NetworkTrafficAnnotationTag traffic_annotation =
+            net::DefineNetworkTrafficAnnotation("icon_cacher", R"(
+              semantics {
+                sender: "Popular Sites New Tab Fetch"
+                description:
+                  "Chrome may display a list of regionally-popular web sites on the "
+                  "New Tab Page. This service fetches icons from those sites."
+                trigger:
+                  "Whenever a popular site would be displayed, but its icon is not "
+                  "yet cached in the browser."
+                data: "The URL for which to retrieve an icon."
+                destination: WEBSITE
+              }
+              policy {
+                cookies_allowed: NO
+                setting: "This feature cannot be disabled in settings."
+                policy_exception_justification: "Not implemented."
+              })");
+      image_fetcher::ImageFetcherParams params(traffic_annotation,
+                                               kImageFetcherUmaClient);
+      _imageFetcher->FetchImage(item.iconURL, std::move(callback), params);
+      return;
+    }
   [self.mostVisitedAttributesProvider fetchFaviconAttributesForURL:item.URL
-                                                        completion:completion];
+                                                              completion:completion];
+
 }
 
 
