diff --git a/ios/chrome/browser/ui/toolbar/primary_toolbar_view.h b/ios/chrome/browser/ui/toolbar/primary_toolbar_view.h
index 4a9f9da0efc28a966ee446f40fbd8e80490ca333..3bcca95a8d6915d13607cfecfbae4ef8e182237b 100644
--- a/ios/chrome/browser/ui/toolbar/primary_toolbar_view.h
+++ b/ios/chrome/browser/ui/toolbar/primary_toolbar_view.h
@@ -73,6 +73,8 @@
 // Removes `fakeOmniboxTarget` from the view hierarchy.
 - (void)removeFakeOmniboxTarget;
 
+- (void)updateMisesAvatar:(UIImage*)image;
+
 @end
 
 #endif  // IOS_CHROME_BROWSER_UI_TOOLBAR_PRIMARY_TOOLBAR_VIEW_H_
