diff --git a/ios/chrome/browser/ui/toolbar/adaptive_toolbar_view.h b/ios/chrome/browser/ui/toolbar/adaptive_toolbar_view.h
index 9c9b38d6fdbde99ac815ac42ff040245dd3f7040..1b8aad89dd4dd964b8689536d4c65092da0f0b53 100644
--- a/ios/chrome/browser/ui/toolbar/adaptive_toolbar_view.h
+++ b/ios/chrome/browser/ui/toolbar/adaptive_toolbar_view.h
@@ -51,6 +51,9 @@
 // Sets the location bar view containing the omnibox.
 - (void)setLocationBarView:(UIView*)locationBarView;
 
+@property(nonatomic, strong, readonly) ToolbarButton* misesButton;
+@property(nonatomic, strong, readonly) ToolbarButton* walletButton;
+
 @end
 
 #endif  // IOS_CHROME_BROWSER_UI_TOOLBAR_ADAPTIVE_TOOLBAR_VIEW_H_
