diff --git a/ios/chrome/browser/ui/scanner/camera_controller.mm b/ios/chrome/browser/ui/scanner/camera_controller.mm
index 7d16fbc832533186fc0aefa9a9b9183f91dd066b..4431b5f976bd6d0cec067a11ad47c620068ffb1b 100644
--- a/ios/chrome/browser/ui/scanner/camera_controller.mm
+++ b/ios/chrome/browser/ui/scanner/camera_controller.mm
@@ -65,7 +65,7 @@
     _cameraState = scanner::CAMERA_NOT_LOADED;
     _delegate = delegate;
     std::string queueName =
-        base::StringPrintf("%s.chrome.ios.QRScannerCaptureSessionQueue",
+        base::StringPrintf("%s.browser.QRScannerCaptureSessionQueue",
                            BUILDFLAG(IOS_APP_BUNDLE_ID_PREFIX));
     _sessionQueue =
         dispatch_queue_create(queueName.c_str(), DISPATCH_QUEUE_SERIAL);
