diff --git a/ios/web/js_messaging/scoped_wk_script_message_handler.h b/ios/web/js_messaging/scoped_wk_script_message_handler.h
index 929724f09cc2edf707da6b212ce6acd28e45e436..63f2909a8eef33c4d9ed4e9de7c811c9714f7c77 100644
--- a/ios/web/js_messaging/scoped_wk_script_message_handler.h
+++ b/ios/web/js_messaging/scoped_wk_script_message_handler.h
@@ -9,11 +9,21 @@
 
 #include "base/functional/callback.h"
 
+namespace base {
+class Value;
+}  // namespace base
+
 @class CRWScriptMessageHandler;
+@class CRWScriptMessageHandlerWithReply;
 
 using ScriptMessageCallback =
     base::RepeatingCallback<void(WKScriptMessage* message)>;
 
+using ScriptMessageReplyHandler = void (^)(const base::Value* reply,
+                                           NSString* error_message);
+using ScriptMessageWithReplyCallback =
+    base::RepeatingCallback<void(WKScriptMessage* message,
+                                 ScriptMessageReplyHandler reply_handler)>;
 // Instances of this class register and unregister itself as a
 // WKUserContentController script message handler upon construction and
 // deconstruction respectively.
@@ -36,6 +46,11 @@ class ScopedWKScriptMessageHandler {
                                WKContentWorld* content_world,
                                ScriptMessageCallback callback);
 
+    ScopedWKScriptMessageHandler(WKUserContentController* user_content_controller,
+                                   NSString* script_handler_name,
+                                   WKContentWorld* content_world,
+                                   ScriptMessageWithReplyCallback callback);
+
   ~ScopedWKScriptMessageHandler();
 
  private:
@@ -48,6 +63,7 @@ class ScopedWKScriptMessageHandler {
   NSString* script_handler_name_;
 
   CRWScriptMessageHandler* script_message_handler_;
+    CRWScriptMessageHandlerWithReply* script_message_handler_with_reply_;
 
   ScopedWKScriptMessageHandler(const ScopedWKScriptMessageHandler&) = delete;
   ScopedWKScriptMessageHandler& operator=(const ScopedWKScriptMessageHandler&) =
