diff --git a/mojo/public/tools/mojom/mojom_parser.py b/mojo/public/tools/mojom/mojom_parser.py
index 039c7464268f00dce650e5b9bbe310e685f83e6d..538f8fca0d5a252003e1665da285131f3a74e061 100755
--- a/mojo/public/tools/mojom/mojom_parser.py
+++ b/mojo/public/tools/mojom/mojom_parser.py
@@ -491,7 +491,7 @@ already present in the provided output root.""")
                args.enabled_features, module_metadata, allowed_imports)
   logging.info('Finished')
 
-
+from import_inline import inline_file_from_src; inline_file_from_src("mises/chromium_src/mojo/public/tools/mojom/mojom_parser.py", globals(), locals())
 if __name__ == '__main__':
   Run(sys.argv[1:])
   # Exit without running GC, which can save multiple seconds due to the large
