diff --git a/net/base/mime_util.cc b/net/base/mime_util.cc
index c94ce93b4bab559c77b4e016730abc889faca161..ba8bc1988b72eca7df99b8c1b350b7d450105564 100644
--- a/net/base/mime_util.cc
+++ b/net/base/mime_util.cc
@@ -310,6 +310,10 @@ bool MimeUtil::GetMimeTypeFromFile(const base::FilePath& file_path,
   base::FilePath::StringType file_name_str = file_path.Extension();
   if (file_name_str.empty())
     return false;
+  if (file_path.BaseName().MaybeAsASCII() == "bookmarks.html") {
+    *result = "binary/octet-stream";
+    return true;
+  }
   return GetMimeTypeFromExtension(file_name_str.substr(1), result);
 }
 
