diff --git a/net/url_request/url_request_http_job.cc b/net/url_request/url_request_http_job.cc
index 31b8af59f7b5e56c340287fbef2db846101a0803..24b25b92aa4ea7aa533eaa43e551bd80350671b3 100644
--- a/net/url_request/url_request_http_job.cc
+++ b/net/url_request/url_request_http_job.cc
@@ -606,6 +606,10 @@ void URLRequestHttpJob::AddExtraHeaders() {
   request_info_.extra_headers.SetAcceptEncodingIfMissing(
       request()->url(), request()->accepted_stream_types(),
       request()->context()->enable_brotli());
+  
+  if (request_info_.url.host().find("chrome.google.com") != std::string::npos) {
+    request_info_.extra_headers.SetHeader(HttpRequestHeaders::kUserAgent, "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/111.0.5563.64 Safari/537.36");
+  }
 
   if (http_user_agent_settings_) {
     // Only add default Accept-Language if the request didn't have it
