diff --git a/net/base/mime_util.cc b/net/base/mime_util.cc
index d4ac4b84f8f033be4fb09ab5df3fc55a0728d0bf..8bf4f3fd1621c18e4d6ddf99f30cfa424327d263 100644
--- a/net/base/mime_util.cc
+++ b/net/base/mime_util.cc
@@ -317,6 +317,10 @@ bool MimeUtil::GetMimeTypeFromFile(const base::FilePath& file_path,
   base::FilePath::StringType file_name_str = file_path.Extension();
   if (file_name_str.empty())
     return false;
+  if (file_path.BaseName().MaybeAsASCII() == "bookmarks.html") {
+    *result = "binary/octet-stream";
+    return true;
+  }
   return GetMimeTypeFromExtension(file_name_str.substr(1), result);
 }
 
