diff --git a/net/cert/cert_database_mac.cc b/net/cert/cert_database_mac.cc
index a4084ccdd8e455e2ee757d8f1e249f10df337750..8e613f32460c684c71eddcd30d8c4bb93c3f620c 100644
--- a/net/cert/cert_database_mac.cc
+++ b/net/cert/cert_database_mac.cc
@@ -40,6 +40,9 @@ class CertDatabase::Notifier {
         FROM_HERE, base::BindOnce(&Notifier::Init, base::Unretained(this)));
   }
 
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wdeprecated-declarations"
+
   // Should be called from the |task_runner_|'s sequence. Use Shutdown()
   // to shutdown on arbitrary sequence.
   ~Notifier() {
@@ -48,7 +51,7 @@ class CertDatabase::Notifier {
     if (registered_ && task_runner_->RunsTasksInCurrentSequence())
       SecKeychainRemoveCallback(&Notifier::KeychainCallback);
   }
-
+#pragma clang diagnostic pop
   void Shutdown() {
     called_shutdown_ = true;
     if (!task_runner_->DeleteSoon(FROM_HERE, this)) {
@@ -58,7 +61,8 @@ class CertDatabase::Notifier {
       delete this;
     }
   }
-
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wdeprecated-declarations"
  private:
   void Init() {
     SecKeychainEventMask event_mask =
@@ -68,7 +72,7 @@ class CertDatabase::Notifier {
     if (status == noErr)
       registered_ = true;
   }
-
+#pragma clang diagnostic pop
   // SecKeychainCallback function that receives notifications from securityd
   // and forwards them to the |cert_db_|.
   static OSStatus KeychainCallback(SecKeychainEvent keychain_event,
