diff --git a/net/cookies/canonical_cookie.cc b/net/cookies/canonical_cookie.cc
index 91d210fa1683da314922a15734d4fabbde8ab873..f6c2013c8013b878e429f95499b5805bdc59a37c 100644
--- a/net/cookies/canonical_cookie.cc
+++ b/net/cookies/canonical_cookie.cc
@@ -665,7 +665,7 @@ std::unique_ptr<CanonicalCookie> CanonicalCookie::Create(
                                          : CookieSourceScheme::kNonSecure;
   // Get the port, this will get a default value if a port isn't provided.
   int source_port = ValidateAndAdjustSourcePort(url.EffectiveIntPort());
-
+  BRAVE_CREATE
   std::unique_ptr<CanonicalCookie> cc = base::WrapUnique(new CanonicalCookie(
       parsed_cookie.Name(), parsed_cookie.Value(), cookie_domain, cookie_path,
       creation_time, cookie_expires, creation_time,
