diff --git a/net/test/keychain_test_util_mac.cc b/net/test/keychain_test_util_mac.cc
index 4a297004e1b96bb9fc56fd3ef45d6641dcd9053e..57ec2089b095c56b180f37f5b691df83214ec1f2 100644
--- a/net/test/keychain_test_util_mac.cc
+++ b/net/test/keychain_test_util_mac.cc
@@ -37,7 +37,8 @@ base::ScopedCFTypeRef<SecIdentityRef> GetSecIdentityRefForCertificate(
 
 ScopedTestKeychain::ScopedTestKeychain() = default;
 ScopedTestKeychain::~ScopedTestKeychain() = default;
-
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wdeprecated-declarations"
 bool ScopedTestKeychain::Initialize() {
   if (!keychain_dir_.CreateUniqueTempDir())
     return false;
@@ -46,7 +47,7 @@ bool ScopedTestKeychain::Initialize() {
   return SecKeychainCreate(keychain_path.value().c_str(), 0, "", FALSE, nullptr,
                            keychain_.InitializeInto()) == noErr;
 }
-
+#pragma clang diagnostic pop
 base::ScopedCFTypeRef<SecIdentityRef> ImportCertAndKeyToKeychain(
     const X509Certificate* cert,
     const std::string pkcs8,
