diff --git a/net/cert/internal/trust_store_mac.cc b/net/cert/internal/trust_store_mac.cc
index 589817b57b1abb629121640207e3f6ca8bbda1c5..7842ee5623b0668d3ae44d1091e8fd85b39fd6d9 100644
--- a/net/cert/internal/trust_store_mac.cc
+++ b/net/cert/internal/trust_store_mac.cc
@@ -30,9 +30,9 @@
 #include "net/cert/x509_util.h"
 #include "net/cert/x509_util_apple.h"
 #include "third_party/boringssl/src/include/openssl/sha.h"
-
 namespace net {
-
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wdeprecated-declarations"
 namespace {
 
 // The rules for interpreting trust settings are documented at:
@@ -613,7 +613,6 @@ class KeychainTrustSettingsChangedNotifier {
 
  private:
   friend base::NoDestructor<KeychainTrustSettingsChangedNotifier>;
-
   KeychainTrustSettingsChangedNotifier() {
     DCHECK(GetNetworkNotificationThreadMac()->RunsTasksInCurrentSequence());
     OSStatus status = SecKeychainAddCallback(
@@ -1357,5 +1356,5 @@ base::ScopedCFTypeRef<CFDataRef> TrustStoreMac::GetMacNormalizedIssuer(
     LOG(ERROR) << "SecCertificateCopyNormalizedIssuerContent";
   return name_data;
 }
-
+#pragma clang diagnostic pop
 }  // namespace net
