diff --git a/remoting/android/java/src/org/chromium/chromoting/PrivacyPolicyActivity.java b/remoting/android/java/src/org/chromium/chromoting/PrivacyPolicyActivity.java
index 8f4ec28b43d52cf43042b8ba7141664c9276c71b..9d01bbe94e77367f88db555dd55a9edb06816895 100644
--- a/remoting/android/java/src/org/chromium/chromoting/PrivacyPolicyActivity.java
+++ b/remoting/android/java/src/org/chromium/chromoting/PrivacyPolicyActivity.java
@@ -8,7 +8,7 @@ package org.chromium.chromoting;
  * The Activity for showing the privacy policy.
  */
 public class PrivacyPolicyActivity extends WebViewActivity {
-    private static final String POLICY_URL = "https://policies.google.com/privacy";
+    private static final String POLICY_URL = "https://www.mises.site/privacy";
 
     public PrivacyPolicyActivity() {
         super(R.string.privacy_policy, POLICY_URL);
