diff --git a/services/device/public/cpp/hid/BUILD.gn b/services/device/public/cpp/hid/BUILD.gn
index fc9faa72e8155d173c960edf6af8ff2c8ffff626..2da950884fbea83b70c662b3f404b9043765e219 100644
--- a/services/device/public/cpp/hid/BUILD.gn
+++ b/services/device/public/cpp/hid/BUILD.gn
@@ -8,7 +8,7 @@ import("//build/config/features.gni")
 source_set("hid") {
   # HID is not implemented on Android and we want to be particularly careful
   # about not bloating binary size by accidentially including it.
-  assert(!is_android)
+  # assert(!is_android)
 
   sources = [
     "hid_blocklist.cc",
