diff --git a/services/device/public/cpp/BUILD.gn b/services/device/public/cpp/BUILD.gn
index 4724c107c4d25bcf00038b4244457c2809759723..7c638524dc95ee1f3d14af5d42ad45498139fa55 100644
--- a/services/device/public/cpp/BUILD.gn
+++ b/services/device/public/cpp/BUILD.gn
@@ -92,6 +92,8 @@ source_set("test_support") {
     sources += [
       "test/scoped_nfc_overrider.cc",
       "test/scoped_nfc_overrider.h",
+      "test/fake_hid_manager.cc",
+      "test/fake_hid_manager.h",
     ]
   } else {
     sources += [
