diff --git a/services/device/public/cpp/hid/BUILD.gn b/services/device/public/cpp/hid/BUILD.gn
index 8585a16c336912ee81e8d84d909043685b16534c..e343e618f664bdd3a8406e51400263ee7c02382e 100644
--- a/services/device/public/cpp/hid/BUILD.gn
+++ b/services/device/public/cpp/hid/BUILD.gn
@@ -8,7 +8,7 @@ import("//build/config/features.gni")
 source_set("hid") {
   # HID is not implemented on Android and we want to be particularly careful
   # about not bloating binary size by accidentially including it.
-  assert(!is_android)
+  # assert(!is_android)
 
   sources = [
     "hid_blocklist.cc",
