diff --git a/services/device/public/mojom/BUILD.gn b/services/device/public/mojom/BUILD.gn
index 7a61ba6cd0fb10ae8370a3611215f1ca87e32d45..38e7262501fa1426f80cda27374b3028e8971a65 100644
--- a/services/device/public/mojom/BUILD.gn
+++ b/services/device/public/mojom/BUILD.gn
@@ -94,7 +94,7 @@ mojom("device_service") {
     enabled_features += [ "enable_input_device_manager" ]
   }
 
-  if (!is_android) {
+  if (is_android || !is_android) {
     enabled_features += [ "enable_hid" ]
   }
 
