diff --git a/services/device/public/cpp/BUILD.gn b/services/device/public/cpp/BUILD.gn
index a08a1dbfa49df17a2809de7248b902dd243a5a2b..33e5fa4d8a8c60c197f14a66e8dc8eb00b7fe77d 100644
--- a/services/device/public/cpp/BUILD.gn
+++ b/services/device/public/cpp/BUILD.gn
@@ -87,6 +87,8 @@ source_set("test_support") {
     sources += [
       "test/scoped_nfc_overrider.cc",
       "test/scoped_nfc_overrider.h",
+      "test/fake_hid_manager.cc",
+      "test/fake_hid_manager.h",
     ]
   } else {
     sources += [
