diff --git a/services/device/public/mojom/BUILD.gn b/services/device/public/mojom/BUILD.gn
index 4428743e695e9f222bcf2d744e236fbbc0187af6..3167a92a7f61a9432151ddc5de37ad8bc56d3e9b 100644
--- a/services/device/public/mojom/BUILD.gn
+++ b/services/device/public/mojom/BUILD.gn
@@ -102,7 +102,7 @@ mojom("device_service") {
     enabled_features += [ "enable_input_device_manager" ]
   }
 
-  if (!is_android) {
+  if (is_android || !is_android) {
     enabled_features += [ "enable_hid" ]
   }
 
