diff --git a/services/device/public/cpp/hid/BUILD.gn b/services/device/public/cpp/hid/BUILD.gn
index 29ca1c8555fba8c62e5e7595e6ff51d00a182c8a..f71bf4735d8dcb7374215cbede38d01a3967350d 100644
--- a/services/device/public/cpp/hid/BUILD.gn
+++ b/services/device/public/cpp/hid/BUILD.gn
@@ -8,7 +8,7 @@ import("//build/config/features.gni")
 source_set("hid") {
   # HID is not implemented on Android and we want to be particularly careful
   # about not bloating binary size by accidentially including it.
-  assert(!is_android)
+  # assert(!is_android)
 
   sources = [
     "hid_blocklist.cc",
