diff --git a/services/device/public/mojom/BUILD.gn b/services/device/public/mojom/BUILD.gn
index f2d13c795f790b905682f0c0bb12df86173d24a6..ab55ad971dcdab9eb57317319861268c8de7b7bf 100644
--- a/services/device/public/mojom/BUILD.gn
+++ b/services/device/public/mojom/BUILD.gn
@@ -93,7 +93,7 @@ mojom("device_service") {
     enabled_features += [ "enable_input_device_manager" ]
   }
 
-  if (!is_android) {
+  if (is_android || !is_android) {
     enabled_features += [ "enable_hid" ]
   }
 
