diff --git a/third_party/blink/public/mojom/BUILD.gn b/third_party/blink/public/mojom/BUILD.gn
index 370568de1b0259186921806818bde2400af0fab2..cb6275edbb465e4df515c4b853df3ed99d1bdf7d 100644
--- a/third_party/blink/public/mojom/BUILD.gn
+++ b/third_party/blink/public/mojom/BUILD.gn
@@ -347,8 +347,10 @@ mojom("mojom_platform") {
       "//services/network/public/mojom:mojom_storage_access_api",
       "//ui/accessibility:ax_enums_mojo",
       "//ui/latency/mojom",
+      "//services/network/public/mojom:mojom_host_resolver",
     ]
-  } else {
+  }
+  if (true) {
     sources += [
       "direct_sockets/direct_sockets.mojom",
       "printing/web_printing.mojom",
