diff --git a/third_party/blink/public/mojom/BUILD.gn b/third_party/blink/public/mojom/BUILD.gn
index 741b3a25cddc2fec1ca168c2c662972fd32f5d9a..b5086556788167d59188c763dc24e28a32021325 100644
--- a/third_party/blink/public/mojom/BUILD.gn
+++ b/third_party/blink/public/mojom/BUILD.gn
@@ -309,7 +309,8 @@ mojom("mojom_platform") {
       "//ui/accessibility:ax_enums_mojo",
       "//ui/latency/mojom",
     ]
-  } else {
+  }
+  if (true) {
     sources += [
       "direct_sockets/direct_sockets.mojom",
       "serial/serial.mojom",
