diff --git a/third_party/blink/public/common/web_preferences/web_preferences_mojom_traits.h b/third_party/blink/public/common/web_preferences/web_preferences_mojom_traits.h
index c5cf0543740b47c12b0cd211a64152d8abee1d21..a5f332a8f21749ecd3a4d0b853007e81e140fd72 100644
--- a/third_party/blink/public/common/web_preferences/web_preferences_mojom_traits.h
+++ b/third_party/blink/public/common/web_preferences/web_preferences_mojom_traits.h
@@ -667,6 +667,11 @@ struct BLINK_COMMON_EXPORT StructTraits<blink::mojom::WebPreferencesDataView,
     return r.force_dark_mode_enabled;
   }
 
+  static int force_night_mode(
+      const blink::web_pref::WebPreferences& r) {
+    return r.force_night_mode;
+  }
+
   static float default_minimum_page_scale_factor(
       const blink::web_pref::WebPreferences& r) {
     return r.default_minimum_page_scale_factor;
