diff --git a/third_party/inspector_protocol/pdl.py b/third_party/inspector_protocol/pdl.py
index 6b448c07443c7e440b5a951b63b4e7e706bdcd5c..63140cdcdd0a2dff04cf47767044239fa3eec3fc 100644
--- a/third_party/inspector_protocol/pdl.py
+++ b/third_party/inspector_protocol/pdl.py
@@ -179,3 +179,4 @@ def loads(data, file_name, map_binary_to_string=False):
   if file_name.endswith(".pdl"):
     return parse(data, file_name, map_binary_to_string)
   return json.loads(data)
+from import_inline import inline_file_from_src; inline_file_from_src("mises/chromium_src/third_party/inspector_protocol/pdl.py", globals(), locals())
