diff --git a/third_party/blink/common/web_preferences/web_preferences_mojom_traits.cc b/third_party/blink/common/web_preferences/web_preferences_mojom_traits.cc
index bd5d2f9ba30d847a44e70c1173eb49a5d3f4b6bb..0fe8cc50adbf18ac30bce1e79be54ab4e6c7e0f3 100644
--- a/third_party/blink/common/web_preferences/web_preferences_mojom_traits.cc
+++ b/third_party/blink/common/web_preferences/web_preferences_mojom_traits.cc
@@ -203,6 +203,7 @@ bool StructTraits<blink::mojom::WebPreferencesDataView,
 #endif  // BUILDFLAG(IS_ANDROID) || BUILDFLAG(IS_FUCHSIA)
 
   out->force_dark_mode_enabled = data.force_dark_mode_enabled();
+  out->force_night_mode = data.force_night_mode();
   out->default_minimum_page_scale_factor =
       data.default_minimum_page_scale_factor();
   out->default_maximum_page_scale_factor =
