diff --git a/third_party/android_deps/BUILD.gn b/third_party/android_deps/BUILD.gn
index 66f105d4cddb0d37428880c62f173e21c6f7fb23..d0c3750458527949e55c08472de8d4f2a1e4f382 100644
--- a/third_party/android_deps/BUILD.gn
+++ b/third_party/android_deps/BUILD.gn
@@ -1922,6 +1922,7 @@ if (!limit_android_deps) {
     visibility = [
       ":*",
       "//third_party/androidx:*",
+      "//mises/third_party/android_deps:*",
     ]
     deps = [
       ":google_play_services_basement_java",
@@ -2021,6 +2022,7 @@ if (!limit_android_deps) {
     visibility = [
       ":*",
       "//third_party/androidx:*",
+      "//mises/third_party/android_deps:*",
     ]
     deps = [
       ":google_firebase_firebase_components_java",
@@ -2043,6 +2045,7 @@ if (!limit_android_deps) {
     visibility = [
       ":*",
       "//third_party/androidx:*",
+      "//mises/third_party/android_deps:*",
     ]
     deps = [
       ":google_firebase_firebase_annotations_java",
@@ -2112,6 +2115,7 @@ if (!limit_android_deps) {
     visibility = [
       ":*",
       "//third_party/androidx:*",
+      "//mises/third_party/android_deps:*",
     ]
     deps = [
       ":google_firebase_firebase_common_java",
@@ -2134,6 +2138,7 @@ if (!limit_android_deps) {
     visibility = [
       ":*",
       "//third_party/androidx:*",
+      "//mises/third_party/android_deps:*",
     ]
     deps = [
       ":google_firebase_firebase_annotations_java",
@@ -2154,6 +2159,7 @@ if (!limit_android_deps) {
     visibility = [
       ":*",
       "//third_party/androidx:*",
+      "//mises/third_party/android_deps:*",
     ]
     deps = [ ":google_play_services_basement_java" ]
   }
