diff --git a/third_party/android_deps/BUILD.gn b/third_party/android_deps/BUILD.gn
index 66f105d4cddb0d37428880c62f173e21c6f7fb23..f7168e44692a555335ed648b2407f14661a49514 100644
--- a/third_party/android_deps/BUILD.gn
+++ b/third_party/android_deps/BUILD.gn
@@ -1922,6 +1922,7 @@ if (!limit_android_deps) {
     visibility = [
       ":*",
       "//third_party/androidx:*",
+      "//mises/third_party/android_deps:*",
     ]
     deps = [
       ":google_play_services_basement_java",
@@ -1988,6 +1989,7 @@ if (!limit_android_deps) {
     # //third_party/android_deps/build.gradle.
     visibility = [
       ":*",
+      "//mises/third_party/android_deps:*",
       "//third_party/androidx:*",
     ]
     deps = [
@@ -2021,6 +2023,7 @@ if (!limit_android_deps) {
     visibility = [
       ":*",
       "//third_party/androidx:*",
+      "//mises/third_party/android_deps:*",
     ]
     deps = [
       ":google_firebase_firebase_components_java",
@@ -2043,6 +2046,7 @@ if (!limit_android_deps) {
     visibility = [
       ":*",
       "//third_party/androidx:*",
+      "//mises/third_party/android_deps:*",
     ]
     deps = [
       ":google_firebase_firebase_annotations_java",
@@ -2112,6 +2116,7 @@ if (!limit_android_deps) {
     visibility = [
       ":*",
       "//third_party/androidx:*",
+      "//mises/third_party/android_deps:*",
     ]
     deps = [
       ":google_firebase_firebase_common_java",
@@ -2134,6 +2139,7 @@ if (!limit_android_deps) {
     visibility = [
       ":*",
       "//third_party/androidx:*",
+      "//mises/third_party/android_deps:*",
     ]
     deps = [
       ":google_firebase_firebase_annotations_java",
@@ -2154,6 +2160,7 @@ if (!limit_android_deps) {
     visibility = [
       ":*",
       "//third_party/androidx:*",
+      "//mises/third_party/android_deps:*",
     ]
     deps = [ ":google_play_services_basement_java" ]
   }
