diff --git a/third_party/blink/public/common/web_preferences/web_preferences_mojom_traits.h b/third_party/blink/public/common/web_preferences/web_preferences_mojom_traits.h
index cc2d2e176476332546b76d33adedbbf760c6e9f6..894ee62160fe81cf32fb648300ff6e3fc8b2d5c1 100644
--- a/third_party/blink/public/common/web_preferences/web_preferences_mojom_traits.h
+++ b/third_party/blink/public/common/web_preferences/web_preferences_mojom_traits.h
@@ -662,6 +662,11 @@ struct BLINK_COMMON_EXPORT StructTraits<blink::mojom::WebPreferencesDataView,
     return r.force_dark_mode_enabled;
   }
 
+  static int force_night_mode(
+      const blink::web_pref::WebPreferences& r) {
+    return r.force_night_mode;
+  }
+
   static float default_minimum_page_scale_factor(
       const blink::web_pref::WebPreferences& r) {
     return r.default_minimum_page_scale_factor;
