diff --git a/third_party/blink/renderer/platform/theme/web_theme_engine_helper.h b/third_party/blink/renderer/platform/theme/web_theme_engine_helper.h
index 39936541a1c79cea6c90e77d6a126df05323dc2c..a497bdc5b2587772c5dcba2a73c2fec2b7d53d45 100644
--- a/third_party/blink/renderer/platform/theme/web_theme_engine_helper.h
+++ b/third_party/blink/renderer/platform/theme/web_theme_engine_helper.h
@@ -20,6 +20,8 @@ class WebThemeEngineHelper {
   // Swaps the current theme engine out returning the old one.
   PLATFORM_EXPORT static std::unique_ptr<WebThemeEngine>
   SwapNativeThemeEngineForTesting(std::unique_ptr<WebThemeEngine> new_theme);
+  PLATFORM_EXPORT static bool GlobalForceDarkModeEnabled();
+  PLATFORM_EXPORT static void SetGlobalForceDarkModeEnabled(bool enable);
 };
 
 }  // namespace blink
