diff --git a/third_party/android_deps/build.gradle b/third_party/android_deps/build.gradle
index 96cf692ff33e26bbc5f84c0d5f37f6c55e62d15f..a24efbbce28f7c9ff58089f41646b7bb3b6ce3bb 100644
--- a/third_party/android_deps/build.gradle
+++ b/third_party/android_deps/build.gradle
@@ -198,5 +198,5 @@ task setUpRepository(type: BuildConfigGenerator) {
     repositoryPath 'third_party/android_deps'
     chromiumSourceRoot '../..'
     cipdBucket 'chromium'
-    internalTargetVisibility = [ ':*', '//third_party/androidx:*' ]
+    internalTargetVisibility = [ ':*', '//third_party/androidx:*', '//mises/third_party/android_deps*' ]
 }
