diff --git a/third_party/lottie/BUILD.gn b/third_party/lottie/BUILD.gn
index 3ba00132768093296d87b5cd7e2a64ecabb55554..ae07d6bb1fa8826d57e4f524370a4a5328ca9b5a 100644
--- a/third_party/lottie/BUILD.gn
+++ b/third_party/lottie/BUILD.gn
@@ -6,7 +6,7 @@ import("//third_party/closure_compiler/compile_js.gni")
 import("//third_party/node/node.gni")
 import("//ui/webui/resources/tools/generate_grd.gni")
 
-assert(!is_android && !is_ios)
+#assert(!is_android && !is_ios)
 
 js_library("lottie_worker") {
 }
