diff --git a/third_party/lottie/BUILD.gn b/third_party/lottie/BUILD.gn
index 3a18ad7f8d6d255f546a5cd4b3b7ce6d62c35a4b..5532d14b6aaf194a890caad02eab3266865c8d47 100644
--- a/third_party/lottie/BUILD.gn
+++ b/third_party/lottie/BUILD.gn
@@ -6,7 +6,7 @@ import("//chrome/browser/resources/tools/optimize_webui.gni")
 import("//third_party/closure_compiler/compile_js.gni")
 import("//ui/webui/resources/tools/generate_grd.gni")
 
-assert(!is_android && !is_ios)
+#assert(!is_android && !is_ios)
 
 js_library("lottie_worker") {
 }
