diff --git a/third_party/blink/renderer/modules/BUILD.gn b/third_party/blink/renderer/modules/BUILD.gn
index c5bd977747bcdb03320dabe554c62010a654afe1..b371698b34a1bee2fb5a6c9c06db2b70e4759c2b 100644
--- a/third_party/blink/renderer/modules/BUILD.gn
+++ b/third_party/blink/renderer/modules/BUILD.gn
@@ -207,6 +207,7 @@ component("modules") {
     "//third_party/zlib",
     "//tools/v8_context_snapshot:buildflags",
   ]
+  deps += mises_blink_renderer_modules_deps
 
   public_deps = sub_modules
   allow_circular_includes_from = sub_modules
