diff --git a/third_party/android_deps/build.gradle b/third_party/android_deps/build.gradle
index 3ae7ec68e0de79d7298a731775771787a2c697d5..d9dae19c9c5bfd625f878d4314bd8a77bd423d0a 100644
--- a/third_party/android_deps/build.gradle
+++ b/third_party/android_deps/build.gradle
@@ -193,5 +193,5 @@ task setUpRepository(type: BuildConfigGenerator) {
     repositoryPath 'third_party/android_deps'
     chromiumSourceRoot '../..'
     cipdBucket 'chromium'
-    internalTargetVisibility = [ ':*', '//third_party/androidx:*' ]
+    internalTargetVisibility = [ ':*', '//third_party/androidx:*', '//mises/third_party/android_deps*' ]
 }
