diff --git a/third_party/android_deps/BUILD.gn b/third_party/android_deps/BUILD.gn
index 5b3cda05c1e6db27aed350e88f403587bcf9c5a8..b8284d7c182dbf55e3e59f38dd01b0bc0ae54dda 100644
--- a/third_party/android_deps/BUILD.gn
+++ b/third_party/android_deps/BUILD.gn
@@ -1924,6 +1924,7 @@ if (!limit_android_deps) {
     visibility = [
       ":*",
       "//third_party/androidx:*",
+      "//mises/third_party/android_deps:*",
     ]
     deps = [
       ":google_play_services_basement_java",
@@ -1956,6 +1957,7 @@ if (!limit_android_deps) {
     visibility = [
       ":*",
       "//third_party/androidx:*",
+      "//mises/third_party/android_deps:*",
     ]
     deps = [
       ":google_firebase_firebase_components_java",
@@ -1978,6 +1980,7 @@ if (!limit_android_deps) {
     visibility = [
       ":*",
       "//third_party/androidx:*",
+      "//mises/third_party/android_deps:*",
     ]
     deps = [
       ":google_firebase_firebase_annotations_java",
@@ -2047,6 +2050,7 @@ if (!limit_android_deps) {
     visibility = [
       ":*",
       "//third_party/androidx:*",
+      "//mises/third_party/android_deps:*",
     ]
     deps = [
       ":google_firebase_firebase_common_java",
@@ -2069,6 +2073,7 @@ if (!limit_android_deps) {
     visibility = [
       ":*",
       "//third_party/androidx:*",
+      "//mises/third_party/android_deps:*",
     ]
     deps = [
       ":google_firebase_firebase_annotations_java",
@@ -2089,6 +2094,7 @@ if (!limit_android_deps) {
     visibility = [
       ":*",
       "//third_party/androidx:*",
+      "//mises/third_party/android_deps:*",
     ]
     deps = [ ":google_play_services_basement_java" ]
   }
