diff --git a/third_party/android_deps/BUILD.gn b/third_party/android_deps/BUILD.gn
index 051c3b106cc9a0eff967c4d9d3c74ab1b5e9fbd5..0f8335d133a26b29751079819e7e3a7e6bc5d3a7 100644
--- a/third_party/android_deps/BUILD.gn
+++ b/third_party/android_deps/BUILD.gn
@@ -1895,6 +1895,7 @@ if (!limit_android_deps) {
     visibility = [
       ":*",
       "//third_party/androidx:*",
+      "//mises/third_party/android_deps:*",
     ]
     deps = [
       ":com_android_support_customview_java",
@@ -2302,6 +2303,7 @@ if (!limit_android_deps) {
     # //third_party/android_deps/build.gradle.
     visibility = [
       ":*",
+      "//mises/third_party/android_deps:*",
       "//third_party/androidx:*",
     ]
     deps = [
@@ -2345,6 +2347,7 @@ if (!limit_android_deps) {
     visibility = [
       ":*",
       "//third_party/androidx:*",
+      "//mises/third_party/android_deps:*",
     ]
     testonly = true
   }
@@ -2361,6 +2364,7 @@ if (!limit_android_deps) {
     visibility = [
       ":*",
       "//third_party/androidx:*",
+      "//mises/third_party/android_deps:*",
     ]
     testonly = true
   }
@@ -2438,6 +2442,7 @@ if (!limit_android_deps) {
     visibility = [
       ":*",
       "//third_party/androidx:*",
+      "//mises/third_party/android_deps:*",
     ]
     testonly = true
     is_robolectric = true
@@ -2454,6 +2459,7 @@ if (!limit_android_deps) {
     visibility = [
       ":*",
       "//third_party/androidx:*",
+      "//mises/third_party/android_deps:*",
     ]
     testonly = true
     deps = [
@@ -2477,6 +2483,7 @@ if (!limit_android_deps) {
     visibility = [
       ":*",
       "//third_party/androidx:*",
+      "//mises/third_party/android_deps:*",
     ]
     testonly = true
     deps = [
