diff --git a/third_party/android_deps/BUILD.gn b/third_party/android_deps/BUILD.gn
index d7f1d1c35a346985e4ded85c4e785cf3914bf9a2..fb003767de1aa9876f08360e10db854efb5a14af 100644
--- a/third_party/android_deps/BUILD.gn
+++ b/third_party/android_deps/BUILD.gn
@@ -1364,6 +1364,7 @@ if (!limit_android_deps) {
       visibility = [
         ":*",
         "//third_party/androidx:*",
+        "//mises/third_party/android_deps:*",
       ]
       deps = [ "//third_party/androidx:androidx_annotation_annotation_java" ]
 
