diff --git a/third_party/blink/renderer/core/css/style_engine.h b/third_party/blink/renderer/core/css/style_engine.h
index 9a71f400ae8330f155d63eb102e2b2cf3d557847..1d26fc03dc6e0b7129c540cd344e4dc0a377e2d3 100644
--- a/third_party/blink/renderer/core/css/style_engine.h
+++ b/third_party/blink/renderer/core/css/style_engine.h
@@ -66,6 +66,7 @@
 #include "third_party/blink/renderer/platform/wtf/allocator/allocator.h"
 #include "third_party/blink/renderer/platform/wtf/text/wtf_string.h"
 #include "third_party/blink/renderer/platform/wtf/vector.h"
+#include "third_party/blink/renderer/platform/theme/web_theme_engine_helper.h"
 
 namespace WTF {
 class TextPosition;
@@ -649,7 +650,7 @@ class CORE_EXPORT StyleEngine final : public GarbageCollected<StyleEngine>,
   mojom::PreferredColorScheme GetPreferredColorScheme() const {
     return preferred_color_scheme_;
   }
-  bool GetForceDarkModeEnabled() const { return force_dark_mode_enabled_; }
+  bool GetForceDarkModeEnabled() const { return force_dark_mode_enabled_ || WebThemeEngineHelper::GlobalForceDarkModeEnabled(); }
   ForcedColors GetForcedColors() const { return forced_colors_; }
   void UpdateColorSchemeBackground(bool color_scheme_changed = false);
   Color ForcedBackgroundColor() const { return forced_background_color_; }
