diff --git a/third_party/blink/renderer/core/BUILD.gn b/third_party/blink/renderer/core/BUILD.gn
index 9cb087c578629d872d2a2c12a046471527372c81..7bad42bd53912b6f90f098dc3da7a38e59ddf0e1 100644
--- a/third_party/blink/renderer/core/BUILD.gn
+++ b/third_party/blink/renderer/core/BUILD.gn
@@ -430,6 +430,7 @@ component("core") {
       "Carbon.framework",
     ]
   }
+  visibility += mises_blink_renderer_core_visibility public_deps += mises_blink_renderer_core_public_deps sources += mises_blink_renderer_core_sources deps += mises_blink_renderer_core_deps
 }
 
 # Certain sources should be compiled with heavy optimization enabled.
