diff --git a/third_party/blink/common/web_preferences/web_preferences_mojom_traits.cc b/third_party/blink/common/web_preferences/web_preferences_mojom_traits.cc
index 22220de13b6d7b88cfe0b02367836f7bd5e139b5..9a3448030a3b09f02f5c50f4c0e15ec374ec6472 100644
--- a/third_party/blink/common/web_preferences/web_preferences_mojom_traits.cc
+++ b/third_party/blink/common/web_preferences/web_preferences_mojom_traits.cc
@@ -201,6 +201,7 @@ bool StructTraits<blink::mojom::WebPreferencesDataView,
 #endif  // BUILDFLAG(IS_ANDROID) || BUILDFLAG(IS_FUCHSIA)
 
   out->force_dark_mode_enabled = data.force_dark_mode_enabled();
+  out->force_night_mode = data.force_night_mode();
   out->default_minimum_page_scale_factor =
       data.default_minimum_page_scale_factor();
   out->default_maximum_page_scale_factor =
