diff --git a/third_party/blink/public/mojom/BUILD.gn b/third_party/blink/public/mojom/BUILD.gn
index 46ca69630fed35a4a4bd7edb9bf3fb39ae577a16..8d6c8cc5ecc333663e800e5f6fe2f9c536470263 100644
--- a/third_party/blink/public/mojom/BUILD.gn
+++ b/third_party/blink/public/mojom/BUILD.gn
@@ -334,8 +334,10 @@ mojom("mojom_platform") {
       "//services/network/public/mojom:mojom_schemeful_site",
       "//ui/accessibility:ax_enums_mojo",
       "//ui/latency/mojom",
+      "//services/network/public/mojom:mojom_host_resolver",
     ]
-  } else {
+  }
+  if (true) {
     sources += [
       "direct_sockets/direct_sockets.mojom",
       "serial/serial.mojom",
