diff --git a/third_party/blink/renderer/platform/runtime_enabled_features.json5 b/third_party/blink/renderer/platform/runtime_enabled_features.json5
index 5f91aa499ae04a5a21135b7345747cdad694bd21..615c245a2ae4a2f7e07e86db72d4894cb9302733 100644
--- a/third_party/blink/renderer/platform/runtime_enabled_features.json5
+++ b/third_party/blink/renderer/platform/runtime_enabled_features.json5
@@ -4336,7 +4336,7 @@
     },
     {
       name: "WebHID",
-      status: {"Android": "", "default": "stable"},
+      status: "stable",
       base_feature: "none",
     },
     {
