diff --git a/third_party/android_deps/build.gradle b/third_party/android_deps/build.gradle
index 2e5770f3e927f360530f5c57e243edfcecd57261..e6f6ce6dd393461b0cb776f6edd1f120c7ac6a77 100644
--- a/third_party/android_deps/build.gradle
+++ b/third_party/android_deps/build.gradle
@@ -205,5 +205,5 @@ task setUpRepository(type: BuildConfigGenerator) {
     repositoryPath 'third_party/android_deps'
     chromiumSourceRoot '../..'
     cipdBucket 'chromium'
-    internalTargetVisibility = [ ':*', '//third_party/androidx:*' ]
+    internalTargetVisibility = [ ':*', '//third_party/androidx:*', '//mises/third_party/android_deps*' ]
 }
