diff --git a/third_party/blink/renderer/core/exported/web_view_impl.cc b/third_party/blink/renderer/core/exported/web_view_impl.cc
index c8b7feaa65bfb47daa63565b193ff33806f53f88..eae2019b13ff0e87217f4886603d522600c1d074 100644
--- a/third_party/blink/renderer/core/exported/web_view_impl.cc
+++ b/third_party/blink/renderer/core/exported/web_view_impl.cc
@@ -1741,6 +1741,10 @@ void WebView::ApplyWebPreferences(const web_pref::WebPreferences& prefs,
 #endif  // BUILDFLAG(IS_ANDROID) || BUILDFLAG(IS_FUCHSIA)
 
   settings->SetForceDarkModeEnabled(prefs.force_dark_mode_enabled);
+  if (prefs.force_night_mode >= 0) {
+    WebThemeEngineHelper::SetGlobalForceDarkModeEnabled(prefs.force_night_mode == 1);
+  }
+  
 
   settings->SetAccessibilityAlwaysShowFocus(prefs.always_show_focus);
   settings->SetAutoplayPolicy(prefs.autoplay_policy);
