diff --git a/third_party/blink/renderer/platform/theme/web_theme_engine_helper.cc b/third_party/blink/renderer/platform/theme/web_theme_engine_helper.cc
index ee409f98c3f66018bb39cfc3a31dc0b92053ff15..04ddeae250cf892346d9fbb001871748f3cd9967 100644
--- a/third_party/blink/renderer/platform/theme/web_theme_engine_helper.cc
+++ b/third_party/blink/renderer/platform/theme/web_theme_engine_helper.cc
@@ -59,4 +59,12 @@ void WebThemeEngineHelper::DidUpdateRendererPreferences(
 #endif
 }
 
+bool global_force_dark_mode_enabled = false;
+bool WebThemeEngineHelper::GlobalForceDarkModeEnabled() {
+  return global_force_dark_mode_enabled;
+}
+void WebThemeEngineHelper::SetGlobalForceDarkModeEnabled(bool enable) {
+  global_force_dark_mode_enabled = enable;
+}
+
 }  // namespace blink
