diff --git a/third_party/blink/renderer/core/exported/web_view_impl.cc b/third_party/blink/renderer/core/exported/web_view_impl.cc
index 8f3603538170a4e66a3f832b1882ed587383cfcc..cbed367e1562895fde1cdaf6bd4a7cba084a0d1f 100644
--- a/third_party/blink/renderer/core/exported/web_view_impl.cc
+++ b/third_party/blink/renderer/core/exported/web_view_impl.cc
@@ -1794,6 +1794,10 @@ void WebView::ApplyWebPreferences(const web_pref::WebPreferences& prefs,
 #endif  // BUILDFLAG(IS_ANDROID) || BUILDFLAG(IS_FUCHSIA)
 
   settings->SetForceDarkModeEnabled(prefs.force_dark_mode_enabled);
+  if (prefs.force_night_mode >= 0) {
+    WebThemeEngineHelper::SetGlobalForceDarkModeEnabled(prefs.force_night_mode == 1);
+  }
+  
 
   settings->SetAccessibilityAlwaysShowFocus(prefs.always_show_focus);
   settings->SetAutoplayPolicy(prefs.autoplay_policy);
