diff --git a/third_party/android_deps/build.gradle b/third_party/android_deps/build.gradle
index 0bdea39e155f0f98b2d3b45ad6f9c54e1bc684d6..b6304ebc690fe05c714d722a84dd4cc824c9e9c3 100644
--- a/third_party/android_deps/build.gradle
+++ b/third_party/android_deps/build.gradle
@@ -197,5 +197,5 @@ task setUpRepository(type: BuildConfigGenerator) {
     repositoryPath 'third_party/android_deps'
     chromiumSourceRoot '../..'
     cipdBucket 'chromium'
-    internalTargetVisibility = [ ':*', '//third_party/androidx:*' ]
+    internalTargetVisibility = [ ':*', '//third_party/androidx:*', '//mises/third_party/android_deps*' ]
 }
