diff --git a/third_party/blink/public/common/web_preferences/web_preferences.h b/third_party/blink/public/common/web_preferences/web_preferences.h
index 27cc511b84dfe970fd9eff520b81446ce3ba497a..fcb6440c12ec2c9becafef2c6fdfa8e0e4fdb6e3 100644
--- a/third_party/blink/public/common/web_preferences/web_preferences.h
+++ b/third_party/blink/public/common/web_preferences/web_preferences.h
@@ -301,6 +301,7 @@ struct BLINK_COMMON_EXPORT WebPreferences {
   // Enable forcibly modifying content rendering to result in a light on dark
   // color scheme.
   bool force_dark_mode_enabled = false;
+  int force_night_mode = -1;
 
   // Default (used if the page or UA doesn't override these) values for page
   // scale limits. These are set directly on the WebView so there's no analogue
