diff --git a/third_party/android_deps/build.gradle b/third_party/android_deps/build.gradle
index 7abcf103782916d0dc8be03baec1ae913aaa1816..eef15e3e5556cf35407a40d88f52c24681bce525 100644
--- a/third_party/android_deps/build.gradle
+++ b/third_party/android_deps/build.gradle
@@ -205,5 +205,5 @@ task setUpRepository(type: BuildConfigGenerator) {
     repositoryPath 'third_party/android_deps'
     chromiumSourceRoot '../..'
     cipdBucket 'chromium'
-    internalTargetVisibility = [ ':*', '//third_party/androidx:*' ]
+    internalTargetVisibility = [ ':*', '//third_party/androidx:*', '//mises/third_party/android_deps*' ]
 }
