diff --git a/third_party/blink/renderer/platform/theme/web_theme_engine_helper.cc b/third_party/blink/renderer/platform/theme/web_theme_engine_helper.cc
index 43fb9329f95dfafb494cb3210c12cd46fa45bc9b..b5bc0bd96c1153aa5f9e1ef133658f1f0353e923 100644
--- a/third_party/blink/renderer/platform/theme/web_theme_engine_helper.cc
+++ b/third_party/blink/renderer/platform/theme/web_theme_engine_helper.cc
@@ -68,4 +68,12 @@ WebThemeEngineHelper::AndroidScrollbarStyle() {
   return style;
 }
 
+bool global_force_dark_mode_enabled = false;
+bool WebThemeEngineHelper::GlobalForceDarkModeEnabled() {
+  return global_force_dark_mode_enabled;
+}
+void WebThemeEngineHelper::SetGlobalForceDarkModeEnabled(bool enable) {
+  global_force_dark_mode_enabled = enable;
+}
+
 }  // namespace blink
