diff --git a/third_party/blink/common/web_preferences/web_preferences_mojom_traits.cc b/third_party/blink/common/web_preferences/web_preferences_mojom_traits.cc
index 5b6208ca0f024410ca9b86444675abdd0a342bae..d9e525cd0b125ffcf20107a3869c3dbeb2c1dc89 100644
--- a/third_party/blink/common/web_preferences/web_preferences_mojom_traits.cc
+++ b/third_party/blink/common/web_preferences/web_preferences_mojom_traits.cc
@@ -201,6 +201,7 @@ bool StructTraits<blink::mojom::WebPreferencesDataView,
 #endif  // BUILDFLAG(IS_ANDROID) || BUILDFLAG(IS_FUCHSIA)
 
   out->force_dark_mode_enabled = data.force_dark_mode_enabled();
+  out->force_night_mode = data.force_night_mode();
   out->default_minimum_page_scale_factor =
       data.default_minimum_page_scale_factor();
   out->default_maximum_page_scale_factor =
