diff --git a/third_party/blink/renderer/platform/runtime_enabled_features.json5 b/third_party/blink/renderer/platform/runtime_enabled_features.json5
index 46369bcb300556a63ea7394943af693d84ff8063..6b22f7a2822e3f5840ba45b9f579f05fe83bceb8 100644
--- a/third_party/blink/renderer/platform/runtime_enabled_features.json5
+++ b/third_party/blink/renderer/platform/runtime_enabled_features.json5
@@ -1990,6 +1990,15 @@
       name: "FindTextInReadonlyTextInput",
       status: "experimental",
     },
+    {
+      name: "FingerprintingClientRectsNoise",
+    },
+    {
+      name: "FingerprintingCanvasMeasureTextNoise",
+    },
+    {
+      name: "FingerprintingCanvasImageDataNoise",
+    },
     {
       name: "Fledge",
       status: "stable",
@@ -4731,7 +4740,7 @@
     },
     {
       name: "WebHID",
-      status: {"Android": "", "default": "stable"},
+      status: "stable",
     },
     {
       // It is only enabled in extension environment for now.
