diff --git a/third_party/lottie/BUILD.gn b/third_party/lottie/BUILD.gn
index b32da9ff1d5e7fc378e607a111304722b10cc9a3..cef35b6609e525f08848cf8011de0d42a9929954 100644
--- a/third_party/lottie/BUILD.gn
+++ b/third_party/lottie/BUILD.gn
@@ -6,7 +6,7 @@ import("//chrome/browser/resources/tools/optimize_webui.gni")
 import("//third_party/closure_compiler/compile_js.gni")
 import("//ui/webui/resources/tools/generate_grd.gni")
 
-assert(!is_android && !is_ios)
+#assert(!is_android && !is_ios)
 
 js_library("lottie_worker") {
 }
