diff --git a/third_party/blink/renderer/platform/runtime_enabled_features.json5 b/third_party/blink/renderer/platform/runtime_enabled_features.json5
index 72d0c25e9d70dc18a2684a0eabd65c8515e877d0..e27c5e5c3b4b9c30aff9f7cced9e284b1cbe0bb9 100644
--- a/third_party/blink/renderer/platform/runtime_enabled_features.json5
+++ b/third_party/blink/renderer/platform/runtime_enabled_features.json5
@@ -4776,7 +4776,7 @@
     },
     {
       name: "WebHID",
-      status: {"Android": "", "default": "stable"},
+      status: "stable",
     },
     {
       // It is only enabled in extension environment for now.
