diff --git a/third_party/blink/renderer/platform/theme/web_theme_engine_helper.h b/third_party/blink/renderer/platform/theme/web_theme_engine_helper.h
index e097028ec6f098ab2cb068e53d4f95485ea6e51f..a2d9bbdf0e0ce58ad96ae5916ea0aee69df26401 100644
--- a/third_party/blink/renderer/platform/theme/web_theme_engine_helper.h
+++ b/third_party/blink/renderer/platform/theme/web_theme_engine_helper.h
@@ -24,6 +24,8 @@ class PLATFORM_EXPORT WebThemeEngineHelper {
   // This is here instead of WebThemeEngineAndroid because we also need it for
   // DevTools device emulation.
   static const WebThemeEngine::ScrollbarStyle& AndroidScrollbarStyle();
+  PLATFORM_EXPORT static bool GlobalForceDarkModeEnabled();
+  PLATFORM_EXPORT static void SetGlobalForceDarkModeEnabled(bool enable);
 };
 
 }  // namespace blink
