diff --git a/tools/gritsettings/resource_ids.spec b/tools/gritsettings/resource_ids.spec
index d8a54f29cdc9bcf16ba02d48d96c7a108d2ccd19..ff94b515a6ee4f93021f0828c022a948817fab17 100644
--- a/tools/gritsettings/resource_ids.spec
+++ b/tools/gritsettings/resource_ids.spec
@@ -44,6 +44,9 @@
   "chrome/app/google_chrome_strings.grd": {
     "messages": [400],
   },
+  "chrome/app/mises_strings.grd": {
+    "messages": [400],
+  },
 
   # Leave lots of space for generated_resources since it has most of our
   # strings.
