diff --git a/tools/json_schema_compiler/feature_compiler.py b/tools/json_schema_compiler/feature_compiler.py
index 8e53077083d2881359b322f880a47688869f76b2..0a3485a9b91f2c9bcc933378ec6ea0e83958c372 100644
--- a/tools/json_schema_compiler/feature_compiler.py
+++ b/tools/json_schema_compiler/feature_compiler.py
@@ -762,6 +762,7 @@ class FeatureCompiler(object):
                   abs_source_file)
         raise
       dupes = set(f_json) & set(self._json)
+      from tools.json_schema_compiler import feature_compiler_helper; feature_compiler_helper.DiscardMisesOverridesFromDupes(dupes)
       assert not dupes, 'Duplicate keys found: %s' % list(dupes)
       self._json.update(f_json)
 
