diff --git a/tools/gritsettings/resource_ids.spec b/tools/gritsettings/resource_ids.spec
index 307e222ab9ce9c4657084da67825dcbb58200d9b..69458d945856a6cdd6903c516b169b6a94391e4a 100644
--- a/tools/gritsettings/resource_ids.spec
+++ b/tools/gritsettings/resource_ids.spec
@@ -38,6 +38,9 @@
   "chrome/app/chromium_strings.grd": {
     "messages": [800],
   },
+  "chrome/app/mises_strings.grd": {
+    "messages": [800],
+  },
   "chrome/app/google_chrome_strings.grd": {
     "messages": [800],
   },
@@ -1007,8 +1010,11 @@
   "components/components_chromium_strings.grd": {
     "messages": [7020],
   },
+  "components/components_mises_strings.grd": {
+    "messages": [6020],
+  },
   "components/components_google_chrome_strings.grd": {
-    "messages": [7020],
+    "messages": [6040],
   },
   "components/components_locale_settings.grd": {
     "META": {"join": 2},
