diff --git a/tools/typescript/ts_library.py b/tools/typescript/ts_library.py
index 4dbbb4455516ac35b0df5204bc182fe798468119..c5623e4b67170608177b7d968c1cc7e97974ff71 100644
--- a/tools/typescript/ts_library.py
+++ b/tools/typescript/ts_library.py
@@ -276,6 +276,6 @@ def main(argv):
           [re.sub(r'\.ts$', '.js', f) for f in manifest_files]
       json.dump(manifest_data, manifest_file)
 
-
+from import_inline import inline_file_from_src; inline_file_from_src("mises/chromium_src/tools/typescript/ts_library.py", globals(), locals())
 if __name__ == '__main__':
   main(sys.argv[1:])
