diff --git a/tools/json_schema_compiler/feature_compiler.py b/tools/json_schema_compiler/feature_compiler.py
index b59a98411b874896492e78d627f6e04f518eddb6..9192963e51cd2a9ef2e4a3efc375c8d9386a6bd3 100644
--- a/tools/json_schema_compiler/feature_compiler.py
+++ b/tools/json_schema_compiler/feature_compiler.py
@@ -973,6 +973,7 @@ class FeatureCompiler(object):
       cc_file.Concat(cc_end)
       f.write(cc_file.Render().strip())
 
+from import_inline import inline_file_from_src; inline_file_from_src("mises/chromium_src/tools/json_schema_compiler/feature_compiler.py", globals(), locals())
 if __name__ == '__main__':
   parser = argparse.ArgumentParser(description='Compile json feature files')
   parser.add_argument('chrome_root', type=str,
