diff --git a/tools/typescript/ts_library.gni b/tools/typescript/ts_library.gni
index b7d218a03f009a36efa2c39fabe5081b50037094..85eba4916402dc7ce141addf7683d21ec224f28f 100644
--- a/tools/typescript/ts_library.gni
+++ b/tools/typescript/ts_library.gni
@@ -150,6 +150,7 @@ template("ts_library") {
       # other ts_library() deps are automatically inferred from |deps| in
       # path_mappings.py. Don't add any such mappings here.
     ]
+    import("//mises/tools/typescript/sources.gni") foreach(mapping, mises_path_mappings) { path_mappings += [ mapping[0] + rebase_path(mapping[1], target_gen_dir) ] }
 
     # The |platform| flag is used in path_mappings.py and validate_tsconfig.py,
     # to limit some logic to certain platforms, like preventing special iOS
