diff --git a/tools/gritsettings/resource_ids.spec b/tools/gritsettings/resource_ids.spec
index 15e77b25cc0b10e59d154d2389d4b8dbaca6391a..f1594610af6d0746a8c688b5095c08534c63c015 100644
--- a/tools/gritsettings/resource_ids.spec
+++ b/tools/gritsettings/resource_ids.spec
@@ -44,6 +44,9 @@
   "chrome/app/google_chrome_strings.grd": {
     "messages": [400],
   },
+  "chrome/app/mises_strings.grd": {
+    "messages": [400],
+  },
 
   # Leave lots of space for generated_resources since it has most of our
   # strings.
@@ -106,434 +109,434 @@
   },
   "<(SHARED_INTERMEDIATE_DIR)/chrome/browser/resources/feed_internals/resources.grd": {
     "META": {"sizes": {"includes": [10],}},
-    "includes": [1270],
+    "includes": [1280],
   },
   "<(SHARED_INTERMEDIATE_DIR)/chrome/browser/resources/app_service_internals/resources.grd": {
     "META": {"sizes": {"includes": [5],}},
-    "includes": [1280],
+    "includes": [1300],
   },
   "<(SHARED_INTERMEDIATE_DIR)/chrome/browser/resources/bookmarks/resources.grd": {
     "META": {"sizes": {"includes": [50],}},
-    "includes": [1300],
+    "includes": [1320],
   },
   "<(SHARED_INTERMEDIATE_DIR)/chrome/browser/resources/browser_switch/resources.grd": {
     "META": {"sizes": {"includes": [10],}},
-    "includes": [1320],
+    "includes": [1340],
   },
   "<(SHARED_INTERMEDIATE_DIR)/chrome/browser/resources/chromeos/arc_account_picker/resources.grd": {
     "META": {"sizes": {"includes": [10],}},
-    "includes": [1330],
+    "includes": [1360],
   },
   "<(SHARED_INTERMEDIATE_DIR)/chrome/browser/resources/chromeos/assistant_optin/assistant_optin_resources.grd": {
     "META": {"sizes": {"includes": [80]}},
-    "includes": [1340],
+    "includes": [1380],
   },
   "<(SHARED_INTERMEDIATE_DIR)/chrome/browser/resources/chromeos/cloud_upload/resources.grd": {
     "META": {"sizes": {"includes": [40]}},
-    "includes": [1345],
+    "includes": [1400],
   },
   "<(SHARED_INTERMEDIATE_DIR)/chrome/browser/resources/chromeos/desk_api/resources.grd": {
     "META": {"sizes": {"includes": [10],}},
-    "includes": [1347],
+    "includes": [1420],
   },
   "<(SHARED_INTERMEDIATE_DIR)/chrome/browser/resources/chromeos/gaia_action_buttons/resources.grd": {
     "META": {"sizes": {"includes": [10],}},
-    "includes": [1350],
+    "includes": [1440],
   },
   "<(SHARED_INTERMEDIATE_DIR)/chrome/browser/resources/chromeos/emoji_picker/resources.grd": {
     "META": {"sizes": {"includes": [50]}},
-    "includes": [1360],
+    "includes": [1460],
   },
   "<(SHARED_INTERMEDIATE_DIR)/chrome/browser/resources/chromeos/launcher_internals/resources.grd": {
     "META": {"sizes": {"includes": [50]}},
-    "includes": [1380],
+    "includes": [1480],
   },
   "<(SHARED_INTERMEDIATE_DIR)/chrome/browser/resources/chromeos/supervision/resources.grd": {
     "META": {"sizes": {"includes": [10],}},
-    "includes": [1383],
+    "includes": [1500],
   },
   "<(SHARED_INTERMEDIATE_DIR)/chrome/browser/resources/profile_internals/resources.grd": {
     "META": {"sizes": {"includes": [10],}},
-    "includes": [1385],
+    "includes": [1520],
   },
   "chrome/browser/resources/chromeos/app_icon/app_icon_resources.grd": {
-    "structures": [1390],
+    "structures": [1540],
   },
   "chrome/browser/resources/chromeos/login/oobe_conditional_resources.grd": {
     "META": {"sizes": {"includes": [150], "structures": [300]}},
-    "includes": [1400],
-    "structures": [1420],
+    "includes": [1560],
+    "structures": [1580],
   },
   "<(SHARED_INTERMEDIATE_DIR)/chrome/browser/resources/chromeos/lock_screen_reauth/resources.grd": {
     "META": {"sizes": {"includes": [30]}},
-    "includes": [1430],
+    "includes": [1600],
   },
   "<(SHARED_INTERMEDIATE_DIR)/chrome/browser/resources/chromeos/login/oobe_unconditional_resources.grd": {
     "META": {"sizes": {"includes": [350]}},
-    "includes": [1440],
+    "includes": [1620],
   },
   "<(SHARED_INTERMEDIATE_DIR)/chrome/browser/resources/chromeos/multidevice_internals/resources.grd": {
     "META": {"sizes": {"includes": [35]}},
-    "includes": [1460],
+    "includes": [1640],
   },
   "<(SHARED_INTERMEDIATE_DIR)/chrome/browser/resources/chromeos/multidevice_setup/multidevice_setup_resources.grd": {
     "META": {"sizes": {"includes": [10]}},
-    "includes": [1480],
+    "includes": [1660],
   },
   "<(SHARED_INTERMEDIATE_DIR)/chrome/browser/resources/chromeos/notification_tester/resources.grd": {
     "META": {"sizes": {"includes": [5]}},
-    "includes": [1490],
+    "includes": [1680],
   },
   "<(SHARED_INTERMEDIATE_DIR)/chrome/browser/resources/chromeos/password_change/resources.grd": {
     "META": {"sizes": {"includes": [30]}},
-    "includes": [1495],
+    "includes": [1700],
   },
   "<(SHARED_INTERMEDIATE_DIR)/chrome/browser/resources/chromeos/vc_tray_tester/resources.grd": {
     "META": {"sizes": {"includes": [5]}},
-    "includes": [1497],
+    "includes": [1720],
   },
   "<(SHARED_INTERMEDIATE_DIR)/chrome/browser/resources/commander/resources.grd": {
     "META": {"sizes": {"includes": [15]}},
-    "includes": [1500],
+    "includes": [1740],
   },
   "chrome/browser/resources/component_extension_resources.grd": {
-    "includes": [1520],
-    "structures": [1540],
+    "includes": [1760],
+    "structures": [1780],
   },
   "<(SHARED_INTERMEDIATE_DIR)/chrome/browser/resources/access_code_cast/resources.grd": {
     "META": {"sizes": {"includes": [50]}},
-    "includes": [1560],
+    "includes": [1800],
   },
   "<(SHARED_INTERMEDIATE_DIR)/chrome/browser/resources/connectors_internals/resources.grd": {
     "META": {"sizes": {"includes": [15]}},
-    "includes": [1580],
+    "includes": [1820],
   },
   "<(SHARED_INTERMEDIATE_DIR)/chrome/browser/resources/discards/discards_resources.grd": {
     "META": {"sizes": {"includes": [20],}},
-    "includes": [1585],
+    "includes": [1840],
   },
   "<(SHARED_INTERMEDIATE_DIR)/chrome/browser/resources/downloads/resources.grd": {
     "META": {"sizes": {"includes": [50],}},
-    "includes": [1590],
+    "includes": [1860],
   },
   "<(SHARED_INTERMEDIATE_DIR)/chrome/browser/resources/extensions/resources.grd": {
     "META": {"sizes": {"includes": [90],}},
-    "includes": [1600],
+    "includes": [1880],
   },
   "<(SHARED_INTERMEDIATE_DIR)/chrome/browser/resources/history/resources.grd": {
     "META": {"sizes": {"includes": [40]}},
-    "includes": [1620],
+    "includes": [1900],
   },
   "<(SHARED_INTERMEDIATE_DIR)/chrome/browser/resources/identity_internals/resources.grd": {
     "META": {"sizes": {"includes": [10]}},
-    "includes": [1621],
+    "includes": [1920],
   },
   "<(SHARED_INTERMEDIATE_DIR)/chrome/browser/resources/image_editor/resources.grd": {
     "META": {"sizes": {"includes": [20]}},
-    "includes": [1625],
+    "includes": [1940],
   },
   "<(SHARED_INTERMEDIATE_DIR)/chrome/browser/resources/image_editor/image_editor_untrusted_resources.grd": {
     "META": {"sizes": {"includes": [20]}},
-    "includes": [1630],
+    "includes": [1960],
   },
   "<(SHARED_INTERMEDIATE_DIR)/chrome/browser/resources/internals/resources.grd": {
     "META": {"sizes": {"includes": [20]}},
-    "includes": [1640],
+    "includes": [1980],
   },
   "<(SHARED_INTERMEDIATE_DIR)/chrome/browser/resources/intro/resources.grd": {
     "META": {"sizes": {"includes": [20],}},
-    "includes": [1650],
+    "includes": [2000],
   },
   "<(SHARED_INTERMEDIATE_DIR)/chrome/browser/resources/management/resources.grd": {
     "META": {"sizes": {"includes": [10]}},
-    "includes": [1660],
+    "includes": [2020],
   },
   "<(SHARED_INTERMEDIATE_DIR)/chrome/browser/resources/new_tab_page_instant/resources.grd": {
-    "META": {"sizes": {"includes": [10]}},
-    "includes": [1680],
+    "META": {"sizes": {"includes": [30]}},
+    "includes": [2040],
   },
   "<(SHARED_INTERMEDIATE_DIR)/chrome/browser/resources/nearby_internals/nearby_internals_resources.grd": {
     "META": {"sizes": {"includes": [20]}},
-    "includes": [1740],
+    "includes": [2060],
   },
   "<(SHARED_INTERMEDIATE_DIR)/chrome/browser/resources/nearby_share/nearby_share_dialog_resources.grd": {
     "META": {"sizes": {"includes": [100]}},
-    "includes": [1760],
+    "includes": [2080],
   },
   "<(SHARED_INTERMEDIATE_DIR)/chrome/browser/resources/media_router/cast_feedback/resources.grd": {
     "META": {"sizes": {"includes": [10],}},
-    "includes": [1780],
+    "includes": [2100],
   },
   "<(SHARED_INTERMEDIATE_DIR)/chrome/browser/resources/new_tab_page/resources.grd": {
     "META": {"sizes": {"includes": [200]}},
-    "includes": [1800],
+    "includes": [2120],
   },
   "<(SHARED_INTERMEDIATE_DIR)/chrome/browser/resources/new_tab_page_third_party/resources.grd": {
     "META": {"sizes": {"includes": [10]}},
-    "includes": [1820],
+    "includes": [2140],
   },
   "<(SHARED_INTERMEDIATE_DIR)/chrome/browser/resources/ntp4/apps_resources.grd": {
     "META": {"sizes": {"includes": [40]}},
-    "includes": [1840],
+    "includes": [2160],
   },
   "chrome/browser/resources/preinstalled_web_apps/resources.grd": {
-    "includes": [1860],
+    "includes": [2180],
   },
   "<(SHARED_INTERMEDIATE_DIR)/chrome/browser/resources/pdf/resources.grd": {
     "META": {"sizes": {"includes": [200]}},
-    "includes": [1880],
+    "includes": [2200],
   },
   "<(SHARED_INTERMEDIATE_DIR)/chrome/browser/resources/print_preview/resources.grd": {
     "META": {"sizes": {"includes": [500],}},
-    "includes": [1900],
+    "includes": [2220],
   },
   "<(SHARED_INTERMEDIATE_DIR)/chrome/browser/resources/privacy_sandbox/resources.grd": {
     "META": {"sizes": {"includes": [30],}},
-    "includes": [1910],
+    "includes": [2240],
   },
   "<(SHARED_INTERMEDIATE_DIR)/chrome/browser/resources/segmentation_internals/resources.grd": {
     "META": {"sizes": {"includes": [10]}},
-    "includes": [1925],
+    "includes": [2260],
   },
   "<(SHARED_INTERMEDIATE_DIR)/chrome/browser/resources/side_panel/bookmarks/resources.grd": {
     "META": {"sizes": {"includes": [45],}},
-    "includes": [1926],
+    "includes": [2280],
   },
   "<(SHARED_INTERMEDIATE_DIR)/chrome/browser/resources/side_panel/customize_chrome/resources.grd": {
     "META": {"sizes": {"includes": [50],}},
-    "includes": [1930],
+    "includes": [2300],
   },
   "<(SHARED_INTERMEDIATE_DIR)/chrome/browser/resources/side_panel/history_clusters/resources.grd": {
     "META": {"sizes": {"includes": [5],}},
-    "includes": [1931],
+    "includes": [2320],
   },
   "<(SHARED_INTERMEDIATE_DIR)/chrome/browser/resources/side_panel/read_anything/resources.grd": {
     "META": {"sizes": {"includes": [10],}},
-    "includes": [1932],
+    "includes": [2340],
   },
   "<(SHARED_INTERMEDIATE_DIR)/chrome/browser/resources/side_panel/reading_list/resources.grd": {
     "META": {"sizes": {"includes": [10],}},
-    "includes": [1933],
+    "includes": [2360],
   },
   "<(SHARED_INTERMEDIATE_DIR)/chrome/browser/resources/side_panel/shared/resources.grd": {
     "META": {"sizes": {"includes": [10],}},
-    "includes": [1934],
+    "includes": [2380],
   },
   "<(SHARED_INTERMEDIATE_DIR)/chrome/browser/resources/side_panel/user_notes/resources.grd": {
     "META": {"sizes": {"includes": [10],}},
-    "includes": [1935],
+    "includes": [2400],
   },
   "<(SHARED_INTERMEDIATE_DIR)/chrome/browser/resources/settings/chromeos/os_settings_resources.grd": {
     "META": {"sizes": {"includes": [1000],}},
-    "includes": [1940],
+    "includes": [2420],
   },
   "<(SHARED_INTERMEDIATE_DIR)/chrome/browser/resources/settings/resources.grd": {
     "META": {"sizes": {"includes": [500],}},
-    "includes": [1960],
+    "includes": [2440],
   },
   "<(SHARED_INTERMEDIATE_DIR)/chrome/browser/resources/signin/profile_picker/resources.grd": {
     "META": {"sizes": {"includes": [50],}},
-    "includes": [1980],
+    "includes": [2460],
   },
   "<(SHARED_INTERMEDIATE_DIR)/chrome/browser/resources/signin/resources.grd": {
     "META": {"sizes": {"includes": [60],}},
-    "includes": [2000],
+    "includes": [2480],
   },
   "<(SHARED_INTERMEDIATE_DIR)/chrome/browser/resources/support_tool/resources.grd": {
     "META": {"sizes": {"includes": [30]}},
-    "includes": [2010],
+    "includes": [2500],
   },
   "<(SHARED_INTERMEDIATE_DIR)/chrome/browser/resources/tab_search/resources.grd": {
     "META": {"sizes": {"includes": [30]}},
-    "includes": [2020],
+    "includes": [2520],
   },
   "<(SHARED_INTERMEDIATE_DIR)/chrome/browser/resources/tab_strip/resources.grd": {
     "META": {"sizes": {"includes": [30]}},
-    "includes": [2040],
+    "includes": [2540],
   },
   "<(SHARED_INTERMEDIATE_DIR)/chrome/browser/resources/welcome/resources.grd": {
     "META": {"sizes": {"includes": [60]}},
-    "includes": [2060],
+    "includes": [2560],
   },
   "chrome/browser/test_dummy/internal/android/resources/resources.grd": {
-    "includes": [2100],
+    "includes": [2580],
   },
   "<(SHARED_INTERMEDIATE_DIR)/chrome/browser/resources/webui_gallery/resources.grd": {
     "META": {"sizes": {"includes": [70]}},
-    "includes": [2130],
+    "includes": [2600],
   },
   "<(SHARED_INTERMEDIATE_DIR)/chrome/browser/resources/web_app_internals/resources.grd": {
     "META": {"sizes": {"includes": [10]}},
-    "includes": [2135],
+    "includes": [2620],
   },
   "<(SHARED_INTERMEDIATE_DIR)/chrome/browser/resources/whats_new/resources.grd": {
     "META": {"sizes": {"includes": [10]}},
-    "includes": [2140],
+    "includes": [2640],
   },
   "<(SHARED_INTERMEDIATE_DIR)/chrome/browser/resources/password_manager/resources.grd": {
     "META": {"sizes": {"includes": [50]}},
-    "includes": [2150],
+    "includes": [2660],
   },
   "<(SHARED_INTERMEDIATE_DIR)/chrome/browser/resources/omnibox_popup/resources.grd": {
     "META": {"sizes": {"includes": [50]}},
-    "includes": [2160],
+    "includes": [2680],
   },
   # END chrome/browser section.
 
   # START chrome/ WebUI resources section
   "<(SHARED_INTERMEDIATE_DIR)/chrome/browser/resources/browsing_topics/resources.grd": {
     "META": {"sizes": {"includes": [10]}},
-    "includes": [2170],
+    "includes": [2700],
   },
   "<(SHARED_INTERMEDIATE_DIR)/chrome/browser/resources/bluetooth_internals/resources.grd": {
     "META": {"sizes": {"includes": [50],}},
-    "includes": [2180],
+    "includes": [2720],
   },
   "<(SHARED_INTERMEDIATE_DIR)/chrome/browser/resources/chromeos/audio/resources.grd": {
     "META": {"sizes": {"includes": [30]}},
-    "includes": [2200],
+    "includes": [2740],
   },
   "<(SHARED_INTERMEDIATE_DIR)/chrome/browser/resources/chromeos/bluetooth_pairing_dialog/bluetooth_pairing_dialog_resources.grd": {
     "META": {"sizes": {"includes": [10],}},
-    "includes": [2220],
+    "includes": [2760],
   },
   "<(SHARED_INTERMEDIATE_DIR)/chrome/browser/resources/chromeos/chromebox_for_meetings/resources.grd": {
     "META": {"sizes": {"includes": [5]}},
-    "includes": [2240],
+    "includes": [2780],
   },
   "<(SHARED_INTERMEDIATE_DIR)/chrome/browser/resources/chromeos/internet_config_dialog/internet_config_dialog_resources.grd": {
     "META": {"sizes": {"includes": [10],}},
-    "includes": [2260],
+    "includes": [2800],
   },
   "<(SHARED_INTERMEDIATE_DIR)/chrome/browser/resources/chromeos/internet_detail_dialog/internet_detail_dialog_resources.grd": {
     "META": {"sizes": {"includes": [10],}},
-    "includes": [2280],
+    "includes": [2820],
   },
   "<(SHARED_INTERMEDIATE_DIR)/chrome/browser/resources/chromeos/network_ui/network_ui_resources.grd": {
     "META": {"sizes": {"includes": [20]}},
-    "includes": [2300],
+    "includes": [2840],
   },
   "<(SHARED_INTERMEDIATE_DIR)/components/history_clusters/history_clusters_internals/resources/resources.grd": {
     "META": {"sizes": {"includes": [10]}},
-    "includes": [2310],
+    "includes": [2860],
   },
   "<(SHARED_INTERMEDIATE_DIR)/components/download/resources/download_internals/resources.grd": {
     "META": {"sizes": {"includes": [10]}},
-    "includes": [2320],
+    "includes": [2880],
   },
   "<(SHARED_INTERMEDIATE_DIR)/components/optimization_guide/optimization_guide_internals/resources/resources.grd": {
     "META": {"sizes": {"includes": [10]}},
-    "includes": [2330],
+    "includes": [2900],
   },
   "<(SHARED_INTERMEDIATE_DIR)/components/policy/resources/webui/resources.grd": {
     "META": {"sizes": {"includes": [20]}},
-    "includes": [2335],
+    "includes": [2920],
   },
   "<(SHARED_INTERMEDIATE_DIR)/chrome/browser/resources/app_home/resources.grd": {
     "META": {"sizes": {"includes": [10]}},
-    "includes": [2340],
+    "includes": [2940],
   },
   "<(SHARED_INTERMEDIATE_DIR)/chrome/browser/resources/gaia_auth_host/resources.grd": {
     "META": {"sizes": {"includes": [20],}},
-    "includes": [2350],
+    "includes": [2960],
   },
   "<(SHARED_INTERMEDIATE_DIR)/chrome/browser/resources/invalidations/resources.grd": {
     "META": {"sizes": {"includes": [10],}},
-    "includes": [2360],
+    "includes": [2980],
   },
   "<(SHARED_INTERMEDIATE_DIR)/chrome/browser/resources/media/resources.grd": {
     "META": {"sizes": {"includes": [20],}},
-    "includes": [2380],
+    "includes": [3000],
   },
   "<(SHARED_INTERMEDIATE_DIR)/chrome/browser/resources/net_internals/resources.grd": {
     "META": {"sizes": {"includes": [20]}},
-    "includes": [2400],
+    "includes": [3020],
   },
   "<(SHARED_INTERMEDIATE_DIR)/chrome/browser/resources/omnibox/resources.grd": {
     "META": {"sizes": {"includes": [20]}},
-    "includes": [2420],
+    "includes": [3040],
   },
   "<(SHARED_INTERMEDIATE_DIR)/chrome/browser/resources/quota_internals/quota_internals_resources.grd": {
     "META": {"sizes": {"includes": [20]}},
-    "includes": [2440],
+    "includes": [3060],
   },
   "<(SHARED_INTERMEDIATE_DIR)/chrome/browser/resources/sync_file_system_internals/resources.grd": {
     "META": {"sizes": {"includes": [20]}},
-    "includes": [2460],
+    "includes": [3080],
   },
   "<(SHARED_INTERMEDIATE_DIR)/chrome/browser/resources/usb_internals/resources.grd": {
     "META": {"sizes": {"includes": [20]}},
-    "includes": [2480],
+    "includes": [3100],
   },
   "<(SHARED_INTERMEDIATE_DIR)/chrome/browser/resources/webapks/resources.grd": {
     "META": {"sizes": {"includes": [10]}},
-    "includes": [2500],
+    "includes": [3120],
   },
   "<(SHARED_INTERMEDIATE_DIR)/chrome/browser/resources/webui_js_error/resources.grd": {
    "META": {"sizes": {"includes": [10],}},
-   "includes": [2520],
+   "includes": [3140],
   },
   "<(SHARED_INTERMEDIATE_DIR)/chrome/browser/resources/app_settings/resources.grd": {
     "META": {"sizes": {"includes": [10]}},
-    "includes": [2530],
+    "includes": [3160],
   },
   "<(SHARED_INTERMEDIATE_DIR)/components/sync/driver/resources/resources.grd": {
    "META": {"sizes": {"includes": [30],}},
-    "includes": [2540],
+    "includes": [3180],
   },
   "components/resources/dev_ui_components_resources.grd": {
-    "includes": [2560],
+    "includes": [3200],
   },
   "<(SHARED_INTERMEDIATE_DIR)/content/browser/resources/aggregation_service/resources.grd": {
     "META": {"sizes": {"includes": [20]}},
-    "includes": [2563],
+    "includes": [3220],
   },
   "<(SHARED_INTERMEDIATE_DIR)/content/browser/resources/attribution_reporting/resources.grd": {
     "META": {"sizes": {"includes": [20]}},
-    "includes": [2565],
+    "includes": [3240],
   },
   "<(SHARED_INTERMEDIATE_DIR)/content/browser/resources/indexed_db/resources.grd": {
     "META": {"sizes": {"includes": [20]}},
-    "includes": [2570],
+    "includes": [3260],
   },
   "<(SHARED_INTERMEDIATE_DIR)/content/browser/resources/media/resources.grd": {
     "META": {"sizes": {"includes": [10],}},
-    "includes": [2580],
+    "includes": [3280],
   },
   "<(SHARED_INTERMEDIATE_DIR)/content/browser/resources/quota/resources.grd": {
     "META": {"sizes": {"includes": [10],}},
-    "includes": [2590],
+    "includes": [3300],
   },
   "<(SHARED_INTERMEDIATE_DIR)/content/browser/webrtc/resources/resources.grd": {
     "META": {"sizes": {"includes": [20],}},
-    "includes": [2600],
+    "includes": [3320],
   },
   "content/dev_ui_content_resources.grd": {
-    "includes": [2620],
+    "includes": [3340],
   },
   "<(SHARED_INTERMEDIATE_DIR)/chrome/browser/resources/feed/resources.grd": {
     "META": {"sizes": {"includes": [20]}},
-    "includes": [2640],
+    "includes": [3360],
   },
   "<(SHARED_INTERMEDIATE_DIR)/chrome/browser/resources/chromeos/manage_mirrorsync/resources.grd": {
     "META": {"sizes": {"includes": [10]}},
-    "includes": [2650],
+    "includes": [3380],
   },
   "<(SHARED_INTERMEDIATE_DIR)/chrome/browser/resources/inline_login/resources.grd": {
     "META": {"sizes": {"includes": [20]}},
-    "includes": [2655],
+    "includes": [3400],
   },
   "<(SHARED_INTERMEDIATE_DIR)/components/metrics/debug/resources.grd": {
     "META": {"sizes": {"includes": [10]}},
-    "includes": [2665],
+    "includes": [3420],
   },
   "<(SHARED_INTERMEDIATE_DIR)/chrome/browser/resources/lens/resources.grd": {
     "META": {"sizes": {"includes": [10]}},
-    "includes": [2675],
+    "includes": [3440],
   },
   "<(SHARED_INTERMEDIATE_DIR)/chrome/browser/resources/lens/untrusted_resources.grd": {
     "META": {"sizes": {"includes": [10]}},
-    "includes": [2685],
+    "includes": [3460],
   },
   # END chrome/ WebUI resources section
 
@@ -541,29 +544,29 @@
   "chrome/common/common_resources.grd": {
     # Big alignment at start of section.
     "META": {"align": 100},
-    "includes": [2700],
+    "includes": [3500],
   },
   "<(SHARED_INTERMEDIATE_DIR)/chrome/common/chromeos/extensions/chromeos_system_extensions_resources.grd": {
     "META": {"sizes": {"includes": [10],}},
-    "includes": [2720],
+    "includes": [3520],
   },
   "chrome/credential_provider/gaiacp/gaia_resources.grd": {
-    "includes": [2740],
-    "messages": [2760],
+    "includes": [3540],
+    "messages": [3560],
   },
   "chrome/renderer/resources/renderer_resources.grd": {
-    "includes": [2780],
-    "structures": [2800],
+    "includes": [3580],
+    "structures": [3600],
   },
   "<(SHARED_INTERMEDIATE_DIR)/chrome/test/data/webui/resources.grd": {
     "META": {"sizes": {"includes": [1400],}},
-    "includes": [2810],
+    "includes": [3620],
   },
   "chrome/test/data/webui_test_resources.grd": {
-    "includes": [2820],
+    "includes": [3640],
   },
   "chrome/test/data/chrome_test_resources.grd": {
-    "messages": [2840],
+    "messages": [3660],
   },
   # END chrome/ miscellaneous section.
 
@@ -571,56 +574,56 @@
   "chromeos/chromeos_strings.grd": {
     # Big alignment at start of section.
     "META": {"align": 100},
-    "messages": [2900],
+    "messages": [3700],
   },
   "<(SHARED_INTERMEDIATE_DIR)/ash/ambient/resources/lottie_resources.grd": {
     "META": {"sizes": {"includes": [100],}},
-    "includes": [2910],
+    "includes": [3720],
   },
   "chromeos/ash/resources/ash_resources.grd": {
-    "includes": [2915],
+    "includes": [3740],
   },
   "<(SHARED_INTERMEDIATE_DIR)/ash/webui/camera_app_ui/ash_camera_app_resources.grd": {
     "META": {"sizes": {"includes": [300],}},
-    "includes": [2920],
+    "includes": [3760],
   },
   "ash/webui/camera_app_ui/resources/strings/camera_strings.grd": {
-    "messages": [2940],
+    "messages": [3780],
   },
   "<(SHARED_INTERMEDIATE_DIR)/ash/webui/color_internals/resources/ash_color_internals_resources.grd": {
     "META": {"sizes": {"includes": [20],}},
-    "includes": [2950],
+    "includes": [3800],
   },
   "<(SHARED_INTERMEDIATE_DIR)/ash/webui/common/resources/office_fallback/resources.grd": {
     "META": {"sizes": {"includes": [5]}},
-    "includes": [2955],
+    "includes": [3820],
   },
   "<(SHARED_INTERMEDIATE_DIR)/ash/webui/connectivity_diagnostics/resources/connectivity_diagnostics_resources.grd": {
     "META": {"sizes": {"includes": [50],}},
-    "includes": [2960],
+    "includes": [3840],
   },
   "<(SHARED_INTERMEDIATE_DIR)/ash/webui/diagnostics_ui/resources/ash_diagnostics_app_resources.grd": {
     "META": {"sizes": {"includes": [200],}},
-    "includes": [2980],
+    "includes": [3860],
   },
   "<(SHARED_INTERMEDIATE_DIR)/ash/webui/file_manager/resources/file_manager_swa_resources.grd": {
     "META": {"sizes": {"includes": [100]}},
-    "includes": [3000],
+    "includes": [3880],
   },
   "<(SHARED_INTERMEDIATE_DIR)/ash/webui/file_manager/untrusted_resources/file_manager_untrusted_resources.grd": {
     "META": {"sizes": {"includes": [20]}},
-    "includes": [3020],
+    "includes": [3900],
   },
   "<(SHARED_INTERMEDIATE_DIR)/ash/webui/files_internals/ash_files_internals_resources.grd": {
     "META": {"sizes": {"includes": [10],}},
-    "includes": [3025],
+    "includes": [3920],
   },
   "<(SHARED_INTERMEDIATE_DIR)/ash/webui/guest_os_installer/resources/resources.grd": {
     "META": {"sizes": {"includes": [10]}},
-    "includes": [3030],
+    "includes": [3940],
   },
   "ash/webui/help_app_ui/resources/help_app_resources.grd": {
-    "includes": [3040],
+    "includes": [3960],
   },
   # Both help_app_kids_magazine_bundle_resources.grd and
   # help_app_kids_magazine_bundle_mock_resources.grd start with the same id
@@ -629,10 +632,10 @@
   # in this case (HTML, JS and CSS file).
   "ash/webui/help_app_ui/resources/prod/help_app_kids_magazine_bundle_resources.grd": {
     "META": {"sizes": {"includes": [15],}},
-    "includes": [3060],
+    "includes": [3980],
   },
   "ash/webui/help_app_ui/resources/mock/help_app_kids_magazine_bundle_mock_resources.grd": {
-    "includes": [3060],
+    "includes": [3980],
   },
   # Both help_app_bundle_resources.grd and help_app_bundle_mock_resources.grd
   # start with the same id because only one of them is built depending on if
@@ -641,14 +644,14 @@
   # and bundled content in the top 25 languages (25 x 2).
   "ash/webui/help_app_ui/resources/prod/help_app_bundle_resources.grd": {
     "META": {"sizes": {"includes": [300],}},  # Relies on src-internal.
-    "includes": [3080],
+    "includes": [4000],
   },
   "ash/webui/help_app_ui/resources/mock/help_app_bundle_mock_resources.grd": {
-    "includes": [3080],
+    "includes": [4000],
   },
   "ash/webui/media_app_ui/resources/media_app_resources.grd": {
     "META": {"join": 2},
-    "includes": [3100],
+    "includes": [4020],
   },
   # Both media_app_bundle_resources.grd and media_app_bundle_mock_resources.grd
   # start with the same id because only one of them is built depending on if
@@ -656,81 +659,81 @@
   # of languages (74).
   "ash/webui/media_app_ui/resources/prod/media_app_bundle_resources.grd": {
     "META": {"sizes": {"includes": [130],}},  # Relies on src-internal.
-    "includes": [3120],
+    "includes": [4040],
   },
   "ash/webui/media_app_ui/resources/mock/media_app_bundle_mock_resources.grd": {
-    "includes": [3120],
+    "includes": [4040],
   },
   "<(SHARED_INTERMEDIATE_DIR)/ash/webui/print_management/resources/resources.grd": {
     "META": {"join": 2, "sizes": {"includes": [20]}},
-    "includes": [3140],
+    "includes": [4060],
   },
   "<(SHARED_INTERMEDIATE_DIR)/ash/webui/sample_system_web_app_ui/resources/trusted/ash_sample_system_web_app_resources.grd": {
     "META": {"sizes": {"includes": [50],}},
-    "includes": [3180],
+    "includes": [4080],
   },
   "<(SHARED_INTERMEDIATE_DIR)/ash/webui/sample_system_web_app_ui/resources/untrusted/ash_sample_system_web_app_untrusted_resources.grd": {
     "META": {"sizes": {"includes": [50],}},
-    "includes": [3200],
+    "includes": [4100],
   },
   "ash/webui/scanning/resources/scanning_app_resources.grd": {
-    "includes": [3220],
-    "structures": [3240],
+    "includes": [4120],
+    "structures": [4140],
   },
   "<(SHARED_INTERMEDIATE_DIR)/ash/webui/system_extensions_internals_ui/ash_system_extensions_internals_resources.grd": {
     "META": {"sizes": {"includes": [10],}},
-    "includes": [3260],
+    "includes": [4160],
   },
   "chromeos/resources/chromeos_resources.grd": {
-    "includes": [3280],
+    "includes": [4180],
   },
   "<(SHARED_INTERMEDIATE_DIR)/ash/webui/eche_app_ui/ash_eche_app_resources.grd": {
     "META": {"sizes": {"includes": [50],}},
-    "includes": [3300],
+    "includes": [4200],
   },
   # Both ash_eche_bundle_resources.grd and ash_eche_bundle_mock_resources.grd
   # start with the same id because only one of them is built depending on if
   # src_internal is available.
   "ash/webui/eche_app_ui/resources/prod/ash_eche_bundle_resources.grd": {
     "META": {"sizes": {"includes": [120],}},
-    "includes": [3320],
+    "includes": [4220],
   },
   "ash/webui/eche_app_ui/resources/mock/ash_eche_bundle_mock_resources.grd": {
     "META": {"sizes": {"includes": [120],}},
-    "includes": [3320],
+    "includes": [4220],
   },
   "ash/webui/multidevice_debug/resources/multidevice_debug_resources.grd": {
     "META": {"join": 2},
-    "includes": [3330],
+    "includes": [4240],
   },
   "<(SHARED_INTERMEDIATE_DIR)/ash/webui/personalization_app/resources/ash_personalization_app_resources.grd": {
     "META": {"sizes": {"includes": [200],}},
-    "includes": [3340],
+    "includes": [4260],
   },
   "<(SHARED_INTERMEDIATE_DIR)/ash/webui/demo_mode_app_ui/ash_demo_mode_app_resources.grd": {
     "META": {"sizes": {"includes": [50],}},
-   "includes": [3360],
+   "includes": [4280],
   },
 
   "<(SHARED_INTERMEDIATE_DIR)/ash/webui/projector_app/resources/app/trusted/ash_projector_app_trusted_resources.grd": {
     "META": {"sizes": {"includes": [50],}},
-    "includes": [3380],
+    "includes": [4300],
   },
  "<(SHARED_INTERMEDIATE_DIR)/ash/webui/projector_app/resources/app/untrusted/ash_projector_app_untrusted_resources.grd": {
     "META": {"sizes": {"includes": [50],}},
-    "includes": [3395],
+    "includes": [4320],
   },
   "<(SHARED_INTERMEDIATE_DIR)/ash/webui/projector_app/resources/annotator/untrusted/ash_projector_annotator_untrusted_resources.grd": {
     "META": {"sizes": {"includes": [50],}},
-    "includes": [3405],
+    "includes": [4340],
   },
   "<(SHARED_INTERMEDIATE_DIR)/ash/webui/projector_app/resources/annotator/trusted/ash_projector_annotator_trusted_resources.grd": {
     "META": {"sizes": {"includes": [50],}},
-    "includes": [3410],
+    "includes": [4360],
   },
   "<(SHARED_INTERMEDIATE_DIR)/ash/webui/projector_app/resources/common/ash_projector_common_resources.grd": {
     "META": {"sizes": {"includes": [50],}},
-    "includes": [3415],
+    "includes": [4380],
   },
 
   # Both projector_app_bundle_resources.grd and projector_app_bundle_mock_resources.grd
@@ -739,28 +742,28 @@
   # of languages (79).
   "ash/webui/projector_app/resources/prod/projector_app_bundle_resources.grd": {
     "META": {"sizes": {"includes": [120],}}, # Relies on src-internal.
-    "includes": [3420],
+    "includes": [4400],
   },
   "ash/webui/projector_app/resources/mock/projector_app_bundle_mock_resources.grd": {
-    "includes": [3420],
+    "includes": [4400],
   },
   "<(SHARED_INTERMEDIATE_DIR)/ash/webui/face_ml_app_ui/resources/trusted/ash_face_ml_app_resources.grd": {
     "META": {"join": 2, "sizes": {"includes": [50],}},
-    "includes": [3440],
+    "includes": [4420],
   },
   # Both face_ml_app_bundle_resources.grd and face_ml_app_bundle_mock_resources.grd
   # start with the same id because only one of them is built depending on if
   # src_internal is available.
   "ash/webui/face_ml_app_ui/resources/prod/face_ml_app_bundle_resources.grd": {
     "META": {"sizes": {"includes": [120],}},  # Relies on src-internal.
-    "includes": [3460],
+    "includes": [4440],
   },
   "ash/webui/face_ml_app_ui/resources/mock/face_ml_app_bundle_mock_resources.grd": {
-    "includes": [3460],
+    "includes": [4440],
   },
   "<(SHARED_INTERMEDIATE_DIR)/ash/webui/face_ml_app_ui/resources/untrusted/ash_face_ml_app_untrusted_resources.grd": {
     "META": {"join": 2, "sizes": {"includes": [50],}},
-    "includes": [3480],
+    "includes": [4460],
   },
   # END chromeos/ section.
 
@@ -770,52 +773,52 @@
   "ash/components/arc/input_overlay/resources/input_overlay_resources.grd": {
     # Big alignment at start of section.
     "META": {"align": 1000},
-    "includes": [3500],
+    "includes": [5000],
   },
   # Chromium strings and Google Chrome strings must start at the same id.
   # We only use one file depending on whether we're building Chromium or
   # Google Chrome.
   "components/components_chromium_strings.grd": {
-    "messages": [3520],
+    "messages": [5020],
   },
   "components/components_google_chrome_strings.grd": {
-    "messages": [3520],
+    "messages": [5020],
   },
   "components/components_locale_settings.grd": {
     "META": {"join": 2},
-    "includes": [3540],
-    "messages": [3560],
+    "includes": [5040],
+    "messages": [5060],
   },
   "components/components_strings.grd": {
-    "messages": [3580],
+    "messages": [5080],
   },
   "components/headless/command_handler/headless_command.grd": {
-    "includes": [3590],
+    "includes": [5100],
   },
   "components/omnibox/resources/omnibox_pedal_synonyms.grd": {
-    "messages": [3600],
+    "messages": [5120],
   },
   # components/policy/resources/policy_templates.grd and
   # components/policy/resources/policy_templates.build.grd must share the same
   # id because they are based on the same structure, however they are used in
   # different pipelines.
   "components/policy/resources/policy_templates.grd": {
-    "structures": [3640],
+    "structures": [5140],
   },
   "components/policy/resources/policy_templates.build.grd": {
-    "structures": [3640],
+    "structures": [5140],
   },
   "components/resources/components_resources.grd": {
-    "includes": [3660],
+    "includes": [5160],
   },
   "components/resources/components_scaled_resources.grd": {
-    "structures": [3680],
+    "structures": [5180],
   },
   "components/embedder_support/android/java/strings/web_contents_delegate_android_strings.grd": {
-    "messages": [3700],
+    "messages": [5200],
   },
   "components/autofill/core/browser/autofill_address_rewriter_resources.grd":{
-    "includes": [3720]
+    "includes": [5220]
   },
   # END components/ section.
 
@@ -887,10 +890,10 @@
   },
   "ios_internal/chrome/app/ios_internal_chromium_strings.grd": {
     "META": {"join": 2},
-    "messages": [3740],
+    "messages": [5240],
   },
   "ios_internal/chrome/app/ios_internal_google_chrome_strings.grd": {
-    "messages": [3740],
+    "messages": [5240],
   },
   # END ios_internal/ section.
 
@@ -899,19 +902,19 @@
   "content/content_resources.grd": {
     # Big alignment at start of section.
     "META": {"join": 2, "align": 100},
-    "includes": [3800],
+    "includes": [5300],
   },
   "content/shell/shell_resources.grd": {
-    "includes": [3820],
+    "includes": [5320],
   },
   "content/test/web_ui_mojo_test_resources.grd": {
-    "includes": [3840],
+    "includes": [5340],
   },
 
   # This file is generated during the build.
   "<(SHARED_INTERMEDIATE_DIR)/content/browser/tracing/tracing_resources.grd": {
     "META": {"sizes": {"includes": [20],}},
-    "includes": [3860],
+    "includes": [5360],
   },
   # END content/ section.
 
@@ -921,10 +924,10 @@
   "ios/web/ios_web_resources.grd": {
     # Big alignment at start of section.
     "META": {"align": 100},
-    "includes": [3800],
+    "includes": [5300],
   },
   "ios/web/test/test_resources.grd": {
-    "includes": [3820],
+    "includes": [5320],
   },
   # END ios/web/ section.
 
@@ -933,163 +936,163 @@
   "ash/ash_strings.grd": {
     # Big alignment at start of section.
     "META": {"join": 2, "align": 100},
-    "messages": [3960],
+    "messages": [5400],
   },
   "<(SHARED_INTERMEDIATE_DIR)/ash/webui/os_feedback_ui/resources/ash_os_feedback_resources.grd": {
     "META": {"sizes": {"includes": [50],}},
-    "includes": [3980],
+    "includes": [5420],
   },
   "<(SHARED_INTERMEDIATE_DIR)/ash/webui/os_feedback_ui/untrusted_resources/ash_os_feedback_untrusted_resources.grd": {
     "META": {"sizes": {"includes": [50],}},
-    "includes": [3982],
+    "includes": [5440],
   },
   "<(SHARED_INTERMEDIATE_DIR)/ash/webui/firmware_update_ui/resources/ash_firmware_update_app_resources.grd": {
     "META": {"sizes": {"includes": [200],}},
-    "includes": [3990],
+    "includes": [5460],
   },
   "<(SHARED_INTERMEDIATE_DIR)/ash/webui/shortcut_customization_ui/resources/ash_shortcut_customization_app_resources.grd": {
     "META": {"sizes": {"includes": [200],}},
-    "includes": [4000],
+    "includes": [5480],
   },
   "ash/shortcut_viewer/shortcut_viewer_strings.grd": {
-    "messages": [4020],
+    "messages": [5500],
   },
   "<(SHARED_INTERMEDIATE_DIR)/ash/webui/shimless_rma/resources/ash_shimless_rma_resources.grd": {
     "META": {"sizes": {"includes": [50],}},
-    "includes": [4040],
+    "includes": [5520],
   },
   "ash/keyboard/ui/keyboard_resources.grd": {
-    "includes": [4060],
+    "includes": [5540],
   },
   "ash/login/resources/login_resources.grd": {
-    "structures": [4080],
+    "structures": [5560],
   },
   "ash/public/cpp/resources/ash_public_unscaled_resources.grd": {
-    "includes": [4100],
-    "structures": [4101],
+    "includes": [5580],
+    "structures": [5600],
   },
   "base/tracing/protos/resources.grd": {
-    "includes": [4120],
+    "includes": [5620],
   },
   "chromecast/app/resources/chromecast_settings.grd": {
-    "messages": [4140],
+    "messages": [5640],
   },
   "chromecast/app/resources/shell_resources.grd": {
-    "includes": [4160],
+    "includes": [5660],
   },
   "chromecast/renderer/resources/extensions_renderer_resources.grd": {
-    "includes": [4180],
+    "includes": [5680],
   },
 
   "device/bluetooth/bluetooth_strings.grd": {
-    "messages": [4240],
+    "messages": [5700],
   },
 
   "device/fido/fido_strings.grd": {
-    "messages": [4260],
+    "messages": [5720],
   },
 
   "extensions/browser/resources/extensions_browser_resources.grd": {
-    "structures": [4280],
+    "structures": [5740],
   },
   "extensions/extensions_resources.grd": {
-    "includes": [4300],
+    "includes": [5760],
   },
   "extensions/renderer/resources/extensions_renderer_resources.grd": {
-    "includes": [4320],
-    "structures": [4340],
+    "includes": [5780],
+    "structures": [5800],
   },
   "extensions/shell/app_shell_resources.grd": {
-    "includes": [4360],
+    "includes": [5820],
   },
   "extensions/strings/extensions_strings.grd": {
-    "messages": [4380],
+    "messages": [5840],
   },
 
   "mojo/public/js/mojo_bindings_resources.grd": {
-    "includes": [4420],
+    "includes": [5860],
   },
 
   "net/base/net_resources.grd": {
-    "includes": [4440],
+    "includes": [5880],
   },
 
   "remoting/resources/remoting_strings.grd": {
-    "messages": [4460],
+    "messages": [5900],
   },
 
   "services/services_strings.grd": {
-    "messages": [4480],
+    "messages": [5920],
   },
   "skia/skia_resources.grd": {
-    "includes": [4500],
+    "includes": [5940],
   },
   "third_party/blink/public/blink_image_resources.grd": {
-    "structures": [4520],
+    "structures": [5960],
   },
   "third_party/blink/public/blink_resources.grd": {
-    "includes": [4540],
+    "includes": [5980],
   },
   "third_party/blink/renderer/modules/media_controls/resources/media_controls_resources.grd": {
-    "includes": [4560],
-    "structures": [4580],
+    "includes": [6000],
+    "structures": [6020],
   },
   "third_party/blink/public/strings/blink_accessibility_strings.grd": {
-    "messages": [4600],
+    "messages": [6040],
   },
   "third_party/blink/public/strings/blink_strings.grd": {
-    "messages": [4610],
+    "messages": [6060],
   },
   "third_party/libaddressinput/chromium/address_input_strings.grd": {
-    "messages": [4620],
+    "messages": [6080],
   },
 
   "ui/base/test/ui_base_test_resources.grd": {
-    "messages": [4640],
+    "messages": [6100],
   },
   "ui/chromeos/resources/ui_chromeos_resources.grd": {
-    "structures": [4660],
+    "structures": [6120],
   },
   "ui/chromeos/ui_chromeos_strings.grd": {
-    "messages": [4680],
+    "messages": [6140],
   },
   "<(SHARED_INTERMEDIATE_DIR)/ui/file_manager/file_manager_gen_resources.grd": {
     "META": {"sizes": {"includes": [2000]}},
-    "includes": [4700],
+    "includes": [6160],
   },
   "ui/file_manager/file_manager_resources.grd": {
-    "includes": [4720],
+    "includes": [6180],
   },
   "ui/resources/ui_resources.grd": {
-    "structures": [4740],
+    "structures": [6200],
   },
   "ui/resources/ui_unscaled_resources.grd": {
-    "includes": [4760],
+    "includes": [6220],
   },
   "ui/strings/app_locale_settings.grd": {
-    "messages": [4780],
+    "messages": [6240],
   },
   "ui/strings/ax_strings.grd": {
-    "messages": [4800],
+    "messages": [6260],
   },
   "ui/strings/ui_strings.grd": {
-    "messages": [4810],
+    "messages": [6280],
   },
   "ui/views/examples/views_examples_resources.grd": {
-    "messages": [4820],
+    "messages": [6300],
   },
   "ui/views/resources/views_resources.grd": {
-    "structures": [4840],
+    "structures": [6320],
   },
   "ui/webui/examples/resources/webui_examples_resources.grd": {
-    "messages": [4850],
+    "messages": [6340],
   },
   "<(SHARED_INTERMEDIATE_DIR)/ui/webui/resources/webui_resources.grd": {
     "META": {"sizes": {"includes": [900]}},
-    "includes": [4900],
+    "includes": [6360],
   },
   "weblayer/weblayer_resources.grd": {
-    "includes": [4920],
+    "includes": [6380],
   },
 
   # This file is generated during the build.
@@ -1098,13 +1101,13 @@
     # In debug build, devtools frontend sources are not bundled and therefore
     # includes a lot of individual resources
     "META": {"sizes": {"includes": [2100],}},
-    "includes": [4940],
+    "includes": [6400],
   },
 
   # This file is generated during the build.
   "<(SHARED_INTERMEDIATE_DIR)/resources/inspector_overlay/inspector_overlay_resources.grd": {
     "META": {"sizes": {"includes": [50],}},
-    "includes": [4960],
+    "includes": [6420],
   },
 
   # END "everything else" section.
