diff --git a/tools/gritsettings/resource_ids.spec b/tools/gritsettings/resource_ids.spec
index ee2a113e2b77873b18d60deee0a373ea335bf149..2ed50d743701e0ae0488c45542f58c1de72f048f 100644
--- a/tools/gritsettings/resource_ids.spec
+++ b/tools/gritsettings/resource_ids.spec
@@ -38,6 +38,9 @@
   "chrome/app/chromium_strings.grd": {
     "messages": [800],
   },
+  "chrome/app/mises_strings.grd": {
+    "messages": [800],
+  },
   "chrome/app/google_chrome_strings.grd": {
     "messages": [800],
   },
@@ -1015,8 +1018,11 @@
   "components/components_chromium_strings.grd": {
     "messages": [7020],
   },
+  "components/components_mises_strings.grd": {
+    "messages": [6020],
+  },
   "components/components_google_chrome_strings.grd": {
-    "messages": [7020],
+    "messages": [6040],
   },
   "components/components_locale_settings.grd": {
     "META": {"join": 2},
