diff --git a/tools/gritsettings/resource_ids.spec b/tools/gritsettings/resource_ids.spec
index 15e77b25cc0b10e59d154d2389d4b8dbaca6391a..07c81af5c35e18c7d13f915890933599849dc0c6 100644
--- a/tools/gritsettings/resource_ids.spec
+++ b/tools/gritsettings/resource_ids.spec
@@ -44,6 +44,9 @@
   "chrome/app/google_chrome_strings.grd": {
     "messages": [400],
   },
+  "chrome/app/mises_strings.grd": {
+    "messages": [400],
+  },
 
   # Leave lots of space for generated_resources since it has most of our
   # strings.
