diff --git a/tools/gritsettings/resource_ids.spec b/tools/gritsettings/resource_ids.spec
index 95e5ac768b9049ca5ce09264cd391f5f7247b5a3..99745e6586d2310e6c55235f7122cc4f78beaab0 100644
--- a/tools/gritsettings/resource_ids.spec
+++ b/tools/gritsettings/resource_ids.spec
@@ -38,6 +38,9 @@
   "chrome/app/chromium_strings.grd": {
     "messages": [800],
   },
+  "chrome/app/mises_strings.grd": {
+    "messages": [800],
+  },
   "chrome/app/google_chrome_strings.grd": {
     "messages": [800],
   },
@@ -927,6 +930,9 @@
   "components/components_chromium_strings.grd": {
     "messages": [7020],
   },
+  "components/components_mises_strings.grd": {
+    "messages": [6020],
+  },
   "components/components_google_chrome_strings.grd": {
     "messages": [7020],
   },
