diff --git a/ui/base/dragdrop/os_exchange_data.h b/ui/base/dragdrop/os_exchange_data.h
index 0c5a6283a322763ce9213b224ff89612a83d09e7..4203b3ad61395dd24c167c1c00b361e3e1095d11 100644
--- a/ui/base/dragdrop/os_exchange_data.h
+++ b/ui/base/dragdrop/os_exchange_data.h
@@ -53,7 +53,7 @@ class COMPONENT_EXPORT(UI_BASE) OSExchangeData {
     FILE_NAME = 1 << 2,
     PICKLED_DATA = 1 << 3,
     FILE_CONTENTS = 1 << 4,
-#if defined(USE_AURA)
+#if BUILDFLAG(IS_ANDROID) || defined(USE_AURA)
     HTML = 1 << 5,
 #endif
 #if BUILDFLAG(IS_CHROMEOS)
