diff --git a/ui/base/dragdrop/os_exchange_data.h b/ui/base/dragdrop/os_exchange_data.h
index c01cc12e761b8a50a5b6af7fec2f5f368799afa8..736ce38cef7879250d91270da6c071bb746a7320 100644
--- a/ui/base/dragdrop/os_exchange_data.h
+++ b/ui/base/dragdrop/os_exchange_data.h
@@ -57,7 +57,7 @@ class COMPONENT_EXPORT(UI_BASE) OSExchangeData {
     FILE_NAME = 1 << 2,
     PICKLED_DATA = 1 << 3,
     FILE_CONTENTS = 1 << 4,
-#if defined(USE_AURA)
+#if BUILDFLAG(IS_ANDROID) || defined(USE_AURA)
     HTML = 1 << 5,
 #endif
 #if BUILDFLAG(IS_CHROMEOS)
