diff --git a/ui/base/dragdrop/os_exchange_data.h b/ui/base/dragdrop/os_exchange_data.h
index 7e0e21766587f4c4c4f46d3cf390eaa7985a05dc..6c4f808dda07e1ab6d8d55ff3bacf41dcbd90e7f 100644
--- a/ui/base/dragdrop/os_exchange_data.h
+++ b/ui/base/dragdrop/os_exchange_data.h
@@ -58,7 +58,7 @@ class COMPONENT_EXPORT(UI_BASE) OSExchangeData {
     FILE_NAME = 1 << 2,
     PICKLED_DATA = 1 << 3,
     FILE_CONTENTS = 1 << 4,
-#if defined(USE_AURA)
+#if BUILDFLAG(IS_ANDROID) || defined(USE_AURA)
     HTML = 1 << 5,
 #endif
 #if BUILDFLAG(IS_CHROMEOS)
