diff --git a/ui/base/BUILD.gn b/ui/base/BUILD.gn
index 3af6a66b319a17f7a5962a3135358dfa3ca5fa00..c4fb9217eff7284b36e90f7fd0d50fcfd45edb36 100644
--- a/ui/base/BUILD.gn
+++ b/ui/base/BUILD.gn
@@ -416,7 +416,7 @@ component("base") {
       "themed_vector_icon.h",
     ]
 
-    if (!is_android && !is_ios) {
+    if (true || !is_android && !is_ios) {
       sources += [
         "menu_source_utils.cc",
         "menu_source_utils.h",
@@ -630,7 +630,7 @@ component("base") {
     ]
   }
 
-  if (!toolkit_views && !use_aura) {
+  if (!is_android && !toolkit_views && !use_aura) {
     sources -= [
       "dragdrop/drop_target_event.cc",
       "dragdrop/drop_target_event.h",
@@ -647,12 +647,18 @@ component("base") {
     deps += [ ":ui_base_jni_headers" ]
     libs += [ "jnigraphics" ]
 
+    if (false) {
     sources -= [
       "l10n/l10n_font_util.cc",
       "models/button_menu_item_model.cc",
       "pointer/touch_editing_controller.cc",
       "theme_provider.cc",
     ]
+    }
+    sources += [
+      "dragdrop/os_exchange_data_provider_non_backed.cc",
+      "dragdrop/os_exchange_data_provider_non_backed.h",
+    ]
   }
 
   if (is_android || is_ios) {
