diff --git a/ui/base/BUILD.gn b/ui/base/BUILD.gn
index 012ab83bb9f4887646decb030b4a18f6b21e0d01..b2ee1ef6950f86393d2fd3982758bfae1dc1a057 100644
--- a/ui/base/BUILD.gn
+++ b/ui/base/BUILD.gn
@@ -591,7 +591,7 @@ component("base") {
     ]
   }
 
-  if (!toolkit_views && !use_aura) {
+  if (!is_android && !toolkit_views && !use_aura) {
     sources -= [
       "dragdrop/drop_target_event.cc",
       "dragdrop/drop_target_event.h",
@@ -608,6 +608,7 @@ component("base") {
     deps += [ ":ui_base_jni_headers" ]
     libs += [ "jnigraphics" ]
 
+    if (false) {
     sources -= [
       "l10n/l10n_font_util.cc",
       "models/button_menu_item_model.cc",
@@ -615,6 +616,11 @@ component("base") {
       "theme_provider.cc",
       "ui_base_types.cc",
     ]
+    }
+    sources += [
+      "dragdrop/os_exchange_data_provider_non_backed.cc",
+      "dragdrop/os_exchange_data_provider_non_backed.h",
+    ]
   }
 
   if (is_android || is_ios) {
