diff --git a/ui/base/clipboard/BUILD.gn b/ui/base/clipboard/BUILD.gn
index 065e8adf7863d829d7455476815e30f1e7639670..62ee5dee34daa994dd002d26338206b2ddf57cf5 100644
--- a/ui/base/clipboard/BUILD.gn
+++ b/ui/base/clipboard/BUILD.gn
@@ -111,7 +111,8 @@ component("clipboard") {
       "clipboard_android.cc",
       "clipboard_android.h",
     ]
-  } else {
+  }
+  if (true) {
     sources += [
       "clipboard_data.cc",
       "clipboard_data.h",
