diff --git a/ui/base/BUILD.gn b/ui/base/BUILD.gn
index acd06385277ac91670e5aa3ad386ccb6740a5a52..248495d56064cd5740627413db95934f6ab3ea3c 100644
--- a/ui/base/BUILD.gn
+++ b/ui/base/BUILD.gn
@@ -600,7 +600,7 @@ component("base") {
     ]
   }
 
-  if (!toolkit_views && !use_aura) {
+  if (!is_android && !toolkit_views && !use_aura) {
     sources -= [
       "dragdrop/drop_target_event.cc",
       "dragdrop/drop_target_event.h",
@@ -617,12 +617,18 @@ component("base") {
     deps += [ ":ui_base_jni_headers" ]
     libs += [ "jnigraphics" ]
 
+    if (false) {
     sources -= [
       "l10n/l10n_font_util.cc",
       "models/button_menu_item_model.cc",
       "pointer/touch_editing_controller.cc",
       "theme_provider.cc",
     ]
+    }
+    sources += [
+      "dragdrop/os_exchange_data_provider_non_backed.cc",
+      "dragdrop/os_exchange_data_provider_non_backed.h",
+    ]
   }
 
   if (is_android || is_ios) {
