diff --git a/ui/base/BUILD.gn b/ui/base/BUILD.gn
index 6a5d2b0d11af3dabf3cdcd21b666fbb5dfe288c1..8745334f667e7cfe08d944e143ec3790139cab4b 100644
--- a/ui/base/BUILD.gn
+++ b/ui/base/BUILD.gn
@@ -402,7 +402,7 @@ component("base") {
       "themed_vector_icon.h",
     ]
 
-    if (!is_android && !is_ios) {
+    if (true || !is_android && !is_ios) {
       sources += [
         "menu_source_utils.cc",
         "menu_source_utils.h",
@@ -612,7 +612,7 @@ component("base") {
     ]
   }
 
-  if (!toolkit_views && !use_aura) {
+  if (!is_android && !toolkit_views && !use_aura) {
     sources -= [
       "dragdrop/drop_target_event.cc",
       "dragdrop/drop_target_event.h",
@@ -629,12 +629,18 @@ component("base") {
     deps += [ ":ui_base_jni_headers" ]
     libs += [ "jnigraphics" ]
 
+    if (false) {
     sources -= [
       "l10n/l10n_font_util.cc",
       "models/button_menu_item_model.cc",
       "pointer/touch_editing_controller.cc",
       "theme_provider.cc",
     ]
+    }
+    sources += [
+      "dragdrop/os_exchange_data_provider_non_backed.cc",
+      "dragdrop/os_exchange_data_provider_non_backed.h",
+    ]
   }
 
   if (is_android || is_ios) {
