diff --git a/ui/base/BUILD.gn b/ui/base/BUILD.gn
index 79937bebda1b3405a8acbf1559ebd994a3f7bf0b..394c17cf919f08d2ce9e52154b04245bdfc385b8 100644
--- a/ui/base/BUILD.gn
+++ b/ui/base/BUILD.gn
@@ -423,7 +423,7 @@ component("base") {
       "themed_vector_icon.h",
     ]
 
-    if (!is_android && !is_ios) {
+    if (true || !is_android && !is_ios) {
       sources += [
         "menu_source_utils.cc",
         "menu_source_utils.h",
@@ -649,7 +649,7 @@ component("base") {
     ]
   }
 
-  if (!toolkit_views && !use_aura) {
+  if (!is_android && !toolkit_views && !use_aura) {
     sources -= [
       "dragdrop/drop_target_event.cc",
       "dragdrop/drop_target_event.h",
@@ -666,11 +666,17 @@ component("base") {
     deps += [ ":ui_base_jni_headers" ]
     libs += [ "jnigraphics" ]
 
+    if (false) {
     sources -= [
       "l10n/l10n_font_util.cc",
       "models/button_menu_item_model.cc",
       "theme_provider.cc",
     ]
+    }
+    sources += [
+      "dragdrop/os_exchange_data_provider_non_backed.cc",
+      "dragdrop/os_exchange_data_provider_non_backed.h",
+    ]
   }
 
   if (is_android || is_ios) {
