diff --git a/ui/gfx/BUILD.gn b/ui/gfx/BUILD.gn
index 5255116a8e8f897607e5c5df2875dbaf275ec919..3f5812805147dbdaf9d70569190f5b64930dca9e 100644
--- a/ui/gfx/BUILD.gn
+++ b/ui/gfx/BUILD.gn
@@ -439,7 +439,7 @@ component("gfx") {
     ]
   }
 
-  if (use_aura || toolkit_views) {
+  if (use_aura || toolkit_views || is_android) {
     sources += [
       "nine_image_painter.cc",
       "nine_image_painter.h",
