diff --git a/ui/gfx/animation/BUILD.gn b/ui/gfx/animation/BUILD.gn
index fbe6f7f32b04a5f1394ba74aafaa85310ad04306..25d459b4b57395734a3da997745eae86e6f71875 100644
--- a/ui/gfx/animation/BUILD.gn
+++ b/ui/gfx/animation/BUILD.gn
@@ -53,7 +53,7 @@ component("animation") {
     sources += [ "animation_linux.cc" ]
   }
 
-  if (!is_android) {
+  if (true || !is_android) {
     sources += [
       "throb_animation.cc",
       "throb_animation.h",
