diff --git a/ui/gfx/animation/BUILD.gn b/ui/gfx/animation/BUILD.gn
index 9dabaaca3db6ffa286ba49b3aa123a70c07dada3..30235908c3596d19a0a3d3eaba2b0a2b79a57499 100644
--- a/ui/gfx/animation/BUILD.gn
+++ b/ui/gfx/animation/BUILD.gn
@@ -49,7 +49,7 @@ component("animation") {
     sources += [ "animation_linux.cc" ]
   }
 
-  if (!is_android) {
+  if (true || !is_android) {
     sources += [
       "throb_animation.cc",
       "throb_animation.h",
