diff --git a/ui/gfx/BUILD.gn b/ui/gfx/BUILD.gn
index 9161edba8266e83480da3c04b63d64a6532517c9..3714bd2a010ced74206e3113cb7e14215ef7f22e 100644
--- a/ui/gfx/BUILD.gn
+++ b/ui/gfx/BUILD.gn
@@ -436,7 +436,7 @@ component("gfx") {
     ]
   }
 
-  if (use_aura || toolkit_views) {
+  if (use_aura || toolkit_views || is_android) {
     sources += [
       "nine_image_painter.cc",
       "nine_image_painter.h",
