diff --git a/ui/gfx/BUILD.gn b/ui/gfx/BUILD.gn
index 9fd5b696fbf8ef4ee2fafc293eb9288763a812f0..4edbb76b473e2805e3732355fa173485e230d98a 100644
--- a/ui/gfx/BUILD.gn
+++ b/ui/gfx/BUILD.gn
@@ -397,7 +397,7 @@ component("gfx") {
     ]
   }
 
-  if ((!use_aura && !toolkit_views) || is_ios) {
+  if ((!use_aura && !toolkit_views && !is_android) || is_ios) {
     sources -= [
       "nine_image_painter.cc",
       "nine_image_painter.h",
