diff --git a/ui/gfx/BUILD.gn b/ui/gfx/BUILD.gn
index c0a85b2cfc69a4f358ae03c65e774cdc45085bf6..45d89442ca0452e5f92462ec93783962032103c2 100644
--- a/ui/gfx/BUILD.gn
+++ b/ui/gfx/BUILD.gn
@@ -439,7 +439,7 @@ component("gfx") {
     ]
   }
 
-  if (use_aura || toolkit_views) {
+  if (use_aura || toolkit_views || is_android) {
     sources += [
       "nine_image_painter.cc",
       "nine_image_painter.h",
