diff --git a/ui/gfx/BUILD.gn b/ui/gfx/BUILD.gn
index d2692c7422e4694a783876addc1b283d451e937f..4ab656b8526d06447f46f401e1b054183d6514f3 100644
--- a/ui/gfx/BUILD.gn
+++ b/ui/gfx/BUILD.gn
@@ -402,7 +402,7 @@ component("gfx") {
     ]
   }
 
-  if ((!use_aura && !toolkit_views) || is_ios) {
+  if ((!use_aura && !toolkit_views && !is_android) || is_ios) {
     sources -= [
       "nine_image_painter.cc",
       "nine_image_painter.h",
