diff --git a/ui/message_center/BUILD.gn b/ui/message_center/BUILD.gn
index 1dc4309115b530b90930c8f53b4bc13412e42608..cb44d78b1be6c5ca8aab927da2917e4173be2e71 100644
--- a/ui/message_center/BUILD.gn
+++ b/ui/message_center/BUILD.gn
@@ -36,7 +36,7 @@ component("message_center") {
 
   defines = [ "MESSAGE_CENTER_IMPLEMENTATION" ]
 
-  if (enable_message_center) {
+  if (is_android || enable_message_center) {
     deps += [
       ":message_center_vector_icons",
       "//base:i18n",
@@ -223,7 +223,7 @@ if (enable_message_center) {
       deps += [ "//ash/constants" ]
     }
 
-    if (toolkit_views) {
+    if (is_android || toolkit_views) {
       sources += [
         "views/large_image_view_unittest.cc",
         "views/message_popup_collection_unittest.cc",
