diff --git a/ui/resources/BUILD.gn b/ui/resources/BUILD.gn
index 67a0edca7b8d654b42363fc9f80da3b6fb45f61e..e22004e1a660474787795aa61f0d52596d10c97e 100644
--- a/ui/resources/BUILD.gn
+++ b/ui/resources/BUILD.gn
@@ -163,7 +163,7 @@ repack("repack_ui_test_pak_100_percent") {
     deps += [ "//ui/chromeos/resources" ]
   }
 
-  if (toolkit_views) {
+  if (is_android || toolkit_views) {
     deps += [ "//ui/views/resources" ]
     sources += [
       "$root_gen_dir/third_party/blink/public/resources/blink_resources.pak",
@@ -181,7 +181,7 @@ repack("repack_ui_test_pak_200_percent") {
 
   deps = [ "//ui/resources" ]
 
-  if (toolkit_views) {
+  if (is_android || toolkit_views) {
     deps += [ "//ui/views/resources" ]
     sources +=
         [ "$root_gen_dir/ui/views/resources/views_resources_200_percent.pak" ]
