diff --git a/ui/resources/BUILD.gn b/ui/resources/BUILD.gn
index 044cfba1ae65ea8918c9469413ace0136f96853b..1539adc92fc3897dc14c814278891fea667d99ce 100644
--- a/ui/resources/BUILD.gn
+++ b/ui/resources/BUILD.gn
@@ -169,7 +169,7 @@ repack("repack_ui_test_pak_100_percent") {
     ]
   }
 
-  if (toolkit_views) {
+  if (is_android || toolkit_views) {
     deps += [ "//ui/views/resources" ]
     sources += [
       "$root_gen_dir/third_party/blink/public/resources/blink_resources.pak",
@@ -187,7 +187,7 @@ repack("repack_ui_test_pak_200_percent") {
 
   deps = [ "//ui/resources" ]
 
-  if (toolkit_views) {
+  if (is_android || toolkit_views) {
     deps += [ "//ui/views/resources" ]
     sources +=
         [ "$root_gen_dir/ui/views/resources/views_resources_200_percent.pak" ]
