diff --git a/ui/views/BUILD.gn b/ui/views/BUILD.gn
index b2b1a783ac0c243ef5cf573b2e73af82b432fbf9..ee2563444d95c5168c2c7aff9eba9a718df8fd2e 100644
--- a/ui/views/BUILD.gn
+++ b/ui/views/BUILD.gn
@@ -15,10 +15,12 @@ import("//testing/test.gni")
 import("//ui/base/ui_features.gni")
 import("//ui/views/features.gni")
 
-assert(toolkit_views)
+#assert(toolkit_views)
 
 config("flags") {
-  defines = [ "TOOLKIT_VIEWS=1" ]
+  if (toolkit_views) {
+    defines = [ "TOOLKIT_VIEWS=1" ]
+  }
 }
 
 aggregate_vector_icons("views_vector_icons") {
@@ -770,6 +772,8 @@ component("views") {
       "//third_party/wtl",
     ]
   }
+ 
+  import("//mises/ui/views/sources.gni") public += mises_ui_view_public  sources += mises_ui_view_sources sources -= mises_ui_view_exception_sources
 
   if (use_aura) {
     public += [
