diff --git a/ui/views/widget/widget.cc b/ui/views/widget/widget.cc
index 968eebaf6719c1da9a796588f487a6ebf0aab876..a12cda2b2b3a319e232f44ffcc30e433b5ec801c 100644
--- a/ui/views/widget/widget.cc
+++ b/ui/views/widget/widget.cc
@@ -438,11 +438,13 @@ void Widget::Init(InitParams params) {
   root_view_.reset(CreateRootView());
 
   // Copy the elements of params that will be used after it is moved.
+#if !BUILDFLAG(IS_ANDROID)
   const InitParams::Type type = params.type;
   const gfx::Rect bounds = params.bounds;
   const ui::WindowShowState show_state = params.show_state;
-  WidgetDelegate* delegate = params.delegate;
   bool should_set_initial_bounds = true;
+#endif
+  WidgetDelegate* delegate = params.delegate;
 #if BUILDFLAG(IS_CHROMEOS)
   // If the target display is specified on ChromeOS, the initial bounds will be
   // set based on the display.
@@ -450,6 +452,7 @@ void Widget::Init(InitParams params) {
 #endif
 
   native_widget_->InitNativeWidget(std::move(params));
+#if !BUILDFLAG(IS_ANDROID)
   if (type == InitParams::TYPE_MENU)
     is_mouse_button_pressed_ = native_widget_->IsMouseButtonDown();
   if (RequiresNonClientView(type)) {
@@ -497,6 +500,7 @@ void Widget::Init(InitParams params) {
   }
 
   native_theme_observation_.Observe(GetNativeTheme());
+#endif
   native_widget_initialized_ = true;
   native_widget_->OnWidgetInitDone();
 
@@ -2062,11 +2066,13 @@ void Widget::OnDragWillStart() {}
 void Widget::OnDragComplete() {}
 
 const ui::NativeTheme* Widget::GetNativeTheme() const {
+#if !BUILDFLAG(IS_ANDROID)
   if (native_theme_)
     return native_theme_;
 
   if (parent_)
     return parent_->GetNativeTheme();
+#endif
 
 #if BUILDFLAG(IS_LINUX)
   if (auto* linux_ui_theme = ui::LinuxUiTheme::GetForWindow(GetNativeWindow()))
