diff --git a/ui/webui/resources/cr_components/most_visited/BUILD.gn b/ui/webui/resources/cr_components/most_visited/BUILD.gn
index cf37adc2a5ca98ab404c9be3af8e94ace89a269e..2ac907fe9c724138cdc2cab1546bea419196397c 100644
--- a/ui/webui/resources/cr_components/most_visited/BUILD.gn
+++ b/ui/webui/resources/cr_components/most_visited/BUILD.gn
@@ -8,7 +8,7 @@ import("//tools/typescript/ts_library.gni")
 import("//ui/webui/resources/tools/generate_grd.gni")
 import("most_visited.gni")
 
-assert(!is_android && !is_ios)
+#assert(!is_android && !is_ios)
 
 preprocess_folder =
     "$root_gen_dir/ui/webui/resources/preprocessed/cr_components/most_visited"
