diff --git a/ui/webui/resources/js/BUILD.gn b/ui/webui/resources/js/BUILD.gn
index c6679f4f34e6bb11f45282e787e446639030982c..b978a7e82b1aec07d8c483057c780872a43d0c58 100644
--- a/ui/webui/resources/js/BUILD.gn
+++ b/ui/webui/resources/js/BUILD.gn
@@ -5,7 +5,7 @@
 import("//build/config/chromeos/ui_mode.gni")
 import("../tools/build_webui.gni")
 
-include_polymer = !is_android && !is_ios
+include_polymer = is_android || !is_android && !is_ios
 
 build_webui("build") {
   grd_prefix = "webui_js"
