diff --git a/ui/webui/resources/images/BUILD.gn b/ui/webui/resources/images/BUILD.gn
index 4f06ce4a6e7f9b997112aa39dae552bcbd6dcb3e..c3787caf48a6c4f266403e83278f9792ce61bc34 100644
--- a/ui/webui/resources/images/BUILD.gn
+++ b/ui/webui/resources/images/BUILD.gn
@@ -48,7 +48,7 @@ generate_grd("build_grdp") {
     "2x/select.png",
   ]
 
-  if (!is_android && !is_ios) {
+  if (!is_ios) {
     input_files += [
       "arrow_down.svg",
       "arrow_right.svg",
