diff --git a/ui/webui/resources/cr_components/app_management/BUILD.gn b/ui/webui/resources/cr_components/app_management/BUILD.gn
index 45328d1f2312cf7d4559c0c0c8c761167dde3241..1d74d3b007a08c2f2b757bd1bd92cb818a46bdbb 100644
--- a/ui/webui/resources/cr_components/app_management/BUILD.gn
+++ b/ui/webui/resources/cr_components/app_management/BUILD.gn
@@ -5,7 +5,7 @@
 import("//mojo/public/tools/bindings/mojom.gni")
 import("//ui/webui/resources/tools/build_webui.gni")
 
-assert(!is_android && !is_ios)
+# assert(!is_android && !is_ios)
 
 mojom("mojo_bindings") {
   sources = [ "app_management.mojom" ]
