diff --git a/ui/webui/resources/cr_components/history_clusters/BUILD.gn b/ui/webui/resources/cr_components/history_clusters/BUILD.gn
index 93ec5c64e0a610aaa54802582058211f4773fd7d..64a393cd79acb1f4d04b23b6cbd19c05bc44c999 100644
--- a/ui/webui/resources/cr_components/history_clusters/BUILD.gn
+++ b/ui/webui/resources/cr_components/history_clusters/BUILD.gn
@@ -6,7 +6,7 @@ import("//mojo/public/tools/bindings/mojom.gni")
 import("//ui/webui/webui_features.gni")
 import("../../tools/build_cr_component.gni")
 
-assert(!is_android)
+# assert(!is_android)
 
 mojom("mojo_bindings") {
   sources = [ "history_clusters.mojom" ]
