diff --git a/ui/webui/resources/cr_components/color_change_listener/BUILD.gn b/ui/webui/resources/cr_components/color_change_listener/BUILD.gn
index 5ff8c3e1f67380c270acce9b6f8b44a79ab41ec0..ffce833cbd696a6a60628909fefdbaf20a674e3b 100644
--- a/ui/webui/resources/cr_components/color_change_listener/BUILD.gn
+++ b/ui/webui/resources/cr_components/color_change_listener/BUILD.gn
@@ -5,7 +5,7 @@
 import("//mojo/public/tools/bindings/mojom.gni")
 import("//ui/webui/resources/tools/build_webui.gni")
 
-assert(!is_android && !is_ios)
+# assert(!is_android && !is_ios)
 
 mojom("mojom") {
   sources = [ "color_change_listener.mojom" ]
