diff --git a/ui/webui/resources/css/BUILD.gn b/ui/webui/resources/css/BUILD.gn
index 816f808edea9e67fee7404a6e905da979d1aff01..cc075da7e7f344923827e5ff2b748a50e9bb3561 100644
--- a/ui/webui/resources/css/BUILD.gn
+++ b/ui/webui/resources/css/BUILD.gn
@@ -5,7 +5,7 @@
 import("//tools/grit/preprocess_if_expr.gni")
 import("//ui/webui/resources/tools/generate_grd.gni")
 
-include_polymer = !is_android && !is_ios
+include_polymer = true || !is_android && !is_ios
 
 preprocess_manifest = "preprocessed_manifest.json"
 
