diff --git a/ui/webui/resources/js/BUILD.gn b/ui/webui/resources/js/BUILD.gn
index acefb77255b811934acb11a0f16aca1d5347d731..ed129dbb3448c6a75c30458d5b7a524c6706644a 100644
--- a/ui/webui/resources/js/BUILD.gn
+++ b/ui/webui/resources/js/BUILD.gn
@@ -5,7 +5,7 @@
 import("//build/config/chromeos/ui_mode.gni")
 import("../tools/build_cr_component.gni")
 
-include_polymer = !is_android && !is_ios
+include_polymer = is_android || !is_android && !is_ios
 
 build_cr_component("build") {
   grd_prefix = "webui_js"
