diff --git a/ui/webui/resources/cr_components/color_change_listener/BUILD.gn b/ui/webui/resources/cr_components/color_change_listener/BUILD.gn
index 1cf9c1ca2ef35ec93149d5b01df719c282f59229..1f6877cc5f43c210d39efd0791ba300a2ca47bc4 100644
--- a/ui/webui/resources/cr_components/color_change_listener/BUILD.gn
+++ b/ui/webui/resources/cr_components/color_change_listener/BUILD.gn
@@ -5,7 +5,7 @@
 import("//mojo/public/tools/bindings/mojom.gni")
 import("//ui/webui/resources/tools/build_webui.gni")
 
-assert(!is_android && !is_ios)
+# assert(!is_android && !is_ios)
 
 mojom("mojom") {
   sources = [ "color_change_listener.mojom" ]
