diff --git a/ui/webui/resources/css/BUILD.gn b/ui/webui/resources/css/BUILD.gn
index e734c2497f9e832f6a8a43f1e5903cfe35b84baf..9878b5264c76430ebde8764a5640d9be5fa299b9 100644
--- a/ui/webui/resources/css/BUILD.gn
+++ b/ui/webui/resources/css/BUILD.gn
@@ -5,7 +5,7 @@
 import("//tools/grit/preprocess_if_expr.gni")
 import("//ui/webui/resources/tools/generate_grd.gni")
 
-include_polymer = !is_android && !is_ios
+include_polymer = is_android || !is_android && !is_ios
 
 preprocess_manifest = "preprocessed_manifest.json"
 
