diff --git a/ui/webui/resources/images/BUILD.gn b/ui/webui/resources/images/BUILD.gn
index 44bad8bc0910080860b0ddb750cbf03c09e15296..21d9febde3bc61a09d80c6b5b2692624f2402938 100644
--- a/ui/webui/resources/images/BUILD.gn
+++ b/ui/webui/resources/images/BUILD.gn
@@ -45,7 +45,7 @@ generate_grd("build_grdp") {
     "2x/select.png",
   ]
 
-  if (!is_android && !is_ios) {
+  if (!is_ios) {
     input_files += [
       "arrow_down.svg",
       "arrow_right.svg",
