diff --git a/ui/webui/resources/cr_components/customize_themes/BUILD.gn b/ui/webui/resources/cr_components/customize_themes/BUILD.gn
index d5f209441d3d320cfffb3d166f90cffb160e1063..de6d395dbd1909947b7a3afbf37c1cdb16bc054a 100644
--- a/ui/webui/resources/cr_components/customize_themes/BUILD.gn
+++ b/ui/webui/resources/cr_components/customize_themes/BUILD.gn
@@ -6,7 +6,7 @@ import("//mojo/public/tools/bindings/mojom.gni")
 import("//ui/webui/webui_features.gni")
 import("../../tools/build_cr_component.gni")
 
-assert(!is_android && !is_ios)
+#assert(!is_android && !is_ios)
 
 mojom("mojom") {
   sources = [ "customize_themes.mojom" ]
