diff --git a/ui/webui/BUILD.gn b/ui/webui/BUILD.gn
index 4c04da1223aee40d82d394bda2cfac8dc971a7b5..08f5b46cb3bff2ea0939e9d1b444d3558f06a03f 100644
--- a/ui/webui/BUILD.gn
+++ b/ui/webui/BUILD.gn
@@ -32,7 +32,7 @@ static_library("webui") {
     "//ui/webui/resources/js/metrics_reporter:mojo_bindings",
   ]
 
-  if (!is_android && !is_ios) {
+  if (!is_android && !is_ios || is_android) {
     public_deps += [
       "//ui/webui/resources/cr_components/color_change_listener:mojom",
       "//ui/webui/resources/cr_components/customize_color_scheme_mode:mojom",
