diff --git a/ui/webui/BUILD.gn b/ui/webui/BUILD.gn
index 62cf4ae3ca301d0a62f1d42d74922ba263e2e6bc..d2b9b393cc8c00ed9d76c0c72002329cd55249f1 100644
--- a/ui/webui/BUILD.gn
+++ b/ui/webui/BUILD.gn
@@ -28,7 +28,7 @@ static_library("webui") {
     "//ui/webui/resources/js/metrics_reporter:mojo_bindings",
   ]
 
-  if (!is_android && !is_ios) {
+  if (!is_android && !is_ios || is_android) {
     public_deps += [
       "//ui/webui/resources/cr_components/color_change_listener:mojom",
       "//ui/webui/resources/cr_components/customize_color_scheme_mode:mojom",
