diff --git a/ui/webui/resources/include_polymer.gni b/ui/webui/resources/include_polymer.gni
index 52daa91cc2319a470692a1d5f24b904842123dcb..b6580b79be9bb72d430c7778399f92bd852325f1 100644
--- a/ui/webui/resources/include_polymer.gni
+++ b/ui/webui/resources/include_polymer.gni
@@ -2,4 +2,4 @@
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
-include_polymer = !is_android && !is_ios
+include_polymer = true || !is_android && !is_ios
