diff --git a/ui/webui/resources/BUILD.gn b/ui/webui/resources/BUILD.gn
index 483defefa69d971e6c8f369263c445ac248297f5..0b32150ee1fed16304b6c37aab7a0b26b47a0291 100644
--- a/ui/webui/resources/BUILD.gn
+++ b/ui/webui/resources/BUILD.gn
@@ -35,7 +35,7 @@ generate_grd("build_grd") {
     "$root_gen_dir/third_party/jstemplate/resources.grdp",
   ]
 
-  if (!is_chromeos_ash && !is_android) {
+  if (!is_chromeos_ash) {
     # Roboto Font. Roboto-Regular and Roboto-Light is already available on
     # Android, and Roboto-Medium is not used on Android. All 6 weights of
     # Roboto are available on Chrome OS.
@@ -47,7 +47,7 @@ generate_grd("build_grd") {
     ]
   }
 
-  if (include_polymer) {
+  if (include_polymer || is_android) {
     public_deps += [
       "cr_components:build_grdp",
       "cr_components/app_management:build_grdp",
@@ -106,7 +106,7 @@ generate_grd("build_grd") {
     ]
   }
 
-  if (!is_android && !is_ios) {
+  if (!is_android && !is_ios || is_android) {
     public_deps += [
       "cr_components/color_change_listener:build_grdp",
       "//third_party/lottie:build_grdp",
@@ -147,7 +147,7 @@ group("closure_compile") {
     "js:closure_compile_modules",
   ]
 
-  if (include_polymer) {
+  if (include_polymer || is_android) {
     deps += [ "cr_elements:closure_compile" ]
 
     if (is_chromeos_ash) {
@@ -156,6 +156,14 @@ group("closure_compile") {
   }
 }
 
+group("modulize") {
+  public_deps = [
+    "js:modulize",
+    "js/cr:modulize",
+    "js/cr/ui:modulize",
+  ]
+}
+
 # TypeScript targets
 
 preprocessed_folder = "$target_gen_dir/preprocessed"
@@ -307,7 +315,7 @@ if (include_polymer) {
   }
 }
 
-if (!is_android) {
+if (true || !is_android) {
   generate_definitions_js_files += [ "js/cr/ui/focus_without_ink.m.js" ]
 }
 
