diff --git a/ui/webui/resources/cr_components/page_image_service/BUILD.gn b/ui/webui/resources/cr_components/page_image_service/BUILD.gn
index 931719eb54c916411f8fcff941b2e7874a66e36a..2e74ccbc168b0d21231697f6a56215a12153182f 100644
--- a/ui/webui/resources/cr_components/page_image_service/BUILD.gn
+++ b/ui/webui/resources/cr_components/page_image_service/BUILD.gn
@@ -4,7 +4,7 @@
 
 import("//ui/webui/resources/tools/build_webui.gni")
 
-assert(!is_android && !is_ios)
+#assert(!is_android && !is_ios)
 
 build_webui("build") {
   grd_prefix = "cr_components_page_image_service"
