diff --git a/ui/webui/resources/cr_components/image_service/BUILD.gn b/ui/webui/resources/cr_components/image_service/BUILD.gn
index 98e36fc2626988a824ffc0b8f326af7f080d18f9..26f7d95f3accda501dbb89ae797637e9c8e0d5b5 100644
--- a/ui/webui/resources/cr_components/image_service/BUILD.gn
+++ b/ui/webui/resources/cr_components/image_service/BUILD.gn
@@ -4,7 +4,7 @@
 
 import("../../tools/build_cr_component.gni")
 
-assert(!is_android && !is_ios)
+# assert(!is_android && !is_ios)
 
 build_cr_component("build") {
   grd_prefix = "cr_components_image_service"
