diff --git a/ui/webui/resources/BUILD.gn b/ui/webui/resources/BUILD.gn
index b9bac52390e72642c485fc9e82d82a02b5b97894..ef6443c87457582ae554de0c6ebd38ca1ff36d0e 100644
--- a/ui/webui/resources/BUILD.gn
+++ b/ui/webui/resources/BUILD.gn
@@ -14,7 +14,7 @@ generate_grd("build_grd") {
   input_files = [ "test_loader.html" ]
   input_files_base_dir = rebase_path(".", "//")
 
-  if (!is_chromeos_ash && !is_android) {
+  if (!is_chromeos_ash) {
     # Roboto Font. Roboto-Regular and Roboto-Light is already available on
     # Android, and Roboto-Medium is not used on Android. All 6 weights of
     # Roboto are available on Chrome OS.
@@ -40,7 +40,7 @@ generate_grd("build_grd") {
     "$target_gen_dir/js/resources.grdp",
   ]
 
-  if (include_polymer) {
+  if (include_polymer || is_android) {
     public_deps += [
       "cr_components/app_management:build_grdp",
       "cr_components/customize_themes:build_grdp",
@@ -109,7 +109,7 @@ generate_grd("build_grd") {
     ]
   }
 
-  if (!is_android && !is_ios) {
+  if (!is_android && !is_ios || is_android) {
     public_deps += [
       "cr_components/color_change_listener:build_grdp",
       "cr_components/image_service:build_grdp",
