diff --git a/ui/webui/resources/cr_components/BUILD.gn b/ui/webui/resources/cr_components/BUILD.gn
index 5f585069f27e9b3037ab1a961daf653edd662406..fcf08f91aca74d0ccb88a643b5f9269d66dc37c3 100644
--- a/ui/webui/resources/cr_components/BUILD.gn
+++ b/ui/webui/resources/cr_components/BUILD.gn
@@ -9,7 +9,7 @@ import("//tools/polymer/html_to_wrapper.gni")
 import("//ui/webui/resources/tools/generate_grd.gni")
 import("//ui/webui/webui_features.gni")
 
-assert(!is_android && !is_ios)
+#assert(!is_android && !is_ios)
 
 preprocess_folder =
     "$root_gen_dir/ui/webui/resources/preprocessed/cr_components"
