diff --git a/ui/webui/resources/cr_components/history_clusters/BUILD.gn b/ui/webui/resources/cr_components/history_clusters/BUILD.gn
index 8fcbccdfa705f910aa33d7a32a5df99fc9e5a158..0588db983ed811533a47b5ef129d7a63cd8535ff 100644
--- a/ui/webui/resources/cr_components/history_clusters/BUILD.gn
+++ b/ui/webui/resources/cr_components/history_clusters/BUILD.gn
@@ -6,7 +6,7 @@ import("//mojo/public/tools/bindings/mojom.gni")
 import("//ui/webui/webui_features.gni")
 import("../../tools/build_cr_component.gni")
 
-assert(!is_android)
+# assert(!is_android)
 
 mojom("mojo_bindings") {
   sources = [ "history_clusters.mojom" ]
