diff --git a/ui/webui/resources/cr_components/most_visited/BUILD.gn b/ui/webui/resources/cr_components/most_visited/BUILD.gn
index ff6eb7acce643597c62613c6dc639b64c64c68d7..88856721ffd370784c96e6901f65f92e17818813 100644
--- a/ui/webui/resources/cr_components/most_visited/BUILD.gn
+++ b/ui/webui/resources/cr_components/most_visited/BUILD.gn
@@ -6,7 +6,7 @@ import("//mojo/public/tools/bindings/mojom.gni")
 import("//ui/webui/webui_features.gni")
 import("../../tools/build_cr_component.gni")
 
-assert(!is_android && !is_ios)
+#assert(!is_android && !is_ios)
 
 mojom("mojom") {
   sources = [ "most_visited.mojom" ]
