diff --git a/ui/webui/resources/cr_components/help_bubble/BUILD.gn b/ui/webui/resources/cr_components/help_bubble/BUILD.gn
index cbe4967bb8cc045c8e49cced9d7e6c646bea8cb1..411edd544eb49719cc2b02a5106870a32034db22 100644
--- a/ui/webui/resources/cr_components/help_bubble/BUILD.gn
+++ b/ui/webui/resources/cr_components/help_bubble/BUILD.gn
@@ -9,7 +9,7 @@ import("//tools/typescript/ts_library.gni")
 import("//ui/webui/resources/tools/generate_grd.gni")
 import("help_bubble.gni")
 
-assert(!is_android && !is_ios)
+#assert(!is_android && !is_ios)
 
 mojom("mojo_bindings") {
   sources = [ "help_bubble.mojom" ]
