diff --git a/ui/webui/BUILD.gn b/ui/webui/BUILD.gn
index 8f400abfbe4645a20cbc6f8b68ecc67173172343..ab30edcbf6941989606501f5e4afd100b58d8c32 100644
--- a/ui/webui/BUILD.gn
+++ b/ui/webui/BUILD.gn
@@ -32,7 +32,7 @@ static_library("webui") {
     "//ui/webui/resources/js/metrics_reporter:mojo_bindings",
   ]
 
-  if (!is_android && !is_ios) {
+  if (!is_android && !is_ios || is_android) {
     public_deps += [
       "//ui/webui/resources/cr_components/color_change_listener:mojom",
       "//ui/webui/resources/cr_components/customize_themes:mojom",
