diff --git a/ui/webui/resources/cr_components/app_management/BUILD.gn b/ui/webui/resources/cr_components/app_management/BUILD.gn
index ee76641d7374139d96b0a79a029eb6cf0e5958a6..4d5d9bd0cfabd3774591a12a2cfe0dd42ed55383 100644
--- a/ui/webui/resources/cr_components/app_management/BUILD.gn
+++ b/ui/webui/resources/cr_components/app_management/BUILD.gn
@@ -6,7 +6,7 @@ import("//mojo/public/tools/bindings/mojom.gni")
 import("//ui/webui/webui_features.gni")
 import("../../tools/build_cr_component.gni")
 
-assert(!is_android && !is_ios)
+# assert(!is_android && !is_ios)
 
 mojom("mojo_bindings") {
   sources = [ "app_management.mojom" ]
