diff --git a/ui/webui/resources/BUILD.gn b/ui/webui/resources/BUILD.gn
index ed6eb9641ff5bb3a25d475929daf04811c10c009..f9ea524fa2843b198814b0cdf45ee8dc8df0cb56 100644
--- a/ui/webui/resources/BUILD.gn
+++ b/ui/webui/resources/BUILD.gn
@@ -15,7 +15,7 @@ generate_grd("build_grd") {
   input_files = [ "test_loader.html" ]
   input_files_base_dir = rebase_path(".", "//")
 
-  if (!is_chromeos_ash && !is_android) {
+  if (!is_chromeos_ash) {
     # Roboto Font. Roboto-Regular and Roboto-Light is already available on
     # Android, and Roboto-Medium is not used on Android. All 6 weights of
     # Roboto are available on Chrome OS.
@@ -43,7 +43,7 @@ generate_grd("build_grd") {
     "$target_gen_dir/js/resources.grdp",
   ]
 
-  if (include_polymer) {
+  if (include_polymer || is_android) {
     public_deps += [
       "cr_components/app_management:build_grdp",
       "cr_components/customize_themes:build_grdp",
@@ -110,7 +110,7 @@ generate_grd("build_grd") {
     ]
   }
 
-  if (!is_android && !is_ios) {
+  if (!is_android && !is_ios || is_android) {
     public_deps += [
       "cr_components/color_change_listener:build_grdp",
       "//third_party/lottie:build_grdp",
@@ -134,6 +134,14 @@ if (is_chromeos_ash) {
   }
 }
 
+group("modulize") {
+  public_deps = [
+    "js:modulize",
+    "js/cr:modulize",
+    "js/cr/ui:modulize",
+  ]
+}
+
 # TypeScript targets
 
 # TODO(crbug/1402829): This target has no input files and just exists to allow
