diff --git a/ui/webui/resources/cr_components/omnibox/BUILD.gn b/ui/webui/resources/cr_components/omnibox/BUILD.gn
index a546b1c3db7c24a7af9db801f727cd4833ba11e3..ad329810edb4dadeab627c45ca70e02f0d6a1c36 100644
--- a/ui/webui/resources/cr_components/omnibox/BUILD.gn
+++ b/ui/webui/resources/cr_components/omnibox/BUILD.gn
@@ -4,7 +4,7 @@
 
 import("../../tools/build_cr_component.gni")
 
-assert(!is_android && !is_ios)
+#assert(!is_android && !is_ios)
 
 build_cr_component("build") {
   grd_prefix = "cr_components_omnibox"
