diff --git a/url/url_canon_ip.cc b/url/url_canon_ip.cc
index f79a03ace424a6ae61d89dcf6df1747f83e4d1da..b53044a1c48da8e94cd6cbd790134496eda11984 100644
--- a/url/url_canon_ip.cc
+++ b/url/url_canon_ip.cc
@@ -164,6 +164,10 @@ CanonHostInfo::Family DoIPv4AddressToNumber(const CHAR* spec,
     // BROKEN IPv4 address, as opposed to a non-IPv4 hostname.
     if (family == CanonHostInfo::NEUTRAL && existing_components == 0)
       return CanonHostInfo::NEUTRAL;
+      
+    // fix for .888 domain
+    if (family == CanonHostInfo::NEUTRAL && existing_components == 1 && component_values[0] == 888)
+        return CanonHostInfo::NEUTRAL;
 
     if (family != CanonHostInfo::IPV4)
       return CanonHostInfo::BROKEN;
