diff --git a/url/url_canon_ip.cc b/url/url_canon_ip.cc
index 069e806f4269bbab8d29ffab0879cd8e85773b1c..f50edb3925f5655f43fbf563045cf9c624aa8c4d 100644
--- a/url/url_canon_ip.cc
+++ b/url/url_canon_ip.cc
@@ -164,6 +164,10 @@ CanonHostInfo::Family DoIPv4AddressToNumber(const CHAR* spec,
     // BROKEN IPv4 address, as opposed to a non-IPv4 hostname.
     if (family == CanonHostInfo::NEUTRAL && existing_components == 0)
       return CanonHostInfo::NEUTRAL;
+      
+    // fix for .888 domain
+    if (family == CanonHostInfo::NEUTRAL && existing_components == 1 && component_values[0] == 888)
+        return CanonHostInfo::NEUTRAL;
 
     if (family != CanonHostInfo::IPV4)
       return CanonHostInfo::BROKEN;
