diff --git a/weblayer/BUILD.gn b/weblayer/BUILD.gn
index 3e43348e924bb966d6736206942afa69d366599b..63b4a38915993e832b1a03958f48c90dea36db19 100644
--- a/weblayer/BUILD.gn
+++ b/weblayer/BUILD.gn
@@ -851,6 +851,7 @@ if (is_android) {
       "//base",
       "//weblayer/browser/java:jni",
       "//weblayer/browser/java:weblayer_jni_registration($default_toolchain)",
+      "//chrome/browser:browser"
     ]
     sources = [
       "browser/web_view_compatibility_helper_impl.cc",
