diff --git a/chrome/app/generated_resources.grd b/chrome/app/generated_resources.grd
index 45bc795e9c51db1352de2630b19fb94fd4bc6260..b4ff76a8af6abcdfe67addddac87e622bcba5a4c 100644
--- a/chrome/app/generated_resources.grd
+++ b/chrome/app/generated_resources.grd
@@ -306,7 +306,7 @@ are declared in tools/grit/grit_rule.gni.
       </if>
 
       <!-- Settings specific strings -->
-      <if expr="not is_android">
+      <if expr="is_android or not is_android">
         <part file="settings_strings.grdp" />
         <part file="shared_settings_strings.grdp"/>
       </if>
