diff --git a/chrome/browser/ui/android/strings/android_chrome_strings.grd b/chrome/browser/ui/android/strings/android_chrome_strings.grd
index cd320c39ca381d190c7b18c5ed9b48d50a82b5b9..b3c21853e3c780c170ea7b29954a472f60981219 100644
--- a/chrome/browser/ui/android/strings/android_chrome_strings.grd
+++ b/chrome/browser/ui/android/strings/android_chrome_strings.grd
@@ -3588,6 +3588,9 @@ To change this setting, <ph name="BEGIN_LINK">&lt;resetlink&gt;</ph>reset sync<p
       <message name="IDS_MENU_STOP_REFRESH" desc="Tooltip on Android for the button to stop refreshing the current web page. [CHAR_LIMIT=27]">
         Stop refreshing
       </message>
+      <message name="IDS_MENU_MISES_FORWARD" desc="Menu item for share to mises. [CHAR_LIMIT=27]">
+        Share to Discover
+      </message>
       <message name="IDS_MENU_NEW_TAB" desc="Menu item for opening a new tab. [CHAR_LIMIT=27]">
         New tab
       </message>
