diff --git a/components/page_info_strings.grdp b/components/page_info_strings.grdp
index c78bc1cca713d70368adbbc3ce46defdddf5ee5a..6672337950a42645c6376d7ad5dcc13c1be644b2 100644
--- a/components/page_info_strings.grdp
+++ b/components/page_info_strings.grdp
@@ -216,7 +216,7 @@
     </message>
 
     <!-- Certificate Viewer link -->
-    <if expr="not is_android">
+    <if expr="not is_android or is_android">
       <message name="IDS_PAGE_INFO_CERTIFICATE" desc="Title of the certificate area in the Page Info bubble, shown when a HTTPS site is loaded.">
         Certificate
       </message>
@@ -258,7 +258,7 @@
     </if>
 
     <!-- Cookies -->
-    <if expr="not is_android">
+    <if expr="not is_android or is_android">
       <message name="IDS_PAGE_INFO_COOKIES" desc="The label for the Cookies setting in the Page Information Window.">
         Cookies
       </message>
@@ -531,7 +531,7 @@
     </message>
 
     <!-- Permission change infobar. -->
-    <if expr="not is_android">
+    <if expr="not is_android or is_android">
       <message name="IDS_PAGE_INFO_INFOBAR_TEXT" desc="The string shown in the infobar after the user has changed site permissions settings, reminding them to reload the page in order for the new settings to take effect.  Deliberately similar to IDS_EXTENSION_SITE_RELOAD_PAGE_BUBBLE_HEADING, but used for an different permission setting so they are not combined. If updating the one, please update the other or consult the extensions and privacy teams.">
         Reload this page to apply your updated settings on this site
       </message>
