diff --git a/components/page_info_strings.grdp b/components/page_info_strings.grdp
index 0feea7dc66d0f..90de8e2126954 100644
--- a/components/page_info_strings.grdp
+++ b/components/page_info_strings.grdp
@@ -230,7 +230,7 @@
     </message>
 
     <!-- Certificate Viewer link -->
-    <if expr="not is_android">
+    <if expr="not is_android or is_android">
       <message name="IDS_PAGE_INFO_CERTIFICATE" desc="Title of the certificate area in the Page Info bubble, shown when a HTTPS site is loaded.">
         Certificate
       </message>
@@ -272,7 +272,7 @@
     </if>
 
     <!-- Cookies -->
-    <if expr="not is_android">
+    <if expr="not is_android or is_android">
       <message name="IDS_PAGE_INFO_COOKIES" desc="The label for the Cookies setting in the Page Information Window.">
         Cookies
       </message>
@@ -530,7 +530,7 @@
     </message>
 
     <!-- Permission change infobar. -->
-    <if expr="not is_android">
+    <if expr="not is_android or is_android">
       <message name="IDS_PAGE_INFO_INFOBAR_TEXT" desc="The string shown in the infobar after the user has changed site permissions settings, reminding them to reload the page in order for the new settings to take effect.">
         To apply your updated settings to this site, reload this page
       </message>
