diff --git a/components/resources/components_scaled_resources.grd b/components/resources/components_scaled_resources.grd
index 939035cad76715757789f1a8878b073522d2d361..4f377061cbaef94a0ea352d753a1fe11415fed16 100644
--- a/components/resources/components_scaled_resources.grd
+++ b/components/resources/components_scaled_resources.grd
@@ -24,7 +24,7 @@
       </if>
 
       <!-- Generic resources -->
-      <if expr="not is_android">
+      <if expr="is_android or not is_android">
         <structure type="chrome_scaled_image" name="IDR_HISTORY_FAVICON" file="favicon_history.png" />
       </if>
       <structure type="chrome_scaled_image" name="IDR_SAD_WEBVIEW" file="webview-crash.png" />
