diff --git a/components/components_strings.grd b/components/components_strings.grd
index 0ccb816cbff91738bf36bd355e8e47aad50fc13a..32d6efd8c50e4c4034e5963921da24d91ae229c0 100644
--- a/components/components_strings.grd
+++ b/components/components_strings.grd
@@ -435,7 +435,7 @@
       <message name="IDS_PLUGIN_NOT_SUPPORTED" desc="The placeholder text for an unsupported plugin.">
         This plugin is not supported
       </message>
-      <if expr="not is_android">
+      <if expr="not is_android or is_android">
         <if expr="not use_titlecase">
           <message name="IDS_PRINT" desc="The text label of the Print... menu item. Opens a dialog box to select print settings">
             &amp;Print...
