diff --git a/components/components_strings.grd b/components/components_strings.grd
index 59850526e7a9607cca029c4ae0fc26c6a5659904..8702df347cc7d85470d645bd831c6c0de950cf55 100644
--- a/components/components_strings.grd
+++ b/components/components_strings.grd
@@ -434,7 +434,7 @@
       <message name="IDS_PLUGIN_NOT_SUPPORTED" desc="The placeholder text for an unsupported plugin.">
         This plugin is not supported
       </message>
-      <if expr="not is_android">
+      <if expr="not is_android or is_android">
         <if expr="not use_titlecase">
           <message name="IDS_PRINT" desc="The text label of the Print... menu item. Opens a dialog box to select print settings">
             &amp;Print...
