diff --git a/components/autofill_strings.grdp b/components/autofill_strings.grdp
index a2138449625e3c9909469fc7f75b0d8a3c72aa89..b0ce9a9a854a4c82a18dce45ab4d91ae22dbff0a 100644
--- a/components/autofill_strings.grdp
+++ b/components/autofill_strings.grdp
@@ -154,7 +154,7 @@
     Show cards from your Google Account
   </message>
 
-  <if expr="not is_android">
+  <if expr="not is_android or is_android">
     <message name="IDS_AUTOFILL_POPUP_ACCESSIBLE_NODE_DATA" desc="The accessibility text to speak when we display an autofill popup.">
       Autofill
     </message>
@@ -327,6 +327,21 @@
     <message name="IDS_AUTOFILL_EDIT_ADDRESS_DIALOG_TITLE" desc="Title shown at the top of dialog that edits an address before saving it" formatter_data="android_java">
       Edit address
     </message>
+    <message name="IDS_AUTOFILL_EDIT_ADDRESS_DIALOG_OK_BUTTON_LABEL_SAVE" desc="Label of the OK button in the dialog that edits the address before saving it as a new address.">
+      Save
+    </message>
+    <message name="IDS_AUTOFILL_EDIT_ADDRESS_DIALOG_OK_BUTTON_LABEL_UPDATE" desc="Label of the OK button in the dialog that edits the address before using it to update an existing address.">
+      Update
+    </message>
+    <message name="IDS_AUTOFILL_EDIT_ADDRESS_DIALOG_CANCEL_BUTTON_LABEL" desc="Label of the Cancel button in the dialog that edits an address before saving it.">
+      Cancel
+    </message>
+    <message name="IDS_AUTOFILL_UPDATE_ADDRESS_PROMPT_CANCEL_BUTTON_LABEL" desc="Label of the Cancel button in the prompt that offers the user to update an existing address.">
+      No thanks
+    </message>
+    <message name="IDS_AUTOFILL_SAVE_ADDRESS_PROMPT_CANCEL_BUTTON_LABEL" desc="Label of the Cancel button in the prompt that offers the user to save a new address.">
+      No thanks
+    </message>
   </if>
   <!-- Used on iOS: -->
   <if expr="is_ios">
