diff --git a/components/new_or_sad_tab_strings.grdp b/components/new_or_sad_tab_strings.grdp
index 6b1088e1e0b36c2a724dbadc3c1d92ca91be0dcb..a6de7a7cd9c3c66737c97527f0badd1ce2f4897e 100644
--- a/components/new_or_sad_tab_strings.grdp
+++ b/components/new_or_sad_tab_strings.grdp
@@ -91,7 +91,7 @@
           Restart Chromium
         </message>
       </if>
-      <if expr="is_android or is_ios">
+      <if expr="not is_android or is_android or is_ios">
 	<then>
           <message name="IDS_SAD_TAB_RELOAD_RESTART_DEVICE" desc="One of the bullet points displayed on the web page if a reload failed to fix the issue, advising the user to restart their device (i.e. phone or tablet)." formatter_data="android_java">
             Restart your device
@@ -99,6 +99,9 @@
           <message name="IDS_SAD_TAB_RELOAD_LEARN_MORE" desc="The link text displayed on the sad tab page pointing the users to a help article if a reload failed to fix the issue." meaning="Learn more about web pages that fail to load" formatter_data="android_java">
             Learn more
           </message>
+          <message name="IDS_SAD_TAB_ERROR_CODE" desc="The message displayed on the crashed web page indicating the type of the crash.">
+            Error code: <ph name="ERROR_CODE">$1<ex>STATUS_ACCESS_VIOLATION</ex></ph>
+          </message>
 	</then>
 	<else>
           <message name="IDS_SAD_TAB_RELOAD_RESTART_DEVICE" desc="One of the bullet points displayed on the web page if a reload failed to fix the issue, advising the user to restart their computer.">
@@ -123,7 +126,7 @@
           desc="Accessibility label for a link labeled 'Learn more' that links to a help article about Incognito mode.">
         Learn more about Incognito
       </message>
-      <if expr="not is_android">
+      <if expr="not is_android or is_android">
         <message name="IDS_NEW_TAB_OTR_HEADING"
                  desc="Heading used when a person opens an OTR window">
           You’ve gone Incognito
