diff --git a/components/payments_strings.grdp b/components/payments_strings.grdp
index 1b05daf93bd3e810105bf19872addcb7cebba3a8..c6e0f91431d4352e7577bd8ee1b29286664baea0 100644
--- a/components/payments_strings.grdp
+++ b/components/payments_strings.grdp
@@ -400,7 +400,7 @@
   </if>
 
   <!-- Payment Request section preview strings. They are slightly different on desktop and android because the pluralization and formatting code behaves differently. -->
-  <if expr="not is_android">
+  <if expr="not is_android or is_android">
     <message name="IDS_PAYMENT_REQUEST_PAYMENT_METHODS_PREVIEW" desc="This is a snippet of a payment method a user has saved to Chrome, plus an indication of the number of additional payment methods the user has saved. Its function is to show the user has payment methods that can be used to complete a payment, and thus doesn't have to type the entire payment method. [ICU Syntax]">
     {PAYMENT_METHOD, plural,
        =0 {<ph name="PAYMENT_METHOD_PREVIEW">{1}<ex>VISA ....1234</ex></ph>}
