diff --git a/components/sync_ui_strings.grdp b/components/sync_ui_strings.grdp
index e9c414a4f5cfba06d09e330c56ef27123e872054..4e62f894686c8b1c77517937d3ad0a93111ac4fa 100644
--- a/components/sync_ui_strings.grdp
+++ b/components/sync_ui_strings.grdp
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="utf-8"?>
 <grit-part>
-  <if expr="not is_android">
+  <if expr="not is_android or is_android">
     <message name="IDS_SYNC_BASIC_ENCRYPTION_DATA" desc="Text of the radio that when selected enables basic encryption.">
       Encrypt synced passwords with your Google Account
     </message>
