diff --git a/components/sync_ui_strings.grdp b/components/sync_ui_strings.grdp
index 730b92cb903043873a5e229c4b6ef9272c0a4afe..95fb0d85c1579e7c40cf7997ebe5cfbadc9d97c5 100644
--- a/components/sync_ui_strings.grdp
+++ b/components/sync_ui_strings.grdp
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="utf-8"?>
 <grit-part>
-  <if expr="not is_android">
+  <if expr="not is_android or is_android">
     <message name="IDS_SYNC_BASIC_ENCRYPTION_DATA" desc="Text of the radio that when selected enables basic encryption.">
       Encrypt synced passwords with your Google Account
     </message>
