diff --git a/components/history_strings.grdp b/components/history_strings.grdp
index 2d9c3b3d94e869686be6c89b613ab50650fd947e..63138afef3822387328f0234a8100cafa1fe809f 100644
--- a/components/history_strings.grdp
+++ b/components/history_strings.grdp
@@ -63,7 +63,7 @@
       Clear Browsing Data...
     </message>
   </if>
-  <if expr="not is_android">
+  <if expr="not is_android or is_android">
     <message name="IDS_HISTORY_OTHER_SESSIONS_COLLAPSE_SESSION" desc="In the 'Other Sessions' menu on the history page, the label for the command to collapse (hide) the list of windows and tabs in a session.">
       Collapse list
     </message>
