diff --git a/components/components_strings.grd b/components/components_strings.grd
index 2751b112ece3d9ad60519c642af05b6db40d17a2..b57bcdc4d217411a18ef4362f202cd17348962ea 100644
--- a/components/components_strings.grd
+++ b/components/components_strings.grd
@@ -338,7 +338,7 @@
       <part file="version_ui_strings.grdp" />
       <part file="webapps_strings.grdp" />
 
-      <if expr="not is_ios">
+      <if expr="not is_ios or is_android">
         <part file="history_clusters_strings.grdp" />
       </if>
       <if expr="is_android">
@@ -434,7 +434,7 @@
       <message name="IDS_PLUGIN_NOT_SUPPORTED" desc="The placeholder text for an unsupported plugin.">
         This plugin is not supported
       </message>
-      <if expr="not is_android">
+      <if expr="not is_android or is_android">
         <if expr="not use_titlecase">
           <message name="IDS_PRINT" desc="The text label of the Print... menu item. Opens a dialog box to select print settings">
             &amp;Print...
