diff --git a/components/site_settings_strings.grdp b/components/site_settings_strings.grdp
index d2dd165579dd2988d52d65a2ff99a987a5e80f33..a9c7af6a4a3b6e43ef6b4f7ba9a1a9bde6c17e57 100644
--- a/components/site_settings_strings.grdp
+++ b/components/site_settings_strings.grdp
@@ -153,7 +153,7 @@
   </message>
 
   <!-- Site Settings desktop-only -->
-  <if expr="not is_android">
+  <if expr="not is_android or is_android">
   <message name="IDS_SITE_SETTINGS_TYPE_FILE_SYSTEM_ACCESS_WRITE" desc="The label used for the File System Access write site settings controls. The write permission determines whether sites are allowed to save to the original file that was selected by the user through the File System Access API.">
     File editing
   </message>
