diff --git a/components/commerce_strings.grdp b/components/commerce_strings.grdp
index 01169f911530d4b1b558d9ae57b2fdc2224dd781..381fda563b9a114848fa9311909976d1b216d3e1 100644
--- a/components/commerce_strings.grdp
+++ b/components/commerce_strings.grdp
@@ -1,7 +1,7 @@
 <?xml version="1.0" encoding="utf-8"?>
 <grit-part>
   <!-- Desktop only -->
-  <if expr="not is_android and not is_ios">
+  <if expr="not is_android and not is_ios or is_android">
     <!-- Discount consent strings -->
     <!-- TODO(crbug.com/1368330): Delete the following strings. -->
     <message name="IDS_DISCOUNT_CONTEXTUAL_CONSENT_TITLE" desc="The title shown on the consent bubble for getting discount.">
