diff --git a/components/page_info_strings.grdp b/components/page_info_strings.grdp
index c68942bf2e0671d84cc8bbc91935e1ca085dc283..2d70075e911ab4d2605bbfe8a31ac57999699994 100644
--- a/components/page_info_strings.grdp
+++ b/components/page_info_strings.grdp
@@ -201,7 +201,7 @@
     </message>
 
     <!-- Certificate Viewer link -->
-    <if expr="not is_android">
+    <if expr="not is_android or is_android">
       <message name="IDS_PAGE_INFO_CERTIFICATE" desc="Title of the certificate area in the Page Info bubble, shown when a HTTPS site is loaded.">
         Certificate
       </message>
@@ -243,7 +243,7 @@
     </if>
 
     <!-- Cookies -->
-    <if expr="not is_android">
+    <if expr="not is_android or is_android">
       <message name="IDS_PAGE_INFO_COOKIES" desc="The label for the Cookies setting in the Page Information Window.">
         Cookies
       </message>
@@ -501,7 +501,7 @@
     </message>
 
     <!-- Permission change infobar. -->
-    <if expr="not is_android">
+    <if expr="not is_android or is_android">
       <message name="IDS_PAGE_INFO_INFOBAR_TEXT" desc="The string shown in the infobar after the user has changed site permissions settings, reminding them to reload the page in order for the new settings to take effect.">
         To apply your updated settings to this site, reload this page
       </message>
