diff --git a/components/commerce_strings.grdp b/components/commerce_strings.grdp
index cc353a70955936b48956bb9296442c54352c1b1c..0b4f1c8e43ffb18170386d67ae8fb90ba5e63d71 100644
--- a/components/commerce_strings.grdp
+++ b/components/commerce_strings.grdp
@@ -1,7 +1,7 @@
 <?xml version="1.0" encoding="utf-8"?>
 <grit-part>
   <!-- Desktop only -->
-  <if expr="not is_android and not is_ios">
+  <if expr="not is_android and not is_ios or is_android">
     <!-- Discount consent strings -->
     <!-- TODO(crbug.com/1368330): Delete the following strings. -->
     <message name="IDS_DISCOUNT_CONTEXTUAL_CONSENT_TITLE" desc="The title shown on the consent bubble for getting discount.">
