diff --git a/components/permissions_strings.grdp b/components/permissions_strings.grdp
index e76fe12706e8d1eee578657843b59117c47cead0..3687c522631ffcc248766a97d6175b6de48f3a6c 100644
--- a/components/permissions_strings.grdp
+++ b/components/permissions_strings.grdp
@@ -169,7 +169,7 @@ This will otherwise be blocked by your privacy settings. This will allow the con
     Never allow
   </message>
   <!-- Device Chooser -->
-  <if expr="not is_android">
+  <if expr="not is_android or is_android">
     <message name="IDS_BLUETOOTH_DEVICE_CHOOSER_PROMPT" desc="The label that is used to introduce Bluetooth chooser details to the user in a popup.">
       <ph name="Origin">$1<ex>www.google.com</ex></ph> wants to pair
     </message>
@@ -191,7 +191,7 @@ This will otherwise be blocked by your privacy settings. This will allow the con
   </message>
 
   <!-- Bluetooth Scanning Prompt -->
-  <if expr="not is_android">
+  <if expr="not is_android or is_android">
     <message name="IDS_BLUETOOTH_SCANNING_PROMPT" desc="The label that is used to introduce Bluetooth scanning prompt details to the user.">
       <ph name="Origin">$1<ex>www.google.com</ex></ph> wants to scan for nearby Bluetooth devices. The following devices have been found:
     </message>
@@ -235,7 +235,7 @@ This will otherwise be blocked by your privacy settings. This will allow the con
     Connect
   </message>
 
-  <if expr="not is_android">
+  <if expr="not is_android or is_android">
     <message name="IDS_GEOLOCATION_PERMISSION_CHIP" desc="Button text representing a request for the user's physical location from a website. When clicked, shows a permission prompt bubble with more information.">
       Use your location?
     </message>
