diff --git a/components/commerce_strings.grdp b/components/commerce_strings.grdp
index 257c88d7a92b0..7b4894d401227 100644
--- a/components/commerce_strings.grdp
+++ b/components/commerce_strings.grdp
@@ -1,7 +1,7 @@
 <?xml version="1.0" encoding="utf-8"?>
 <grit-part>
   <!-- Desktop only -->
-  <if expr="not is_android and not is_ios">
+  <if expr="not is_android and not is_ios or is_android">
     <message name="IDS_DISCOUNT_CONTEXTUAL_CONSENT_TITLE" desc="The title shown on the consent bubble for getting discount.">
       Get discount on <ph name="MERCHANT_NAME">$1</ph> and more
     </message>
@@ -33,4 +33,4 @@
       Let Google find personalized discounts on your carts and on online stores you visit. When available, discounts will automatically show up on your carts and on sites.
     </message>
   </if>
-</grit-part>
\ No newline at end of file
+</grit-part>
