diff --git a/components/page_info_strings.grdp b/components/page_info_strings.grdp
index 3abb567a79c289ce6f95ed25e0181c052a87526f..96a85fdaf521c81c604f40500f780dd0a9e2821d 100644
--- a/components/page_info_strings.grdp
+++ b/components/page_info_strings.grdp
@@ -189,7 +189,7 @@
     </message>
 
     <!-- Certificate Viewer link -->
-    <if expr="not is_android">
+    <if expr="not is_android or is_android">
       <message name="IDS_PAGE_INFO_CERTIFICATE" desc="Title of the certificate area in the Page Info bubble, shown when a HTTPS site is loaded.">
         Certificate
       </message>
@@ -231,7 +231,7 @@
     </if>
 
     <!-- Cookies -->
-    <if expr="not is_android">
+    <if expr="not is_android or is_android">
       <message name="IDS_PAGE_INFO_COOKIES" desc="The label for the Cookies setting in the Page Information Window.">
         Cookies
       </message>
@@ -532,7 +532,7 @@
     </message>
 
     <!-- Permission change infobar. -->
-    <if expr="not is_android">
+    <if expr="not is_android or is_android">
       <message name="IDS_PAGE_INFO_INFOBAR_TEXT" desc="The string shown in the infobar after the user has changed site permissions settings, reminding them to reload the page in order for the new settings to take effect.  Deliberately similar to IDS_EXTENSION_SITE_RELOAD_PAGE_BUBBLE_HEADING, but used for an different permission setting so they are not combined. If updating the one, please update the other or consult the extensions and privacy teams.">
         Reload this page to apply your updated settings on this site
       </message>
