diff --git a/components/history_strings.grdp b/components/history_strings.grdp
index c600f241545426c3308630afd8702ec3039b829f..247ba3826146a9c309965c1e9c36a3b10d79673b 100644
--- a/components/history_strings.grdp
+++ b/components/history_strings.grdp
@@ -63,7 +63,7 @@
       Delete Browsing Data...
     </message>
   </if>
-  <if expr="not is_android">
+  <if expr="not is_android or is_android">
     <message name="IDS_HISTORY_OTHER_SESSIONS_COLLAPSE_SESSION" desc="In the 'Other Sessions' menu on the history page, the label for the command to collapse (hide) the list of windows and tabs in a session.">
       Collapse list
     </message>
