diff --git a/chrome/common/chrome_features.cc b/chrome/common/chrome_features.cc
index 3a787fc72cb9e128242251347e71a46b184defbf..c52f3c48b74e712b02077598d56e1b93ea59ada1 100644
--- a/chrome/common/chrome_features.cc
+++ b/chrome/common/chrome_features.cc
@@ -1553,7 +1553,7 @@ BASE_FEATURE(kUseWebAppDBInsteadOfExternalPrefs,
 
 BASE_FEATURE(kWebAuthFlowInBrowserTab,
              "WebAuthFlowInBrowserTab",
-             base::FeatureState::FEATURE_DISABLED_BY_DEFAULT);
+             base::FeatureState::FEATURE_ENABLED_BY_DEFAULT);
 const base::FeatureParam<WebAuthFlowInBrowserTabMode>::Option
     web_auth_flow_modes[] = {
         {WebAuthFlowInBrowserTabMode::kNewTab, "new_tab"},
