diff --git a/ui/gfx/BUILD.gn b/ui/gfx/BUILD.gn
index b9ea2dc1321018f9ef821fc0dfa5165b255f244d..f1899eced41f371bbd245422ff071c5161502e33 100644
--- a/ui/gfx/BUILD.gn
+++ b/ui/gfx/BUILD.gn
@@ -413,7 +413,7 @@ component("gfx") {
     ]
   }
 
-  if ((!use_aura && !toolkit_views) || is_ios) {
+  if ((!use_aura && !toolkit_views && !is_android) || is_ios) {
     sources -= [
       "nine_image_painter.cc",
       "nine_image_painter.h",
